// @react-spectrum/textfield/dist/main.js
require("./main.css");
var $fMXlK$reactariautils = require("@react-aria/utils");
var $fMXlK$react = require("react");
var $fMXlK$reactstatelyutils = require("@react-stately/utils");
var $fMXlK$reactspectrumprovider = require("@react-spectrum/provider");
var $fMXlK$reactariatextfield = require("@react-aria/textfield");
var $fMXlK$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $fMXlK$spectrumiconsuiCheckmarkMedium = require("@spectrum-icons/ui/CheckmarkMedium");
var $fMXlK$reactspectrumutils = require("@react-spectrum/utils");
var $fMXlK$reactspectrumlabel = require("@react-spectrum/label");
var $fMXlK$reactariafocus = require("@react-aria/focus");
var $fMXlK$reactariainteractions = require("@react-aria/interactions");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $0771a1439a157af4$exports = {};

$parcel$export($0771a1439a157af4$exports, "TextArea", () => $0771a1439a157af4$export$f5c9f3c2c4054eec);


var $f728b3265180f1f9$exports = {};

$parcel$export($f728b3265180f1f9$exports, "TextFieldBase", () => $f728b3265180f1f9$export$d22444a338b6e3c2);







var $484f319272acc7c5$exports = {};

$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield-input", () => $484f319272acc7c5$export$1ac009feb2b0bd0c, (v) => $484f319272acc7c5$export$1ac009feb2b0bd0c = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield", () => $484f319272acc7c5$export$2c8a10299d8b3418, (v) => $484f319272acc7c5$export$2c8a10299d8b3418 = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield--quiet", () => $484f319272acc7c5$export$bffd5e3d61a81737, (v) => $484f319272acc7c5$export$bffd5e3d61a81737 = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield--multiline", () => $484f319272acc7c5$export$9699f1c538a448a0, (v) => $484f319272acc7c5$export$9699f1c538a448a0 = v);
$parcel$export($484f319272acc7c5$exports, "is-placeholder", () => $484f319272acc7c5$export$e5b2f5233e4e5194, (v) => $484f319272acc7c5$export$e5b2f5233e4e5194 = v);
$parcel$export($484f319272acc7c5$exports, "is-hovered", () => $484f319272acc7c5$export$b8813cd5d7824ce7, (v) => $484f319272acc7c5$export$b8813cd5d7824ce7 = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield--valid", () => $484f319272acc7c5$export$c2abbc3f970170b7, (v) => $484f319272acc7c5$export$c2abbc3f970170b7 = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield--invalid", () => $484f319272acc7c5$export$58e50b10d30123df, (v) => $484f319272acc7c5$export$58e50b10d30123df = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield--loadable", () => $484f319272acc7c5$export$14fd30fea6f1348c, (v) => $484f319272acc7c5$export$14fd30fea6f1348c = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield-validationIcon", () => $484f319272acc7c5$export$cf0ead5b44db0da3, (v) => $484f319272acc7c5$export$cf0ead5b44db0da3 = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield-icon", () => $484f319272acc7c5$export$222d1ccd1870be1d, (v) => $484f319272acc7c5$export$222d1ccd1870be1d = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield-inputIcon", () => $484f319272acc7c5$export$c1cdcaa5fe76a871, (v) => $484f319272acc7c5$export$c1cdcaa5fe76a871 = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield-circleLoader", () => $484f319272acc7c5$export$82a249fb4d6127, (v) => $484f319272acc7c5$export$82a249fb4d6127 = v);
$parcel$export($484f319272acc7c5$exports, "is-focused", () => $484f319272acc7c5$export$e7dc768d35940237, (v) => $484f319272acc7c5$export$e7dc768d35940237 = v);
$parcel$export($484f319272acc7c5$exports, "focus-ring", () => $484f319272acc7c5$export$f39a09f249340e2a, (v) => $484f319272acc7c5$export$f39a09f249340e2a = v);
$parcel$export($484f319272acc7c5$exports, "is-disabled", () => $484f319272acc7c5$export$d35bc1e505d1ebbf, (v) => $484f319272acc7c5$export$d35bc1e505d1ebbf = v);
var $484f319272acc7c5$export$1ac009feb2b0bd0c;
var $484f319272acc7c5$export$2c8a10299d8b3418;
var $484f319272acc7c5$export$bffd5e3d61a81737;
var $484f319272acc7c5$export$9699f1c538a448a0;
var $484f319272acc7c5$export$e5b2f5233e4e5194;
var $484f319272acc7c5$export$b8813cd5d7824ce7;
var $484f319272acc7c5$export$c2abbc3f970170b7;
var $484f319272acc7c5$export$58e50b10d30123df;
var $484f319272acc7c5$export$14fd30fea6f1348c;
var $484f319272acc7c5$export$cf0ead5b44db0da3;
var $484f319272acc7c5$export$222d1ccd1870be1d;
var $484f319272acc7c5$export$c1cdcaa5fe76a871;
var $484f319272acc7c5$export$82a249fb4d6127;
var $484f319272acc7c5$export$e7dc768d35940237;
var $484f319272acc7c5$export$f39a09f249340e2a;
var $484f319272acc7c5$export$d35bc1e505d1ebbf;
$484f319272acc7c5$export$1ac009feb2b0bd0c = "spectrum-Textfield-input_73bc77";
$484f319272acc7c5$export$2c8a10299d8b3418 = "spectrum-Textfield_73bc77";
$484f319272acc7c5$export$bffd5e3d61a81737 = "spectrum-Textfield--quiet_73bc77";
$484f319272acc7c5$export$9699f1c538a448a0 = "spectrum-Textfield--multiline_73bc77";
$484f319272acc7c5$export$e5b2f5233e4e5194 = "is-placeholder_73bc77";
$484f319272acc7c5$export$b8813cd5d7824ce7 = "is-hovered_73bc77";
$484f319272acc7c5$export$c2abbc3f970170b7 = "spectrum-Textfield--valid_73bc77";
$484f319272acc7c5$export$58e50b10d30123df = "spectrum-Textfield--invalid_73bc77";
$484f319272acc7c5$export$14fd30fea6f1348c = "spectrum-Textfield--loadable_73bc77";
$484f319272acc7c5$export$cf0ead5b44db0da3 = "spectrum-Textfield-validationIcon_73bc77";
$484f319272acc7c5$export$222d1ccd1870be1d = "spectrum-Textfield-icon_73bc77";
$484f319272acc7c5$export$c1cdcaa5fe76a871 = "spectrum-Textfield-inputIcon_73bc77";
$484f319272acc7c5$export$82a249fb4d6127 = "spectrum-Textfield-circleLoader_73bc77";
$484f319272acc7c5$export$e7dc768d35940237 = "is-focused_73bc77";
$484f319272acc7c5$export$f39a09f249340e2a = "focus-ring_73bc77";
$484f319272acc7c5$export$d35bc1e505d1ebbf = "is-disabled_73bc77";



function $f728b3265180f1f9$var$TextFieldBase(props, ref) {
    let { label: label , validationState: validationState , icon: icon , isQuiet: isQuiet = false , isDisabled: isDisabled , multiLine: multiLine , autoFocus: autoFocus , inputClassName: inputClassName , wrapperChildren: wrapperChildren , labelProps: labelProps , inputProps: inputProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps , inputRef: inputRef , isLoading: isLoading , loadingIndicator: loadingIndicator , validationIconClassName: validationIconClassName  } = props;
    let { hoverProps: hoverProps , isHovered: isHovered  } = $fMXlK$reactariainteractions.useHover({
        isDisabled: isDisabled
    });
    let domRef = $fMXlK$react.useRef(null);
    let defaultInputRef = $fMXlK$react.useRef(null);
    inputRef = inputRef || defaultInputRef;
    // Expose imperative interface for ref
    $fMXlK$react.useImperativeHandle(ref, ()=>({
            ...$fMXlK$reactspectrumutils.createFocusableRef(domRef, inputRef),
            select () {
                if (inputRef.current) inputRef.current.select();
            },
            getInputElement () {
                return inputRef.current;
            }
        })
    );
    let ElementType = multiLine ? 'textarea' : 'input';
    let isInvalid = validationState === 'invalid';
    if (icon) {
        let UNSAFE_className = $fMXlK$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($484f319272acc7c5$exports)), icon.props && icon.props.UNSAFE_className, 'spectrum-Textfield-icon');
        icon = /*#__PURE__*/ $fMXlK$react.cloneElement(icon, {
            UNSAFE_className: UNSAFE_className,
            size: 'S'
        });
    }
    let validationIcon = isInvalid ? /*#__PURE__*/ ($parcel$interopDefault($fMXlK$react)).createElement(($parcel$interopDefault($fMXlK$spectrumiconsuiAlertMedium)), null) : /*#__PURE__*/ ($parcel$interopDefault($fMXlK$react)).createElement(($parcel$interopDefault($fMXlK$spectrumiconsuiCheckmarkMedium)), null);
    let validation = /*#__PURE__*/ $fMXlK$react.cloneElement(validationIcon, {
        UNSAFE_className: $fMXlK$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($484f319272acc7c5$exports)), 'spectrum-Textfield-validationIcon', validationIconClassName)
    });
    let textField = /*#__PURE__*/ ($parcel$interopDefault($fMXlK$react)).createElement("div", {
        className: $fMXlK$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($484f319272acc7c5$exports)), 'spectrum-Textfield', {
            'spectrum-Textfield--invalid': isInvalid,
            'spectrum-Textfield--valid': validationState === 'valid',
            'spectrum-Textfield--loadable': loadingIndicator,
            'spectrum-Textfield--quiet': isQuiet,
            'spectrum-Textfield--multiline': multiLine
        })
    }, /*#__PURE__*/ ($parcel$interopDefault($fMXlK$react)).createElement($fMXlK$reactariafocus.FocusRing, {
        focusRingClass: $fMXlK$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($484f319272acc7c5$exports)), 'focus-ring'),
        isTextInput: true,
        autoFocus: autoFocus
    }, /*#__PURE__*/ ($parcel$interopDefault($fMXlK$react)).createElement(ElementType, {
        ...$fMXlK$reactariautils.mergeProps(inputProps, hoverProps),
        ref: inputRef,
        rows: multiLine ? 1 : undefined,
        className: $fMXlK$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($484f319272acc7c5$exports)), 'spectrum-Textfield-input', {
            'spectrum-Textfield-inputIcon': icon,
            'is-hovered': isHovered
        }, inputClassName)
    })), icon, validationState && !isLoading ? validation : null, isLoading && loadingIndicator, wrapperChildren);
    if (label) textField = /*#__PURE__*/ ($parcel$interopDefault($fMXlK$react)).cloneElement(textField, $fMXlK$reactariautils.mergeProps(textField.props, {
        className: multiLine ? 'spectrum-Field-field--multiline' : ''
    }));
    return(/*#__PURE__*/ ($parcel$interopDefault($fMXlK$react)).createElement($fMXlK$reactspectrumlabel.Field, {
        ...props,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        showErrorIcon: false,
        ref: domRef
    }, textField));
}
const $f728b3265180f1f9$export$d22444a338b6e3c2 = /*#__PURE__*/ $fMXlK$react.forwardRef($f728b3265180f1f9$var$TextFieldBase);





function $0771a1439a157af4$var$TextArea(props, ref) {
    props = $fMXlK$reactspectrumprovider.useProviderProps(props);
    let { isDisabled: isDisabled = false , isQuiet: isQuiet = false , isReadOnly: isReadOnly = false , isRequired: isRequired = false , onChange: onChange , ...otherProps } = props;
    // not in stately because this is so we know when to re-measure, which is a spectrum design
    let [inputValue, setInputValue] = $fMXlK$reactstatelyutils.useControlledState(props.value, props.defaultValue, ()=>{
    });
    let inputRef = $fMXlK$react.useRef();
    let onHeightChange = $fMXlK$react.useCallback(()=>{
        if (isQuiet) {
            let input = inputRef.current;
            let prevAlignment = input.style.alignSelf;
            input.style.alignSelf = 'start';
            input.style.height = 'auto';
            input.style.height = `${input.scrollHeight}px`;
            input.style.alignSelf = prevAlignment;
        }
    }, [
        isQuiet,
        inputRef
    ]);
    $fMXlK$reactariautils.useLayoutEffect(()=>{
        if (inputRef.current) onHeightChange();
    }, [
        onHeightChange,
        inputValue,
        inputRef
    ]);
    let { labelProps: labelProps , inputProps: inputProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $fMXlK$reactariatextfield.useTextField({
        ...props,
        onChange: $fMXlK$reactariautils.chain(onChange, setInputValue),
        inputElementType: 'textarea'
    }, inputRef);
    return(/*#__PURE__*/ ($parcel$interopDefault($fMXlK$react)).createElement($f728b3265180f1f9$export$d22444a338b6e3c2, {
        ...otherProps,
        ref: ref,
        inputRef: inputRef,
        labelProps: labelProps,
        inputProps: inputProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        multiLine: true,
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        isReadOnly: isReadOnly,
        isRequired: isRequired
    }));
}
/**
 * TextAreas are multiline text inputs, useful for cases where users have
 * a sizable amount of text to enter. They allow for all customizations that
 * are available to text fields.
 */ let $0771a1439a157af4$export$f5c9f3c2c4054eec = /*#__PURE__*/ ($parcel$interopDefault($fMXlK$react)).forwardRef($0771a1439a157af4$var$TextArea);


var $27469905ba6c02ac$exports = {};

$parcel$export($27469905ba6c02ac$exports, "TextField", () => $27469905ba6c02ac$export$2c73285ae9390cec);




function $27469905ba6c02ac$var$TextField(props, ref) {
    props = $fMXlK$reactspectrumprovider.useProviderProps(props);
    let inputRef = $fMXlK$react.useRef();
    let { labelProps: labelProps , inputProps: inputProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $fMXlK$reactariatextfield.useTextField(props, inputRef);
    return(/*#__PURE__*/ ($parcel$interopDefault($fMXlK$react)).createElement($f728b3265180f1f9$export$d22444a338b6e3c2, {
        ...props,
        labelProps: labelProps,
        inputProps: inputProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        ref: ref,
        inputRef: inputRef
    }));
}
/**
 * TextFields are text inputs that allow users to input custom text entries
 * with a keyboard. Various decorations can be displayed around the field to
 * communicate the entry requirements.
 */ const $27469905ba6c02ac$export$2c73285ae9390cec = /*#__PURE__*/ $fMXlK$react.forwardRef($27469905ba6c02ac$var$TextField);



$parcel$exportWildcard(module.exports, $0771a1439a157af4$exports);
$parcel$exportWildcard(module.exports, $27469905ba6c02ac$exports);
$parcel$exportWildcard(module.exports, $f728b3265180f1f9$exports);
