// @react-spectrum/textfield/dist/module.js
import "./main.css";
import {useLayoutEffect as $bVKqV$useLayoutEffect, chain as $bVKqV$chain, mergeProps as $bVKqV$mergeProps} from "@react-aria/utils";
import $bVKqV$react, {useRef as $bVKqV$useRef, useCallback as $bVKqV$useCallback, useImperativeHandle as $bVKqV$useImperativeHandle, cloneElement as $bVKqV$cloneElement, forwardRef as $bVKqV$forwardRef} from "react";
import {useControlledState as $bVKqV$useControlledState} from "@react-stately/utils";
import {useProviderProps as $bVKqV$useProviderProps} from "@react-spectrum/provider";
import {useTextField as $bVKqV$useTextField} from "@react-aria/textfield";
import $bVKqV$spectrumiconsuiAlertMedium from "@spectrum-icons/ui/AlertMedium";
import $bVKqV$spectrumiconsuiCheckmarkMedium from "@spectrum-icons/ui/CheckmarkMedium";
import {createFocusableRef as $bVKqV$createFocusableRef, classNames as $bVKqV$classNames} from "@react-spectrum/utils";
import {Field as $bVKqV$Field} from "@react-spectrum/label";
import {FocusRing as $bVKqV$FocusRing} from "@react-aria/focus";
import {useHover as $bVKqV$useHover} from "@react-aria/interactions";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
var $de5a8a929fce58e7$exports = {};

$parcel$export($de5a8a929fce58e7$exports, "TextArea", () => $de5a8a929fce58e7$export$f5c9f3c2c4054eec);


var $83651c7a03742561$exports = {};

$parcel$export($83651c7a03742561$exports, "TextFieldBase", () => $83651c7a03742561$export$d22444a338b6e3c2);







var $302bd214da97cdcb$exports = {};

$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield-input", () => $302bd214da97cdcb$export$1ac009feb2b0bd0c, (v) => $302bd214da97cdcb$export$1ac009feb2b0bd0c = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield", () => $302bd214da97cdcb$export$2c8a10299d8b3418, (v) => $302bd214da97cdcb$export$2c8a10299d8b3418 = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield--quiet", () => $302bd214da97cdcb$export$bffd5e3d61a81737, (v) => $302bd214da97cdcb$export$bffd5e3d61a81737 = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield--multiline", () => $302bd214da97cdcb$export$9699f1c538a448a0, (v) => $302bd214da97cdcb$export$9699f1c538a448a0 = v);
$parcel$export($302bd214da97cdcb$exports, "is-placeholder", () => $302bd214da97cdcb$export$e5b2f5233e4e5194, (v) => $302bd214da97cdcb$export$e5b2f5233e4e5194 = v);
$parcel$export($302bd214da97cdcb$exports, "is-hovered", () => $302bd214da97cdcb$export$b8813cd5d7824ce7, (v) => $302bd214da97cdcb$export$b8813cd5d7824ce7 = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield--valid", () => $302bd214da97cdcb$export$c2abbc3f970170b7, (v) => $302bd214da97cdcb$export$c2abbc3f970170b7 = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield--invalid", () => $302bd214da97cdcb$export$58e50b10d30123df, (v) => $302bd214da97cdcb$export$58e50b10d30123df = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield--loadable", () => $302bd214da97cdcb$export$14fd30fea6f1348c, (v) => $302bd214da97cdcb$export$14fd30fea6f1348c = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield-validationIcon", () => $302bd214da97cdcb$export$cf0ead5b44db0da3, (v) => $302bd214da97cdcb$export$cf0ead5b44db0da3 = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield-icon", () => $302bd214da97cdcb$export$222d1ccd1870be1d, (v) => $302bd214da97cdcb$export$222d1ccd1870be1d = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield-inputIcon", () => $302bd214da97cdcb$export$c1cdcaa5fe76a871, (v) => $302bd214da97cdcb$export$c1cdcaa5fe76a871 = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield-circleLoader", () => $302bd214da97cdcb$export$82a249fb4d6127, (v) => $302bd214da97cdcb$export$82a249fb4d6127 = v);
$parcel$export($302bd214da97cdcb$exports, "is-focused", () => $302bd214da97cdcb$export$e7dc768d35940237, (v) => $302bd214da97cdcb$export$e7dc768d35940237 = v);
$parcel$export($302bd214da97cdcb$exports, "focus-ring", () => $302bd214da97cdcb$export$f39a09f249340e2a, (v) => $302bd214da97cdcb$export$f39a09f249340e2a = v);
$parcel$export($302bd214da97cdcb$exports, "is-disabled", () => $302bd214da97cdcb$export$d35bc1e505d1ebbf, (v) => $302bd214da97cdcb$export$d35bc1e505d1ebbf = v);
var $302bd214da97cdcb$export$1ac009feb2b0bd0c;
var $302bd214da97cdcb$export$2c8a10299d8b3418;
var $302bd214da97cdcb$export$bffd5e3d61a81737;
var $302bd214da97cdcb$export$9699f1c538a448a0;
var $302bd214da97cdcb$export$e5b2f5233e4e5194;
var $302bd214da97cdcb$export$b8813cd5d7824ce7;
var $302bd214da97cdcb$export$c2abbc3f970170b7;
var $302bd214da97cdcb$export$58e50b10d30123df;
var $302bd214da97cdcb$export$14fd30fea6f1348c;
var $302bd214da97cdcb$export$cf0ead5b44db0da3;
var $302bd214da97cdcb$export$222d1ccd1870be1d;
var $302bd214da97cdcb$export$c1cdcaa5fe76a871;
var $302bd214da97cdcb$export$82a249fb4d6127;
var $302bd214da97cdcb$export$e7dc768d35940237;
var $302bd214da97cdcb$export$f39a09f249340e2a;
var $302bd214da97cdcb$export$d35bc1e505d1ebbf;
$302bd214da97cdcb$export$1ac009feb2b0bd0c = "spectrum-Textfield-input_73bc77";
$302bd214da97cdcb$export$2c8a10299d8b3418 = "spectrum-Textfield_73bc77";
$302bd214da97cdcb$export$bffd5e3d61a81737 = "spectrum-Textfield--quiet_73bc77";
$302bd214da97cdcb$export$9699f1c538a448a0 = "spectrum-Textfield--multiline_73bc77";
$302bd214da97cdcb$export$e5b2f5233e4e5194 = "is-placeholder_73bc77";
$302bd214da97cdcb$export$b8813cd5d7824ce7 = "is-hovered_73bc77";
$302bd214da97cdcb$export$c2abbc3f970170b7 = "spectrum-Textfield--valid_73bc77";
$302bd214da97cdcb$export$58e50b10d30123df = "spectrum-Textfield--invalid_73bc77";
$302bd214da97cdcb$export$14fd30fea6f1348c = "spectrum-Textfield--loadable_73bc77";
$302bd214da97cdcb$export$cf0ead5b44db0da3 = "spectrum-Textfield-validationIcon_73bc77";
$302bd214da97cdcb$export$222d1ccd1870be1d = "spectrum-Textfield-icon_73bc77";
$302bd214da97cdcb$export$c1cdcaa5fe76a871 = "spectrum-Textfield-inputIcon_73bc77";
$302bd214da97cdcb$export$82a249fb4d6127 = "spectrum-Textfield-circleLoader_73bc77";
$302bd214da97cdcb$export$e7dc768d35940237 = "is-focused_73bc77";
$302bd214da97cdcb$export$f39a09f249340e2a = "focus-ring_73bc77";
$302bd214da97cdcb$export$d35bc1e505d1ebbf = "is-disabled_73bc77";



function $83651c7a03742561$var$TextFieldBase(props, ref) {
    let { label: label , validationState: validationState , icon: icon , isQuiet: isQuiet = false , isDisabled: isDisabled , multiLine: multiLine , autoFocus: autoFocus , inputClassName: inputClassName , wrapperChildren: wrapperChildren , labelProps: labelProps , inputProps: inputProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps , inputRef: inputRef , isLoading: isLoading , loadingIndicator: loadingIndicator , validationIconClassName: validationIconClassName  } = props;
    let { hoverProps: hoverProps , isHovered: isHovered  } = $bVKqV$useHover({
        isDisabled: isDisabled
    });
    let domRef = $bVKqV$useRef(null);
    let defaultInputRef = $bVKqV$useRef(null);
    inputRef = inputRef || defaultInputRef;
    // Expose imperative interface for ref
    $bVKqV$useImperativeHandle(ref, ()=>({
            ...$bVKqV$createFocusableRef(domRef, inputRef),
            select () {
                if (inputRef.current) inputRef.current.select();
            },
            getInputElement () {
                return inputRef.current;
            }
        })
    );
    let ElementType = multiLine ? 'textarea' : 'input';
    let isInvalid = validationState === 'invalid';
    if (icon) {
        let UNSAFE_className = $bVKqV$classNames((/*@__PURE__*/$parcel$interopDefault($302bd214da97cdcb$exports)), icon.props && icon.props.UNSAFE_className, 'spectrum-Textfield-icon');
        icon = /*#__PURE__*/ $bVKqV$cloneElement(icon, {
            UNSAFE_className: UNSAFE_className,
            size: 'S'
        });
    }
    let validationIcon = isInvalid ? /*#__PURE__*/ $bVKqV$react.createElement($bVKqV$spectrumiconsuiAlertMedium, null) : /*#__PURE__*/ $bVKqV$react.createElement($bVKqV$spectrumiconsuiCheckmarkMedium, null);
    let validation = /*#__PURE__*/ $bVKqV$cloneElement(validationIcon, {
        UNSAFE_className: $bVKqV$classNames((/*@__PURE__*/$parcel$interopDefault($302bd214da97cdcb$exports)), 'spectrum-Textfield-validationIcon', validationIconClassName)
    });
    let textField = /*#__PURE__*/ $bVKqV$react.createElement("div", {
        className: $bVKqV$classNames((/*@__PURE__*/$parcel$interopDefault($302bd214da97cdcb$exports)), 'spectrum-Textfield', {
            'spectrum-Textfield--invalid': isInvalid,
            'spectrum-Textfield--valid': validationState === 'valid',
            'spectrum-Textfield--loadable': loadingIndicator,
            'spectrum-Textfield--quiet': isQuiet,
            'spectrum-Textfield--multiline': multiLine
        })
    }, /*#__PURE__*/ $bVKqV$react.createElement($bVKqV$FocusRing, {
        focusRingClass: $bVKqV$classNames((/*@__PURE__*/$parcel$interopDefault($302bd214da97cdcb$exports)), 'focus-ring'),
        isTextInput: true,
        autoFocus: autoFocus
    }, /*#__PURE__*/ $bVKqV$react.createElement(ElementType, {
        ...$bVKqV$mergeProps(inputProps, hoverProps),
        ref: inputRef,
        rows: multiLine ? 1 : undefined,
        className: $bVKqV$classNames((/*@__PURE__*/$parcel$interopDefault($302bd214da97cdcb$exports)), 'spectrum-Textfield-input', {
            'spectrum-Textfield-inputIcon': icon,
            'is-hovered': isHovered
        }, inputClassName)
    })), icon, validationState && !isLoading ? validation : null, isLoading && loadingIndicator, wrapperChildren);
    if (label) textField = /*#__PURE__*/ $bVKqV$react.cloneElement(textField, $bVKqV$mergeProps(textField.props, {
        className: multiLine ? 'spectrum-Field-field--multiline' : ''
    }));
    return(/*#__PURE__*/ $bVKqV$react.createElement($bVKqV$Field, {
        ...props,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        showErrorIcon: false,
        ref: domRef
    }, textField));
}
const $83651c7a03742561$export$d22444a338b6e3c2 = /*#__PURE__*/ $bVKqV$forwardRef($83651c7a03742561$var$TextFieldBase);





function $de5a8a929fce58e7$var$TextArea(props, ref) {
    props = $bVKqV$useProviderProps(props);
    let { isDisabled: isDisabled = false , isQuiet: isQuiet = false , isReadOnly: isReadOnly = false , isRequired: isRequired = false , onChange: onChange , ...otherProps } = props;
    // not in stately because this is so we know when to re-measure, which is a spectrum design
    let [inputValue, setInputValue] = $bVKqV$useControlledState(props.value, props.defaultValue, ()=>{
    });
    let inputRef = $bVKqV$useRef();
    let onHeightChange = $bVKqV$useCallback(()=>{
        if (isQuiet) {
            let input = inputRef.current;
            let prevAlignment = input.style.alignSelf;
            input.style.alignSelf = 'start';
            input.style.height = 'auto';
            input.style.height = `${input.scrollHeight}px`;
            input.style.alignSelf = prevAlignment;
        }
    }, [
        isQuiet,
        inputRef
    ]);
    $bVKqV$useLayoutEffect(()=>{
        if (inputRef.current) onHeightChange();
    }, [
        onHeightChange,
        inputValue,
        inputRef
    ]);
    let { labelProps: labelProps , inputProps: inputProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $bVKqV$useTextField({
        ...props,
        onChange: $bVKqV$chain(onChange, setInputValue),
        inputElementType: 'textarea'
    }, inputRef);
    return(/*#__PURE__*/ $bVKqV$react.createElement($83651c7a03742561$export$d22444a338b6e3c2, {
        ...otherProps,
        ref: ref,
        inputRef: inputRef,
        labelProps: labelProps,
        inputProps: inputProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        multiLine: true,
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        isReadOnly: isReadOnly,
        isRequired: isRequired
    }));
}
/**
 * TextAreas are multiline text inputs, useful for cases where users have
 * a sizable amount of text to enter. They allow for all customizations that
 * are available to text fields.
 */ let $de5a8a929fce58e7$export$f5c9f3c2c4054eec = /*#__PURE__*/ $bVKqV$react.forwardRef($de5a8a929fce58e7$var$TextArea);


var $2d97b2be3a1f9c23$exports = {};

$parcel$export($2d97b2be3a1f9c23$exports, "TextField", () => $2d97b2be3a1f9c23$export$2c73285ae9390cec);




function $2d97b2be3a1f9c23$var$TextField(props, ref) {
    props = $bVKqV$useProviderProps(props);
    let inputRef = $bVKqV$useRef();
    let { labelProps: labelProps , inputProps: inputProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $bVKqV$useTextField(props, inputRef);
    return(/*#__PURE__*/ $bVKqV$react.createElement($83651c7a03742561$export$d22444a338b6e3c2, {
        ...props,
        labelProps: labelProps,
        inputProps: inputProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        ref: ref,
        inputRef: inputRef
    }));
}
/**
 * TextFields are text inputs that allow users to input custom text entries
 * with a keyboard. Various decorations can be displayed around the field to
 * communicate the entry requirements.
 */ const $2d97b2be3a1f9c23$export$2c73285ae9390cec = /*#__PURE__*/ $bVKqV$forwardRef($2d97b2be3a1f9c23$var$TextField);





export {$de5a8a929fce58e7$export$f5c9f3c2c4054eec as TextArea, $2d97b2be3a1f9c23$export$2c73285ae9390cec as TextField, $83651c7a03742561$export$d22444a338b6e3c2 as TextFieldBase};
