// @react-spectrum/form/dist/main.js
require("./main.css");
var $etHtu$reactspectrumutils = require("@react-spectrum/utils");
var $etHtu$reactariautils = require("@react-aria/utils");
var $etHtu$reactspectrumprovider = require("@react-spectrum/provider");
var $etHtu$react = require("react");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $8f20d6b8c0818d0b$exports = {};

$parcel$export($8f20d6b8c0818d0b$exports, "useFormProps", () => $8f20d6b8c0818d0b$export$a6b5be5c6b451665);
$parcel$export($8f20d6b8c0818d0b$exports, "Form", () => $8f20d6b8c0818d0b$export$a7fed597f4b8afd8);




var $0043d8d062ef413c$exports = {};

$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel", () => $0043d8d062ef413c$export$493d18e796ae054, (v) => $0043d8d062ef413c$export$493d18e796ae054 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel--positionSide", () => $0043d8d062ef413c$export$9e6f19dc21f22f2e, (v) => $0043d8d062ef413c$export$9e6f19dc21f22f2e = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel-requiredIcon", () => $0043d8d062ef413c$export$7c47642c7d46f3c9, (v) => $0043d8d062ef413c$export$7c47642c7d46f3c9 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel--alignEnd", () => $0043d8d062ef413c$export$885efcc08143a987, (v) => $0043d8d062ef413c$export$885efcc08143a987 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field", () => $0043d8d062ef413c$export$f6d480ae1e56eba0, (v) => $0043d8d062ef413c$export$f6d480ae1e56eba0 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field--positionTop", () => $0043d8d062ef413c$export$a4ea780a9064d7f9, (v) => $0043d8d062ef413c$export$a4ea780a9064d7f9 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field-field", () => $0043d8d062ef413c$export$3ab8a3dc7f0563d, (v) => $0043d8d062ef413c$export$3ab8a3dc7f0563d = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field-field--multiline", () => $0043d8d062ef413c$export$f58a1e966a92ba5e, (v) => $0043d8d062ef413c$export$f58a1e966a92ba5e = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field--positionSide", () => $0043d8d062ef413c$export$2963225c91179589, (v) => $0043d8d062ef413c$export$2963225c91179589 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field-wrapper", () => $0043d8d062ef413c$export$127df0b7290eb8ef, (v) => $0043d8d062ef413c$export$127df0b7290eb8ef = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form", () => $0043d8d062ef413c$export$94d2bcc94afabd89, (v) => $0043d8d062ef413c$export$94d2bcc94afabd89 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form--positionSide", () => $0043d8d062ef413c$export$7f44db659563c8f4, (v) => $0043d8d062ef413c$export$7f44db659563c8f4 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form--positionTop", () => $0043d8d062ef413c$export$11b8bc81fe551057, (v) => $0043d8d062ef413c$export$11b8bc81fe551057 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form-itemLabel", () => $0043d8d062ef413c$export$51ec27881a1e3928, (v) => $0043d8d062ef413c$export$51ec27881a1e3928 = v);
$parcel$export($0043d8d062ef413c$exports, "is-disabled", () => $0043d8d062ef413c$export$d35bc1e505d1ebbf, (v) => $0043d8d062ef413c$export$d35bc1e505d1ebbf = v);
var $0043d8d062ef413c$export$493d18e796ae054;
var $0043d8d062ef413c$export$9e6f19dc21f22f2e;
var $0043d8d062ef413c$export$7c47642c7d46f3c9;
var $0043d8d062ef413c$export$885efcc08143a987;
var $0043d8d062ef413c$export$f6d480ae1e56eba0;
var $0043d8d062ef413c$export$a4ea780a9064d7f9;
var $0043d8d062ef413c$export$3ab8a3dc7f0563d;
var $0043d8d062ef413c$export$f58a1e966a92ba5e;
var $0043d8d062ef413c$export$2963225c91179589;
var $0043d8d062ef413c$export$127df0b7290eb8ef;
var $0043d8d062ef413c$export$94d2bcc94afabd89;
var $0043d8d062ef413c$export$7f44db659563c8f4;
var $0043d8d062ef413c$export$11b8bc81fe551057;
var $0043d8d062ef413c$export$51ec27881a1e3928;
var $0043d8d062ef413c$export$d35bc1e505d1ebbf;
$0043d8d062ef413c$export$493d18e796ae054 = "spectrum-FieldLabel_d2db1f";
$0043d8d062ef413c$export$9e6f19dc21f22f2e = "spectrum-FieldLabel--positionSide_d2db1f";
$0043d8d062ef413c$export$7c47642c7d46f3c9 = "spectrum-FieldLabel-requiredIcon_d2db1f";
$0043d8d062ef413c$export$885efcc08143a987 = "spectrum-FieldLabel--alignEnd_d2db1f";
$0043d8d062ef413c$export$f6d480ae1e56eba0 = "spectrum-Field_d2db1f";
$0043d8d062ef413c$export$a4ea780a9064d7f9 = "spectrum-Field--positionTop_d2db1f";
$0043d8d062ef413c$export$3ab8a3dc7f0563d = "spectrum-Field-field_d2db1f";
$0043d8d062ef413c$export$f58a1e966a92ba5e = "spectrum-Field-field--multiline_d2db1f";
$0043d8d062ef413c$export$2963225c91179589 = "spectrum-Field--positionSide_d2db1f";
$0043d8d062ef413c$export$127df0b7290eb8ef = "spectrum-Field-wrapper_d2db1f";
$0043d8d062ef413c$export$94d2bcc94afabd89 = "spectrum-Form_d2db1f";
$0043d8d062ef413c$export$7f44db659563c8f4 = "spectrum-Form--positionSide_d2db1f";
$0043d8d062ef413c$export$11b8bc81fe551057 = "spectrum-Form--positionTop_d2db1f";
$0043d8d062ef413c$export$51ec27881a1e3928 = "spectrum-Form-itemLabel_d2db1f";
$0043d8d062ef413c$export$d35bc1e505d1ebbf = "is-disabled_d2db1f";


let $8f20d6b8c0818d0b$var$FormContext = /*#__PURE__*/ ($parcel$interopDefault($etHtu$react)).createContext({
});
function $8f20d6b8c0818d0b$export$a6b5be5c6b451665(props) {
    let ctx = $etHtu$react.useContext($8f20d6b8c0818d0b$var$FormContext);
    return {
        ...ctx,
        ...props
    };
}
const $8f20d6b8c0818d0b$var$formPropNames = new Set([
    'action',
    'autoComplete',
    'encType',
    'method',
    'target',
    'onSubmit'
]);
function $8f20d6b8c0818d0b$var$Form(props, ref) {
    props = $etHtu$reactspectrumprovider.useProviderProps(props);
    let { children: children , labelPosition: labelPosition = 'top' , labelAlign: labelAlign = 'start' , isRequired: isRequired , necessityIndicator: necessityIndicator , isQuiet: isQuiet , isEmphasized: isEmphasized , isDisabled: isDisabled , isReadOnly: isReadOnly , validationState: validationState , ...otherProps } = props;
    let { styleProps: styleProps  } = $etHtu$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $etHtu$reactspectrumutils.useDOMRef(ref);
    let ctx = {
        labelPosition: labelPosition,
        labelAlign: labelAlign,
        necessityIndicator: necessityIndicator
    };
    return(/*#__PURE__*/ ($parcel$interopDefault($etHtu$react)).createElement("form", {
        ...$etHtu$reactariautils.filterDOMProps(otherProps, {
            labelable: true,
            propNames: $8f20d6b8c0818d0b$var$formPropNames
        }),
        ...styleProps,
        noValidate: true,
        ref: domRef,
        className: $etHtu$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($0043d8d062ef413c$exports)), 'spectrum-Form', {
            'spectrum-Form--positionSide': labelPosition === 'side',
            'spectrum-Form--positionTop': labelPosition === 'top'
        }, styleProps.className)
    }, /*#__PURE__*/ ($parcel$interopDefault($etHtu$react)).createElement($8f20d6b8c0818d0b$var$FormContext.Provider, {
        value: ctx
    }, /*#__PURE__*/ ($parcel$interopDefault($etHtu$react)).createElement($etHtu$reactspectrumprovider.Provider, {
        isQuiet: isQuiet,
        isEmphasized: isEmphasized,
        isDisabled: isDisabled,
        isReadOnly: isReadOnly,
        isRequired: isRequired,
        validationState: validationState
    }, children))));
}
/**
 * Forms allow users to enter data that can be submitted while providing alignment and styling for form fields.
 */ const $8f20d6b8c0818d0b$export$a7fed597f4b8afd8 = /*#__PURE__*/ ($parcel$interopDefault($etHtu$react)).forwardRef($8f20d6b8c0818d0b$var$Form);


$parcel$exportWildcard(module.exports, $8f20d6b8c0818d0b$exports);
