// @react-spectrum/form/dist/module.js
import "./main.css";
import {useStyleProps as $8Y8y4$useStyleProps, useDOMRef as $8Y8y4$useDOMRef, classNames as $8Y8y4$classNames} from "@react-spectrum/utils";
import {filterDOMProps as $8Y8y4$filterDOMProps} from "@react-aria/utils";
import {useProviderProps as $8Y8y4$useProviderProps, Provider as $8Y8y4$Provider} from "@react-spectrum/provider";
import $8Y8y4$react, {useContext as $8Y8y4$useContext} from "react";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $a5eb5d835477bd21$exports = {};

$parcel$export($a5eb5d835477bd21$exports, "useFormProps", () => $a5eb5d835477bd21$export$a6b5be5c6b451665);
$parcel$export($a5eb5d835477bd21$exports, "Form", () => $a5eb5d835477bd21$export$a7fed597f4b8afd8);




var $4a84444d5a56ab41$exports = {};

$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel", () => $4a84444d5a56ab41$export$493d18e796ae054, (v) => $4a84444d5a56ab41$export$493d18e796ae054 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel--positionSide", () => $4a84444d5a56ab41$export$9e6f19dc21f22f2e, (v) => $4a84444d5a56ab41$export$9e6f19dc21f22f2e = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel-requiredIcon", () => $4a84444d5a56ab41$export$7c47642c7d46f3c9, (v) => $4a84444d5a56ab41$export$7c47642c7d46f3c9 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel--alignEnd", () => $4a84444d5a56ab41$export$885efcc08143a987, (v) => $4a84444d5a56ab41$export$885efcc08143a987 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field", () => $4a84444d5a56ab41$export$f6d480ae1e56eba0, (v) => $4a84444d5a56ab41$export$f6d480ae1e56eba0 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field--positionTop", () => $4a84444d5a56ab41$export$a4ea780a9064d7f9, (v) => $4a84444d5a56ab41$export$a4ea780a9064d7f9 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field-field", () => $4a84444d5a56ab41$export$3ab8a3dc7f0563d, (v) => $4a84444d5a56ab41$export$3ab8a3dc7f0563d = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field-field--multiline", () => $4a84444d5a56ab41$export$f58a1e966a92ba5e, (v) => $4a84444d5a56ab41$export$f58a1e966a92ba5e = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field--positionSide", () => $4a84444d5a56ab41$export$2963225c91179589, (v) => $4a84444d5a56ab41$export$2963225c91179589 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field-wrapper", () => $4a84444d5a56ab41$export$127df0b7290eb8ef, (v) => $4a84444d5a56ab41$export$127df0b7290eb8ef = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form", () => $4a84444d5a56ab41$export$94d2bcc94afabd89, (v) => $4a84444d5a56ab41$export$94d2bcc94afabd89 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form--positionSide", () => $4a84444d5a56ab41$export$7f44db659563c8f4, (v) => $4a84444d5a56ab41$export$7f44db659563c8f4 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form--positionTop", () => $4a84444d5a56ab41$export$11b8bc81fe551057, (v) => $4a84444d5a56ab41$export$11b8bc81fe551057 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form-itemLabel", () => $4a84444d5a56ab41$export$51ec27881a1e3928, (v) => $4a84444d5a56ab41$export$51ec27881a1e3928 = v);
$parcel$export($4a84444d5a56ab41$exports, "is-disabled", () => $4a84444d5a56ab41$export$d35bc1e505d1ebbf, (v) => $4a84444d5a56ab41$export$d35bc1e505d1ebbf = v);
var $4a84444d5a56ab41$export$493d18e796ae054;
var $4a84444d5a56ab41$export$9e6f19dc21f22f2e;
var $4a84444d5a56ab41$export$7c47642c7d46f3c9;
var $4a84444d5a56ab41$export$885efcc08143a987;
var $4a84444d5a56ab41$export$f6d480ae1e56eba0;
var $4a84444d5a56ab41$export$a4ea780a9064d7f9;
var $4a84444d5a56ab41$export$3ab8a3dc7f0563d;
var $4a84444d5a56ab41$export$f58a1e966a92ba5e;
var $4a84444d5a56ab41$export$2963225c91179589;
var $4a84444d5a56ab41$export$127df0b7290eb8ef;
var $4a84444d5a56ab41$export$94d2bcc94afabd89;
var $4a84444d5a56ab41$export$7f44db659563c8f4;
var $4a84444d5a56ab41$export$11b8bc81fe551057;
var $4a84444d5a56ab41$export$51ec27881a1e3928;
var $4a84444d5a56ab41$export$d35bc1e505d1ebbf;
$4a84444d5a56ab41$export$493d18e796ae054 = "spectrum-FieldLabel_d2db1f";
$4a84444d5a56ab41$export$9e6f19dc21f22f2e = "spectrum-FieldLabel--positionSide_d2db1f";
$4a84444d5a56ab41$export$7c47642c7d46f3c9 = "spectrum-FieldLabel-requiredIcon_d2db1f";
$4a84444d5a56ab41$export$885efcc08143a987 = "spectrum-FieldLabel--alignEnd_d2db1f";
$4a84444d5a56ab41$export$f6d480ae1e56eba0 = "spectrum-Field_d2db1f";
$4a84444d5a56ab41$export$a4ea780a9064d7f9 = "spectrum-Field--positionTop_d2db1f";
$4a84444d5a56ab41$export$3ab8a3dc7f0563d = "spectrum-Field-field_d2db1f";
$4a84444d5a56ab41$export$f58a1e966a92ba5e = "spectrum-Field-field--multiline_d2db1f";
$4a84444d5a56ab41$export$2963225c91179589 = "spectrum-Field--positionSide_d2db1f";
$4a84444d5a56ab41$export$127df0b7290eb8ef = "spectrum-Field-wrapper_d2db1f";
$4a84444d5a56ab41$export$94d2bcc94afabd89 = "spectrum-Form_d2db1f";
$4a84444d5a56ab41$export$7f44db659563c8f4 = "spectrum-Form--positionSide_d2db1f";
$4a84444d5a56ab41$export$11b8bc81fe551057 = "spectrum-Form--positionTop_d2db1f";
$4a84444d5a56ab41$export$51ec27881a1e3928 = "spectrum-Form-itemLabel_d2db1f";
$4a84444d5a56ab41$export$d35bc1e505d1ebbf = "is-disabled_d2db1f";


let $a5eb5d835477bd21$var$FormContext = /*#__PURE__*/ $8Y8y4$react.createContext({
});
function $a5eb5d835477bd21$export$a6b5be5c6b451665(props) {
    let ctx = $8Y8y4$useContext($a5eb5d835477bd21$var$FormContext);
    return {
        ...ctx,
        ...props
    };
}
const $a5eb5d835477bd21$var$formPropNames = new Set([
    'action',
    'autoComplete',
    'encType',
    'method',
    'target',
    'onSubmit'
]);
function $a5eb5d835477bd21$var$Form(props, ref) {
    props = $8Y8y4$useProviderProps(props);
    let { children: children , labelPosition: labelPosition = 'top' , labelAlign: labelAlign = 'start' , isRequired: isRequired , necessityIndicator: necessityIndicator , isQuiet: isQuiet , isEmphasized: isEmphasized , isDisabled: isDisabled , isReadOnly: isReadOnly , validationState: validationState , ...otherProps } = props;
    let { styleProps: styleProps  } = $8Y8y4$useStyleProps(otherProps);
    let domRef = $8Y8y4$useDOMRef(ref);
    let ctx = {
        labelPosition: labelPosition,
        labelAlign: labelAlign,
        necessityIndicator: necessityIndicator
    };
    return(/*#__PURE__*/ $8Y8y4$react.createElement("form", {
        ...$8Y8y4$filterDOMProps(otherProps, {
            labelable: true,
            propNames: $a5eb5d835477bd21$var$formPropNames
        }),
        ...styleProps,
        noValidate: true,
        ref: domRef,
        className: $8Y8y4$classNames((/*@__PURE__*/$parcel$interopDefault($4a84444d5a56ab41$exports)), 'spectrum-Form', {
            'spectrum-Form--positionSide': labelPosition === 'side',
            'spectrum-Form--positionTop': labelPosition === 'top'
        }, styleProps.className)
    }, /*#__PURE__*/ $8Y8y4$react.createElement($a5eb5d835477bd21$var$FormContext.Provider, {
        value: ctx
    }, /*#__PURE__*/ $8Y8y4$react.createElement($8Y8y4$Provider, {
        isQuiet: isQuiet,
        isEmphasized: isEmphasized,
        isDisabled: isDisabled,
        isReadOnly: isReadOnly,
        isRequired: isRequired,
        validationState: validationState
    }, children))));
}
/**
 * Forms allow users to enter data that can be submitted while providing alignment and styling for form fields.
 */ const $a5eb5d835477bd21$export$a7fed597f4b8afd8 = /*#__PURE__*/ $8Y8y4$react.forwardRef($a5eb5d835477bd21$var$Form);




export {$a5eb5d835477bd21$export$a6b5be5c6b451665 as useFormProps, $a5eb5d835477bd21$export$a7fed597f4b8afd8 as Form};
