// @react-spectrum/well/dist/main.js
require("./main.css");
var $4HELH$reactspectrumutils = require("@react-spectrum/utils");
var $4HELH$reactariautils = require("@react-aria/utils");
var $4HELH$react = require("react");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $a7f58e5eb71010ff$exports = {};

$parcel$export($a7f58e5eb71010ff$exports, "Well", () => $a7f58e5eb71010ff$export$d89c679ee2f73242);



var $841d34b2d7b814bb$exports = {};

$parcel$export($841d34b2d7b814bb$exports, "spectrum-Well", () => $841d34b2d7b814bb$export$98e76d3873e01aa9, (v) => $841d34b2d7b814bb$export$98e76d3873e01aa9 = v);
var $841d34b2d7b814bb$export$98e76d3873e01aa9;
$841d34b2d7b814bb$export$98e76d3873e01aa9 = "spectrum-Well_d48698";


function $a7f58e5eb71010ff$var$Well(props, ref) {
    let { children: children , role: role , ...otherProps } = props;
    let domRef = $4HELH$reactspectrumutils.useDOMRef(ref);
    let { styleProps: styleProps  } = $4HELH$reactspectrumutils.useStyleProps(otherProps);
    if (!role && (props['aria-label'] || props['aria-labelledby'])) console.warn('A labelled Well must have a role.');
    return(/*#__PURE__*/ ($parcel$interopDefault($4HELH$react)).createElement("div", {
        ...$4HELH$reactariautils.filterDOMProps(otherProps, {
            labelable: !!role
        }),
        ...styleProps,
        role: role,
        ref: domRef,
        className: $4HELH$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($841d34b2d7b814bb$exports)), 'spectrum-Well', styleProps.className)
    }, children));
}
/**
 * A Well is a content container that displays non-editable content separate from other content on the screen.
 * Often this is used to display preformatted text, such as code/markup examples on a documentation page.
 */ const $a7f58e5eb71010ff$export$d89c679ee2f73242 = /*#__PURE__*/ $4HELH$react.forwardRef($a7f58e5eb71010ff$var$Well);


$parcel$exportWildcard(module.exports, $a7f58e5eb71010ff$exports);
