// @react-spectrum/well/dist/module.js
import "./main.css";
import {useDOMRef as $f4Z9U$useDOMRef, useStyleProps as $f4Z9U$useStyleProps, classNames as $f4Z9U$classNames} from "@react-spectrum/utils";
import {filterDOMProps as $f4Z9U$filterDOMProps} from "@react-aria/utils";
import $f4Z9U$react, {forwardRef as $f4Z9U$forwardRef} from "react";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $727f016123196149$exports = {};

$parcel$export($727f016123196149$exports, "Well", () => $727f016123196149$export$d89c679ee2f73242);



var $2caa32d79596cf78$exports = {};

$parcel$export($2caa32d79596cf78$exports, "spectrum-Well", () => $2caa32d79596cf78$export$98e76d3873e01aa9, (v) => $2caa32d79596cf78$export$98e76d3873e01aa9 = v);
var $2caa32d79596cf78$export$98e76d3873e01aa9;
$2caa32d79596cf78$export$98e76d3873e01aa9 = "spectrum-Well_d48698";


function $727f016123196149$var$Well(props, ref) {
    let { children: children , role: role , ...otherProps } = props;
    let domRef = $f4Z9U$useDOMRef(ref);
    let { styleProps: styleProps  } = $f4Z9U$useStyleProps(otherProps);
    if (!role && (props['aria-label'] || props['aria-labelledby'])) console.warn('A labelled Well must have a role.');
    return(/*#__PURE__*/ $f4Z9U$react.createElement("div", {
        ...$f4Z9U$filterDOMProps(otherProps, {
            labelable: !!role
        }),
        ...styleProps,
        role: role,
        ref: domRef,
        className: $f4Z9U$classNames((/*@__PURE__*/$parcel$interopDefault($2caa32d79596cf78$exports)), 'spectrum-Well', styleProps.className)
    }, children));
}
/**
 * A Well is a content container that displays non-editable content separate from other content on the screen.
 * Often this is used to display preformatted text, such as code/markup examples on a documentation page.
 */ const $727f016123196149$export$d89c679ee2f73242 = /*#__PURE__*/ $f4Z9U$forwardRef($727f016123196149$var$Well);




export {$727f016123196149$export$d89c679ee2f73242 as Well};
