// @react-spectrum/tooltip/dist/main.js
require("./main.css");
var $60prd$spectrumiconsuiAlertSmall = require("@spectrum-icons/ui/AlertSmall");
var $60prd$reactspectrumutils = require("@react-spectrum/utils");
var $60prd$spectrumiconsuiInfoSmall = require("@spectrum-icons/ui/InfoSmall");
var $60prd$reactariautils = require("@react-aria/utils");
var $60prd$react = require("react");
var $60prd$spectrumiconsuiSuccessSmall = require("@spectrum-icons/ui/SuccessSmall");
var $60prd$reactariatooltip = require("@react-aria/tooltip");
var $60prd$reactariafocus = require("@react-aria/focus");
var $60prd$reactspectrumoverlays = require("@react-spectrum/overlays");
var $60prd$reactariaoverlays = require("@react-aria/overlays");
var $60prd$reactstatelytooltip = require("@react-stately/tooltip");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $a3079462de72782a$exports = {};

$parcel$export($a3079462de72782a$exports, "Tooltip", () => $a3079462de72782a$export$28c660c63b792dea);





var $9da618d120423c4f$exports = {};

$parcel$export($9da618d120423c4f$exports, "spectrum-Tooltip", () => $9da618d120423c4f$export$f7bf7078d3d03f05, (v) => $9da618d120423c4f$export$f7bf7078d3d03f05 = v);
$parcel$export($9da618d120423c4f$exports, "is-open", () => $9da618d120423c4f$export$a9781837241c946d, (v) => $9da618d120423c4f$export$a9781837241c946d = v);
$parcel$export($9da618d120423c4f$exports, "spectrum-Tooltip--bottom", () => $9da618d120423c4f$export$4f420a9bbb442953, (v) => $9da618d120423c4f$export$4f420a9bbb442953 = v);
$parcel$export($9da618d120423c4f$exports, "spectrum-Tooltip--top", () => $9da618d120423c4f$export$d7516df59e198fc7, (v) => $9da618d120423c4f$export$d7516df59e198fc7 = v);
$parcel$export($9da618d120423c4f$exports, "spectrum-Tooltip--right", () => $9da618d120423c4f$export$c094187772823b2e, (v) => $9da618d120423c4f$export$c094187772823b2e = v);
$parcel$export($9da618d120423c4f$exports, "spectrum-Tooltip--left", () => $9da618d120423c4f$export$47f060b68b4cab2c, (v) => $9da618d120423c4f$export$47f060b68b4cab2c = v);
$parcel$export($9da618d120423c4f$exports, "spectrum-Tooltip-tip", () => $9da618d120423c4f$export$9deb89a7a91de30, (v) => $9da618d120423c4f$export$9deb89a7a91de30 = v);
$parcel$export($9da618d120423c4f$exports, "spectrum-Tooltip-typeIcon", () => $9da618d120423c4f$export$ff23d46fee8272e4, (v) => $9da618d120423c4f$export$ff23d46fee8272e4 = v);
$parcel$export($9da618d120423c4f$exports, "spectrum-Tooltip-label", () => $9da618d120423c4f$export$f4096e0dc445b255, (v) => $9da618d120423c4f$export$f4096e0dc445b255 = v);
$parcel$export($9da618d120423c4f$exports, "u-tooltip-showOnHover", () => $9da618d120423c4f$export$6ea3237f62800d00, (v) => $9da618d120423c4f$export$6ea3237f62800d00 = v);
$parcel$export($9da618d120423c4f$exports, "is-hovered", () => $9da618d120423c4f$export$b8813cd5d7824ce7, (v) => $9da618d120423c4f$export$b8813cd5d7824ce7 = v);
$parcel$export($9da618d120423c4f$exports, "is-focused", () => $9da618d120423c4f$export$e7dc768d35940237, (v) => $9da618d120423c4f$export$e7dc768d35940237 = v);
$parcel$export($9da618d120423c4f$exports, "spectrum-Tooltip--negative", () => $9da618d120423c4f$export$92fe45b4260594f6, (v) => $9da618d120423c4f$export$92fe45b4260594f6 = v);
$parcel$export($9da618d120423c4f$exports, "spectrum-Tooltip--error", () => $9da618d120423c4f$export$2b291890a5577893, (v) => $9da618d120423c4f$export$2b291890a5577893 = v);
$parcel$export($9da618d120423c4f$exports, "spectrum-Tooltip--info", () => $9da618d120423c4f$export$e995d5780591d0a8, (v) => $9da618d120423c4f$export$e995d5780591d0a8 = v);
$parcel$export($9da618d120423c4f$exports, "spectrum-Tooltip--help", () => $9da618d120423c4f$export$d33ca445ae528241, (v) => $9da618d120423c4f$export$d33ca445ae528241 = v);
$parcel$export($9da618d120423c4f$exports, "spectrum-Tooltip--positive", () => $9da618d120423c4f$export$11cbf31e1b6e51ac, (v) => $9da618d120423c4f$export$11cbf31e1b6e51ac = v);
$parcel$export($9da618d120423c4f$exports, "spectrum-Tooltip--success", () => $9da618d120423c4f$export$4eae622bc9b08cac, (v) => $9da618d120423c4f$export$4eae622bc9b08cac = v);
var $9da618d120423c4f$export$f7bf7078d3d03f05;
var $9da618d120423c4f$export$a9781837241c946d;
var $9da618d120423c4f$export$4f420a9bbb442953;
var $9da618d120423c4f$export$d7516df59e198fc7;
var $9da618d120423c4f$export$c094187772823b2e;
var $9da618d120423c4f$export$47f060b68b4cab2c;
var $9da618d120423c4f$export$9deb89a7a91de30;
var $9da618d120423c4f$export$ff23d46fee8272e4;
var $9da618d120423c4f$export$f4096e0dc445b255;
var $9da618d120423c4f$export$6ea3237f62800d00;
var $9da618d120423c4f$export$b8813cd5d7824ce7;
var $9da618d120423c4f$export$e7dc768d35940237;
var $9da618d120423c4f$export$92fe45b4260594f6;
var $9da618d120423c4f$export$2b291890a5577893;
var $9da618d120423c4f$export$e995d5780591d0a8;
var $9da618d120423c4f$export$d33ca445ae528241;
var $9da618d120423c4f$export$11cbf31e1b6e51ac;
var $9da618d120423c4f$export$4eae622bc9b08cac;
$9da618d120423c4f$export$f7bf7078d3d03f05 = "spectrum-Tooltip_22ae94";
$9da618d120423c4f$export$a9781837241c946d = "is-open_22ae94";
$9da618d120423c4f$export$4f420a9bbb442953 = "spectrum-Tooltip--bottom_22ae94";
$9da618d120423c4f$export$d7516df59e198fc7 = "spectrum-Tooltip--top_22ae94";
$9da618d120423c4f$export$c094187772823b2e = "spectrum-Tooltip--right_22ae94";
$9da618d120423c4f$export$47f060b68b4cab2c = "spectrum-Tooltip--left_22ae94";
$9da618d120423c4f$export$9deb89a7a91de30 = "spectrum-Tooltip-tip_22ae94";
$9da618d120423c4f$export$ff23d46fee8272e4 = "spectrum-Tooltip-typeIcon_22ae94";
$9da618d120423c4f$export$f4096e0dc445b255 = "spectrum-Tooltip-label_22ae94";
$9da618d120423c4f$export$6ea3237f62800d00 = "u-tooltip-showOnHover_22ae94";
$9da618d120423c4f$export$b8813cd5d7824ce7 = "is-hovered_22ae94";
$9da618d120423c4f$export$e7dc768d35940237 = "is-focused_22ae94";
$9da618d120423c4f$export$92fe45b4260594f6 = "spectrum-Tooltip--negative_22ae94";
$9da618d120423c4f$export$2b291890a5577893 = "spectrum-Tooltip--error_22ae94";
$9da618d120423c4f$export$e995d5780591d0a8 = "spectrum-Tooltip--info_22ae94";
$9da618d120423c4f$export$d33ca445ae528241 = "spectrum-Tooltip--help_22ae94";
$9da618d120423c4f$export$11cbf31e1b6e51ac = "spectrum-Tooltip--positive_22ae94";
$9da618d120423c4f$export$4eae622bc9b08cac = "spectrum-Tooltip--success_22ae94";




const $d1ffd047957dcedf$export$39ae08fa83328b12 = ($parcel$interopDefault($60prd$react)).createContext({
});



let $a3079462de72782a$var$iconMap = {
    info: ($parcel$interopDefault($60prd$spectrumiconsuiInfoSmall)),
    positive: ($parcel$interopDefault($60prd$spectrumiconsuiSuccessSmall)),
    negative: ($parcel$interopDefault($60prd$spectrumiconsuiAlertSmall))
};
function $a3079462de72782a$var$Tooltip(props, ref) {
    let { ref: overlayRef , arrowProps: arrowProps , state: state , ...tooltipProviderProps } = $60prd$react.useContext($d1ffd047957dcedf$export$39ae08fa83328b12);
    let defaultRef = $60prd$react.useRef();
    overlayRef = overlayRef || defaultRef;
    props = $60prd$reactariautils.mergeProps(props, tooltipProviderProps);
    let { variant: variant = 'neutral' , placement: placement = 'top' , isOpen: isOpen , showIcon: showIcon , ...otherProps } = props;
    let { styleProps: styleProps  } = $60prd$reactspectrumutils.useStyleProps(otherProps);
    let { tooltipProps: tooltipProps  } = $60prd$reactariatooltip.useTooltip(props, state);
    // Sync ref with overlayRef from context.
    $60prd$react.useImperativeHandle(ref, ()=>$60prd$reactspectrumutils.createDOMRef(overlayRef)
    );
    let Icon = $a3079462de72782a$var$iconMap[variant];
    return(/*#__PURE__*/ ($parcel$interopDefault($60prd$react)).createElement("div", {
        ...styleProps,
        ...tooltipProps,
        className: $60prd$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($9da618d120423c4f$exports)), 'spectrum-Tooltip', `spectrum-Tooltip--${variant}`, `spectrum-Tooltip--${placement}`, {
            'is-open': isOpen
        }, styleProps.className),
        ref: overlayRef
    }, showIcon && variant !== 'neutral' && /*#__PURE__*/ ($parcel$interopDefault($60prd$react)).createElement(Icon, {
        UNSAFE_className: $60prd$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($9da618d120423c4f$exports)), 'spectrum-Tooltip-typeIcon'),
        "aria-hidden": true
    }), props.children && /*#__PURE__*/ ($parcel$interopDefault($60prd$react)).createElement("span", {
        className: $60prd$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($9da618d120423c4f$exports)), 'spectrum-Tooltip-label')
    }, props.children), /*#__PURE__*/ ($parcel$interopDefault($60prd$react)).createElement("span", {
        ...arrowProps,
        className: $60prd$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($9da618d120423c4f$exports)), 'spectrum-Tooltip-tip')
    })));
}
/**
 * Display container for Tooltip content. Has a directional arrow dependent on its placement.
 */ let $a3079462de72782a$export$28c660c63b792dea = /*#__PURE__*/ ($parcel$interopDefault($60prd$react)).forwardRef($a3079462de72782a$var$Tooltip);


var $c1431f05b5e3cfe8$exports = {};

$parcel$export($c1431f05b5e3cfe8$exports, "TooltipTrigger", () => $c1431f05b5e3cfe8$export$8c610744efcf8a1d);







const $c1431f05b5e3cfe8$var$DEFAULT_OFFSET = -1; // Offset needed to reach 4px/5px (med/large) distance between tooltip and trigger button
const $c1431f05b5e3cfe8$var$DEFAULT_CROSS_OFFSET = 0;
function $c1431f05b5e3cfe8$var$TooltipTrigger(props) {
    let { children: children , crossOffset: crossOffset = $c1431f05b5e3cfe8$var$DEFAULT_CROSS_OFFSET , isDisabled: isDisabled , offset: offset = $c1431f05b5e3cfe8$var$DEFAULT_OFFSET , trigger: triggerAction  } = props;
    let [trigger, tooltip] = ($parcel$interopDefault($60prd$react)).Children.toArray(children);
    let state = $60prd$reactstatelytooltip.useTooltipTriggerState(props);
    let tooltipTriggerRef = $60prd$react.useRef();
    let overlayRef = $60prd$react.useRef();
    let { triggerProps: triggerProps , tooltipProps: tooltipProps  } = $60prd$reactariatooltip.useTooltipTrigger({
        isDisabled: isDisabled,
        trigger: triggerAction
    }, state, tooltipTriggerRef);
    let { overlayProps: overlayProps , arrowProps: arrowProps , placement: placement  } = $60prd$reactariaoverlays.useOverlayPosition({
        placement: props.placement || 'top',
        targetRef: tooltipTriggerRef,
        overlayRef: overlayRef,
        offset: offset,
        crossOffset: crossOffset,
        isOpen: state.isOpen
    });
    return(/*#__PURE__*/ ($parcel$interopDefault($60prd$react)).createElement($60prd$reactariafocus.FocusableProvider, {
        ...triggerProps,
        ref: tooltipTriggerRef
    }, trigger, /*#__PURE__*/ ($parcel$interopDefault($60prd$react)).createElement($d1ffd047957dcedf$export$39ae08fa83328b12.Provider, {
        value: {
            state: state,
            placement: placement,
            ref: overlayRef,
            UNSAFE_style: overlayProps.style,
            arrowProps: arrowProps,
            ...tooltipProps
        }
    }, /*#__PURE__*/ ($parcel$interopDefault($60prd$react)).createElement($60prd$reactspectrumoverlays.Overlay, {
        isOpen: state.isOpen
    }, tooltip))));
}
// Support TooltipTrigger inside components using CollectionBuilder.
$c1431f05b5e3cfe8$var$TooltipTrigger.getCollectionNode = function*(props) {
    // Replaced the use of React.Childern.toArray because it mutates the key prop.
    let childArray = [];
    ($parcel$interopDefault($60prd$react)).Children.forEach(props.children, (child)=>{
        if (/*#__PURE__*/ ($parcel$interopDefault($60prd$react)).isValidElement(child)) childArray.push(child);
    });
    let [trigger, tooltip] = childArray;
    yield {
        element: trigger,
        wrapper: (element)=>/*#__PURE__*/ ($parcel$interopDefault($60prd$react)).createElement($c1431f05b5e3cfe8$var$TooltipTrigger, {
                key: element.key,
                ...props
            }, element, tooltip)
    };
};
/**
 * TooltipTrigger wraps around a trigger element and a Tooltip. It handles opening and closing
 * the Tooltip when the user hovers over or focuses the trigger, and positioning the Tooltip
 * relative to the trigger.
 */ // We don't want getCollectionNode to show up in the type definition
let $c1431f05b5e3cfe8$export$8c610744efcf8a1d = $c1431f05b5e3cfe8$var$TooltipTrigger;


$parcel$exportWildcard(module.exports, $a3079462de72782a$exports);
$parcel$exportWildcard(module.exports, $c1431f05b5e3cfe8$exports);
