// @react-spectrum/tooltip/dist/module.js
import "./main.css";
import $3jE8v$spectrumiconsuiAlertSmall from "@spectrum-icons/ui/AlertSmall";
import {useStyleProps as $3jE8v$useStyleProps, createDOMRef as $3jE8v$createDOMRef, classNames as $3jE8v$classNames} from "@react-spectrum/utils";
import $3jE8v$spectrumiconsuiInfoSmall from "@spectrum-icons/ui/InfoSmall";
import {mergeProps as $3jE8v$mergeProps} from "@react-aria/utils";
import $3jE8v$react, {useContext as $3jE8v$useContext, useRef as $3jE8v$useRef, useImperativeHandle as $3jE8v$useImperativeHandle} from "react";
import $3jE8v$spectrumiconsuiSuccessSmall from "@spectrum-icons/ui/SuccessSmall";
import {useTooltip as $3jE8v$useTooltip, useTooltipTrigger as $3jE8v$useTooltipTrigger} from "@react-aria/tooltip";
import {FocusableProvider as $3jE8v$FocusableProvider} from "@react-aria/focus";
import {Overlay as $3jE8v$Overlay} from "@react-spectrum/overlays";
import {useOverlayPosition as $3jE8v$useOverlayPosition} from "@react-aria/overlays";
import {useTooltipTriggerState as $3jE8v$useTooltipTriggerState} from "@react-stately/tooltip";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $9128be99fa3a92ab$exports = {};

$parcel$export($9128be99fa3a92ab$exports, "Tooltip", () => $9128be99fa3a92ab$export$28c660c63b792dea);





var $838bbe40f48a4d98$exports = {};

$parcel$export($838bbe40f48a4d98$exports, "spectrum-Tooltip", () => $838bbe40f48a4d98$export$f7bf7078d3d03f05, (v) => $838bbe40f48a4d98$export$f7bf7078d3d03f05 = v);
$parcel$export($838bbe40f48a4d98$exports, "is-open", () => $838bbe40f48a4d98$export$a9781837241c946d, (v) => $838bbe40f48a4d98$export$a9781837241c946d = v);
$parcel$export($838bbe40f48a4d98$exports, "spectrum-Tooltip--bottom", () => $838bbe40f48a4d98$export$4f420a9bbb442953, (v) => $838bbe40f48a4d98$export$4f420a9bbb442953 = v);
$parcel$export($838bbe40f48a4d98$exports, "spectrum-Tooltip--top", () => $838bbe40f48a4d98$export$d7516df59e198fc7, (v) => $838bbe40f48a4d98$export$d7516df59e198fc7 = v);
$parcel$export($838bbe40f48a4d98$exports, "spectrum-Tooltip--right", () => $838bbe40f48a4d98$export$c094187772823b2e, (v) => $838bbe40f48a4d98$export$c094187772823b2e = v);
$parcel$export($838bbe40f48a4d98$exports, "spectrum-Tooltip--left", () => $838bbe40f48a4d98$export$47f060b68b4cab2c, (v) => $838bbe40f48a4d98$export$47f060b68b4cab2c = v);
$parcel$export($838bbe40f48a4d98$exports, "spectrum-Tooltip-tip", () => $838bbe40f48a4d98$export$9deb89a7a91de30, (v) => $838bbe40f48a4d98$export$9deb89a7a91de30 = v);
$parcel$export($838bbe40f48a4d98$exports, "spectrum-Tooltip-typeIcon", () => $838bbe40f48a4d98$export$ff23d46fee8272e4, (v) => $838bbe40f48a4d98$export$ff23d46fee8272e4 = v);
$parcel$export($838bbe40f48a4d98$exports, "spectrum-Tooltip-label", () => $838bbe40f48a4d98$export$f4096e0dc445b255, (v) => $838bbe40f48a4d98$export$f4096e0dc445b255 = v);
$parcel$export($838bbe40f48a4d98$exports, "u-tooltip-showOnHover", () => $838bbe40f48a4d98$export$6ea3237f62800d00, (v) => $838bbe40f48a4d98$export$6ea3237f62800d00 = v);
$parcel$export($838bbe40f48a4d98$exports, "is-hovered", () => $838bbe40f48a4d98$export$b8813cd5d7824ce7, (v) => $838bbe40f48a4d98$export$b8813cd5d7824ce7 = v);
$parcel$export($838bbe40f48a4d98$exports, "is-focused", () => $838bbe40f48a4d98$export$e7dc768d35940237, (v) => $838bbe40f48a4d98$export$e7dc768d35940237 = v);
$parcel$export($838bbe40f48a4d98$exports, "spectrum-Tooltip--negative", () => $838bbe40f48a4d98$export$92fe45b4260594f6, (v) => $838bbe40f48a4d98$export$92fe45b4260594f6 = v);
$parcel$export($838bbe40f48a4d98$exports, "spectrum-Tooltip--error", () => $838bbe40f48a4d98$export$2b291890a5577893, (v) => $838bbe40f48a4d98$export$2b291890a5577893 = v);
$parcel$export($838bbe40f48a4d98$exports, "spectrum-Tooltip--info", () => $838bbe40f48a4d98$export$e995d5780591d0a8, (v) => $838bbe40f48a4d98$export$e995d5780591d0a8 = v);
$parcel$export($838bbe40f48a4d98$exports, "spectrum-Tooltip--help", () => $838bbe40f48a4d98$export$d33ca445ae528241, (v) => $838bbe40f48a4d98$export$d33ca445ae528241 = v);
$parcel$export($838bbe40f48a4d98$exports, "spectrum-Tooltip--positive", () => $838bbe40f48a4d98$export$11cbf31e1b6e51ac, (v) => $838bbe40f48a4d98$export$11cbf31e1b6e51ac = v);
$parcel$export($838bbe40f48a4d98$exports, "spectrum-Tooltip--success", () => $838bbe40f48a4d98$export$4eae622bc9b08cac, (v) => $838bbe40f48a4d98$export$4eae622bc9b08cac = v);
var $838bbe40f48a4d98$export$f7bf7078d3d03f05;
var $838bbe40f48a4d98$export$a9781837241c946d;
var $838bbe40f48a4d98$export$4f420a9bbb442953;
var $838bbe40f48a4d98$export$d7516df59e198fc7;
var $838bbe40f48a4d98$export$c094187772823b2e;
var $838bbe40f48a4d98$export$47f060b68b4cab2c;
var $838bbe40f48a4d98$export$9deb89a7a91de30;
var $838bbe40f48a4d98$export$ff23d46fee8272e4;
var $838bbe40f48a4d98$export$f4096e0dc445b255;
var $838bbe40f48a4d98$export$6ea3237f62800d00;
var $838bbe40f48a4d98$export$b8813cd5d7824ce7;
var $838bbe40f48a4d98$export$e7dc768d35940237;
var $838bbe40f48a4d98$export$92fe45b4260594f6;
var $838bbe40f48a4d98$export$2b291890a5577893;
var $838bbe40f48a4d98$export$e995d5780591d0a8;
var $838bbe40f48a4d98$export$d33ca445ae528241;
var $838bbe40f48a4d98$export$11cbf31e1b6e51ac;
var $838bbe40f48a4d98$export$4eae622bc9b08cac;
$838bbe40f48a4d98$export$f7bf7078d3d03f05 = "spectrum-Tooltip_22ae94";
$838bbe40f48a4d98$export$a9781837241c946d = "is-open_22ae94";
$838bbe40f48a4d98$export$4f420a9bbb442953 = "spectrum-Tooltip--bottom_22ae94";
$838bbe40f48a4d98$export$d7516df59e198fc7 = "spectrum-Tooltip--top_22ae94";
$838bbe40f48a4d98$export$c094187772823b2e = "spectrum-Tooltip--right_22ae94";
$838bbe40f48a4d98$export$47f060b68b4cab2c = "spectrum-Tooltip--left_22ae94";
$838bbe40f48a4d98$export$9deb89a7a91de30 = "spectrum-Tooltip-tip_22ae94";
$838bbe40f48a4d98$export$ff23d46fee8272e4 = "spectrum-Tooltip-typeIcon_22ae94";
$838bbe40f48a4d98$export$f4096e0dc445b255 = "spectrum-Tooltip-label_22ae94";
$838bbe40f48a4d98$export$6ea3237f62800d00 = "u-tooltip-showOnHover_22ae94";
$838bbe40f48a4d98$export$b8813cd5d7824ce7 = "is-hovered_22ae94";
$838bbe40f48a4d98$export$e7dc768d35940237 = "is-focused_22ae94";
$838bbe40f48a4d98$export$92fe45b4260594f6 = "spectrum-Tooltip--negative_22ae94";
$838bbe40f48a4d98$export$2b291890a5577893 = "spectrum-Tooltip--error_22ae94";
$838bbe40f48a4d98$export$e995d5780591d0a8 = "spectrum-Tooltip--info_22ae94";
$838bbe40f48a4d98$export$d33ca445ae528241 = "spectrum-Tooltip--help_22ae94";
$838bbe40f48a4d98$export$11cbf31e1b6e51ac = "spectrum-Tooltip--positive_22ae94";
$838bbe40f48a4d98$export$4eae622bc9b08cac = "spectrum-Tooltip--success_22ae94";




const $ddbdbe5044e23e94$export$39ae08fa83328b12 = $3jE8v$react.createContext({
});



let $9128be99fa3a92ab$var$iconMap = {
    info: $3jE8v$spectrumiconsuiInfoSmall,
    positive: $3jE8v$spectrumiconsuiSuccessSmall,
    negative: $3jE8v$spectrumiconsuiAlertSmall
};
function $9128be99fa3a92ab$var$Tooltip(props, ref) {
    let { ref: overlayRef , arrowProps: arrowProps , state: state , ...tooltipProviderProps } = $3jE8v$useContext($ddbdbe5044e23e94$export$39ae08fa83328b12);
    let defaultRef = $3jE8v$useRef();
    overlayRef = overlayRef || defaultRef;
    props = $3jE8v$mergeProps(props, tooltipProviderProps);
    let { variant: variant = 'neutral' , placement: placement = 'top' , isOpen: isOpen , showIcon: showIcon , ...otherProps } = props;
    let { styleProps: styleProps  } = $3jE8v$useStyleProps(otherProps);
    let { tooltipProps: tooltipProps  } = $3jE8v$useTooltip(props, state);
    // Sync ref with overlayRef from context.
    $3jE8v$useImperativeHandle(ref, ()=>$3jE8v$createDOMRef(overlayRef)
    );
    let Icon = $9128be99fa3a92ab$var$iconMap[variant];
    return(/*#__PURE__*/ $3jE8v$react.createElement("div", {
        ...styleProps,
        ...tooltipProps,
        className: $3jE8v$classNames((/*@__PURE__*/$parcel$interopDefault($838bbe40f48a4d98$exports)), 'spectrum-Tooltip', `spectrum-Tooltip--${variant}`, `spectrum-Tooltip--${placement}`, {
            'is-open': isOpen
        }, styleProps.className),
        ref: overlayRef
    }, showIcon && variant !== 'neutral' && /*#__PURE__*/ $3jE8v$react.createElement(Icon, {
        UNSAFE_className: $3jE8v$classNames((/*@__PURE__*/$parcel$interopDefault($838bbe40f48a4d98$exports)), 'spectrum-Tooltip-typeIcon'),
        "aria-hidden": true
    }), props.children && /*#__PURE__*/ $3jE8v$react.createElement("span", {
        className: $3jE8v$classNames((/*@__PURE__*/$parcel$interopDefault($838bbe40f48a4d98$exports)), 'spectrum-Tooltip-label')
    }, props.children), /*#__PURE__*/ $3jE8v$react.createElement("span", {
        ...arrowProps,
        className: $3jE8v$classNames((/*@__PURE__*/$parcel$interopDefault($838bbe40f48a4d98$exports)), 'spectrum-Tooltip-tip')
    })));
}
/**
 * Display container for Tooltip content. Has a directional arrow dependent on its placement.
 */ let $9128be99fa3a92ab$export$28c660c63b792dea = /*#__PURE__*/ $3jE8v$react.forwardRef($9128be99fa3a92ab$var$Tooltip);


var $a354ae152307fd5d$exports = {};

$parcel$export($a354ae152307fd5d$exports, "TooltipTrigger", () => $a354ae152307fd5d$export$8c610744efcf8a1d);







const $a354ae152307fd5d$var$DEFAULT_OFFSET = -1; // Offset needed to reach 4px/5px (med/large) distance between tooltip and trigger button
const $a354ae152307fd5d$var$DEFAULT_CROSS_OFFSET = 0;
function $a354ae152307fd5d$var$TooltipTrigger(props) {
    let { children: children , crossOffset: crossOffset = $a354ae152307fd5d$var$DEFAULT_CROSS_OFFSET , isDisabled: isDisabled , offset: offset = $a354ae152307fd5d$var$DEFAULT_OFFSET , trigger: triggerAction  } = props;
    let [trigger, tooltip] = $3jE8v$react.Children.toArray(children);
    let state = $3jE8v$useTooltipTriggerState(props);
    let tooltipTriggerRef = $3jE8v$useRef();
    let overlayRef = $3jE8v$useRef();
    let { triggerProps: triggerProps , tooltipProps: tooltipProps  } = $3jE8v$useTooltipTrigger({
        isDisabled: isDisabled,
        trigger: triggerAction
    }, state, tooltipTriggerRef);
    let { overlayProps: overlayProps , arrowProps: arrowProps , placement: placement  } = $3jE8v$useOverlayPosition({
        placement: props.placement || 'top',
        targetRef: tooltipTriggerRef,
        overlayRef: overlayRef,
        offset: offset,
        crossOffset: crossOffset,
        isOpen: state.isOpen
    });
    return(/*#__PURE__*/ $3jE8v$react.createElement($3jE8v$FocusableProvider, {
        ...triggerProps,
        ref: tooltipTriggerRef
    }, trigger, /*#__PURE__*/ $3jE8v$react.createElement($ddbdbe5044e23e94$export$39ae08fa83328b12.Provider, {
        value: {
            state: state,
            placement: placement,
            ref: overlayRef,
            UNSAFE_style: overlayProps.style,
            arrowProps: arrowProps,
            ...tooltipProps
        }
    }, /*#__PURE__*/ $3jE8v$react.createElement($3jE8v$Overlay, {
        isOpen: state.isOpen
    }, tooltip))));
}
// Support TooltipTrigger inside components using CollectionBuilder.
$a354ae152307fd5d$var$TooltipTrigger.getCollectionNode = function*(props) {
    // Replaced the use of React.Childern.toArray because it mutates the key prop.
    let childArray = [];
    $3jE8v$react.Children.forEach(props.children, (child)=>{
        if (/*#__PURE__*/ $3jE8v$react.isValidElement(child)) childArray.push(child);
    });
    let [trigger, tooltip] = childArray;
    yield {
        element: trigger,
        wrapper: (element)=>/*#__PURE__*/ $3jE8v$react.createElement($a354ae152307fd5d$var$TooltipTrigger, {
                key: element.key,
                ...props
            }, element, tooltip)
    };
};
/**
 * TooltipTrigger wraps around a trigger element and a Tooltip. It handles opening and closing
 * the Tooltip when the user hovers over or focuses the trigger, and positioning the Tooltip
 * relative to the trigger.
 */ // We don't want getCollectionNode to show up in the type definition
let $a354ae152307fd5d$export$8c610744efcf8a1d = $a354ae152307fd5d$var$TooltipTrigger;




export {$9128be99fa3a92ab$export$28c660c63b792dea as Tooltip, $a354ae152307fd5d$export$8c610744efcf8a1d as TooltipTrigger};
