// @react-spectrum/breadcrumbs/dist/main.js
require("./main.css");
var $6zINN$reactstatelycollections = require("@react-stately/collections");
var $6zINN$reactspectrumbutton = require("@react-spectrum/button");
var $6zINN$reactspectrumutils = require("@react-spectrum/utils");
var $6zINN$spectrumiconsuiFolderBreadcrumb = require("@spectrum-icons/ui/FolderBreadcrumb");
var $6zINN$reactspectrummenu = require("@react-spectrum/menu");
var $6zINN$react = require("react");
var $6zINN$reactariabreadcrumbs = require("@react-aria/breadcrumbs");
var $6zINN$reactariautils = require("@react-aria/utils");
var $6zINN$reactspectrumprovider = require("@react-spectrum/provider");
var $6zINN$spectrumiconsuiChevronRightSmall = require("@spectrum-icons/ui/ChevronRightSmall");
var $6zINN$reactariafocus = require("@react-aria/focus");
var $6zINN$reactariainteractions = require("@react-aria/interactions");
var $6zINN$reactariai18n = require("@react-aria/i18n");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Item", () => $6zINN$reactstatelycollections.Item);
var $ce45029bf0c8e0f0$exports = {};

$parcel$export($ce45029bf0c8e0f0$exports, "Breadcrumbs", () => $ce45029bf0c8e0f0$export$2dc68d50d56fbbd);






var $7d405af0aae01906$exports = {};

$parcel$export($7d405af0aae01906$exports, "spectrum-Breadcrumbs", () => $7d405af0aae01906$export$8c6320d2db82081f, (v) => $7d405af0aae01906$export$8c6320d2db82081f = v);
$parcel$export($7d405af0aae01906$exports, "spectrum-Breadcrumbs-itemSeparator", () => $7d405af0aae01906$export$f72910e36cd91d1e, (v) => $7d405af0aae01906$export$f72910e36cd91d1e = v);
$parcel$export($7d405af0aae01906$exports, "is-reversed", () => $7d405af0aae01906$export$a4561bd665abb5db, (v) => $7d405af0aae01906$export$a4561bd665abb5db = v);
$parcel$export($7d405af0aae01906$exports, "spectrum-Breadcrumbs-item", () => $7d405af0aae01906$export$492a5983c5fbf603, (v) => $7d405af0aae01906$export$492a5983c5fbf603 = v);
$parcel$export($7d405af0aae01906$exports, "spectrum-ActionButton", () => $7d405af0aae01906$export$53da69f51b770d3, (v) => $7d405af0aae01906$export$53da69f51b770d3 = v);
$parcel$export($7d405af0aae01906$exports, "spectrum-Breadcrumbs-itemLink", () => $7d405af0aae01906$export$5af6a1bcd207d9f6, (v) => $7d405af0aae01906$export$5af6a1bcd207d9f6 = v);
$parcel$export($7d405af0aae01906$exports, "is-hovered", () => $7d405af0aae01906$export$b8813cd5d7824ce7, (v) => $7d405af0aae01906$export$b8813cd5d7824ce7 = v);
$parcel$export($7d405af0aae01906$exports, "focus-ring", () => $7d405af0aae01906$export$f39a09f249340e2a, (v) => $7d405af0aae01906$export$f39a09f249340e2a = v);
$parcel$export($7d405af0aae01906$exports, "is-dragged", () => $7d405af0aae01906$export$8778c911bed6c759, (v) => $7d405af0aae01906$export$8778c911bed6c759 = v);
$parcel$export($7d405af0aae01906$exports, "spectrum-Breadcrumbs--small", () => $7d405af0aae01906$export$e01337e7944312f6, (v) => $7d405af0aae01906$export$e01337e7944312f6 = v);
$parcel$export($7d405af0aae01906$exports, "spectrum-Breadcrumbs--medium", () => $7d405af0aae01906$export$19f957201d941895, (v) => $7d405af0aae01906$export$19f957201d941895 = v);
$parcel$export($7d405af0aae01906$exports, "spectrum-Breadcrumbs--multiline", () => $7d405af0aae01906$export$713961e8e432825b, (v) => $7d405af0aae01906$export$713961e8e432825b = v);
$parcel$export($7d405af0aae01906$exports, "spectrum-Heading--pageTitle", () => $7d405af0aae01906$export$63e9e087a73f808b, (v) => $7d405af0aae01906$export$63e9e087a73f808b = v);
$parcel$export($7d405af0aae01906$exports, "is-disabled", () => $7d405af0aae01906$export$d35bc1e505d1ebbf, (v) => $7d405af0aae01906$export$d35bc1e505d1ebbf = v);
$parcel$export($7d405af0aae01906$exports, "is-selected", () => $7d405af0aae01906$export$1e0fb04f31d3c22a, (v) => $7d405af0aae01906$export$1e0fb04f31d3c22a = v);
$parcel$export($7d405af0aae01906$exports, "spectrum-Breadcrumb", () => $7d405af0aae01906$export$45a5316056cfac3c, (v) => $7d405af0aae01906$export$45a5316056cfac3c = v);
var $7d405af0aae01906$export$8c6320d2db82081f;
var $7d405af0aae01906$export$f72910e36cd91d1e;
var $7d405af0aae01906$export$a4561bd665abb5db;
var $7d405af0aae01906$export$492a5983c5fbf603;
var $7d405af0aae01906$export$53da69f51b770d3;
var $7d405af0aae01906$export$5af6a1bcd207d9f6;
var $7d405af0aae01906$export$b8813cd5d7824ce7;
var $7d405af0aae01906$export$f39a09f249340e2a;
var $7d405af0aae01906$export$8778c911bed6c759;
var $7d405af0aae01906$export$e01337e7944312f6;
var $7d405af0aae01906$export$19f957201d941895;
var $7d405af0aae01906$export$713961e8e432825b;
var $7d405af0aae01906$export$63e9e087a73f808b;
var $7d405af0aae01906$export$d35bc1e505d1ebbf;
var $7d405af0aae01906$export$1e0fb04f31d3c22a;
var $7d405af0aae01906$export$45a5316056cfac3c;
$7d405af0aae01906$export$8c6320d2db82081f = "spectrum-Breadcrumbs_3fe6e8";
$7d405af0aae01906$export$f72910e36cd91d1e = "spectrum-Breadcrumbs-itemSeparator_3fe6e8";
$7d405af0aae01906$export$a4561bd665abb5db = "is-reversed_3fe6e8";
$7d405af0aae01906$export$492a5983c5fbf603 = "spectrum-Breadcrumbs-item_3fe6e8";
$7d405af0aae01906$export$53da69f51b770d3 = "spectrum-ActionButton_3fe6e8";
$7d405af0aae01906$export$5af6a1bcd207d9f6 = "spectrum-Breadcrumbs-itemLink_3fe6e8";
$7d405af0aae01906$export$b8813cd5d7824ce7 = "is-hovered_3fe6e8";
$7d405af0aae01906$export$f39a09f249340e2a = "focus-ring_3fe6e8";
$7d405af0aae01906$export$8778c911bed6c759 = "is-dragged_3fe6e8";
$7d405af0aae01906$export$e01337e7944312f6 = "spectrum-Breadcrumbs--small_3fe6e8";
$7d405af0aae01906$export$19f957201d941895 = "spectrum-Breadcrumbs--medium_3fe6e8";
$7d405af0aae01906$export$713961e8e432825b = "spectrum-Breadcrumbs--multiline_3fe6e8";
$7d405af0aae01906$export$63e9e087a73f808b = "spectrum-Heading--pageTitle_3fe6e8";
$7d405af0aae01906$export$d35bc1e505d1ebbf = "is-disabled_3fe6e8";
$7d405af0aae01906$export$1e0fb04f31d3c22a = "is-selected_3fe6e8";
$7d405af0aae01906$export$45a5316056cfac3c = "spectrum-Breadcrumb_3fe6e8";





function $01a6adf6454cb0d0$export$c13f210c706eb549(props) {
    let { children: children , isCurrent: isCurrent , isDisabled: isDisabled  } = props;
    let { direction: direction  } = $6zINN$reactariai18n.useLocale();
    let ref = $6zINN$react.useRef();
    let { itemProps: itemProps  } = $6zINN$reactariabreadcrumbs.useBreadcrumbItem({
        ...props,
        elementType: typeof children === 'string' ? 'span' : 'a'
    }, ref);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $6zINN$reactariainteractions.useHover(props);
    let element = /*#__PURE__*/ ($parcel$interopDefault($6zINN$react)).cloneElement($6zINN$reactspectrumutils.getWrappedElement(children), {
        ...$6zINN$reactariautils.mergeProps(itemProps, hoverProps),
        ref: ref,
        className: $6zINN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($7d405af0aae01906$exports)), 'spectrum-Breadcrumbs-itemLink', {
            'is-disabled': !isCurrent && isDisabled,
            'is-hovered': isHovered
        })
    });
    return(/*#__PURE__*/ ($parcel$interopDefault($6zINN$react)).createElement($6zINN$react.Fragment, null, /*#__PURE__*/ ($parcel$interopDefault($6zINN$react)).createElement($6zINN$reactariafocus.FocusRing, {
        focusRingClass: $6zINN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($7d405af0aae01906$exports)), 'focus-ring')
    }, element), isCurrent === false && /*#__PURE__*/ ($parcel$interopDefault($6zINN$react)).createElement(($parcel$interopDefault($6zINN$spectrumiconsuiChevronRightSmall)), {
        UNSAFE_className: $6zINN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($7d405af0aae01906$exports)), 'spectrum-Breadcrumbs-itemSeparator', {
            'is-reversed': direction === 'rtl'
        })
    })));
}











const $ce45029bf0c8e0f0$var$MIN_VISIBLE_ITEMS = 1;
const $ce45029bf0c8e0f0$var$MAX_VISIBLE_ITEMS = 4;
function $ce45029bf0c8e0f0$var$Breadcrumbs(props, ref) {
    props = $6zINN$reactspectrumprovider.useProviderProps(props);
    let { size: size = 'L' , isMultiline: isMultiline , children: children , showRoot: showRoot , isDisabled: isDisabled , onAction: onAction , ...otherProps } = props;
    // Not using React.Children.toArray because it mutates the key prop.
    let childArray = [];
    ($parcel$interopDefault($6zINN$react)).Children.forEach(children, (child)=>{
        if (/*#__PURE__*/ ($parcel$interopDefault($6zINN$react)).isValidElement(child)) childArray.push(child);
    });
    let domRef = $6zINN$reactspectrumutils.useDOMRef(ref);
    let listRef = $6zINN$react.useRef(null);
    let [visibleItems1, setVisibleItems] = $6zINN$reactariautils.useValueEffect(childArray.length);
    let { navProps: navProps  } = $6zINN$reactariabreadcrumbs.useBreadcrumbs(props);
    let { styleProps: styleProps  } = $6zINN$reactspectrumutils.useStyleProps(otherProps);
    let updateOverflow = $6zINN$react.useCallback(()=>{
        let computeVisibleItems = (visibleItems)=>{
            // Refs can be null at runtime.
            let currListRef = listRef.current;
            if (!currListRef) return;
            let listItems = Array.from(currListRef.children);
            let containerWidth = currListRef.offsetWidth;
            let isShowingMenu = childArray.length > visibleItems;
            let calculatedWidth = 0;
            let newVisibleItems = 0;
            let maxVisibleItems = $ce45029bf0c8e0f0$var$MAX_VISIBLE_ITEMS;
            if (showRoot) {
                calculatedWidth += listItems.shift().offsetWidth;
                newVisibleItems++;
            }
            if (isShowingMenu) {
                calculatedWidth += listItems.shift().offsetWidth;
                maxVisibleItems--;
            }
            if (showRoot && calculatedWidth >= containerWidth) newVisibleItems--;
            // TODO: what if multiline and only one breadcrumb??
            if (isMultiline) {
                listItems.pop();
                newVisibleItems++;
            } else {
                // Ensure the last breadcrumb isn't truncated when we measure it.
                let last = listItems.pop();
                last.style.overflow = 'visible';
                calculatedWidth += last.offsetWidth;
                if (calculatedWidth < containerWidth) newVisibleItems++;
                last.style.overflow = '';
            }
            for (let breadcrumb of listItems.reverse()){
                calculatedWidth += breadcrumb.offsetWidth;
                if (calculatedWidth < containerWidth) newVisibleItems++;
            }
            return Math.max($ce45029bf0c8e0f0$var$MIN_VISIBLE_ITEMS, Math.min(maxVisibleItems, newVisibleItems));
        };
        setVisibleItems(function*() {
            // Update to show all items.
            yield childArray.length;
            // Measure, and update to show the items that fit.
            let newVisibleItems = computeVisibleItems(childArray.length);
            yield newVisibleItems;
            // If the number of items is less than the number of children,
            // then update again to ensure that the menu fits.
            if (newVisibleItems < childArray.length && newVisibleItems > 1) yield computeVisibleItems(newVisibleItems);
        });
    }, [
        listRef,
        children,
        setVisibleItems,
        showRoot,
        isMultiline
    ]);
    $6zINN$reactariautils.useResizeObserver({
        ref: domRef,
        onResize: updateOverflow
    });
    $6zINN$reactariautils.useLayoutEffect(updateOverflow, [
        children
    ]);
    let contents = childArray;
    if (childArray.length > visibleItems1) {
        let selectedItem = childArray[childArray.length - 1];
        var _key;
        let selectedKey = (_key = selectedItem.key) !== null && _key !== void 0 ? _key : childArray.length - 1;
        let onMenuAction = (key)=>{
            // Don't fire onAction when clicking on the last item
            if (key !== selectedKey && onAction) onAction(key);
        };
        let menuItem = /*#__PURE__*/ ($parcel$interopDefault($6zINN$react)).createElement($01a6adf6454cb0d0$export$c13f210c706eb549, {
            key: "menu"
        }, /*#__PURE__*/ ($parcel$interopDefault($6zINN$react)).createElement($6zINN$reactspectrummenu.MenuTrigger, null, /*#__PURE__*/ ($parcel$interopDefault($6zINN$react)).createElement($6zINN$reactspectrumbutton.ActionButton, {
            "aria-label": "…",
            isQuiet: true,
            isDisabled: isDisabled
        }, /*#__PURE__*/ ($parcel$interopDefault($6zINN$react)).createElement(($parcel$interopDefault($6zINN$spectrumiconsuiFolderBreadcrumb)), null)), /*#__PURE__*/ ($parcel$interopDefault($6zINN$react)).createElement($6zINN$reactspectrummenu.Menu, {
            selectionMode: "single",
            selectedKeys: [
                selectedKey
            ],
            onAction: onMenuAction
        }, childArray)));
        contents = [
            menuItem
        ];
        let breadcrumbs = [
            ...childArray
        ];
        let endItems = visibleItems1;
        if (showRoot && visibleItems1 > 1) {
            contents.unshift(breadcrumbs.shift());
            endItems--;
        }
        contents.push(...breadcrumbs.slice(-endItems));
    }
    let lastIndex = contents.length - 1;
    let breadcrumbItems = contents.map((child, index)=>{
        let isCurrent = index === lastIndex;
        var _key;
        let key = (_key = child.key) !== null && _key !== void 0 ? _key : index;
        let onPress = ()=>{
            if (onAction) onAction(key);
        };
        return(/*#__PURE__*/ ($parcel$interopDefault($6zINN$react)).createElement("li", {
            key: key,
            className: $6zINN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($7d405af0aae01906$exports)), 'spectrum-Breadcrumbs-item')
        }, /*#__PURE__*/ ($parcel$interopDefault($6zINN$react)).createElement($01a6adf6454cb0d0$export$c13f210c706eb549, {
            isCurrent: isCurrent,
            isDisabled: isDisabled,
            onPress: onPress
        }, child.props.children)));
    });
    return(/*#__PURE__*/ ($parcel$interopDefault($6zINN$react)).createElement("nav", {
        ...styleProps,
        ...navProps,
        ref: domRef
    }, /*#__PURE__*/ ($parcel$interopDefault($6zINN$react)).createElement("ul", {
        ref: listRef,
        className: $6zINN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($7d405af0aae01906$exports)), 'spectrum-Breadcrumbs', {
            'spectrum-Breadcrumbs--small': size === 'S',
            'spectrum-Breadcrumbs--medium': size === 'M',
            'spectrum-Breadcrumbs--multiline': isMultiline,
            'spectrum-Breadcrumbs--showRoot': showRoot,
            'is-disabled': isDisabled
        }, styleProps.className)
    }, breadcrumbItems)));
}
/**
 * Breadcrumbs show hierarchy and navigational context for a user’s location within an application.
 */ let $ce45029bf0c8e0f0$export$2dc68d50d56fbbd = /*#__PURE__*/ ($parcel$interopDefault($6zINN$react)).forwardRef($ce45029bf0c8e0f0$var$Breadcrumbs);



$parcel$exportWildcard(module.exports, $ce45029bf0c8e0f0$exports);
