// @react-spectrum/label/dist/main.js
require("./main.css");
var $eT0Yl$reactspectrumutils = require("@react-spectrum/utils");
var $eT0Yl$reactspectrumlayout = require("@react-spectrum/layout");
var $eT0Yl$reactariautils = require("@react-aria/utils");
var $eT0Yl$react = require("react");
var $eT0Yl$reactspectrumform = require("@react-spectrum/form");
var $eT0Yl$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $eT0Yl$spectrumiconsuiAsterisk = require("@spectrum-icons/ui/Asterisk");
var $eT0Yl$reactariai18n = require("@react-aria/i18n");
var $eT0Yl$reactspectrumprovider = require("@react-spectrum/provider");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $1adbcef7dae5150b$exports = {};

$parcel$export($1adbcef7dae5150b$exports, "Field", () => $1adbcef7dae5150b$export$a455218a85c89869);


var $12367bf3577b934b$exports = {};

$parcel$export($12367bf3577b934b$exports, "HelpText", () => $12367bf3577b934b$export$a67c0bc59081311a);



var $cad320f23e2f8a9c$exports = {};

$parcel$export($cad320f23e2f8a9c$exports, "spectrum-HelpText", () => $cad320f23e2f8a9c$export$c0875e83c57c7736, (v) => $cad320f23e2f8a9c$export$c0875e83c57c7736 = v);
$parcel$export($cad320f23e2f8a9c$exports, "spectrum-HelpText-validationIcon", () => $cad320f23e2f8a9c$export$527776642b333691, (v) => $cad320f23e2f8a9c$export$527776642b333691 = v);
$parcel$export($cad320f23e2f8a9c$exports, "spectrum-HelpText-text", () => $cad320f23e2f8a9c$export$a512452beb4231bf, (v) => $cad320f23e2f8a9c$export$a512452beb4231bf = v);
$parcel$export($cad320f23e2f8a9c$exports, "spectrum-HelpText--neutral", () => $cad320f23e2f8a9c$export$6e9351b1ad13c, (v) => $cad320f23e2f8a9c$export$6e9351b1ad13c = v);
$parcel$export($cad320f23e2f8a9c$exports, "is-disabled", () => $cad320f23e2f8a9c$export$d35bc1e505d1ebbf, (v) => $cad320f23e2f8a9c$export$d35bc1e505d1ebbf = v);
$parcel$export($cad320f23e2f8a9c$exports, "spectrum-HelpText--negative", () => $cad320f23e2f8a9c$export$d31dd875d69daee0, (v) => $cad320f23e2f8a9c$export$d31dd875d69daee0 = v);
var $cad320f23e2f8a9c$export$c0875e83c57c7736;
var $cad320f23e2f8a9c$export$527776642b333691;
var $cad320f23e2f8a9c$export$a512452beb4231bf;
var $cad320f23e2f8a9c$export$6e9351b1ad13c;
var $cad320f23e2f8a9c$export$d35bc1e505d1ebbf;
var $cad320f23e2f8a9c$export$d31dd875d69daee0;
$cad320f23e2f8a9c$export$c0875e83c57c7736 = "spectrum-HelpText_f395de";
$cad320f23e2f8a9c$export$527776642b333691 = "spectrum-HelpText-validationIcon_f395de";
$cad320f23e2f8a9c$export$a512452beb4231bf = "spectrum-HelpText-text_f395de";
$cad320f23e2f8a9c$export$6e9351b1ad13c = "spectrum-HelpText--neutral_f395de";
$cad320f23e2f8a9c$export$d35bc1e505d1ebbf = "is-disabled_f395de";
$cad320f23e2f8a9c$export$d31dd875d69daee0 = "spectrum-HelpText--negative_f395de";


function $12367bf3577b934b$var$HelpText(props, ref) {
    let { description: description , errorMessage: errorMessage , validationState: validationState , isDisabled: isDisabled , showErrorIcon: showErrorIcon , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = props;
    let domRef = $eT0Yl$reactspectrumutils.useDOMRef(ref);
    let isErrorMessage = errorMessage && validationState === 'invalid';
    return(/*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).createElement("div", {
        className: $eT0Yl$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($cad320f23e2f8a9c$exports)), 'spectrum-HelpText', `spectrum-HelpText--${isErrorMessage ? 'negative' : 'neutral'}`, {
            'is-disabled': isDisabled
        }),
        ref: domRef
    }, isErrorMessage ? /*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).createElement(($parcel$interopDefault($eT0Yl$react)).Fragment, null, showErrorIcon && /*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).createElement(($parcel$interopDefault($eT0Yl$spectrumiconsuiAlertMedium)), {
        UNSAFE_className: $eT0Yl$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($cad320f23e2f8a9c$exports)), 'spectrum-HelpText-validationIcon')
    }), /*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).createElement("div", {
        ...errorMessageProps,
        className: $eT0Yl$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($cad320f23e2f8a9c$exports)), 'spectrum-HelpText-text')
    }, errorMessage)) : /*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).createElement("div", {
        ...descriptionProps,
        className: $eT0Yl$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($cad320f23e2f8a9c$exports)), 'spectrum-HelpText-text')
    }, description)));
}
/**
 * Help text provides either an informative description or an error message that gives more context about what a user needs to input. It's commonly used in forms.
 */ const $12367bf3577b934b$export$a67c0bc59081311a = /*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).forwardRef($12367bf3577b934b$var$HelpText);


var $bdf64f6c6923b819$exports = {};

$parcel$export($bdf64f6c6923b819$exports, "Label", () => $bdf64f6c6923b819$export$b04be29aa201d4f5);



var $72494c6c371fadd8$exports = {};
var $af7686f623deed76$exports = {};
$af7686f623deed76$exports = JSON.parse("{\"(optional)\":\"(اختياري)\",\"(required)\":\"(مطلوب)\"}");


var $5397be85c60fe8d8$exports = {};
$5397be85c60fe8d8$exports = JSON.parse("{\"(optional)\":\"(незадължително)\",\"(required)\":\"(задължително)\"}");


var $d8a4155623286f1e$exports = {};
$d8a4155623286f1e$exports = JSON.parse("{\"(optional)\":\"(volitelně)\",\"(required)\":\"(požadováno)\"}");


var $3f78342917481aa4$exports = {};
$3f78342917481aa4$exports = JSON.parse("{\"(optional)\":\"(valgfrit)\",\"(required)\":\"(obligatorisk)\"}");


var $49356611faa2f106$exports = {};
$49356611faa2f106$exports = JSON.parse("{\"(optional)\":\"(optional)\",\"(required)\":\"(erforderlich)\"}");


var $27c7a662312b259c$exports = {};
$27c7a662312b259c$exports = JSON.parse("{\"(optional)\":\"(προαιρετικό)\",\"(required)\":\"(απαιτείται)\"}");


var $c34233a395a0c542$exports = {};
$c34233a395a0c542$exports = JSON.parse("{\"(required)\":\"(required)\",\"(optional)\":\"(optional)\"}");


var $65bc4451773ce40c$exports = {};
$65bc4451773ce40c$exports = JSON.parse("{\"(optional)\":\"(opcional)\",\"(required)\":\"(necesario)\"}");


var $84144ad7b50634c8$exports = {};
$84144ad7b50634c8$exports = JSON.parse("{\"(optional)\":\"(valikuline)\",\"(required)\":\"(nõutav)\"}");


var $6bf934182ee7d2d6$exports = {};
$6bf934182ee7d2d6$exports = JSON.parse("{\"(optional)\":\"(valinnainen)\",\"(required)\":\"(pakollinen)\"}");


var $448b2190a4abf502$exports = {};
$448b2190a4abf502$exports = JSON.parse("{\"(optional)\":\"(facultatif)\",\"(required)\":\"(requis)\"}");


var $293e80b0d2d8203e$exports = {};
$293e80b0d2d8203e$exports = JSON.parse("{\"(optional)\":\"(אופציונלי)\",\"(required)\":\"(נדרש)\"}");


var $f44dff8a12d0ca85$exports = {};
$f44dff8a12d0ca85$exports = JSON.parse("{\"(optional)\":\"(opcionalno)\",\"(required)\":\"(obvezno)\"}");


var $0fc163e9b778059d$exports = {};
$0fc163e9b778059d$exports = JSON.parse("{\"(optional)\":\"(opcionális)\",\"(required)\":\"(kötelező)\"}");


var $103edcdcaf838ee9$exports = {};
$103edcdcaf838ee9$exports = JSON.parse("{\"(optional)\":\"(facoltativo)\",\"(required)\":\"(obbligatorio)\"}");


var $1bc401fd6d96b93c$exports = {};
$1bc401fd6d96b93c$exports = JSON.parse("{\"(optional)\":\"（オプション）\",\"(required)\":\"（必須）\"}");


var $b1839f0277876fc7$exports = {};
$b1839f0277876fc7$exports = JSON.parse("{\"(optional)\":\"(선택 사항)\",\"(required)\":\"(필수 사항)\"}");


var $0f5d2d05589c4186$exports = {};
$0f5d2d05589c4186$exports = JSON.parse("{\"(optional)\":\"(pasirenkama)\",\"(required)\":\"(privaloma)\"}");


var $2d66e6f111e57533$exports = {};
$2d66e6f111e57533$exports = JSON.parse("{\"(optional)\":\"(neobligāti)\",\"(required)\":\"(obligāti)\"}");


var $dcfe08cebd691ea5$exports = {};
$dcfe08cebd691ea5$exports = JSON.parse("{\"(optional)\":\"(valgfritt)\",\"(required)\":\"(obligatorisk)\"}");


var $e5fd40566f6c30cd$exports = {};
$e5fd40566f6c30cd$exports = JSON.parse("{\"(optional)\":\"(optioneel)\",\"(required)\":\"(vereist)\"}");


var $95afbdf107fda73e$exports = {};
$95afbdf107fda73e$exports = JSON.parse("{\"(optional)\":\"(opcjonalne)\",\"(required)\":\"(wymagane)\"}");


var $ff5fc0b870b67f3b$exports = {};
$ff5fc0b870b67f3b$exports = JSON.parse("{\"(optional)\":\"(opcional)\",\"(required)\":\"(obrigatório)\"}");


var $4848a9385066c020$exports = {};
$4848a9385066c020$exports = JSON.parse("{\"(optional)\":\"(opcional)\",\"(required)\":\"(obrigatório)\"}");


var $a422fc242d68aca6$exports = {};
$a422fc242d68aca6$exports = JSON.parse("{\"(optional)\":\"(opţional)\",\"(required)\":\"(obligatoriu)\"}");


var $f8ea3888ce39d033$exports = {};
$f8ea3888ce39d033$exports = JSON.parse("{\"(optional)\":\"(дополнительно)\",\"(required)\":\"(обязательно)\"}");


var $eeaf555d0f138d4c$exports = {};
$eeaf555d0f138d4c$exports = JSON.parse("{\"(optional)\":\"(nepovinné)\",\"(required)\":\"(povinné)\"}");


var $3af7208b025a1d9a$exports = {};
$3af7208b025a1d9a$exports = JSON.parse("{\"(optional)\":\"(opcijsko)\",\"(required)\":\"(obvezno)\"}");


var $24e496759eb78093$exports = {};
$24e496759eb78093$exports = JSON.parse("{\"(optional)\":\"(opciono)\",\"(required)\":\"(obavezno)\"}");


var $d86fa5aae3615712$exports = {};
$d86fa5aae3615712$exports = JSON.parse("{\"(optional)\":\"(valfritt)\",\"(required)\":\"(krävs)\"}");


var $22aa31c7e86b7a27$exports = {};
$22aa31c7e86b7a27$exports = JSON.parse("{\"(optional)\":\"(isteğe bağlı)\",\"(required)\":\"(gerekli)\"}");


var $24ffa7599aac844d$exports = {};
$24ffa7599aac844d$exports = JSON.parse("{\"(optional)\":\"(необов’язково)\",\"(required)\":\"(обов’язково)\"}");


var $64ae5a7173fed017$exports = {};
$64ae5a7173fed017$exports = JSON.parse("{\"(optional)\":\"（可选）\",\"(required)\":\"（必填）\"}");


var $069d9eaa99d29729$exports = {};
$069d9eaa99d29729$exports = JSON.parse("{\"(optional)\":\"(選填)\",\"(required)\":\"(必填)\"}");


$72494c6c371fadd8$exports = {
    "ar-AE": $af7686f623deed76$exports,
    "bg-BG": $5397be85c60fe8d8$exports,
    "cs-CZ": $d8a4155623286f1e$exports,
    "da-DK": $3f78342917481aa4$exports,
    "de-DE": $49356611faa2f106$exports,
    "el-GR": $27c7a662312b259c$exports,
    "en-US": $c34233a395a0c542$exports,
    "es-ES": $65bc4451773ce40c$exports,
    "et-EE": $84144ad7b50634c8$exports,
    "fi-FI": $6bf934182ee7d2d6$exports,
    "fr-FR": $448b2190a4abf502$exports,
    "he-IL": $293e80b0d2d8203e$exports,
    "hr-HR": $f44dff8a12d0ca85$exports,
    "hu-HU": $0fc163e9b778059d$exports,
    "it-IT": $103edcdcaf838ee9$exports,
    "ja-JP": $1bc401fd6d96b93c$exports,
    "ko-KR": $b1839f0277876fc7$exports,
    "lt-LT": $0f5d2d05589c4186$exports,
    "lv-LV": $2d66e6f111e57533$exports,
    "nb-NO": $dcfe08cebd691ea5$exports,
    "nl-NL": $e5fd40566f6c30cd$exports,
    "pl-PL": $95afbdf107fda73e$exports,
    "pt-BR": $ff5fc0b870b67f3b$exports,
    "pt-PT": $4848a9385066c020$exports,
    "ro-RO": $a422fc242d68aca6$exports,
    "ru-RU": $f8ea3888ce39d033$exports,
    "sk-SK": $eeaf555d0f138d4c$exports,
    "sl-SI": $3af7208b025a1d9a$exports,
    "sr-SP": $24e496759eb78093$exports,
    "sv-SE": $d86fa5aae3615712$exports,
    "tr-TR": $22aa31c7e86b7a27$exports,
    "uk-UA": $24ffa7599aac844d$exports,
    "zh-CN": $64ae5a7173fed017$exports,
    "zh-TW": $069d9eaa99d29729$exports
};



var $0043d8d062ef413c$exports = {};

$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel", () => $0043d8d062ef413c$export$493d18e796ae054, (v) => $0043d8d062ef413c$export$493d18e796ae054 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel--positionSide", () => $0043d8d062ef413c$export$9e6f19dc21f22f2e, (v) => $0043d8d062ef413c$export$9e6f19dc21f22f2e = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel-requiredIcon", () => $0043d8d062ef413c$export$7c47642c7d46f3c9, (v) => $0043d8d062ef413c$export$7c47642c7d46f3c9 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel--alignEnd", () => $0043d8d062ef413c$export$885efcc08143a987, (v) => $0043d8d062ef413c$export$885efcc08143a987 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field", () => $0043d8d062ef413c$export$f6d480ae1e56eba0, (v) => $0043d8d062ef413c$export$f6d480ae1e56eba0 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field--positionTop", () => $0043d8d062ef413c$export$a4ea780a9064d7f9, (v) => $0043d8d062ef413c$export$a4ea780a9064d7f9 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field-field", () => $0043d8d062ef413c$export$3ab8a3dc7f0563d, (v) => $0043d8d062ef413c$export$3ab8a3dc7f0563d = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field-field--multiline", () => $0043d8d062ef413c$export$f58a1e966a92ba5e, (v) => $0043d8d062ef413c$export$f58a1e966a92ba5e = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field--positionSide", () => $0043d8d062ef413c$export$2963225c91179589, (v) => $0043d8d062ef413c$export$2963225c91179589 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field-wrapper", () => $0043d8d062ef413c$export$127df0b7290eb8ef, (v) => $0043d8d062ef413c$export$127df0b7290eb8ef = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form", () => $0043d8d062ef413c$export$94d2bcc94afabd89, (v) => $0043d8d062ef413c$export$94d2bcc94afabd89 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form--positionSide", () => $0043d8d062ef413c$export$7f44db659563c8f4, (v) => $0043d8d062ef413c$export$7f44db659563c8f4 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form--positionTop", () => $0043d8d062ef413c$export$11b8bc81fe551057, (v) => $0043d8d062ef413c$export$11b8bc81fe551057 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form-itemLabel", () => $0043d8d062ef413c$export$51ec27881a1e3928, (v) => $0043d8d062ef413c$export$51ec27881a1e3928 = v);
$parcel$export($0043d8d062ef413c$exports, "is-disabled", () => $0043d8d062ef413c$export$d35bc1e505d1ebbf, (v) => $0043d8d062ef413c$export$d35bc1e505d1ebbf = v);
var $0043d8d062ef413c$export$493d18e796ae054;
var $0043d8d062ef413c$export$9e6f19dc21f22f2e;
var $0043d8d062ef413c$export$7c47642c7d46f3c9;
var $0043d8d062ef413c$export$885efcc08143a987;
var $0043d8d062ef413c$export$f6d480ae1e56eba0;
var $0043d8d062ef413c$export$a4ea780a9064d7f9;
var $0043d8d062ef413c$export$3ab8a3dc7f0563d;
var $0043d8d062ef413c$export$f58a1e966a92ba5e;
var $0043d8d062ef413c$export$2963225c91179589;
var $0043d8d062ef413c$export$127df0b7290eb8ef;
var $0043d8d062ef413c$export$94d2bcc94afabd89;
var $0043d8d062ef413c$export$7f44db659563c8f4;
var $0043d8d062ef413c$export$11b8bc81fe551057;
var $0043d8d062ef413c$export$51ec27881a1e3928;
var $0043d8d062ef413c$export$d35bc1e505d1ebbf;
$0043d8d062ef413c$export$493d18e796ae054 = "spectrum-FieldLabel_d2db1f";
$0043d8d062ef413c$export$9e6f19dc21f22f2e = "spectrum-FieldLabel--positionSide_d2db1f";
$0043d8d062ef413c$export$7c47642c7d46f3c9 = "spectrum-FieldLabel-requiredIcon_d2db1f";
$0043d8d062ef413c$export$885efcc08143a987 = "spectrum-FieldLabel--alignEnd_d2db1f";
$0043d8d062ef413c$export$f6d480ae1e56eba0 = "spectrum-Field_d2db1f";
$0043d8d062ef413c$export$a4ea780a9064d7f9 = "spectrum-Field--positionTop_d2db1f";
$0043d8d062ef413c$export$3ab8a3dc7f0563d = "spectrum-Field-field_d2db1f";
$0043d8d062ef413c$export$f58a1e966a92ba5e = "spectrum-Field-field--multiline_d2db1f";
$0043d8d062ef413c$export$2963225c91179589 = "spectrum-Field--positionSide_d2db1f";
$0043d8d062ef413c$export$127df0b7290eb8ef = "spectrum-Field-wrapper_d2db1f";
$0043d8d062ef413c$export$94d2bcc94afabd89 = "spectrum-Form_d2db1f";
$0043d8d062ef413c$export$7f44db659563c8f4 = "spectrum-Form--positionSide_d2db1f";
$0043d8d062ef413c$export$11b8bc81fe551057 = "spectrum-Form--positionTop_d2db1f";
$0043d8d062ef413c$export$51ec27881a1e3928 = "spectrum-Form-itemLabel_d2db1f";
$0043d8d062ef413c$export$d35bc1e505d1ebbf = "is-disabled_d2db1f";




function $bdf64f6c6923b819$var$Label(props, ref) {
    props = $eT0Yl$reactspectrumprovider.useProviderProps(props);
    let { children: children , labelPosition: labelPosition = 'top' , labelAlign: labelAlign = labelPosition === 'side' ? 'start' : null , isRequired: isRequired , necessityIndicator: necessityIndicator = isRequired != null ? 'icon' : null , includeNecessityIndicatorInAccessibilityName: includeNecessityIndicatorInAccessibilityName = false , htmlFor: htmlFor , for: labelFor , elementType: ElementType = 'label' , onClick: onClick , ...otherProps } = props;
    let domRef = $eT0Yl$reactspectrumutils.useDOMRef(ref);
    let { styleProps: styleProps  } = $eT0Yl$reactspectrumutils.useStyleProps(otherProps);
    let formatMessage = $eT0Yl$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($72494c6c371fadd8$exports)));
    let necessityLabel = isRequired ? formatMessage('(required)') : formatMessage('(optional)');
    let icon = /*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).createElement(($parcel$interopDefault($eT0Yl$spectrumiconsuiAsterisk)), {
        UNSAFE_className: $eT0Yl$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($0043d8d062ef413c$exports)), 'spectrum-FieldLabel-requiredIcon'),
        "aria-label": includeNecessityIndicatorInAccessibilityName ? formatMessage('(required)') : undefined
    });
    let labelClassNames = $eT0Yl$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($0043d8d062ef413c$exports)), 'spectrum-FieldLabel', {
        'spectrum-FieldLabel--positionSide': labelPosition === 'side',
        'spectrum-FieldLabel--alignEnd': labelAlign === 'end'
    }, styleProps.className);
    return(/*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).createElement(ElementType, {
        ...$eT0Yl$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        onClick: onClick,
        ref: domRef,
        className: labelClassNames,
        htmlFor: ElementType === 'label' ? labelFor || htmlFor : undefined
    }, children, (necessityIndicator === 'label' || necessityIndicator === 'icon' && isRequired) && ' \u200b', necessityIndicator === 'label' && /*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).createElement("span", {
        "aria-hidden": !includeNecessityIndicatorInAccessibilityName ? isRequired : undefined
    }, necessityLabel), necessityIndicator === 'icon' && isRequired && icon));
}
let $bdf64f6c6923b819$export$b04be29aa201d4f5 = /*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).forwardRef($bdf64f6c6923b819$var$Label);






function $1adbcef7dae5150b$var$Field(props, ref) {
    props = $eT0Yl$reactspectrumform.useFormProps(props);
    let { label: label , labelPosition: labelPosition = 'top' , labelAlign: labelAlign , isRequired: isRequired , necessityIndicator: necessityIndicator , includeNecessityIndicatorInAccessibilityName: includeNecessityIndicatorInAccessibilityName , validationState: validationState , description: description , errorMessage: errorMessage , isDisabled: isDisabled , showErrorIcon: showErrorIcon , children: children , labelProps: labelProps , descriptionProps: // Not every component that uses <Field> supports help text.
    descriptionProps = {
    } , errorMessageProps: errorMessageProps = {
    } , elementType: elementType , wrapperClassName: wrapperClassName , ...otherProps } = props;
    let { styleProps: styleProps  } = $eT0Yl$reactspectrumutils.useStyleProps(otherProps);
    let hasHelpText = !!description || errorMessage && validationState === 'invalid';
    if (label || hasHelpText) {
        let labelWrapperClass = $eT0Yl$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($0043d8d062ef413c$exports)), 'spectrum-Field', {
            'spectrum-Field--positionTop': labelPosition === 'top',
            'spectrum-Field--positionSide': labelPosition === 'side'
        }, styleProps.className, wrapperClassName);
        children = /*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).cloneElement(children, $eT0Yl$reactariautils.mergeProps(children.props, {
            className: $eT0Yl$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($0043d8d062ef413c$exports)), 'spectrum-Field-field')
        }));
        let renderHelpText = ()=>/*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).createElement($12367bf3577b934b$export$a67c0bc59081311a, {
                descriptionProps: descriptionProps,
                errorMessageProps: errorMessageProps,
                description: description,
                errorMessage: errorMessage,
                validationState: validationState,
                isDisabled: isDisabled,
                showErrorIcon: showErrorIcon
            })
        ;
        let renderChildren = ()=>/*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).createElement($eT0Yl$reactspectrumlayout.Flex, {
                direction: "column",
                UNSAFE_className: $eT0Yl$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($0043d8d062ef413c$exports)), 'spectrum-Field-wrapper')
            }, children, hasHelpText && renderHelpText())
        ;
        return(/*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).createElement("div", {
            ...styleProps,
            ref: ref,
            className: labelWrapperClass
        }, label && /*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).createElement($bdf64f6c6923b819$export$b04be29aa201d4f5, {
            ...labelProps,
            labelPosition: labelPosition,
            labelAlign: labelAlign,
            isRequired: isRequired,
            necessityIndicator: necessityIndicator,
            includeNecessityIndicatorInAccessibilityName: includeNecessityIndicatorInAccessibilityName,
            elementType: elementType
        }, label), renderChildren()));
    }
    return(/*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).cloneElement(children, $eT0Yl$reactariautils.mergeProps(children.props, {
        ...styleProps,
        ref: ref
    })));
}
let $1adbcef7dae5150b$export$a455218a85c89869 = /*#__PURE__*/ ($parcel$interopDefault($eT0Yl$react)).forwardRef($1adbcef7dae5150b$var$Field);




$parcel$exportWildcard(module.exports, $1adbcef7dae5150b$exports);
$parcel$exportWildcard(module.exports, $bdf64f6c6923b819$exports);
$parcel$exportWildcard(module.exports, $12367bf3577b934b$exports);
