// @react-spectrum/label/dist/module.js
import "./main.css";
import {useStyleProps as $9yNCC$useStyleProps, classNames as $9yNCC$classNames, useDOMRef as $9yNCC$useDOMRef} from "@react-spectrum/utils";
import {Flex as $9yNCC$Flex} from "@react-spectrum/layout";
import {mergeProps as $9yNCC$mergeProps, filterDOMProps as $9yNCC$filterDOMProps} from "@react-aria/utils";
import $9yNCC$react from "react";
import {useFormProps as $9yNCC$useFormProps} from "@react-spectrum/form";
import $9yNCC$spectrumiconsuiAlertMedium from "@spectrum-icons/ui/AlertMedium";
import $9yNCC$spectrumiconsuiAsterisk from "@spectrum-icons/ui/Asterisk";
import {useMessageFormatter as $9yNCC$useMessageFormatter} from "@react-aria/i18n";
import {useProviderProps as $9yNCC$useProviderProps} from "@react-spectrum/provider";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $6a18e978bf9cef17$exports = {};

$parcel$export($6a18e978bf9cef17$exports, "Field", () => $6a18e978bf9cef17$export$a455218a85c89869);


var $0687fc2f81035995$exports = {};

$parcel$export($0687fc2f81035995$exports, "HelpText", () => $0687fc2f81035995$export$a67c0bc59081311a);



var $9123ddab9d7f6116$exports = {};

$parcel$export($9123ddab9d7f6116$exports, "spectrum-HelpText", () => $9123ddab9d7f6116$export$c0875e83c57c7736, (v) => $9123ddab9d7f6116$export$c0875e83c57c7736 = v);
$parcel$export($9123ddab9d7f6116$exports, "spectrum-HelpText-validationIcon", () => $9123ddab9d7f6116$export$527776642b333691, (v) => $9123ddab9d7f6116$export$527776642b333691 = v);
$parcel$export($9123ddab9d7f6116$exports, "spectrum-HelpText-text", () => $9123ddab9d7f6116$export$a512452beb4231bf, (v) => $9123ddab9d7f6116$export$a512452beb4231bf = v);
$parcel$export($9123ddab9d7f6116$exports, "spectrum-HelpText--neutral", () => $9123ddab9d7f6116$export$6e9351b1ad13c, (v) => $9123ddab9d7f6116$export$6e9351b1ad13c = v);
$parcel$export($9123ddab9d7f6116$exports, "is-disabled", () => $9123ddab9d7f6116$export$d35bc1e505d1ebbf, (v) => $9123ddab9d7f6116$export$d35bc1e505d1ebbf = v);
$parcel$export($9123ddab9d7f6116$exports, "spectrum-HelpText--negative", () => $9123ddab9d7f6116$export$d31dd875d69daee0, (v) => $9123ddab9d7f6116$export$d31dd875d69daee0 = v);
var $9123ddab9d7f6116$export$c0875e83c57c7736;
var $9123ddab9d7f6116$export$527776642b333691;
var $9123ddab9d7f6116$export$a512452beb4231bf;
var $9123ddab9d7f6116$export$6e9351b1ad13c;
var $9123ddab9d7f6116$export$d35bc1e505d1ebbf;
var $9123ddab9d7f6116$export$d31dd875d69daee0;
$9123ddab9d7f6116$export$c0875e83c57c7736 = "spectrum-HelpText_f395de";
$9123ddab9d7f6116$export$527776642b333691 = "spectrum-HelpText-validationIcon_f395de";
$9123ddab9d7f6116$export$a512452beb4231bf = "spectrum-HelpText-text_f395de";
$9123ddab9d7f6116$export$6e9351b1ad13c = "spectrum-HelpText--neutral_f395de";
$9123ddab9d7f6116$export$d35bc1e505d1ebbf = "is-disabled_f395de";
$9123ddab9d7f6116$export$d31dd875d69daee0 = "spectrum-HelpText--negative_f395de";


function $0687fc2f81035995$var$HelpText(props, ref) {
    let { description: description , errorMessage: errorMessage , validationState: validationState , isDisabled: isDisabled , showErrorIcon: showErrorIcon , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = props;
    let domRef = $9yNCC$useDOMRef(ref);
    let isErrorMessage = errorMessage && validationState === 'invalid';
    return(/*#__PURE__*/ $9yNCC$react.createElement("div", {
        className: $9yNCC$classNames((/*@__PURE__*/$parcel$interopDefault($9123ddab9d7f6116$exports)), 'spectrum-HelpText', `spectrum-HelpText--${isErrorMessage ? 'negative' : 'neutral'}`, {
            'is-disabled': isDisabled
        }),
        ref: domRef
    }, isErrorMessage ? /*#__PURE__*/ $9yNCC$react.createElement($9yNCC$react.Fragment, null, showErrorIcon && /*#__PURE__*/ $9yNCC$react.createElement($9yNCC$spectrumiconsuiAlertMedium, {
        UNSAFE_className: $9yNCC$classNames((/*@__PURE__*/$parcel$interopDefault($9123ddab9d7f6116$exports)), 'spectrum-HelpText-validationIcon')
    }), /*#__PURE__*/ $9yNCC$react.createElement("div", {
        ...errorMessageProps,
        className: $9yNCC$classNames((/*@__PURE__*/$parcel$interopDefault($9123ddab9d7f6116$exports)), 'spectrum-HelpText-text')
    }, errorMessage)) : /*#__PURE__*/ $9yNCC$react.createElement("div", {
        ...descriptionProps,
        className: $9yNCC$classNames((/*@__PURE__*/$parcel$interopDefault($9123ddab9d7f6116$exports)), 'spectrum-HelpText-text')
    }, description)));
}
/**
 * Help text provides either an informative description or an error message that gives more context about what a user needs to input. It's commonly used in forms.
 */ const $0687fc2f81035995$export$a67c0bc59081311a = /*#__PURE__*/ $9yNCC$react.forwardRef($0687fc2f81035995$var$HelpText);


var $18838bfc3495f5b6$exports = {};

$parcel$export($18838bfc3495f5b6$exports, "Label", () => $18838bfc3495f5b6$export$b04be29aa201d4f5);



var $943eee8113729748$exports = {};
var $14f966d0de292b65$exports = {};
$14f966d0de292b65$exports = JSON.parse("{\"(optional)\":\"(اختياري)\",\"(required)\":\"(مطلوب)\"}");


var $799d252dfb1bd2be$exports = {};
$799d252dfb1bd2be$exports = JSON.parse("{\"(optional)\":\"(незадължително)\",\"(required)\":\"(задължително)\"}");


var $e3ddbc33af143efa$exports = {};
$e3ddbc33af143efa$exports = JSON.parse("{\"(optional)\":\"(volitelně)\",\"(required)\":\"(požadováno)\"}");


var $3405fab23afb5e06$exports = {};
$3405fab23afb5e06$exports = JSON.parse("{\"(optional)\":\"(valgfrit)\",\"(required)\":\"(obligatorisk)\"}");


var $1f19432f2a803ee3$exports = {};
$1f19432f2a803ee3$exports = JSON.parse("{\"(optional)\":\"(optional)\",\"(required)\":\"(erforderlich)\"}");


var $02bca5cf49f7287d$exports = {};
$02bca5cf49f7287d$exports = JSON.parse("{\"(optional)\":\"(προαιρετικό)\",\"(required)\":\"(απαιτείται)\"}");


var $9a3df3930cb7ef76$exports = {};
$9a3df3930cb7ef76$exports = JSON.parse("{\"(required)\":\"(required)\",\"(optional)\":\"(optional)\"}");


var $bbe80d930a957f9d$exports = {};
$bbe80d930a957f9d$exports = JSON.parse("{\"(optional)\":\"(opcional)\",\"(required)\":\"(necesario)\"}");


var $ced0a79906fe0931$exports = {};
$ced0a79906fe0931$exports = JSON.parse("{\"(optional)\":\"(valikuline)\",\"(required)\":\"(nõutav)\"}");


var $6b5ed60869b96a1d$exports = {};
$6b5ed60869b96a1d$exports = JSON.parse("{\"(optional)\":\"(valinnainen)\",\"(required)\":\"(pakollinen)\"}");


var $54b48da0cb8cc793$exports = {};
$54b48da0cb8cc793$exports = JSON.parse("{\"(optional)\":\"(facultatif)\",\"(required)\":\"(requis)\"}");


var $99c44842d07f3615$exports = {};
$99c44842d07f3615$exports = JSON.parse("{\"(optional)\":\"(אופציונלי)\",\"(required)\":\"(נדרש)\"}");


var $ad02c5bc69864c04$exports = {};
$ad02c5bc69864c04$exports = JSON.parse("{\"(optional)\":\"(opcionalno)\",\"(required)\":\"(obvezno)\"}");


var $9476bbd54d427237$exports = {};
$9476bbd54d427237$exports = JSON.parse("{\"(optional)\":\"(opcionális)\",\"(required)\":\"(kötelező)\"}");


var $f862a0d9d8389065$exports = {};
$f862a0d9d8389065$exports = JSON.parse("{\"(optional)\":\"(facoltativo)\",\"(required)\":\"(obbligatorio)\"}");


var $594c3e1ba5c6c082$exports = {};
$594c3e1ba5c6c082$exports = JSON.parse("{\"(optional)\":\"（オプション）\",\"(required)\":\"（必須）\"}");


var $1d4fbc6c8ac790e1$exports = {};
$1d4fbc6c8ac790e1$exports = JSON.parse("{\"(optional)\":\"(선택 사항)\",\"(required)\":\"(필수 사항)\"}");


var $6e00eec94bcc697d$exports = {};
$6e00eec94bcc697d$exports = JSON.parse("{\"(optional)\":\"(pasirenkama)\",\"(required)\":\"(privaloma)\"}");


var $814c727e3d9836f7$exports = {};
$814c727e3d9836f7$exports = JSON.parse("{\"(optional)\":\"(neobligāti)\",\"(required)\":\"(obligāti)\"}");


var $45ed56b9d96b9500$exports = {};
$45ed56b9d96b9500$exports = JSON.parse("{\"(optional)\":\"(valgfritt)\",\"(required)\":\"(obligatorisk)\"}");


var $6cdb8be5e67e6b11$exports = {};
$6cdb8be5e67e6b11$exports = JSON.parse("{\"(optional)\":\"(optioneel)\",\"(required)\":\"(vereist)\"}");


var $a6b9c5fb565dfd11$exports = {};
$a6b9c5fb565dfd11$exports = JSON.parse("{\"(optional)\":\"(opcjonalne)\",\"(required)\":\"(wymagane)\"}");


var $089d37311709beed$exports = {};
$089d37311709beed$exports = JSON.parse("{\"(optional)\":\"(opcional)\",\"(required)\":\"(obrigatório)\"}");


var $f62f719657ea7fbc$exports = {};
$f62f719657ea7fbc$exports = JSON.parse("{\"(optional)\":\"(opcional)\",\"(required)\":\"(obrigatório)\"}");


var $fea0f2b4442f5e81$exports = {};
$fea0f2b4442f5e81$exports = JSON.parse("{\"(optional)\":\"(opţional)\",\"(required)\":\"(obligatoriu)\"}");


var $c1326e1e4b2b829a$exports = {};
$c1326e1e4b2b829a$exports = JSON.parse("{\"(optional)\":\"(дополнительно)\",\"(required)\":\"(обязательно)\"}");


var $62d94dc6705d6e6d$exports = {};
$62d94dc6705d6e6d$exports = JSON.parse("{\"(optional)\":\"(nepovinné)\",\"(required)\":\"(povinné)\"}");


var $fa8fd958864796dd$exports = {};
$fa8fd958864796dd$exports = JSON.parse("{\"(optional)\":\"(opcijsko)\",\"(required)\":\"(obvezno)\"}");


var $6c98002b668ccee0$exports = {};
$6c98002b668ccee0$exports = JSON.parse("{\"(optional)\":\"(opciono)\",\"(required)\":\"(obavezno)\"}");


var $7afb11ff86e27dc5$exports = {};
$7afb11ff86e27dc5$exports = JSON.parse("{\"(optional)\":\"(valfritt)\",\"(required)\":\"(krävs)\"}");


var $8e63b1fbb80d79fb$exports = {};
$8e63b1fbb80d79fb$exports = JSON.parse("{\"(optional)\":\"(isteğe bağlı)\",\"(required)\":\"(gerekli)\"}");


var $6ed586786e83b45a$exports = {};
$6ed586786e83b45a$exports = JSON.parse("{\"(optional)\":\"(необов’язково)\",\"(required)\":\"(обов’язково)\"}");


var $77e417f6de33263e$exports = {};
$77e417f6de33263e$exports = JSON.parse("{\"(optional)\":\"（可选）\",\"(required)\":\"（必填）\"}");


var $6c347e16cb3d51a8$exports = {};
$6c347e16cb3d51a8$exports = JSON.parse("{\"(optional)\":\"(選填)\",\"(required)\":\"(必填)\"}");


$943eee8113729748$exports = {
    "ar-AE": $14f966d0de292b65$exports,
    "bg-BG": $799d252dfb1bd2be$exports,
    "cs-CZ": $e3ddbc33af143efa$exports,
    "da-DK": $3405fab23afb5e06$exports,
    "de-DE": $1f19432f2a803ee3$exports,
    "el-GR": $02bca5cf49f7287d$exports,
    "en-US": $9a3df3930cb7ef76$exports,
    "es-ES": $bbe80d930a957f9d$exports,
    "et-EE": $ced0a79906fe0931$exports,
    "fi-FI": $6b5ed60869b96a1d$exports,
    "fr-FR": $54b48da0cb8cc793$exports,
    "he-IL": $99c44842d07f3615$exports,
    "hr-HR": $ad02c5bc69864c04$exports,
    "hu-HU": $9476bbd54d427237$exports,
    "it-IT": $f862a0d9d8389065$exports,
    "ja-JP": $594c3e1ba5c6c082$exports,
    "ko-KR": $1d4fbc6c8ac790e1$exports,
    "lt-LT": $6e00eec94bcc697d$exports,
    "lv-LV": $814c727e3d9836f7$exports,
    "nb-NO": $45ed56b9d96b9500$exports,
    "nl-NL": $6cdb8be5e67e6b11$exports,
    "pl-PL": $a6b9c5fb565dfd11$exports,
    "pt-BR": $089d37311709beed$exports,
    "pt-PT": $f62f719657ea7fbc$exports,
    "ro-RO": $fea0f2b4442f5e81$exports,
    "ru-RU": $c1326e1e4b2b829a$exports,
    "sk-SK": $62d94dc6705d6e6d$exports,
    "sl-SI": $fa8fd958864796dd$exports,
    "sr-SP": $6c98002b668ccee0$exports,
    "sv-SE": $7afb11ff86e27dc5$exports,
    "tr-TR": $8e63b1fbb80d79fb$exports,
    "uk-UA": $6ed586786e83b45a$exports,
    "zh-CN": $77e417f6de33263e$exports,
    "zh-TW": $6c347e16cb3d51a8$exports
};



var $4a84444d5a56ab41$exports = {};

$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel", () => $4a84444d5a56ab41$export$493d18e796ae054, (v) => $4a84444d5a56ab41$export$493d18e796ae054 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel--positionSide", () => $4a84444d5a56ab41$export$9e6f19dc21f22f2e, (v) => $4a84444d5a56ab41$export$9e6f19dc21f22f2e = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel-requiredIcon", () => $4a84444d5a56ab41$export$7c47642c7d46f3c9, (v) => $4a84444d5a56ab41$export$7c47642c7d46f3c9 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel--alignEnd", () => $4a84444d5a56ab41$export$885efcc08143a987, (v) => $4a84444d5a56ab41$export$885efcc08143a987 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field", () => $4a84444d5a56ab41$export$f6d480ae1e56eba0, (v) => $4a84444d5a56ab41$export$f6d480ae1e56eba0 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field--positionTop", () => $4a84444d5a56ab41$export$a4ea780a9064d7f9, (v) => $4a84444d5a56ab41$export$a4ea780a9064d7f9 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field-field", () => $4a84444d5a56ab41$export$3ab8a3dc7f0563d, (v) => $4a84444d5a56ab41$export$3ab8a3dc7f0563d = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field-field--multiline", () => $4a84444d5a56ab41$export$f58a1e966a92ba5e, (v) => $4a84444d5a56ab41$export$f58a1e966a92ba5e = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field--positionSide", () => $4a84444d5a56ab41$export$2963225c91179589, (v) => $4a84444d5a56ab41$export$2963225c91179589 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field-wrapper", () => $4a84444d5a56ab41$export$127df0b7290eb8ef, (v) => $4a84444d5a56ab41$export$127df0b7290eb8ef = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form", () => $4a84444d5a56ab41$export$94d2bcc94afabd89, (v) => $4a84444d5a56ab41$export$94d2bcc94afabd89 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form--positionSide", () => $4a84444d5a56ab41$export$7f44db659563c8f4, (v) => $4a84444d5a56ab41$export$7f44db659563c8f4 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form--positionTop", () => $4a84444d5a56ab41$export$11b8bc81fe551057, (v) => $4a84444d5a56ab41$export$11b8bc81fe551057 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form-itemLabel", () => $4a84444d5a56ab41$export$51ec27881a1e3928, (v) => $4a84444d5a56ab41$export$51ec27881a1e3928 = v);
$parcel$export($4a84444d5a56ab41$exports, "is-disabled", () => $4a84444d5a56ab41$export$d35bc1e505d1ebbf, (v) => $4a84444d5a56ab41$export$d35bc1e505d1ebbf = v);
var $4a84444d5a56ab41$export$493d18e796ae054;
var $4a84444d5a56ab41$export$9e6f19dc21f22f2e;
var $4a84444d5a56ab41$export$7c47642c7d46f3c9;
var $4a84444d5a56ab41$export$885efcc08143a987;
var $4a84444d5a56ab41$export$f6d480ae1e56eba0;
var $4a84444d5a56ab41$export$a4ea780a9064d7f9;
var $4a84444d5a56ab41$export$3ab8a3dc7f0563d;
var $4a84444d5a56ab41$export$f58a1e966a92ba5e;
var $4a84444d5a56ab41$export$2963225c91179589;
var $4a84444d5a56ab41$export$127df0b7290eb8ef;
var $4a84444d5a56ab41$export$94d2bcc94afabd89;
var $4a84444d5a56ab41$export$7f44db659563c8f4;
var $4a84444d5a56ab41$export$11b8bc81fe551057;
var $4a84444d5a56ab41$export$51ec27881a1e3928;
var $4a84444d5a56ab41$export$d35bc1e505d1ebbf;
$4a84444d5a56ab41$export$493d18e796ae054 = "spectrum-FieldLabel_d2db1f";
$4a84444d5a56ab41$export$9e6f19dc21f22f2e = "spectrum-FieldLabel--positionSide_d2db1f";
$4a84444d5a56ab41$export$7c47642c7d46f3c9 = "spectrum-FieldLabel-requiredIcon_d2db1f";
$4a84444d5a56ab41$export$885efcc08143a987 = "spectrum-FieldLabel--alignEnd_d2db1f";
$4a84444d5a56ab41$export$f6d480ae1e56eba0 = "spectrum-Field_d2db1f";
$4a84444d5a56ab41$export$a4ea780a9064d7f9 = "spectrum-Field--positionTop_d2db1f";
$4a84444d5a56ab41$export$3ab8a3dc7f0563d = "spectrum-Field-field_d2db1f";
$4a84444d5a56ab41$export$f58a1e966a92ba5e = "spectrum-Field-field--multiline_d2db1f";
$4a84444d5a56ab41$export$2963225c91179589 = "spectrum-Field--positionSide_d2db1f";
$4a84444d5a56ab41$export$127df0b7290eb8ef = "spectrum-Field-wrapper_d2db1f";
$4a84444d5a56ab41$export$94d2bcc94afabd89 = "spectrum-Form_d2db1f";
$4a84444d5a56ab41$export$7f44db659563c8f4 = "spectrum-Form--positionSide_d2db1f";
$4a84444d5a56ab41$export$11b8bc81fe551057 = "spectrum-Form--positionTop_d2db1f";
$4a84444d5a56ab41$export$51ec27881a1e3928 = "spectrum-Form-itemLabel_d2db1f";
$4a84444d5a56ab41$export$d35bc1e505d1ebbf = "is-disabled_d2db1f";




function $18838bfc3495f5b6$var$Label(props, ref) {
    props = $9yNCC$useProviderProps(props);
    let { children: children , labelPosition: labelPosition = 'top' , labelAlign: labelAlign = labelPosition === 'side' ? 'start' : null , isRequired: isRequired , necessityIndicator: necessityIndicator = isRequired != null ? 'icon' : null , includeNecessityIndicatorInAccessibilityName: includeNecessityIndicatorInAccessibilityName = false , htmlFor: htmlFor , for: labelFor , elementType: ElementType = 'label' , onClick: onClick , ...otherProps } = props;
    let domRef = $9yNCC$useDOMRef(ref);
    let { styleProps: styleProps  } = $9yNCC$useStyleProps(otherProps);
    let formatMessage = $9yNCC$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($943eee8113729748$exports)));
    let necessityLabel = isRequired ? formatMessage('(required)') : formatMessage('(optional)');
    let icon = /*#__PURE__*/ $9yNCC$react.createElement($9yNCC$spectrumiconsuiAsterisk, {
        UNSAFE_className: $9yNCC$classNames((/*@__PURE__*/$parcel$interopDefault($4a84444d5a56ab41$exports)), 'spectrum-FieldLabel-requiredIcon'),
        "aria-label": includeNecessityIndicatorInAccessibilityName ? formatMessage('(required)') : undefined
    });
    let labelClassNames = $9yNCC$classNames((/*@__PURE__*/$parcel$interopDefault($4a84444d5a56ab41$exports)), 'spectrum-FieldLabel', {
        'spectrum-FieldLabel--positionSide': labelPosition === 'side',
        'spectrum-FieldLabel--alignEnd': labelAlign === 'end'
    }, styleProps.className);
    return(/*#__PURE__*/ $9yNCC$react.createElement(ElementType, {
        ...$9yNCC$filterDOMProps(otherProps),
        ...styleProps,
        onClick: onClick,
        ref: domRef,
        className: labelClassNames,
        htmlFor: ElementType === 'label' ? labelFor || htmlFor : undefined
    }, children, (necessityIndicator === 'label' || necessityIndicator === 'icon' && isRequired) && ' \u200b', necessityIndicator === 'label' && /*#__PURE__*/ $9yNCC$react.createElement("span", {
        "aria-hidden": !includeNecessityIndicatorInAccessibilityName ? isRequired : undefined
    }, necessityLabel), necessityIndicator === 'icon' && isRequired && icon));
}
let $18838bfc3495f5b6$export$b04be29aa201d4f5 = /*#__PURE__*/ $9yNCC$react.forwardRef($18838bfc3495f5b6$var$Label);






function $6a18e978bf9cef17$var$Field(props, ref) {
    props = $9yNCC$useFormProps(props);
    let { label: label , labelPosition: labelPosition = 'top' , labelAlign: labelAlign , isRequired: isRequired , necessityIndicator: necessityIndicator , includeNecessityIndicatorInAccessibilityName: includeNecessityIndicatorInAccessibilityName , validationState: validationState , description: description , errorMessage: errorMessage , isDisabled: isDisabled , showErrorIcon: showErrorIcon , children: children , labelProps: labelProps , descriptionProps: // Not every component that uses <Field> supports help text.
    descriptionProps = {
    } , errorMessageProps: errorMessageProps = {
    } , elementType: elementType , wrapperClassName: wrapperClassName , ...otherProps } = props;
    let { styleProps: styleProps  } = $9yNCC$useStyleProps(otherProps);
    let hasHelpText = !!description || errorMessage && validationState === 'invalid';
    if (label || hasHelpText) {
        let labelWrapperClass = $9yNCC$classNames((/*@__PURE__*/$parcel$interopDefault($4a84444d5a56ab41$exports)), 'spectrum-Field', {
            'spectrum-Field--positionTop': labelPosition === 'top',
            'spectrum-Field--positionSide': labelPosition === 'side'
        }, styleProps.className, wrapperClassName);
        children = /*#__PURE__*/ $9yNCC$react.cloneElement(children, $9yNCC$mergeProps(children.props, {
            className: $9yNCC$classNames((/*@__PURE__*/$parcel$interopDefault($4a84444d5a56ab41$exports)), 'spectrum-Field-field')
        }));
        let renderHelpText = ()=>/*#__PURE__*/ $9yNCC$react.createElement($0687fc2f81035995$export$a67c0bc59081311a, {
                descriptionProps: descriptionProps,
                errorMessageProps: errorMessageProps,
                description: description,
                errorMessage: errorMessage,
                validationState: validationState,
                isDisabled: isDisabled,
                showErrorIcon: showErrorIcon
            })
        ;
        let renderChildren = ()=>/*#__PURE__*/ $9yNCC$react.createElement($9yNCC$Flex, {
                direction: "column",
                UNSAFE_className: $9yNCC$classNames((/*@__PURE__*/$parcel$interopDefault($4a84444d5a56ab41$exports)), 'spectrum-Field-wrapper')
            }, children, hasHelpText && renderHelpText())
        ;
        return(/*#__PURE__*/ $9yNCC$react.createElement("div", {
            ...styleProps,
            ref: ref,
            className: labelWrapperClass
        }, label && /*#__PURE__*/ $9yNCC$react.createElement($18838bfc3495f5b6$export$b04be29aa201d4f5, {
            ...labelProps,
            labelPosition: labelPosition,
            labelAlign: labelAlign,
            isRequired: isRequired,
            necessityIndicator: necessityIndicator,
            includeNecessityIndicatorInAccessibilityName: includeNecessityIndicatorInAccessibilityName,
            elementType: elementType
        }, label), renderChildren()));
    }
    return(/*#__PURE__*/ $9yNCC$react.cloneElement(children, $9yNCC$mergeProps(children.props, {
        ...styleProps,
        ref: ref
    })));
}
let $6a18e978bf9cef17$export$a455218a85c89869 = /*#__PURE__*/ $9yNCC$react.forwardRef($6a18e978bf9cef17$var$Field);






export {$6a18e978bf9cef17$export$a455218a85c89869 as Field, $18838bfc3495f5b6$export$b04be29aa201d4f5 as Label, $0687fc2f81035995$export$a67c0bc59081311a as HelpText};
