// @react-spectrum/searchfield/dist/main.js
require("./main.css");
var $1bmMG$reactspectrumutils = require("@react-spectrum/utils");
var $1bmMG$reactspectrumbutton = require("@react-spectrum/button");
var $1bmMG$spectrumiconsuiMagnifier = require("@spectrum-icons/ui/Magnifier");
var $1bmMG$react = require("react");
var $1bmMG$reactspectrumtextfield = require("@react-spectrum/textfield");
var $1bmMG$reactspectrumprovider = require("@react-spectrum/provider");
var $1bmMG$reactariasearchfield = require("@react-aria/searchfield");
var $1bmMG$reactstatelysearchfield = require("@react-stately/searchfield");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $f3bd5c9e824ee5fa$exports = {};

$parcel$export($f3bd5c9e824ee5fa$exports, "SearchField", () => $f3bd5c9e824ee5fa$export$b94867ecbd698f21);




var $7cdb83ac3ccb92ab$exports = {};

$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Search", () => $7cdb83ac3ccb92ab$export$713911d754e1e4ae, (v) => $7cdb83ac3ccb92ab$export$713911d754e1e4ae = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-ClearButton", () => $7cdb83ac3ccb92ab$export$b752ce409e5660c, (v) => $7cdb83ac3ccb92ab$export$b752ce409e5660c = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "is-quiet", () => $7cdb83ac3ccb92ab$export$6d5c5433f0da4241, (v) => $7cdb83ac3ccb92ab$export$6d5c5433f0da4241 = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Search-input", () => $7cdb83ac3ccb92ab$export$50e02603f2d97ae1, (v) => $7cdb83ac3ccb92ab$export$50e02603f2d97ae1 = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Search--invalid", () => $7cdb83ac3ccb92ab$export$77a6151a0b46a6a3, (v) => $7cdb83ac3ccb92ab$export$77a6151a0b46a6a3 = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Search--valid", () => $7cdb83ac3ccb92ab$export$aa81e3ab40f37567, (v) => $7cdb83ac3ccb92ab$export$aa81e3ab40f37567 = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Search--loadable", () => $7cdb83ac3ccb92ab$export$9c74d64afae3b2d4, (v) => $7cdb83ac3ccb92ab$export$9c74d64afae3b2d4 = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Search-circleLoader", () => $7cdb83ac3ccb92ab$export$70ae4353602451dc, (v) => $7cdb83ac3ccb92ab$export$70ae4353602451dc = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Search-validationIcon", () => $7cdb83ac3ccb92ab$export$938f0981825eae6e, (v) => $7cdb83ac3ccb92ab$export$938f0981825eae6e = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Textfield", () => $7cdb83ac3ccb92ab$export$2c8a10299d8b3418, (v) => $7cdb83ac3ccb92ab$export$2c8a10299d8b3418 = v);
var $7cdb83ac3ccb92ab$export$713911d754e1e4ae;
var $7cdb83ac3ccb92ab$export$b752ce409e5660c;
var $7cdb83ac3ccb92ab$export$6d5c5433f0da4241;
var $7cdb83ac3ccb92ab$export$50e02603f2d97ae1;
var $7cdb83ac3ccb92ab$export$77a6151a0b46a6a3;
var $7cdb83ac3ccb92ab$export$aa81e3ab40f37567;
var $7cdb83ac3ccb92ab$export$9c74d64afae3b2d4;
var $7cdb83ac3ccb92ab$export$70ae4353602451dc;
var $7cdb83ac3ccb92ab$export$938f0981825eae6e;
var $7cdb83ac3ccb92ab$export$2c8a10299d8b3418;
$7cdb83ac3ccb92ab$export$713911d754e1e4ae = "spectrum-Search_d3c0f3";
$7cdb83ac3ccb92ab$export$b752ce409e5660c = "spectrum-ClearButton_d3c0f3";
$7cdb83ac3ccb92ab$export$6d5c5433f0da4241 = "is-quiet_d3c0f3";
$7cdb83ac3ccb92ab$export$50e02603f2d97ae1 = "spectrum-Search-input_d3c0f3";
$7cdb83ac3ccb92ab$export$77a6151a0b46a6a3 = "spectrum-Search--invalid_d3c0f3";
$7cdb83ac3ccb92ab$export$aa81e3ab40f37567 = "spectrum-Search--valid_d3c0f3";
$7cdb83ac3ccb92ab$export$9c74d64afae3b2d4 = "spectrum-Search--loadable_d3c0f3";
$7cdb83ac3ccb92ab$export$70ae4353602451dc = "spectrum-Search-circleLoader_d3c0f3";
$7cdb83ac3ccb92ab$export$938f0981825eae6e = "spectrum-Search-validationIcon_d3c0f3";
$7cdb83ac3ccb92ab$export$2c8a10299d8b3418 = "spectrum-Textfield_d3c0f3";






function $f3bd5c9e824ee5fa$var$SearchField(props, ref) {
    props = $1bmMG$reactspectrumutils.useSlotProps(props, 'searchfield');
    props = $1bmMG$reactspectrumprovider.useProviderProps(props);
    let defaultIcon = /*#__PURE__*/ ($parcel$interopDefault($1bmMG$react)).createElement(($parcel$interopDefault($1bmMG$spectrumiconsuiMagnifier)), {
        "data-testid": "searchicon"
    });
    let { icon: icon = defaultIcon , isDisabled: isDisabled , UNSAFE_className: UNSAFE_className , ...otherProps } = props;
    let state = $1bmMG$reactstatelysearchfield.useSearchFieldState(props);
    let inputRef = $1bmMG$react.useRef();
    let { labelProps: labelProps , inputProps: inputProps , clearButtonProps: clearButtonProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $1bmMG$reactariasearchfield.useSearchField(props, state, inputRef);
    let clearButton = /*#__PURE__*/ ($parcel$interopDefault($1bmMG$react)).createElement($1bmMG$reactspectrumbutton.ClearButton, {
        ...clearButtonProps,
        preventFocus: true,
        UNSAFE_className: $1bmMG$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($7cdb83ac3ccb92ab$exports)), 'spectrum-ClearButton'),
        isDisabled: isDisabled
    });
    return(/*#__PURE__*/ ($parcel$interopDefault($1bmMG$react)).createElement($1bmMG$reactspectrumtextfield.TextFieldBase, {
        ...otherProps,
        labelProps: labelProps,
        inputProps: inputProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        UNSAFE_className: $1bmMG$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($7cdb83ac3ccb92ab$exports)), 'spectrum-Search', 'spectrum-Textfield', {
            'is-disabled': isDisabled,
            'is-quiet': props.isQuiet,
            'spectrum-Search--invalid': props.validationState === 'invalid',
            'spectrum-Search--valid': props.validationState === 'valid'
        }, UNSAFE_className),
        inputClassName: $1bmMG$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($7cdb83ac3ccb92ab$exports)), 'spectrum-Search-input'),
        ref: ref,
        inputRef: inputRef,
        isDisabled: isDisabled,
        icon: icon,
        wrapperChildren: state.value !== '' && !props.isReadOnly && clearButton
    }));
}
/**
 * A SearchField is a text field designed for searches.
 */ let $f3bd5c9e824ee5fa$export$b94867ecbd698f21 = /*#__PURE__*/ $1bmMG$react.forwardRef($f3bd5c9e824ee5fa$var$SearchField);


$parcel$exportWildcard(module.exports, $f3bd5c9e824ee5fa$exports);
