// @react-spectrum/searchfield/dist/module.js
import "./main.css";
import {useSlotProps as $5fE9B$useSlotProps, classNames as $5fE9B$classNames} from "@react-spectrum/utils";
import {ClearButton as $5fE9B$ClearButton} from "@react-spectrum/button";
import $5fE9B$spectrumiconsuiMagnifier from "@spectrum-icons/ui/Magnifier";
import $5fE9B$react, {useRef as $5fE9B$useRef, forwardRef as $5fE9B$forwardRef} from "react";
import {TextFieldBase as $5fE9B$TextFieldBase} from "@react-spectrum/textfield";
import {useProviderProps as $5fE9B$useProviderProps} from "@react-spectrum/provider";
import {useSearchField as $5fE9B$useSearchField} from "@react-aria/searchfield";
import {useSearchFieldState as $5fE9B$useSearchFieldState} from "@react-stately/searchfield";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $3538a3933fb2b56d$exports = {};

$parcel$export($3538a3933fb2b56d$exports, "SearchField", () => $3538a3933fb2b56d$export$b94867ecbd698f21);




var $fe88933789dbeaab$exports = {};

$parcel$export($fe88933789dbeaab$exports, "spectrum-Search", () => $fe88933789dbeaab$export$713911d754e1e4ae, (v) => $fe88933789dbeaab$export$713911d754e1e4ae = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-ClearButton", () => $fe88933789dbeaab$export$b752ce409e5660c, (v) => $fe88933789dbeaab$export$b752ce409e5660c = v);
$parcel$export($fe88933789dbeaab$exports, "is-quiet", () => $fe88933789dbeaab$export$6d5c5433f0da4241, (v) => $fe88933789dbeaab$export$6d5c5433f0da4241 = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-Search-input", () => $fe88933789dbeaab$export$50e02603f2d97ae1, (v) => $fe88933789dbeaab$export$50e02603f2d97ae1 = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-Search--invalid", () => $fe88933789dbeaab$export$77a6151a0b46a6a3, (v) => $fe88933789dbeaab$export$77a6151a0b46a6a3 = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-Search--valid", () => $fe88933789dbeaab$export$aa81e3ab40f37567, (v) => $fe88933789dbeaab$export$aa81e3ab40f37567 = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-Search--loadable", () => $fe88933789dbeaab$export$9c74d64afae3b2d4, (v) => $fe88933789dbeaab$export$9c74d64afae3b2d4 = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-Search-circleLoader", () => $fe88933789dbeaab$export$70ae4353602451dc, (v) => $fe88933789dbeaab$export$70ae4353602451dc = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-Search-validationIcon", () => $fe88933789dbeaab$export$938f0981825eae6e, (v) => $fe88933789dbeaab$export$938f0981825eae6e = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-Textfield", () => $fe88933789dbeaab$export$2c8a10299d8b3418, (v) => $fe88933789dbeaab$export$2c8a10299d8b3418 = v);
var $fe88933789dbeaab$export$713911d754e1e4ae;
var $fe88933789dbeaab$export$b752ce409e5660c;
var $fe88933789dbeaab$export$6d5c5433f0da4241;
var $fe88933789dbeaab$export$50e02603f2d97ae1;
var $fe88933789dbeaab$export$77a6151a0b46a6a3;
var $fe88933789dbeaab$export$aa81e3ab40f37567;
var $fe88933789dbeaab$export$9c74d64afae3b2d4;
var $fe88933789dbeaab$export$70ae4353602451dc;
var $fe88933789dbeaab$export$938f0981825eae6e;
var $fe88933789dbeaab$export$2c8a10299d8b3418;
$fe88933789dbeaab$export$713911d754e1e4ae = "spectrum-Search_d3c0f3";
$fe88933789dbeaab$export$b752ce409e5660c = "spectrum-ClearButton_d3c0f3";
$fe88933789dbeaab$export$6d5c5433f0da4241 = "is-quiet_d3c0f3";
$fe88933789dbeaab$export$50e02603f2d97ae1 = "spectrum-Search-input_d3c0f3";
$fe88933789dbeaab$export$77a6151a0b46a6a3 = "spectrum-Search--invalid_d3c0f3";
$fe88933789dbeaab$export$aa81e3ab40f37567 = "spectrum-Search--valid_d3c0f3";
$fe88933789dbeaab$export$9c74d64afae3b2d4 = "spectrum-Search--loadable_d3c0f3";
$fe88933789dbeaab$export$70ae4353602451dc = "spectrum-Search-circleLoader_d3c0f3";
$fe88933789dbeaab$export$938f0981825eae6e = "spectrum-Search-validationIcon_d3c0f3";
$fe88933789dbeaab$export$2c8a10299d8b3418 = "spectrum-Textfield_d3c0f3";






function $3538a3933fb2b56d$var$SearchField(props, ref) {
    props = $5fE9B$useSlotProps(props, 'searchfield');
    props = $5fE9B$useProviderProps(props);
    let defaultIcon = /*#__PURE__*/ $5fE9B$react.createElement($5fE9B$spectrumiconsuiMagnifier, {
        "data-testid": "searchicon"
    });
    let { icon: icon = defaultIcon , isDisabled: isDisabled , UNSAFE_className: UNSAFE_className , ...otherProps } = props;
    let state = $5fE9B$useSearchFieldState(props);
    let inputRef = $5fE9B$useRef();
    let { labelProps: labelProps , inputProps: inputProps , clearButtonProps: clearButtonProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $5fE9B$useSearchField(props, state, inputRef);
    let clearButton = /*#__PURE__*/ $5fE9B$react.createElement($5fE9B$ClearButton, {
        ...clearButtonProps,
        preventFocus: true,
        UNSAFE_className: $5fE9B$classNames((/*@__PURE__*/$parcel$interopDefault($fe88933789dbeaab$exports)), 'spectrum-ClearButton'),
        isDisabled: isDisabled
    });
    return(/*#__PURE__*/ $5fE9B$react.createElement($5fE9B$TextFieldBase, {
        ...otherProps,
        labelProps: labelProps,
        inputProps: inputProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        UNSAFE_className: $5fE9B$classNames((/*@__PURE__*/$parcel$interopDefault($fe88933789dbeaab$exports)), 'spectrum-Search', 'spectrum-Textfield', {
            'is-disabled': isDisabled,
            'is-quiet': props.isQuiet,
            'spectrum-Search--invalid': props.validationState === 'invalid',
            'spectrum-Search--valid': props.validationState === 'valid'
        }, UNSAFE_className),
        inputClassName: $5fE9B$classNames((/*@__PURE__*/$parcel$interopDefault($fe88933789dbeaab$exports)), 'spectrum-Search-input'),
        ref: ref,
        inputRef: inputRef,
        isDisabled: isDisabled,
        icon: icon,
        wrapperChildren: state.value !== '' && !props.isReadOnly && clearButton
    }));
}
/**
 * A SearchField is a text field designed for searches.
 */ let $3538a3933fb2b56d$export$b94867ecbd698f21 = /*#__PURE__*/ $5fE9B$forwardRef($3538a3933fb2b56d$var$SearchField);




export {$3538a3933fb2b56d$export$b94867ecbd698f21 as SearchField};
