// @react-spectrum/utils/dist/main.js
var $lNqMH$clsx = require("clsx");
var $lNqMH$react = require("react");
var $lNqMH$reactariassr = require("@react-aria/ssr");
var $lNqMH$reactariai18n = require("@react-aria/i18n");
var $lNqMH$reactariautils = require("@react-aria/utils");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $ee2244f86238f046$exports = {};

$parcel$export($ee2244f86238f046$exports, "shouldKeepSpectrumClassNames", () => $ee2244f86238f046$export$46d604dce8bf8724);
$parcel$export($ee2244f86238f046$exports, "keepSpectrumClassNames", () => $ee2244f86238f046$export$f9d3bfd10703eb31);
$parcel$export($ee2244f86238f046$exports, "classNames", () => $ee2244f86238f046$export$ce4ab0c55987d1ff);

let $ee2244f86238f046$export$46d604dce8bf8724 = false;
function $ee2244f86238f046$export$f9d3bfd10703eb31() {
    $ee2244f86238f046$export$46d604dce8bf8724 = true;
    console.warn("Legacy spectrum-prefixed class names enabled for backward compatibility. We recommend replacing instances of CSS overrides targeting spectrum selectors in your app with custom class names of your own, and disabling this flag.");
}
function $ee2244f86238f046$export$ce4ab0c55987d1ff(cssModule, ...values) {
    let classes = [];
    for (let value of values){
        if (typeof value === 'object' && value) {
            let mapped = {
            };
            for(let key in value){
                if (cssModule[key]) mapped[cssModule[key]] = value[key];
                if ($ee2244f86238f046$export$46d604dce8bf8724 || !cssModule[key]) mapped[key] = value[key];
            }
            classes.push(mapped);
        } else if (typeof value === 'string') {
            if (cssModule[value]) classes.push(cssModule[value]);
            if ($ee2244f86238f046$export$46d604dce8bf8724 || !cssModule[value]) classes.push(value);
        } else classes.push(value);
    }
    return ($parcel$interopDefault($lNqMH$clsx))(...classes);
}


var $70e73537f8916bcc$exports = {};

$parcel$export($70e73537f8916bcc$exports, "getWrappedElement", () => $70e73537f8916bcc$export$a5f5a6912b18861c);

function $70e73537f8916bcc$export$a5f5a6912b18861c(children) {
    let element;
    if (typeof children === 'string') element = /*#__PURE__*/ ($parcel$interopDefault($lNqMH$react)).createElement("span", null, children);
    else element = ($parcel$interopDefault($lNqMH$react)).Children.only(children);
    return element;
}


var $1b6371e033fc95ef$exports = {};

$parcel$export($1b6371e033fc95ef$exports, "useMediaQuery", () => $1b6371e033fc95ef$export$32d5543ab307c01);


function $1b6371e033fc95ef$export$32d5543ab307c01(query) {
    let supportsMatchMedia = typeof window !== 'undefined' && typeof window.matchMedia === 'function';
    let [matches, setMatches] = $lNqMH$react.useState(()=>supportsMatchMedia ? window.matchMedia(query).matches : false
    );
    $lNqMH$react.useEffect(()=>{
        if (!supportsMatchMedia) return;
        let mq = window.matchMedia(query);
        let onChange = (evt)=>{
            setMatches(evt.matches);
        };
        mq.addListener(onChange);
        return ()=>{
            mq.removeListener(onChange);
        };
    }, [
        supportsMatchMedia,
        query
    ]);
    // If in SSR, the media query should never match. Once the page hydrates,
    // this will update and the real value will be returned.
    let isSSR = $lNqMH$reactariassr.useIsSSR();
    return isSSR ? false : matches;
}


var $c62409ad469b3817$exports = {};

$parcel$export($c62409ad469b3817$exports, "createDOMRef", () => $c62409ad469b3817$export$a5795cc979dfae80);
$parcel$export($c62409ad469b3817$exports, "createFocusableRef", () => $c62409ad469b3817$export$79d69eee6ae4b329);
$parcel$export($c62409ad469b3817$exports, "useDOMRef", () => $c62409ad469b3817$export$c2c55ef9111cafd8);
$parcel$export($c62409ad469b3817$exports, "useFocusableRef", () => $c62409ad469b3817$export$96a734597687c040);
$parcel$export($c62409ad469b3817$exports, "unwrapDOMRef", () => $c62409ad469b3817$export$c7e28c72a4823176);
$parcel$export($c62409ad469b3817$exports, "useUnwrapDOMRef", () => $c62409ad469b3817$export$1d5cc31d9d8df817);

function $c62409ad469b3817$export$a5795cc979dfae80(ref) {
    return {
        UNSAFE_getDOMNode () {
            return ref.current;
        }
    };
}
function $c62409ad469b3817$export$79d69eee6ae4b329(domRef, focusableRef = domRef) {
    return {
        ...$c62409ad469b3817$export$a5795cc979dfae80(domRef),
        focus () {
            if (focusableRef.current) focusableRef.current.focus();
        }
    };
}
function $c62409ad469b3817$export$c2c55ef9111cafd8(ref) {
    let domRef = $lNqMH$react.useRef(null);
    $lNqMH$react.useImperativeHandle(ref, ()=>$c62409ad469b3817$export$a5795cc979dfae80(domRef)
    );
    return domRef;
}
function $c62409ad469b3817$export$96a734597687c040(ref, focusableRef) {
    let domRef = $lNqMH$react.useRef(null);
    $lNqMH$react.useImperativeHandle(ref, ()=>$c62409ad469b3817$export$79d69eee6ae4b329(domRef, focusableRef)
    );
    return domRef;
}
function $c62409ad469b3817$export$c7e28c72a4823176(ref) {
    return {
        get current () {
            return ref.current && ref.current.UNSAFE_getDOMNode();
        }
    };
}
function $c62409ad469b3817$export$1d5cc31d9d8df817(ref) {
    return $lNqMH$react.useMemo(()=>$c62409ad469b3817$export$c7e28c72a4823176(ref)
    , [
        ref
    ]);
}


var $23d6bcdf1d0b93b7$exports = {};

$parcel$export($23d6bcdf1d0b93b7$exports, "baseStyleProps", () => $23d6bcdf1d0b93b7$export$fe9c6e915565b4e8);
$parcel$export($23d6bcdf1d0b93b7$exports, "dimensionValue", () => $23d6bcdf1d0b93b7$export$abc24f5b99744ea6);
$parcel$export($23d6bcdf1d0b93b7$exports, "passthroughStyle", () => $23d6bcdf1d0b93b7$export$46b6c81d11d2c30a);
$parcel$export($23d6bcdf1d0b93b7$exports, "viewStyleProps", () => $23d6bcdf1d0b93b7$export$e0705d1a55f297c);
$parcel$export($23d6bcdf1d0b93b7$exports, "responsiveDimensionValue", () => $23d6bcdf1d0b93b7$export$f348bec194f2e6b5);
$parcel$export($23d6bcdf1d0b93b7$exports, "getResponsiveProp", () => $23d6bcdf1d0b93b7$export$52dbfdbe1b2c3541);
$parcel$export($23d6bcdf1d0b93b7$exports, "convertStyleProps", () => $23d6bcdf1d0b93b7$export$f3c39bb9534218d0);
$parcel$export($23d6bcdf1d0b93b7$exports, "useStyleProps", () => $23d6bcdf1d0b93b7$export$b8e6fb9d2dff3f41);
var $f2846dd3c9067392$exports = {};

$parcel$export($f2846dd3c9067392$exports, "BreakpointProvider", () => $f2846dd3c9067392$export$8214320346cf5104);
$parcel$export($f2846dd3c9067392$exports, "useMatchedBreakpoints", () => $f2846dd3c9067392$export$140ae7baa51cca23);
$parcel$export($f2846dd3c9067392$exports, "useBreakpoint", () => $f2846dd3c9067392$export$199d6754bdf4e1e3);


const $f2846dd3c9067392$var$Context = /*#__PURE__*/ ($parcel$interopDefault($lNqMH$react)).createContext(null);
$f2846dd3c9067392$var$Context.displayName = 'BreakpointContext';
function $f2846dd3c9067392$export$8214320346cf5104(props) {
    let { children: children , matchedBreakpoints: matchedBreakpoints  } = props;
    return(/*#__PURE__*/ ($parcel$interopDefault($lNqMH$react)).createElement($f2846dd3c9067392$var$Context.Provider, {
        value: {
            matchedBreakpoints: matchedBreakpoints
        }
    }, children));
}
function $f2846dd3c9067392$export$140ae7baa51cca23(breakpoints) {
    let entries = Object.entries(breakpoints).sort(([, valueA], [, valueB])=>valueB - valueA
    );
    let breakpointQueries = entries.map(([, value])=>`(min-width: ${value}px)`
    );
    let supportsMatchMedia = typeof window !== 'undefined' && typeof window.matchMedia === 'function';
    let getBreakpointHandler = ()=>{
        let matched = [];
        for(let i in breakpointQueries){
            let query = breakpointQueries[i];
            if (window.matchMedia(query).matches) matched.push(entries[i][0]);
        }
        matched.push('base');
        return matched;
    };
    let [breakpoint1, setBreakpoint] = $lNqMH$react.useState(()=>supportsMatchMedia ? getBreakpointHandler() : [
            'base'
        ]
    );
    $lNqMH$react.useEffect(()=>{
        if (!supportsMatchMedia) return;
        let onResize = ()=>{
            const breakpointHandler = getBreakpointHandler();
            setBreakpoint((previousBreakpointHandler)=>{
                if (previousBreakpointHandler.length !== breakpointHandler.length || previousBreakpointHandler.some((breakpoint, idx)=>breakpoint !== breakpointHandler[idx]
                )) return [
                    ...breakpointHandler
                ]; // Return a new array to force state change
                return previousBreakpointHandler;
            });
        };
        window.addEventListener('resize', onResize);
        return ()=>{
            window.removeEventListener('resize', onResize);
        };
    }, [
        supportsMatchMedia
    ]);
    // If in SSR, the media query should never match. Once the page hydrates,
    // this will update and the real value will be returned.
    let isSSR = $lNqMH$reactariassr.useIsSSR();
    return isSSR ? [
        'base'
    ] : breakpoint1;
}
function $f2846dd3c9067392$export$199d6754bdf4e1e3() {
    return $lNqMH$react.useContext($f2846dd3c9067392$var$Context);
}



const $23d6bcdf1d0b93b7$export$fe9c6e915565b4e8 = {
    margin: [
        'margin',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    marginStart: [
        $23d6bcdf1d0b93b7$var$rtl('marginLeft', 'marginRight'),
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    marginEnd: [
        $23d6bcdf1d0b93b7$var$rtl('marginRight', 'marginLeft'),
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    // marginLeft: ['marginLeft', dimensionValue],
    // marginRight: ['marginRight', dimensionValue],
    marginTop: [
        'marginTop',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    marginBottom: [
        'marginBottom',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    marginX: [
        [
            'marginLeft',
            'marginRight'
        ],
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    marginY: [
        [
            'marginTop',
            'marginBottom'
        ],
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    width: [
        'width',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    height: [
        'height',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    minWidth: [
        'minWidth',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    minHeight: [
        'minHeight',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    maxWidth: [
        'maxWidth',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    maxHeight: [
        'maxHeight',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    isHidden: [
        'display',
        $23d6bcdf1d0b93b7$var$hiddenValue
    ],
    alignSelf: [
        'alignSelf',
        $23d6bcdf1d0b93b7$export$46b6c81d11d2c30a
    ],
    justifySelf: [
        'justifySelf',
        $23d6bcdf1d0b93b7$export$46b6c81d11d2c30a
    ],
    position: [
        'position',
        $23d6bcdf1d0b93b7$var$anyValue
    ],
    zIndex: [
        'zIndex',
        $23d6bcdf1d0b93b7$var$anyValue
    ],
    top: [
        'top',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    bottom: [
        'bottom',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    start: [
        $23d6bcdf1d0b93b7$var$rtl('left', 'right'),
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    end: [
        $23d6bcdf1d0b93b7$var$rtl('right', 'left'),
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    left: [
        'left',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    right: [
        'right',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    order: [
        'order',
        $23d6bcdf1d0b93b7$var$anyValue
    ],
    flex: [
        'flex',
        $23d6bcdf1d0b93b7$var$flexValue
    ],
    flexGrow: [
        'flexGrow',
        $23d6bcdf1d0b93b7$export$46b6c81d11d2c30a
    ],
    flexShrink: [
        'flexShrink',
        $23d6bcdf1d0b93b7$export$46b6c81d11d2c30a
    ],
    flexBasis: [
        'flexBasis',
        $23d6bcdf1d0b93b7$export$46b6c81d11d2c30a
    ],
    gridArea: [
        'gridArea',
        $23d6bcdf1d0b93b7$export$46b6c81d11d2c30a
    ],
    gridColumn: [
        'gridColumn',
        $23d6bcdf1d0b93b7$export$46b6c81d11d2c30a
    ],
    gridColumnEnd: [
        'gridColumnEnd',
        $23d6bcdf1d0b93b7$export$46b6c81d11d2c30a
    ],
    gridColumnStart: [
        'gridColumnStart',
        $23d6bcdf1d0b93b7$export$46b6c81d11d2c30a
    ],
    gridRow: [
        'gridRow',
        $23d6bcdf1d0b93b7$export$46b6c81d11d2c30a
    ],
    gridRowEnd: [
        'gridRowEnd',
        $23d6bcdf1d0b93b7$export$46b6c81d11d2c30a
    ],
    gridRowStart: [
        'gridRowStart',
        $23d6bcdf1d0b93b7$export$46b6c81d11d2c30a
    ]
};
const $23d6bcdf1d0b93b7$export$e0705d1a55f297c = {
    ...$23d6bcdf1d0b93b7$export$fe9c6e915565b4e8,
    backgroundColor: [
        'backgroundColor',
        $23d6bcdf1d0b93b7$var$backgroundColorValue
    ],
    borderWidth: [
        'borderWidth',
        $23d6bcdf1d0b93b7$var$borderSizeValue
    ],
    borderStartWidth: [
        $23d6bcdf1d0b93b7$var$rtl('borderLeftWidth', 'borderRightWidth'),
        $23d6bcdf1d0b93b7$var$borderSizeValue
    ],
    borderEndWidth: [
        $23d6bcdf1d0b93b7$var$rtl('borderRightWidth', 'borderLeftWidth'),
        $23d6bcdf1d0b93b7$var$borderSizeValue
    ],
    borderLeftWidth: [
        'borderLeftWidth',
        $23d6bcdf1d0b93b7$var$borderSizeValue
    ],
    borderRightWidth: [
        'borderRightWidth',
        $23d6bcdf1d0b93b7$var$borderSizeValue
    ],
    borderTopWidth: [
        'borderTopWidth',
        $23d6bcdf1d0b93b7$var$borderSizeValue
    ],
    borderBottomWidth: [
        'borderBottomWidth',
        $23d6bcdf1d0b93b7$var$borderSizeValue
    ],
    borderXWidth: [
        [
            'borderLeftWidth',
            'borderRightWidth'
        ],
        $23d6bcdf1d0b93b7$var$borderSizeValue
    ],
    borderYWidth: [
        [
            'borderTopWidth',
            'borderBottomWidth'
        ],
        $23d6bcdf1d0b93b7$var$borderSizeValue
    ],
    borderColor: [
        'borderColor',
        $23d6bcdf1d0b93b7$var$borderColorValue
    ],
    borderStartColor: [
        $23d6bcdf1d0b93b7$var$rtl('borderLeftColor', 'borderRightColor'),
        $23d6bcdf1d0b93b7$var$borderColorValue
    ],
    borderEndColor: [
        $23d6bcdf1d0b93b7$var$rtl('borderRightColor', 'borderLeftColor'),
        $23d6bcdf1d0b93b7$var$borderColorValue
    ],
    borderLeftColor: [
        'borderLeftColor',
        $23d6bcdf1d0b93b7$var$borderColorValue
    ],
    borderRightColor: [
        'borderRightColor',
        $23d6bcdf1d0b93b7$var$borderColorValue
    ],
    borderTopColor: [
        'borderTopColor',
        $23d6bcdf1d0b93b7$var$borderColorValue
    ],
    borderBottomColor: [
        'borderBottomColor',
        $23d6bcdf1d0b93b7$var$borderColorValue
    ],
    borderXColor: [
        [
            'borderLeftColor',
            'borderRightColor'
        ],
        $23d6bcdf1d0b93b7$var$borderColorValue
    ],
    borderYColor: [
        [
            'borderTopColor',
            'borderBottomColor'
        ],
        $23d6bcdf1d0b93b7$var$borderColorValue
    ],
    borderRadius: [
        'borderRadius',
        $23d6bcdf1d0b93b7$var$borderRadiusValue
    ],
    borderTopStartRadius: [
        $23d6bcdf1d0b93b7$var$rtl('borderTopLeftRadius', 'borderTopRightRadius'),
        $23d6bcdf1d0b93b7$var$borderRadiusValue
    ],
    borderTopEndRadius: [
        $23d6bcdf1d0b93b7$var$rtl('borderTopRightRadius', 'borderTopLeftRadius'),
        $23d6bcdf1d0b93b7$var$borderRadiusValue
    ],
    borderBottomStartRadius: [
        $23d6bcdf1d0b93b7$var$rtl('borderBottomLeftRadius', 'borderBottomRightRadius'),
        $23d6bcdf1d0b93b7$var$borderRadiusValue
    ],
    borderBottomEndRadius: [
        $23d6bcdf1d0b93b7$var$rtl('borderBottomRightRadius', 'borderBottomLeftRadius'),
        $23d6bcdf1d0b93b7$var$borderRadiusValue
    ],
    borderTopLeftRadius: [
        'borderTopLeftRadius',
        $23d6bcdf1d0b93b7$var$borderRadiusValue
    ],
    borderTopRightRadius: [
        'borderTopRightRadius',
        $23d6bcdf1d0b93b7$var$borderRadiusValue
    ],
    borderBottomLeftRadius: [
        'borderBottomLeftRadius',
        $23d6bcdf1d0b93b7$var$borderRadiusValue
    ],
    borderBottomRightRadius: [
        'borderBottomRightRadius',
        $23d6bcdf1d0b93b7$var$borderRadiusValue
    ],
    padding: [
        'padding',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    paddingStart: [
        $23d6bcdf1d0b93b7$var$rtl('paddingLeft', 'paddingRight'),
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    paddingEnd: [
        $23d6bcdf1d0b93b7$var$rtl('paddingRight', 'paddingLeft'),
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    paddingLeft: [
        'paddingLeft',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    paddingRight: [
        'paddingRight',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    paddingTop: [
        'paddingTop',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    paddingBottom: [
        'paddingBottom',
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    paddingX: [
        [
            'paddingLeft',
            'paddingRight'
        ],
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    paddingY: [
        [
            'paddingTop',
            'paddingBottom'
        ],
        $23d6bcdf1d0b93b7$export$abc24f5b99744ea6
    ],
    overflow: [
        'overflow',
        $23d6bcdf1d0b93b7$export$46b6c81d11d2c30a
    ]
};
const $23d6bcdf1d0b93b7$var$borderStyleProps = {
    borderWidth: 'borderStyle',
    borderLeftWidth: 'borderLeftStyle',
    borderRightWidth: 'borderRightStyle',
    borderTopWidth: 'borderTopStyle',
    borderBottomWidth: 'borderBottomStyle'
};
function $23d6bcdf1d0b93b7$var$rtl(ltr, rtl) {
    return (direction)=>direction === 'rtl' ? rtl : ltr
    ;
}
const $23d6bcdf1d0b93b7$var$UNIT_RE = /(%|px|em|rem|vw|vh|auto|cm|mm|in|pt|pc|ex|ch|rem|vmin|vmax|fr)$/;
const $23d6bcdf1d0b93b7$var$FUNC_RE = /^\s*\w+\(/;
const $23d6bcdf1d0b93b7$var$SPECTRUM_VARIABLE_RE = /(static-)?size-\d+|single-line-(height|width)/g;
function $23d6bcdf1d0b93b7$export$abc24f5b99744ea6(value) {
    if (typeof value === 'number') return value + 'px';
    if ($23d6bcdf1d0b93b7$var$UNIT_RE.test(value)) return value;
    if ($23d6bcdf1d0b93b7$var$FUNC_RE.test(value)) return value.replace($23d6bcdf1d0b93b7$var$SPECTRUM_VARIABLE_RE, 'var(--spectrum-global-dimension-$&, var(--spectrum-alias-$&))');
    return `var(--spectrum-global-dimension-${value}, var(--spectrum-alias-${value}))`;
}
function $23d6bcdf1d0b93b7$export$f348bec194f2e6b5(value, matchedBreakpoints) {
    value = $23d6bcdf1d0b93b7$export$52dbfdbe1b2c3541(value, matchedBreakpoints);
    return $23d6bcdf1d0b93b7$export$abc24f5b99744ea6(value);
}
function $23d6bcdf1d0b93b7$var$colorValue(value, type = 'default') {
    return `var(--spectrum-global-color-${value}, var(--spectrum-semantic-${value}-color-${type}))`;
}
function $23d6bcdf1d0b93b7$var$backgroundColorValue(value) {
    return `var(--spectrum-alias-background-color-${value}, ${$23d6bcdf1d0b93b7$var$colorValue(value, 'background')})`;
}
function $23d6bcdf1d0b93b7$var$borderColorValue(value) {
    if (value === 'default') return 'var(--spectrum-alias-border-color)';
    return `var(--spectrum-alias-border-color-${value}, ${$23d6bcdf1d0b93b7$var$colorValue(value, 'border')})`;
}
function $23d6bcdf1d0b93b7$var$borderSizeValue(value) {
    return `var(--spectrum-alias-border-size-${value})`;
}
function $23d6bcdf1d0b93b7$var$borderRadiusValue(value) {
    return `var(--spectrum-alias-border-radius-${value})`;
}
function $23d6bcdf1d0b93b7$var$hiddenValue(value) {
    return value ? 'none' : undefined;
}
function $23d6bcdf1d0b93b7$var$anyValue(value) {
    return value;
}
function $23d6bcdf1d0b93b7$var$flexValue(value) {
    if (typeof value === 'boolean') return value ? '1' : undefined;
    return '' + value;
}
function $23d6bcdf1d0b93b7$export$f3c39bb9534218d0(props, handlers, direction, matchedBreakpoints) {
    let style = {
    };
    for(let key in props){
        let styleProp = handlers[key];
        if (!styleProp || props[key] == null) continue;
        let [name, convert] = styleProp;
        if (typeof name === 'function') name = name(direction);
        let prop = $23d6bcdf1d0b93b7$export$52dbfdbe1b2c3541(props[key], matchedBreakpoints);
        let value = convert(prop);
        if (Array.isArray(name)) for (let k of name)style[k] = value;
        else style[name] = value;
    }
    for(let prop in $23d6bcdf1d0b93b7$var$borderStyleProps)if (style[prop]) {
        style[$23d6bcdf1d0b93b7$var$borderStyleProps[prop]] = 'solid';
        style.boxSizing = 'border-box';
    }
    return style;
}
function $23d6bcdf1d0b93b7$export$b8e6fb9d2dff3f41(props, handlers = $23d6bcdf1d0b93b7$export$fe9c6e915565b4e8, options = {
}) {
    let { UNSAFE_className: UNSAFE_className , UNSAFE_style: UNSAFE_style , ...otherProps } = props;
    let breakpointProvider = $f2846dd3c9067392$export$199d6754bdf4e1e3();
    let { direction: direction  } = $lNqMH$reactariai18n.useLocale();
    let { matchedBreakpoints: matchedBreakpoints = (breakpointProvider === null || breakpointProvider === void 0 ? void 0 : breakpointProvider.matchedBreakpoints) || [
        'base'
    ]  } = options;
    let styles = $23d6bcdf1d0b93b7$export$f3c39bb9534218d0(props, handlers, direction, matchedBreakpoints);
    let style = {
        ...UNSAFE_style,
        ...styles
    };
    // @ts-ignore
    if (otherProps.className) console.warn("The className prop is unsafe and is unsupported in React Spectrum v3. Please use style props with Spectrum variables, or UNSAFE_className if you absolutely must do something custom. Note that this may break in future versions due to DOM structure changes.");
    // @ts-ignore
    if (otherProps.style) console.warn("The style prop is unsafe and is unsupported in React Spectrum v3. Please use style props with Spectrum variables, or UNSAFE_style if you absolutely must do something custom. Note that this may break in future versions due to DOM structure changes.");
    let styleProps = {
        style: style,
        className: UNSAFE_className
    };
    if ($23d6bcdf1d0b93b7$export$52dbfdbe1b2c3541(props.isHidden, matchedBreakpoints)) styleProps.hidden = true;
    return {
        styleProps: styleProps
    };
}
function $23d6bcdf1d0b93b7$export$46b6c81d11d2c30a(value) {
    return value;
}
function $23d6bcdf1d0b93b7$export$52dbfdbe1b2c3541(prop, matchedBreakpoints) {
    if (prop && typeof prop === 'object' && !Array.isArray(prop)) {
        for(let i = 0; i < matchedBreakpoints.length; i++){
            let breakpoint = matchedBreakpoints[i];
            if (prop[breakpoint] != null) return prop[breakpoint];
        }
        return prop.base;
    }
    return prop;
}


var $f789ad0ec2caffd0$exports = {};

$parcel$export($f789ad0ec2caffd0$exports, "useSlotProps", () => $f789ad0ec2caffd0$export$1e5c9e6e4e15efe3);
$parcel$export($f789ad0ec2caffd0$exports, "cssModuleToSlots", () => $f789ad0ec2caffd0$export$365cf34cda9978e2);
$parcel$export($f789ad0ec2caffd0$exports, "SlotProvider", () => $f789ad0ec2caffd0$export$8107b24b91795686);
$parcel$export($f789ad0ec2caffd0$exports, "ClearSlots", () => $f789ad0ec2caffd0$export$ceb145244332b7a2);


let $f789ad0ec2caffd0$var$SlotContext = /*#__PURE__*/ ($parcel$interopDefault($lNqMH$react)).createContext(null);
function $f789ad0ec2caffd0$export$1e5c9e6e4e15efe3(props, defaultSlot) {
    let slot = props.slot || defaultSlot;
    let { [slot]: slotProps = {
    }  } = $lNqMH$react.useContext($f789ad0ec2caffd0$var$SlotContext) || {
    };
    return $lNqMH$reactariautils.mergeProps(props, $lNqMH$reactariautils.mergeProps(slotProps, {
        id: props.id
    }));
}
function $f789ad0ec2caffd0$export$365cf34cda9978e2(cssModule) {
    return Object.keys(cssModule).reduce((acc, slot)=>{
        acc[slot] = {
            UNSAFE_className: cssModule[slot]
        };
        return acc;
    }, {
    });
}
function $f789ad0ec2caffd0$export$8107b24b91795686(props) {
    let parentSlots = $lNqMH$react.useContext($f789ad0ec2caffd0$var$SlotContext) || {
    };
    let { slots: slots = {
    } , children: children  } = props;
    // Merge props for each slot from parent context and props
    let value = $lNqMH$react.useMemo(()=>Object.keys(parentSlots).concat(Object.keys(slots)).reduce((o, p)=>({
                ...o,
                [p]: $lNqMH$reactariautils.mergeProps(parentSlots[p] || {
                }, slots[p] || {
                })
            })
        , {
        })
    , [
        parentSlots,
        slots
    ]);
    return(/*#__PURE__*/ ($parcel$interopDefault($lNqMH$react)).createElement($f789ad0ec2caffd0$var$SlotContext.Provider, {
        value: value
    }, children));
}
function $f789ad0ec2caffd0$export$ceb145244332b7a2(props) {
    let { children: children , ...otherProps } = props;
    let content = children;
    if (($parcel$interopDefault($lNqMH$react)).Children.toArray(children).length <= 1) {
        if (typeof children === 'function') content = /*#__PURE__*/ ($parcel$interopDefault($lNqMH$react)).cloneElement(($parcel$interopDefault($lNqMH$react)).Children.only(children), otherProps);
    }
    return(/*#__PURE__*/ ($parcel$interopDefault($lNqMH$react)).createElement($f789ad0ec2caffd0$var$SlotContext.Provider, {
        value: {
        }
    }, content));
}


var $47091953f88214e5$exports = {};

$parcel$export($47091953f88214e5$exports, "useHasChild", () => $47091953f88214e5$export$e52e2242b6d0f1d4);


function $47091953f88214e5$export$e52e2242b6d0f1d4(query, ref) {
    let [hasChild, setHasChild] = $lNqMH$react.useState(true);
    $lNqMH$reactariautils.useLayoutEffect(()=>{
        setHasChild(!!(ref.current && ref.current.querySelector(query)));
    }, [
        setHasChild,
        query,
        ref
    ]);
    return hasChild;
}


var $2423c85918cf93dc$exports = {};

$parcel$export($2423c85918cf93dc$exports, "useResizeObserver", () => $lNqMH$reactariautils.useResizeObserver);



var $0e63baace47ba08f$exports = {};

$parcel$export($0e63baace47ba08f$exports, "useIsMobileDevice", () => $0e63baace47ba08f$export$736bf165441b18c7);

const $0e63baace47ba08f$var$MOBILE_SCREEN_WIDTH = 700;
function $0e63baace47ba08f$export$736bf165441b18c7() {
    let isSSR = $lNqMH$reactariassr.useIsSSR();
    if (isSSR || typeof window === 'undefined') return false;
    return window.screen.width <= $0e63baace47ba08f$var$MOBILE_SCREEN_WIDTH;
}


var $55eab923392975f0$exports = {};

$parcel$export($55eab923392975f0$exports, "useValueEffect", () => $lNqMH$reactariautils.useValueEffect);




$parcel$exportWildcard(module.exports, $ee2244f86238f046$exports);
$parcel$exportWildcard(module.exports, $70e73537f8916bcc$exports);
$parcel$exportWildcard(module.exports, $1b6371e033fc95ef$exports);
$parcel$exportWildcard(module.exports, $c62409ad469b3817$exports);
$parcel$exportWildcard(module.exports, $23d6bcdf1d0b93b7$exports);
$parcel$exportWildcard(module.exports, $f789ad0ec2caffd0$exports);
$parcel$exportWildcard(module.exports, $47091953f88214e5$exports);
$parcel$exportWildcard(module.exports, $2423c85918cf93dc$exports);
$parcel$exportWildcard(module.exports, $0e63baace47ba08f$exports);
$parcel$exportWildcard(module.exports, $55eab923392975f0$exports);
$parcel$exportWildcard(module.exports, $f2846dd3c9067392$exports);
