// @react-spectrum/utils/dist/module.js
import $1MhIz$clsx from "clsx";
import $1MhIz$react, {useState as $1MhIz$useState, useEffect as $1MhIz$useEffect, useRef as $1MhIz$useRef, useImperativeHandle as $1MhIz$useImperativeHandle, useMemo as $1MhIz$useMemo, useContext as $1MhIz$useContext} from "react";
import {useIsSSR as $1MhIz$useIsSSR} from "@react-aria/ssr";
import {useLocale as $1MhIz$useLocale} from "@react-aria/i18n";
import {mergeProps as $1MhIz$mergeProps, useLayoutEffect as $1MhIz$useLayoutEffect, useResizeObserver as $470e61c66b07f487$re_export$useResizeObserver, useValueEffect as $5b569f8bd87e05f1$re_export$useValueEffect} from "@react-aria/utils";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $368fcc465babe045$exports = {};

$parcel$export($368fcc465babe045$exports, "shouldKeepSpectrumClassNames", () => $368fcc465babe045$export$46d604dce8bf8724);
$parcel$export($368fcc465babe045$exports, "keepSpectrumClassNames", () => $368fcc465babe045$export$f9d3bfd10703eb31);
$parcel$export($368fcc465babe045$exports, "classNames", () => $368fcc465babe045$export$ce4ab0c55987d1ff);

let $368fcc465babe045$export$46d604dce8bf8724 = false;
function $368fcc465babe045$export$f9d3bfd10703eb31() {
    $368fcc465babe045$export$46d604dce8bf8724 = true;
    console.warn("Legacy spectrum-prefixed class names enabled for backward compatibility. We recommend replacing instances of CSS overrides targeting spectrum selectors in your app with custom class names of your own, and disabling this flag.");
}
function $368fcc465babe045$export$ce4ab0c55987d1ff(cssModule, ...values) {
    let classes = [];
    for (let value of values){
        if (typeof value === 'object' && value) {
            let mapped = {
            };
            for(let key in value){
                if (cssModule[key]) mapped[cssModule[key]] = value[key];
                if ($368fcc465babe045$export$46d604dce8bf8724 || !cssModule[key]) mapped[key] = value[key];
            }
            classes.push(mapped);
        } else if (typeof value === 'string') {
            if (cssModule[value]) classes.push(cssModule[value]);
            if ($368fcc465babe045$export$46d604dce8bf8724 || !cssModule[value]) classes.push(value);
        } else classes.push(value);
    }
    return $1MhIz$clsx(...classes);
}


var $040be1e92b750588$exports = {};

$parcel$export($040be1e92b750588$exports, "getWrappedElement", () => $040be1e92b750588$export$a5f5a6912b18861c);

function $040be1e92b750588$export$a5f5a6912b18861c(children) {
    let element;
    if (typeof children === 'string') element = /*#__PURE__*/ $1MhIz$react.createElement("span", null, children);
    else element = $1MhIz$react.Children.only(children);
    return element;
}


var $4b8bd32804da4aaa$exports = {};

$parcel$export($4b8bd32804da4aaa$exports, "useMediaQuery", () => $4b8bd32804da4aaa$export$32d5543ab307c01);


function $4b8bd32804da4aaa$export$32d5543ab307c01(query) {
    let supportsMatchMedia = typeof window !== 'undefined' && typeof window.matchMedia === 'function';
    let [matches, setMatches] = $1MhIz$useState(()=>supportsMatchMedia ? window.matchMedia(query).matches : false
    );
    $1MhIz$useEffect(()=>{
        if (!supportsMatchMedia) return;
        let mq = window.matchMedia(query);
        let onChange = (evt)=>{
            setMatches(evt.matches);
        };
        mq.addListener(onChange);
        return ()=>{
            mq.removeListener(onChange);
        };
    }, [
        supportsMatchMedia,
        query
    ]);
    // If in SSR, the media query should never match. Once the page hydrates,
    // this will update and the real value will be returned.
    let isSSR = $1MhIz$useIsSSR();
    return isSSR ? false : matches;
}


var $48195d35898e8e33$exports = {};

$parcel$export($48195d35898e8e33$exports, "createDOMRef", () => $48195d35898e8e33$export$a5795cc979dfae80);
$parcel$export($48195d35898e8e33$exports, "createFocusableRef", () => $48195d35898e8e33$export$79d69eee6ae4b329);
$parcel$export($48195d35898e8e33$exports, "useDOMRef", () => $48195d35898e8e33$export$c2c55ef9111cafd8);
$parcel$export($48195d35898e8e33$exports, "useFocusableRef", () => $48195d35898e8e33$export$96a734597687c040);
$parcel$export($48195d35898e8e33$exports, "unwrapDOMRef", () => $48195d35898e8e33$export$c7e28c72a4823176);
$parcel$export($48195d35898e8e33$exports, "useUnwrapDOMRef", () => $48195d35898e8e33$export$1d5cc31d9d8df817);

function $48195d35898e8e33$export$a5795cc979dfae80(ref) {
    return {
        UNSAFE_getDOMNode () {
            return ref.current;
        }
    };
}
function $48195d35898e8e33$export$79d69eee6ae4b329(domRef, focusableRef = domRef) {
    return {
        ...$48195d35898e8e33$export$a5795cc979dfae80(domRef),
        focus () {
            if (focusableRef.current) focusableRef.current.focus();
        }
    };
}
function $48195d35898e8e33$export$c2c55ef9111cafd8(ref) {
    let domRef = $1MhIz$useRef(null);
    $1MhIz$useImperativeHandle(ref, ()=>$48195d35898e8e33$export$a5795cc979dfae80(domRef)
    );
    return domRef;
}
function $48195d35898e8e33$export$96a734597687c040(ref, focusableRef) {
    let domRef = $1MhIz$useRef(null);
    $1MhIz$useImperativeHandle(ref, ()=>$48195d35898e8e33$export$79d69eee6ae4b329(domRef, focusableRef)
    );
    return domRef;
}
function $48195d35898e8e33$export$c7e28c72a4823176(ref) {
    return {
        get current () {
            return ref.current && ref.current.UNSAFE_getDOMNode();
        }
    };
}
function $48195d35898e8e33$export$1d5cc31d9d8df817(ref) {
    return $1MhIz$useMemo(()=>$48195d35898e8e33$export$c7e28c72a4823176(ref)
    , [
        ref
    ]);
}


var $fe256ae6f45aa375$exports = {};

$parcel$export($fe256ae6f45aa375$exports, "baseStyleProps", () => $fe256ae6f45aa375$export$fe9c6e915565b4e8);
$parcel$export($fe256ae6f45aa375$exports, "dimensionValue", () => $fe256ae6f45aa375$export$abc24f5b99744ea6);
$parcel$export($fe256ae6f45aa375$exports, "passthroughStyle", () => $fe256ae6f45aa375$export$46b6c81d11d2c30a);
$parcel$export($fe256ae6f45aa375$exports, "viewStyleProps", () => $fe256ae6f45aa375$export$e0705d1a55f297c);
$parcel$export($fe256ae6f45aa375$exports, "responsiveDimensionValue", () => $fe256ae6f45aa375$export$f348bec194f2e6b5);
$parcel$export($fe256ae6f45aa375$exports, "getResponsiveProp", () => $fe256ae6f45aa375$export$52dbfdbe1b2c3541);
$parcel$export($fe256ae6f45aa375$exports, "convertStyleProps", () => $fe256ae6f45aa375$export$f3c39bb9534218d0);
$parcel$export($fe256ae6f45aa375$exports, "useStyleProps", () => $fe256ae6f45aa375$export$b8e6fb9d2dff3f41);
var $31dbf1c5576cf635$exports = {};

$parcel$export($31dbf1c5576cf635$exports, "BreakpointProvider", () => $31dbf1c5576cf635$export$8214320346cf5104);
$parcel$export($31dbf1c5576cf635$exports, "useMatchedBreakpoints", () => $31dbf1c5576cf635$export$140ae7baa51cca23);
$parcel$export($31dbf1c5576cf635$exports, "useBreakpoint", () => $31dbf1c5576cf635$export$199d6754bdf4e1e3);


const $31dbf1c5576cf635$var$Context = /*#__PURE__*/ $1MhIz$react.createContext(null);
$31dbf1c5576cf635$var$Context.displayName = 'BreakpointContext';
function $31dbf1c5576cf635$export$8214320346cf5104(props) {
    let { children: children , matchedBreakpoints: matchedBreakpoints  } = props;
    return(/*#__PURE__*/ $1MhIz$react.createElement($31dbf1c5576cf635$var$Context.Provider, {
        value: {
            matchedBreakpoints: matchedBreakpoints
        }
    }, children));
}
function $31dbf1c5576cf635$export$140ae7baa51cca23(breakpoints) {
    let entries = Object.entries(breakpoints).sort(([, valueA], [, valueB])=>valueB - valueA
    );
    let breakpointQueries = entries.map(([, value])=>`(min-width: ${value}px)`
    );
    let supportsMatchMedia = typeof window !== 'undefined' && typeof window.matchMedia === 'function';
    let getBreakpointHandler = ()=>{
        let matched = [];
        for(let i in breakpointQueries){
            let query = breakpointQueries[i];
            if (window.matchMedia(query).matches) matched.push(entries[i][0]);
        }
        matched.push('base');
        return matched;
    };
    let [breakpoint1, setBreakpoint] = $1MhIz$useState(()=>supportsMatchMedia ? getBreakpointHandler() : [
            'base'
        ]
    );
    $1MhIz$useEffect(()=>{
        if (!supportsMatchMedia) return;
        let onResize = ()=>{
            const breakpointHandler = getBreakpointHandler();
            setBreakpoint((previousBreakpointHandler)=>{
                if (previousBreakpointHandler.length !== breakpointHandler.length || previousBreakpointHandler.some((breakpoint, idx)=>breakpoint !== breakpointHandler[idx]
                )) return [
                    ...breakpointHandler
                ]; // Return a new array to force state change
                return previousBreakpointHandler;
            });
        };
        window.addEventListener('resize', onResize);
        return ()=>{
            window.removeEventListener('resize', onResize);
        };
    }, [
        supportsMatchMedia
    ]);
    // If in SSR, the media query should never match. Once the page hydrates,
    // this will update and the real value will be returned.
    let isSSR = $1MhIz$useIsSSR();
    return isSSR ? [
        'base'
    ] : breakpoint1;
}
function $31dbf1c5576cf635$export$199d6754bdf4e1e3() {
    return $1MhIz$useContext($31dbf1c5576cf635$var$Context);
}



const $fe256ae6f45aa375$export$fe9c6e915565b4e8 = {
    margin: [
        'margin',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    marginStart: [
        $fe256ae6f45aa375$var$rtl('marginLeft', 'marginRight'),
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    marginEnd: [
        $fe256ae6f45aa375$var$rtl('marginRight', 'marginLeft'),
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    // marginLeft: ['marginLeft', dimensionValue],
    // marginRight: ['marginRight', dimensionValue],
    marginTop: [
        'marginTop',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    marginBottom: [
        'marginBottom',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    marginX: [
        [
            'marginLeft',
            'marginRight'
        ],
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    marginY: [
        [
            'marginTop',
            'marginBottom'
        ],
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    width: [
        'width',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    height: [
        'height',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    minWidth: [
        'minWidth',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    minHeight: [
        'minHeight',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    maxWidth: [
        'maxWidth',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    maxHeight: [
        'maxHeight',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    isHidden: [
        'display',
        $fe256ae6f45aa375$var$hiddenValue
    ],
    alignSelf: [
        'alignSelf',
        $fe256ae6f45aa375$export$46b6c81d11d2c30a
    ],
    justifySelf: [
        'justifySelf',
        $fe256ae6f45aa375$export$46b6c81d11d2c30a
    ],
    position: [
        'position',
        $fe256ae6f45aa375$var$anyValue
    ],
    zIndex: [
        'zIndex',
        $fe256ae6f45aa375$var$anyValue
    ],
    top: [
        'top',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    bottom: [
        'bottom',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    start: [
        $fe256ae6f45aa375$var$rtl('left', 'right'),
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    end: [
        $fe256ae6f45aa375$var$rtl('right', 'left'),
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    left: [
        'left',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    right: [
        'right',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    order: [
        'order',
        $fe256ae6f45aa375$var$anyValue
    ],
    flex: [
        'flex',
        $fe256ae6f45aa375$var$flexValue
    ],
    flexGrow: [
        'flexGrow',
        $fe256ae6f45aa375$export$46b6c81d11d2c30a
    ],
    flexShrink: [
        'flexShrink',
        $fe256ae6f45aa375$export$46b6c81d11d2c30a
    ],
    flexBasis: [
        'flexBasis',
        $fe256ae6f45aa375$export$46b6c81d11d2c30a
    ],
    gridArea: [
        'gridArea',
        $fe256ae6f45aa375$export$46b6c81d11d2c30a
    ],
    gridColumn: [
        'gridColumn',
        $fe256ae6f45aa375$export$46b6c81d11d2c30a
    ],
    gridColumnEnd: [
        'gridColumnEnd',
        $fe256ae6f45aa375$export$46b6c81d11d2c30a
    ],
    gridColumnStart: [
        'gridColumnStart',
        $fe256ae6f45aa375$export$46b6c81d11d2c30a
    ],
    gridRow: [
        'gridRow',
        $fe256ae6f45aa375$export$46b6c81d11d2c30a
    ],
    gridRowEnd: [
        'gridRowEnd',
        $fe256ae6f45aa375$export$46b6c81d11d2c30a
    ],
    gridRowStart: [
        'gridRowStart',
        $fe256ae6f45aa375$export$46b6c81d11d2c30a
    ]
};
const $fe256ae6f45aa375$export$e0705d1a55f297c = {
    ...$fe256ae6f45aa375$export$fe9c6e915565b4e8,
    backgroundColor: [
        'backgroundColor',
        $fe256ae6f45aa375$var$backgroundColorValue
    ],
    borderWidth: [
        'borderWidth',
        $fe256ae6f45aa375$var$borderSizeValue
    ],
    borderStartWidth: [
        $fe256ae6f45aa375$var$rtl('borderLeftWidth', 'borderRightWidth'),
        $fe256ae6f45aa375$var$borderSizeValue
    ],
    borderEndWidth: [
        $fe256ae6f45aa375$var$rtl('borderRightWidth', 'borderLeftWidth'),
        $fe256ae6f45aa375$var$borderSizeValue
    ],
    borderLeftWidth: [
        'borderLeftWidth',
        $fe256ae6f45aa375$var$borderSizeValue
    ],
    borderRightWidth: [
        'borderRightWidth',
        $fe256ae6f45aa375$var$borderSizeValue
    ],
    borderTopWidth: [
        'borderTopWidth',
        $fe256ae6f45aa375$var$borderSizeValue
    ],
    borderBottomWidth: [
        'borderBottomWidth',
        $fe256ae6f45aa375$var$borderSizeValue
    ],
    borderXWidth: [
        [
            'borderLeftWidth',
            'borderRightWidth'
        ],
        $fe256ae6f45aa375$var$borderSizeValue
    ],
    borderYWidth: [
        [
            'borderTopWidth',
            'borderBottomWidth'
        ],
        $fe256ae6f45aa375$var$borderSizeValue
    ],
    borderColor: [
        'borderColor',
        $fe256ae6f45aa375$var$borderColorValue
    ],
    borderStartColor: [
        $fe256ae6f45aa375$var$rtl('borderLeftColor', 'borderRightColor'),
        $fe256ae6f45aa375$var$borderColorValue
    ],
    borderEndColor: [
        $fe256ae6f45aa375$var$rtl('borderRightColor', 'borderLeftColor'),
        $fe256ae6f45aa375$var$borderColorValue
    ],
    borderLeftColor: [
        'borderLeftColor',
        $fe256ae6f45aa375$var$borderColorValue
    ],
    borderRightColor: [
        'borderRightColor',
        $fe256ae6f45aa375$var$borderColorValue
    ],
    borderTopColor: [
        'borderTopColor',
        $fe256ae6f45aa375$var$borderColorValue
    ],
    borderBottomColor: [
        'borderBottomColor',
        $fe256ae6f45aa375$var$borderColorValue
    ],
    borderXColor: [
        [
            'borderLeftColor',
            'borderRightColor'
        ],
        $fe256ae6f45aa375$var$borderColorValue
    ],
    borderYColor: [
        [
            'borderTopColor',
            'borderBottomColor'
        ],
        $fe256ae6f45aa375$var$borderColorValue
    ],
    borderRadius: [
        'borderRadius',
        $fe256ae6f45aa375$var$borderRadiusValue
    ],
    borderTopStartRadius: [
        $fe256ae6f45aa375$var$rtl('borderTopLeftRadius', 'borderTopRightRadius'),
        $fe256ae6f45aa375$var$borderRadiusValue
    ],
    borderTopEndRadius: [
        $fe256ae6f45aa375$var$rtl('borderTopRightRadius', 'borderTopLeftRadius'),
        $fe256ae6f45aa375$var$borderRadiusValue
    ],
    borderBottomStartRadius: [
        $fe256ae6f45aa375$var$rtl('borderBottomLeftRadius', 'borderBottomRightRadius'),
        $fe256ae6f45aa375$var$borderRadiusValue
    ],
    borderBottomEndRadius: [
        $fe256ae6f45aa375$var$rtl('borderBottomRightRadius', 'borderBottomLeftRadius'),
        $fe256ae6f45aa375$var$borderRadiusValue
    ],
    borderTopLeftRadius: [
        'borderTopLeftRadius',
        $fe256ae6f45aa375$var$borderRadiusValue
    ],
    borderTopRightRadius: [
        'borderTopRightRadius',
        $fe256ae6f45aa375$var$borderRadiusValue
    ],
    borderBottomLeftRadius: [
        'borderBottomLeftRadius',
        $fe256ae6f45aa375$var$borderRadiusValue
    ],
    borderBottomRightRadius: [
        'borderBottomRightRadius',
        $fe256ae6f45aa375$var$borderRadiusValue
    ],
    padding: [
        'padding',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    paddingStart: [
        $fe256ae6f45aa375$var$rtl('paddingLeft', 'paddingRight'),
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    paddingEnd: [
        $fe256ae6f45aa375$var$rtl('paddingRight', 'paddingLeft'),
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    paddingLeft: [
        'paddingLeft',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    paddingRight: [
        'paddingRight',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    paddingTop: [
        'paddingTop',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    paddingBottom: [
        'paddingBottom',
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    paddingX: [
        [
            'paddingLeft',
            'paddingRight'
        ],
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    paddingY: [
        [
            'paddingTop',
            'paddingBottom'
        ],
        $fe256ae6f45aa375$export$abc24f5b99744ea6
    ],
    overflow: [
        'overflow',
        $fe256ae6f45aa375$export$46b6c81d11d2c30a
    ]
};
const $fe256ae6f45aa375$var$borderStyleProps = {
    borderWidth: 'borderStyle',
    borderLeftWidth: 'borderLeftStyle',
    borderRightWidth: 'borderRightStyle',
    borderTopWidth: 'borderTopStyle',
    borderBottomWidth: 'borderBottomStyle'
};
function $fe256ae6f45aa375$var$rtl(ltr, rtl) {
    return (direction)=>direction === 'rtl' ? rtl : ltr
    ;
}
const $fe256ae6f45aa375$var$UNIT_RE = /(%|px|em|rem|vw|vh|auto|cm|mm|in|pt|pc|ex|ch|rem|vmin|vmax|fr)$/;
const $fe256ae6f45aa375$var$FUNC_RE = /^\s*\w+\(/;
const $fe256ae6f45aa375$var$SPECTRUM_VARIABLE_RE = /(static-)?size-\d+|single-line-(height|width)/g;
function $fe256ae6f45aa375$export$abc24f5b99744ea6(value) {
    if (typeof value === 'number') return value + 'px';
    if ($fe256ae6f45aa375$var$UNIT_RE.test(value)) return value;
    if ($fe256ae6f45aa375$var$FUNC_RE.test(value)) return value.replace($fe256ae6f45aa375$var$SPECTRUM_VARIABLE_RE, 'var(--spectrum-global-dimension-$&, var(--spectrum-alias-$&))');
    return `var(--spectrum-global-dimension-${value}, var(--spectrum-alias-${value}))`;
}
function $fe256ae6f45aa375$export$f348bec194f2e6b5(value, matchedBreakpoints) {
    value = $fe256ae6f45aa375$export$52dbfdbe1b2c3541(value, matchedBreakpoints);
    return $fe256ae6f45aa375$export$abc24f5b99744ea6(value);
}
function $fe256ae6f45aa375$var$colorValue(value, type = 'default') {
    return `var(--spectrum-global-color-${value}, var(--spectrum-semantic-${value}-color-${type}))`;
}
function $fe256ae6f45aa375$var$backgroundColorValue(value) {
    return `var(--spectrum-alias-background-color-${value}, ${$fe256ae6f45aa375$var$colorValue(value, 'background')})`;
}
function $fe256ae6f45aa375$var$borderColorValue(value) {
    if (value === 'default') return 'var(--spectrum-alias-border-color)';
    return `var(--spectrum-alias-border-color-${value}, ${$fe256ae6f45aa375$var$colorValue(value, 'border')})`;
}
function $fe256ae6f45aa375$var$borderSizeValue(value) {
    return `var(--spectrum-alias-border-size-${value})`;
}
function $fe256ae6f45aa375$var$borderRadiusValue(value) {
    return `var(--spectrum-alias-border-radius-${value})`;
}
function $fe256ae6f45aa375$var$hiddenValue(value) {
    return value ? 'none' : undefined;
}
function $fe256ae6f45aa375$var$anyValue(value) {
    return value;
}
function $fe256ae6f45aa375$var$flexValue(value) {
    if (typeof value === 'boolean') return value ? '1' : undefined;
    return '' + value;
}
function $fe256ae6f45aa375$export$f3c39bb9534218d0(props, handlers, direction, matchedBreakpoints) {
    let style = {
    };
    for(let key in props){
        let styleProp = handlers[key];
        if (!styleProp || props[key] == null) continue;
        let [name, convert] = styleProp;
        if (typeof name === 'function') name = name(direction);
        let prop = $fe256ae6f45aa375$export$52dbfdbe1b2c3541(props[key], matchedBreakpoints);
        let value = convert(prop);
        if (Array.isArray(name)) for (let k of name)style[k] = value;
        else style[name] = value;
    }
    for(let prop in $fe256ae6f45aa375$var$borderStyleProps)if (style[prop]) {
        style[$fe256ae6f45aa375$var$borderStyleProps[prop]] = 'solid';
        style.boxSizing = 'border-box';
    }
    return style;
}
function $fe256ae6f45aa375$export$b8e6fb9d2dff3f41(props, handlers = $fe256ae6f45aa375$export$fe9c6e915565b4e8, options = {
}) {
    let { UNSAFE_className: UNSAFE_className , UNSAFE_style: UNSAFE_style , ...otherProps } = props;
    let breakpointProvider = $31dbf1c5576cf635$export$199d6754bdf4e1e3();
    let { direction: direction  } = $1MhIz$useLocale();
    let { matchedBreakpoints: matchedBreakpoints = (breakpointProvider === null || breakpointProvider === void 0 ? void 0 : breakpointProvider.matchedBreakpoints) || [
        'base'
    ]  } = options;
    let styles = $fe256ae6f45aa375$export$f3c39bb9534218d0(props, handlers, direction, matchedBreakpoints);
    let style = {
        ...UNSAFE_style,
        ...styles
    };
    // @ts-ignore
    if (otherProps.className) console.warn("The className prop is unsafe and is unsupported in React Spectrum v3. Please use style props with Spectrum variables, or UNSAFE_className if you absolutely must do something custom. Note that this may break in future versions due to DOM structure changes.");
    // @ts-ignore
    if (otherProps.style) console.warn("The style prop is unsafe and is unsupported in React Spectrum v3. Please use style props with Spectrum variables, or UNSAFE_style if you absolutely must do something custom. Note that this may break in future versions due to DOM structure changes.");
    let styleProps = {
        style: style,
        className: UNSAFE_className
    };
    if ($fe256ae6f45aa375$export$52dbfdbe1b2c3541(props.isHidden, matchedBreakpoints)) styleProps.hidden = true;
    return {
        styleProps: styleProps
    };
}
function $fe256ae6f45aa375$export$46b6c81d11d2c30a(value) {
    return value;
}
function $fe256ae6f45aa375$export$52dbfdbe1b2c3541(prop, matchedBreakpoints) {
    if (prop && typeof prop === 'object' && !Array.isArray(prop)) {
        for(let i = 0; i < matchedBreakpoints.length; i++){
            let breakpoint = matchedBreakpoints[i];
            if (prop[breakpoint] != null) return prop[breakpoint];
        }
        return prop.base;
    }
    return prop;
}


var $142bd45dbe45bbab$exports = {};

$parcel$export($142bd45dbe45bbab$exports, "useSlotProps", () => $142bd45dbe45bbab$export$1e5c9e6e4e15efe3);
$parcel$export($142bd45dbe45bbab$exports, "cssModuleToSlots", () => $142bd45dbe45bbab$export$365cf34cda9978e2);
$parcel$export($142bd45dbe45bbab$exports, "SlotProvider", () => $142bd45dbe45bbab$export$8107b24b91795686);
$parcel$export($142bd45dbe45bbab$exports, "ClearSlots", () => $142bd45dbe45bbab$export$ceb145244332b7a2);


let $142bd45dbe45bbab$var$SlotContext = /*#__PURE__*/ $1MhIz$react.createContext(null);
function $142bd45dbe45bbab$export$1e5c9e6e4e15efe3(props, defaultSlot) {
    let slot = props.slot || defaultSlot;
    let { [slot]: slotProps = {
    }  } = $1MhIz$useContext($142bd45dbe45bbab$var$SlotContext) || {
    };
    return $1MhIz$mergeProps(props, $1MhIz$mergeProps(slotProps, {
        id: props.id
    }));
}
function $142bd45dbe45bbab$export$365cf34cda9978e2(cssModule) {
    return Object.keys(cssModule).reduce((acc, slot)=>{
        acc[slot] = {
            UNSAFE_className: cssModule[slot]
        };
        return acc;
    }, {
    });
}
function $142bd45dbe45bbab$export$8107b24b91795686(props) {
    let parentSlots = $1MhIz$useContext($142bd45dbe45bbab$var$SlotContext) || {
    };
    let { slots: slots = {
    } , children: children  } = props;
    // Merge props for each slot from parent context and props
    let value = $1MhIz$useMemo(()=>Object.keys(parentSlots).concat(Object.keys(slots)).reduce((o, p)=>({
                ...o,
                [p]: $1MhIz$mergeProps(parentSlots[p] || {
                }, slots[p] || {
                })
            })
        , {
        })
    , [
        parentSlots,
        slots
    ]);
    return(/*#__PURE__*/ $1MhIz$react.createElement($142bd45dbe45bbab$var$SlotContext.Provider, {
        value: value
    }, children));
}
function $142bd45dbe45bbab$export$ceb145244332b7a2(props) {
    let { children: children , ...otherProps } = props;
    let content = children;
    if ($1MhIz$react.Children.toArray(children).length <= 1) {
        if (typeof children === 'function') content = /*#__PURE__*/ $1MhIz$react.cloneElement($1MhIz$react.Children.only(children), otherProps);
    }
    return(/*#__PURE__*/ $1MhIz$react.createElement($142bd45dbe45bbab$var$SlotContext.Provider, {
        value: {
        }
    }, content));
}


var $e30b74a2c392bf42$exports = {};

$parcel$export($e30b74a2c392bf42$exports, "useHasChild", () => $e30b74a2c392bf42$export$e52e2242b6d0f1d4);


function $e30b74a2c392bf42$export$e52e2242b6d0f1d4(query, ref) {
    let [hasChild, setHasChild] = $1MhIz$useState(true);
    $1MhIz$useLayoutEffect(()=>{
        setHasChild(!!(ref.current && ref.current.querySelector(query)));
    }, [
        setHasChild,
        query,
        ref
    ]);
    return hasChild;
}


var $470e61c66b07f487$exports = {};

$parcel$export($470e61c66b07f487$exports, "useResizeObserver", () => $470e61c66b07f487$re_export$useResizeObserver);



var $abf7096307f33b22$exports = {};

$parcel$export($abf7096307f33b22$exports, "useIsMobileDevice", () => $abf7096307f33b22$export$736bf165441b18c7);

const $abf7096307f33b22$var$MOBILE_SCREEN_WIDTH = 700;
function $abf7096307f33b22$export$736bf165441b18c7() {
    let isSSR = $1MhIz$useIsSSR();
    if (isSSR || typeof window === 'undefined') return false;
    return window.screen.width <= $abf7096307f33b22$var$MOBILE_SCREEN_WIDTH;
}


var $5b569f8bd87e05f1$exports = {};

$parcel$export($5b569f8bd87e05f1$exports, "useValueEffect", () => $5b569f8bd87e05f1$re_export$useValueEffect);






export {$368fcc465babe045$export$46d604dce8bf8724 as shouldKeepSpectrumClassNames, $368fcc465babe045$export$f9d3bfd10703eb31 as keepSpectrumClassNames, $368fcc465babe045$export$ce4ab0c55987d1ff as classNames, $040be1e92b750588$export$a5f5a6912b18861c as getWrappedElement, $4b8bd32804da4aaa$export$32d5543ab307c01 as useMediaQuery, $48195d35898e8e33$export$a5795cc979dfae80 as createDOMRef, $48195d35898e8e33$export$79d69eee6ae4b329 as createFocusableRef, $48195d35898e8e33$export$c2c55ef9111cafd8 as useDOMRef, $48195d35898e8e33$export$96a734597687c040 as useFocusableRef, $48195d35898e8e33$export$c7e28c72a4823176 as unwrapDOMRef, $48195d35898e8e33$export$1d5cc31d9d8df817 as useUnwrapDOMRef, $fe256ae6f45aa375$export$fe9c6e915565b4e8 as baseStyleProps, $fe256ae6f45aa375$export$abc24f5b99744ea6 as dimensionValue, $fe256ae6f45aa375$export$46b6c81d11d2c30a as passthroughStyle, $fe256ae6f45aa375$export$e0705d1a55f297c as viewStyleProps, $fe256ae6f45aa375$export$f348bec194f2e6b5 as responsiveDimensionValue, $fe256ae6f45aa375$export$52dbfdbe1b2c3541 as getResponsiveProp, $fe256ae6f45aa375$export$f3c39bb9534218d0 as convertStyleProps, $fe256ae6f45aa375$export$b8e6fb9d2dff3f41 as useStyleProps, $142bd45dbe45bbab$export$1e5c9e6e4e15efe3 as useSlotProps, $142bd45dbe45bbab$export$365cf34cda9978e2 as cssModuleToSlots, $142bd45dbe45bbab$export$8107b24b91795686 as SlotProvider, $142bd45dbe45bbab$export$ceb145244332b7a2 as ClearSlots, $e30b74a2c392bf42$export$e52e2242b6d0f1d4 as useHasChild, $470e61c66b07f487$re_export$useResizeObserver as useResizeObserver, $abf7096307f33b22$export$736bf165441b18c7 as useIsMobileDevice, $5b569f8bd87e05f1$re_export$useValueEffect as useValueEffect, $31dbf1c5576cf635$export$8214320346cf5104 as BreakpointProvider, $31dbf1c5576cf635$export$140ae7baa51cca23 as useMatchedBreakpoints, $31dbf1c5576cf635$export$199d6754bdf4e1e3 as useBreakpoint};
