// @react-spectrum/slider/dist/main.js
require("./main.css");
var $dfBtN$reactariautils = require("@react-aria/utils");
var $dfBtN$reactspectrumutils = require("@react-spectrum/utils");
var $dfBtN$react = require("react");
var $dfBtN$reactariai18n = require("@react-aria/i18n");
var $dfBtN$reactstatelyslider = require("@react-stately/slider");
var $dfBtN$reactspectrumprovider = require("@react-spectrum/provider");
var $dfBtN$reactariaslider = require("@react-aria/slider");
var $dfBtN$reactariafocus = require("@react-aria/focus");
var $dfBtN$reactariainteractions = require("@react-aria/interactions");
var $dfBtN$reactariavisuallyhidden = require("@react-aria/visually-hidden");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $c093838529f935d7$exports = {};

$parcel$export($c093838529f935d7$exports, "Slider", () => $c093838529f935d7$export$472062a354075cee);






var $130287b05156917f$exports = {};

$parcel$export($130287b05156917f$exports, "spectrum-Slider", () => $130287b05156917f$export$fc844a18ff837543, (v) => $130287b05156917f$export$fc844a18ff837543 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Dial", () => $130287b05156917f$export$db619d28010f1dd7, (v) => $130287b05156917f$export$db619d28010f1dd7 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider--positionTop", () => $130287b05156917f$export$3d0239f4af6225c0, (v) => $130287b05156917f$export$3d0239f4af6225c0 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider-controls", () => $130287b05156917f$export$eed070f7d409a4eb, (v) => $130287b05156917f$export$eed070f7d409a4eb = v);
$parcel$export($130287b05156917f$exports, "spectrum-Dial-controls", () => $130287b05156917f$export$9a4ea6c1b342a483, (v) => $130287b05156917f$export$9a4ea6c1b342a483 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider--positionSide", () => $130287b05156917f$export$7e6c269c6bbb2ffa, (v) => $130287b05156917f$export$7e6c269c6bbb2ffa = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider-labelContainer", () => $130287b05156917f$export$6a95ef18f88e804b, (v) => $130287b05156917f$export$6a95ef18f88e804b = v);
$parcel$export($130287b05156917f$exports, "spectrum-Dial-labelContainer", () => $130287b05156917f$export$f64ec32f85626909, (v) => $130287b05156917f$export$f64ec32f85626909 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider-valueLabelContainer", () => $130287b05156917f$export$2183ef1701a88013, (v) => $130287b05156917f$export$2183ef1701a88013 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider-value", () => $130287b05156917f$export$d1c0699c6d985532, (v) => $130287b05156917f$export$d1c0699c6d985532 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider-track", () => $130287b05156917f$export$16e664b3091a0b47, (v) => $130287b05156917f$export$16e664b3091a0b47 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider-buffer", () => $130287b05156917f$export$8ac5a980ecf86d70, (v) => $130287b05156917f$export$8ac5a980ecf86d70 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider-ramp", () => $130287b05156917f$export$74af90f7b59a4a0b, (v) => $130287b05156917f$export$74af90f7b59a4a0b = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider-fill", () => $130287b05156917f$export$79e384500d3b9c9d, (v) => $130287b05156917f$export$79e384500d3b9c9d = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider-fill--right", () => $130287b05156917f$export$4ad5f741edcaf768, (v) => $130287b05156917f$export$4ad5f741edcaf768 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider--range", () => $130287b05156917f$export$91bbf9ca430efa38, (v) => $130287b05156917f$export$91bbf9ca430efa38 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider-handle", () => $130287b05156917f$export$4e56e324ec89fe53, (v) => $130287b05156917f$export$4e56e324ec89fe53 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Dial-handle", () => $130287b05156917f$export$41c59246a9248fa9, (v) => $130287b05156917f$export$41c59246a9248fa9 = v);
$parcel$export($130287b05156917f$exports, "is-dragged", () => $130287b05156917f$export$8778c911bed6c759, (v) => $130287b05156917f$export$8778c911bed6c759 = v);
$parcel$export($130287b05156917f$exports, "is-focused", () => $130287b05156917f$export$e7dc768d35940237, (v) => $130287b05156917f$export$e7dc768d35940237 = v);
$parcel$export($130287b05156917f$exports, "is-tophandle", () => $130287b05156917f$export$e7c8b64dd65a8bd6, (v) => $130287b05156917f$export$e7c8b64dd65a8bd6 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider-input", () => $130287b05156917f$export$fdecbf00e674772b, (v) => $130287b05156917f$export$fdecbf00e674772b = v);
$parcel$export($130287b05156917f$exports, "spectrum-Dial-input", () => $130287b05156917f$export$39043e8612543a9e, (v) => $130287b05156917f$export$39043e8612543a9e = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider-label", () => $130287b05156917f$export$87d97f82d3be9540, (v) => $130287b05156917f$export$87d97f82d3be9540 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Dial-label", () => $130287b05156917f$export$d93200a6e46e9ce7, (v) => $130287b05156917f$export$d93200a6e46e9ce7 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Dial-value", () => $130287b05156917f$export$be548803c10dddf5, (v) => $130287b05156917f$export$be548803c10dddf5 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider-ticks", () => $130287b05156917f$export$8015a78273599416, (v) => $130287b05156917f$export$8015a78273599416 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider-tick", () => $130287b05156917f$export$daf7ddf4a129876d, (v) => $130287b05156917f$export$daf7ddf4a129876d = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider-tickLabel", () => $130287b05156917f$export$e045256579534563, (v) => $130287b05156917f$export$e045256579534563 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider--color", () => $130287b05156917f$export$c0271c7b757e2c6d, (v) => $130287b05156917f$export$c0271c7b757e2c6d = v);
$parcel$export($130287b05156917f$exports, "spectrum-Dial--small", () => $130287b05156917f$export$40db493959f853b2, (v) => $130287b05156917f$export$40db493959f853b2 = v);
$parcel$export($130287b05156917f$exports, "is-disabled", () => $130287b05156917f$export$d35bc1e505d1ebbf, (v) => $130287b05156917f$export$d35bc1e505d1ebbf = v);
$parcel$export($130287b05156917f$exports, "is-hovered", () => $130287b05156917f$export$b8813cd5d7824ce7, (v) => $130287b05156917f$export$b8813cd5d7824ce7 = v);
$parcel$export($130287b05156917f$exports, "u-isGrabbing", () => $130287b05156917f$export$af9e6e484daea602, (v) => $130287b05156917f$export$af9e6e484daea602 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider--filled", () => $130287b05156917f$export$743466ce5e272176, (v) => $130287b05156917f$export$743466ce5e272176 = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider--ramp", () => $130287b05156917f$export$87c47a8a21ea9aa, (v) => $130287b05156917f$export$87c47a8a21ea9aa = v);
$parcel$export($130287b05156917f$exports, "spectrum-Slider--label-side", () => $130287b05156917f$export$fc5a803c4caf36a, (v) => $130287b05156917f$export$fc5a803c4caf36a = v);
var $130287b05156917f$export$fc844a18ff837543;
var $130287b05156917f$export$db619d28010f1dd7;
var $130287b05156917f$export$3d0239f4af6225c0;
var $130287b05156917f$export$eed070f7d409a4eb;
var $130287b05156917f$export$9a4ea6c1b342a483;
var $130287b05156917f$export$7e6c269c6bbb2ffa;
var $130287b05156917f$export$6a95ef18f88e804b;
var $130287b05156917f$export$f64ec32f85626909;
var $130287b05156917f$export$2183ef1701a88013;
var $130287b05156917f$export$d1c0699c6d985532;
var $130287b05156917f$export$16e664b3091a0b47;
var $130287b05156917f$export$8ac5a980ecf86d70;
var $130287b05156917f$export$74af90f7b59a4a0b;
var $130287b05156917f$export$79e384500d3b9c9d;
var $130287b05156917f$export$4ad5f741edcaf768;
var $130287b05156917f$export$91bbf9ca430efa38;
var $130287b05156917f$export$4e56e324ec89fe53;
var $130287b05156917f$export$41c59246a9248fa9;
var $130287b05156917f$export$8778c911bed6c759;
var $130287b05156917f$export$e7dc768d35940237;
var $130287b05156917f$export$e7c8b64dd65a8bd6;
var $130287b05156917f$export$fdecbf00e674772b;
var $130287b05156917f$export$39043e8612543a9e;
var $130287b05156917f$export$87d97f82d3be9540;
var $130287b05156917f$export$d93200a6e46e9ce7;
var $130287b05156917f$export$be548803c10dddf5;
var $130287b05156917f$export$8015a78273599416;
var $130287b05156917f$export$daf7ddf4a129876d;
var $130287b05156917f$export$e045256579534563;
var $130287b05156917f$export$c0271c7b757e2c6d;
var $130287b05156917f$export$40db493959f853b2;
var $130287b05156917f$export$d35bc1e505d1ebbf;
var $130287b05156917f$export$b8813cd5d7824ce7;
var $130287b05156917f$export$af9e6e484daea602;
var $130287b05156917f$export$743466ce5e272176;
var $130287b05156917f$export$87c47a8a21ea9aa;
var $130287b05156917f$export$fc5a803c4caf36a;
$130287b05156917f$export$fc844a18ff837543 = "spectrum-Slider_e4b6ba";
$130287b05156917f$export$db619d28010f1dd7 = "spectrum-Dial_e4b6ba";
$130287b05156917f$export$3d0239f4af6225c0 = "spectrum-Slider--positionTop_e4b6ba";
$130287b05156917f$export$eed070f7d409a4eb = "spectrum-Slider-controls_e4b6ba";
$130287b05156917f$export$9a4ea6c1b342a483 = "spectrum-Dial-controls_e4b6ba";
$130287b05156917f$export$7e6c269c6bbb2ffa = "spectrum-Slider--positionSide_e4b6ba";
$130287b05156917f$export$6a95ef18f88e804b = "spectrum-Slider-labelContainer_e4b6ba";
$130287b05156917f$export$f64ec32f85626909 = "spectrum-Dial-labelContainer_e4b6ba";
$130287b05156917f$export$2183ef1701a88013 = "spectrum-Slider-valueLabelContainer_e4b6ba";
$130287b05156917f$export$d1c0699c6d985532 = "spectrum-Slider-value_e4b6ba";
$130287b05156917f$export$16e664b3091a0b47 = "spectrum-Slider-track_e4b6ba";
$130287b05156917f$export$8ac5a980ecf86d70 = "spectrum-Slider-buffer_e4b6ba";
$130287b05156917f$export$74af90f7b59a4a0b = "spectrum-Slider-ramp_e4b6ba";
$130287b05156917f$export$79e384500d3b9c9d = "spectrum-Slider-fill_e4b6ba";
$130287b05156917f$export$4ad5f741edcaf768 = "spectrum-Slider-fill--right_e4b6ba";
$130287b05156917f$export$91bbf9ca430efa38 = "spectrum-Slider--range_e4b6ba";
$130287b05156917f$export$4e56e324ec89fe53 = "spectrum-Slider-handle_e4b6ba";
$130287b05156917f$export$41c59246a9248fa9 = "spectrum-Dial-handle_e4b6ba";
$130287b05156917f$export$8778c911bed6c759 = "is-dragged_e4b6ba";
$130287b05156917f$export$e7dc768d35940237 = "is-focused_e4b6ba";
$130287b05156917f$export$e7c8b64dd65a8bd6 = "is-tophandle_e4b6ba";
$130287b05156917f$export$fdecbf00e674772b = "spectrum-Slider-input_e4b6ba";
$130287b05156917f$export$39043e8612543a9e = "spectrum-Dial-input_e4b6ba";
$130287b05156917f$export$87d97f82d3be9540 = "spectrum-Slider-label_e4b6ba";
$130287b05156917f$export$d93200a6e46e9ce7 = "spectrum-Dial-label_e4b6ba";
$130287b05156917f$export$be548803c10dddf5 = "spectrum-Dial-value_e4b6ba";
$130287b05156917f$export$8015a78273599416 = "spectrum-Slider-ticks_e4b6ba";
$130287b05156917f$export$daf7ddf4a129876d = "spectrum-Slider-tick_e4b6ba";
$130287b05156917f$export$e045256579534563 = "spectrum-Slider-tickLabel_e4b6ba";
$130287b05156917f$export$c0271c7b757e2c6d = "spectrum-Slider--color_e4b6ba";
$130287b05156917f$export$40db493959f853b2 = "spectrum-Dial--small_e4b6ba";
$130287b05156917f$export$d35bc1e505d1ebbf = "is-disabled_e4b6ba";
$130287b05156917f$export$b8813cd5d7824ce7 = "is-hovered_e4b6ba";
$130287b05156917f$export$af9e6e484daea602 = "u-isGrabbing_e4b6ba";
$130287b05156917f$export$743466ce5e272176 = "spectrum-Slider--filled_e4b6ba";
$130287b05156917f$export$87c47a8a21ea9aa = "spectrum-Slider--ramp_e4b6ba";
$130287b05156917f$export$fc5a803c4caf36a = "spectrum-Slider--label-side_e4b6ba";





function $128bb8e2e0591582$var$SliderBase(props, ref) {
    props = $dfBtN$reactspectrumprovider.useProviderProps(props);
    let { isDisabled: isDisabled , children: children , classes: classes , style: style , labelPosition: labelPosition = 'top' , getValueLabel: getValueLabel , showValueLabel: showValueLabel = !!props.label , formatOptions: formatOptions , minValue: minValue = 0 , maxValue: maxValue = 100 , ...otherProps } = props;
    let { styleProps: styleProps  } = $dfBtN$reactspectrumutils.useStyleProps(otherProps);
    // `Math.abs(Math.sign(a) - Math.sign(b)) === 2` is true if the values have a different sign.
    let alwaysDisplaySign = Math.abs(Math.sign(minValue) - Math.sign(maxValue)) === 2;
    if (alwaysDisplaySign) {
        if (formatOptions != null) {
            if (!('signDisplay' in formatOptions)) formatOptions = {
                ...formatOptions,
                // @ts-ignore
                signDisplay: 'exceptZero'
            };
        } else // @ts-ignore
        formatOptions = {
            signDisplay: 'exceptZero'
        };
    }
    const formatter = $dfBtN$reactariai18n.useNumberFormatter(formatOptions);
    const state = $dfBtN$reactstatelyslider.useSliderState({
        ...props,
        numberFormatter: formatter,
        minValue: minValue,
        maxValue: maxValue
    });
    let trackRef = $dfBtN$react.useRef();
    let { groupProps: groupProps , trackProps: trackProps , labelProps: labelProps , outputProps: outputProps  } = $dfBtN$reactariaslider.useSlider(props, state, trackRef);
    let inputRef = $dfBtN$react.useRef();
    let domRef = $dfBtN$reactspectrumutils.useFocusableRef(ref, inputRef);
    let displayValue = '';
    let maxLabelLength = undefined;
    if (typeof getValueLabel === 'function') {
        displayValue = getValueLabel(state.values);
        switch(state.values.length){
            case 1:
                maxLabelLength = Math.max(getValueLabel([
                    minValue
                ]).length, getValueLabel([
                    maxValue
                ]).length);
                break;
            case 2:
                // Try all possible combinations of min and max values.
                maxLabelLength = Math.max(getValueLabel([
                    minValue,
                    minValue
                ]).length, getValueLabel([
                    minValue,
                    maxValue
                ]).length, getValueLabel([
                    maxValue,
                    minValue
                ]).length, getValueLabel([
                    maxValue,
                    maxValue
                ]).length);
                break;
            default:
                throw new Error('Only sliders with 1 or 2 handles are supported!');
        }
    } else {
        maxLabelLength = Math.max([
            ...formatter.format(minValue)
        ].length, [
            ...formatter.format(maxValue)
        ].length);
        switch(state.values.length){
            case 1:
                displayValue = state.getThumbValueLabel(0);
                break;
            case 2:
                // This should really use the NumberFormat#formatRange proposal...
                // https://github.com/tc39/ecma402/issues/393
                // https://github.com/tc39/proposal-intl-numberformat-v3#formatrange-ecma-402-393
                displayValue = `${state.getThumbValueLabel(0)} – ${state.getThumbValueLabel(1)}`;
                maxLabelLength = 3 + 2 * Math.max(maxLabelLength, [
                    ...formatter.format(minValue)
                ].length, [
                    ...formatter.format(maxValue)
                ].length);
                break;
            default:
                throw new Error('Only sliders with 1 or 2 handles are supported!');
        }
    }
    let labelNode = /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement("label", {
        className: $dfBtN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($130287b05156917f$exports)), 'spectrum-Slider-label'),
        ...labelProps
    }, props.label);
    let valueNode = /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement("output", {
        ...outputProps,
        className: $dfBtN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($130287b05156917f$exports)), 'spectrum-Slider-value'),
        style: maxLabelLength && {
            width: `${maxLabelLength}ch`,
            minWidth: `${maxLabelLength}ch`
        }
    }, displayValue);
    return(/*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement("div", {
        ref: domRef,
        className: $dfBtN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($130287b05156917f$exports)), 'spectrum-Slider', {
            'spectrum-Slider--positionTop': labelPosition === 'top',
            'spectrum-Slider--positionSide': labelPosition === 'side',
            'is-disabled': isDisabled
        }, classes, styleProps.className),
        style: {
            ...style,
            ...styleProps.style
        },
        ...groupProps
    }, props.label && /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement("div", {
        className: $dfBtN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($130287b05156917f$exports)), 'spectrum-Slider-labelContainer'),
        role: "presentation"
    }, props.label && labelNode, labelPosition === 'top' && showValueLabel && valueNode), /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement("div", {
        className: $dfBtN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($130287b05156917f$exports)), 'spectrum-Slider-controls'),
        ref: trackRef,
        ...trackProps,
        role: "presentation"
    }, children({
        trackRef: trackRef,
        inputRef: inputRef,
        state: state
    })), labelPosition === 'side' && /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement("div", {
        className: $dfBtN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($130287b05156917f$exports)), 'spectrum-Slider-valueLabelContainer'),
        role: "presentation"
    }, showValueLabel && valueNode)));
}
const $128bb8e2e0591582$export$9418495bb635ebde = /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).forwardRef($128bb8e2e0591582$var$SliderBase);











function $dc2a2a316853e6ae$export$2c1b491743890dec(props) {
    let { index: index , inputRef: inputRef , state: state  } = props;
    let backupRef = $dfBtN$react.useRef();
    inputRef = inputRef || backupRef;
    let { thumbProps: thumbProps , inputProps: inputProps  } = $dfBtN$reactariaslider.useSliderThumb({
        ...props,
        inputRef: inputRef
    }, state);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $dfBtN$reactariainteractions.useHover({
    });
    let { direction: direction  } = $dfBtN$reactariai18n.useLocale();
    let cssDirection = direction === 'rtl' ? 'right' : 'left';
    return(/*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement($dfBtN$reactariafocus.FocusRing, {
        within: true,
        focusRingClass: $dfBtN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($130287b05156917f$exports)), 'is-focused')
    }, /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement("div", {
        className: $dfBtN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($130287b05156917f$exports)), 'spectrum-Slider-handle', {
            'is-hovered': isHovered,
            'is-dragged': state.isThumbDragging(index),
            'is-tophandle': state.focusedThumb === index
        }),
        style: {
            [cssDirection]: `${state.getThumbPercent(index) * 100}%`
        },
        ...$dfBtN$reactariautils.mergeProps(thumbProps, hoverProps),
        role: "presentation"
    }, /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement($dfBtN$reactariavisuallyhidden.VisuallyHidden, null, /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement("input", {
        className: $dfBtN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($130287b05156917f$exports)), 'spectrum-Slider-input'),
        ref: inputRef,
        ...inputProps
    })))));
}




function $c093838529f935d7$var$Slider(props, ref) {
    let { onChange: onChange , onChangeEnd: onChangeEnd , value: value , defaultValue: defaultValue , isFilled: isFilled , fillOffset: fillOffset , trackGradient: trackGradient , getValueLabel: getValueLabel , ...otherProps } = props;
    let baseProps = {
        ...otherProps,
        // Normalize `value: number[]` to `value: number`
        value: value != null ? [
            value
        ] : undefined,
        defaultValue: defaultValue != null ? [
            defaultValue
        ] : undefined,
        onChange: (v)=>{
            onChange === null || onChange === void 0 ? void 0 : onChange(v[0]);
        },
        onChangeEnd: (v)=>{
            onChangeEnd === null || onChangeEnd === void 0 ? void 0 : onChangeEnd(v[0]);
        },
        getValueLabel: getValueLabel ? ([v])=>getValueLabel(v)
         : undefined
    };
    let { direction: direction  } = $dfBtN$reactariai18n.useLocale();
    return(/*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement($128bb8e2e0591582$export$9418495bb635ebde, {
        ...baseProps,
        ref: ref,
        classes: {
            'spectrum-Slider--filled': isFilled && fillOffset == null
        },
        style: // @ts-ignore
        {
            '--spectrum-slider-track-gradient': trackGradient && `linear-gradient(to ${direction === 'ltr' ? 'right' : 'left'}, ${trackGradient.join(', ')})`
        }
    }, ({ trackRef: trackRef , inputRef: inputRef , state: state  })=>{
        fillOffset = fillOffset != null ? $dfBtN$reactariautils.clamp(fillOffset, state.getThumbMinValue(0), state.getThumbMaxValue(0)) : fillOffset;
        let cssDirection = direction === 'rtl' ? 'right' : 'left';
        let lowerTrack = /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement("div", {
            className: $dfBtN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($130287b05156917f$exports)), 'spectrum-Slider-track'),
            style: {
                width: `${state.getThumbPercent(0) * 100}%`,
                // TODO not sure if it has advantages, but this could also be implemented as CSS calc():
                // .track::before {
                //    background-size: calc((1/ (var(--width)/100)) * 100%);
                //    width: calc(var(--width) * 1%)M
                // }
                // @ts-ignore
                '--spectrum-track-background-size': `${1 / state.getThumbPercent(0) * 100}%`,
                '--spectrum-track-background-position': direction === 'ltr' ? '0' : '100%'
            }
        });
        let upperTrack = /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement("div", {
            className: $dfBtN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($130287b05156917f$exports)), 'spectrum-Slider-track'),
            style: {
                width: `${(1 - state.getThumbPercent(0)) * 100}%`,
                // @ts-ignore
                '--spectrum-track-background-size': `${1 / (1 - state.getThumbPercent(0)) * 100}%`,
                '--spectrum-track-background-position': direction === 'ltr' ? '100%' : '0'
            }
        });
        let filledTrack = null;
        if (isFilled && fillOffset != null) {
            let width = state.getThumbPercent(0) - state.getValuePercent(fillOffset);
            let isRightOfOffset = width > 0;
            let offset = isRightOfOffset ? state.getValuePercent(fillOffset) : state.getThumbPercent(0);
            filledTrack = /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement("div", {
                className: $dfBtN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($130287b05156917f$exports)), 'spectrum-Slider-fill', {
                    'spectrum-Slider-fill--right': isRightOfOffset
                }),
                style: {
                    [cssDirection]: `${offset * 100}%`,
                    width: `${Math.abs(width) * 100}%`
                }
            });
        }
        return(/*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement(($parcel$interopDefault($dfBtN$react)).Fragment, null, lowerTrack, /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement($dc2a2a316853e6ae$export$2c1b491743890dec, {
            index: 0,
            isDisabled: props.isDisabled,
            trackRef: trackRef,
            inputRef: inputRef,
            state: state
        }), upperTrack, filledTrack));
    }));
}
/**
 * Sliders allow users to quickly select a value within a range. They should be used when the upper and lower bounds to the range are invariable.
 */ const $c093838529f935d7$export$472062a354075cee = /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).forwardRef($c093838529f935d7$var$Slider);


var $77bc4557d72910aa$exports = {};

$parcel$export($77bc4557d72910aa$exports, "RangeSlider", () => $77bc4557d72910aa$export$826424dabc3dd705);

var $8f4823194e789f2a$exports = {};
var $afbae878687d22d0$exports = {};
$afbae878687d22d0$exports = JSON.parse("{\"maximum\":\"أقصى\",\"minimum\":\"أدنى\"}");


var $8a45cff7692bf866$exports = {};
$8a45cff7692bf866$exports = JSON.parse("{\"maximum\":\"Максимум\",\"minimum\":\"Минимум\"}");


var $d0d5061873899691$exports = {};
$d0d5061873899691$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $39cac25782a97424$exports = {};
$39cac25782a97424$exports = JSON.parse("{\"maximum\":\"Maksimum\",\"minimum\":\"Minimum\"}");


var $927ef2de41e1a5ef$exports = {};
$927ef2de41e1a5ef$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $4265d07b885097bb$exports = {};
$4265d07b885097bb$exports = JSON.parse("{\"maximum\":\"Μέγιστο\",\"minimum\":\"Ελάχιστο\"}");


var $a15c4ab52d24ba76$exports = {};
$a15c4ab52d24ba76$exports = JSON.parse("{\"minimum\":\"Minimum\",\"maximum\":\"Maximum\"}");


var $36f88d3d3e042e80$exports = {};
$36f88d3d3e042e80$exports = JSON.parse("{\"maximum\":\"Máximo\",\"minimum\":\"Mínimo\"}");


var $ebbbdfb387cda5a4$exports = {};
$ebbbdfb387cda5a4$exports = JSON.parse("{\"maximum\":\"Maksimaalne\",\"minimum\":\"Minimaalne\"}");


var $c76d1be8afb1b49d$exports = {};
$c76d1be8afb1b49d$exports = JSON.parse("{\"maximum\":\"Maksimi\",\"minimum\":\"Minimi\"}");


var $015a66367d1be428$exports = {};
$015a66367d1be428$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $534e983c9eda24c4$exports = {};
$534e983c9eda24c4$exports = JSON.parse("{\"maximum\":\"מקסימום\",\"minimum\":\"מינימום\"}");


var $0e0dc0980c6b8750$exports = {};
$0e0dc0980c6b8750$exports = JSON.parse("{\"maximum\":\"Najviše\",\"minimum\":\"Najmanje\"}");


var $469d0067c3eb56f9$exports = {};
$469d0067c3eb56f9$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $d59bf35f1b6b19ec$exports = {};
$d59bf35f1b6b19ec$exports = JSON.parse("{\"maximum\":\"Massimo\",\"minimum\":\"Minimo\"}");


var $4d8028a67c9fde10$exports = {};
$4d8028a67c9fde10$exports = JSON.parse("{\"maximum\":\"最大\",\"minimum\":\"最小\"}");


var $905160aa28a8bd70$exports = {};
$905160aa28a8bd70$exports = JSON.parse("{\"maximum\":\"최대\",\"minimum\":\"최소\"}");


var $e1714547a8bfba04$exports = {};
$e1714547a8bfba04$exports = JSON.parse("{\"maximum\":\"Daugiausia\",\"minimum\":\"Mažiausia\"}");


var $a381bff81c735f69$exports = {};
$a381bff81c735f69$exports = JSON.parse("{\"maximum\":\"Maksimālā vērtība\",\"minimum\":\"Minimālā vērtība\"}");


var $d23b8896f20dde1b$exports = {};
$d23b8896f20dde1b$exports = JSON.parse("{\"maximum\":\"Maksimum\",\"minimum\":\"Minimum\"}");


var $e4e96ad4ec9a87e9$exports = {};
$e4e96ad4ec9a87e9$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $b0afa5684d9f42da$exports = {};
$b0afa5684d9f42da$exports = JSON.parse("{\"maximum\":\"Maksimum\",\"minimum\":\"Minimum\"}");


var $35711c7629e7d503$exports = {};
$35711c7629e7d503$exports = JSON.parse("{\"maximum\":\"Máximo\",\"minimum\":\"Mínimo\"}");


var $cd84f60ed4be3b68$exports = {};
$cd84f60ed4be3b68$exports = JSON.parse("{\"maximum\":\"Máximo\",\"minimum\":\"Mínimo\"}");


var $38d4782ead65e4cc$exports = {};
$38d4782ead65e4cc$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $24a421571950784e$exports = {};
$24a421571950784e$exports = JSON.parse("{\"maximum\":\"Максимум\",\"minimum\":\"Минимум\"}");


var $d806c6e4a24a42f4$exports = {};
$d806c6e4a24a42f4$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $a110594a0b604c05$exports = {};
$a110594a0b604c05$exports = JSON.parse("{\"maximum\":\"Največji\",\"minimum\":\"Najmanj\"}");


var $cef15df81f815f63$exports = {};
$cef15df81f815f63$exports = JSON.parse("{\"maximum\":\"Najviše\",\"minimum\":\"Najmanje\"}");


var $f4faf67554887424$exports = {};
$f4faf67554887424$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $3cc6ae0c3901a6b1$exports = {};
$3cc6ae0c3901a6b1$exports = JSON.parse("{\"maximum\":\"Maksimum\",\"minimum\":\"Minimum\"}");


var $5b5cc1955e71c0ee$exports = {};
$5b5cc1955e71c0ee$exports = JSON.parse("{\"maximum\":\"Максимум\",\"minimum\":\"Мінімум\"}");


var $97bd628395b7acd8$exports = {};
$97bd628395b7acd8$exports = JSON.parse("{\"maximum\":\"最大\",\"minimum\":\"最小\"}");


var $7b3f3571e5212b0c$exports = {};
$7b3f3571e5212b0c$exports = JSON.parse("{\"maximum\":\"最大值\",\"minimum\":\"最小值\"}");


$8f4823194e789f2a$exports = {
    "ar-AE": $afbae878687d22d0$exports,
    "bg-BG": $8a45cff7692bf866$exports,
    "cs-CZ": $d0d5061873899691$exports,
    "da-DK": $39cac25782a97424$exports,
    "de-DE": $927ef2de41e1a5ef$exports,
    "el-GR": $4265d07b885097bb$exports,
    "en-US": $a15c4ab52d24ba76$exports,
    "es-ES": $36f88d3d3e042e80$exports,
    "et-EE": $ebbbdfb387cda5a4$exports,
    "fi-FI": $c76d1be8afb1b49d$exports,
    "fr-FR": $015a66367d1be428$exports,
    "he-IL": $534e983c9eda24c4$exports,
    "hr-HR": $0e0dc0980c6b8750$exports,
    "hu-HU": $469d0067c3eb56f9$exports,
    "it-IT": $d59bf35f1b6b19ec$exports,
    "ja-JP": $4d8028a67c9fde10$exports,
    "ko-KR": $905160aa28a8bd70$exports,
    "lt-LT": $e1714547a8bfba04$exports,
    "lv-LV": $a381bff81c735f69$exports,
    "nb-NO": $d23b8896f20dde1b$exports,
    "nl-NL": $e4e96ad4ec9a87e9$exports,
    "pl-PL": $b0afa5684d9f42da$exports,
    "pt-BR": $35711c7629e7d503$exports,
    "pt-PT": $cd84f60ed4be3b68$exports,
    "ro-RO": $38d4782ead65e4cc$exports,
    "ru-RU": $24a421571950784e$exports,
    "sk-SK": $d806c6e4a24a42f4$exports,
    "sl-SI": $a110594a0b604c05$exports,
    "sr-SP": $cef15df81f815f63$exports,
    "sv-SE": $f4faf67554887424$exports,
    "tr-TR": $3cc6ae0c3901a6b1$exports,
    "uk-UA": $5b5cc1955e71c0ee$exports,
    "zh-CN": $97bd628395b7acd8$exports,
    "zh-TW": $7b3f3571e5212b0c$exports
};







function $77bc4557d72910aa$var$RangeSlider(props, ref) {
    let { onChange: onChange , onChangeEnd: onChangeEnd , value: value , defaultValue: defaultValue , getValueLabel: getValueLabel , ...otherProps } = props;
    var _minValue, _maxValue;
    let baseProps = {
        ...otherProps,
        value: value != null ? [
            value.start,
            value.end
        ] : undefined,
        defaultValue: defaultValue != null ? [
            defaultValue.start,
            defaultValue.end
        ] : [
            (_minValue = props.minValue) !== null && _minValue !== void 0 ? _minValue : 0,
            (_maxValue = props.maxValue) !== null && _maxValue !== void 0 ? _maxValue : 100
        ],
        onChange (v) {
            onChange === null || onChange === void 0 ? void 0 : onChange({
                start: v[0],
                end: v[1]
            });
        },
        onChangeEnd (v) {
            onChangeEnd === null || onChangeEnd === void 0 ? void 0 : onChangeEnd({
                start: v[0],
                end: v[1]
            });
        },
        getValueLabel: getValueLabel ? ([start, end])=>getValueLabel({
                start: start,
                end: end
            })
         : undefined
    };
    let formatter = $dfBtN$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($8f4823194e789f2a$exports)));
    let { direction: direction  } = $dfBtN$reactariai18n.useLocale();
    return(/*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement($128bb8e2e0591582$export$9418495bb635ebde, {
        ...baseProps,
        classes: 'spectrum-Slider--range',
        ref: ref
    }, ({ trackRef: trackRef , inputRef: inputRef , state: state  })=>{
        let cssDirection = direction === 'rtl' ? 'right' : 'left';
        return(/*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement(($parcel$interopDefault($dfBtN$react)).Fragment, null, /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement("div", {
            className: $dfBtN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($130287b05156917f$exports)), 'spectrum-Slider-track'),
            style: {
                width: `${state.getThumbPercent(0) * 100}%`
            }
        }), /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement($dc2a2a316853e6ae$export$2c1b491743890dec, {
            index: 0,
            "aria-label": formatter('minimum'),
            isDisabled: props.isDisabled,
            trackRef: trackRef,
            inputRef: inputRef,
            state: state
        }), /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement("div", {
            className: $dfBtN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($130287b05156917f$exports)), 'spectrum-Slider-track'),
            style: {
                [cssDirection]: `${state.getThumbPercent(0) * 100}%`,
                width: `${Math.abs(state.getThumbPercent(0) - state.getThumbPercent(1)) * 100}%`
            }
        }), /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement($dc2a2a316853e6ae$export$2c1b491743890dec, {
            index: 1,
            "aria-label": formatter('maximum'),
            isDisabled: props.isDisabled,
            trackRef: trackRef,
            state: state
        }), /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).createElement("div", {
            className: $dfBtN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($130287b05156917f$exports)), 'spectrum-Slider-track'),
            style: {
                width: `${(1 - state.getThumbPercent(1)) * 100}%`
            }
        })));
    }));
}
/**
 * RangeSliders allow users to quickly select a subset range. They should be used when the upper and lower bounds to the range are invariable.
 */ const $77bc4557d72910aa$export$826424dabc3dd705 = /*#__PURE__*/ ($parcel$interopDefault($dfBtN$react)).forwardRef($77bc4557d72910aa$var$RangeSlider);


$parcel$exportWildcard(module.exports, $c093838529f935d7$exports);
$parcel$exportWildcard(module.exports, $77bc4557d72910aa$exports);
