// @react-spectrum/slider/dist/module.js
import "./main.css";
import {clamp as $hMsoe$clamp, mergeProps as $hMsoe$mergeProps} from "@react-aria/utils";
import {classNames as $hMsoe$classNames, useStyleProps as $hMsoe$useStyleProps, useFocusableRef as $hMsoe$useFocusableRef} from "@react-spectrum/utils";
import $hMsoe$react, {useRef as $hMsoe$useRef} from "react";
import {useLocale as $hMsoe$useLocale, useNumberFormatter as $hMsoe$useNumberFormatter, useMessageFormatter as $hMsoe$useMessageFormatter} from "@react-aria/i18n";
import {useSliderState as $hMsoe$useSliderState} from "@react-stately/slider";
import {useProviderProps as $hMsoe$useProviderProps} from "@react-spectrum/provider";
import {useSlider as $hMsoe$useSlider, useSliderThumb as $hMsoe$useSliderThumb} from "@react-aria/slider";
import {FocusRing as $hMsoe$FocusRing} from "@react-aria/focus";
import {useHover as $hMsoe$useHover} from "@react-aria/interactions";
import {VisuallyHidden as $hMsoe$VisuallyHidden} from "@react-aria/visually-hidden";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $b48a457725a03d8f$exports = {};

$parcel$export($b48a457725a03d8f$exports, "Slider", () => $b48a457725a03d8f$export$472062a354075cee);






var $f6685e21d89782f2$exports = {};

$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider", () => $f6685e21d89782f2$export$fc844a18ff837543, (v) => $f6685e21d89782f2$export$fc844a18ff837543 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Dial", () => $f6685e21d89782f2$export$db619d28010f1dd7, (v) => $f6685e21d89782f2$export$db619d28010f1dd7 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider--positionTop", () => $f6685e21d89782f2$export$3d0239f4af6225c0, (v) => $f6685e21d89782f2$export$3d0239f4af6225c0 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider-controls", () => $f6685e21d89782f2$export$eed070f7d409a4eb, (v) => $f6685e21d89782f2$export$eed070f7d409a4eb = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Dial-controls", () => $f6685e21d89782f2$export$9a4ea6c1b342a483, (v) => $f6685e21d89782f2$export$9a4ea6c1b342a483 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider--positionSide", () => $f6685e21d89782f2$export$7e6c269c6bbb2ffa, (v) => $f6685e21d89782f2$export$7e6c269c6bbb2ffa = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider-labelContainer", () => $f6685e21d89782f2$export$6a95ef18f88e804b, (v) => $f6685e21d89782f2$export$6a95ef18f88e804b = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Dial-labelContainer", () => $f6685e21d89782f2$export$f64ec32f85626909, (v) => $f6685e21d89782f2$export$f64ec32f85626909 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider-valueLabelContainer", () => $f6685e21d89782f2$export$2183ef1701a88013, (v) => $f6685e21d89782f2$export$2183ef1701a88013 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider-value", () => $f6685e21d89782f2$export$d1c0699c6d985532, (v) => $f6685e21d89782f2$export$d1c0699c6d985532 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider-track", () => $f6685e21d89782f2$export$16e664b3091a0b47, (v) => $f6685e21d89782f2$export$16e664b3091a0b47 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider-buffer", () => $f6685e21d89782f2$export$8ac5a980ecf86d70, (v) => $f6685e21d89782f2$export$8ac5a980ecf86d70 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider-ramp", () => $f6685e21d89782f2$export$74af90f7b59a4a0b, (v) => $f6685e21d89782f2$export$74af90f7b59a4a0b = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider-fill", () => $f6685e21d89782f2$export$79e384500d3b9c9d, (v) => $f6685e21d89782f2$export$79e384500d3b9c9d = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider-fill--right", () => $f6685e21d89782f2$export$4ad5f741edcaf768, (v) => $f6685e21d89782f2$export$4ad5f741edcaf768 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider--range", () => $f6685e21d89782f2$export$91bbf9ca430efa38, (v) => $f6685e21d89782f2$export$91bbf9ca430efa38 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider-handle", () => $f6685e21d89782f2$export$4e56e324ec89fe53, (v) => $f6685e21d89782f2$export$4e56e324ec89fe53 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Dial-handle", () => $f6685e21d89782f2$export$41c59246a9248fa9, (v) => $f6685e21d89782f2$export$41c59246a9248fa9 = v);
$parcel$export($f6685e21d89782f2$exports, "is-dragged", () => $f6685e21d89782f2$export$8778c911bed6c759, (v) => $f6685e21d89782f2$export$8778c911bed6c759 = v);
$parcel$export($f6685e21d89782f2$exports, "is-focused", () => $f6685e21d89782f2$export$e7dc768d35940237, (v) => $f6685e21d89782f2$export$e7dc768d35940237 = v);
$parcel$export($f6685e21d89782f2$exports, "is-tophandle", () => $f6685e21d89782f2$export$e7c8b64dd65a8bd6, (v) => $f6685e21d89782f2$export$e7c8b64dd65a8bd6 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider-input", () => $f6685e21d89782f2$export$fdecbf00e674772b, (v) => $f6685e21d89782f2$export$fdecbf00e674772b = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Dial-input", () => $f6685e21d89782f2$export$39043e8612543a9e, (v) => $f6685e21d89782f2$export$39043e8612543a9e = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider-label", () => $f6685e21d89782f2$export$87d97f82d3be9540, (v) => $f6685e21d89782f2$export$87d97f82d3be9540 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Dial-label", () => $f6685e21d89782f2$export$d93200a6e46e9ce7, (v) => $f6685e21d89782f2$export$d93200a6e46e9ce7 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Dial-value", () => $f6685e21d89782f2$export$be548803c10dddf5, (v) => $f6685e21d89782f2$export$be548803c10dddf5 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider-ticks", () => $f6685e21d89782f2$export$8015a78273599416, (v) => $f6685e21d89782f2$export$8015a78273599416 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider-tick", () => $f6685e21d89782f2$export$daf7ddf4a129876d, (v) => $f6685e21d89782f2$export$daf7ddf4a129876d = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider-tickLabel", () => $f6685e21d89782f2$export$e045256579534563, (v) => $f6685e21d89782f2$export$e045256579534563 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider--color", () => $f6685e21d89782f2$export$c0271c7b757e2c6d, (v) => $f6685e21d89782f2$export$c0271c7b757e2c6d = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Dial--small", () => $f6685e21d89782f2$export$40db493959f853b2, (v) => $f6685e21d89782f2$export$40db493959f853b2 = v);
$parcel$export($f6685e21d89782f2$exports, "is-disabled", () => $f6685e21d89782f2$export$d35bc1e505d1ebbf, (v) => $f6685e21d89782f2$export$d35bc1e505d1ebbf = v);
$parcel$export($f6685e21d89782f2$exports, "is-hovered", () => $f6685e21d89782f2$export$b8813cd5d7824ce7, (v) => $f6685e21d89782f2$export$b8813cd5d7824ce7 = v);
$parcel$export($f6685e21d89782f2$exports, "u-isGrabbing", () => $f6685e21d89782f2$export$af9e6e484daea602, (v) => $f6685e21d89782f2$export$af9e6e484daea602 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider--filled", () => $f6685e21d89782f2$export$743466ce5e272176, (v) => $f6685e21d89782f2$export$743466ce5e272176 = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider--ramp", () => $f6685e21d89782f2$export$87c47a8a21ea9aa, (v) => $f6685e21d89782f2$export$87c47a8a21ea9aa = v);
$parcel$export($f6685e21d89782f2$exports, "spectrum-Slider--label-side", () => $f6685e21d89782f2$export$fc5a803c4caf36a, (v) => $f6685e21d89782f2$export$fc5a803c4caf36a = v);
var $f6685e21d89782f2$export$fc844a18ff837543;
var $f6685e21d89782f2$export$db619d28010f1dd7;
var $f6685e21d89782f2$export$3d0239f4af6225c0;
var $f6685e21d89782f2$export$eed070f7d409a4eb;
var $f6685e21d89782f2$export$9a4ea6c1b342a483;
var $f6685e21d89782f2$export$7e6c269c6bbb2ffa;
var $f6685e21d89782f2$export$6a95ef18f88e804b;
var $f6685e21d89782f2$export$f64ec32f85626909;
var $f6685e21d89782f2$export$2183ef1701a88013;
var $f6685e21d89782f2$export$d1c0699c6d985532;
var $f6685e21d89782f2$export$16e664b3091a0b47;
var $f6685e21d89782f2$export$8ac5a980ecf86d70;
var $f6685e21d89782f2$export$74af90f7b59a4a0b;
var $f6685e21d89782f2$export$79e384500d3b9c9d;
var $f6685e21d89782f2$export$4ad5f741edcaf768;
var $f6685e21d89782f2$export$91bbf9ca430efa38;
var $f6685e21d89782f2$export$4e56e324ec89fe53;
var $f6685e21d89782f2$export$41c59246a9248fa9;
var $f6685e21d89782f2$export$8778c911bed6c759;
var $f6685e21d89782f2$export$e7dc768d35940237;
var $f6685e21d89782f2$export$e7c8b64dd65a8bd6;
var $f6685e21d89782f2$export$fdecbf00e674772b;
var $f6685e21d89782f2$export$39043e8612543a9e;
var $f6685e21d89782f2$export$87d97f82d3be9540;
var $f6685e21d89782f2$export$d93200a6e46e9ce7;
var $f6685e21d89782f2$export$be548803c10dddf5;
var $f6685e21d89782f2$export$8015a78273599416;
var $f6685e21d89782f2$export$daf7ddf4a129876d;
var $f6685e21d89782f2$export$e045256579534563;
var $f6685e21d89782f2$export$c0271c7b757e2c6d;
var $f6685e21d89782f2$export$40db493959f853b2;
var $f6685e21d89782f2$export$d35bc1e505d1ebbf;
var $f6685e21d89782f2$export$b8813cd5d7824ce7;
var $f6685e21d89782f2$export$af9e6e484daea602;
var $f6685e21d89782f2$export$743466ce5e272176;
var $f6685e21d89782f2$export$87c47a8a21ea9aa;
var $f6685e21d89782f2$export$fc5a803c4caf36a;
$f6685e21d89782f2$export$fc844a18ff837543 = "spectrum-Slider_e4b6ba";
$f6685e21d89782f2$export$db619d28010f1dd7 = "spectrum-Dial_e4b6ba";
$f6685e21d89782f2$export$3d0239f4af6225c0 = "spectrum-Slider--positionTop_e4b6ba";
$f6685e21d89782f2$export$eed070f7d409a4eb = "spectrum-Slider-controls_e4b6ba";
$f6685e21d89782f2$export$9a4ea6c1b342a483 = "spectrum-Dial-controls_e4b6ba";
$f6685e21d89782f2$export$7e6c269c6bbb2ffa = "spectrum-Slider--positionSide_e4b6ba";
$f6685e21d89782f2$export$6a95ef18f88e804b = "spectrum-Slider-labelContainer_e4b6ba";
$f6685e21d89782f2$export$f64ec32f85626909 = "spectrum-Dial-labelContainer_e4b6ba";
$f6685e21d89782f2$export$2183ef1701a88013 = "spectrum-Slider-valueLabelContainer_e4b6ba";
$f6685e21d89782f2$export$d1c0699c6d985532 = "spectrum-Slider-value_e4b6ba";
$f6685e21d89782f2$export$16e664b3091a0b47 = "spectrum-Slider-track_e4b6ba";
$f6685e21d89782f2$export$8ac5a980ecf86d70 = "spectrum-Slider-buffer_e4b6ba";
$f6685e21d89782f2$export$74af90f7b59a4a0b = "spectrum-Slider-ramp_e4b6ba";
$f6685e21d89782f2$export$79e384500d3b9c9d = "spectrum-Slider-fill_e4b6ba";
$f6685e21d89782f2$export$4ad5f741edcaf768 = "spectrum-Slider-fill--right_e4b6ba";
$f6685e21d89782f2$export$91bbf9ca430efa38 = "spectrum-Slider--range_e4b6ba";
$f6685e21d89782f2$export$4e56e324ec89fe53 = "spectrum-Slider-handle_e4b6ba";
$f6685e21d89782f2$export$41c59246a9248fa9 = "spectrum-Dial-handle_e4b6ba";
$f6685e21d89782f2$export$8778c911bed6c759 = "is-dragged_e4b6ba";
$f6685e21d89782f2$export$e7dc768d35940237 = "is-focused_e4b6ba";
$f6685e21d89782f2$export$e7c8b64dd65a8bd6 = "is-tophandle_e4b6ba";
$f6685e21d89782f2$export$fdecbf00e674772b = "spectrum-Slider-input_e4b6ba";
$f6685e21d89782f2$export$39043e8612543a9e = "spectrum-Dial-input_e4b6ba";
$f6685e21d89782f2$export$87d97f82d3be9540 = "spectrum-Slider-label_e4b6ba";
$f6685e21d89782f2$export$d93200a6e46e9ce7 = "spectrum-Dial-label_e4b6ba";
$f6685e21d89782f2$export$be548803c10dddf5 = "spectrum-Dial-value_e4b6ba";
$f6685e21d89782f2$export$8015a78273599416 = "spectrum-Slider-ticks_e4b6ba";
$f6685e21d89782f2$export$daf7ddf4a129876d = "spectrum-Slider-tick_e4b6ba";
$f6685e21d89782f2$export$e045256579534563 = "spectrum-Slider-tickLabel_e4b6ba";
$f6685e21d89782f2$export$c0271c7b757e2c6d = "spectrum-Slider--color_e4b6ba";
$f6685e21d89782f2$export$40db493959f853b2 = "spectrum-Dial--small_e4b6ba";
$f6685e21d89782f2$export$d35bc1e505d1ebbf = "is-disabled_e4b6ba";
$f6685e21d89782f2$export$b8813cd5d7824ce7 = "is-hovered_e4b6ba";
$f6685e21d89782f2$export$af9e6e484daea602 = "u-isGrabbing_e4b6ba";
$f6685e21d89782f2$export$743466ce5e272176 = "spectrum-Slider--filled_e4b6ba";
$f6685e21d89782f2$export$87c47a8a21ea9aa = "spectrum-Slider--ramp_e4b6ba";
$f6685e21d89782f2$export$fc5a803c4caf36a = "spectrum-Slider--label-side_e4b6ba";





function $3dd7a0627a64a325$var$SliderBase(props, ref) {
    props = $hMsoe$useProviderProps(props);
    let { isDisabled: isDisabled , children: children , classes: classes , style: style , labelPosition: labelPosition = 'top' , getValueLabel: getValueLabel , showValueLabel: showValueLabel = !!props.label , formatOptions: formatOptions , minValue: minValue = 0 , maxValue: maxValue = 100 , ...otherProps } = props;
    let { styleProps: styleProps  } = $hMsoe$useStyleProps(otherProps);
    // `Math.abs(Math.sign(a) - Math.sign(b)) === 2` is true if the values have a different sign.
    let alwaysDisplaySign = Math.abs(Math.sign(minValue) - Math.sign(maxValue)) === 2;
    if (alwaysDisplaySign) {
        if (formatOptions != null) {
            if (!('signDisplay' in formatOptions)) formatOptions = {
                ...formatOptions,
                // @ts-ignore
                signDisplay: 'exceptZero'
            };
        } else // @ts-ignore
        formatOptions = {
            signDisplay: 'exceptZero'
        };
    }
    const formatter = $hMsoe$useNumberFormatter(formatOptions);
    const state = $hMsoe$useSliderState({
        ...props,
        numberFormatter: formatter,
        minValue: minValue,
        maxValue: maxValue
    });
    let trackRef = $hMsoe$useRef();
    let { groupProps: groupProps , trackProps: trackProps , labelProps: labelProps , outputProps: outputProps  } = $hMsoe$useSlider(props, state, trackRef);
    let inputRef = $hMsoe$useRef();
    let domRef = $hMsoe$useFocusableRef(ref, inputRef);
    let displayValue = '';
    let maxLabelLength = undefined;
    if (typeof getValueLabel === 'function') {
        displayValue = getValueLabel(state.values);
        switch(state.values.length){
            case 1:
                maxLabelLength = Math.max(getValueLabel([
                    minValue
                ]).length, getValueLabel([
                    maxValue
                ]).length);
                break;
            case 2:
                // Try all possible combinations of min and max values.
                maxLabelLength = Math.max(getValueLabel([
                    minValue,
                    minValue
                ]).length, getValueLabel([
                    minValue,
                    maxValue
                ]).length, getValueLabel([
                    maxValue,
                    minValue
                ]).length, getValueLabel([
                    maxValue,
                    maxValue
                ]).length);
                break;
            default:
                throw new Error('Only sliders with 1 or 2 handles are supported!');
        }
    } else {
        maxLabelLength = Math.max([
            ...formatter.format(minValue)
        ].length, [
            ...formatter.format(maxValue)
        ].length);
        switch(state.values.length){
            case 1:
                displayValue = state.getThumbValueLabel(0);
                break;
            case 2:
                // This should really use the NumberFormat#formatRange proposal...
                // https://github.com/tc39/ecma402/issues/393
                // https://github.com/tc39/proposal-intl-numberformat-v3#formatrange-ecma-402-393
                displayValue = `${state.getThumbValueLabel(0)} – ${state.getThumbValueLabel(1)}`;
                maxLabelLength = 3 + 2 * Math.max(maxLabelLength, [
                    ...formatter.format(minValue)
                ].length, [
                    ...formatter.format(maxValue)
                ].length);
                break;
            default:
                throw new Error('Only sliders with 1 or 2 handles are supported!');
        }
    }
    let labelNode = /*#__PURE__*/ $hMsoe$react.createElement("label", {
        className: $hMsoe$classNames((/*@__PURE__*/$parcel$interopDefault($f6685e21d89782f2$exports)), 'spectrum-Slider-label'),
        ...labelProps
    }, props.label);
    let valueNode = /*#__PURE__*/ $hMsoe$react.createElement("output", {
        ...outputProps,
        className: $hMsoe$classNames((/*@__PURE__*/$parcel$interopDefault($f6685e21d89782f2$exports)), 'spectrum-Slider-value'),
        style: maxLabelLength && {
            width: `${maxLabelLength}ch`,
            minWidth: `${maxLabelLength}ch`
        }
    }, displayValue);
    return(/*#__PURE__*/ $hMsoe$react.createElement("div", {
        ref: domRef,
        className: $hMsoe$classNames((/*@__PURE__*/$parcel$interopDefault($f6685e21d89782f2$exports)), 'spectrum-Slider', {
            'spectrum-Slider--positionTop': labelPosition === 'top',
            'spectrum-Slider--positionSide': labelPosition === 'side',
            'is-disabled': isDisabled
        }, classes, styleProps.className),
        style: {
            ...style,
            ...styleProps.style
        },
        ...groupProps
    }, props.label && /*#__PURE__*/ $hMsoe$react.createElement("div", {
        className: $hMsoe$classNames((/*@__PURE__*/$parcel$interopDefault($f6685e21d89782f2$exports)), 'spectrum-Slider-labelContainer'),
        role: "presentation"
    }, props.label && labelNode, labelPosition === 'top' && showValueLabel && valueNode), /*#__PURE__*/ $hMsoe$react.createElement("div", {
        className: $hMsoe$classNames((/*@__PURE__*/$parcel$interopDefault($f6685e21d89782f2$exports)), 'spectrum-Slider-controls'),
        ref: trackRef,
        ...trackProps,
        role: "presentation"
    }, children({
        trackRef: trackRef,
        inputRef: inputRef,
        state: state
    })), labelPosition === 'side' && /*#__PURE__*/ $hMsoe$react.createElement("div", {
        className: $hMsoe$classNames((/*@__PURE__*/$parcel$interopDefault($f6685e21d89782f2$exports)), 'spectrum-Slider-valueLabelContainer'),
        role: "presentation"
    }, showValueLabel && valueNode)));
}
const $3dd7a0627a64a325$export$9418495bb635ebde = /*#__PURE__*/ $hMsoe$react.forwardRef($3dd7a0627a64a325$var$SliderBase);











function $4a1b4dbf5b2701f7$export$2c1b491743890dec(props) {
    let { index: index , inputRef: inputRef , state: state  } = props;
    let backupRef = $hMsoe$useRef();
    inputRef = inputRef || backupRef;
    let { thumbProps: thumbProps , inputProps: inputProps  } = $hMsoe$useSliderThumb({
        ...props,
        inputRef: inputRef
    }, state);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $hMsoe$useHover({
    });
    let { direction: direction  } = $hMsoe$useLocale();
    let cssDirection = direction === 'rtl' ? 'right' : 'left';
    return(/*#__PURE__*/ $hMsoe$react.createElement($hMsoe$FocusRing, {
        within: true,
        focusRingClass: $hMsoe$classNames((/*@__PURE__*/$parcel$interopDefault($f6685e21d89782f2$exports)), 'is-focused')
    }, /*#__PURE__*/ $hMsoe$react.createElement("div", {
        className: $hMsoe$classNames((/*@__PURE__*/$parcel$interopDefault($f6685e21d89782f2$exports)), 'spectrum-Slider-handle', {
            'is-hovered': isHovered,
            'is-dragged': state.isThumbDragging(index),
            'is-tophandle': state.focusedThumb === index
        }),
        style: {
            [cssDirection]: `${state.getThumbPercent(index) * 100}%`
        },
        ...$hMsoe$mergeProps(thumbProps, hoverProps),
        role: "presentation"
    }, /*#__PURE__*/ $hMsoe$react.createElement($hMsoe$VisuallyHidden, null, /*#__PURE__*/ $hMsoe$react.createElement("input", {
        className: $hMsoe$classNames((/*@__PURE__*/$parcel$interopDefault($f6685e21d89782f2$exports)), 'spectrum-Slider-input'),
        ref: inputRef,
        ...inputProps
    })))));
}




function $b48a457725a03d8f$var$Slider(props, ref) {
    let { onChange: onChange , onChangeEnd: onChangeEnd , value: value , defaultValue: defaultValue , isFilled: isFilled , fillOffset: fillOffset , trackGradient: trackGradient , getValueLabel: getValueLabel , ...otherProps } = props;
    let baseProps = {
        ...otherProps,
        // Normalize `value: number[]` to `value: number`
        value: value != null ? [
            value
        ] : undefined,
        defaultValue: defaultValue != null ? [
            defaultValue
        ] : undefined,
        onChange: (v)=>{
            onChange === null || onChange === void 0 ? void 0 : onChange(v[0]);
        },
        onChangeEnd: (v)=>{
            onChangeEnd === null || onChangeEnd === void 0 ? void 0 : onChangeEnd(v[0]);
        },
        getValueLabel: getValueLabel ? ([v])=>getValueLabel(v)
         : undefined
    };
    let { direction: direction  } = $hMsoe$useLocale();
    return(/*#__PURE__*/ $hMsoe$react.createElement($3dd7a0627a64a325$export$9418495bb635ebde, {
        ...baseProps,
        ref: ref,
        classes: {
            'spectrum-Slider--filled': isFilled && fillOffset == null
        },
        style: // @ts-ignore
        {
            '--spectrum-slider-track-gradient': trackGradient && `linear-gradient(to ${direction === 'ltr' ? 'right' : 'left'}, ${trackGradient.join(', ')})`
        }
    }, ({ trackRef: trackRef , inputRef: inputRef , state: state  })=>{
        fillOffset = fillOffset != null ? $hMsoe$clamp(fillOffset, state.getThumbMinValue(0), state.getThumbMaxValue(0)) : fillOffset;
        let cssDirection = direction === 'rtl' ? 'right' : 'left';
        let lowerTrack = /*#__PURE__*/ $hMsoe$react.createElement("div", {
            className: $hMsoe$classNames((/*@__PURE__*/$parcel$interopDefault($f6685e21d89782f2$exports)), 'spectrum-Slider-track'),
            style: {
                width: `${state.getThumbPercent(0) * 100}%`,
                // TODO not sure if it has advantages, but this could also be implemented as CSS calc():
                // .track::before {
                //    background-size: calc((1/ (var(--width)/100)) * 100%);
                //    width: calc(var(--width) * 1%)M
                // }
                // @ts-ignore
                '--spectrum-track-background-size': `${1 / state.getThumbPercent(0) * 100}%`,
                '--spectrum-track-background-position': direction === 'ltr' ? '0' : '100%'
            }
        });
        let upperTrack = /*#__PURE__*/ $hMsoe$react.createElement("div", {
            className: $hMsoe$classNames((/*@__PURE__*/$parcel$interopDefault($f6685e21d89782f2$exports)), 'spectrum-Slider-track'),
            style: {
                width: `${(1 - state.getThumbPercent(0)) * 100}%`,
                // @ts-ignore
                '--spectrum-track-background-size': `${1 / (1 - state.getThumbPercent(0)) * 100}%`,
                '--spectrum-track-background-position': direction === 'ltr' ? '100%' : '0'
            }
        });
        let filledTrack = null;
        if (isFilled && fillOffset != null) {
            let width = state.getThumbPercent(0) - state.getValuePercent(fillOffset);
            let isRightOfOffset = width > 0;
            let offset = isRightOfOffset ? state.getValuePercent(fillOffset) : state.getThumbPercent(0);
            filledTrack = /*#__PURE__*/ $hMsoe$react.createElement("div", {
                className: $hMsoe$classNames((/*@__PURE__*/$parcel$interopDefault($f6685e21d89782f2$exports)), 'spectrum-Slider-fill', {
                    'spectrum-Slider-fill--right': isRightOfOffset
                }),
                style: {
                    [cssDirection]: `${offset * 100}%`,
                    width: `${Math.abs(width) * 100}%`
                }
            });
        }
        return(/*#__PURE__*/ $hMsoe$react.createElement($hMsoe$react.Fragment, null, lowerTrack, /*#__PURE__*/ $hMsoe$react.createElement($4a1b4dbf5b2701f7$export$2c1b491743890dec, {
            index: 0,
            isDisabled: props.isDisabled,
            trackRef: trackRef,
            inputRef: inputRef,
            state: state
        }), upperTrack, filledTrack));
    }));
}
/**
 * Sliders allow users to quickly select a value within a range. They should be used when the upper and lower bounds to the range are invariable.
 */ const $b48a457725a03d8f$export$472062a354075cee = /*#__PURE__*/ $hMsoe$react.forwardRef($b48a457725a03d8f$var$Slider);


var $af48336cc068c938$exports = {};

$parcel$export($af48336cc068c938$exports, "RangeSlider", () => $af48336cc068c938$export$826424dabc3dd705);

var $56bc35cee43d0a96$exports = {};
var $e70fa802cd62850f$exports = {};
$e70fa802cd62850f$exports = JSON.parse("{\"maximum\":\"أقصى\",\"minimum\":\"أدنى\"}");


var $c5305d0c63208513$exports = {};
$c5305d0c63208513$exports = JSON.parse("{\"maximum\":\"Максимум\",\"minimum\":\"Минимум\"}");


var $c493ee0975a7e4e1$exports = {};
$c493ee0975a7e4e1$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $7e974560d9c03619$exports = {};
$7e974560d9c03619$exports = JSON.parse("{\"maximum\":\"Maksimum\",\"minimum\":\"Minimum\"}");


var $0b5611b85ddde06a$exports = {};
$0b5611b85ddde06a$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $4f20d261b54669f1$exports = {};
$4f20d261b54669f1$exports = JSON.parse("{\"maximum\":\"Μέγιστο\",\"minimum\":\"Ελάχιστο\"}");


var $bc112e3cebbb3b07$exports = {};
$bc112e3cebbb3b07$exports = JSON.parse("{\"minimum\":\"Minimum\",\"maximum\":\"Maximum\"}");


var $790c5de70e9ef7b5$exports = {};
$790c5de70e9ef7b5$exports = JSON.parse("{\"maximum\":\"Máximo\",\"minimum\":\"Mínimo\"}");


var $6a03307048cc26d5$exports = {};
$6a03307048cc26d5$exports = JSON.parse("{\"maximum\":\"Maksimaalne\",\"minimum\":\"Minimaalne\"}");


var $5a13a379ee14c4eb$exports = {};
$5a13a379ee14c4eb$exports = JSON.parse("{\"maximum\":\"Maksimi\",\"minimum\":\"Minimi\"}");


var $dc26f518641b6b77$exports = {};
$dc26f518641b6b77$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $ad6a4ec77462025e$exports = {};
$ad6a4ec77462025e$exports = JSON.parse("{\"maximum\":\"מקסימום\",\"minimum\":\"מינימום\"}");


var $c4c4d1619844926d$exports = {};
$c4c4d1619844926d$exports = JSON.parse("{\"maximum\":\"Najviše\",\"minimum\":\"Najmanje\"}");


var $f3d9dfcb2828d7cf$exports = {};
$f3d9dfcb2828d7cf$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $23b86d63df559407$exports = {};
$23b86d63df559407$exports = JSON.parse("{\"maximum\":\"Massimo\",\"minimum\":\"Minimo\"}");


var $5866e882bbd1a905$exports = {};
$5866e882bbd1a905$exports = JSON.parse("{\"maximum\":\"最大\",\"minimum\":\"最小\"}");


var $c66dc37189d83b05$exports = {};
$c66dc37189d83b05$exports = JSON.parse("{\"maximum\":\"최대\",\"minimum\":\"최소\"}");


var $64060cfd7f53a4ad$exports = {};
$64060cfd7f53a4ad$exports = JSON.parse("{\"maximum\":\"Daugiausia\",\"minimum\":\"Mažiausia\"}");


var $933de1f060fd4728$exports = {};
$933de1f060fd4728$exports = JSON.parse("{\"maximum\":\"Maksimālā vērtība\",\"minimum\":\"Minimālā vērtība\"}");


var $e6d150b173d33537$exports = {};
$e6d150b173d33537$exports = JSON.parse("{\"maximum\":\"Maksimum\",\"minimum\":\"Minimum\"}");


var $c654be22d188eb67$exports = {};
$c654be22d188eb67$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $595acdd4dfa020e3$exports = {};
$595acdd4dfa020e3$exports = JSON.parse("{\"maximum\":\"Maksimum\",\"minimum\":\"Minimum\"}");


var $3cb02588657d1914$exports = {};
$3cb02588657d1914$exports = JSON.parse("{\"maximum\":\"Máximo\",\"minimum\":\"Mínimo\"}");


var $ea1ffb313093bb81$exports = {};
$ea1ffb313093bb81$exports = JSON.parse("{\"maximum\":\"Máximo\",\"minimum\":\"Mínimo\"}");


var $c628b515a0065540$exports = {};
$c628b515a0065540$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $20260ab29a599681$exports = {};
$20260ab29a599681$exports = JSON.parse("{\"maximum\":\"Максимум\",\"minimum\":\"Минимум\"}");


var $798ac2f2c238a2a9$exports = {};
$798ac2f2c238a2a9$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $cc59db37f5cdb9c4$exports = {};
$cc59db37f5cdb9c4$exports = JSON.parse("{\"maximum\":\"Največji\",\"minimum\":\"Najmanj\"}");


var $0e356fc23b1ec7d4$exports = {};
$0e356fc23b1ec7d4$exports = JSON.parse("{\"maximum\":\"Najviše\",\"minimum\":\"Najmanje\"}");


var $34570d94fde9fcc2$exports = {};
$34570d94fde9fcc2$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $2edabe82af80e671$exports = {};
$2edabe82af80e671$exports = JSON.parse("{\"maximum\":\"Maksimum\",\"minimum\":\"Minimum\"}");


var $338172cf7c3663fa$exports = {};
$338172cf7c3663fa$exports = JSON.parse("{\"maximum\":\"Максимум\",\"minimum\":\"Мінімум\"}");


var $654352ad02fecb42$exports = {};
$654352ad02fecb42$exports = JSON.parse("{\"maximum\":\"最大\",\"minimum\":\"最小\"}");


var $b24eb6a67ee3ab96$exports = {};
$b24eb6a67ee3ab96$exports = JSON.parse("{\"maximum\":\"最大值\",\"minimum\":\"最小值\"}");


$56bc35cee43d0a96$exports = {
    "ar-AE": $e70fa802cd62850f$exports,
    "bg-BG": $c5305d0c63208513$exports,
    "cs-CZ": $c493ee0975a7e4e1$exports,
    "da-DK": $7e974560d9c03619$exports,
    "de-DE": $0b5611b85ddde06a$exports,
    "el-GR": $4f20d261b54669f1$exports,
    "en-US": $bc112e3cebbb3b07$exports,
    "es-ES": $790c5de70e9ef7b5$exports,
    "et-EE": $6a03307048cc26d5$exports,
    "fi-FI": $5a13a379ee14c4eb$exports,
    "fr-FR": $dc26f518641b6b77$exports,
    "he-IL": $ad6a4ec77462025e$exports,
    "hr-HR": $c4c4d1619844926d$exports,
    "hu-HU": $f3d9dfcb2828d7cf$exports,
    "it-IT": $23b86d63df559407$exports,
    "ja-JP": $5866e882bbd1a905$exports,
    "ko-KR": $c66dc37189d83b05$exports,
    "lt-LT": $64060cfd7f53a4ad$exports,
    "lv-LV": $933de1f060fd4728$exports,
    "nb-NO": $e6d150b173d33537$exports,
    "nl-NL": $c654be22d188eb67$exports,
    "pl-PL": $595acdd4dfa020e3$exports,
    "pt-BR": $3cb02588657d1914$exports,
    "pt-PT": $ea1ffb313093bb81$exports,
    "ro-RO": $c628b515a0065540$exports,
    "ru-RU": $20260ab29a599681$exports,
    "sk-SK": $798ac2f2c238a2a9$exports,
    "sl-SI": $cc59db37f5cdb9c4$exports,
    "sr-SP": $0e356fc23b1ec7d4$exports,
    "sv-SE": $34570d94fde9fcc2$exports,
    "tr-TR": $2edabe82af80e671$exports,
    "uk-UA": $338172cf7c3663fa$exports,
    "zh-CN": $654352ad02fecb42$exports,
    "zh-TW": $b24eb6a67ee3ab96$exports
};







function $af48336cc068c938$var$RangeSlider(props, ref) {
    let { onChange: onChange , onChangeEnd: onChangeEnd , value: value , defaultValue: defaultValue , getValueLabel: getValueLabel , ...otherProps } = props;
    var _minValue, _maxValue;
    let baseProps = {
        ...otherProps,
        value: value != null ? [
            value.start,
            value.end
        ] : undefined,
        defaultValue: defaultValue != null ? [
            defaultValue.start,
            defaultValue.end
        ] : [
            (_minValue = props.minValue) !== null && _minValue !== void 0 ? _minValue : 0,
            (_maxValue = props.maxValue) !== null && _maxValue !== void 0 ? _maxValue : 100
        ],
        onChange (v) {
            onChange === null || onChange === void 0 ? void 0 : onChange({
                start: v[0],
                end: v[1]
            });
        },
        onChangeEnd (v) {
            onChangeEnd === null || onChangeEnd === void 0 ? void 0 : onChangeEnd({
                start: v[0],
                end: v[1]
            });
        },
        getValueLabel: getValueLabel ? ([start, end])=>getValueLabel({
                start: start,
                end: end
            })
         : undefined
    };
    let formatter = $hMsoe$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($56bc35cee43d0a96$exports)));
    let { direction: direction  } = $hMsoe$useLocale();
    return(/*#__PURE__*/ $hMsoe$react.createElement($3dd7a0627a64a325$export$9418495bb635ebde, {
        ...baseProps,
        classes: 'spectrum-Slider--range',
        ref: ref
    }, ({ trackRef: trackRef , inputRef: inputRef , state: state  })=>{
        let cssDirection = direction === 'rtl' ? 'right' : 'left';
        return(/*#__PURE__*/ $hMsoe$react.createElement($hMsoe$react.Fragment, null, /*#__PURE__*/ $hMsoe$react.createElement("div", {
            className: $hMsoe$classNames((/*@__PURE__*/$parcel$interopDefault($f6685e21d89782f2$exports)), 'spectrum-Slider-track'),
            style: {
                width: `${state.getThumbPercent(0) * 100}%`
            }
        }), /*#__PURE__*/ $hMsoe$react.createElement($4a1b4dbf5b2701f7$export$2c1b491743890dec, {
            index: 0,
            "aria-label": formatter('minimum'),
            isDisabled: props.isDisabled,
            trackRef: trackRef,
            inputRef: inputRef,
            state: state
        }), /*#__PURE__*/ $hMsoe$react.createElement("div", {
            className: $hMsoe$classNames((/*@__PURE__*/$parcel$interopDefault($f6685e21d89782f2$exports)), 'spectrum-Slider-track'),
            style: {
                [cssDirection]: `${state.getThumbPercent(0) * 100}%`,
                width: `${Math.abs(state.getThumbPercent(0) - state.getThumbPercent(1)) * 100}%`
            }
        }), /*#__PURE__*/ $hMsoe$react.createElement($4a1b4dbf5b2701f7$export$2c1b491743890dec, {
            index: 1,
            "aria-label": formatter('maximum'),
            isDisabled: props.isDisabled,
            trackRef: trackRef,
            state: state
        }), /*#__PURE__*/ $hMsoe$react.createElement("div", {
            className: $hMsoe$classNames((/*@__PURE__*/$parcel$interopDefault($f6685e21d89782f2$exports)), 'spectrum-Slider-track'),
            style: {
                width: `${(1 - state.getThumbPercent(1)) * 100}%`
            }
        })));
    }));
}
/**
 * RangeSliders allow users to quickly select a subset range. They should be used when the upper and lower bounds to the range are invariable.
 */ const $af48336cc068c938$export$826424dabc3dd705 = /*#__PURE__*/ $hMsoe$react.forwardRef($af48336cc068c938$var$RangeSlider);




export {$b48a457725a03d8f$export$472062a354075cee as Slider, $af48336cc068c938$export$826424dabc3dd705 as RangeSlider};
