// @react-spectrum/layout/dist/main.js
require("./main.css");
var $jzyg4$reactspectrumutils = require("@react-spectrum/utils");
var $jzyg4$reactariautils = require("@react-aria/utils");
var $jzyg4$react = require("react");
var $jzyg4$reactariassr = require("@react-aria/ssr");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $0450960891042a15$exports = {};

$parcel$export($0450960891042a15$exports, "repeat", () => $0450960891042a15$export$76d90c956114f2c2);
$parcel$export($0450960891042a15$exports, "minmax", () => $0450960891042a15$export$9c1b655deaca4988);
$parcel$export($0450960891042a15$exports, "fitContent", () => $0450960891042a15$export$2f0b47b0911ce698);
$parcel$export($0450960891042a15$exports, "Grid", () => $0450960891042a15$export$ef2184bd89960b14);



const $0450960891042a15$var$gridStyleProps = {
    ...$jzyg4$reactspectrumutils.baseStyleProps,
    autoFlow: [
        'gridAutoFlow',
        $jzyg4$reactspectrumutils.passthroughStyle
    ],
    autoColumns: [
        'gridAutoColumns',
        $0450960891042a15$var$gridDimensionValue
    ],
    autoRows: [
        'gridAutoRows',
        $0450960891042a15$var$gridDimensionValue
    ],
    areas: [
        'gridTemplateAreas',
        $0450960891042a15$var$gridTemplateAreasValue
    ],
    columns: [
        'gridTemplateColumns',
        $0450960891042a15$var$gridTemplateValue
    ],
    rows: [
        'gridTemplateRows',
        $0450960891042a15$var$gridTemplateValue
    ],
    gap: [
        'gap',
        $jzyg4$reactspectrumutils.dimensionValue
    ],
    rowGap: [
        'rowGap',
        $jzyg4$reactspectrumutils.dimensionValue
    ],
    columnGap: [
        'columnGap',
        $jzyg4$reactspectrumutils.dimensionValue
    ],
    justifyItems: [
        'justifyItems',
        $jzyg4$reactspectrumutils.passthroughStyle
    ],
    justifyContent: [
        'justifyContent',
        $jzyg4$reactspectrumutils.passthroughStyle
    ],
    alignItems: [
        'alignItems',
        $jzyg4$reactspectrumutils.passthroughStyle
    ],
    alignContent: [
        'alignContent',
        $jzyg4$reactspectrumutils.passthroughStyle
    ]
};
function $0450960891042a15$var$Grid(props, ref) {
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $jzyg4$reactspectrumutils.useStyleProps(otherProps, $0450960891042a15$var$gridStyleProps);
    styleProps.style.display = 'grid'; // inline-grid?
    let domRef = $jzyg4$reactspectrumutils.useDOMRef(ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($jzyg4$react)).createElement("div", {
        ...$jzyg4$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, children));
}
function $0450960891042a15$export$76d90c956114f2c2(count, $0450960891042a15$export$76d90c956114f2c2) {
    return `repeat(${count}, ${$0450960891042a15$var$gridTemplateValue($0450960891042a15$export$76d90c956114f2c2)})`;
}
function $0450960891042a15$export$9c1b655deaca4988(min, max) {
    return `minmax(${$0450960891042a15$var$gridDimensionValue(min)}, ${$0450960891042a15$var$gridDimensionValue(max)})`;
}
function $0450960891042a15$export$2f0b47b0911ce698(dimension) {
    return `fit-content(${$0450960891042a15$var$gridDimensionValue(dimension)})`;
}
function $0450960891042a15$var$gridTemplateAreasValue(value) {
    return value.map((v)=>`"${v}"`
    ).join('\n');
}
function $0450960891042a15$var$gridDimensionValue(value) {
    if (/^max-content|min-content|minmax|auto|fit-content|repeat|subgrid/.test(value)) return value;
    return $jzyg4$reactspectrumutils.dimensionValue(value);
}
function $0450960891042a15$var$gridTemplateValue(value) {
    if (Array.isArray(value)) return value.map($0450960891042a15$var$gridDimensionValue).join(' ');
    return $0450960891042a15$var$gridDimensionValue(value);
}
/**
 * A layout container using CSS grid. Supports Spectrum dimensions as values to
 * ensure consistent and adaptive sizing and spacing.
 */ const $0450960891042a15$export$ef2184bd89960b14 = /*#__PURE__*/ $jzyg4$react.forwardRef($0450960891042a15$var$Grid);


var $3ae3a86b899a71dc$exports = {};

$parcel$export($3ae3a86b899a71dc$exports, "Flex", () => $3ae3a86b899a71dc$export$f51f4c4ede09e011);



var $55d404924f6e8155$exports = {};

$parcel$export($55d404924f6e8155$exports, "flex-container", () => $55d404924f6e8155$export$69d7a39fa31a000b, (v) => $55d404924f6e8155$export$69d7a39fa31a000b = v);
$parcel$export($55d404924f6e8155$exports, "flex", () => $55d404924f6e8155$export$97691fbb80847c19, (v) => $55d404924f6e8155$export$97691fbb80847c19 = v);
$parcel$export($55d404924f6e8155$exports, "flex-gap", () => $55d404924f6e8155$export$31a9da8b58047a44, (v) => $55d404924f6e8155$export$31a9da8b58047a44 = v);
var $55d404924f6e8155$export$69d7a39fa31a000b;
var $55d404924f6e8155$export$97691fbb80847c19;
var $55d404924f6e8155$export$31a9da8b58047a44;
$55d404924f6e8155$export$69d7a39fa31a000b = "flex-container_e15493";
$55d404924f6e8155$export$97691fbb80847c19 = "flex_e15493";
$55d404924f6e8155$export$31a9da8b58047a44 = "flex-gap_e15493";



const $3ae3a86b899a71dc$var$flexStyleProps = {
    direction: [
        'flexDirection',
        $jzyg4$reactspectrumutils.passthroughStyle
    ],
    wrap: [
        'flexWrap',
        $3ae3a86b899a71dc$var$flexWrapValue
    ],
    justifyContent: [
        'justifyContent',
        $3ae3a86b899a71dc$var$flexAlignValue
    ],
    alignItems: [
        'alignItems',
        $3ae3a86b899a71dc$var$flexAlignValue
    ],
    alignContent: [
        'alignContent',
        $3ae3a86b899a71dc$var$flexAlignValue
    ]
};
function $3ae3a86b899a71dc$var$Flex(props, ref) {
    let { children: children , ...otherProps } = props;
    let breakpointProvider = $jzyg4$reactspectrumutils.useBreakpoint();
    let matchedBreakpoints = (breakpointProvider === null || breakpointProvider === void 0 ? void 0 : breakpointProvider.matchedBreakpoints) || [
        'base'
    ];
    let { styleProps: styleProps  } = $jzyg4$reactspectrumutils.useStyleProps(otherProps);
    let { styleProps: flexStyle  } = $jzyg4$reactspectrumutils.useStyleProps(otherProps, $3ae3a86b899a71dc$var$flexStyleProps);
    let domRef = $jzyg4$reactspectrumutils.useDOMRef(ref);
    let isSSR = $jzyg4$reactariassr.useIsSSR();
    // If a gap property is specified, and there is no native support or we're in SSR, use a shim.
    // Two divs are required for this: the outer one contains most style properties, and the inner
    // one is the flex container. Each item inside the flex container gets a margin around it based
    // on the gap, and the flex container has a negative margin to counteract this. The outer container
    // is necessary to allow nesting of flex containers with gaps, so that the inner CSS variable doesn't
    // override the outer one.
    if ((props.gap || props.rowGap || props.columnGap) && (isSSR || !$3ae3a86b899a71dc$var$isFlexGapSupported())) {
        let style = {
            ...flexStyle.style,
            '--column-gap': props.columnGap != null ? $jzyg4$reactspectrumutils.responsiveDimensionValue(props.columnGap, matchedBreakpoints) : undefined,
            '--row-gap': props.rowGap != null ? $jzyg4$reactspectrumutils.responsiveDimensionValue(props.rowGap, matchedBreakpoints) : undefined,
            '--gap': props.gap != null ? $jzyg4$reactspectrumutils.responsiveDimensionValue(props.gap, matchedBreakpoints) : undefined
        };
        return(/*#__PURE__*/ ($parcel$interopDefault($jzyg4$react)).createElement("div", {
            ...$jzyg4$reactariautils.filterDOMProps(otherProps),
            ...styleProps,
            className: $jzyg4$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($55d404924f6e8155$exports)), 'flex-container', styleProps.className),
            ref: domRef
        }, /*#__PURE__*/ ($parcel$interopDefault($jzyg4$react)).createElement("div", {
            className: $jzyg4$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($55d404924f6e8155$exports)), 'flex', 'flex-gap'),
            style: style
        }, children)));
    }
    // If no gaps, or native support exists, then we only need to render a single div.
    let style = {
        ...styleProps.style,
        ...flexStyle.style
    };
    if (props.gap != null) style.gap = $jzyg4$reactspectrumutils.responsiveDimensionValue(props.gap, matchedBreakpoints);
    if (props.columnGap != null) style.columnGap = $jzyg4$reactspectrumutils.responsiveDimensionValue(props.columnGap, matchedBreakpoints);
    if (props.rowGap != null) style.rowGap = $jzyg4$reactspectrumutils.responsiveDimensionValue(props.rowGap, matchedBreakpoints);
    return(/*#__PURE__*/ ($parcel$interopDefault($jzyg4$react)).createElement("div", {
        ...$jzyg4$reactariautils.filterDOMProps(otherProps),
        className: $jzyg4$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($55d404924f6e8155$exports)), 'flex', styleProps.className),
        style: style,
        ref: domRef
    }, children));
}
/**
 * Normalize 'start' and 'end' alignment values to 'flex-start' and 'flex-end'
 * in flex containers for browser compatibility.
 */ function $3ae3a86b899a71dc$var$flexAlignValue(value) {
    if (value === 'start') return 'flex-start';
    if (value === 'end') return 'flex-end';
    return value;
}
/**
 * Takes a boolean and translates it to flex wrap or nowrap.
 */ function $3ae3a86b899a71dc$var$flexWrapValue(value) {
    if (typeof value === 'boolean') return value ? 'wrap' : 'nowrap';
    return value;
}
// Original licensing for the following method can be found in the
// NOTICE file in the root directory of this source tree.
// See https://github.com/Modernizr/Modernizr/blob/7efb9d0edd66815fb115fdce95fabaf019ce8db5/feature-detects/css/flexgap.js
let $3ae3a86b899a71dc$var$_isFlexGapSupported = null;
function $3ae3a86b899a71dc$var$isFlexGapSupported() {
    if ($3ae3a86b899a71dc$var$_isFlexGapSupported != null) return $3ae3a86b899a71dc$var$_isFlexGapSupported;
    if (typeof document === 'undefined') return false;
    // create flex container with row-gap set
    var flex = document.createElement('div');
    flex.style.display = 'flex';
    flex.style.flexDirection = 'column';
    flex.style.rowGap = '1px';
    // create two, elements inside it
    flex.appendChild(document.createElement('div'));
    flex.appendChild(document.createElement('div'));
    // append to the DOM (needed to obtain scrollHeight)
    document.body.appendChild(flex);
    $3ae3a86b899a71dc$var$_isFlexGapSupported = flex.scrollHeight === 1; // flex container should be 1px high from the row-gap
    flex.parentNode.removeChild(flex);
    return $3ae3a86b899a71dc$var$_isFlexGapSupported;
}
/**
 * A layout container using flexbox. Provides Spectrum dimension values, and supports the gap
 * property to define consistent spacing between items.
 */ const $3ae3a86b899a71dc$export$f51f4c4ede09e011 = /*#__PURE__*/ $jzyg4$react.forwardRef($3ae3a86b899a71dc$var$Flex);


$parcel$exportWildcard(module.exports, $0450960891042a15$exports);
$parcel$exportWildcard(module.exports, $3ae3a86b899a71dc$exports);
