// @react-spectrum/layout/dist/module.js
import "./main.css";
import {baseStyleProps as $5XBri$baseStyleProps, passthroughStyle as $5XBri$passthroughStyle, dimensionValue as $5XBri$dimensionValue, useStyleProps as $5XBri$useStyleProps, useDOMRef as $5XBri$useDOMRef, useBreakpoint as $5XBri$useBreakpoint, responsiveDimensionValue as $5XBri$responsiveDimensionValue, classNames as $5XBri$classNames} from "@react-spectrum/utils";
import {filterDOMProps as $5XBri$filterDOMProps} from "@react-aria/utils";
import $5XBri$react, {forwardRef as $5XBri$forwardRef} from "react";
import {useIsSSR as $5XBri$useIsSSR} from "@react-aria/ssr";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
var $adc7dbc11bf4e86c$exports = {};

$parcel$export($adc7dbc11bf4e86c$exports, "repeat", () => $adc7dbc11bf4e86c$export$76d90c956114f2c2);
$parcel$export($adc7dbc11bf4e86c$exports, "minmax", () => $adc7dbc11bf4e86c$export$9c1b655deaca4988);
$parcel$export($adc7dbc11bf4e86c$exports, "fitContent", () => $adc7dbc11bf4e86c$export$2f0b47b0911ce698);
$parcel$export($adc7dbc11bf4e86c$exports, "Grid", () => $adc7dbc11bf4e86c$export$ef2184bd89960b14);



const $adc7dbc11bf4e86c$var$gridStyleProps = {
    ...$5XBri$baseStyleProps,
    autoFlow: [
        'gridAutoFlow',
        $5XBri$passthroughStyle
    ],
    autoColumns: [
        'gridAutoColumns',
        $adc7dbc11bf4e86c$var$gridDimensionValue
    ],
    autoRows: [
        'gridAutoRows',
        $adc7dbc11bf4e86c$var$gridDimensionValue
    ],
    areas: [
        'gridTemplateAreas',
        $adc7dbc11bf4e86c$var$gridTemplateAreasValue
    ],
    columns: [
        'gridTemplateColumns',
        $adc7dbc11bf4e86c$var$gridTemplateValue
    ],
    rows: [
        'gridTemplateRows',
        $adc7dbc11bf4e86c$var$gridTemplateValue
    ],
    gap: [
        'gap',
        $5XBri$dimensionValue
    ],
    rowGap: [
        'rowGap',
        $5XBri$dimensionValue
    ],
    columnGap: [
        'columnGap',
        $5XBri$dimensionValue
    ],
    justifyItems: [
        'justifyItems',
        $5XBri$passthroughStyle
    ],
    justifyContent: [
        'justifyContent',
        $5XBri$passthroughStyle
    ],
    alignItems: [
        'alignItems',
        $5XBri$passthroughStyle
    ],
    alignContent: [
        'alignContent',
        $5XBri$passthroughStyle
    ]
};
function $adc7dbc11bf4e86c$var$Grid(props, ref) {
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $5XBri$useStyleProps(otherProps, $adc7dbc11bf4e86c$var$gridStyleProps);
    styleProps.style.display = 'grid'; // inline-grid?
    let domRef = $5XBri$useDOMRef(ref);
    return(/*#__PURE__*/ $5XBri$react.createElement("div", {
        ...$5XBri$filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, children));
}
function $adc7dbc11bf4e86c$export$76d90c956114f2c2(count, $adc7dbc11bf4e86c$export$76d90c956114f2c2) {
    return `repeat(${count}, ${$adc7dbc11bf4e86c$var$gridTemplateValue($adc7dbc11bf4e86c$export$76d90c956114f2c2)})`;
}
function $adc7dbc11bf4e86c$export$9c1b655deaca4988(min, max) {
    return `minmax(${$adc7dbc11bf4e86c$var$gridDimensionValue(min)}, ${$adc7dbc11bf4e86c$var$gridDimensionValue(max)})`;
}
function $adc7dbc11bf4e86c$export$2f0b47b0911ce698(dimension) {
    return `fit-content(${$adc7dbc11bf4e86c$var$gridDimensionValue(dimension)})`;
}
function $adc7dbc11bf4e86c$var$gridTemplateAreasValue(value) {
    return value.map((v)=>`"${v}"`
    ).join('\n');
}
function $adc7dbc11bf4e86c$var$gridDimensionValue(value) {
    if (/^max-content|min-content|minmax|auto|fit-content|repeat|subgrid/.test(value)) return value;
    return $5XBri$dimensionValue(value);
}
function $adc7dbc11bf4e86c$var$gridTemplateValue(value) {
    if (Array.isArray(value)) return value.map($adc7dbc11bf4e86c$var$gridDimensionValue).join(' ');
    return $adc7dbc11bf4e86c$var$gridDimensionValue(value);
}
/**
 * A layout container using CSS grid. Supports Spectrum dimensions as values to
 * ensure consistent and adaptive sizing and spacing.
 */ const $adc7dbc11bf4e86c$export$ef2184bd89960b14 = /*#__PURE__*/ $5XBri$forwardRef($adc7dbc11bf4e86c$var$Grid);


var $1b39fab7603c3bcf$exports = {};

$parcel$export($1b39fab7603c3bcf$exports, "Flex", () => $1b39fab7603c3bcf$export$f51f4c4ede09e011);



var $fbc4eea22f41fc14$exports = {};

$parcel$export($fbc4eea22f41fc14$exports, "flex-container", () => $fbc4eea22f41fc14$export$69d7a39fa31a000b, (v) => $fbc4eea22f41fc14$export$69d7a39fa31a000b = v);
$parcel$export($fbc4eea22f41fc14$exports, "flex", () => $fbc4eea22f41fc14$export$97691fbb80847c19, (v) => $fbc4eea22f41fc14$export$97691fbb80847c19 = v);
$parcel$export($fbc4eea22f41fc14$exports, "flex-gap", () => $fbc4eea22f41fc14$export$31a9da8b58047a44, (v) => $fbc4eea22f41fc14$export$31a9da8b58047a44 = v);
var $fbc4eea22f41fc14$export$69d7a39fa31a000b;
var $fbc4eea22f41fc14$export$97691fbb80847c19;
var $fbc4eea22f41fc14$export$31a9da8b58047a44;
$fbc4eea22f41fc14$export$69d7a39fa31a000b = "flex-container_e15493";
$fbc4eea22f41fc14$export$97691fbb80847c19 = "flex_e15493";
$fbc4eea22f41fc14$export$31a9da8b58047a44 = "flex-gap_e15493";



const $1b39fab7603c3bcf$var$flexStyleProps = {
    direction: [
        'flexDirection',
        $5XBri$passthroughStyle
    ],
    wrap: [
        'flexWrap',
        $1b39fab7603c3bcf$var$flexWrapValue
    ],
    justifyContent: [
        'justifyContent',
        $1b39fab7603c3bcf$var$flexAlignValue
    ],
    alignItems: [
        'alignItems',
        $1b39fab7603c3bcf$var$flexAlignValue
    ],
    alignContent: [
        'alignContent',
        $1b39fab7603c3bcf$var$flexAlignValue
    ]
};
function $1b39fab7603c3bcf$var$Flex(props, ref) {
    let { children: children , ...otherProps } = props;
    let breakpointProvider = $5XBri$useBreakpoint();
    let matchedBreakpoints = (breakpointProvider === null || breakpointProvider === void 0 ? void 0 : breakpointProvider.matchedBreakpoints) || [
        'base'
    ];
    let { styleProps: styleProps  } = $5XBri$useStyleProps(otherProps);
    let { styleProps: flexStyle  } = $5XBri$useStyleProps(otherProps, $1b39fab7603c3bcf$var$flexStyleProps);
    let domRef = $5XBri$useDOMRef(ref);
    let isSSR = $5XBri$useIsSSR();
    // If a gap property is specified, and there is no native support or we're in SSR, use a shim.
    // Two divs are required for this: the outer one contains most style properties, and the inner
    // one is the flex container. Each item inside the flex container gets a margin around it based
    // on the gap, and the flex container has a negative margin to counteract this. The outer container
    // is necessary to allow nesting of flex containers with gaps, so that the inner CSS variable doesn't
    // override the outer one.
    if ((props.gap || props.rowGap || props.columnGap) && (isSSR || !$1b39fab7603c3bcf$var$isFlexGapSupported())) {
        let style = {
            ...flexStyle.style,
            '--column-gap': props.columnGap != null ? $5XBri$responsiveDimensionValue(props.columnGap, matchedBreakpoints) : undefined,
            '--row-gap': props.rowGap != null ? $5XBri$responsiveDimensionValue(props.rowGap, matchedBreakpoints) : undefined,
            '--gap': props.gap != null ? $5XBri$responsiveDimensionValue(props.gap, matchedBreakpoints) : undefined
        };
        return(/*#__PURE__*/ $5XBri$react.createElement("div", {
            ...$5XBri$filterDOMProps(otherProps),
            ...styleProps,
            className: $5XBri$classNames((/*@__PURE__*/$parcel$interopDefault($fbc4eea22f41fc14$exports)), 'flex-container', styleProps.className),
            ref: domRef
        }, /*#__PURE__*/ $5XBri$react.createElement("div", {
            className: $5XBri$classNames((/*@__PURE__*/$parcel$interopDefault($fbc4eea22f41fc14$exports)), 'flex', 'flex-gap'),
            style: style
        }, children)));
    }
    // If no gaps, or native support exists, then we only need to render a single div.
    let style = {
        ...styleProps.style,
        ...flexStyle.style
    };
    if (props.gap != null) style.gap = $5XBri$responsiveDimensionValue(props.gap, matchedBreakpoints);
    if (props.columnGap != null) style.columnGap = $5XBri$responsiveDimensionValue(props.columnGap, matchedBreakpoints);
    if (props.rowGap != null) style.rowGap = $5XBri$responsiveDimensionValue(props.rowGap, matchedBreakpoints);
    return(/*#__PURE__*/ $5XBri$react.createElement("div", {
        ...$5XBri$filterDOMProps(otherProps),
        className: $5XBri$classNames((/*@__PURE__*/$parcel$interopDefault($fbc4eea22f41fc14$exports)), 'flex', styleProps.className),
        style: style,
        ref: domRef
    }, children));
}
/**
 * Normalize 'start' and 'end' alignment values to 'flex-start' and 'flex-end'
 * in flex containers for browser compatibility.
 */ function $1b39fab7603c3bcf$var$flexAlignValue(value) {
    if (value === 'start') return 'flex-start';
    if (value === 'end') return 'flex-end';
    return value;
}
/**
 * Takes a boolean and translates it to flex wrap or nowrap.
 */ function $1b39fab7603c3bcf$var$flexWrapValue(value) {
    if (typeof value === 'boolean') return value ? 'wrap' : 'nowrap';
    return value;
}
// Original licensing for the following method can be found in the
// NOTICE file in the root directory of this source tree.
// See https://github.com/Modernizr/Modernizr/blob/7efb9d0edd66815fb115fdce95fabaf019ce8db5/feature-detects/css/flexgap.js
let $1b39fab7603c3bcf$var$_isFlexGapSupported = null;
function $1b39fab7603c3bcf$var$isFlexGapSupported() {
    if ($1b39fab7603c3bcf$var$_isFlexGapSupported != null) return $1b39fab7603c3bcf$var$_isFlexGapSupported;
    if (typeof document === 'undefined') return false;
    // create flex container with row-gap set
    var flex = document.createElement('div');
    flex.style.display = 'flex';
    flex.style.flexDirection = 'column';
    flex.style.rowGap = '1px';
    // create two, elements inside it
    flex.appendChild(document.createElement('div'));
    flex.appendChild(document.createElement('div'));
    // append to the DOM (needed to obtain scrollHeight)
    document.body.appendChild(flex);
    $1b39fab7603c3bcf$var$_isFlexGapSupported = flex.scrollHeight === 1; // flex container should be 1px high from the row-gap
    flex.parentNode.removeChild(flex);
    return $1b39fab7603c3bcf$var$_isFlexGapSupported;
}
/**
 * A layout container using flexbox. Provides Spectrum dimension values, and supports the gap
 * property to define consistent spacing between items.
 */ const $1b39fab7603c3bcf$export$f51f4c4ede09e011 = /*#__PURE__*/ $5XBri$forwardRef($1b39fab7603c3bcf$var$Flex);




export {$adc7dbc11bf4e86c$export$76d90c956114f2c2 as repeat, $adc7dbc11bf4e86c$export$9c1b655deaca4988 as minmax, $adc7dbc11bf4e86c$export$2f0b47b0911ce698 as fitContent, $adc7dbc11bf4e86c$export$ef2184bd89960b14 as Grid, $1b39fab7603c3bcf$export$f51f4c4ede09e011 as Flex};
