// @react-aria/link/dist/module.js
import {filterDOMProps as $k1pim$filterDOMProps, mergeProps as $k1pim$mergeProps} from "@react-aria/utils";
import {useFocusable as $k1pim$useFocusable} from "@react-aria/focus";
import {usePress as $k1pim$usePress} from "@react-aria/interactions";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $c80650969d556788$exports = {};

$parcel$export($c80650969d556788$exports, "useLink", () => $c80650969d556788$export$dcf14c9974fe2767);



function $c80650969d556788$export$dcf14c9974fe2767(props, ref) {
    let { elementType: elementType = 'a' , onPress: onPress , onPressStart: onPressStart , onPressEnd: onPressEnd , // @ts-ignore
    onClick: deprecatedOnClick , isDisabled: isDisabled , ...otherProps } = props;
    let linkProps;
    if (elementType !== 'a') linkProps = {
        role: 'link',
        tabIndex: !isDisabled ? 0 : undefined
    };
    let { focusableProps: focusableProps  } = $k1pim$useFocusable(props, ref);
    let { pressProps: pressProps , isPressed: isPressed  } = $k1pim$usePress({
        onPress: onPress,
        onPressStart: onPressStart,
        onPressEnd: onPressEnd,
        isDisabled: isDisabled,
        ref: ref
    });
    let domProps = $k1pim$filterDOMProps(otherProps, {
        labelable: true
    });
    let interactionHandlers = $k1pim$mergeProps(focusableProps, pressProps);
    return {
        isPressed: isPressed,
        linkProps: $k1pim$mergeProps(domProps, {
            ...interactionHandlers,
            ...linkProps,
            'aria-disabled': isDisabled || undefined,
            onClick: (e)=>{
                pressProps.onClick(e);
                if (deprecatedOnClick) {
                    deprecatedOnClick(e);
                    console.warn('onClick is deprecated, please use onPress');
                }
            }
        })
    };
}




export {$c80650969d556788$export$dcf14c9974fe2767 as useLink};
