// @react-spectrum/radio/dist/main.js
require("./main.css");
var $iLD3i$reactspectrumutils = require("@react-spectrum/utils");
var $iLD3i$reactariafocus = require("@react-aria/focus");
var $iLD3i$react = require("react");
var $iLD3i$reactariainteractions = require("@react-aria/interactions");
var $iLD3i$reactariaradio = require("@react-aria/radio");
var $iLD3i$reactspectrumlabel = require("@react-spectrum/label");
var $iLD3i$reactspectrumform = require("@react-spectrum/form");
var $iLD3i$reactspectrumprovider = require("@react-spectrum/provider");
var $iLD3i$reactstatelyradio = require("@react-stately/radio");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $2033c9c68ea215df$exports = {};

$parcel$export($2033c9c68ea215df$exports, "Radio", () => $2033c9c68ea215df$export$d7b12c4107be0d61);



var $a0557b52f593b6c8$exports = {};

$parcel$export($a0557b52f593b6c8$exports, "spectrum-Radio", () => $a0557b52f593b6c8$export$26ebd4dbfef068c5, (v) => $a0557b52f593b6c8$export$26ebd4dbfef068c5 = v);
$parcel$export($a0557b52f593b6c8$exports, "spectrum-Radio-input", () => $a0557b52f593b6c8$export$c0ce6f9a6a2de1cc, (v) => $a0557b52f593b6c8$export$c0ce6f9a6a2de1cc = v);
$parcel$export($a0557b52f593b6c8$exports, "spectrum-Radio-button", () => $a0557b52f593b6c8$export$bf9d5e902c33bf05, (v) => $a0557b52f593b6c8$export$bf9d5e902c33bf05 = v);
$parcel$export($a0557b52f593b6c8$exports, "focus-ring", () => $a0557b52f593b6c8$export$f39a09f249340e2a, (v) => $a0557b52f593b6c8$export$f39a09f249340e2a = v);
$parcel$export($a0557b52f593b6c8$exports, "spectrum-Radio-label", () => $a0557b52f593b6c8$export$b0f8395a9667922d, (v) => $a0557b52f593b6c8$export$b0f8395a9667922d = v);
$parcel$export($a0557b52f593b6c8$exports, "spectrum-Radio--labelBelow", () => $a0557b52f593b6c8$export$9443e7082698950d, (v) => $a0557b52f593b6c8$export$9443e7082698950d = v);
$parcel$export($a0557b52f593b6c8$exports, "is-hovered", () => $a0557b52f593b6c8$export$b8813cd5d7824ce7, (v) => $a0557b52f593b6c8$export$b8813cd5d7824ce7 = v);
$parcel$export($a0557b52f593b6c8$exports, "spectrum-Radio--quiet", () => $a0557b52f593b6c8$export$6db66b55a897f36d, (v) => $a0557b52f593b6c8$export$6db66b55a897f36d = v);
$parcel$export($a0557b52f593b6c8$exports, "is-invalid", () => $a0557b52f593b6c8$export$14a30de2866685fa, (v) => $a0557b52f593b6c8$export$14a30de2866685fa = v);
var $a0557b52f593b6c8$export$26ebd4dbfef068c5;
var $a0557b52f593b6c8$export$c0ce6f9a6a2de1cc;
var $a0557b52f593b6c8$export$bf9d5e902c33bf05;
var $a0557b52f593b6c8$export$f39a09f249340e2a;
var $a0557b52f593b6c8$export$b0f8395a9667922d;
var $a0557b52f593b6c8$export$9443e7082698950d;
var $a0557b52f593b6c8$export$b8813cd5d7824ce7;
var $a0557b52f593b6c8$export$6db66b55a897f36d;
var $a0557b52f593b6c8$export$14a30de2866685fa;
$a0557b52f593b6c8$export$26ebd4dbfef068c5 = "spectrum-Radio_bf46c2";
$a0557b52f593b6c8$export$c0ce6f9a6a2de1cc = "spectrum-Radio-input_bf46c2";
$a0557b52f593b6c8$export$bf9d5e902c33bf05 = "spectrum-Radio-button_bf46c2";
$a0557b52f593b6c8$export$f39a09f249340e2a = "focus-ring_bf46c2";
$a0557b52f593b6c8$export$b0f8395a9667922d = "spectrum-Radio-label_bf46c2";
$a0557b52f593b6c8$export$9443e7082698950d = "spectrum-Radio--labelBelow_bf46c2";
$a0557b52f593b6c8$export$b8813cd5d7824ce7 = "is-hovered_bf46c2";
$a0557b52f593b6c8$export$6db66b55a897f36d = "spectrum-Radio--quiet_bf46c2";
$a0557b52f593b6c8$export$14a30de2866685fa = "is-invalid_bf46c2";





const $e0caaa82aa917f07$export$b118023277d4a5c3 = ($parcel$interopDefault($iLD3i$react)).createContext(null);
function $e0caaa82aa917f07$export$b054eba74077a826() {
    return $iLD3i$react.useContext($e0caaa82aa917f07$export$b118023277d4a5c3);
}


function $2033c9c68ea215df$var$Radio(props, ref) {
    let { isDisabled: isDisabled , children: children , autoFocus: autoFocus , ...otherProps } = props;
    let { styleProps: styleProps  } = $iLD3i$reactspectrumutils.useStyleProps(otherProps);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $iLD3i$reactariainteractions.useHover({
        isDisabled: isDisabled
    });
    let inputRef = $iLD3i$react.useRef(null);
    let domRef = $iLD3i$reactspectrumutils.useFocusableRef(ref, inputRef);
    let radioGroupProps = $e0caaa82aa917f07$export$b054eba74077a826();
    let { isEmphasized: isEmphasized , validationState: validationState , state: state  } = radioGroupProps;
    let { inputProps: inputProps  } = $iLD3i$reactariaradio.useRadio({
        ...props,
        ...radioGroupProps,
        isDisabled: isDisabled
    }, state, inputRef);
    return(/*#__PURE__*/ ($parcel$interopDefault($iLD3i$react)).createElement("label", {
        ...styleProps,
        ...hoverProps,
        ref: domRef,
        className: $iLD3i$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($a0557b52f593b6c8$exports)), 'spectrum-Radio', {
            // Removing. Pending design feedback.
            // 'spectrum-Radio--labelBelow': labelPosition === 'bottom',
            'spectrum-Radio--quiet': !isEmphasized,
            'is-disabled': isDisabled,
            'is-invalid': validationState === 'invalid',
            'is-hovered': isHovered
        }, styleProps.className)
    }, /*#__PURE__*/ ($parcel$interopDefault($iLD3i$react)).createElement($iLD3i$reactariafocus.FocusRing, {
        focusRingClass: $iLD3i$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($a0557b52f593b6c8$exports)), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ ($parcel$interopDefault($iLD3i$react)).createElement("input", {
        ...inputProps,
        ref: inputRef,
        className: $iLD3i$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($a0557b52f593b6c8$exports)), 'spectrum-Radio-input')
    })), /*#__PURE__*/ ($parcel$interopDefault($iLD3i$react)).createElement("span", {
        className: $iLD3i$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($a0557b52f593b6c8$exports)), 'spectrum-Radio-button')
    }), children && /*#__PURE__*/ ($parcel$interopDefault($iLD3i$react)).createElement("span", {
        className: $iLD3i$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($a0557b52f593b6c8$exports)), 'spectrum-Radio-label')
    }, children)));
}
/**
 * Radio buttons allow users to select a single option from a list of mutually exclusive options.
 * All possible options are exposed up front for users to compare.
 */ const $2033c9c68ea215df$export$d7b12c4107be0d61 = /*#__PURE__*/ $iLD3i$react.forwardRef($2033c9c68ea215df$var$Radio);


var $daaa8e2da7e1109e$exports = {};

$parcel$export($daaa8e2da7e1109e$exports, "RadioGroup", () => $daaa8e2da7e1109e$export$a98f0dcb43a68a25);


var $0043d8d062ef413c$exports = {};

$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel", () => $0043d8d062ef413c$export$493d18e796ae054, (v) => $0043d8d062ef413c$export$493d18e796ae054 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel--positionSide", () => $0043d8d062ef413c$export$9e6f19dc21f22f2e, (v) => $0043d8d062ef413c$export$9e6f19dc21f22f2e = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel-requiredIcon", () => $0043d8d062ef413c$export$7c47642c7d46f3c9, (v) => $0043d8d062ef413c$export$7c47642c7d46f3c9 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel--alignEnd", () => $0043d8d062ef413c$export$885efcc08143a987, (v) => $0043d8d062ef413c$export$885efcc08143a987 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field", () => $0043d8d062ef413c$export$f6d480ae1e56eba0, (v) => $0043d8d062ef413c$export$f6d480ae1e56eba0 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field--positionTop", () => $0043d8d062ef413c$export$a4ea780a9064d7f9, (v) => $0043d8d062ef413c$export$a4ea780a9064d7f9 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field-field", () => $0043d8d062ef413c$export$3ab8a3dc7f0563d, (v) => $0043d8d062ef413c$export$3ab8a3dc7f0563d = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field-field--multiline", () => $0043d8d062ef413c$export$f58a1e966a92ba5e, (v) => $0043d8d062ef413c$export$f58a1e966a92ba5e = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field--positionSide", () => $0043d8d062ef413c$export$2963225c91179589, (v) => $0043d8d062ef413c$export$2963225c91179589 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field-wrapper", () => $0043d8d062ef413c$export$127df0b7290eb8ef, (v) => $0043d8d062ef413c$export$127df0b7290eb8ef = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form", () => $0043d8d062ef413c$export$94d2bcc94afabd89, (v) => $0043d8d062ef413c$export$94d2bcc94afabd89 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form--positionSide", () => $0043d8d062ef413c$export$7f44db659563c8f4, (v) => $0043d8d062ef413c$export$7f44db659563c8f4 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form--positionTop", () => $0043d8d062ef413c$export$11b8bc81fe551057, (v) => $0043d8d062ef413c$export$11b8bc81fe551057 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form-itemLabel", () => $0043d8d062ef413c$export$51ec27881a1e3928, (v) => $0043d8d062ef413c$export$51ec27881a1e3928 = v);
$parcel$export($0043d8d062ef413c$exports, "is-disabled", () => $0043d8d062ef413c$export$d35bc1e505d1ebbf, (v) => $0043d8d062ef413c$export$d35bc1e505d1ebbf = v);
var $0043d8d062ef413c$export$493d18e796ae054;
var $0043d8d062ef413c$export$9e6f19dc21f22f2e;
var $0043d8d062ef413c$export$7c47642c7d46f3c9;
var $0043d8d062ef413c$export$885efcc08143a987;
var $0043d8d062ef413c$export$f6d480ae1e56eba0;
var $0043d8d062ef413c$export$a4ea780a9064d7f9;
var $0043d8d062ef413c$export$3ab8a3dc7f0563d;
var $0043d8d062ef413c$export$f58a1e966a92ba5e;
var $0043d8d062ef413c$export$2963225c91179589;
var $0043d8d062ef413c$export$127df0b7290eb8ef;
var $0043d8d062ef413c$export$94d2bcc94afabd89;
var $0043d8d062ef413c$export$7f44db659563c8f4;
var $0043d8d062ef413c$export$11b8bc81fe551057;
var $0043d8d062ef413c$export$51ec27881a1e3928;
var $0043d8d062ef413c$export$d35bc1e505d1ebbf;
$0043d8d062ef413c$export$493d18e796ae054 = "spectrum-FieldLabel_d2db1f";
$0043d8d062ef413c$export$9e6f19dc21f22f2e = "spectrum-FieldLabel--positionSide_d2db1f";
$0043d8d062ef413c$export$7c47642c7d46f3c9 = "spectrum-FieldLabel-requiredIcon_d2db1f";
$0043d8d062ef413c$export$885efcc08143a987 = "spectrum-FieldLabel--alignEnd_d2db1f";
$0043d8d062ef413c$export$f6d480ae1e56eba0 = "spectrum-Field_d2db1f";
$0043d8d062ef413c$export$a4ea780a9064d7f9 = "spectrum-Field--positionTop_d2db1f";
$0043d8d062ef413c$export$3ab8a3dc7f0563d = "spectrum-Field-field_d2db1f";
$0043d8d062ef413c$export$f58a1e966a92ba5e = "spectrum-Field-field--multiline_d2db1f";
$0043d8d062ef413c$export$2963225c91179589 = "spectrum-Field--positionSide_d2db1f";
$0043d8d062ef413c$export$127df0b7290eb8ef = "spectrum-Field-wrapper_d2db1f";
$0043d8d062ef413c$export$94d2bcc94afabd89 = "spectrum-Form_d2db1f";
$0043d8d062ef413c$export$7f44db659563c8f4 = "spectrum-Form--positionSide_d2db1f";
$0043d8d062ef413c$export$11b8bc81fe551057 = "spectrum-Form--positionTop_d2db1f";
$0043d8d062ef413c$export$51ec27881a1e3928 = "spectrum-Form-itemLabel_d2db1f";
$0043d8d062ef413c$export$d35bc1e505d1ebbf = "is-disabled_d2db1f";




var $d2824def51242b29$exports = {};

$parcel$export($d2824def51242b29$exports, "spectrum-FieldGroup", () => $d2824def51242b29$export$75eb16cbfc18279c, (v) => $d2824def51242b29$export$75eb16cbfc18279c = v);
$parcel$export($d2824def51242b29$exports, "spectrum-FieldGroup--positionSide", () => $d2824def51242b29$export$859b606bd9f45f81, (v) => $d2824def51242b29$export$859b606bd9f45f81 = v);
$parcel$export($d2824def51242b29$exports, "spectrum-FieldGroup-group", () => $d2824def51242b29$export$984c7bbbfc7ddffc, (v) => $d2824def51242b29$export$984c7bbbfc7ddffc = v);
$parcel$export($d2824def51242b29$exports, "spectrum-FieldGroup-group--horizontal", () => $d2824def51242b29$export$1ad8646bcba21c0e, (v) => $d2824def51242b29$export$1ad8646bcba21c0e = v);
var $d2824def51242b29$export$75eb16cbfc18279c;
var $d2824def51242b29$export$859b606bd9f45f81;
var $d2824def51242b29$export$984c7bbbfc7ddffc;
var $d2824def51242b29$export$1ad8646bcba21c0e;
$d2824def51242b29$export$75eb16cbfc18279c = "spectrum-FieldGroup_41d350";
$d2824def51242b29$export$859b606bd9f45f81 = "spectrum-FieldGroup--positionSide_41d350";
$d2824def51242b29$export$984c7bbbfc7ddffc = "spectrum-FieldGroup-group_41d350";
$d2824def51242b29$export$1ad8646bcba21c0e = "spectrum-FieldGroup-group--horizontal_41d350";






function $daaa8e2da7e1109e$var$RadioGroup(props, ref) {
    props = $iLD3i$reactspectrumprovider.useProviderProps(props);
    props = $iLD3i$reactspectrumform.useFormProps(props);
    let { isEmphasized: isEmphasized , isRequired: isRequired , necessityIndicator: necessityIndicator , label: label , labelPosition: labelPosition = 'top' , labelAlign: labelAlign , validationState: validationState , children: children , orientation: orientation = 'vertical' , ...otherProps } = props;
    let domRef = $iLD3i$reactspectrumutils.useDOMRef(ref);
    let { styleProps: styleProps  } = $iLD3i$reactspectrumutils.useStyleProps(otherProps);
    let state = $iLD3i$reactstatelyradio.useRadioGroupState(props);
    let { radioGroupProps: radioGroupProps , labelProps: labelProps  } = $iLD3i$reactariaradio.useRadioGroup(props, state);
    return(/*#__PURE__*/ ($parcel$interopDefault($iLD3i$react)).createElement("div", {
        ...styleProps,
        ...radioGroupProps,
        className: $iLD3i$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d2824def51242b29$exports)), 'spectrum-FieldGroup', {
            'spectrum-FieldGroup--positionSide': labelPosition === 'side'
        }, // This is so radio works inside a <Form>
        $iLD3i$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($0043d8d062ef413c$exports)), 'spectrum-Field'), styleProps.className),
        ref: domRef
    }, label && /*#__PURE__*/ ($parcel$interopDefault($iLD3i$react)).createElement($iLD3i$reactspectrumlabel.Label, {
        ...labelProps,
        elementType: "span",
        labelPosition: labelPosition,
        labelAlign: labelAlign,
        isRequired: isRequired,
        necessityIndicator: necessityIndicator
    }, label), /*#__PURE__*/ ($parcel$interopDefault($iLD3i$react)).createElement("div", {
        className: $iLD3i$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d2824def51242b29$exports)), 'spectrum-FieldGroup-group', {
            'spectrum-FieldGroup-group--horizontal': orientation === 'horizontal'
        })
    }, /*#__PURE__*/ ($parcel$interopDefault($iLD3i$react)).createElement($e0caaa82aa917f07$export$b118023277d4a5c3.Provider, {
        value: {
            isEmphasized: isEmphasized,
            validationState: validationState,
            state: state
        }
    }, children))));
}
/**
 * Radio groups allow users to select a single option from a list of mutually exclusive options.
 * All possible options are exposed up front for users to compare.
 */ const $daaa8e2da7e1109e$export$a98f0dcb43a68a25 = /*#__PURE__*/ ($parcel$interopDefault($iLD3i$react)).forwardRef($daaa8e2da7e1109e$var$RadioGroup);


$parcel$exportWildcard(module.exports, $2033c9c68ea215df$exports);
$parcel$exportWildcard(module.exports, $daaa8e2da7e1109e$exports);
