// @react-spectrum/radio/dist/module.js
import "./main.css";
import {useStyleProps as $fZLUC$useStyleProps, useFocusableRef as $fZLUC$useFocusableRef, classNames as $fZLUC$classNames, useDOMRef as $fZLUC$useDOMRef} from "@react-spectrum/utils";
import {FocusRing as $fZLUC$FocusRing} from "@react-aria/focus";
import $fZLUC$react, {useRef as $fZLUC$useRef, forwardRef as $fZLUC$forwardRef, useContext as $fZLUC$useContext} from "react";
import {useHover as $fZLUC$useHover} from "@react-aria/interactions";
import {useRadio as $fZLUC$useRadio, useRadioGroup as $fZLUC$useRadioGroup} from "@react-aria/radio";
import {Label as $fZLUC$Label} from "@react-spectrum/label";
import {useFormProps as $fZLUC$useFormProps} from "@react-spectrum/form";
import {useProviderProps as $fZLUC$useProviderProps} from "@react-spectrum/provider";
import {useRadioGroupState as $fZLUC$useRadioGroupState} from "@react-stately/radio";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $6713fd2c3526c9dd$exports = {};

$parcel$export($6713fd2c3526c9dd$exports, "Radio", () => $6713fd2c3526c9dd$export$d7b12c4107be0d61);



var $79b1f1d44d442660$exports = {};

$parcel$export($79b1f1d44d442660$exports, "spectrum-Radio", () => $79b1f1d44d442660$export$26ebd4dbfef068c5, (v) => $79b1f1d44d442660$export$26ebd4dbfef068c5 = v);
$parcel$export($79b1f1d44d442660$exports, "spectrum-Radio-input", () => $79b1f1d44d442660$export$c0ce6f9a6a2de1cc, (v) => $79b1f1d44d442660$export$c0ce6f9a6a2de1cc = v);
$parcel$export($79b1f1d44d442660$exports, "spectrum-Radio-button", () => $79b1f1d44d442660$export$bf9d5e902c33bf05, (v) => $79b1f1d44d442660$export$bf9d5e902c33bf05 = v);
$parcel$export($79b1f1d44d442660$exports, "focus-ring", () => $79b1f1d44d442660$export$f39a09f249340e2a, (v) => $79b1f1d44d442660$export$f39a09f249340e2a = v);
$parcel$export($79b1f1d44d442660$exports, "spectrum-Radio-label", () => $79b1f1d44d442660$export$b0f8395a9667922d, (v) => $79b1f1d44d442660$export$b0f8395a9667922d = v);
$parcel$export($79b1f1d44d442660$exports, "spectrum-Radio--labelBelow", () => $79b1f1d44d442660$export$9443e7082698950d, (v) => $79b1f1d44d442660$export$9443e7082698950d = v);
$parcel$export($79b1f1d44d442660$exports, "is-hovered", () => $79b1f1d44d442660$export$b8813cd5d7824ce7, (v) => $79b1f1d44d442660$export$b8813cd5d7824ce7 = v);
$parcel$export($79b1f1d44d442660$exports, "spectrum-Radio--quiet", () => $79b1f1d44d442660$export$6db66b55a897f36d, (v) => $79b1f1d44d442660$export$6db66b55a897f36d = v);
$parcel$export($79b1f1d44d442660$exports, "is-invalid", () => $79b1f1d44d442660$export$14a30de2866685fa, (v) => $79b1f1d44d442660$export$14a30de2866685fa = v);
var $79b1f1d44d442660$export$26ebd4dbfef068c5;
var $79b1f1d44d442660$export$c0ce6f9a6a2de1cc;
var $79b1f1d44d442660$export$bf9d5e902c33bf05;
var $79b1f1d44d442660$export$f39a09f249340e2a;
var $79b1f1d44d442660$export$b0f8395a9667922d;
var $79b1f1d44d442660$export$9443e7082698950d;
var $79b1f1d44d442660$export$b8813cd5d7824ce7;
var $79b1f1d44d442660$export$6db66b55a897f36d;
var $79b1f1d44d442660$export$14a30de2866685fa;
$79b1f1d44d442660$export$26ebd4dbfef068c5 = "spectrum-Radio_bf46c2";
$79b1f1d44d442660$export$c0ce6f9a6a2de1cc = "spectrum-Radio-input_bf46c2";
$79b1f1d44d442660$export$bf9d5e902c33bf05 = "spectrum-Radio-button_bf46c2";
$79b1f1d44d442660$export$f39a09f249340e2a = "focus-ring_bf46c2";
$79b1f1d44d442660$export$b0f8395a9667922d = "spectrum-Radio-label_bf46c2";
$79b1f1d44d442660$export$9443e7082698950d = "spectrum-Radio--labelBelow_bf46c2";
$79b1f1d44d442660$export$b8813cd5d7824ce7 = "is-hovered_bf46c2";
$79b1f1d44d442660$export$6db66b55a897f36d = "spectrum-Radio--quiet_bf46c2";
$79b1f1d44d442660$export$14a30de2866685fa = "is-invalid_bf46c2";





const $c70c0aebae77adbf$export$b118023277d4a5c3 = $fZLUC$react.createContext(null);
function $c70c0aebae77adbf$export$b054eba74077a826() {
    return $fZLUC$useContext($c70c0aebae77adbf$export$b118023277d4a5c3);
}


function $6713fd2c3526c9dd$var$Radio(props, ref) {
    let { isDisabled: isDisabled , children: children , autoFocus: autoFocus , ...otherProps } = props;
    let { styleProps: styleProps  } = $fZLUC$useStyleProps(otherProps);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $fZLUC$useHover({
        isDisabled: isDisabled
    });
    let inputRef = $fZLUC$useRef(null);
    let domRef = $fZLUC$useFocusableRef(ref, inputRef);
    let radioGroupProps = $c70c0aebae77adbf$export$b054eba74077a826();
    let { isEmphasized: isEmphasized , validationState: validationState , state: state  } = radioGroupProps;
    let { inputProps: inputProps  } = $fZLUC$useRadio({
        ...props,
        ...radioGroupProps,
        isDisabled: isDisabled
    }, state, inputRef);
    return(/*#__PURE__*/ $fZLUC$react.createElement("label", {
        ...styleProps,
        ...hoverProps,
        ref: domRef,
        className: $fZLUC$classNames((/*@__PURE__*/$parcel$interopDefault($79b1f1d44d442660$exports)), 'spectrum-Radio', {
            // Removing. Pending design feedback.
            // 'spectrum-Radio--labelBelow': labelPosition === 'bottom',
            'spectrum-Radio--quiet': !isEmphasized,
            'is-disabled': isDisabled,
            'is-invalid': validationState === 'invalid',
            'is-hovered': isHovered
        }, styleProps.className)
    }, /*#__PURE__*/ $fZLUC$react.createElement($fZLUC$FocusRing, {
        focusRingClass: $fZLUC$classNames((/*@__PURE__*/$parcel$interopDefault($79b1f1d44d442660$exports)), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ $fZLUC$react.createElement("input", {
        ...inputProps,
        ref: inputRef,
        className: $fZLUC$classNames((/*@__PURE__*/$parcel$interopDefault($79b1f1d44d442660$exports)), 'spectrum-Radio-input')
    })), /*#__PURE__*/ $fZLUC$react.createElement("span", {
        className: $fZLUC$classNames((/*@__PURE__*/$parcel$interopDefault($79b1f1d44d442660$exports)), 'spectrum-Radio-button')
    }), children && /*#__PURE__*/ $fZLUC$react.createElement("span", {
        className: $fZLUC$classNames((/*@__PURE__*/$parcel$interopDefault($79b1f1d44d442660$exports)), 'spectrum-Radio-label')
    }, children)));
}
/**
 * Radio buttons allow users to select a single option from a list of mutually exclusive options.
 * All possible options are exposed up front for users to compare.
 */ const $6713fd2c3526c9dd$export$d7b12c4107be0d61 = /*#__PURE__*/ $fZLUC$forwardRef($6713fd2c3526c9dd$var$Radio);


var $3c8c9af9884883ec$exports = {};

$parcel$export($3c8c9af9884883ec$exports, "RadioGroup", () => $3c8c9af9884883ec$export$a98f0dcb43a68a25);


var $4a84444d5a56ab41$exports = {};

$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel", () => $4a84444d5a56ab41$export$493d18e796ae054, (v) => $4a84444d5a56ab41$export$493d18e796ae054 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel--positionSide", () => $4a84444d5a56ab41$export$9e6f19dc21f22f2e, (v) => $4a84444d5a56ab41$export$9e6f19dc21f22f2e = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel-requiredIcon", () => $4a84444d5a56ab41$export$7c47642c7d46f3c9, (v) => $4a84444d5a56ab41$export$7c47642c7d46f3c9 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel--alignEnd", () => $4a84444d5a56ab41$export$885efcc08143a987, (v) => $4a84444d5a56ab41$export$885efcc08143a987 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field", () => $4a84444d5a56ab41$export$f6d480ae1e56eba0, (v) => $4a84444d5a56ab41$export$f6d480ae1e56eba0 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field--positionTop", () => $4a84444d5a56ab41$export$a4ea780a9064d7f9, (v) => $4a84444d5a56ab41$export$a4ea780a9064d7f9 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field-field", () => $4a84444d5a56ab41$export$3ab8a3dc7f0563d, (v) => $4a84444d5a56ab41$export$3ab8a3dc7f0563d = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field-field--multiline", () => $4a84444d5a56ab41$export$f58a1e966a92ba5e, (v) => $4a84444d5a56ab41$export$f58a1e966a92ba5e = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field--positionSide", () => $4a84444d5a56ab41$export$2963225c91179589, (v) => $4a84444d5a56ab41$export$2963225c91179589 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field-wrapper", () => $4a84444d5a56ab41$export$127df0b7290eb8ef, (v) => $4a84444d5a56ab41$export$127df0b7290eb8ef = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form", () => $4a84444d5a56ab41$export$94d2bcc94afabd89, (v) => $4a84444d5a56ab41$export$94d2bcc94afabd89 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form--positionSide", () => $4a84444d5a56ab41$export$7f44db659563c8f4, (v) => $4a84444d5a56ab41$export$7f44db659563c8f4 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form--positionTop", () => $4a84444d5a56ab41$export$11b8bc81fe551057, (v) => $4a84444d5a56ab41$export$11b8bc81fe551057 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form-itemLabel", () => $4a84444d5a56ab41$export$51ec27881a1e3928, (v) => $4a84444d5a56ab41$export$51ec27881a1e3928 = v);
$parcel$export($4a84444d5a56ab41$exports, "is-disabled", () => $4a84444d5a56ab41$export$d35bc1e505d1ebbf, (v) => $4a84444d5a56ab41$export$d35bc1e505d1ebbf = v);
var $4a84444d5a56ab41$export$493d18e796ae054;
var $4a84444d5a56ab41$export$9e6f19dc21f22f2e;
var $4a84444d5a56ab41$export$7c47642c7d46f3c9;
var $4a84444d5a56ab41$export$885efcc08143a987;
var $4a84444d5a56ab41$export$f6d480ae1e56eba0;
var $4a84444d5a56ab41$export$a4ea780a9064d7f9;
var $4a84444d5a56ab41$export$3ab8a3dc7f0563d;
var $4a84444d5a56ab41$export$f58a1e966a92ba5e;
var $4a84444d5a56ab41$export$2963225c91179589;
var $4a84444d5a56ab41$export$127df0b7290eb8ef;
var $4a84444d5a56ab41$export$94d2bcc94afabd89;
var $4a84444d5a56ab41$export$7f44db659563c8f4;
var $4a84444d5a56ab41$export$11b8bc81fe551057;
var $4a84444d5a56ab41$export$51ec27881a1e3928;
var $4a84444d5a56ab41$export$d35bc1e505d1ebbf;
$4a84444d5a56ab41$export$493d18e796ae054 = "spectrum-FieldLabel_d2db1f";
$4a84444d5a56ab41$export$9e6f19dc21f22f2e = "spectrum-FieldLabel--positionSide_d2db1f";
$4a84444d5a56ab41$export$7c47642c7d46f3c9 = "spectrum-FieldLabel-requiredIcon_d2db1f";
$4a84444d5a56ab41$export$885efcc08143a987 = "spectrum-FieldLabel--alignEnd_d2db1f";
$4a84444d5a56ab41$export$f6d480ae1e56eba0 = "spectrum-Field_d2db1f";
$4a84444d5a56ab41$export$a4ea780a9064d7f9 = "spectrum-Field--positionTop_d2db1f";
$4a84444d5a56ab41$export$3ab8a3dc7f0563d = "spectrum-Field-field_d2db1f";
$4a84444d5a56ab41$export$f58a1e966a92ba5e = "spectrum-Field-field--multiline_d2db1f";
$4a84444d5a56ab41$export$2963225c91179589 = "spectrum-Field--positionSide_d2db1f";
$4a84444d5a56ab41$export$127df0b7290eb8ef = "spectrum-Field-wrapper_d2db1f";
$4a84444d5a56ab41$export$94d2bcc94afabd89 = "spectrum-Form_d2db1f";
$4a84444d5a56ab41$export$7f44db659563c8f4 = "spectrum-Form--positionSide_d2db1f";
$4a84444d5a56ab41$export$11b8bc81fe551057 = "spectrum-Form--positionTop_d2db1f";
$4a84444d5a56ab41$export$51ec27881a1e3928 = "spectrum-Form-itemLabel_d2db1f";
$4a84444d5a56ab41$export$d35bc1e505d1ebbf = "is-disabled_d2db1f";




var $003163fb2d58eced$exports = {};

$parcel$export($003163fb2d58eced$exports, "spectrum-FieldGroup", () => $003163fb2d58eced$export$75eb16cbfc18279c, (v) => $003163fb2d58eced$export$75eb16cbfc18279c = v);
$parcel$export($003163fb2d58eced$exports, "spectrum-FieldGroup--positionSide", () => $003163fb2d58eced$export$859b606bd9f45f81, (v) => $003163fb2d58eced$export$859b606bd9f45f81 = v);
$parcel$export($003163fb2d58eced$exports, "spectrum-FieldGroup-group", () => $003163fb2d58eced$export$984c7bbbfc7ddffc, (v) => $003163fb2d58eced$export$984c7bbbfc7ddffc = v);
$parcel$export($003163fb2d58eced$exports, "spectrum-FieldGroup-group--horizontal", () => $003163fb2d58eced$export$1ad8646bcba21c0e, (v) => $003163fb2d58eced$export$1ad8646bcba21c0e = v);
var $003163fb2d58eced$export$75eb16cbfc18279c;
var $003163fb2d58eced$export$859b606bd9f45f81;
var $003163fb2d58eced$export$984c7bbbfc7ddffc;
var $003163fb2d58eced$export$1ad8646bcba21c0e;
$003163fb2d58eced$export$75eb16cbfc18279c = "spectrum-FieldGroup_41d350";
$003163fb2d58eced$export$859b606bd9f45f81 = "spectrum-FieldGroup--positionSide_41d350";
$003163fb2d58eced$export$984c7bbbfc7ddffc = "spectrum-FieldGroup-group_41d350";
$003163fb2d58eced$export$1ad8646bcba21c0e = "spectrum-FieldGroup-group--horizontal_41d350";






function $3c8c9af9884883ec$var$RadioGroup(props, ref) {
    props = $fZLUC$useProviderProps(props);
    props = $fZLUC$useFormProps(props);
    let { isEmphasized: isEmphasized , isRequired: isRequired , necessityIndicator: necessityIndicator , label: label , labelPosition: labelPosition = 'top' , labelAlign: labelAlign , validationState: validationState , children: children , orientation: orientation = 'vertical' , ...otherProps } = props;
    let domRef = $fZLUC$useDOMRef(ref);
    let { styleProps: styleProps  } = $fZLUC$useStyleProps(otherProps);
    let state = $fZLUC$useRadioGroupState(props);
    let { radioGroupProps: radioGroupProps , labelProps: labelProps  } = $fZLUC$useRadioGroup(props, state);
    return(/*#__PURE__*/ $fZLUC$react.createElement("div", {
        ...styleProps,
        ...radioGroupProps,
        className: $fZLUC$classNames((/*@__PURE__*/$parcel$interopDefault($003163fb2d58eced$exports)), 'spectrum-FieldGroup', {
            'spectrum-FieldGroup--positionSide': labelPosition === 'side'
        }, // This is so radio works inside a <Form>
        $fZLUC$classNames((/*@__PURE__*/$parcel$interopDefault($4a84444d5a56ab41$exports)), 'spectrum-Field'), styleProps.className),
        ref: domRef
    }, label && /*#__PURE__*/ $fZLUC$react.createElement($fZLUC$Label, {
        ...labelProps,
        elementType: "span",
        labelPosition: labelPosition,
        labelAlign: labelAlign,
        isRequired: isRequired,
        necessityIndicator: necessityIndicator
    }, label), /*#__PURE__*/ $fZLUC$react.createElement("div", {
        className: $fZLUC$classNames((/*@__PURE__*/$parcel$interopDefault($003163fb2d58eced$exports)), 'spectrum-FieldGroup-group', {
            'spectrum-FieldGroup-group--horizontal': orientation === 'horizontal'
        })
    }, /*#__PURE__*/ $fZLUC$react.createElement($c70c0aebae77adbf$export$b118023277d4a5c3.Provider, {
        value: {
            isEmphasized: isEmphasized,
            validationState: validationState,
            state: state
        }
    }, children))));
}
/**
 * Radio groups allow users to select a single option from a list of mutually exclusive options.
 * All possible options are exposed up front for users to compare.
 */ const $3c8c9af9884883ec$export$a98f0dcb43a68a25 = /*#__PURE__*/ $fZLUC$react.forwardRef($3c8c9af9884883ec$var$RadioGroup);




export {$6713fd2c3526c9dd$export$d7b12c4107be0d61 as Radio, $3c8c9af9884883ec$export$a98f0dcb43a68a25 as RadioGroup};
