// @react-spectrum/image/dist/main.js
require("./main.css");
var $jh4In$reactspectrumutils = require("@react-spectrum/utils");
var $jh4In$reactariautils = require("@react-aria/utils");
var $jh4In$react = require("react");
var $jh4In$reactspectrumprovider = require("@react-spectrum/provider");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $84193f4614b6f53f$exports = {};

$parcel$export($84193f4614b6f53f$exports, "Image", () => $84193f4614b6f53f$export$3e431a229df88919);



var $09f08cc5c8862136$exports = {};

$parcel$export($09f08cc5c8862136$exports, "spectrum-Image-img", () => $09f08cc5c8862136$export$3238177972e11200, (v) => $09f08cc5c8862136$export$3238177972e11200 = v);
var $09f08cc5c8862136$export$3238177972e11200;
$09f08cc5c8862136$export$3238177972e11200 = "spectrum-Image-img_fdc794";



// incomplete component for show right now
function $84193f4614b6f53f$var$Image(props, ref) {
    /* Slots should be able to pass an alt for default behavior, but in Images, the child may know better. */ let userProvidedAlt = props.alt;
    props = $jh4In$reactspectrumutils.useSlotProps(props, 'image');
    props = $jh4In$reactspectrumprovider.useProviderProps(props);
    let { objectFit: objectFit , src: src , alt: alt , ...otherProps } = props;
    let { styleProps: styleProps  } = $jh4In$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $jh4In$reactspectrumutils.useDOMRef(ref);
    if (alt == null) console.warn("The `alt` prop was not provided to an image. Add `alt` text for screen readers, or set `alt=\"\"` prop to indicate that the image is decorative or redundant with displayed text and should not be announced by screen readers.");
    return(/*#__PURE__*/ ($parcel$interopDefault($jh4In$react)).createElement("div", {
        ...$jh4In$reactariautils.filterDOMProps(props),
        ...styleProps,
        className: $jh4In$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($09f08cc5c8862136$exports)), styleProps.className),
        style: {
            ...styleProps.style,
            overflow: 'hidden'
        },
        ref: domRef
    }, /*#__PURE__*/ ($parcel$interopDefault($jh4In$react)).createElement("img", {
        src: src,
        alt: userProvidedAlt || alt,
        style: {
            objectFit: objectFit
        },
        className: $jh4In$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($09f08cc5c8862136$exports)), 'spectrum-Image-img')
    })));
}
/**
 * Image is used to insert and display an image within a component.
 */ const $84193f4614b6f53f$export$3e431a229df88919 = /*#__PURE__*/ ($parcel$interopDefault($jh4In$react)).forwardRef($84193f4614b6f53f$var$Image);


$parcel$exportWildcard(module.exports, $84193f4614b6f53f$exports);
