// @react-spectrum/image/dist/module.js
import "./main.css";
import {useSlotProps as $41gns$useSlotProps, useStyleProps as $41gns$useStyleProps, useDOMRef as $41gns$useDOMRef, classNames as $41gns$classNames} from "@react-spectrum/utils";
import {filterDOMProps as $41gns$filterDOMProps} from "@react-aria/utils";
import $41gns$react from "react";
import {useProviderProps as $41gns$useProviderProps} from "@react-spectrum/provider";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $f37cef3013440181$exports = {};

$parcel$export($f37cef3013440181$exports, "Image", () => $f37cef3013440181$export$3e431a229df88919);



var $31018244cc3e1a7c$exports = {};

$parcel$export($31018244cc3e1a7c$exports, "spectrum-Image-img", () => $31018244cc3e1a7c$export$3238177972e11200, (v) => $31018244cc3e1a7c$export$3238177972e11200 = v);
var $31018244cc3e1a7c$export$3238177972e11200;
$31018244cc3e1a7c$export$3238177972e11200 = "spectrum-Image-img_fdc794";



// incomplete component for show right now
function $f37cef3013440181$var$Image(props, ref) {
    /* Slots should be able to pass an alt for default behavior, but in Images, the child may know better. */ let userProvidedAlt = props.alt;
    props = $41gns$useSlotProps(props, 'image');
    props = $41gns$useProviderProps(props);
    let { objectFit: objectFit , src: src , alt: alt , ...otherProps } = props;
    let { styleProps: styleProps  } = $41gns$useStyleProps(otherProps);
    let domRef = $41gns$useDOMRef(ref);
    if (alt == null) console.warn("The `alt` prop was not provided to an image. Add `alt` text for screen readers, or set `alt=\"\"` prop to indicate that the image is decorative or redundant with displayed text and should not be announced by screen readers.");
    return(/*#__PURE__*/ $41gns$react.createElement("div", {
        ...$41gns$filterDOMProps(props),
        ...styleProps,
        className: $41gns$classNames((/*@__PURE__*/$parcel$interopDefault($31018244cc3e1a7c$exports)), styleProps.className),
        style: {
            ...styleProps.style,
            overflow: 'hidden'
        },
        ref: domRef
    }, /*#__PURE__*/ $41gns$react.createElement("img", {
        src: src,
        alt: userProvidedAlt || alt,
        style: {
            objectFit: objectFit
        },
        className: $41gns$classNames((/*@__PURE__*/$parcel$interopDefault($31018244cc3e1a7c$exports)), 'spectrum-Image-img')
    })));
}
/**
 * Image is used to insert and display an image within a component.
 */ const $f37cef3013440181$export$3e431a229df88919 = /*#__PURE__*/ $41gns$react.forwardRef($f37cef3013440181$var$Image);




export {$f37cef3013440181$export$3e431a229df88919 as Image};
