// @react-aria/breadcrumbs/dist/main.js
var $b3Yra$reactarialink = require("@react-aria/link");
var $b3Yra$reactariautils = require("@react-aria/utils");
var $b3Yra$reactariai18n = require("@react-aria/i18n");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
var $d70894c1656a4ec4$exports = {};

$parcel$export($d70894c1656a4ec4$exports, "useBreadcrumbItem", () => $d70894c1656a4ec4$export$452b38fce62c9384);

function $d70894c1656a4ec4$export$452b38fce62c9384(props, ref) {
    let { isCurrent: isCurrent , isDisabled: isDisabled , 'aria-current': ariaCurrent , elementType: elementType = 'a' , ...otherProps } = props;
    let { linkProps: linkProps  } = $b3Yra$reactarialink.useLink({
        isDisabled: isDisabled || isCurrent,
        elementType: elementType,
        ...otherProps
    }, ref);
    let isHeading = /^h[1-6]$/.test(elementType);
    let itemProps = {
    };
    if (!isHeading) itemProps = linkProps;
    if (isCurrent) itemProps['aria-current'] = ariaCurrent || 'page';
    return {
        itemProps: {
            'aria-disabled': isDisabled,
            ...itemProps
        }
    };
}


var $4d1b15d144f129fa$exports = {};

$parcel$export($4d1b15d144f129fa$exports, "useBreadcrumbs", () => $4d1b15d144f129fa$export$8cefe241bd876ca0);

var $b34eb249cf7dd066$exports = {};
var $3960fa96293bcc59$exports = {};
$3960fa96293bcc59$exports = JSON.parse("{\"breadcrumbs\":\"عناصر الواجهة\"}");


var $abe11536efa5b429$exports = {};
$abe11536efa5b429$exports = JSON.parse("{\"breadcrumbs\":\"Трохи хляб\"}");


var $613fc9b567be2236$exports = {};
$613fc9b567be2236$exports = JSON.parse("{\"breadcrumbs\":\"Popis cesty\"}");


var $bdd3c166d2a526fc$exports = {};
$bdd3c166d2a526fc$exports = JSON.parse("{\"breadcrumbs\":\"Brødkrummer\"}");


var $f9f3a8f9fc414a2e$exports = {};
$f9f3a8f9fc414a2e$exports = JSON.parse("{\"breadcrumbs\":\"Breadcrumbs\"}");


var $bca873223d1547c4$exports = {};
$bca873223d1547c4$exports = JSON.parse("{\"breadcrumbs\":\"Πλοηγήσεις breadcrumb\"}");


var $b1888ffa78c0636f$exports = {};
$b1888ffa78c0636f$exports = JSON.parse("{\"breadcrumbs\":\"Breadcrumbs\"}");


var $ec1d30b7d0e67553$exports = {};
$ec1d30b7d0e67553$exports = JSON.parse("{\"breadcrumbs\":\"Migas de pan\"}");


var $7559c1bb50789552$exports = {};
$7559c1bb50789552$exports = JSON.parse("{\"breadcrumbs\":\"Lingiread\"}");


var $a0dfea45588f258c$exports = {};
$a0dfea45588f258c$exports = JSON.parse("{\"breadcrumbs\":\"Navigointilinkit\"}");


var $7b78eb6c350ec1a2$exports = {};
$7b78eb6c350ec1a2$exports = JSON.parse("{\"breadcrumbs\":\"Chemin de navigation\"}");


var $317d86c10ef00c53$exports = {};
$317d86c10ef00c53$exports = JSON.parse("{\"breadcrumbs\":\"שבילי ניווט\"}");


var $76ed1525b36d7359$exports = {};
$76ed1525b36d7359$exports = JSON.parse("{\"breadcrumbs\":\"Navigacijski putovi\"}");


var $7ed389cea9ea9cad$exports = {};
$7ed389cea9ea9cad$exports = JSON.parse("{\"breadcrumbs\":\"Morzsamenü\"}");


var $e9c200f66f708f0a$exports = {};
$e9c200f66f708f0a$exports = JSON.parse("{\"breadcrumbs\":\"Breadcrumb\"}");


var $afeba701c750c7ba$exports = {};
$afeba701c750c7ba$exports = JSON.parse("{\"breadcrumbs\":\"パンくずリスト\"}");


var $cda0c61aeb5ee9ab$exports = {};
$cda0c61aeb5ee9ab$exports = JSON.parse("{\"breadcrumbs\":\"탐색 표시\"}");


var $64f83b78a6cfc936$exports = {};
$64f83b78a6cfc936$exports = JSON.parse("{\"breadcrumbs\":\"Naršymo kelias\"}");


var $3128809d716ae49c$exports = {};
$3128809d716ae49c$exports = JSON.parse("{\"breadcrumbs\":\"Atpakaļceļi\"}");


var $75f0c2000e867857$exports = {};
$75f0c2000e867857$exports = JSON.parse("{\"breadcrumbs\":\"Navigasjonsstier\"}");


var $fc413c04d0285015$exports = {};
$fc413c04d0285015$exports = JSON.parse("{\"breadcrumbs\":\"Broodkruimels\"}");


var $8acbe8283539522c$exports = {};
$8acbe8283539522c$exports = JSON.parse("{\"breadcrumbs\":\"Struktura nawigacyjna\"}");


var $5abc9f869b726ad0$exports = {};
$5abc9f869b726ad0$exports = JSON.parse("{\"breadcrumbs\":\"Caminho detalhado\"}");


var $fe3a9eafd0ec09e0$exports = {};
$fe3a9eafd0ec09e0$exports = JSON.parse("{\"breadcrumbs\":\"Categorias\"}");


var $af20fe3e9eb21a57$exports = {};
$af20fe3e9eb21a57$exports = JSON.parse("{\"breadcrumbs\":\"Miez de pâine\"}");


var $fc498831bbe78929$exports = {};
$fc498831bbe78929$exports = JSON.parse("{\"breadcrumbs\":\"Навигация\"}");


var $8812e4a7e7ae1504$exports = {};
$8812e4a7e7ae1504$exports = JSON.parse("{\"breadcrumbs\":\"Navigačné prvky Breadcrumbs\"}");


var $c90e515ce6a2c597$exports = {};
$c90e515ce6a2c597$exports = JSON.parse("{\"breadcrumbs\":\"Drobtine\"}");


var $02f0627c4a814427$exports = {};
$02f0627c4a814427$exports = JSON.parse("{\"breadcrumbs\":\"Putanje navigacije\"}");


var $df6acacb5bc495da$exports = {};
$df6acacb5bc495da$exports = JSON.parse("{\"breadcrumbs\":\"Sökvägar\"}");


var $8f68b6110fd6c54b$exports = {};
$8f68b6110fd6c54b$exports = JSON.parse("{\"breadcrumbs\":\"İçerik haritaları\"}");


var $b0f95bace8e9a153$exports = {};
$b0f95bace8e9a153$exports = JSON.parse("{\"breadcrumbs\":\"Навігаційна стежка\"}");


var $105eb1dc1c69b569$exports = {};
$105eb1dc1c69b569$exports = JSON.parse("{\"breadcrumbs\":\"导航栏\"}");


var $1a5e00a39c7c9c2f$exports = {};
$1a5e00a39c7c9c2f$exports = JSON.parse("{\"breadcrumbs\":\"導覽列\"}");


$b34eb249cf7dd066$exports = {
    "ar-AE": $3960fa96293bcc59$exports,
    "bg-BG": $abe11536efa5b429$exports,
    "cs-CZ": $613fc9b567be2236$exports,
    "da-DK": $bdd3c166d2a526fc$exports,
    "de-DE": $f9f3a8f9fc414a2e$exports,
    "el-GR": $bca873223d1547c4$exports,
    "en-US": $b1888ffa78c0636f$exports,
    "es-ES": $ec1d30b7d0e67553$exports,
    "et-EE": $7559c1bb50789552$exports,
    "fi-FI": $a0dfea45588f258c$exports,
    "fr-FR": $7b78eb6c350ec1a2$exports,
    "he-IL": $317d86c10ef00c53$exports,
    "hr-HR": $76ed1525b36d7359$exports,
    "hu-HU": $7ed389cea9ea9cad$exports,
    "it-IT": $e9c200f66f708f0a$exports,
    "ja-JP": $afeba701c750c7ba$exports,
    "ko-KR": $cda0c61aeb5ee9ab$exports,
    "lt-LT": $64f83b78a6cfc936$exports,
    "lv-LV": $3128809d716ae49c$exports,
    "nb-NO": $75f0c2000e867857$exports,
    "nl-NL": $fc413c04d0285015$exports,
    "pl-PL": $8acbe8283539522c$exports,
    "pt-BR": $5abc9f869b726ad0$exports,
    "pt-PT": $fe3a9eafd0ec09e0$exports,
    "ro-RO": $af20fe3e9eb21a57$exports,
    "ru-RU": $fc498831bbe78929$exports,
    "sk-SK": $8812e4a7e7ae1504$exports,
    "sl-SI": $c90e515ce6a2c597$exports,
    "sr-SP": $02f0627c4a814427$exports,
    "sv-SE": $df6acacb5bc495da$exports,
    "tr-TR": $8f68b6110fd6c54b$exports,
    "uk-UA": $b0f95bace8e9a153$exports,
    "zh-CN": $105eb1dc1c69b569$exports,
    "zh-TW": $1a5e00a39c7c9c2f$exports
};



function $4d1b15d144f129fa$export$8cefe241bd876ca0(props) {
    let { 'aria-label': ariaLabel , ...otherProps } = props;
    let formatMessage = $b3Yra$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($b34eb249cf7dd066$exports)));
    return {
        navProps: {
            ...$b3Yra$reactariautils.filterDOMProps(otherProps, {
                labelable: true
            }),
            'aria-label': ariaLabel || formatMessage('breadcrumbs')
        }
    };
}


$parcel$exportWildcard(module.exports, $d70894c1656a4ec4$exports);
$parcel$exportWildcard(module.exports, $4d1b15d144f129fa$exports);
