// @react-aria/breadcrumbs/dist/module.js
import {useLink as $7vd6z$useLink} from "@react-aria/link";
import {filterDOMProps as $7vd6z$filterDOMProps} from "@react-aria/utils";
import {useMessageFormatter as $7vd6z$useMessageFormatter} from "@react-aria/i18n";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
var $c229b6f7e13bf1fc$exports = {};

$parcel$export($c229b6f7e13bf1fc$exports, "useBreadcrumbItem", () => $c229b6f7e13bf1fc$export$452b38fce62c9384);

function $c229b6f7e13bf1fc$export$452b38fce62c9384(props, ref) {
    let { isCurrent: isCurrent , isDisabled: isDisabled , 'aria-current': ariaCurrent , elementType: elementType = 'a' , ...otherProps } = props;
    let { linkProps: linkProps  } = $7vd6z$useLink({
        isDisabled: isDisabled || isCurrent,
        elementType: elementType,
        ...otherProps
    }, ref);
    let isHeading = /^h[1-6]$/.test(elementType);
    let itemProps = {
    };
    if (!isHeading) itemProps = linkProps;
    if (isCurrent) itemProps['aria-current'] = ariaCurrent || 'page';
    return {
        itemProps: {
            'aria-disabled': isDisabled,
            ...itemProps
        }
    };
}


var $5d2fecae9926feeb$exports = {};

$parcel$export($5d2fecae9926feeb$exports, "useBreadcrumbs", () => $5d2fecae9926feeb$export$8cefe241bd876ca0);

var $485710c529557361$exports = {};
var $f4882d7260ca8f9d$exports = {};
$f4882d7260ca8f9d$exports = JSON.parse("{\"breadcrumbs\":\"عناصر الواجهة\"}");


var $4820186fc7eb481b$exports = {};
$4820186fc7eb481b$exports = JSON.parse("{\"breadcrumbs\":\"Трохи хляб\"}");


var $cd21fe388d89ec0b$exports = {};
$cd21fe388d89ec0b$exports = JSON.parse("{\"breadcrumbs\":\"Popis cesty\"}");


var $a073ed7897856adc$exports = {};
$a073ed7897856adc$exports = JSON.parse("{\"breadcrumbs\":\"Brødkrummer\"}");


var $6269ed27c37560a5$exports = {};
$6269ed27c37560a5$exports = JSON.parse("{\"breadcrumbs\":\"Breadcrumbs\"}");


var $533ddb487767d55d$exports = {};
$533ddb487767d55d$exports = JSON.parse("{\"breadcrumbs\":\"Πλοηγήσεις breadcrumb\"}");


var $f2db6bd9510488c7$exports = {};
$f2db6bd9510488c7$exports = JSON.parse("{\"breadcrumbs\":\"Breadcrumbs\"}");


var $fb38c4846c09e98f$exports = {};
$fb38c4846c09e98f$exports = JSON.parse("{\"breadcrumbs\":\"Migas de pan\"}");


var $022a9f75c44b56af$exports = {};
$022a9f75c44b56af$exports = JSON.parse("{\"breadcrumbs\":\"Lingiread\"}");


var $94b8da7c6bc99373$exports = {};
$94b8da7c6bc99373$exports = JSON.parse("{\"breadcrumbs\":\"Navigointilinkit\"}");


var $438cfb04e4fb7b1d$exports = {};
$438cfb04e4fb7b1d$exports = JSON.parse("{\"breadcrumbs\":\"Chemin de navigation\"}");


var $b89382922d84873c$exports = {};
$b89382922d84873c$exports = JSON.parse("{\"breadcrumbs\":\"שבילי ניווט\"}");


var $13c375f718e9ab73$exports = {};
$13c375f718e9ab73$exports = JSON.parse("{\"breadcrumbs\":\"Navigacijski putovi\"}");


var $99db8e40d6794512$exports = {};
$99db8e40d6794512$exports = JSON.parse("{\"breadcrumbs\":\"Morzsamenü\"}");


var $c5ec56fcffed7c2e$exports = {};
$c5ec56fcffed7c2e$exports = JSON.parse("{\"breadcrumbs\":\"Breadcrumb\"}");


var $4c818075da01c5fb$exports = {};
$4c818075da01c5fb$exports = JSON.parse("{\"breadcrumbs\":\"パンくずリスト\"}");


var $020bec590bb8378d$exports = {};
$020bec590bb8378d$exports = JSON.parse("{\"breadcrumbs\":\"탐색 표시\"}");


var $9679a565775519e8$exports = {};
$9679a565775519e8$exports = JSON.parse("{\"breadcrumbs\":\"Naršymo kelias\"}");


var $d42942efb802b9f2$exports = {};
$d42942efb802b9f2$exports = JSON.parse("{\"breadcrumbs\":\"Atpakaļceļi\"}");


var $f1b2d49b5f1a53ce$exports = {};
$f1b2d49b5f1a53ce$exports = JSON.parse("{\"breadcrumbs\":\"Navigasjonsstier\"}");


var $6335c9f4b1a860c8$exports = {};
$6335c9f4b1a860c8$exports = JSON.parse("{\"breadcrumbs\":\"Broodkruimels\"}");


var $5467dc99696e6e12$exports = {};
$5467dc99696e6e12$exports = JSON.parse("{\"breadcrumbs\":\"Struktura nawigacyjna\"}");


var $04df9be94af1c63e$exports = {};
$04df9be94af1c63e$exports = JSON.parse("{\"breadcrumbs\":\"Caminho detalhado\"}");


var $93938dc85cfd9f53$exports = {};
$93938dc85cfd9f53$exports = JSON.parse("{\"breadcrumbs\":\"Categorias\"}");


var $0f892a5062257f67$exports = {};
$0f892a5062257f67$exports = JSON.parse("{\"breadcrumbs\":\"Miez de pâine\"}");


var $11748af6b7efd7ce$exports = {};
$11748af6b7efd7ce$exports = JSON.parse("{\"breadcrumbs\":\"Навигация\"}");


var $63c4fc1394d16ef6$exports = {};
$63c4fc1394d16ef6$exports = JSON.parse("{\"breadcrumbs\":\"Navigačné prvky Breadcrumbs\"}");


var $d3f065ea8b3e92d2$exports = {};
$d3f065ea8b3e92d2$exports = JSON.parse("{\"breadcrumbs\":\"Drobtine\"}");


var $ed32fb8efdd4ae31$exports = {};
$ed32fb8efdd4ae31$exports = JSON.parse("{\"breadcrumbs\":\"Putanje navigacije\"}");


var $c194c93964d34aa6$exports = {};
$c194c93964d34aa6$exports = JSON.parse("{\"breadcrumbs\":\"Sökvägar\"}");


var $8d07143566be6228$exports = {};
$8d07143566be6228$exports = JSON.parse("{\"breadcrumbs\":\"İçerik haritaları\"}");


var $a9fe916ff1200a3e$exports = {};
$a9fe916ff1200a3e$exports = JSON.parse("{\"breadcrumbs\":\"Навігаційна стежка\"}");


var $9e50d72eef976ccc$exports = {};
$9e50d72eef976ccc$exports = JSON.parse("{\"breadcrumbs\":\"导航栏\"}");


var $62b0442a1ca24c49$exports = {};
$62b0442a1ca24c49$exports = JSON.parse("{\"breadcrumbs\":\"導覽列\"}");


$485710c529557361$exports = {
    "ar-AE": $f4882d7260ca8f9d$exports,
    "bg-BG": $4820186fc7eb481b$exports,
    "cs-CZ": $cd21fe388d89ec0b$exports,
    "da-DK": $a073ed7897856adc$exports,
    "de-DE": $6269ed27c37560a5$exports,
    "el-GR": $533ddb487767d55d$exports,
    "en-US": $f2db6bd9510488c7$exports,
    "es-ES": $fb38c4846c09e98f$exports,
    "et-EE": $022a9f75c44b56af$exports,
    "fi-FI": $94b8da7c6bc99373$exports,
    "fr-FR": $438cfb04e4fb7b1d$exports,
    "he-IL": $b89382922d84873c$exports,
    "hr-HR": $13c375f718e9ab73$exports,
    "hu-HU": $99db8e40d6794512$exports,
    "it-IT": $c5ec56fcffed7c2e$exports,
    "ja-JP": $4c818075da01c5fb$exports,
    "ko-KR": $020bec590bb8378d$exports,
    "lt-LT": $9679a565775519e8$exports,
    "lv-LV": $d42942efb802b9f2$exports,
    "nb-NO": $f1b2d49b5f1a53ce$exports,
    "nl-NL": $6335c9f4b1a860c8$exports,
    "pl-PL": $5467dc99696e6e12$exports,
    "pt-BR": $04df9be94af1c63e$exports,
    "pt-PT": $93938dc85cfd9f53$exports,
    "ro-RO": $0f892a5062257f67$exports,
    "ru-RU": $11748af6b7efd7ce$exports,
    "sk-SK": $63c4fc1394d16ef6$exports,
    "sl-SI": $d3f065ea8b3e92d2$exports,
    "sr-SP": $ed32fb8efdd4ae31$exports,
    "sv-SE": $c194c93964d34aa6$exports,
    "tr-TR": $8d07143566be6228$exports,
    "uk-UA": $a9fe916ff1200a3e$exports,
    "zh-CN": $9e50d72eef976ccc$exports,
    "zh-TW": $62b0442a1ca24c49$exports
};



function $5d2fecae9926feeb$export$8cefe241bd876ca0(props) {
    let { 'aria-label': ariaLabel , ...otherProps } = props;
    let formatMessage = $7vd6z$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($485710c529557361$exports)));
    return {
        navProps: {
            ...$7vd6z$filterDOMProps(otherProps, {
                labelable: true
            }),
            'aria-label': ariaLabel || formatMessage('breadcrumbs')
        }
    };
}




export {$c229b6f7e13bf1fc$export$452b38fce62c9384 as useBreadcrumbItem, $5d2fecae9926feeb$export$8cefe241bd876ca0 as useBreadcrumbs};
