// @react-spectrum/contextualhelp/dist/main.js
require("./main.css");
var $47LQ2$reactspectrumbutton = require("@react-spectrum/button");
var $47LQ2$reactspectrumdialog = require("@react-spectrum/dialog");
var $47LQ2$spectrumiconsworkflowHelpOutline = require("@spectrum-icons/workflow/HelpOutline");
var $47LQ2$spectrumiconsworkflowInfoOutline = require("@spectrum-icons/workflow/InfoOutline");
var $47LQ2$react = require("react");
var $47LQ2$reactspectrumutils = require("@react-spectrum/utils");
var $47LQ2$reactariai18n = require("@react-aria/i18n");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $529741ddac2715df$exports = {};

$parcel$export($529741ddac2715df$exports, "ContextualHelp", () => $529741ddac2715df$export$7d3cdb256c2ba320);



var $62308c1230be17df$exports = {};

$parcel$export($62308c1230be17df$exports, "react-spectrum-ContextualHelp-button", () => $62308c1230be17df$export$20d9583c9f1529c1, (v) => $62308c1230be17df$export$20d9583c9f1529c1 = v);
$parcel$export($62308c1230be17df$exports, "react-spectrum-ContextualHelp-dialog", () => $62308c1230be17df$export$ab2d873674d20d28, (v) => $62308c1230be17df$export$ab2d873674d20d28 = v);
$parcel$export($62308c1230be17df$exports, "react-spectrum-ContextualHelp-content", () => $62308c1230be17df$export$a1f805890cf58c4, (v) => $62308c1230be17df$export$a1f805890cf58c4 = v);
$parcel$export($62308c1230be17df$exports, "react-spectrum-ContextualHelp-footer", () => $62308c1230be17df$export$d901bc3fa82b87a6, (v) => $62308c1230be17df$export$d901bc3fa82b87a6 = v);
var $62308c1230be17df$export$20d9583c9f1529c1;
var $62308c1230be17df$export$ab2d873674d20d28;
var $62308c1230be17df$export$a1f805890cf58c4;
var $62308c1230be17df$export$d901bc3fa82b87a6;
$62308c1230be17df$export$20d9583c9f1529c1 = "react-spectrum-ContextualHelp-button_d88303";
$62308c1230be17df$export$ab2d873674d20d28 = "react-spectrum-ContextualHelp-dialog_d88303";
$62308c1230be17df$export$a1f805890cf58c4 = "react-spectrum-ContextualHelp-content_d88303";
$62308c1230be17df$export$d901bc3fa82b87a6 = "react-spectrum-ContextualHelp-footer_d88303";



var $75fed20cc58284fb$exports = {};
var $9d0f904cf907f473$exports = {};
$9d0f904cf907f473$exports = JSON.parse("{\"help\":\"مساعدة\",\"info\":\"معلومات\"}");


var $f44b31e3fc637268$exports = {};
$f44b31e3fc637268$exports = JSON.parse("{\"help\":\"Помощ\",\"info\":\"Информация\"}");


var $ebd70f09596f59ce$exports = {};
$ebd70f09596f59ce$exports = JSON.parse("{\"help\":\"Nápověda\",\"info\":\"Informace\"}");


var $5e5108ea55a0bba2$exports = {};
$5e5108ea55a0bba2$exports = JSON.parse("{\"help\":\"Hjælp\",\"info\":\"Oplysninger\"}");


var $f764c963b1fed4a3$exports = {};
$f764c963b1fed4a3$exports = JSON.parse("{\"help\":\"Hilfe\",\"info\":\"Informationen\"}");


var $fc0a85a2bef8904b$exports = {};
$fc0a85a2bef8904b$exports = JSON.parse("{\"help\":\"Βοήθεια\",\"info\":\"Πληροφορίες\"}");


var $b415f39a89ab7322$exports = {};
$b415f39a89ab7322$exports = JSON.parse("{\"info\":\"Information\",\"help\":\"Help\"}");


var $9286fc6a2e307e8c$exports = {};
$9286fc6a2e307e8c$exports = JSON.parse("{\"help\":\"Ayuda\",\"info\":\"Información\"}");


var $5bbf0ab2e77f17c7$exports = {};
$5bbf0ab2e77f17c7$exports = JSON.parse("{\"help\":\"Spikker\",\"info\":\"Teave\"}");


var $153eda102fd46f8a$exports = {};
$153eda102fd46f8a$exports = JSON.parse("{\"help\":\"Ohje\",\"info\":\"Tiedot\"}");


var $3fb5afbfb942e285$exports = {};
$3fb5afbfb942e285$exports = JSON.parse("{\"help\":\"Aide\",\"info\":\"Informations\"}");


var $9ffe1069c5f575cb$exports = {};
$9ffe1069c5f575cb$exports = JSON.parse("{\"help\":\"עזרה\",\"info\":\"מידע\"}");


var $d9dc906cdbcb2a32$exports = {};
$d9dc906cdbcb2a32$exports = JSON.parse("{\"help\":\"Pomoć\",\"info\":\"Informacije\"}");


var $6308a6e65e3a0aad$exports = {};
$6308a6e65e3a0aad$exports = JSON.parse("{\"help\":\"Súgó\",\"info\":\"Információ\"}");


var $e95fb1c204cf617c$exports = {};
$e95fb1c204cf617c$exports = JSON.parse("{\"help\":\"Aiuto\",\"info\":\"Informazioni\"}");


var $e0a02d4d593fe2dc$exports = {};
$e0a02d4d593fe2dc$exports = JSON.parse("{\"help\":\"ヘルプ\",\"info\":\"情報\"}");


var $2e6f41cbf081d755$exports = {};
$2e6f41cbf081d755$exports = JSON.parse("{\"help\":\"도움말\",\"info\":\"정보\"}");


var $8b90749b48a76c7c$exports = {};
$8b90749b48a76c7c$exports = JSON.parse("{\"help\":\"Žinynas\",\"info\":\"Informacija\"}");


var $dec50ac4c2d60ea9$exports = {};
$dec50ac4c2d60ea9$exports = JSON.parse("{\"help\":\"Palīdzība\",\"info\":\"Informācija\"}");


var $93ebcc444cd7fe5a$exports = {};
$93ebcc444cd7fe5a$exports = JSON.parse("{\"help\":\"Hjelp\",\"info\":\"Informasjon\"}");


var $fa291e62ce894832$exports = {};
$fa291e62ce894832$exports = JSON.parse("{\"help\":\"Help\",\"info\":\"Informatie\"}");


var $9d5ddfe0f49dc937$exports = {};
$9d5ddfe0f49dc937$exports = JSON.parse("{\"help\":\"Pomoc\",\"info\":\"Informacja\"}");


var $14a47a53df35c968$exports = {};
$14a47a53df35c968$exports = JSON.parse("{\"help\":\"Ajuda\",\"info\":\"Informações\"}");


var $29607a9fe863aaf9$exports = {};
$29607a9fe863aaf9$exports = JSON.parse("{\"help\":\"Ajuda\",\"info\":\"Informação\"}");


var $3f909eb30ec7d44e$exports = {};
$3f909eb30ec7d44e$exports = JSON.parse("{\"help\":\"Ajutor\",\"info\":\"Informaţii\"}");


var $fa33993fe533c3db$exports = {};
$fa33993fe533c3db$exports = JSON.parse("{\"help\":\"Справка\",\"info\":\"Информация\"}");


var $25cfb35bf74b51ae$exports = {};
$25cfb35bf74b51ae$exports = JSON.parse("{\"help\":\"Pomoc\",\"info\":\"Informácie\"}");


var $f2acb957e6882090$exports = {};
$f2acb957e6882090$exports = JSON.parse("{\"help\":\"Pomoč\",\"info\":\"Informacije\"}");


var $c3f0157a3097a6b2$exports = {};
$c3f0157a3097a6b2$exports = JSON.parse("{\"help\":\"Pomoć\",\"info\":\"Informacije\"}");


var $d4e1abb832f956d9$exports = {};
$d4e1abb832f956d9$exports = JSON.parse("{\"help\":\"Hjälp\",\"info\":\"Information\"}");


var $2a41cc67db3cd150$exports = {};
$2a41cc67db3cd150$exports = JSON.parse("{\"help\":\"Yardım\",\"info\":\"Bilgiler\"}");


var $ecc37fac2aa7f045$exports = {};
$ecc37fac2aa7f045$exports = JSON.parse("{\"help\":\"Довідка\",\"info\":\"Інформація\"}");


var $1fc7db1cdfa67fc4$exports = {};
$1fc7db1cdfa67fc4$exports = JSON.parse("{\"help\":\"帮助\",\"info\":\"信息\"}");


var $f324da332dae8649$exports = {};
$f324da332dae8649$exports = JSON.parse("{\"help\":\"說明\",\"info\":\"資訊\"}");


$75fed20cc58284fb$exports = {
    "ar-AE": $9d0f904cf907f473$exports,
    "bg-BG": $f44b31e3fc637268$exports,
    "cs-CZ": $ebd70f09596f59ce$exports,
    "da-DK": $5e5108ea55a0bba2$exports,
    "de-DE": $f764c963b1fed4a3$exports,
    "el-GR": $fc0a85a2bef8904b$exports,
    "en-US": $b415f39a89ab7322$exports,
    "es-ES": $9286fc6a2e307e8c$exports,
    "et-EE": $5bbf0ab2e77f17c7$exports,
    "fi-FI": $153eda102fd46f8a$exports,
    "fr-FR": $3fb5afbfb942e285$exports,
    "he-IL": $9ffe1069c5f575cb$exports,
    "hr-HR": $d9dc906cdbcb2a32$exports,
    "hu-HU": $6308a6e65e3a0aad$exports,
    "it-IT": $e95fb1c204cf617c$exports,
    "ja-JP": $e0a02d4d593fe2dc$exports,
    "ko-KR": $2e6f41cbf081d755$exports,
    "lt-LT": $8b90749b48a76c7c$exports,
    "lv-LV": $dec50ac4c2d60ea9$exports,
    "nb-NO": $93ebcc444cd7fe5a$exports,
    "nl-NL": $fa291e62ce894832$exports,
    "pl-PL": $9d5ddfe0f49dc937$exports,
    "pt-BR": $14a47a53df35c968$exports,
    "pt-PT": $29607a9fe863aaf9$exports,
    "ro-RO": $3f909eb30ec7d44e$exports,
    "ru-RU": $fa33993fe533c3db$exports,
    "sk-SK": $25cfb35bf74b51ae$exports,
    "sl-SI": $f2acb957e6882090$exports,
    "sr-SP": $c3f0157a3097a6b2$exports,
    "sv-SE": $d4e1abb832f956d9$exports,
    "tr-TR": $2a41cc67db3cd150$exports,
    "uk-UA": $ecc37fac2aa7f045$exports,
    "zh-CN": $1fc7db1cdfa67fc4$exports,
    "zh-TW": $f324da332dae8649$exports
};





function $529741ddac2715df$var$ContextualHelp(props, ref) {
    let { variant: variant = 'help' , placement: placement = 'bottom start' , children: children , ...otherProps } = props;
    let formatMessage = $47LQ2$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($75fed20cc58284fb$exports)));
    let icon = variant === 'info' ? /*#__PURE__*/ ($parcel$interopDefault($47LQ2$react)).createElement(($parcel$interopDefault($47LQ2$spectrumiconsworkflowInfoOutline)), null) : /*#__PURE__*/ ($parcel$interopDefault($47LQ2$react)).createElement(($parcel$interopDefault($47LQ2$spectrumiconsworkflowHelpOutline)), null);
    let slots = {
        content: {
            UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($62308c1230be17df$exports))['react-spectrum-ContextualHelp-content']
        },
        footer: {
            UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($62308c1230be17df$exports))['react-spectrum-ContextualHelp-footer']
        }
    };
    let ariaLabel = otherProps['aria-label'];
    if (!ariaLabel && !otherProps['aria-labelledby']) ariaLabel = formatMessage(variant);
    return(/*#__PURE__*/ ($parcel$interopDefault($47LQ2$react)).createElement($47LQ2$reactspectrumdialog.DialogTrigger, {
        ...otherProps,
        type: "popover",
        placement: placement,
        hideArrow: true
    }, /*#__PURE__*/ ($parcel$interopDefault($47LQ2$react)).createElement($47LQ2$reactspectrumbutton.ActionButton, {
        ...otherProps,
        ref: ref,
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($62308c1230be17df$exports))['react-spectrum-ContextualHelp-button'],
        isQuiet: true,
        "aria-label": ariaLabel
    }, icon), /*#__PURE__*/ ($parcel$interopDefault($47LQ2$react)).createElement($47LQ2$reactspectrumutils.SlotProvider, {
        slots: slots
    }, /*#__PURE__*/ ($parcel$interopDefault($47LQ2$react)).createElement($47LQ2$reactspectrumdialog.Dialog, {
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($62308c1230be17df$exports))['react-spectrum-ContextualHelp-dialog']
    }, children))));
}
/**
 * Contextual help shows a user extra information about the state of an adjacent component, or a total view.
 */ let $529741ddac2715df$export$7d3cdb256c2ba320 = /*#__PURE__*/ ($parcel$interopDefault($47LQ2$react)).forwardRef($529741ddac2715df$var$ContextualHelp);


$parcel$exportWildcard(module.exports, $529741ddac2715df$exports);
