// @react-spectrum/contextualhelp/dist/module.js
import "./main.css";
import {ActionButton as $gha0s$ActionButton} from "@react-spectrum/button";
import {DialogTrigger as $gha0s$DialogTrigger, Dialog as $gha0s$Dialog} from "@react-spectrum/dialog";
import $gha0s$spectrumiconsworkflowHelpOutline from "@spectrum-icons/workflow/HelpOutline";
import $gha0s$spectrumiconsworkflowInfoOutline from "@spectrum-icons/workflow/InfoOutline";
import $gha0s$react from "react";
import {SlotProvider as $gha0s$SlotProvider} from "@react-spectrum/utils";
import {useMessageFormatter as $gha0s$useMessageFormatter} from "@react-aria/i18n";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $cb65d59c2630eefe$exports = {};

$parcel$export($cb65d59c2630eefe$exports, "ContextualHelp", () => $cb65d59c2630eefe$export$7d3cdb256c2ba320);



var $8d7cfb941bc48111$exports = {};

$parcel$export($8d7cfb941bc48111$exports, "react-spectrum-ContextualHelp-button", () => $8d7cfb941bc48111$export$20d9583c9f1529c1, (v) => $8d7cfb941bc48111$export$20d9583c9f1529c1 = v);
$parcel$export($8d7cfb941bc48111$exports, "react-spectrum-ContextualHelp-dialog", () => $8d7cfb941bc48111$export$ab2d873674d20d28, (v) => $8d7cfb941bc48111$export$ab2d873674d20d28 = v);
$parcel$export($8d7cfb941bc48111$exports, "react-spectrum-ContextualHelp-content", () => $8d7cfb941bc48111$export$a1f805890cf58c4, (v) => $8d7cfb941bc48111$export$a1f805890cf58c4 = v);
$parcel$export($8d7cfb941bc48111$exports, "react-spectrum-ContextualHelp-footer", () => $8d7cfb941bc48111$export$d901bc3fa82b87a6, (v) => $8d7cfb941bc48111$export$d901bc3fa82b87a6 = v);
var $8d7cfb941bc48111$export$20d9583c9f1529c1;
var $8d7cfb941bc48111$export$ab2d873674d20d28;
var $8d7cfb941bc48111$export$a1f805890cf58c4;
var $8d7cfb941bc48111$export$d901bc3fa82b87a6;
$8d7cfb941bc48111$export$20d9583c9f1529c1 = "react-spectrum-ContextualHelp-button_d88303";
$8d7cfb941bc48111$export$ab2d873674d20d28 = "react-spectrum-ContextualHelp-dialog_d88303";
$8d7cfb941bc48111$export$a1f805890cf58c4 = "react-spectrum-ContextualHelp-content_d88303";
$8d7cfb941bc48111$export$d901bc3fa82b87a6 = "react-spectrum-ContextualHelp-footer_d88303";



var $f7905ea23de83350$exports = {};
var $bffc06ab6989610b$exports = {};
$bffc06ab6989610b$exports = JSON.parse("{\"help\":\"مساعدة\",\"info\":\"معلومات\"}");


var $e3643fac7ad02019$exports = {};
$e3643fac7ad02019$exports = JSON.parse("{\"help\":\"Помощ\",\"info\":\"Информация\"}");


var $9560804b088a40eb$exports = {};
$9560804b088a40eb$exports = JSON.parse("{\"help\":\"Nápověda\",\"info\":\"Informace\"}");


var $af4809a8597564f3$exports = {};
$af4809a8597564f3$exports = JSON.parse("{\"help\":\"Hjælp\",\"info\":\"Oplysninger\"}");


var $e6c3c9f4783ad1c5$exports = {};
$e6c3c9f4783ad1c5$exports = JSON.parse("{\"help\":\"Hilfe\",\"info\":\"Informationen\"}");


var $dab66e1f93263686$exports = {};
$dab66e1f93263686$exports = JSON.parse("{\"help\":\"Βοήθεια\",\"info\":\"Πληροφορίες\"}");


var $ab0793759b97b2e9$exports = {};
$ab0793759b97b2e9$exports = JSON.parse("{\"info\":\"Information\",\"help\":\"Help\"}");


var $c2b4e17d882372f8$exports = {};
$c2b4e17d882372f8$exports = JSON.parse("{\"help\":\"Ayuda\",\"info\":\"Información\"}");


var $cae5e6f9ff51e197$exports = {};
$cae5e6f9ff51e197$exports = JSON.parse("{\"help\":\"Spikker\",\"info\":\"Teave\"}");


var $d463b67d3808048e$exports = {};
$d463b67d3808048e$exports = JSON.parse("{\"help\":\"Ohje\",\"info\":\"Tiedot\"}");


var $4d9bf0ad8b28534f$exports = {};
$4d9bf0ad8b28534f$exports = JSON.parse("{\"help\":\"Aide\",\"info\":\"Informations\"}");


var $de77025adac2049f$exports = {};
$de77025adac2049f$exports = JSON.parse("{\"help\":\"עזרה\",\"info\":\"מידע\"}");


var $85468a04ff2ea533$exports = {};
$85468a04ff2ea533$exports = JSON.parse("{\"help\":\"Pomoć\",\"info\":\"Informacije\"}");


var $f07191871bb5d0ac$exports = {};
$f07191871bb5d0ac$exports = JSON.parse("{\"help\":\"Súgó\",\"info\":\"Információ\"}");


var $adbf38d2282d27d3$exports = {};
$adbf38d2282d27d3$exports = JSON.parse("{\"help\":\"Aiuto\",\"info\":\"Informazioni\"}");


var $8ebf6ed01562d999$exports = {};
$8ebf6ed01562d999$exports = JSON.parse("{\"help\":\"ヘルプ\",\"info\":\"情報\"}");


var $6176ab19851742d4$exports = {};
$6176ab19851742d4$exports = JSON.parse("{\"help\":\"도움말\",\"info\":\"정보\"}");


var $c773eca8e6660e14$exports = {};
$c773eca8e6660e14$exports = JSON.parse("{\"help\":\"Žinynas\",\"info\":\"Informacija\"}");


var $814ebb503229ae7e$exports = {};
$814ebb503229ae7e$exports = JSON.parse("{\"help\":\"Palīdzība\",\"info\":\"Informācija\"}");


var $8ce839121a3242ab$exports = {};
$8ce839121a3242ab$exports = JSON.parse("{\"help\":\"Hjelp\",\"info\":\"Informasjon\"}");


var $33619b5f9e88c898$exports = {};
$33619b5f9e88c898$exports = JSON.parse("{\"help\":\"Help\",\"info\":\"Informatie\"}");


var $104b2de588b62ad1$exports = {};
$104b2de588b62ad1$exports = JSON.parse("{\"help\":\"Pomoc\",\"info\":\"Informacja\"}");


var $2b7799038d6cf857$exports = {};
$2b7799038d6cf857$exports = JSON.parse("{\"help\":\"Ajuda\",\"info\":\"Informações\"}");


var $069428e451eb3954$exports = {};
$069428e451eb3954$exports = JSON.parse("{\"help\":\"Ajuda\",\"info\":\"Informação\"}");


var $e9d29e6aae387edb$exports = {};
$e9d29e6aae387edb$exports = JSON.parse("{\"help\":\"Ajutor\",\"info\":\"Informaţii\"}");


var $88dad3837e3c24d9$exports = {};
$88dad3837e3c24d9$exports = JSON.parse("{\"help\":\"Справка\",\"info\":\"Информация\"}");


var $001ec3df43549174$exports = {};
$001ec3df43549174$exports = JSON.parse("{\"help\":\"Pomoc\",\"info\":\"Informácie\"}");


var $841863ad959f1399$exports = {};
$841863ad959f1399$exports = JSON.parse("{\"help\":\"Pomoč\",\"info\":\"Informacije\"}");


var $9ef935374ffd6cdc$exports = {};
$9ef935374ffd6cdc$exports = JSON.parse("{\"help\":\"Pomoć\",\"info\":\"Informacije\"}");


var $fbad1fbbe9ae2f18$exports = {};
$fbad1fbbe9ae2f18$exports = JSON.parse("{\"help\":\"Hjälp\",\"info\":\"Information\"}");


var $d89c91685a1e83f5$exports = {};
$d89c91685a1e83f5$exports = JSON.parse("{\"help\":\"Yardım\",\"info\":\"Bilgiler\"}");


var $14e79ee58db28b6b$exports = {};
$14e79ee58db28b6b$exports = JSON.parse("{\"help\":\"Довідка\",\"info\":\"Інформація\"}");


var $5678b90291eb59fc$exports = {};
$5678b90291eb59fc$exports = JSON.parse("{\"help\":\"帮助\",\"info\":\"信息\"}");


var $e876a95b7da3501c$exports = {};
$e876a95b7da3501c$exports = JSON.parse("{\"help\":\"說明\",\"info\":\"資訊\"}");


$f7905ea23de83350$exports = {
    "ar-AE": $bffc06ab6989610b$exports,
    "bg-BG": $e3643fac7ad02019$exports,
    "cs-CZ": $9560804b088a40eb$exports,
    "da-DK": $af4809a8597564f3$exports,
    "de-DE": $e6c3c9f4783ad1c5$exports,
    "el-GR": $dab66e1f93263686$exports,
    "en-US": $ab0793759b97b2e9$exports,
    "es-ES": $c2b4e17d882372f8$exports,
    "et-EE": $cae5e6f9ff51e197$exports,
    "fi-FI": $d463b67d3808048e$exports,
    "fr-FR": $4d9bf0ad8b28534f$exports,
    "he-IL": $de77025adac2049f$exports,
    "hr-HR": $85468a04ff2ea533$exports,
    "hu-HU": $f07191871bb5d0ac$exports,
    "it-IT": $adbf38d2282d27d3$exports,
    "ja-JP": $8ebf6ed01562d999$exports,
    "ko-KR": $6176ab19851742d4$exports,
    "lt-LT": $c773eca8e6660e14$exports,
    "lv-LV": $814ebb503229ae7e$exports,
    "nb-NO": $8ce839121a3242ab$exports,
    "nl-NL": $33619b5f9e88c898$exports,
    "pl-PL": $104b2de588b62ad1$exports,
    "pt-BR": $2b7799038d6cf857$exports,
    "pt-PT": $069428e451eb3954$exports,
    "ro-RO": $e9d29e6aae387edb$exports,
    "ru-RU": $88dad3837e3c24d9$exports,
    "sk-SK": $001ec3df43549174$exports,
    "sl-SI": $841863ad959f1399$exports,
    "sr-SP": $9ef935374ffd6cdc$exports,
    "sv-SE": $fbad1fbbe9ae2f18$exports,
    "tr-TR": $d89c91685a1e83f5$exports,
    "uk-UA": $14e79ee58db28b6b$exports,
    "zh-CN": $5678b90291eb59fc$exports,
    "zh-TW": $e876a95b7da3501c$exports
};





function $cb65d59c2630eefe$var$ContextualHelp(props, ref) {
    let { variant: variant = 'help' , placement: placement = 'bottom start' , children: children , ...otherProps } = props;
    let formatMessage = $gha0s$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($f7905ea23de83350$exports)));
    let icon = variant === 'info' ? /*#__PURE__*/ $gha0s$react.createElement($gha0s$spectrumiconsworkflowInfoOutline, null) : /*#__PURE__*/ $gha0s$react.createElement($gha0s$spectrumiconsworkflowHelpOutline, null);
    let slots = {
        content: {
            UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($8d7cfb941bc48111$exports))['react-spectrum-ContextualHelp-content']
        },
        footer: {
            UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($8d7cfb941bc48111$exports))['react-spectrum-ContextualHelp-footer']
        }
    };
    let ariaLabel = otherProps['aria-label'];
    if (!ariaLabel && !otherProps['aria-labelledby']) ariaLabel = formatMessage(variant);
    return(/*#__PURE__*/ $gha0s$react.createElement($gha0s$DialogTrigger, {
        ...otherProps,
        type: "popover",
        placement: placement,
        hideArrow: true
    }, /*#__PURE__*/ $gha0s$react.createElement($gha0s$ActionButton, {
        ...otherProps,
        ref: ref,
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($8d7cfb941bc48111$exports))['react-spectrum-ContextualHelp-button'],
        isQuiet: true,
        "aria-label": ariaLabel
    }, icon), /*#__PURE__*/ $gha0s$react.createElement($gha0s$SlotProvider, {
        slots: slots
    }, /*#__PURE__*/ $gha0s$react.createElement($gha0s$Dialog, {
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($8d7cfb941bc48111$exports))['react-spectrum-ContextualHelp-dialog']
    }, children))));
}
/**
 * Contextual help shows a user extra information about the state of an adjacent component, or a total view.
 */ let $cb65d59c2630eefe$export$7d3cdb256c2ba320 = /*#__PURE__*/ $gha0s$react.forwardRef($cb65d59c2630eefe$var$ContextualHelp);




export {$cb65d59c2630eefe$export$7d3cdb256c2ba320 as ContextualHelp};
