// @react-spectrum/button/dist/module.js
import "./main.css";
import {useSlotProps as $gkQn6$useSlotProps, useFocusableRef as $gkQn6$useFocusableRef, useStyleProps as $gkQn6$useStyleProps, classNames as $gkQn6$classNames, SlotProvider as $gkQn6$SlotProvider} from "@react-spectrum/utils";
import {FocusRing as $gkQn6$FocusRing} from "@react-aria/focus";
import {mergeProps as $gkQn6$mergeProps} from "@react-aria/utils";
import $gkQn6$react from "react";
import {Text as $gkQn6$Text} from "@react-spectrum/text";
import {useButton as $gkQn6$useButton, useToggleButton as $gkQn6$useToggleButton} from "@react-aria/button";
import {useHover as $gkQn6$useHover} from "@react-aria/interactions";
import {useProviderProps as $gkQn6$useProviderProps} from "@react-spectrum/provider";
import $gkQn6$spectrumiconsuiCornerTriangle from "@spectrum-icons/ui/CornerTriangle";
import $gkQn6$spectrumiconsuiCrossSmall from "@spectrum-icons/ui/CrossSmall";
import {useToggleState as $gkQn6$useToggleState} from "@react-stately/toggle";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $c8dd5cb5a06d28a6$exports = {};

$parcel$export($c8dd5cb5a06d28a6$exports, "Button", () => $c8dd5cb5a06d28a6$export$353f5b6fc5456de1);




var $8682ec7d9494680f$exports = {};

$parcel$export($8682ec7d9494680f$exports, "spectrum-Button", () => $8682ec7d9494680f$export$1db4cca5b4ade39a, (v) => $8682ec7d9494680f$export$1db4cca5b4ade39a = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton", () => $8682ec7d9494680f$export$53da69f51b770d3, (v) => $8682ec7d9494680f$export$53da69f51b770d3 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-LogicButton", () => $8682ec7d9494680f$export$36ae67a131730482, (v) => $8682ec7d9494680f$export$36ae67a131730482 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-FieldButton", () => $8682ec7d9494680f$export$a024c99ecf6b5741, (v) => $8682ec7d9494680f$export$a024c99ecf6b5741 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ClearButton", () => $8682ec7d9494680f$export$b752ce409e5660c, (v) => $8682ec7d9494680f$export$b752ce409e5660c = v);
$parcel$export($8682ec7d9494680f$exports, "focus-ring", () => $8682ec7d9494680f$export$f39a09f249340e2a, (v) => $8682ec7d9494680f$export$f39a09f249340e2a = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Icon", () => $8682ec7d9494680f$export$d374b04f30360026, (v) => $8682ec7d9494680f$export$d374b04f30360026 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton--emphasized", () => $8682ec7d9494680f$export$5aabb61a966e266f, (v) => $8682ec7d9494680f$export$5aabb61a966e266f = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton--staticColor", () => $8682ec7d9494680f$export$29b7b5d6a5a497c5, (v) => $8682ec7d9494680f$export$29b7b5d6a5a497c5 = v);
$parcel$export($8682ec7d9494680f$exports, "is-hovered", () => $8682ec7d9494680f$export$b8813cd5d7824ce7, (v) => $8682ec7d9494680f$export$b8813cd5d7824ce7 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button-label", () => $8682ec7d9494680f$export$c081ba562e204942, (v) => $8682ec7d9494680f$export$c081ba562e204942 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton-label", () => $8682ec7d9494680f$export$ebb4635c66bacdb9, (v) => $8682ec7d9494680f$export$ebb4635c66bacdb9 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton-hold", () => $8682ec7d9494680f$export$4a4f01c443eff992, (v) => $8682ec7d9494680f$export$4a4f01c443eff992 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton--quiet", () => $8682ec7d9494680f$export$40e1f009544bacf3, (v) => $8682ec7d9494680f$export$40e1f009544bacf3 = v);
$parcel$export($8682ec7d9494680f$exports, "is-disabled", () => $8682ec7d9494680f$export$d35bc1e505d1ebbf, (v) => $8682ec7d9494680f$export$d35bc1e505d1ebbf = v);
$parcel$export($8682ec7d9494680f$exports, "is-open", () => $8682ec7d9494680f$export$a9781837241c946d, (v) => $8682ec7d9494680f$export$a9781837241c946d = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-FieldButton--quiet", () => $8682ec7d9494680f$export$5ca7b274e4fff9fc, (v) => $8682ec7d9494680f$export$5ca7b274e4fff9fc = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ClearButton--overBackground", () => $8682ec7d9494680f$export$116f2fccf941d34d, (v) => $8682ec7d9494680f$export$116f2fccf941d34d = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ClearButton--small", () => $8682ec7d9494680f$export$49280ffa5236ba8b, (v) => $8682ec7d9494680f$export$49280ffa5236ba8b = v);
$parcel$export($8682ec7d9494680f$exports, "is-focused", () => $8682ec7d9494680f$export$e7dc768d35940237, (v) => $8682ec7d9494680f$export$e7dc768d35940237 = v);
$parcel$export($8682ec7d9494680f$exports, "is-active", () => $8682ec7d9494680f$export$20fd0f7cd4e6112f, (v) => $8682ec7d9494680f$export$20fd0f7cd4e6112f = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--cta", () => $8682ec7d9494680f$export$b818fdf0052a2cab, (v) => $8682ec7d9494680f$export$b818fdf0052a2cab = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--primary", () => $8682ec7d9494680f$export$c1c3511f5e96e926, (v) => $8682ec7d9494680f$export$c1c3511f5e96e926 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--secondary", () => $8682ec7d9494680f$export$4f1b75526afd8ff3, (v) => $8682ec7d9494680f$export$4f1b75526afd8ff3 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--warning", () => $8682ec7d9494680f$export$6f97cef817ff1cac, (v) => $8682ec7d9494680f$export$6f97cef817ff1cac = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--overBackground", () => $8682ec7d9494680f$export$5cda1128f9112d2e, (v) => $8682ec7d9494680f$export$5cda1128f9112d2e = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--quiet", () => $8682ec7d9494680f$export$25f049c049371040, (v) => $8682ec7d9494680f$export$25f049c049371040 = v);
$parcel$export($8682ec7d9494680f$exports, "is-selected", () => $8682ec7d9494680f$export$1e0fb04f31d3c22a, (v) => $8682ec7d9494680f$export$1e0fb04f31d3c22a = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton--staticWhite", () => $8682ec7d9494680f$export$ccc536e1adbdc059, (v) => $8682ec7d9494680f$export$ccc536e1adbdc059 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton--staticBlack", () => $8682ec7d9494680f$export$8f7c9db66978d20f, (v) => $8682ec7d9494680f$export$8f7c9db66978d20f = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-LogicButton--and", () => $8682ec7d9494680f$export$e52daab43a62c528, (v) => $8682ec7d9494680f$export$e52daab43a62c528 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-LogicButton--or", () => $8682ec7d9494680f$export$da76be6126aaa633, (v) => $8682ec7d9494680f$export$da76be6126aaa633 = v);
$parcel$export($8682ec7d9494680f$exports, "is-placeholder", () => $8682ec7d9494680f$export$e5b2f5233e4e5194, (v) => $8682ec7d9494680f$export$e5b2f5233e4e5194 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-FieldButton--invalid", () => $8682ec7d9494680f$export$c9f503f672e8a3c1, (v) => $8682ec7d9494680f$export$c9f503f672e8a3c1 = v);
var $8682ec7d9494680f$export$1db4cca5b4ade39a;
var $8682ec7d9494680f$export$53da69f51b770d3;
var $8682ec7d9494680f$export$36ae67a131730482;
var $8682ec7d9494680f$export$a024c99ecf6b5741;
var $8682ec7d9494680f$export$b752ce409e5660c;
var $8682ec7d9494680f$export$f39a09f249340e2a;
var $8682ec7d9494680f$export$d374b04f30360026;
var $8682ec7d9494680f$export$5aabb61a966e266f;
var $8682ec7d9494680f$export$29b7b5d6a5a497c5;
var $8682ec7d9494680f$export$b8813cd5d7824ce7;
var $8682ec7d9494680f$export$c081ba562e204942;
var $8682ec7d9494680f$export$ebb4635c66bacdb9;
var $8682ec7d9494680f$export$4a4f01c443eff992;
var $8682ec7d9494680f$export$40e1f009544bacf3;
var $8682ec7d9494680f$export$d35bc1e505d1ebbf;
var $8682ec7d9494680f$export$a9781837241c946d;
var $8682ec7d9494680f$export$5ca7b274e4fff9fc;
var $8682ec7d9494680f$export$116f2fccf941d34d;
var $8682ec7d9494680f$export$49280ffa5236ba8b;
var $8682ec7d9494680f$export$e7dc768d35940237;
var $8682ec7d9494680f$export$20fd0f7cd4e6112f;
var $8682ec7d9494680f$export$b818fdf0052a2cab;
var $8682ec7d9494680f$export$c1c3511f5e96e926;
var $8682ec7d9494680f$export$4f1b75526afd8ff3;
var $8682ec7d9494680f$export$6f97cef817ff1cac;
var $8682ec7d9494680f$export$5cda1128f9112d2e;
var $8682ec7d9494680f$export$25f049c049371040;
var $8682ec7d9494680f$export$1e0fb04f31d3c22a;
var $8682ec7d9494680f$export$ccc536e1adbdc059;
var $8682ec7d9494680f$export$8f7c9db66978d20f;
var $8682ec7d9494680f$export$e52daab43a62c528;
var $8682ec7d9494680f$export$da76be6126aaa633;
var $8682ec7d9494680f$export$e5b2f5233e4e5194;
var $8682ec7d9494680f$export$c9f503f672e8a3c1;
$8682ec7d9494680f$export$1db4cca5b4ade39a = "spectrum-Button_e2d99e";
$8682ec7d9494680f$export$53da69f51b770d3 = "spectrum-ActionButton_e2d99e";
$8682ec7d9494680f$export$36ae67a131730482 = "spectrum-LogicButton_e2d99e";
$8682ec7d9494680f$export$a024c99ecf6b5741 = "spectrum-FieldButton_e2d99e";
$8682ec7d9494680f$export$b752ce409e5660c = "spectrum-ClearButton_e2d99e";
$8682ec7d9494680f$export$f39a09f249340e2a = "focus-ring_e2d99e";
$8682ec7d9494680f$export$d374b04f30360026 = "spectrum-Icon_e2d99e";
$8682ec7d9494680f$export$5aabb61a966e266f = "spectrum-ActionButton--emphasized_e2d99e";
$8682ec7d9494680f$export$29b7b5d6a5a497c5 = "spectrum-ActionButton--staticColor_e2d99e";
$8682ec7d9494680f$export$b8813cd5d7824ce7 = "is-hovered_e2d99e";
$8682ec7d9494680f$export$c081ba562e204942 = "spectrum-Button-label_e2d99e";
$8682ec7d9494680f$export$ebb4635c66bacdb9 = "spectrum-ActionButton-label_e2d99e";
$8682ec7d9494680f$export$4a4f01c443eff992 = "spectrum-ActionButton-hold_e2d99e";
$8682ec7d9494680f$export$40e1f009544bacf3 = "spectrum-ActionButton--quiet_e2d99e";
$8682ec7d9494680f$export$d35bc1e505d1ebbf = "is-disabled_e2d99e";
$8682ec7d9494680f$export$a9781837241c946d = "is-open_e2d99e";
$8682ec7d9494680f$export$5ca7b274e4fff9fc = "spectrum-FieldButton--quiet_e2d99e";
$8682ec7d9494680f$export$116f2fccf941d34d = "spectrum-ClearButton--overBackground_e2d99e";
$8682ec7d9494680f$export$49280ffa5236ba8b = "spectrum-ClearButton--small_e2d99e";
$8682ec7d9494680f$export$e7dc768d35940237 = "is-focused_e2d99e";
$8682ec7d9494680f$export$20fd0f7cd4e6112f = "is-active_e2d99e";
$8682ec7d9494680f$export$b818fdf0052a2cab = "spectrum-Button--cta_e2d99e";
$8682ec7d9494680f$export$c1c3511f5e96e926 = "spectrum-Button--primary_e2d99e";
$8682ec7d9494680f$export$4f1b75526afd8ff3 = "spectrum-Button--secondary_e2d99e";
$8682ec7d9494680f$export$6f97cef817ff1cac = "spectrum-Button--warning_e2d99e";
$8682ec7d9494680f$export$5cda1128f9112d2e = "spectrum-Button--overBackground_e2d99e";
$8682ec7d9494680f$export$25f049c049371040 = "spectrum-Button--quiet_e2d99e";
$8682ec7d9494680f$export$1e0fb04f31d3c22a = "is-selected_e2d99e";
$8682ec7d9494680f$export$ccc536e1adbdc059 = "spectrum-ActionButton--staticWhite_e2d99e";
$8682ec7d9494680f$export$8f7c9db66978d20f = "spectrum-ActionButton--staticBlack_e2d99e";
$8682ec7d9494680f$export$e52daab43a62c528 = "spectrum-LogicButton--and_e2d99e";
$8682ec7d9494680f$export$da76be6126aaa633 = "spectrum-LogicButton--or_e2d99e";
$8682ec7d9494680f$export$e5b2f5233e4e5194 = "is-placeholder_e2d99e";
$8682ec7d9494680f$export$c9f503f672e8a3c1 = "spectrum-FieldButton--invalid_e2d99e";






// todo: CSS hasn't caught up yet, map
let $c8dd5cb5a06d28a6$var$VARIANT_MAPPING = {
    negative: 'warning'
};
function $c8dd5cb5a06d28a6$var$Button(props, ref) {
    props = $gkQn6$useProviderProps(props);
    props = $gkQn6$useSlotProps(props, 'button');
    let { elementType: ElementType = 'button' , children: children , variant: variant , isQuiet: isQuiet , isDisabled: isDisabled , autoFocus: autoFocus , ...otherProps } = props;
    let domRef = $gkQn6$useFocusableRef(ref);
    let { buttonProps: buttonProps , isPressed: isPressed  } = $gkQn6$useButton(props, domRef);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $gkQn6$useHover({
        isDisabled: isDisabled
    });
    let { styleProps: styleProps  } = $gkQn6$useStyleProps(otherProps);
    let buttonVariant = variant;
    if ($c8dd5cb5a06d28a6$var$VARIANT_MAPPING[variant]) buttonVariant = $c8dd5cb5a06d28a6$var$VARIANT_MAPPING[variant];
    return(/*#__PURE__*/ $gkQn6$react.createElement($gkQn6$FocusRing, {
        focusRingClass: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ $gkQn6$react.createElement(ElementType, {
        ...styleProps,
        ...$gkQn6$mergeProps(buttonProps, hoverProps),
        ref: domRef,
        className: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-Button', `spectrum-Button--${buttonVariant}`, {
            'spectrum-Button--quiet': isQuiet,
            'is-disabled': isDisabled,
            'is-active': isPressed,
            'is-hovered': isHovered
        }, styleProps.className)
    }, /*#__PURE__*/ $gkQn6$react.createElement($gkQn6$SlotProvider, {
        slots: {
            icon: {
                size: 'S',
                UNSAFE_className: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-Icon')
            },
            text: {
                UNSAFE_className: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-Button-label')
            }
        }
    }, typeof children === 'string' ? /*#__PURE__*/ $gkQn6$react.createElement($gkQn6$Text, null, children) : children))));
}
/**
 * Buttons allow users to perform an action or to navigate to another page.
 * They have multiple styles for various needs, and are ideal for calling attention to
 * where a user needs to do something in order to move forward in a flow.
 */ let $c8dd5cb5a06d28a6$export$353f5b6fc5456de1 = /*#__PURE__*/ $gkQn6$react.forwardRef($c8dd5cb5a06d28a6$var$Button);


var $5bd673c069f530a7$exports = {};

$parcel$export($5bd673c069f530a7$exports, "ActionButton", () => $5bd673c069f530a7$export$cfc7921d29ef7b80);










function $5bd673c069f530a7$var$ActionButton(props, ref) {
    props = $gkQn6$useProviderProps(props);
    props = $gkQn6$useSlotProps(props, 'actionButton');
    let { isQuiet: isQuiet , isDisabled: isDisabled , staticColor: staticColor , children: children , autoFocus: autoFocus , holdAffordance: // @ts-ignore (private)
    holdAffordance , ...otherProps } = props;
    let domRef = $gkQn6$useFocusableRef(ref);
    let { buttonProps: buttonProps , isPressed: isPressed  } = $gkQn6$useButton(props, domRef);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $gkQn6$useHover({
        isDisabled: isDisabled
    });
    let { styleProps: styleProps  } = $gkQn6$useStyleProps(otherProps);
    let isTextOnly = $gkQn6$react.Children.toArray(props.children).every((c)=>!/*#__PURE__*/ $gkQn6$react.isValidElement(c)
    );
    return(/*#__PURE__*/ $gkQn6$react.createElement($gkQn6$FocusRing, {
        focusRingClass: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ $gkQn6$react.createElement("button", {
        ...styleProps,
        ...$gkQn6$mergeProps(buttonProps, hoverProps),
        ref: domRef,
        className: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-ActionButton', {
            'spectrum-ActionButton--quiet': isQuiet,
            'spectrum-ActionButton--staticColor': !!staticColor,
            'spectrum-ActionButton--staticWhite': staticColor === 'white',
            'spectrum-ActionButton--staticBlack': staticColor === 'black',
            'is-active': isPressed,
            'is-disabled': isDisabled,
            'is-hovered': isHovered
        }, styleProps.className)
    }, holdAffordance && /*#__PURE__*/ $gkQn6$react.createElement($gkQn6$spectrumiconsuiCornerTriangle, {
        UNSAFE_className: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-ActionButton-hold')
    }), /*#__PURE__*/ $gkQn6$react.createElement($gkQn6$SlotProvider, {
        slots: {
            icon: {
                size: 'S',
                UNSAFE_className: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-Icon')
            },
            text: {
                UNSAFE_className: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-ActionButton-label')
            }
        }
    }, typeof children === 'string' || isTextOnly ? /*#__PURE__*/ $gkQn6$react.createElement($gkQn6$Text, null, children) : children))));
}
/**
 * ActionButtons allow users to perform an action.
 * They’re used for similar, task-based options within a workflow, and are ideal for interfaces where buttons aren’t meant to draw a lot of attention.
 */ let $5bd673c069f530a7$export$cfc7921d29ef7b80 = /*#__PURE__*/ $gkQn6$react.forwardRef($5bd673c069f530a7$var$ActionButton);


var $980303e6527fb30a$exports = {};

$parcel$export($980303e6527fb30a$exports, "FieldButton", () => $980303e6527fb30a$export$47dc48f595b075da);







// @private
function $980303e6527fb30a$var$FieldButton(props, ref) {
    props = $gkQn6$useSlotProps(props, 'button');
    let { isQuiet: isQuiet , isDisabled: isDisabled , validationState: validationState , children: children , autoFocus: autoFocus , isActive: isActive , focusRingClass: focusRingClass , ...otherProps } = props;
    let domRef = $gkQn6$useFocusableRef(ref);
    let { buttonProps: buttonProps , isPressed: isPressed  } = $gkQn6$useButton(props, domRef);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $gkQn6$useHover({
        isDisabled: isDisabled
    });
    let { styleProps: styleProps  } = $gkQn6$useStyleProps(otherProps);
    return(/*#__PURE__*/ $gkQn6$react.createElement($gkQn6$FocusRing, {
        focusRingClass: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'focus-ring', focusRingClass),
        autoFocus: autoFocus
    }, /*#__PURE__*/ $gkQn6$react.createElement("button", {
        ...$gkQn6$mergeProps(buttonProps, hoverProps),
        ref: domRef,
        className: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-FieldButton', {
            'spectrum-FieldButton--quiet': isQuiet,
            'is-active': isActive || isPressed,
            'is-disabled': isDisabled,
            'spectrum-FieldButton--invalid': validationState === 'invalid',
            'is-hovered': isHovered
        }, styleProps.className)
    }, /*#__PURE__*/ $gkQn6$react.createElement($gkQn6$SlotProvider, {
        slots: {
            icon: {
                size: 'S',
                UNSAFE_className: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-Icon')
            }
        }
    }, children))));
}
let $980303e6527fb30a$export$47dc48f595b075da = /*#__PURE__*/ $gkQn6$react.forwardRef($980303e6527fb30a$var$FieldButton);


var $8f2ce7ade3eac4ab$exports = {};

$parcel$export($8f2ce7ade3eac4ab$exports, "LogicButton", () => $8f2ce7ade3eac4ab$export$9b0b80fed00ba8b1);








function $8f2ce7ade3eac4ab$var$LogicButton(props, ref) {
    props = $gkQn6$useProviderProps(props);
    let { variant: variant , children: children , isDisabled: isDisabled , autoFocus: autoFocus , ...otherProps } = props;
    let domRef = $gkQn6$useFocusableRef(ref);
    let { buttonProps: buttonProps , isPressed: isPressed  } = $gkQn6$useButton(props, domRef);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $gkQn6$useHover({
        isDisabled: isDisabled
    });
    let { styleProps: styleProps  } = $gkQn6$useStyleProps(otherProps);
    return(/*#__PURE__*/ $gkQn6$react.createElement($gkQn6$FocusRing, {
        focusRingClass: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ $gkQn6$react.createElement("button", {
        ...styleProps,
        ...$gkQn6$mergeProps(buttonProps, hoverProps),
        ref: domRef,
        className: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-LogicButton', {
            [`spectrum-LogicButton--${variant}`]: variant,
            'is-disabled': isDisabled,
            'is-active': isPressed,
            'is-hovered': isHovered
        }, styleProps.className)
    }, /*#__PURE__*/ $gkQn6$react.createElement("span", {
        className: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-Button-label')
    }, children))));
}
/**
 * A LogicButton displays an operator within a boolean logic sequence.
 */ let $8f2ce7ade3eac4ab$export$9b0b80fed00ba8b1 = /*#__PURE__*/ $gkQn6$react.forwardRef($8f2ce7ade3eac4ab$var$LogicButton);


var $644b5097aad32977$exports = {};

$parcel$export($644b5097aad32977$exports, "ClearButton", () => $644b5097aad32977$export$13ec83e50bf04290);








function $644b5097aad32977$var$ClearButton(props, ref) {
    let { children: children = /*#__PURE__*/ $gkQn6$react.createElement($gkQn6$spectrumiconsuiCrossSmall, {
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports))['spectrum-Icon']
    }) , focusClassName: focusClassName , variant: variant , autoFocus: autoFocus , isDisabled: isDisabled , preventFocus: preventFocus , elementType: elementType = preventFocus ? 'div' : 'button' , ...otherProps } = props;
    let domRef = $gkQn6$useFocusableRef(ref);
    let { buttonProps: buttonProps , isPressed: isPressed  } = $gkQn6$useButton({
        ...props,
        elementType: elementType
    }, domRef);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $gkQn6$useHover({
        isDisabled: isDisabled
    });
    let { styleProps: styleProps  } = $gkQn6$useStyleProps(otherProps);
    // For cases like the clear button in a search field, remove the tabIndex so
    // iOS 14 with VoiceOver doesn't focus the button and hide the keyboard when
    // moving the cursor over the clear button.
    if (preventFocus) delete buttonProps.tabIndex;
    let ElementType = elementType;
    return(/*#__PURE__*/ $gkQn6$react.createElement($gkQn6$FocusRing, {
        focusRingClass: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'focus-ring', focusClassName),
        autoFocus: autoFocus
    }, /*#__PURE__*/ $gkQn6$react.createElement(ElementType, {
        ...styleProps,
        ...$gkQn6$mergeProps(buttonProps, hoverProps),
        ref: domRef,
        className: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-ClearButton', {
            [`spectrum-ClearButton--${variant}`]: variant,
            'is-disabled': isDisabled,
            'is-active': isPressed,
            'is-hovered': isHovered
        }, styleProps.className)
    }, children)));
}
let $644b5097aad32977$export$13ec83e50bf04290 = /*#__PURE__*/ $gkQn6$react.forwardRef($644b5097aad32977$var$ClearButton);


var $a646a64e8f7197de$exports = {};

$parcel$export($a646a64e8f7197de$exports, "ToggleButton", () => $a646a64e8f7197de$export$d2b052e7b4be1756);










function $a646a64e8f7197de$var$ToggleButton(props, ref) {
    props = $gkQn6$useProviderProps(props);
    let { isQuiet: isQuiet , isDisabled: isDisabled , isEmphasized: isEmphasized , staticColor: staticColor , children: children , autoFocus: autoFocus , ...otherProps } = props;
    let domRef = $gkQn6$useFocusableRef(ref);
    let state = $gkQn6$useToggleState(props);
    let { buttonProps: buttonProps , isPressed: isPressed  } = $gkQn6$useToggleButton(props, state, domRef);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $gkQn6$useHover({
        isDisabled: isDisabled
    });
    let { styleProps: styleProps  } = $gkQn6$useStyleProps(otherProps);
    let isTextOnly = $gkQn6$react.Children.toArray(props.children).every((c)=>!/*#__PURE__*/ $gkQn6$react.isValidElement(c)
    );
    return(/*#__PURE__*/ $gkQn6$react.createElement($gkQn6$FocusRing, {
        focusRingClass: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ $gkQn6$react.createElement("button", {
        ...styleProps,
        ...$gkQn6$mergeProps(buttonProps, hoverProps),
        ref: domRef,
        className: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-ActionButton', {
            'spectrum-ActionButton--quiet': isQuiet,
            'spectrum-ActionButton--emphasized': isEmphasized,
            'spectrum-ActionButton--staticColor': !!staticColor,
            'spectrum-ActionButton--staticWhite': staticColor === 'white',
            'spectrum-ActionButton--staticBlack': staticColor === 'black',
            'is-active': isPressed,
            'is-disabled': isDisabled,
            'is-hovered': isHovered,
            'is-selected': state.isSelected
        }, styleProps.className)
    }, /*#__PURE__*/ $gkQn6$react.createElement($gkQn6$SlotProvider, {
        slots: {
            icon: {
                size: 'S',
                UNSAFE_className: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-Icon')
            },
            text: {
                UNSAFE_className: $gkQn6$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-ActionButton-label')
            }
        }
    }, typeof children === 'string' || isTextOnly ? /*#__PURE__*/ $gkQn6$react.createElement($gkQn6$Text, null, children) : children))));
}
/**
 * ToggleButtons allow users to toggle a selection on or off, for example
 * switching between two states or modes.
 */ let $a646a64e8f7197de$export$d2b052e7b4be1756 = /*#__PURE__*/ $gkQn6$react.forwardRef($a646a64e8f7197de$var$ToggleButton);




export {$c8dd5cb5a06d28a6$export$353f5b6fc5456de1 as Button, $5bd673c069f530a7$export$cfc7921d29ef7b80 as ActionButton, $980303e6527fb30a$export$47dc48f595b075da as FieldButton, $8f2ce7ade3eac4ab$export$9b0b80fed00ba8b1 as LogicButton, $644b5097aad32977$export$13ec83e50bf04290 as ClearButton, $a646a64e8f7197de$export$d2b052e7b4be1756 as ToggleButton};
