// @react-spectrum/overlays/dist/main.js
require("./main.css");
var $g65uJ$reactspectrumprovider = require("@react-spectrum/provider");
var $g65uJ$react = require("react");
var $g65uJ$reactdom = require("react-dom");
var $g65uJ$reacttransitiongroupTransition = require("react-transition-group/Transition");
var $g65uJ$reactspectrumutils = require("@react-spectrum/utils");
var $g65uJ$reactariautils = require("@react-aria/utils");
var $g65uJ$reactariaoverlays = require("@react-aria/overlays");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $9827034998823bc2$exports = {};

$parcel$export($9827034998823bc2$exports, "Overlay", () => $9827034998823bc2$export$c6fdb837b070b4ff);
var $ac0466425dbf7f76$exports = {};

$parcel$export($ac0466425dbf7f76$exports, "OpenTransition", () => $ac0466425dbf7f76$export$b847a40ee92eff38);


const $ac0466425dbf7f76$var$OPEN_STATES = {
    entering: false,
    entered: true
};
function $ac0466425dbf7f76$export$b847a40ee92eff38(props) {
    return(/*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement(($parcel$interopDefault($g65uJ$reacttransitiongroupTransition)), {
        timeout: {
            enter: 0,
            exit: 350
        },
        ...props
    }, (state)=>($parcel$interopDefault($g65uJ$react)).Children.map(props.children, (child)=>child && /*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).cloneElement(child, {
                isOpen: !!$ac0466425dbf7f76$var$OPEN_STATES[state]
            })
        )
    ));
}





function $9827034998823bc2$var$Overlay(props, ref) {
    let { children: children , isOpen: isOpen , container: container , onEnter: onEnter , onEntering: onEntering , onEntered: onEntered , onExit: onExit , onExiting: onExiting , onExited: onExited  } = props;
    let [exited, setExited] = $g65uJ$react.useState(!isOpen);
    let handleEntered = $g65uJ$react.useCallback(()=>{
        setExited(false);
        if (onEntered) onEntered();
    }, [
        onEntered
    ]);
    let handleExited = $g65uJ$react.useCallback(()=>{
        setExited(true);
        if (onExited) onExited();
    }, [
        onExited
    ]);
    // Don't un-render the overlay while it's transitioning out.
    let mountOverlay = isOpen || !exited;
    if (!mountOverlay) // Don't bother showing anything if we don't have to.
    return null;
    let contents = /*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement($g65uJ$reactspectrumprovider.Provider, {
        ref: ref,
        UNSAFE_style: {
            background: 'transparent',
            isolation: 'isolate'
        }
    }, /*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement($ac0466425dbf7f76$export$b847a40ee92eff38, {
        in: isOpen,
        appear: true,
        onExit: onExit,
        onExiting: onExiting,
        onExited: handleExited,
        onEnter: onEnter,
        onEntering: onEntering,
        onEntered: handleEntered
    }, children));
    return(/*#__PURE__*/ ($parcel$interopDefault($g65uJ$reactdom)).createPortal(contents, container || document.body));
}
let $9827034998823bc2$export$c6fdb837b070b4ff = /*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).forwardRef($9827034998823bc2$var$Overlay);


var $f4e3e1a9e3fff0e3$exports = {};

$parcel$export($f4e3e1a9e3fff0e3$exports, "Popover", () => $f4e3e1a9e3fff0e3$export$5b6b19405a83ff9d);



var $92f1a2282fd787c0$exports = {};

$parcel$export($92f1a2282fd787c0$exports, "spectrum-Popover", () => $92f1a2282fd787c0$export$a62dc1b6ab23a7bb, (v) => $92f1a2282fd787c0$export$a62dc1b6ab23a7bb = v);
$parcel$export($92f1a2282fd787c0$exports, "react-spectrum-Popover", () => $92f1a2282fd787c0$export$884f6b5065bdc41a, (v) => $92f1a2282fd787c0$export$884f6b5065bdc41a = v);
$parcel$export($92f1a2282fd787c0$exports, "spectrum-Dialog-content", () => $92f1a2282fd787c0$export$cb8eccd0f3639238, (v) => $92f1a2282fd787c0$export$cb8eccd0f3639238 = v);
$parcel$export($92f1a2282fd787c0$exports, "spectrum-Modal-wrapper", () => $92f1a2282fd787c0$export$35bea187ff802494, (v) => $92f1a2282fd787c0$export$35bea187ff802494 = v);
$parcel$export($92f1a2282fd787c0$exports, "react-spectrum-Modal-wrapper", () => $92f1a2282fd787c0$export$e088717c213c31d7, (v) => $92f1a2282fd787c0$export$e088717c213c31d7 = v);
$parcel$export($92f1a2282fd787c0$exports, "spectrum-Modal", () => $92f1a2282fd787c0$export$4c0b1e44c3834c85, (v) => $92f1a2282fd787c0$export$4c0b1e44c3834c85 = v);
$parcel$export($92f1a2282fd787c0$exports, "react-spectrum-Modal", () => $92f1a2282fd787c0$export$f8edeb62d7fed8c1, (v) => $92f1a2282fd787c0$export$f8edeb62d7fed8c1 = v);
$parcel$export($92f1a2282fd787c0$exports, "spectrum-Tray", () => $92f1a2282fd787c0$export$ce0704cd084c4f0d, (v) => $92f1a2282fd787c0$export$ce0704cd084c4f0d = v);
$parcel$export($92f1a2282fd787c0$exports, "react-spectrum-Tray", () => $92f1a2282fd787c0$export$69dafa24343974dd, (v) => $92f1a2282fd787c0$export$69dafa24343974dd = v);
var $92f1a2282fd787c0$export$a62dc1b6ab23a7bb;
var $92f1a2282fd787c0$export$884f6b5065bdc41a;
var $92f1a2282fd787c0$export$cb8eccd0f3639238;
var $92f1a2282fd787c0$export$35bea187ff802494;
var $92f1a2282fd787c0$export$e088717c213c31d7;
var $92f1a2282fd787c0$export$4c0b1e44c3834c85;
var $92f1a2282fd787c0$export$f8edeb62d7fed8c1;
var $92f1a2282fd787c0$export$ce0704cd084c4f0d;
var $92f1a2282fd787c0$export$69dafa24343974dd;
$92f1a2282fd787c0$export$a62dc1b6ab23a7bb = "spectrum-Popover_38efea";
$92f1a2282fd787c0$export$884f6b5065bdc41a = "react-spectrum-Popover_38efea";
$92f1a2282fd787c0$export$cb8eccd0f3639238 = "spectrum-Dialog-content_38efea";
$92f1a2282fd787c0$export$35bea187ff802494 = "spectrum-Modal-wrapper_38efea";
$92f1a2282fd787c0$export$e088717c213c31d7 = "react-spectrum-Modal-wrapper_38efea";
$92f1a2282fd787c0$export$4c0b1e44c3834c85 = "spectrum-Modal_38efea";
$92f1a2282fd787c0$export$f8edeb62d7fed8c1 = "react-spectrum-Modal_38efea";
$92f1a2282fd787c0$export$ce0704cd084c4f0d = "spectrum-Tray_38efea";
$92f1a2282fd787c0$export$69dafa24343974dd = "react-spectrum-Tray_38efea";



var $712eff1af44443cd$exports = {};

$parcel$export($712eff1af44443cd$exports, "spectrum-Popover", () => $712eff1af44443cd$export$a62dc1b6ab23a7bb, (v) => $712eff1af44443cd$export$a62dc1b6ab23a7bb = v);
$parcel$export($712eff1af44443cd$exports, "is-open", () => $712eff1af44443cd$export$a9781837241c946d, (v) => $712eff1af44443cd$export$a9781837241c946d = v);
$parcel$export($712eff1af44443cd$exports, "spectrum-Popover--bottom", () => $712eff1af44443cd$export$6a19b19b14780d4a, (v) => $712eff1af44443cd$export$6a19b19b14780d4a = v);
$parcel$export($712eff1af44443cd$exports, "spectrum-Popover--top", () => $712eff1af44443cd$export$f30becfb1df0cae4, (v) => $712eff1af44443cd$export$f30becfb1df0cae4 = v);
$parcel$export($712eff1af44443cd$exports, "spectrum-Popover--right", () => $712eff1af44443cd$export$5fc8b429396c9f58, (v) => $712eff1af44443cd$export$5fc8b429396c9f58 = v);
$parcel$export($712eff1af44443cd$exports, "spectrum-Popover--left", () => $712eff1af44443cd$export$79cc55906ac5b00, (v) => $712eff1af44443cd$export$79cc55906ac5b00 = v);
$parcel$export($712eff1af44443cd$exports, "spectrum-Popover-tip", () => $712eff1af44443cd$export$3100ec4dc1668df1, (v) => $712eff1af44443cd$export$3100ec4dc1668df1 = v);
$parcel$export($712eff1af44443cd$exports, "spectrum-Popover-tip-triangle", () => $712eff1af44443cd$export$ed733fa6cf19ab80, (v) => $712eff1af44443cd$export$ed733fa6cf19ab80 = v);
$parcel$export($712eff1af44443cd$exports, "spectrum-Popover--dialog", () => $712eff1af44443cd$export$cf634262e726cd19, (v) => $712eff1af44443cd$export$cf634262e726cd19 = v);
$parcel$export($712eff1af44443cd$exports, "spectrum-Popover--withTip", () => $712eff1af44443cd$export$d8b16c5377728262, (v) => $712eff1af44443cd$export$d8b16c5377728262 = v);
var $712eff1af44443cd$export$a62dc1b6ab23a7bb;
var $712eff1af44443cd$export$a9781837241c946d;
var $712eff1af44443cd$export$6a19b19b14780d4a;
var $712eff1af44443cd$export$f30becfb1df0cae4;
var $712eff1af44443cd$export$5fc8b429396c9f58;
var $712eff1af44443cd$export$79cc55906ac5b00;
var $712eff1af44443cd$export$3100ec4dc1668df1;
var $712eff1af44443cd$export$ed733fa6cf19ab80;
var $712eff1af44443cd$export$cf634262e726cd19;
var $712eff1af44443cd$export$d8b16c5377728262;
$712eff1af44443cd$export$a62dc1b6ab23a7bb = "spectrum-Popover_6115b8";
$712eff1af44443cd$export$a9781837241c946d = "is-open_6115b8";
$712eff1af44443cd$export$6a19b19b14780d4a = "spectrum-Popover--bottom_6115b8";
$712eff1af44443cd$export$f30becfb1df0cae4 = "spectrum-Popover--top_6115b8";
$712eff1af44443cd$export$5fc8b429396c9f58 = "spectrum-Popover--right_6115b8";
$712eff1af44443cd$export$79cc55906ac5b00 = "spectrum-Popover--left_6115b8";
$712eff1af44443cd$export$3100ec4dc1668df1 = "spectrum-Popover-tip_6115b8";
$712eff1af44443cd$export$ed733fa6cf19ab80 = "spectrum-Popover-tip-triangle_6115b8";
$712eff1af44443cd$export$cf634262e726cd19 = "spectrum-Popover--dialog_6115b8";
$712eff1af44443cd$export$d8b16c5377728262 = "spectrum-Popover--withTip_6115b8";



/**
 * Arrow placement can be done pointing right or down because those paths start at 0, x or y. Because the
 * other two don't, they start at a fractional pixel value, it introduces rounding differences between browsers and
 * between display types (retina with subpixels vs not retina). By flipping them with CSS we can ensure that
 * the path always starts at 0 so that it perfectly overlaps the popover's border.
 * See bottom of file for more explanation.
 */ let $f4e3e1a9e3fff0e3$var$arrowPlacement = {
    left: 'right',
    right: 'right',
    top: 'bottom',
    bottom: 'bottom'
};
function $f4e3e1a9e3fff0e3$var$Popover(props, ref) {
    let { children: children , placement: placement , arrowProps: arrowProps , onClose: onClose , shouldCloseOnBlur: shouldCloseOnBlur , hideArrow: hideArrow , isKeyboardDismissDisabled: isKeyboardDismissDisabled , isNonModal: isNonModal , isDismissable: isDismissable = true , ...otherProps } = props;
    let domRef = $g65uJ$reactspectrumutils.useDOMRef(ref);
    let { styleProps: styleProps  } = $g65uJ$reactspectrumutils.useStyleProps(props);
    return(/*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement($9827034998823bc2$export$c6fdb837b070b4ff, otherProps, /*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement($f4e3e1a9e3fff0e3$var$PopoverWrapper, {
        ...styleProps,
        ref: domRef,
        placement: placement,
        arrowProps: arrowProps,
        onClose: onClose,
        shouldCloseOnBlur: shouldCloseOnBlur,
        isKeyboardDismissDisabled: isKeyboardDismissDisabled,
        hideArrow: hideArrow,
        isNonModal: isNonModal,
        isDismissable: isDismissable
    }, children)));
}
const $f4e3e1a9e3fff0e3$var$PopoverWrapper = /*#__PURE__*/ $g65uJ$react.forwardRef((props, ref)=>{
    let { children: children , placement: placement = 'bottom' , arrowProps: arrowProps , isOpen: isOpen , hideArrow: hideArrow , shouldCloseOnBlur: // eslint-disable-next-line @typescript-eslint/no-unused-vars
    shouldCloseOnBlur , isKeyboardDismissDisabled: // eslint-disable-next-line @typescript-eslint/no-unused-vars
    isKeyboardDismissDisabled , isNonModal: isNonModal , isDismissable: // eslint-disable-next-line @typescript-eslint/no-unused-vars
    isDismissable , ...otherProps } = props;
    let { overlayProps: overlayProps  } = $g65uJ$reactariaoverlays.useOverlay({
        ...props,
        isDismissable: isDismissable && isOpen
    }, ref);
    let { modalProps: modalProps  } = $g65uJ$reactariaoverlays.useModal({
        isDisabled: isNonModal
    });
    return(/*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement("div", {
        ...$g65uJ$reactariautils.mergeProps(otherProps, overlayProps, modalProps),
        ref: ref,
        className: $g65uJ$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($712eff1af44443cd$exports)), 'spectrum-Popover', `spectrum-Popover--${placement}`, {
            'spectrum-Popover--withTip': !hideArrow,
            'is-open': isOpen
        }, $g65uJ$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($92f1a2282fd787c0$exports)), 'spectrum-Popover', 'react-spectrum-Popover'), otherProps.className),
        role: "presentation",
        "data-testid": "popover"
    }, children, hideArrow ? null : /*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement($f4e3e1a9e3fff0e3$var$Arrow, {
        arrowProps: arrowProps,
        direction: $f4e3e1a9e3fff0e3$var$arrowPlacement[placement]
    })));
});
let $f4e3e1a9e3fff0e3$var$ROOT_2 = Math.sqrt(2);
function $f4e3e1a9e3fff0e3$var$Arrow(props) {
    let [size, setSize] = $g65uJ$react.useState(20);
    let [borderWidth, setBorderWidth] = $g65uJ$react.useState(1);
    let ref = $g65uJ$react.useRef();
    // get the css value for the tip size and divide it by 2 for this arrow implementation
    $g65uJ$reactariautils.useLayoutEffect(()=>{
        if (ref.current) {
            let spectrumTipWidth = window.getComputedStyle(ref.current).getPropertyValue('--spectrum-popover-tip-size');
            if (spectrumTipWidth !== '') setSize(parseInt(spectrumTipWidth, 10) / 2);
            let spectrumBorderWidth = window.getComputedStyle(ref.current).getPropertyValue('--spectrum-popover-tip-borderWidth');
            if (spectrumBorderWidth !== '') setBorderWidth(parseInt(spectrumBorderWidth, 10));
        }
    }, [
        ref
    ]);
    let landscape = props.direction === 'top' || props.direction === 'bottom';
    let mirror = props.direction === 'left' || props.direction === 'top';
    let borderDiagonal = borderWidth * $f4e3e1a9e3fff0e3$var$ROOT_2;
    let halfBorderDiagonal = borderDiagonal / 2;
    let secondary = 2 * size + 2 * borderDiagonal;
    let primary = size + borderDiagonal;
    let primaryStart = mirror ? primary : 0;
    let primaryEnd = mirror ? halfBorderDiagonal : primary - halfBorderDiagonal;
    let secondaryStart = halfBorderDiagonal;
    let secondaryMiddle = secondary / 2;
    let secondaryEnd = secondary - halfBorderDiagonal;
    let pathData = landscape ? [
        'M',
        secondaryStart,
        primaryStart,
        'L',
        secondaryMiddle,
        primaryEnd,
        'L',
        secondaryEnd,
        primaryStart
    ] : [
        'M',
        primaryStart,
        secondaryStart,
        'L',
        primaryEnd,
        secondaryMiddle,
        'L',
        primaryStart,
        secondaryEnd
    ];
    let arrowProps = props.arrowProps;
    /* use ceil because the svg needs to always accomodate the path inside it */ return(/*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement("svg", {
        xmlns: "http://www.w3.org/svg/2000",
        width: Math.ceil(landscape ? secondary : primary),
        height: Math.ceil(landscape ? primary : secondary),
        style: props.style,
        className: $g65uJ$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($712eff1af44443cd$exports)), 'spectrum-Popover-tip'),
        ref: ref,
        ...arrowProps
    }, /*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement("path", {
        className: $g65uJ$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($712eff1af44443cd$exports)), 'spectrum-Popover-tip-triangle'),
        d: pathData.join(' ')
    })));
}
let $f4e3e1a9e3fff0e3$export$5b6b19405a83ff9d = /*#__PURE__*/ $g65uJ$react.forwardRef($f4e3e1a9e3fff0e3$var$Popover);


var $d399df4a2ef8c1fb$exports = {};

$parcel$export($d399df4a2ef8c1fb$exports, "Modal", () => $d399df4a2ef8c1fb$export$2b77a92f1a5ad772);


var $b2cb6d428c05324b$exports = {};

$parcel$export($b2cb6d428c05324b$exports, "spectrum-Modal", () => $b2cb6d428c05324b$export$4c0b1e44c3834c85, (v) => $b2cb6d428c05324b$export$4c0b1e44c3834c85 = v);
$parcel$export($b2cb6d428c05324b$exports, "is-open", () => $b2cb6d428c05324b$export$a9781837241c946d, (v) => $b2cb6d428c05324b$export$a9781837241c946d = v);
$parcel$export($b2cb6d428c05324b$exports, "spectrum-Modal-wrapper", () => $b2cb6d428c05324b$export$35bea187ff802494, (v) => $b2cb6d428c05324b$export$35bea187ff802494 = v);
$parcel$export($b2cb6d428c05324b$exports, "spectrum-Modal--responsive", () => $b2cb6d428c05324b$export$fd1e42aa8ea02a1, (v) => $b2cb6d428c05324b$export$fd1e42aa8ea02a1 = v);
$parcel$export($b2cb6d428c05324b$exports, "spectrum-Modal--fullscreen", () => $b2cb6d428c05324b$export$a4423fa77727431a, (v) => $b2cb6d428c05324b$export$a4423fa77727431a = v);
$parcel$export($b2cb6d428c05324b$exports, "spectrum-Modal--fullscreenTakeover", () => $b2cb6d428c05324b$export$175dd699720b5fb7, (v) => $b2cb6d428c05324b$export$175dd699720b5fb7 = v);
var $b2cb6d428c05324b$export$4c0b1e44c3834c85;
var $b2cb6d428c05324b$export$a9781837241c946d;
var $b2cb6d428c05324b$export$35bea187ff802494;
var $b2cb6d428c05324b$export$fd1e42aa8ea02a1;
var $b2cb6d428c05324b$export$a4423fa77727431a;
var $b2cb6d428c05324b$export$175dd699720b5fb7;
$b2cb6d428c05324b$export$4c0b1e44c3834c85 = "spectrum-Modal_f81956";
$b2cb6d428c05324b$export$a9781837241c946d = "is-open_f81956";
$b2cb6d428c05324b$export$35bea187ff802494 = "spectrum-Modal-wrapper_f81956";
$b2cb6d428c05324b$export$fd1e42aa8ea02a1 = "spectrum-Modal--responsive_f81956";
$b2cb6d428c05324b$export$a4423fa77727431a = "spectrum-Modal--fullscreen_f81956";
$b2cb6d428c05324b$export$175dd699720b5fb7 = "spectrum-Modal--fullscreenTakeover_f81956";







var $9bf2d76dc881d3e5$exports = {};

$parcel$export($9bf2d76dc881d3e5$exports, "spectrum-Underlay", () => $9bf2d76dc881d3e5$export$74470528f463af97, (v) => $9bf2d76dc881d3e5$export$74470528f463af97 = v);
$parcel$export($9bf2d76dc881d3e5$exports, "is-open", () => $9bf2d76dc881d3e5$export$a9781837241c946d, (v) => $9bf2d76dc881d3e5$export$a9781837241c946d = v);
var $9bf2d76dc881d3e5$export$74470528f463af97;
var $9bf2d76dc881d3e5$export$a9781837241c946d;
$9bf2d76dc881d3e5$export$74470528f463af97 = "spectrum-Underlay_eb7615";
$9bf2d76dc881d3e5$export$a9781837241c946d = "is-open_eb7615";


function $edec6ef6d7d4c7ce$export$f360afc887607b02({ isOpen: isOpen  }) {
    return(/*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement("div", {
        className: $g65uJ$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($9bf2d76dc881d3e5$exports)), 'spectrum-Underlay', {
            'is-open': isOpen
        })
    }));
}



function $d399df4a2ef8c1fb$var$Modal(props, ref) {
    let { children: children , onClose: onClose , type: type , ...otherProps } = props;
    let domRef = $g65uJ$reactspectrumutils.useDOMRef(ref);
    let { styleProps: styleProps  } = $g65uJ$reactspectrumutils.useStyleProps(props);
    let { overlayProps: overlayProps , underlayProps: underlayProps  } = $g65uJ$reactariaoverlays.useOverlay(props, domRef);
    return(/*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement($9827034998823bc2$export$c6fdb837b070b4ff, otherProps, /*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement($edec6ef6d7d4c7ce$export$f360afc887607b02, underlayProps), /*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement($d399df4a2ef8c1fb$var$ModalWrapper, {
        ...styleProps,
        onClose: onClose,
        type: type,
        ref: domRef,
        overlayProps: overlayProps
    }, children)));
}
let $d399df4a2ef8c1fb$var$typeMap = {
    fullscreen: 'fullscreen',
    fullscreenTakeover: 'fullscreenTakeover'
};
let $d399df4a2ef8c1fb$var$ModalWrapper = /*#__PURE__*/ $g65uJ$react.forwardRef(function(props, ref) {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    let { children: children , isOpen: isOpen , type: type , overlayProps: overlayProps , ...otherProps } = props;
    let typeVariant = $d399df4a2ef8c1fb$var$typeMap[type];
    $g65uJ$reactariaoverlays.usePreventScroll();
    let { modalProps: modalProps  } = $g65uJ$reactariaoverlays.useModal();
    let wrapperClassName = $g65uJ$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($b2cb6d428c05324b$exports)), 'spectrum-Modal-wrapper', $g65uJ$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($92f1a2282fd787c0$exports)), 'spectrum-Modal-wrapper', 'react-spectrum-Modal-wrapper'));
    let modalClassName = $g65uJ$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($b2cb6d428c05324b$exports)), 'spectrum-Modal', {
        'is-open': isOpen
    }, $g65uJ$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($92f1a2282fd787c0$exports)), 'spectrum-Modal', 'react-spectrum-Modal'), {
        [`spectrum-Modal--${typeVariant}`]: typeVariant
    }, otherProps.className);
    let viewport = $g65uJ$reactariautils.useViewportSize();
    let style = {
        '--spectrum-visual-viewport-height': viewport.height + 'px'
    };
    return(/*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement("div", {
        className: wrapperClassName,
        style: style
    }, /*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement("div", {
        ...$g65uJ$reactariautils.mergeProps(otherProps, overlayProps, modalProps),
        ref: ref,
        className: modalClassName,
        "data-testid": "modal"
    }, children)));
});
let $d399df4a2ef8c1fb$export$2b77a92f1a5ad772 = /*#__PURE__*/ $g65uJ$react.forwardRef($d399df4a2ef8c1fb$var$Modal);


var $ce4be73fc3209851$exports = {};

$parcel$export($ce4be73fc3209851$exports, "Tray", () => $ce4be73fc3209851$export$4589ed81930b555c);





var $14a50570408c0f12$exports = {};

$parcel$export($14a50570408c0f12$exports, "spectrum-Tray", () => $14a50570408c0f12$export$ce0704cd084c4f0d, (v) => $14a50570408c0f12$export$ce0704cd084c4f0d = v);
$parcel$export($14a50570408c0f12$exports, "is-open", () => $14a50570408c0f12$export$a9781837241c946d, (v) => $14a50570408c0f12$export$a9781837241c946d = v);
$parcel$export($14a50570408c0f12$exports, "spectrum-Tray-wrapper", () => $14a50570408c0f12$export$641b9b70c23dce66, (v) => $14a50570408c0f12$export$641b9b70c23dce66 = v);
$parcel$export($14a50570408c0f12$exports, "spectrum-Tray--fixedHeight", () => $14a50570408c0f12$export$79bfb05e59a300b, (v) => $14a50570408c0f12$export$79bfb05e59a300b = v);
var $14a50570408c0f12$export$ce0704cd084c4f0d;
var $14a50570408c0f12$export$a9781837241c946d;
var $14a50570408c0f12$export$641b9b70c23dce66;
var $14a50570408c0f12$export$79bfb05e59a300b;
$14a50570408c0f12$export$ce0704cd084c4f0d = "spectrum-Tray_23bf66";
$14a50570408c0f12$export$a9781837241c946d = "is-open_23bf66";
$14a50570408c0f12$export$641b9b70c23dce66 = "spectrum-Tray-wrapper_23bf66";
$14a50570408c0f12$export$79bfb05e59a300b = "spectrum-Tray--fixedHeight_23bf66";




function $ce4be73fc3209851$var$Tray(props, ref) {
    let { children: children , onClose: onClose , isFixedHeight: isFixedHeight , isNonModal: isNonModal , ...otherProps } = props;
    let domRef = $g65uJ$reactspectrumutils.useDOMRef(ref);
    let { styleProps: styleProps  } = $g65uJ$reactspectrumutils.useStyleProps(props);
    let { overlayProps: overlayProps , underlayProps: underlayProps  } = $g65uJ$reactariaoverlays.useOverlay({
        ...props,
        isDismissable: true
    }, domRef);
    return(/*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement($9827034998823bc2$export$c6fdb837b070b4ff, otherProps, /*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement($edec6ef6d7d4c7ce$export$f360afc887607b02, underlayProps), /*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement($ce4be73fc3209851$var$TrayWrapper, {
        ...styleProps,
        onClose: onClose,
        ref: domRef,
        overlayProps: overlayProps,
        isFixedHeight: isFixedHeight,
        isNonModal: isNonModal
    }, children)));
}
let $ce4be73fc3209851$var$TrayWrapper = /*#__PURE__*/ $g65uJ$react.forwardRef(function(props, ref) {
    let { children: children , isOpen: isOpen , isFixedHeight: isFixedHeight , isNonModal: isNonModal , overlayProps: overlayProps , ...otherProps } = props;
    $g65uJ$reactariaoverlays.usePreventScroll();
    let { modalProps: modalProps  } = $g65uJ$reactariaoverlays.useModal({
        isDisabled: isNonModal
    });
    // We need to measure the window's height in JS rather than using percentages in CSS
    // so that contents (e.g. menu) can inherit the max-height properly. Using percentages
    // does not work properly because there is nothing to base the percentage on.
    // We cannot use vh units because mobile browsers adjust the window height dynamically
    // when the address bar/bottom toolbars show and hide on scroll and vh units are fixed.
    // Also, the visual viewport is smaller than the layout viewport when the virtual keyboard
    // is up, so use the VisualViewport API to ensure the tray is displayed above the keyboard.
    let viewport = $g65uJ$reactariautils.useViewportSize();
    let [height, setHeight] = $g65uJ$react.useState(viewport.height);
    let timeoutRef = $g65uJ$react.useRef();
    $g65uJ$react.useEffect(()=>{
        clearTimeout(timeoutRef.current);
        // When the height is decreasing, and the keyboard is visible
        // (visual viewport smaller than layout viewport), delay setting
        // the new max height until after the animation is complete
        // so that there isn't an empty space under the tray briefly.
        if (viewport.height < height && viewport.height < window.innerHeight) timeoutRef.current = setTimeout(()=>{
            setHeight(viewport.height);
        }, 500);
        else setHeight(viewport.height);
    }, [
        height,
        viewport.height
    ]);
    let wrapperStyle = {
        '--spectrum-visual-viewport-height': height + 'px'
    };
    let wrapperClassName = $g65uJ$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($14a50570408c0f12$exports)), 'spectrum-Tray-wrapper');
    let className = $g65uJ$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($14a50570408c0f12$exports)), 'spectrum-Tray', {
        'is-open': isOpen,
        'spectrum-Tray--fixedHeight': isFixedHeight
    }, $g65uJ$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($92f1a2282fd787c0$exports)), 'spectrum-Tray', 'react-spectrum-Tray'), otherProps.className);
    let domProps = $g65uJ$reactariautils.mergeProps(otherProps, overlayProps);
    return(/*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement("div", {
        className: wrapperClassName,
        style: wrapperStyle
    }, /*#__PURE__*/ ($parcel$interopDefault($g65uJ$react)).createElement("div", {
        ...domProps,
        ...modalProps,
        className: className,
        ref: ref,
        "data-testid": "tray"
    }, children)));
});
let $ce4be73fc3209851$export$4589ed81930b555c = /*#__PURE__*/ $g65uJ$react.forwardRef($ce4be73fc3209851$var$Tray);



$parcel$exportWildcard(module.exports, $9827034998823bc2$exports);
$parcel$exportWildcard(module.exports, $f4e3e1a9e3fff0e3$exports);
$parcel$exportWildcard(module.exports, $d399df4a2ef8c1fb$exports);
$parcel$exportWildcard(module.exports, $ce4be73fc3209851$exports);
$parcel$exportWildcard(module.exports, $ac0466425dbf7f76$exports);
