// @react-spectrum/overlays/dist/module.js
import "./main.css";
import {Provider as $4Qwqh$Provider} from "@react-spectrum/provider";
import $4Qwqh$react, {useState as $4Qwqh$useState, useCallback as $4Qwqh$useCallback, forwardRef as $4Qwqh$forwardRef, useRef as $4Qwqh$useRef, useEffect as $4Qwqh$useEffect} from "react";
import $4Qwqh$reactdom from "react-dom";
import $4Qwqh$reacttransitiongroupTransition from "react-transition-group/Transition";
import {useDOMRef as $4Qwqh$useDOMRef, useStyleProps as $4Qwqh$useStyleProps, classNames as $4Qwqh$classNames} from "@react-spectrum/utils";
import {mergeProps as $4Qwqh$mergeProps, useLayoutEffect as $4Qwqh$useLayoutEffect, useViewportSize as $4Qwqh$useViewportSize} from "@react-aria/utils";
import {useOverlay as $4Qwqh$useOverlay, useModal as $4Qwqh$useModal, usePreventScroll as $4Qwqh$usePreventScroll} from "@react-aria/overlays";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
var $80816efa7007e5bf$exports = {};

$parcel$export($80816efa7007e5bf$exports, "Overlay", () => $80816efa7007e5bf$export$c6fdb837b070b4ff);
var $647de0c873a185d0$exports = {};

$parcel$export($647de0c873a185d0$exports, "OpenTransition", () => $647de0c873a185d0$export$b847a40ee92eff38);


const $647de0c873a185d0$var$OPEN_STATES = {
    entering: false,
    entered: true
};
function $647de0c873a185d0$export$b847a40ee92eff38(props) {
    return(/*#__PURE__*/ $4Qwqh$react.createElement($4Qwqh$reacttransitiongroupTransition, {
        timeout: {
            enter: 0,
            exit: 350
        },
        ...props
    }, (state)=>$4Qwqh$react.Children.map(props.children, (child)=>child && /*#__PURE__*/ $4Qwqh$react.cloneElement(child, {
                isOpen: !!$647de0c873a185d0$var$OPEN_STATES[state]
            })
        )
    ));
}





function $80816efa7007e5bf$var$Overlay(props, ref) {
    let { children: children , isOpen: isOpen , container: container , onEnter: onEnter , onEntering: onEntering , onEntered: onEntered , onExit: onExit , onExiting: onExiting , onExited: onExited  } = props;
    let [exited, setExited] = $4Qwqh$useState(!isOpen);
    let handleEntered = $4Qwqh$useCallback(()=>{
        setExited(false);
        if (onEntered) onEntered();
    }, [
        onEntered
    ]);
    let handleExited = $4Qwqh$useCallback(()=>{
        setExited(true);
        if (onExited) onExited();
    }, [
        onExited
    ]);
    // Don't un-render the overlay while it's transitioning out.
    let mountOverlay = isOpen || !exited;
    if (!mountOverlay) // Don't bother showing anything if we don't have to.
    return null;
    let contents = /*#__PURE__*/ $4Qwqh$react.createElement($4Qwqh$Provider, {
        ref: ref,
        UNSAFE_style: {
            background: 'transparent',
            isolation: 'isolate'
        }
    }, /*#__PURE__*/ $4Qwqh$react.createElement($647de0c873a185d0$export$b847a40ee92eff38, {
        in: isOpen,
        appear: true,
        onExit: onExit,
        onExiting: onExiting,
        onExited: handleExited,
        onEnter: onEnter,
        onEntering: onEntering,
        onEntered: handleEntered
    }, children));
    return(/*#__PURE__*/ $4Qwqh$reactdom.createPortal(contents, container || document.body));
}
let $80816efa7007e5bf$export$c6fdb837b070b4ff = /*#__PURE__*/ $4Qwqh$react.forwardRef($80816efa7007e5bf$var$Overlay);


var $5b3c29c81f767e0c$exports = {};

$parcel$export($5b3c29c81f767e0c$exports, "Popover", () => $5b3c29c81f767e0c$export$5b6b19405a83ff9d);



var $9ec42a7475cdf6b0$exports = {};

$parcel$export($9ec42a7475cdf6b0$exports, "spectrum-Popover", () => $9ec42a7475cdf6b0$export$a62dc1b6ab23a7bb, (v) => $9ec42a7475cdf6b0$export$a62dc1b6ab23a7bb = v);
$parcel$export($9ec42a7475cdf6b0$exports, "react-spectrum-Popover", () => $9ec42a7475cdf6b0$export$884f6b5065bdc41a, (v) => $9ec42a7475cdf6b0$export$884f6b5065bdc41a = v);
$parcel$export($9ec42a7475cdf6b0$exports, "spectrum-Dialog-content", () => $9ec42a7475cdf6b0$export$cb8eccd0f3639238, (v) => $9ec42a7475cdf6b0$export$cb8eccd0f3639238 = v);
$parcel$export($9ec42a7475cdf6b0$exports, "spectrum-Modal-wrapper", () => $9ec42a7475cdf6b0$export$35bea187ff802494, (v) => $9ec42a7475cdf6b0$export$35bea187ff802494 = v);
$parcel$export($9ec42a7475cdf6b0$exports, "react-spectrum-Modal-wrapper", () => $9ec42a7475cdf6b0$export$e088717c213c31d7, (v) => $9ec42a7475cdf6b0$export$e088717c213c31d7 = v);
$parcel$export($9ec42a7475cdf6b0$exports, "spectrum-Modal", () => $9ec42a7475cdf6b0$export$4c0b1e44c3834c85, (v) => $9ec42a7475cdf6b0$export$4c0b1e44c3834c85 = v);
$parcel$export($9ec42a7475cdf6b0$exports, "react-spectrum-Modal", () => $9ec42a7475cdf6b0$export$f8edeb62d7fed8c1, (v) => $9ec42a7475cdf6b0$export$f8edeb62d7fed8c1 = v);
$parcel$export($9ec42a7475cdf6b0$exports, "spectrum-Tray", () => $9ec42a7475cdf6b0$export$ce0704cd084c4f0d, (v) => $9ec42a7475cdf6b0$export$ce0704cd084c4f0d = v);
$parcel$export($9ec42a7475cdf6b0$exports, "react-spectrum-Tray", () => $9ec42a7475cdf6b0$export$69dafa24343974dd, (v) => $9ec42a7475cdf6b0$export$69dafa24343974dd = v);
var $9ec42a7475cdf6b0$export$a62dc1b6ab23a7bb;
var $9ec42a7475cdf6b0$export$884f6b5065bdc41a;
var $9ec42a7475cdf6b0$export$cb8eccd0f3639238;
var $9ec42a7475cdf6b0$export$35bea187ff802494;
var $9ec42a7475cdf6b0$export$e088717c213c31d7;
var $9ec42a7475cdf6b0$export$4c0b1e44c3834c85;
var $9ec42a7475cdf6b0$export$f8edeb62d7fed8c1;
var $9ec42a7475cdf6b0$export$ce0704cd084c4f0d;
var $9ec42a7475cdf6b0$export$69dafa24343974dd;
$9ec42a7475cdf6b0$export$a62dc1b6ab23a7bb = "spectrum-Popover_38efea";
$9ec42a7475cdf6b0$export$884f6b5065bdc41a = "react-spectrum-Popover_38efea";
$9ec42a7475cdf6b0$export$cb8eccd0f3639238 = "spectrum-Dialog-content_38efea";
$9ec42a7475cdf6b0$export$35bea187ff802494 = "spectrum-Modal-wrapper_38efea";
$9ec42a7475cdf6b0$export$e088717c213c31d7 = "react-spectrum-Modal-wrapper_38efea";
$9ec42a7475cdf6b0$export$4c0b1e44c3834c85 = "spectrum-Modal_38efea";
$9ec42a7475cdf6b0$export$f8edeb62d7fed8c1 = "react-spectrum-Modal_38efea";
$9ec42a7475cdf6b0$export$ce0704cd084c4f0d = "spectrum-Tray_38efea";
$9ec42a7475cdf6b0$export$69dafa24343974dd = "react-spectrum-Tray_38efea";



var $9126ba5fe017d182$exports = {};

$parcel$export($9126ba5fe017d182$exports, "spectrum-Popover", () => $9126ba5fe017d182$export$a62dc1b6ab23a7bb, (v) => $9126ba5fe017d182$export$a62dc1b6ab23a7bb = v);
$parcel$export($9126ba5fe017d182$exports, "is-open", () => $9126ba5fe017d182$export$a9781837241c946d, (v) => $9126ba5fe017d182$export$a9781837241c946d = v);
$parcel$export($9126ba5fe017d182$exports, "spectrum-Popover--bottom", () => $9126ba5fe017d182$export$6a19b19b14780d4a, (v) => $9126ba5fe017d182$export$6a19b19b14780d4a = v);
$parcel$export($9126ba5fe017d182$exports, "spectrum-Popover--top", () => $9126ba5fe017d182$export$f30becfb1df0cae4, (v) => $9126ba5fe017d182$export$f30becfb1df0cae4 = v);
$parcel$export($9126ba5fe017d182$exports, "spectrum-Popover--right", () => $9126ba5fe017d182$export$5fc8b429396c9f58, (v) => $9126ba5fe017d182$export$5fc8b429396c9f58 = v);
$parcel$export($9126ba5fe017d182$exports, "spectrum-Popover--left", () => $9126ba5fe017d182$export$79cc55906ac5b00, (v) => $9126ba5fe017d182$export$79cc55906ac5b00 = v);
$parcel$export($9126ba5fe017d182$exports, "spectrum-Popover-tip", () => $9126ba5fe017d182$export$3100ec4dc1668df1, (v) => $9126ba5fe017d182$export$3100ec4dc1668df1 = v);
$parcel$export($9126ba5fe017d182$exports, "spectrum-Popover-tip-triangle", () => $9126ba5fe017d182$export$ed733fa6cf19ab80, (v) => $9126ba5fe017d182$export$ed733fa6cf19ab80 = v);
$parcel$export($9126ba5fe017d182$exports, "spectrum-Popover--dialog", () => $9126ba5fe017d182$export$cf634262e726cd19, (v) => $9126ba5fe017d182$export$cf634262e726cd19 = v);
$parcel$export($9126ba5fe017d182$exports, "spectrum-Popover--withTip", () => $9126ba5fe017d182$export$d8b16c5377728262, (v) => $9126ba5fe017d182$export$d8b16c5377728262 = v);
var $9126ba5fe017d182$export$a62dc1b6ab23a7bb;
var $9126ba5fe017d182$export$a9781837241c946d;
var $9126ba5fe017d182$export$6a19b19b14780d4a;
var $9126ba5fe017d182$export$f30becfb1df0cae4;
var $9126ba5fe017d182$export$5fc8b429396c9f58;
var $9126ba5fe017d182$export$79cc55906ac5b00;
var $9126ba5fe017d182$export$3100ec4dc1668df1;
var $9126ba5fe017d182$export$ed733fa6cf19ab80;
var $9126ba5fe017d182$export$cf634262e726cd19;
var $9126ba5fe017d182$export$d8b16c5377728262;
$9126ba5fe017d182$export$a62dc1b6ab23a7bb = "spectrum-Popover_6115b8";
$9126ba5fe017d182$export$a9781837241c946d = "is-open_6115b8";
$9126ba5fe017d182$export$6a19b19b14780d4a = "spectrum-Popover--bottom_6115b8";
$9126ba5fe017d182$export$f30becfb1df0cae4 = "spectrum-Popover--top_6115b8";
$9126ba5fe017d182$export$5fc8b429396c9f58 = "spectrum-Popover--right_6115b8";
$9126ba5fe017d182$export$79cc55906ac5b00 = "spectrum-Popover--left_6115b8";
$9126ba5fe017d182$export$3100ec4dc1668df1 = "spectrum-Popover-tip_6115b8";
$9126ba5fe017d182$export$ed733fa6cf19ab80 = "spectrum-Popover-tip-triangle_6115b8";
$9126ba5fe017d182$export$cf634262e726cd19 = "spectrum-Popover--dialog_6115b8";
$9126ba5fe017d182$export$d8b16c5377728262 = "spectrum-Popover--withTip_6115b8";



/**
 * Arrow placement can be done pointing right or down because those paths start at 0, x or y. Because the
 * other two don't, they start at a fractional pixel value, it introduces rounding differences between browsers and
 * between display types (retina with subpixels vs not retina). By flipping them with CSS we can ensure that
 * the path always starts at 0 so that it perfectly overlaps the popover's border.
 * See bottom of file for more explanation.
 */ let $5b3c29c81f767e0c$var$arrowPlacement = {
    left: 'right',
    right: 'right',
    top: 'bottom',
    bottom: 'bottom'
};
function $5b3c29c81f767e0c$var$Popover(props, ref) {
    let { children: children , placement: placement , arrowProps: arrowProps , onClose: onClose , shouldCloseOnBlur: shouldCloseOnBlur , hideArrow: hideArrow , isKeyboardDismissDisabled: isKeyboardDismissDisabled , isNonModal: isNonModal , isDismissable: isDismissable = true , ...otherProps } = props;
    let domRef = $4Qwqh$useDOMRef(ref);
    let { styleProps: styleProps  } = $4Qwqh$useStyleProps(props);
    return(/*#__PURE__*/ $4Qwqh$react.createElement($80816efa7007e5bf$export$c6fdb837b070b4ff, otherProps, /*#__PURE__*/ $4Qwqh$react.createElement($5b3c29c81f767e0c$var$PopoverWrapper, {
        ...styleProps,
        ref: domRef,
        placement: placement,
        arrowProps: arrowProps,
        onClose: onClose,
        shouldCloseOnBlur: shouldCloseOnBlur,
        isKeyboardDismissDisabled: isKeyboardDismissDisabled,
        hideArrow: hideArrow,
        isNonModal: isNonModal,
        isDismissable: isDismissable
    }, children)));
}
const $5b3c29c81f767e0c$var$PopoverWrapper = /*#__PURE__*/ $4Qwqh$forwardRef((props, ref)=>{
    let { children: children , placement: placement = 'bottom' , arrowProps: arrowProps , isOpen: isOpen , hideArrow: hideArrow , shouldCloseOnBlur: // eslint-disable-next-line @typescript-eslint/no-unused-vars
    shouldCloseOnBlur , isKeyboardDismissDisabled: // eslint-disable-next-line @typescript-eslint/no-unused-vars
    isKeyboardDismissDisabled , isNonModal: isNonModal , isDismissable: // eslint-disable-next-line @typescript-eslint/no-unused-vars
    isDismissable , ...otherProps } = props;
    let { overlayProps: overlayProps  } = $4Qwqh$useOverlay({
        ...props,
        isDismissable: isDismissable && isOpen
    }, ref);
    let { modalProps: modalProps  } = $4Qwqh$useModal({
        isDisabled: isNonModal
    });
    return(/*#__PURE__*/ $4Qwqh$react.createElement("div", {
        ...$4Qwqh$mergeProps(otherProps, overlayProps, modalProps),
        ref: ref,
        className: $4Qwqh$classNames((/*@__PURE__*/$parcel$interopDefault($9126ba5fe017d182$exports)), 'spectrum-Popover', `spectrum-Popover--${placement}`, {
            'spectrum-Popover--withTip': !hideArrow,
            'is-open': isOpen
        }, $4Qwqh$classNames((/*@__PURE__*/$parcel$interopDefault($9ec42a7475cdf6b0$exports)), 'spectrum-Popover', 'react-spectrum-Popover'), otherProps.className),
        role: "presentation",
        "data-testid": "popover"
    }, children, hideArrow ? null : /*#__PURE__*/ $4Qwqh$react.createElement($5b3c29c81f767e0c$var$Arrow, {
        arrowProps: arrowProps,
        direction: $5b3c29c81f767e0c$var$arrowPlacement[placement]
    })));
});
let $5b3c29c81f767e0c$var$ROOT_2 = Math.sqrt(2);
function $5b3c29c81f767e0c$var$Arrow(props) {
    let [size, setSize] = $4Qwqh$useState(20);
    let [borderWidth, setBorderWidth] = $4Qwqh$useState(1);
    let ref = $4Qwqh$useRef();
    // get the css value for the tip size and divide it by 2 for this arrow implementation
    $4Qwqh$useLayoutEffect(()=>{
        if (ref.current) {
            let spectrumTipWidth = window.getComputedStyle(ref.current).getPropertyValue('--spectrum-popover-tip-size');
            if (spectrumTipWidth !== '') setSize(parseInt(spectrumTipWidth, 10) / 2);
            let spectrumBorderWidth = window.getComputedStyle(ref.current).getPropertyValue('--spectrum-popover-tip-borderWidth');
            if (spectrumBorderWidth !== '') setBorderWidth(parseInt(spectrumBorderWidth, 10));
        }
    }, [
        ref
    ]);
    let landscape = props.direction === 'top' || props.direction === 'bottom';
    let mirror = props.direction === 'left' || props.direction === 'top';
    let borderDiagonal = borderWidth * $5b3c29c81f767e0c$var$ROOT_2;
    let halfBorderDiagonal = borderDiagonal / 2;
    let secondary = 2 * size + 2 * borderDiagonal;
    let primary = size + borderDiagonal;
    let primaryStart = mirror ? primary : 0;
    let primaryEnd = mirror ? halfBorderDiagonal : primary - halfBorderDiagonal;
    let secondaryStart = halfBorderDiagonal;
    let secondaryMiddle = secondary / 2;
    let secondaryEnd = secondary - halfBorderDiagonal;
    let pathData = landscape ? [
        'M',
        secondaryStart,
        primaryStart,
        'L',
        secondaryMiddle,
        primaryEnd,
        'L',
        secondaryEnd,
        primaryStart
    ] : [
        'M',
        primaryStart,
        secondaryStart,
        'L',
        primaryEnd,
        secondaryMiddle,
        'L',
        primaryStart,
        secondaryEnd
    ];
    let arrowProps = props.arrowProps;
    /* use ceil because the svg needs to always accomodate the path inside it */ return(/*#__PURE__*/ $4Qwqh$react.createElement("svg", {
        xmlns: "http://www.w3.org/svg/2000",
        width: Math.ceil(landscape ? secondary : primary),
        height: Math.ceil(landscape ? primary : secondary),
        style: props.style,
        className: $4Qwqh$classNames((/*@__PURE__*/$parcel$interopDefault($9126ba5fe017d182$exports)), 'spectrum-Popover-tip'),
        ref: ref,
        ...arrowProps
    }, /*#__PURE__*/ $4Qwqh$react.createElement("path", {
        className: $4Qwqh$classNames((/*@__PURE__*/$parcel$interopDefault($9126ba5fe017d182$exports)), 'spectrum-Popover-tip-triangle'),
        d: pathData.join(' ')
    })));
}
let $5b3c29c81f767e0c$export$5b6b19405a83ff9d = /*#__PURE__*/ $4Qwqh$forwardRef($5b3c29c81f767e0c$var$Popover);


var $c6a2ea12be2a8505$exports = {};

$parcel$export($c6a2ea12be2a8505$exports, "Modal", () => $c6a2ea12be2a8505$export$2b77a92f1a5ad772);


var $b65929eb65b6e471$exports = {};

$parcel$export($b65929eb65b6e471$exports, "spectrum-Modal", () => $b65929eb65b6e471$export$4c0b1e44c3834c85, (v) => $b65929eb65b6e471$export$4c0b1e44c3834c85 = v);
$parcel$export($b65929eb65b6e471$exports, "is-open", () => $b65929eb65b6e471$export$a9781837241c946d, (v) => $b65929eb65b6e471$export$a9781837241c946d = v);
$parcel$export($b65929eb65b6e471$exports, "spectrum-Modal-wrapper", () => $b65929eb65b6e471$export$35bea187ff802494, (v) => $b65929eb65b6e471$export$35bea187ff802494 = v);
$parcel$export($b65929eb65b6e471$exports, "spectrum-Modal--responsive", () => $b65929eb65b6e471$export$fd1e42aa8ea02a1, (v) => $b65929eb65b6e471$export$fd1e42aa8ea02a1 = v);
$parcel$export($b65929eb65b6e471$exports, "spectrum-Modal--fullscreen", () => $b65929eb65b6e471$export$a4423fa77727431a, (v) => $b65929eb65b6e471$export$a4423fa77727431a = v);
$parcel$export($b65929eb65b6e471$exports, "spectrum-Modal--fullscreenTakeover", () => $b65929eb65b6e471$export$175dd699720b5fb7, (v) => $b65929eb65b6e471$export$175dd699720b5fb7 = v);
var $b65929eb65b6e471$export$4c0b1e44c3834c85;
var $b65929eb65b6e471$export$a9781837241c946d;
var $b65929eb65b6e471$export$35bea187ff802494;
var $b65929eb65b6e471$export$fd1e42aa8ea02a1;
var $b65929eb65b6e471$export$a4423fa77727431a;
var $b65929eb65b6e471$export$175dd699720b5fb7;
$b65929eb65b6e471$export$4c0b1e44c3834c85 = "spectrum-Modal_f81956";
$b65929eb65b6e471$export$a9781837241c946d = "is-open_f81956";
$b65929eb65b6e471$export$35bea187ff802494 = "spectrum-Modal-wrapper_f81956";
$b65929eb65b6e471$export$fd1e42aa8ea02a1 = "spectrum-Modal--responsive_f81956";
$b65929eb65b6e471$export$a4423fa77727431a = "spectrum-Modal--fullscreen_f81956";
$b65929eb65b6e471$export$175dd699720b5fb7 = "spectrum-Modal--fullscreenTakeover_f81956";







var $b4c2aa75f00572f6$exports = {};

$parcel$export($b4c2aa75f00572f6$exports, "spectrum-Underlay", () => $b4c2aa75f00572f6$export$74470528f463af97, (v) => $b4c2aa75f00572f6$export$74470528f463af97 = v);
$parcel$export($b4c2aa75f00572f6$exports, "is-open", () => $b4c2aa75f00572f6$export$a9781837241c946d, (v) => $b4c2aa75f00572f6$export$a9781837241c946d = v);
var $b4c2aa75f00572f6$export$74470528f463af97;
var $b4c2aa75f00572f6$export$a9781837241c946d;
$b4c2aa75f00572f6$export$74470528f463af97 = "spectrum-Underlay_eb7615";
$b4c2aa75f00572f6$export$a9781837241c946d = "is-open_eb7615";


function $6856e8fa42ff7496$export$f360afc887607b02({ isOpen: isOpen  }) {
    return(/*#__PURE__*/ $4Qwqh$react.createElement("div", {
        className: $4Qwqh$classNames((/*@__PURE__*/$parcel$interopDefault($b4c2aa75f00572f6$exports)), 'spectrum-Underlay', {
            'is-open': isOpen
        })
    }));
}



function $c6a2ea12be2a8505$var$Modal(props, ref) {
    let { children: children , onClose: onClose , type: type , ...otherProps } = props;
    let domRef = $4Qwqh$useDOMRef(ref);
    let { styleProps: styleProps  } = $4Qwqh$useStyleProps(props);
    let { overlayProps: overlayProps , underlayProps: underlayProps  } = $4Qwqh$useOverlay(props, domRef);
    return(/*#__PURE__*/ $4Qwqh$react.createElement($80816efa7007e5bf$export$c6fdb837b070b4ff, otherProps, /*#__PURE__*/ $4Qwqh$react.createElement($6856e8fa42ff7496$export$f360afc887607b02, underlayProps), /*#__PURE__*/ $4Qwqh$react.createElement($c6a2ea12be2a8505$var$ModalWrapper, {
        ...styleProps,
        onClose: onClose,
        type: type,
        ref: domRef,
        overlayProps: overlayProps
    }, children)));
}
let $c6a2ea12be2a8505$var$typeMap = {
    fullscreen: 'fullscreen',
    fullscreenTakeover: 'fullscreenTakeover'
};
let $c6a2ea12be2a8505$var$ModalWrapper = /*#__PURE__*/ $4Qwqh$forwardRef(function(props, ref) {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    let { children: children , isOpen: isOpen , type: type , overlayProps: overlayProps , ...otherProps } = props;
    let typeVariant = $c6a2ea12be2a8505$var$typeMap[type];
    $4Qwqh$usePreventScroll();
    let { modalProps: modalProps  } = $4Qwqh$useModal();
    let wrapperClassName = $4Qwqh$classNames((/*@__PURE__*/$parcel$interopDefault($b65929eb65b6e471$exports)), 'spectrum-Modal-wrapper', $4Qwqh$classNames((/*@__PURE__*/$parcel$interopDefault($9ec42a7475cdf6b0$exports)), 'spectrum-Modal-wrapper', 'react-spectrum-Modal-wrapper'));
    let modalClassName = $4Qwqh$classNames((/*@__PURE__*/$parcel$interopDefault($b65929eb65b6e471$exports)), 'spectrum-Modal', {
        'is-open': isOpen
    }, $4Qwqh$classNames((/*@__PURE__*/$parcel$interopDefault($9ec42a7475cdf6b0$exports)), 'spectrum-Modal', 'react-spectrum-Modal'), {
        [`spectrum-Modal--${typeVariant}`]: typeVariant
    }, otherProps.className);
    let viewport = $4Qwqh$useViewportSize();
    let style = {
        '--spectrum-visual-viewport-height': viewport.height + 'px'
    };
    return(/*#__PURE__*/ $4Qwqh$react.createElement("div", {
        className: wrapperClassName,
        style: style
    }, /*#__PURE__*/ $4Qwqh$react.createElement("div", {
        ...$4Qwqh$mergeProps(otherProps, overlayProps, modalProps),
        ref: ref,
        className: modalClassName,
        "data-testid": "modal"
    }, children)));
});
let $c6a2ea12be2a8505$export$2b77a92f1a5ad772 = /*#__PURE__*/ $4Qwqh$forwardRef($c6a2ea12be2a8505$var$Modal);


var $115ab8a5c0552804$exports = {};

$parcel$export($115ab8a5c0552804$exports, "Tray", () => $115ab8a5c0552804$export$4589ed81930b555c);





var $02af7fbcfbd0696e$exports = {};

$parcel$export($02af7fbcfbd0696e$exports, "spectrum-Tray", () => $02af7fbcfbd0696e$export$ce0704cd084c4f0d, (v) => $02af7fbcfbd0696e$export$ce0704cd084c4f0d = v);
$parcel$export($02af7fbcfbd0696e$exports, "is-open", () => $02af7fbcfbd0696e$export$a9781837241c946d, (v) => $02af7fbcfbd0696e$export$a9781837241c946d = v);
$parcel$export($02af7fbcfbd0696e$exports, "spectrum-Tray-wrapper", () => $02af7fbcfbd0696e$export$641b9b70c23dce66, (v) => $02af7fbcfbd0696e$export$641b9b70c23dce66 = v);
$parcel$export($02af7fbcfbd0696e$exports, "spectrum-Tray--fixedHeight", () => $02af7fbcfbd0696e$export$79bfb05e59a300b, (v) => $02af7fbcfbd0696e$export$79bfb05e59a300b = v);
var $02af7fbcfbd0696e$export$ce0704cd084c4f0d;
var $02af7fbcfbd0696e$export$a9781837241c946d;
var $02af7fbcfbd0696e$export$641b9b70c23dce66;
var $02af7fbcfbd0696e$export$79bfb05e59a300b;
$02af7fbcfbd0696e$export$ce0704cd084c4f0d = "spectrum-Tray_23bf66";
$02af7fbcfbd0696e$export$a9781837241c946d = "is-open_23bf66";
$02af7fbcfbd0696e$export$641b9b70c23dce66 = "spectrum-Tray-wrapper_23bf66";
$02af7fbcfbd0696e$export$79bfb05e59a300b = "spectrum-Tray--fixedHeight_23bf66";




function $115ab8a5c0552804$var$Tray(props, ref) {
    let { children: children , onClose: onClose , isFixedHeight: isFixedHeight , isNonModal: isNonModal , ...otherProps } = props;
    let domRef = $4Qwqh$useDOMRef(ref);
    let { styleProps: styleProps  } = $4Qwqh$useStyleProps(props);
    let { overlayProps: overlayProps , underlayProps: underlayProps  } = $4Qwqh$useOverlay({
        ...props,
        isDismissable: true
    }, domRef);
    return(/*#__PURE__*/ $4Qwqh$react.createElement($80816efa7007e5bf$export$c6fdb837b070b4ff, otherProps, /*#__PURE__*/ $4Qwqh$react.createElement($6856e8fa42ff7496$export$f360afc887607b02, underlayProps), /*#__PURE__*/ $4Qwqh$react.createElement($115ab8a5c0552804$var$TrayWrapper, {
        ...styleProps,
        onClose: onClose,
        ref: domRef,
        overlayProps: overlayProps,
        isFixedHeight: isFixedHeight,
        isNonModal: isNonModal
    }, children)));
}
let $115ab8a5c0552804$var$TrayWrapper = /*#__PURE__*/ $4Qwqh$forwardRef(function(props, ref) {
    let { children: children , isOpen: isOpen , isFixedHeight: isFixedHeight , isNonModal: isNonModal , overlayProps: overlayProps , ...otherProps } = props;
    $4Qwqh$usePreventScroll();
    let { modalProps: modalProps  } = $4Qwqh$useModal({
        isDisabled: isNonModal
    });
    // We need to measure the window's height in JS rather than using percentages in CSS
    // so that contents (e.g. menu) can inherit the max-height properly. Using percentages
    // does not work properly because there is nothing to base the percentage on.
    // We cannot use vh units because mobile browsers adjust the window height dynamically
    // when the address bar/bottom toolbars show and hide on scroll and vh units are fixed.
    // Also, the visual viewport is smaller than the layout viewport when the virtual keyboard
    // is up, so use the VisualViewport API to ensure the tray is displayed above the keyboard.
    let viewport = $4Qwqh$useViewportSize();
    let [height, setHeight] = $4Qwqh$useState(viewport.height);
    let timeoutRef = $4Qwqh$useRef();
    $4Qwqh$useEffect(()=>{
        clearTimeout(timeoutRef.current);
        // When the height is decreasing, and the keyboard is visible
        // (visual viewport smaller than layout viewport), delay setting
        // the new max height until after the animation is complete
        // so that there isn't an empty space under the tray briefly.
        if (viewport.height < height && viewport.height < window.innerHeight) timeoutRef.current = setTimeout(()=>{
            setHeight(viewport.height);
        }, 500);
        else setHeight(viewport.height);
    }, [
        height,
        viewport.height
    ]);
    let wrapperStyle = {
        '--spectrum-visual-viewport-height': height + 'px'
    };
    let wrapperClassName = $4Qwqh$classNames((/*@__PURE__*/$parcel$interopDefault($02af7fbcfbd0696e$exports)), 'spectrum-Tray-wrapper');
    let className = $4Qwqh$classNames((/*@__PURE__*/$parcel$interopDefault($02af7fbcfbd0696e$exports)), 'spectrum-Tray', {
        'is-open': isOpen,
        'spectrum-Tray--fixedHeight': isFixedHeight
    }, $4Qwqh$classNames((/*@__PURE__*/$parcel$interopDefault($9ec42a7475cdf6b0$exports)), 'spectrum-Tray', 'react-spectrum-Tray'), otherProps.className);
    let domProps = $4Qwqh$mergeProps(otherProps, overlayProps);
    return(/*#__PURE__*/ $4Qwqh$react.createElement("div", {
        className: wrapperClassName,
        style: wrapperStyle
    }, /*#__PURE__*/ $4Qwqh$react.createElement("div", {
        ...domProps,
        ...modalProps,
        className: className,
        ref: ref,
        "data-testid": "tray"
    }, children)));
});
let $115ab8a5c0552804$export$4589ed81930b555c = /*#__PURE__*/ $4Qwqh$forwardRef($115ab8a5c0552804$var$Tray);





export {$80816efa7007e5bf$export$c6fdb837b070b4ff as Overlay, $5b3c29c81f767e0c$export$5b6b19405a83ff9d as Popover, $c6a2ea12be2a8505$export$2b77a92f1a5ad772 as Modal, $115ab8a5c0552804$export$4589ed81930b555c as Tray, $647de0c873a185d0$export$b847a40ee92eff38 as OpenTransition};
