// @react-spectrum/combobox/dist/main.js
require("./main.css");
var $dyK7H$reactstatelycollections = require("@react-stately/collections");
var $dyK7H$spectrumiconsuiChevronDownMedium = require("@spectrum-icons/ui/ChevronDownMedium");
var $dyK7H$reactspectrumutils = require("@react-spectrum/utils");
var $dyK7H$reactariaoverlays = require("@react-aria/overlays");
var $dyK7H$reactspectrumlabel = require("@react-spectrum/label");
var $dyK7H$reactspectrumbutton = require("@react-spectrum/button");
var $dyK7H$reactariafocus = require("@react-aria/focus");
var $dyK7H$reactspectrumlistbox = require("@react-spectrum/listbox");
var $dyK7H$reactspectrumoverlays = require("@react-spectrum/overlays");
var $dyK7H$reactariainteractions = require("@react-aria/interactions");
var $dyK7H$reactspectrumprogress = require("@react-spectrum/progress");
var $dyK7H$react = require("react");
var $dyK7H$reactspectrumtextfield = require("@react-spectrum/textfield");
var $dyK7H$reactariacombobox = require("@react-aria/combobox");
var $dyK7H$reactstatelycombobox = require("@react-stately/combobox");
var $dyK7H$reactariai18n = require("@react-aria/i18n");
var $dyK7H$reactariautils = require("@react-aria/utils");
var $dyK7H$reactspectrumprovider = require("@react-spectrum/provider");
var $dyK7H$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $dyK7H$spectrumiconsuiCheckmarkMedium = require("@spectrum-icons/ui/CheckmarkMedium");
var $dyK7H$reactariabutton = require("@react-aria/button");
var $dyK7H$reactariadialog = require("@react-aria/dialog");
var $dyK7H$reactarialabel = require("@react-aria/label");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Item", () => $dyK7H$reactstatelycollections.Item);
$parcel$export(module.exports, "Section", () => $dyK7H$reactstatelycollections.Section);
var $3740f8e8ff9aff46$exports = {};

$parcel$export($3740f8e8ff9aff46$exports, "ComboBox", () => $3740f8e8ff9aff46$export$72b9695b8216309a);


var $ddda48ae832cd17d$exports = {};

$parcel$export($ddda48ae832cd17d$exports, "no-results", () => $ddda48ae832cd17d$export$5b1939da91820a47, (v) => $ddda48ae832cd17d$export$5b1939da91820a47 = v);
$parcel$export($ddda48ae832cd17d$exports, "mobile-combobox", () => $ddda48ae832cd17d$export$31f2f19e6c50453d, (v) => $ddda48ae832cd17d$export$31f2f19e6c50453d = v);
$parcel$export($ddda48ae832cd17d$exports, "mobile-input", () => $ddda48ae832cd17d$export$bededa1c0ac414c5, (v) => $ddda48ae832cd17d$export$bededa1c0ac414c5 = v);
$parcel$export($ddda48ae832cd17d$exports, "mobile-value", () => $ddda48ae832cd17d$export$e32ca2bb9f795355, (v) => $ddda48ae832cd17d$export$e32ca2bb9f795355 = v);
$parcel$export($ddda48ae832cd17d$exports, "tray-dialog", () => $ddda48ae832cd17d$export$3d4198eea215577a, (v) => $ddda48ae832cd17d$export$3d4198eea215577a = v);
$parcel$export($ddda48ae832cd17d$exports, "tray-textfield", () => $ddda48ae832cd17d$export$73cc97e04e76a6d0, (v) => $ddda48ae832cd17d$export$73cc97e04e76a6d0 = v);
$parcel$export($ddda48ae832cd17d$exports, "has-label", () => $ddda48ae832cd17d$export$b8e014ff9ad5e5e2, (v) => $ddda48ae832cd17d$export$b8e014ff9ad5e5e2 = v);
$parcel$export($ddda48ae832cd17d$exports, "tray-textfield-input", () => $ddda48ae832cd17d$export$feb70b7bc717e2b8, (v) => $ddda48ae832cd17d$export$feb70b7bc717e2b8 = v);
$parcel$export($ddda48ae832cd17d$exports, "tray-listbox", () => $ddda48ae832cd17d$export$7281dc54d38d4bed, (v) => $ddda48ae832cd17d$export$7281dc54d38d4bed = v);
var $ddda48ae832cd17d$export$5b1939da91820a47;
var $ddda48ae832cd17d$export$31f2f19e6c50453d;
var $ddda48ae832cd17d$export$bededa1c0ac414c5;
var $ddda48ae832cd17d$export$e32ca2bb9f795355;
var $ddda48ae832cd17d$export$3d4198eea215577a;
var $ddda48ae832cd17d$export$73cc97e04e76a6d0;
var $ddda48ae832cd17d$export$b8e014ff9ad5e5e2;
var $ddda48ae832cd17d$export$feb70b7bc717e2b8;
var $ddda48ae832cd17d$export$7281dc54d38d4bed;
$ddda48ae832cd17d$export$5b1939da91820a47 = "no-results_743018";
$ddda48ae832cd17d$export$31f2f19e6c50453d = "mobile-combobox_743018";
$ddda48ae832cd17d$export$bededa1c0ac414c5 = "mobile-input_743018";
$ddda48ae832cd17d$export$e32ca2bb9f795355 = "mobile-value_743018";
$ddda48ae832cd17d$export$3d4198eea215577a = "tray-dialog_743018";
$ddda48ae832cd17d$export$73cc97e04e76a6d0 = "tray-textfield_743018";
$ddda48ae832cd17d$export$b8e014ff9ad5e5e2 = "has-label_743018";
$ddda48ae832cd17d$export$feb70b7bc717e2b8 = "tray-textfield-input_743018";
$ddda48ae832cd17d$export$7281dc54d38d4bed = "tray-listbox_743018";






var $44bf33e45cbed0d5$exports = {};
var $c12113d2d694ce1c$exports = {};
$c12113d2d694ce1c$exports = JSON.parse("{\"clear\":\"مسح\",\"invalid\":\"(غير صالح)\",\"loading\":\"جارٍ التحميل...\",\"noResults\":\"لا توجد نتائج\"}");


var $f4ab0a1456298174$exports = {};
$f4ab0a1456298174$exports = JSON.parse("{\"clear\":\"Изчисти\",\"invalid\":\"(невалиден)\",\"loading\":\"Зареждане...\",\"noResults\":\"Няма резултати\"}");


var $78ee82ab81be24de$exports = {};
$78ee82ab81be24de$exports = JSON.parse("{\"clear\":\"Vymazat\",\"invalid\":\"(neplatné)\",\"loading\":\"Načítání...\",\"noResults\":\"Žádné výsledky\"}");


var $57ca434ad64e415f$exports = {};
$57ca434ad64e415f$exports = JSON.parse("{\"clear\":\"Ryd\",\"invalid\":\"(ugyldig)\",\"loading\":\"Indlæser ...\",\"noResults\":\"Ingen resultater\"}");


var $f96a02434d48bc86$exports = {};
$f96a02434d48bc86$exports = JSON.parse("{\"clear\":\"Löschen\",\"invalid\":\"(ungültig)\",\"loading\":\"Wird geladen...\",\"noResults\":\"Keine Ergebnisse\"}");


var $6a1083b6b25aa271$exports = {};
$6a1083b6b25aa271$exports = JSON.parse("{\"clear\":\"Καθαρισμός\",\"invalid\":\"(δεν ισχύει)\",\"loading\":\"Φόρτωση...\",\"noResults\":\"Χωρίς αποτέλεσμα\"}");


var $aacf3b7de80670b6$exports = {};
$aacf3b7de80670b6$exports = JSON.parse("{\"loading\":\"Loading...\",\"noResults\":\"No results\",\"clear\":\"Clear\",\"invalid\":\"(invalid)\"}");


var $880abf9993dee87e$exports = {};
$880abf9993dee87e$exports = JSON.parse("{\"clear\":\"Borrar\",\"invalid\":\"(no válido)\",\"loading\":\"Cargando...\",\"noResults\":\"Sin resultados\"}");


var $830efdd0ea207b5d$exports = {};
$830efdd0ea207b5d$exports = JSON.parse("{\"clear\":\"Puhasta\",\"invalid\":\"(kehtetu)\",\"loading\":\"Laadimine...\",\"noResults\":\"Tulemusi pole\"}");


var $c2b787e4d0298787$exports = {};
$c2b787e4d0298787$exports = JSON.parse("{\"clear\":\"Kirkas\",\"invalid\":\"(epäkelpo)\",\"loading\":\"Ladataan...\",\"noResults\":\"Ei tuloksia\"}");


var $94c6c8332f959284$exports = {};
$94c6c8332f959284$exports = JSON.parse("{\"clear\":\"Effacer\",\"invalid\":\"(non valide)\",\"loading\":\"Chargement en cours...\",\"noResults\":\"Aucun résultat\"}");


var $b00f1fdaf73b98dc$exports = {};
$b00f1fdaf73b98dc$exports = JSON.parse("{\"clear\":\"נקי\",\"invalid\":\"(לא חוקי)\",\"loading\":\"טוען...\",\"noResults\":\"אין תוצאות\"}");


var $74dfbb9591b40514$exports = {};
$74dfbb9591b40514$exports = JSON.parse("{\"clear\":\"Izbriši\",\"invalid\":\"(nevažeće)\",\"loading\":\"Učitavam...\",\"noResults\":\"Nema rezultata\"}");


var $6f9dd1959e5c5978$exports = {};
$6f9dd1959e5c5978$exports = JSON.parse("{\"clear\":\"Törlés\",\"invalid\":\"(érvénytelen)\",\"loading\":\"Betöltés folyamatban…\",\"noResults\":\"Nincsenek találatok\"}");


var $9a35e1d3b27df84d$exports = {};
$9a35e1d3b27df84d$exports = JSON.parse("{\"clear\":\"Cancella\",\"invalid\":\"(non valido)\",\"loading\":\"Caricamento in corso...\",\"noResults\":\"Nessun risultato\"}");


var $872a90f64b389264$exports = {};
$872a90f64b389264$exports = JSON.parse("{\"clear\":\"クリア\",\"invalid\":\"(無効)\",\"loading\":\"読み込み中...\",\"noResults\":\"結果なし\"}");


var $7c3c7dd826f32afa$exports = {};
$7c3c7dd826f32afa$exports = JSON.parse("{\"clear\":\"지우기\",\"invalid\":\"(유효하지 않음)\",\"loading\":\"로드 중...\",\"noResults\":\"결과 없음\"}");


var $bfe55744defe24ce$exports = {};
$bfe55744defe24ce$exports = JSON.parse("{\"clear\":\"Skaidrus\",\"invalid\":\"(netinkama)\",\"loading\":\"Įkeliama...\",\"noResults\":\"Be rezultatų\"}");


var $a3fe5b109f584088$exports = {};
$a3fe5b109f584088$exports = JSON.parse("{\"clear\":\"Notīrīt\",\"invalid\":\"(nederīgs)\",\"loading\":\"Notiek ielāde...\",\"noResults\":\"Nav rezultātu\"}");


var $04471ddf0fa9077c$exports = {};
$04471ddf0fa9077c$exports = JSON.parse("{\"clear\":\"Tøm\",\"invalid\":\"(ugyldig)\",\"loading\":\"Laster inn ...\",\"noResults\":\"Ingen resultater\"}");


var $e7a6a77bdc3f498e$exports = {};
$e7a6a77bdc3f498e$exports = JSON.parse("{\"clear\":\"Helder\",\"invalid\":\"(ongeldig)\",\"loading\":\"Laden...\",\"noResults\":\"Geen resultaten\"}");


var $c7fd89af12a5eba5$exports = {};
$c7fd89af12a5eba5$exports = JSON.parse("{\"clear\":\"Wyczyść\",\"invalid\":\"(nieprawidłowy)\",\"loading\":\"Trwa ładowanie...\",\"noResults\":\"Brak wyników\"}");


var $848f5c17d4f2be33$exports = {};
$848f5c17d4f2be33$exports = JSON.parse("{\"clear\":\"Limpar\",\"invalid\":\"(inválido)\",\"loading\":\"Carregando...\",\"noResults\":\"Nenhum resultado\"}");


var $f830b605c05751ee$exports = {};
$f830b605c05751ee$exports = JSON.parse("{\"clear\":\"Limpar\",\"invalid\":\"(inválido)\",\"loading\":\"A carregar...\",\"noResults\":\"Sem resultados\"}");


var $ed561c617b37a04b$exports = {};
$ed561c617b37a04b$exports = JSON.parse("{\"clear\":\"Golire\",\"invalid\":\"(nevalid)\",\"loading\":\"Se încarcă...\",\"noResults\":\"Niciun rezultat\"}");


var $d243261bd85e058d$exports = {};
$d243261bd85e058d$exports = JSON.parse("{\"clear\":\"Очистить\",\"invalid\":\"(недействительно)\",\"loading\":\"Загрузка...\",\"noResults\":\"Результаты отсутствуют\"}");


var $0d1fcaa384d6fa0d$exports = {};
$0d1fcaa384d6fa0d$exports = JSON.parse("{\"clear\":\"Vymazať\",\"invalid\":\"(neplatné)\",\"loading\":\"Načítava sa...\",\"noResults\":\"Žiadne výsledky\"}");


var $c572796800752b8d$exports = {};
$c572796800752b8d$exports = JSON.parse("{\"clear\":\"Jasen\",\"invalid\":\"(neveljavno)\",\"loading\":\"Nalaganje...\",\"noResults\":\"Ni rezultatov\"}");


var $1cfb0c5857a2c200$exports = {};
$1cfb0c5857a2c200$exports = JSON.parse("{\"clear\":\"Izbriši\",\"invalid\":\"(nevažeće)\",\"loading\":\"Učitavam...\",\"noResults\":\"Nema rezultata\"}");


var $a8b4704c5be4f8f7$exports = {};
$a8b4704c5be4f8f7$exports = JSON.parse("{\"clear\":\"Rensa\",\"invalid\":\"(ogiltigt)\",\"loading\":\"Läser in...\",\"noResults\":\"Inga resultat\"}");


var $e184409a5e614952$exports = {};
$e184409a5e614952$exports = JSON.parse("{\"clear\":\"Temizle\",\"invalid\":\"(geçersiz)\",\"loading\":\"Yükleniyor...\",\"noResults\":\"Sonuç yok\"}");


var $5a38b995b335b0ef$exports = {};
$5a38b995b335b0ef$exports = JSON.parse("{\"clear\":\"Очистити\",\"invalid\":\"(недійсне)\",\"loading\":\"Завантаження...\",\"noResults\":\"Результатів немає\"}");


var $fa23b594a91a782d$exports = {};
$fa23b594a91a782d$exports = JSON.parse("{\"clear\":\"透明\",\"invalid\":\"（无效）\",\"loading\":\"正在加载...\",\"noResults\":\"无结果\"}");


var $1fd677b56c80dbf5$exports = {};
$1fd677b56c80dbf5$exports = JSON.parse("{\"clear\":\"清除\",\"invalid\":\"(無效)\",\"loading\":\"正在載入...\",\"noResults\":\"無任何結果\"}");


$44bf33e45cbed0d5$exports = {
    "ar-AE": $c12113d2d694ce1c$exports,
    "bg-BG": $f4ab0a1456298174$exports,
    "cs-CZ": $78ee82ab81be24de$exports,
    "da-DK": $57ca434ad64e415f$exports,
    "de-DE": $f96a02434d48bc86$exports,
    "el-GR": $6a1083b6b25aa271$exports,
    "en-US": $aacf3b7de80670b6$exports,
    "es-ES": $880abf9993dee87e$exports,
    "et-EE": $830efdd0ea207b5d$exports,
    "fi-FI": $c2b787e4d0298787$exports,
    "fr-FR": $94c6c8332f959284$exports,
    "he-IL": $b00f1fdaf73b98dc$exports,
    "hr-HR": $74dfbb9591b40514$exports,
    "hu-HU": $6f9dd1959e5c5978$exports,
    "it-IT": $9a35e1d3b27df84d$exports,
    "ja-JP": $872a90f64b389264$exports,
    "ko-KR": $7c3c7dd826f32afa$exports,
    "lt-LT": $bfe55744defe24ce$exports,
    "lv-LV": $a3fe5b109f584088$exports,
    "nb-NO": $04471ddf0fa9077c$exports,
    "nl-NL": $e7a6a77bdc3f498e$exports,
    "pl-PL": $c7fd89af12a5eba5$exports,
    "pt-BR": $848f5c17d4f2be33$exports,
    "pt-PT": $f830b605c05751ee$exports,
    "ro-RO": $ed561c617b37a04b$exports,
    "ru-RU": $d243261bd85e058d$exports,
    "sk-SK": $0d1fcaa384d6fa0d$exports,
    "sl-SI": $c572796800752b8d$exports,
    "sr-SP": $1cfb0c5857a2c200$exports,
    "sv-SE": $a8b4704c5be4f8f7$exports,
    "tr-TR": $e184409a5e614952$exports,
    "uk-UA": $5a38b995b335b0ef$exports,
    "zh-CN": $fa23b594a91a782d$exports,
    "zh-TW": $1fd677b56c80dbf5$exports
};




var $fdb409b291c9958f$exports = {};

$parcel$export($fdb409b291c9958f$exports, "spectrum-Button", () => $fdb409b291c9958f$export$1db4cca5b4ade39a, (v) => $fdb409b291c9958f$export$1db4cca5b4ade39a = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton", () => $fdb409b291c9958f$export$53da69f51b770d3, (v) => $fdb409b291c9958f$export$53da69f51b770d3 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-LogicButton", () => $fdb409b291c9958f$export$36ae67a131730482, (v) => $fdb409b291c9958f$export$36ae67a131730482 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-FieldButton", () => $fdb409b291c9958f$export$a024c99ecf6b5741, (v) => $fdb409b291c9958f$export$a024c99ecf6b5741 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ClearButton", () => $fdb409b291c9958f$export$b752ce409e5660c, (v) => $fdb409b291c9958f$export$b752ce409e5660c = v);
$parcel$export($fdb409b291c9958f$exports, "focus-ring", () => $fdb409b291c9958f$export$f39a09f249340e2a, (v) => $fdb409b291c9958f$export$f39a09f249340e2a = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Icon", () => $fdb409b291c9958f$export$d374b04f30360026, (v) => $fdb409b291c9958f$export$d374b04f30360026 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton--emphasized", () => $fdb409b291c9958f$export$5aabb61a966e266f, (v) => $fdb409b291c9958f$export$5aabb61a966e266f = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton--staticColor", () => $fdb409b291c9958f$export$29b7b5d6a5a497c5, (v) => $fdb409b291c9958f$export$29b7b5d6a5a497c5 = v);
$parcel$export($fdb409b291c9958f$exports, "is-hovered", () => $fdb409b291c9958f$export$b8813cd5d7824ce7, (v) => $fdb409b291c9958f$export$b8813cd5d7824ce7 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Button-label", () => $fdb409b291c9958f$export$c081ba562e204942, (v) => $fdb409b291c9958f$export$c081ba562e204942 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton-label", () => $fdb409b291c9958f$export$ebb4635c66bacdb9, (v) => $fdb409b291c9958f$export$ebb4635c66bacdb9 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton-hold", () => $fdb409b291c9958f$export$4a4f01c443eff992, (v) => $fdb409b291c9958f$export$4a4f01c443eff992 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton--quiet", () => $fdb409b291c9958f$export$40e1f009544bacf3, (v) => $fdb409b291c9958f$export$40e1f009544bacf3 = v);
$parcel$export($fdb409b291c9958f$exports, "is-disabled", () => $fdb409b291c9958f$export$d35bc1e505d1ebbf, (v) => $fdb409b291c9958f$export$d35bc1e505d1ebbf = v);
$parcel$export($fdb409b291c9958f$exports, "is-open", () => $fdb409b291c9958f$export$a9781837241c946d, (v) => $fdb409b291c9958f$export$a9781837241c946d = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-FieldButton--quiet", () => $fdb409b291c9958f$export$5ca7b274e4fff9fc, (v) => $fdb409b291c9958f$export$5ca7b274e4fff9fc = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ClearButton--overBackground", () => $fdb409b291c9958f$export$116f2fccf941d34d, (v) => $fdb409b291c9958f$export$116f2fccf941d34d = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ClearButton--small", () => $fdb409b291c9958f$export$49280ffa5236ba8b, (v) => $fdb409b291c9958f$export$49280ffa5236ba8b = v);
$parcel$export($fdb409b291c9958f$exports, "is-focused", () => $fdb409b291c9958f$export$e7dc768d35940237, (v) => $fdb409b291c9958f$export$e7dc768d35940237 = v);
$parcel$export($fdb409b291c9958f$exports, "is-active", () => $fdb409b291c9958f$export$20fd0f7cd4e6112f, (v) => $fdb409b291c9958f$export$20fd0f7cd4e6112f = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Button--cta", () => $fdb409b291c9958f$export$b818fdf0052a2cab, (v) => $fdb409b291c9958f$export$b818fdf0052a2cab = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Button--primary", () => $fdb409b291c9958f$export$c1c3511f5e96e926, (v) => $fdb409b291c9958f$export$c1c3511f5e96e926 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Button--secondary", () => $fdb409b291c9958f$export$4f1b75526afd8ff3, (v) => $fdb409b291c9958f$export$4f1b75526afd8ff3 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Button--warning", () => $fdb409b291c9958f$export$6f97cef817ff1cac, (v) => $fdb409b291c9958f$export$6f97cef817ff1cac = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Button--overBackground", () => $fdb409b291c9958f$export$5cda1128f9112d2e, (v) => $fdb409b291c9958f$export$5cda1128f9112d2e = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Button--quiet", () => $fdb409b291c9958f$export$25f049c049371040, (v) => $fdb409b291c9958f$export$25f049c049371040 = v);
$parcel$export($fdb409b291c9958f$exports, "is-selected", () => $fdb409b291c9958f$export$1e0fb04f31d3c22a, (v) => $fdb409b291c9958f$export$1e0fb04f31d3c22a = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton--staticWhite", () => $fdb409b291c9958f$export$ccc536e1adbdc059, (v) => $fdb409b291c9958f$export$ccc536e1adbdc059 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton--staticBlack", () => $fdb409b291c9958f$export$8f7c9db66978d20f, (v) => $fdb409b291c9958f$export$8f7c9db66978d20f = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-LogicButton--and", () => $fdb409b291c9958f$export$e52daab43a62c528, (v) => $fdb409b291c9958f$export$e52daab43a62c528 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-LogicButton--or", () => $fdb409b291c9958f$export$da76be6126aaa633, (v) => $fdb409b291c9958f$export$da76be6126aaa633 = v);
$parcel$export($fdb409b291c9958f$exports, "is-placeholder", () => $fdb409b291c9958f$export$e5b2f5233e4e5194, (v) => $fdb409b291c9958f$export$e5b2f5233e4e5194 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-FieldButton--invalid", () => $fdb409b291c9958f$export$c9f503f672e8a3c1, (v) => $fdb409b291c9958f$export$c9f503f672e8a3c1 = v);
var $fdb409b291c9958f$export$1db4cca5b4ade39a;
var $fdb409b291c9958f$export$53da69f51b770d3;
var $fdb409b291c9958f$export$36ae67a131730482;
var $fdb409b291c9958f$export$a024c99ecf6b5741;
var $fdb409b291c9958f$export$b752ce409e5660c;
var $fdb409b291c9958f$export$f39a09f249340e2a;
var $fdb409b291c9958f$export$d374b04f30360026;
var $fdb409b291c9958f$export$5aabb61a966e266f;
var $fdb409b291c9958f$export$29b7b5d6a5a497c5;
var $fdb409b291c9958f$export$b8813cd5d7824ce7;
var $fdb409b291c9958f$export$c081ba562e204942;
var $fdb409b291c9958f$export$ebb4635c66bacdb9;
var $fdb409b291c9958f$export$4a4f01c443eff992;
var $fdb409b291c9958f$export$40e1f009544bacf3;
var $fdb409b291c9958f$export$d35bc1e505d1ebbf;
var $fdb409b291c9958f$export$a9781837241c946d;
var $fdb409b291c9958f$export$5ca7b274e4fff9fc;
var $fdb409b291c9958f$export$116f2fccf941d34d;
var $fdb409b291c9958f$export$49280ffa5236ba8b;
var $fdb409b291c9958f$export$e7dc768d35940237;
var $fdb409b291c9958f$export$20fd0f7cd4e6112f;
var $fdb409b291c9958f$export$b818fdf0052a2cab;
var $fdb409b291c9958f$export$c1c3511f5e96e926;
var $fdb409b291c9958f$export$4f1b75526afd8ff3;
var $fdb409b291c9958f$export$6f97cef817ff1cac;
var $fdb409b291c9958f$export$5cda1128f9112d2e;
var $fdb409b291c9958f$export$25f049c049371040;
var $fdb409b291c9958f$export$1e0fb04f31d3c22a;
var $fdb409b291c9958f$export$ccc536e1adbdc059;
var $fdb409b291c9958f$export$8f7c9db66978d20f;
var $fdb409b291c9958f$export$e52daab43a62c528;
var $fdb409b291c9958f$export$da76be6126aaa633;
var $fdb409b291c9958f$export$e5b2f5233e4e5194;
var $fdb409b291c9958f$export$c9f503f672e8a3c1;
$fdb409b291c9958f$export$1db4cca5b4ade39a = "spectrum-Button_e2d99e";
$fdb409b291c9958f$export$53da69f51b770d3 = "spectrum-ActionButton_e2d99e";
$fdb409b291c9958f$export$36ae67a131730482 = "spectrum-LogicButton_e2d99e";
$fdb409b291c9958f$export$a024c99ecf6b5741 = "spectrum-FieldButton_e2d99e";
$fdb409b291c9958f$export$b752ce409e5660c = "spectrum-ClearButton_e2d99e";
$fdb409b291c9958f$export$f39a09f249340e2a = "focus-ring_e2d99e";
$fdb409b291c9958f$export$d374b04f30360026 = "spectrum-Icon_e2d99e";
$fdb409b291c9958f$export$5aabb61a966e266f = "spectrum-ActionButton--emphasized_e2d99e";
$fdb409b291c9958f$export$29b7b5d6a5a497c5 = "spectrum-ActionButton--staticColor_e2d99e";
$fdb409b291c9958f$export$b8813cd5d7824ce7 = "is-hovered_e2d99e";
$fdb409b291c9958f$export$c081ba562e204942 = "spectrum-Button-label_e2d99e";
$fdb409b291c9958f$export$ebb4635c66bacdb9 = "spectrum-ActionButton-label_e2d99e";
$fdb409b291c9958f$export$4a4f01c443eff992 = "spectrum-ActionButton-hold_e2d99e";
$fdb409b291c9958f$export$40e1f009544bacf3 = "spectrum-ActionButton--quiet_e2d99e";
$fdb409b291c9958f$export$d35bc1e505d1ebbf = "is-disabled_e2d99e";
$fdb409b291c9958f$export$a9781837241c946d = "is-open_e2d99e";
$fdb409b291c9958f$export$5ca7b274e4fff9fc = "spectrum-FieldButton--quiet_e2d99e";
$fdb409b291c9958f$export$116f2fccf941d34d = "spectrum-ClearButton--overBackground_e2d99e";
$fdb409b291c9958f$export$49280ffa5236ba8b = "spectrum-ClearButton--small_e2d99e";
$fdb409b291c9958f$export$e7dc768d35940237 = "is-focused_e2d99e";
$fdb409b291c9958f$export$20fd0f7cd4e6112f = "is-active_e2d99e";
$fdb409b291c9958f$export$b818fdf0052a2cab = "spectrum-Button--cta_e2d99e";
$fdb409b291c9958f$export$c1c3511f5e96e926 = "spectrum-Button--primary_e2d99e";
$fdb409b291c9958f$export$4f1b75526afd8ff3 = "spectrum-Button--secondary_e2d99e";
$fdb409b291c9958f$export$6f97cef817ff1cac = "spectrum-Button--warning_e2d99e";
$fdb409b291c9958f$export$5cda1128f9112d2e = "spectrum-Button--overBackground_e2d99e";
$fdb409b291c9958f$export$25f049c049371040 = "spectrum-Button--quiet_e2d99e";
$fdb409b291c9958f$export$1e0fb04f31d3c22a = "is-selected_e2d99e";
$fdb409b291c9958f$export$ccc536e1adbdc059 = "spectrum-ActionButton--staticWhite_e2d99e";
$fdb409b291c9958f$export$8f7c9db66978d20f = "spectrum-ActionButton--staticBlack_e2d99e";
$fdb409b291c9958f$export$e52daab43a62c528 = "spectrum-LogicButton--and_e2d99e";
$fdb409b291c9958f$export$da76be6126aaa633 = "spectrum-LogicButton--or_e2d99e";
$fdb409b291c9958f$export$e5b2f5233e4e5194 = "is-placeholder_e2d99e";
$fdb409b291c9958f$export$c9f503f672e8a3c1 = "spectrum-FieldButton--invalid_e2d99e";













var $0043d8d062ef413c$exports = {};

$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel", () => $0043d8d062ef413c$export$493d18e796ae054, (v) => $0043d8d062ef413c$export$493d18e796ae054 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel--positionSide", () => $0043d8d062ef413c$export$9e6f19dc21f22f2e, (v) => $0043d8d062ef413c$export$9e6f19dc21f22f2e = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel-requiredIcon", () => $0043d8d062ef413c$export$7c47642c7d46f3c9, (v) => $0043d8d062ef413c$export$7c47642c7d46f3c9 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel--alignEnd", () => $0043d8d062ef413c$export$885efcc08143a987, (v) => $0043d8d062ef413c$export$885efcc08143a987 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field", () => $0043d8d062ef413c$export$f6d480ae1e56eba0, (v) => $0043d8d062ef413c$export$f6d480ae1e56eba0 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field--positionTop", () => $0043d8d062ef413c$export$a4ea780a9064d7f9, (v) => $0043d8d062ef413c$export$a4ea780a9064d7f9 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field-field", () => $0043d8d062ef413c$export$3ab8a3dc7f0563d, (v) => $0043d8d062ef413c$export$3ab8a3dc7f0563d = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field-field--multiline", () => $0043d8d062ef413c$export$f58a1e966a92ba5e, (v) => $0043d8d062ef413c$export$f58a1e966a92ba5e = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field--positionSide", () => $0043d8d062ef413c$export$2963225c91179589, (v) => $0043d8d062ef413c$export$2963225c91179589 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field-wrapper", () => $0043d8d062ef413c$export$127df0b7290eb8ef, (v) => $0043d8d062ef413c$export$127df0b7290eb8ef = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form", () => $0043d8d062ef413c$export$94d2bcc94afabd89, (v) => $0043d8d062ef413c$export$94d2bcc94afabd89 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form--positionSide", () => $0043d8d062ef413c$export$7f44db659563c8f4, (v) => $0043d8d062ef413c$export$7f44db659563c8f4 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form--positionTop", () => $0043d8d062ef413c$export$11b8bc81fe551057, (v) => $0043d8d062ef413c$export$11b8bc81fe551057 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form-itemLabel", () => $0043d8d062ef413c$export$51ec27881a1e3928, (v) => $0043d8d062ef413c$export$51ec27881a1e3928 = v);
$parcel$export($0043d8d062ef413c$exports, "is-disabled", () => $0043d8d062ef413c$export$d35bc1e505d1ebbf, (v) => $0043d8d062ef413c$export$d35bc1e505d1ebbf = v);
var $0043d8d062ef413c$export$493d18e796ae054;
var $0043d8d062ef413c$export$9e6f19dc21f22f2e;
var $0043d8d062ef413c$export$7c47642c7d46f3c9;
var $0043d8d062ef413c$export$885efcc08143a987;
var $0043d8d062ef413c$export$f6d480ae1e56eba0;
var $0043d8d062ef413c$export$a4ea780a9064d7f9;
var $0043d8d062ef413c$export$3ab8a3dc7f0563d;
var $0043d8d062ef413c$export$f58a1e966a92ba5e;
var $0043d8d062ef413c$export$2963225c91179589;
var $0043d8d062ef413c$export$127df0b7290eb8ef;
var $0043d8d062ef413c$export$94d2bcc94afabd89;
var $0043d8d062ef413c$export$7f44db659563c8f4;
var $0043d8d062ef413c$export$11b8bc81fe551057;
var $0043d8d062ef413c$export$51ec27881a1e3928;
var $0043d8d062ef413c$export$d35bc1e505d1ebbf;
$0043d8d062ef413c$export$493d18e796ae054 = "spectrum-FieldLabel_d2db1f";
$0043d8d062ef413c$export$9e6f19dc21f22f2e = "spectrum-FieldLabel--positionSide_d2db1f";
$0043d8d062ef413c$export$7c47642c7d46f3c9 = "spectrum-FieldLabel-requiredIcon_d2db1f";
$0043d8d062ef413c$export$885efcc08143a987 = "spectrum-FieldLabel--alignEnd_d2db1f";
$0043d8d062ef413c$export$f6d480ae1e56eba0 = "spectrum-Field_d2db1f";
$0043d8d062ef413c$export$a4ea780a9064d7f9 = "spectrum-Field--positionTop_d2db1f";
$0043d8d062ef413c$export$3ab8a3dc7f0563d = "spectrum-Field-field_d2db1f";
$0043d8d062ef413c$export$f58a1e966a92ba5e = "spectrum-Field-field--multiline_d2db1f";
$0043d8d062ef413c$export$2963225c91179589 = "spectrum-Field--positionSide_d2db1f";
$0043d8d062ef413c$export$127df0b7290eb8ef = "spectrum-Field-wrapper_d2db1f";
$0043d8d062ef413c$export$94d2bcc94afabd89 = "spectrum-Form_d2db1f";
$0043d8d062ef413c$export$7f44db659563c8f4 = "spectrum-Form--positionSide_d2db1f";
$0043d8d062ef413c$export$11b8bc81fe551057 = "spectrum-Form--positionTop_d2db1f";
$0043d8d062ef413c$export$51ec27881a1e3928 = "spectrum-Form-itemLabel_d2db1f";
$0043d8d062ef413c$export$d35bc1e505d1ebbf = "is-disabled_d2db1f";






var $7cdb83ac3ccb92ab$exports = {};

$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Search", () => $7cdb83ac3ccb92ab$export$713911d754e1e4ae, (v) => $7cdb83ac3ccb92ab$export$713911d754e1e4ae = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-ClearButton", () => $7cdb83ac3ccb92ab$export$b752ce409e5660c, (v) => $7cdb83ac3ccb92ab$export$b752ce409e5660c = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "is-quiet", () => $7cdb83ac3ccb92ab$export$6d5c5433f0da4241, (v) => $7cdb83ac3ccb92ab$export$6d5c5433f0da4241 = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Search-input", () => $7cdb83ac3ccb92ab$export$50e02603f2d97ae1, (v) => $7cdb83ac3ccb92ab$export$50e02603f2d97ae1 = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Search--invalid", () => $7cdb83ac3ccb92ab$export$77a6151a0b46a6a3, (v) => $7cdb83ac3ccb92ab$export$77a6151a0b46a6a3 = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Search--valid", () => $7cdb83ac3ccb92ab$export$aa81e3ab40f37567, (v) => $7cdb83ac3ccb92ab$export$aa81e3ab40f37567 = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Search--loadable", () => $7cdb83ac3ccb92ab$export$9c74d64afae3b2d4, (v) => $7cdb83ac3ccb92ab$export$9c74d64afae3b2d4 = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Search-circleLoader", () => $7cdb83ac3ccb92ab$export$70ae4353602451dc, (v) => $7cdb83ac3ccb92ab$export$70ae4353602451dc = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Search-validationIcon", () => $7cdb83ac3ccb92ab$export$938f0981825eae6e, (v) => $7cdb83ac3ccb92ab$export$938f0981825eae6e = v);
$parcel$export($7cdb83ac3ccb92ab$exports, "spectrum-Textfield", () => $7cdb83ac3ccb92ab$export$2c8a10299d8b3418, (v) => $7cdb83ac3ccb92ab$export$2c8a10299d8b3418 = v);
var $7cdb83ac3ccb92ab$export$713911d754e1e4ae;
var $7cdb83ac3ccb92ab$export$b752ce409e5660c;
var $7cdb83ac3ccb92ab$export$6d5c5433f0da4241;
var $7cdb83ac3ccb92ab$export$50e02603f2d97ae1;
var $7cdb83ac3ccb92ab$export$77a6151a0b46a6a3;
var $7cdb83ac3ccb92ab$export$aa81e3ab40f37567;
var $7cdb83ac3ccb92ab$export$9c74d64afae3b2d4;
var $7cdb83ac3ccb92ab$export$70ae4353602451dc;
var $7cdb83ac3ccb92ab$export$938f0981825eae6e;
var $7cdb83ac3ccb92ab$export$2c8a10299d8b3418;
$7cdb83ac3ccb92ab$export$713911d754e1e4ae = "spectrum-Search_d3c0f3";
$7cdb83ac3ccb92ab$export$b752ce409e5660c = "spectrum-ClearButton_d3c0f3";
$7cdb83ac3ccb92ab$export$6d5c5433f0da4241 = "is-quiet_d3c0f3";
$7cdb83ac3ccb92ab$export$50e02603f2d97ae1 = "spectrum-Search-input_d3c0f3";
$7cdb83ac3ccb92ab$export$77a6151a0b46a6a3 = "spectrum-Search--invalid_d3c0f3";
$7cdb83ac3ccb92ab$export$aa81e3ab40f37567 = "spectrum-Search--valid_d3c0f3";
$7cdb83ac3ccb92ab$export$9c74d64afae3b2d4 = "spectrum-Search--loadable_d3c0f3";
$7cdb83ac3ccb92ab$export$70ae4353602451dc = "spectrum-Search-circleLoader_d3c0f3";
$7cdb83ac3ccb92ab$export$938f0981825eae6e = "spectrum-Search-validationIcon_d3c0f3";
$7cdb83ac3ccb92ab$export$2c8a10299d8b3418 = "spectrum-Textfield_d3c0f3";



var $ae1ecdab5fc4ef1b$exports = {};

$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-InputGroup", () => $ae1ecdab5fc4ef1b$export$5be359c1227eedcb, (v) => $ae1ecdab5fc4ef1b$export$5be359c1227eedcb = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-FieldButton", () => $ae1ecdab5fc4ef1b$export$a024c99ecf6b5741, (v) => $ae1ecdab5fc4ef1b$export$a024c99ecf6b5741 = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "is-disabled", () => $ae1ecdab5fc4ef1b$export$d35bc1e505d1ebbf, (v) => $ae1ecdab5fc4ef1b$export$d35bc1e505d1ebbf = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-InputGroup--invalid", () => $ae1ecdab5fc4ef1b$export$3b48509739a9cda0, (v) => $ae1ecdab5fc4ef1b$export$3b48509739a9cda0 = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-InputGroup--quiet", () => $ae1ecdab5fc4ef1b$export$ebf265ffc062029b, (v) => $ae1ecdab5fc4ef1b$export$ebf265ffc062029b = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-InputGroup-field", () => $ae1ecdab5fc4ef1b$export$ea77459595e26bec, (v) => $ae1ecdab5fc4ef1b$export$ea77459595e26bec = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-InputGroup-input-validationIcon", () => $ae1ecdab5fc4ef1b$export$f7ab13ee113f4884, (v) => $ae1ecdab5fc4ef1b$export$f7ab13ee113f4884 = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-InputGroup-input-circleLoader", () => $ae1ecdab5fc4ef1b$export$dcc9e00cdad967fa, (v) => $ae1ecdab5fc4ef1b$export$dcc9e00cdad967fa = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-InputGroup-input", () => $ae1ecdab5fc4ef1b$export$92706f5c6fe4074, (v) => $ae1ecdab5fc4ef1b$export$92706f5c6fe4074 = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-InputGroup-icon", () => $ae1ecdab5fc4ef1b$export$b91b6af799fd93a8, (v) => $ae1ecdab5fc4ef1b$export$b91b6af799fd93a8 = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-InputGroup-popover--quiet", () => $ae1ecdab5fc4ef1b$export$5e116178f9b1556a, (v) => $ae1ecdab5fc4ef1b$export$5e116178f9b1556a = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-Datepicker--range", () => $ae1ecdab5fc4ef1b$export$1d8b0b29e11d7cd, (v) => $ae1ecdab5fc4ef1b$export$1d8b0b29e11d7cd = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-Datepicker--datetimeRange", () => $ae1ecdab5fc4ef1b$export$58dd7cd23c92699f, (v) => $ae1ecdab5fc4ef1b$export$58dd7cd23c92699f = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-Datepicker-startField", () => $ae1ecdab5fc4ef1b$export$80f9c838a8406469, (v) => $ae1ecdab5fc4ef1b$export$80f9c838a8406469 = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-Datepicker-endField", () => $ae1ecdab5fc4ef1b$export$1c72fb41fa4f21ee, (v) => $ae1ecdab5fc4ef1b$export$1c72fb41fa4f21ee = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-Datepicker--rangeDash", () => $ae1ecdab5fc4ef1b$export$202113e883898dc, (v) => $ae1ecdab5fc4ef1b$export$202113e883898dc = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "is-focused", () => $ae1ecdab5fc4ef1b$export$e7dc768d35940237, (v) => $ae1ecdab5fc4ef1b$export$e7dc768d35940237 = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-Datepicker-focusRing", () => $ae1ecdab5fc4ef1b$export$eb262b15b9261fde, (v) => $ae1ecdab5fc4ef1b$export$eb262b15b9261fde = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "is-hovered", () => $ae1ecdab5fc4ef1b$export$b8813cd5d7824ce7, (v) => $ae1ecdab5fc4ef1b$export$b8813cd5d7824ce7 = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "focus-ring", () => $ae1ecdab5fc4ef1b$export$f39a09f249340e2a, (v) => $ae1ecdab5fc4ef1b$export$f39a09f249340e2a = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "is-selected", () => $ae1ecdab5fc4ef1b$export$1e0fb04f31d3c22a, (v) => $ae1ecdab5fc4ef1b$export$1e0fb04f31d3c22a = v);
$parcel$export($ae1ecdab5fc4ef1b$exports, "spectrum-FieldButton--invalid", () => $ae1ecdab5fc4ef1b$export$c9f503f672e8a3c1, (v) => $ae1ecdab5fc4ef1b$export$c9f503f672e8a3c1 = v);
var $ae1ecdab5fc4ef1b$export$5be359c1227eedcb;
var $ae1ecdab5fc4ef1b$export$a024c99ecf6b5741;
var $ae1ecdab5fc4ef1b$export$d35bc1e505d1ebbf;
var $ae1ecdab5fc4ef1b$export$3b48509739a9cda0;
var $ae1ecdab5fc4ef1b$export$ebf265ffc062029b;
var $ae1ecdab5fc4ef1b$export$ea77459595e26bec;
var $ae1ecdab5fc4ef1b$export$f7ab13ee113f4884;
var $ae1ecdab5fc4ef1b$export$dcc9e00cdad967fa;
var $ae1ecdab5fc4ef1b$export$92706f5c6fe4074;
var $ae1ecdab5fc4ef1b$export$b91b6af799fd93a8;
var $ae1ecdab5fc4ef1b$export$5e116178f9b1556a;
var $ae1ecdab5fc4ef1b$export$1d8b0b29e11d7cd;
var $ae1ecdab5fc4ef1b$export$58dd7cd23c92699f;
var $ae1ecdab5fc4ef1b$export$80f9c838a8406469;
var $ae1ecdab5fc4ef1b$export$1c72fb41fa4f21ee;
var $ae1ecdab5fc4ef1b$export$202113e883898dc;
var $ae1ecdab5fc4ef1b$export$e7dc768d35940237;
var $ae1ecdab5fc4ef1b$export$eb262b15b9261fde;
var $ae1ecdab5fc4ef1b$export$b8813cd5d7824ce7;
var $ae1ecdab5fc4ef1b$export$f39a09f249340e2a;
var $ae1ecdab5fc4ef1b$export$1e0fb04f31d3c22a;
var $ae1ecdab5fc4ef1b$export$c9f503f672e8a3c1;
$ae1ecdab5fc4ef1b$export$5be359c1227eedcb = "spectrum-InputGroup_a0942f";
$ae1ecdab5fc4ef1b$export$a024c99ecf6b5741 = "spectrum-FieldButton_a0942f";
$ae1ecdab5fc4ef1b$export$d35bc1e505d1ebbf = "is-disabled_a0942f";
$ae1ecdab5fc4ef1b$export$3b48509739a9cda0 = "spectrum-InputGroup--invalid_a0942f";
$ae1ecdab5fc4ef1b$export$ebf265ffc062029b = "spectrum-InputGroup--quiet_a0942f";
$ae1ecdab5fc4ef1b$export$ea77459595e26bec = "spectrum-InputGroup-field_a0942f";
$ae1ecdab5fc4ef1b$export$f7ab13ee113f4884 = "spectrum-InputGroup-input-validationIcon_a0942f";
$ae1ecdab5fc4ef1b$export$dcc9e00cdad967fa = "spectrum-InputGroup-input-circleLoader_a0942f";
$ae1ecdab5fc4ef1b$export$92706f5c6fe4074 = "spectrum-InputGroup-input_a0942f";
$ae1ecdab5fc4ef1b$export$b91b6af799fd93a8 = "spectrum-InputGroup-icon_a0942f";
$ae1ecdab5fc4ef1b$export$5e116178f9b1556a = "spectrum-InputGroup-popover--quiet_a0942f";
$ae1ecdab5fc4ef1b$export$1d8b0b29e11d7cd = "spectrum-Datepicker--range_a0942f";
$ae1ecdab5fc4ef1b$export$58dd7cd23c92699f = "spectrum-Datepicker--datetimeRange_a0942f";
$ae1ecdab5fc4ef1b$export$80f9c838a8406469 = "spectrum-Datepicker-startField_a0942f";
$ae1ecdab5fc4ef1b$export$1c72fb41fa4f21ee = "spectrum-Datepicker-endField_a0942f";
$ae1ecdab5fc4ef1b$export$202113e883898dc = "spectrum-Datepicker--rangeDash_a0942f";
$ae1ecdab5fc4ef1b$export$e7dc768d35940237 = "is-focused_a0942f";
$ae1ecdab5fc4ef1b$export$eb262b15b9261fde = "spectrum-Datepicker-focusRing_a0942f";
$ae1ecdab5fc4ef1b$export$b8813cd5d7824ce7 = "is-hovered_a0942f";
$ae1ecdab5fc4ef1b$export$f39a09f249340e2a = "focus-ring_a0942f";
$ae1ecdab5fc4ef1b$export$1e0fb04f31d3c22a = "is-selected_a0942f";
$ae1ecdab5fc4ef1b$export$c9f503f672e8a3c1 = "spectrum-FieldButton--invalid_a0942f";



var $484f319272acc7c5$exports = {};

$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield-input", () => $484f319272acc7c5$export$1ac009feb2b0bd0c, (v) => $484f319272acc7c5$export$1ac009feb2b0bd0c = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield", () => $484f319272acc7c5$export$2c8a10299d8b3418, (v) => $484f319272acc7c5$export$2c8a10299d8b3418 = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield--quiet", () => $484f319272acc7c5$export$bffd5e3d61a81737, (v) => $484f319272acc7c5$export$bffd5e3d61a81737 = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield--multiline", () => $484f319272acc7c5$export$9699f1c538a448a0, (v) => $484f319272acc7c5$export$9699f1c538a448a0 = v);
$parcel$export($484f319272acc7c5$exports, "is-placeholder", () => $484f319272acc7c5$export$e5b2f5233e4e5194, (v) => $484f319272acc7c5$export$e5b2f5233e4e5194 = v);
$parcel$export($484f319272acc7c5$exports, "is-hovered", () => $484f319272acc7c5$export$b8813cd5d7824ce7, (v) => $484f319272acc7c5$export$b8813cd5d7824ce7 = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield--valid", () => $484f319272acc7c5$export$c2abbc3f970170b7, (v) => $484f319272acc7c5$export$c2abbc3f970170b7 = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield--invalid", () => $484f319272acc7c5$export$58e50b10d30123df, (v) => $484f319272acc7c5$export$58e50b10d30123df = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield--loadable", () => $484f319272acc7c5$export$14fd30fea6f1348c, (v) => $484f319272acc7c5$export$14fd30fea6f1348c = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield-validationIcon", () => $484f319272acc7c5$export$cf0ead5b44db0da3, (v) => $484f319272acc7c5$export$cf0ead5b44db0da3 = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield-icon", () => $484f319272acc7c5$export$222d1ccd1870be1d, (v) => $484f319272acc7c5$export$222d1ccd1870be1d = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield-inputIcon", () => $484f319272acc7c5$export$c1cdcaa5fe76a871, (v) => $484f319272acc7c5$export$c1cdcaa5fe76a871 = v);
$parcel$export($484f319272acc7c5$exports, "spectrum-Textfield-circleLoader", () => $484f319272acc7c5$export$82a249fb4d6127, (v) => $484f319272acc7c5$export$82a249fb4d6127 = v);
$parcel$export($484f319272acc7c5$exports, "is-focused", () => $484f319272acc7c5$export$e7dc768d35940237, (v) => $484f319272acc7c5$export$e7dc768d35940237 = v);
$parcel$export($484f319272acc7c5$exports, "focus-ring", () => $484f319272acc7c5$export$f39a09f249340e2a, (v) => $484f319272acc7c5$export$f39a09f249340e2a = v);
$parcel$export($484f319272acc7c5$exports, "is-disabled", () => $484f319272acc7c5$export$d35bc1e505d1ebbf, (v) => $484f319272acc7c5$export$d35bc1e505d1ebbf = v);
var $484f319272acc7c5$export$1ac009feb2b0bd0c;
var $484f319272acc7c5$export$2c8a10299d8b3418;
var $484f319272acc7c5$export$bffd5e3d61a81737;
var $484f319272acc7c5$export$9699f1c538a448a0;
var $484f319272acc7c5$export$e5b2f5233e4e5194;
var $484f319272acc7c5$export$b8813cd5d7824ce7;
var $484f319272acc7c5$export$c2abbc3f970170b7;
var $484f319272acc7c5$export$58e50b10d30123df;
var $484f319272acc7c5$export$14fd30fea6f1348c;
var $484f319272acc7c5$export$cf0ead5b44db0da3;
var $484f319272acc7c5$export$222d1ccd1870be1d;
var $484f319272acc7c5$export$c1cdcaa5fe76a871;
var $484f319272acc7c5$export$82a249fb4d6127;
var $484f319272acc7c5$export$e7dc768d35940237;
var $484f319272acc7c5$export$f39a09f249340e2a;
var $484f319272acc7c5$export$d35bc1e505d1ebbf;
$484f319272acc7c5$export$1ac009feb2b0bd0c = "spectrum-Textfield-input_73bc77";
$484f319272acc7c5$export$2c8a10299d8b3418 = "spectrum-Textfield_73bc77";
$484f319272acc7c5$export$bffd5e3d61a81737 = "spectrum-Textfield--quiet_73bc77";
$484f319272acc7c5$export$9699f1c538a448a0 = "spectrum-Textfield--multiline_73bc77";
$484f319272acc7c5$export$e5b2f5233e4e5194 = "is-placeholder_73bc77";
$484f319272acc7c5$export$b8813cd5d7824ce7 = "is-hovered_73bc77";
$484f319272acc7c5$export$c2abbc3f970170b7 = "spectrum-Textfield--valid_73bc77";
$484f319272acc7c5$export$58e50b10d30123df = "spectrum-Textfield--invalid_73bc77";
$484f319272acc7c5$export$14fd30fea6f1348c = "spectrum-Textfield--loadable_73bc77";
$484f319272acc7c5$export$cf0ead5b44db0da3 = "spectrum-Textfield-validationIcon_73bc77";
$484f319272acc7c5$export$222d1ccd1870be1d = "spectrum-Textfield-icon_73bc77";
$484f319272acc7c5$export$c1cdcaa5fe76a871 = "spectrum-Textfield-inputIcon_73bc77";
$484f319272acc7c5$export$82a249fb4d6127 = "spectrum-Textfield-circleLoader_73bc77";
$484f319272acc7c5$export$e7dc768d35940237 = "is-focused_73bc77";
$484f319272acc7c5$export$f39a09f249340e2a = "focus-ring_73bc77";
$484f319272acc7c5$export$d35bc1e505d1ebbf = "is-disabled_73bc77";












const $e504d125e7664084$export$7637df911c069b4d = /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).forwardRef(function $e504d125e7664084$export$7637df911c069b4d(props, ref) {
    props = $dyK7H$reactspectrumprovider.useProviderProps(props);
    let { isQuiet: isQuiet , isDisabled: isDisabled , validationState: validationState , isReadOnly: isReadOnly  } = props;
    let { contains: contains  } = $dyK7H$reactariai18n.useFilter({
        sensitivity: 'base'
    });
    let state = $dyK7H$reactstatelycombobox.useComboBoxState({
        ...props,
        defaultFilter: contains,
        allowsEmptyCollection: true,
        // Needs to be false here otherwise we double up on commitSelection/commitCustomValue calls when
        // user taps on underlay (i.e. initial tap will call setFocused(false) -> commitSelection/commitCustomValue via onBlur,
        // then the closing of the tray will call setFocused(false) again due to cleanup effect)
        shouldCloseOnBlur: false
    });
    let buttonRef = $dyK7H$react.useRef();
    let domRef = $dyK7H$reactspectrumutils.useFocusableRef(ref, buttonRef);
    let { triggerProps: triggerProps , overlayProps: overlayProps  } = $dyK7H$reactariaoverlays.useOverlayTrigger({
        type: 'listbox'
    }, state, buttonRef);
    let { labelProps: labelProps , fieldProps: fieldProps  } = $dyK7H$reactarialabel.useField({
        ...props,
        labelElementType: 'span'
    });
    // Focus the button and show focus ring when clicking on the label
    labelProps.onClick = ()=>{
        if (!props.isDisabled) {
            buttonRef.current.focus();
            $dyK7H$reactariainteractions.setInteractionModality('keyboard');
        }
    };
    let onClose = ()=>state.commit()
    ;
    return(/*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement(($parcel$interopDefault($dyK7H$react)).Fragment, null, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactspectrumlabel.Field, {
        ...props,
        labelProps: labelProps,
        elementType: "span",
        ref: domRef,
        includeNecessityIndicatorInAccessibilityName: true
    }, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($e504d125e7664084$var$ComboBoxButton, {
        ...$dyK7H$reactariautils.mergeProps(triggerProps, fieldProps, {
            autoFocus: props.autoFocus
        }),
        ref: buttonRef,
        isQuiet: isQuiet,
        isDisabled: isDisabled,
        isPlaceholder: !state.inputValue,
        validationState: validationState,
        onPress: ()=>!isReadOnly && state.open(null, 'manual')
    }, state.inputValue || props.placeholder || '')), /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactspectrumoverlays.Tray, {
        isOpen: state.isOpen,
        onClose: onClose,
        isFixedHeight: true,
        isNonModal: true,
        ...overlayProps
    }, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($e504d125e7664084$var$ComboBoxTray, {
        ...props,
        onClose: onClose,
        overlayProps: overlayProps,
        state: state
    }))));
});
const $e504d125e7664084$var$ComboBoxButton = /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).forwardRef(function ComboBoxButton(props, ref) {
    let { isQuiet: isQuiet , isDisabled: isDisabled , isPlaceholder: isPlaceholder , validationState: validationState , children: children , style: style , className: className  } = props;
    let formatMessage = $dyK7H$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($44bf33e45cbed0d5$exports)));
    let valueId = $dyK7H$reactariautils.useId();
    let invalidId = $dyK7H$reactariautils.useId();
    let validationIcon = validationState === 'invalid' ? /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement(($parcel$interopDefault($dyK7H$spectrumiconsuiAlertMedium)), {
        id: invalidId,
        "aria-label": formatMessage('invalid')
    }) : /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement(($parcel$interopDefault($dyK7H$spectrumiconsuiCheckmarkMedium)), null);
    let validation = /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).cloneElement(validationIcon, {
        UNSAFE_className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($484f319272acc7c5$exports)), 'spectrum-Textfield-validationIcon', $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'spectrum-InputGroup-input-validationIcon'))
    });
    let { hoverProps: hoverProps , isHovered: isHovered  } = $dyK7H$reactariainteractions.useHover({
    });
    let { buttonProps: buttonProps , isPressed: isPressed  } = $dyK7H$reactariabutton.useButton({
        ...props,
        'aria-labelledby': [
            props['aria-labelledby'],
            props['aria-label'] && !props['aria-labelledby'] ? props.id : null,
            valueId,
            validationState === 'invalid' ? invalidId : null
        ].filter(Boolean).join(' '),
        elementType: 'div'
    }, ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactariafocus.FocusRing, {
        focusClass: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'is-focused'),
        focusRingClass: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement("div", {
        ...$dyK7H$reactariautils.mergeProps(hoverProps, buttonProps),
        "aria-haspopup": "dialog",
        ref: ref,
        style: {
            ...style,
            outline: 'none'
        },
        className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'spectrum-InputGroup', {
            'spectrum-InputGroup--quiet': isQuiet,
            'is-disabled': isDisabled,
            'spectrum-InputGroup--invalid': validationState === 'invalid',
            'is-hovered': isHovered
        }, $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ddda48ae832cd17d$exports)), 'mobile-combobox'), className)
    }, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement("div", {
        className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($484f319272acc7c5$exports)), 'spectrum-Textfield', {
            'spectrum-Textfield--invalid': validationState === 'invalid',
            'spectrum-Textfield--valid': validationState === 'valid',
            'spectrum-Textfield--quiet': isQuiet
        }, $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'spectrum-InputGroup-field'))
    }, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement("div", {
        className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($484f319272acc7c5$exports)), 'spectrum-Textfield-input', {
            'is-hovered': isHovered,
            'is-placeholder': isPlaceholder,
            'is-disabled': isDisabled
        }, $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'spectrum-InputGroup-input', $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($0043d8d062ef413c$exports)), 'spectrum-Field-field')), $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ddda48ae832cd17d$exports)), 'mobile-input'))
    }, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement("span", {
        id: valueId,
        className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ddda48ae832cd17d$exports)), 'mobile-value')
    }, children)), validationState ? validation : null), /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement("div", {
        className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($fdb409b291c9958f$exports)), 'spectrum-FieldButton', {
            'spectrum-FieldButton--quiet': isQuiet,
            'is-active': isPressed,
            'is-disabled': isDisabled,
            'spectrum-FieldButton--invalid': validationState === 'invalid',
            'is-hovered': isHovered
        }, $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'spectrum-FieldButton'))
    }, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement(($parcel$interopDefault($dyK7H$spectrumiconsuiChevronDownMedium)), {
        UNSAFE_className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'spectrum-Dropdown-chevron')
    })))));
});
function $e504d125e7664084$var$ComboBoxTray(props) {
    let { state: // completionMode = 'suggest',
    state , isDisabled: isDisabled , validationState: validationState , label: label , overlayProps: overlayProps , loadingState: loadingState , onLoadMore: onLoadMore , onClose: onClose  } = props;
    let timeout = $dyK7H$react.useRef(null);
    let [showLoading, setShowLoading] = $dyK7H$react.useState(false);
    let inputRef = $dyK7H$react.useRef();
    let buttonRef = $dyK7H$react.useRef();
    let popoverRef = $dyK7H$react.useRef();
    let listBoxRef = $dyK7H$react.useRef();
    let layout = $dyK7H$reactspectrumlistbox.useListBoxLayout(state);
    let formatMessage = $dyK7H$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($44bf33e45cbed0d5$exports)));
    let { inputProps: inputProps , listBoxProps: listBoxProps , labelProps: labelProps  } = $dyK7H$reactariacombobox.useComboBox({
        ...props,
        // completionMode,
        keyboardDelegate: layout,
        buttonRef: $dyK7H$reactspectrumutils.unwrapDOMRef(buttonRef),
        popoverRef: popoverRef,
        listBoxRef: listBoxRef,
        inputRef: inputRef
    }, state);
    ($parcel$interopDefault($dyK7H$react)).useEffect(()=>{
        $dyK7H$reactariafocus.focusSafely(inputRef.current);
        // When the tray unmounts, set state.isFocused (i.e. the tray input's focus tracker) to false.
        // This is to prevent state.isFocused from being set to true when the tray closes via tapping on the underlay
        // (FocusScope attempts to restore focus to the tray input when tapping outside the tray due to "contain")
        // Have to do this manually since React doesn't call onBlur when a component is unmounted: https://github.com/facebook/react/issues/12363
        return ()=>{
            state.setFocused(false);
        };
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
    let { dialogProps: dialogProps  } = $dyK7H$reactariadialog.useDialog({
        'aria-labelledby': $dyK7H$reactariautils.useId(labelProps.id)
    }, popoverRef);
    // Override the role of the input to "searchbox" instead of "combobox".
    // Since the listbox is always visible, the combobox role doesn't really give us anything.
    // VoiceOver on iOS reads "double tap to collapse" when focused on the input rather than
    // "double tap to edit text", as with a textbox or searchbox. We'd like double tapping to
    // open the virtual keyboard rather than closing the tray.
    inputProps.role = 'searchbox';
    inputProps['aria-haspopup'] = 'listbox';
    delete inputProps.onTouchEnd;
    let clearButton = /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactspectrumbutton.ClearButton, {
        preventFocus: true,
        "aria-label": formatMessage('clear'),
        excludeFromTabOrder: true,
        onPress: ()=>{
            state.setInputValue('');
            inputRef.current.focus();
        },
        UNSAFE_className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($7cdb83ac3ccb92ab$exports)), 'spectrum-ClearButton'),
        isDisabled: isDisabled
    });
    let loadingCircle = /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactspectrumprogress.ProgressCircle, {
        "aria-label": formatMessage('loading'),
        size: "S",
        isIndeterminate: true,
        UNSAFE_className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($7cdb83ac3ccb92ab$exports)), 'spectrum-Search-circleLoader', $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($484f319272acc7c5$exports)), 'spectrum-Textfield-circleLoader'))
    });
    // Close the software keyboard on scroll to give the user a bigger area to scroll.
    // But only do this if scrolling with touch, otherwise it can cause issues with touch
    // screen readers.
    let isTouchDown = $dyK7H$react.useRef(false);
    let onTouchStart = ()=>{
        isTouchDown.current = true;
    };
    let onTouchEnd = ()=>{
        isTouchDown.current = false;
    };
    let onScroll = $dyK7H$react.useCallback(()=>{
        if (!inputRef.current || document.activeElement !== inputRef.current || !isTouchDown.current) return;
        popoverRef.current.focus();
    }, [
        inputRef,
        popoverRef,
        isTouchDown
    ]);
    let inputValue = inputProps.value;
    let lastInputValue = $dyK7H$react.useRef(inputValue);
    $dyK7H$react.useEffect(()=>{
        if (loadingState === 'filtering' && !showLoading) {
            if (timeout.current === null) timeout.current = setTimeout(()=>{
                setShowLoading(true);
            }, 500);
            // If user is typing, clear the timer and restart since it is a new request
            if (inputValue !== lastInputValue.current) {
                clearTimeout(timeout.current);
                timeout.current = setTimeout(()=>{
                    setShowLoading(true);
                }, 500);
            }
        } else if (loadingState !== 'filtering') {
            // If loading is no longer happening, clear any timers and hide the loading circle
            setShowLoading(false);
            clearTimeout(timeout.current);
            timeout.current = null;
        }
        lastInputValue.current = inputValue;
    }, [
        loadingState,
        inputValue,
        showLoading
    ]);
    let onKeyDown = (e)=>{
        // Close virtual keyboard if user hits Enter w/o any focused options
        if (e.key === 'Enter' && state.selectionManager.focusedKey == null) popoverRef.current.focus();
        else inputProps.onKeyDown(e);
    };
    return(/*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactariafocus.FocusScope, {
        restoreFocus: true,
        contain: true
    }, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement("div", {
        ...$dyK7H$reactariautils.mergeProps(overlayProps, dialogProps),
        ref: popoverRef,
        className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ddda48ae832cd17d$exports)), 'tray-dialog')
    }, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactariaoverlays.DismissButton, {
        onDismiss: onClose
    }), /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactspectrumtextfield.TextFieldBase, {
        label: label,
        labelProps: labelProps,
        inputProps: {
            ...inputProps,
            onKeyDown: onKeyDown
        },
        inputRef: inputRef,
        isDisabled: isDisabled,
        isLoading: showLoading && loadingState === 'filtering',
        loadingIndicator: loadingState != null && loadingCircle,
        validationState: validationState,
        labelAlign: "start",
        labelPosition: "top",
        wrapperChildren: (state.inputValue !== '' || loadingState === 'filtering' || validationState != null) && !props.isReadOnly && clearButton,
        UNSAFE_className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($7cdb83ac3ccb92ab$exports)), 'spectrum-Search', 'spectrum-Textfield', 'spectrum-Search--loadable', {
            'spectrum-Search--invalid': validationState === 'invalid',
            'spectrum-Search--valid': validationState === 'valid'
        }, $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ddda48ae832cd17d$exports)), 'tray-textfield', {
            'has-label': !!props.label
        })),
        inputClassName: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ddda48ae832cd17d$exports)), 'tray-textfield-input', $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($7cdb83ac3ccb92ab$exports)), 'spectrum-Search-input')),
        validationIconClassName: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($7cdb83ac3ccb92ab$exports)), 'spectrum-Search-validationIcon')
    }), /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactspectrumlistbox.ListBoxBase, {
        ...listBoxProps,
        domProps: {
            onTouchStart: onTouchStart,
            onTouchEnd: onTouchEnd
        },
        disallowEmptySelection: true,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        shouldUseVirtualFocus: true,
        renderEmptyState: ()=>loadingState !== 'loading' && /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement("span", {
                className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ddda48ae832cd17d$exports)), 'no-results')
            }, formatMessage('noResults'))
        ,
        UNSAFE_className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ddda48ae832cd17d$exports)), 'tray-listbox'),
        ref: listBoxRef,
        onScroll: onScroll,
        onLoadMore: onLoadMore,
        isLoading: loadingState === 'loading' || loadingState === 'loadingMore'
    }), /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactariaoverlays.DismissButton, {
        onDismiss: onClose
    }))));
}















function $3740f8e8ff9aff46$var$ComboBox(props, ref) {
    props = $dyK7H$reactspectrumprovider.useProviderProps(props);
    let isMobile = $dyK7H$reactspectrumutils.useIsMobileDevice();
    if (isMobile) // menuTrigger=focus/manual don't apply to mobile combobox
    return(/*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($e504d125e7664084$export$7637df911c069b4d, {
        ...props,
        menuTrigger: "input",
        ref: ref
    }));
    else return(/*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($3740f8e8ff9aff46$var$ComboBoxBase, {
        ...props,
        ref: ref
    }));
}
const $3740f8e8ff9aff46$var$ComboBoxBase = /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).forwardRef(function ComboBoxBase(props, ref) {
    let { menuTrigger: menuTrigger = 'input' , shouldFlip: shouldFlip = true , direction: direction = 'bottom' , isQuiet: isQuiet , loadingState: loadingState , onLoadMore: onLoadMore  } = props;
    let formatMessage = $dyK7H$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($44bf33e45cbed0d5$exports)));
    let isAsync = loadingState != null;
    let popoverRef = $dyK7H$react.useRef();
    let unwrappedPopoverRef = $dyK7H$reactspectrumutils.useUnwrapDOMRef(popoverRef);
    let buttonRef = $dyK7H$react.useRef();
    let unwrappedButtonRef = $dyK7H$reactspectrumutils.useUnwrapDOMRef(buttonRef);
    let listBoxRef = $dyK7H$react.useRef();
    let inputRef = $dyK7H$react.useRef();
    let domRef = $dyK7H$reactspectrumutils.useFocusableRef(ref, inputRef);
    let { contains: contains  } = $dyK7H$reactariai18n.useFilter({
        sensitivity: 'base'
    });
    let state = $dyK7H$reactstatelycombobox.useComboBoxState({
        ...props,
        defaultFilter: contains,
        allowsEmptyCollection: isAsync
    });
    let layout = $dyK7H$reactspectrumlistbox.useListBoxLayout(state);
    let { buttonProps: buttonProps , inputProps: inputProps , listBoxProps: listBoxProps , labelProps: labelProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $dyK7H$reactariacombobox.useComboBox({
        ...props,
        keyboardDelegate: layout,
        buttonRef: unwrappedButtonRef,
        popoverRef: unwrappedPopoverRef,
        listBoxRef: listBoxRef,
        inputRef: inputRef,
        menuTrigger: menuTrigger
    }, state);
    let { overlayProps: overlayProps , placement: placement , updatePosition: updatePosition  } = $dyK7H$reactariaoverlays.useOverlayPosition({
        targetRef: unwrappedButtonRef,
        overlayRef: unwrappedPopoverRef,
        scrollRef: listBoxRef,
        placement: `${direction} end`,
        shouldFlip: shouldFlip,
        isOpen: state.isOpen,
        onClose: state.close
    });
    // Measure the width of the inputfield and the button to inform the width of the menu (below).
    let [menuWidth, setMenuWidth] = $dyK7H$react.useState(null);
    let { scale: scale  } = $dyK7H$reactspectrumprovider.useProvider();
    let onResize = $dyK7H$react.useCallback(()=>{
        if (unwrappedButtonRef.current && inputRef.current) {
            let buttonWidth = unwrappedButtonRef.current.offsetWidth;
            let inputWidth = inputRef.current.offsetWidth;
            setMenuWidth(buttonWidth + inputWidth);
        }
    }, [
        unwrappedButtonRef,
        inputRef,
        setMenuWidth
    ]);
    $dyK7H$reactspectrumutils.useResizeObserver({
        ref: domRef,
        onResize: onResize
    });
    $dyK7H$reactariautils.useLayoutEffect(onResize, [
        scale,
        onResize
    ]);
    // Update position once the ListBox has rendered. This ensures that
    // it flips properly when it doesn't fit in the available space.
    // TODO: add ResizeObserver to useOverlayPosition so we don't need this.
    $dyK7H$reactariautils.useLayoutEffect(()=>{
        if (state.isOpen) requestAnimationFrame(()=>{
            updatePosition();
        });
    }, [
        state.isOpen,
        updatePosition
    ]);
    let style = {
        ...overlayProps.style,
        width: isQuiet ? null : menuWidth,
        minWidth: isQuiet ? `calc(${menuWidth}px + calc(2 * var(--spectrum-dropdown-quiet-offset)))` : menuWidth
    };
    return(/*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement(($parcel$interopDefault($dyK7H$react)).Fragment, null, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactspectrumlabel.Field, {
        ...props,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        labelProps: labelProps,
        ref: domRef
    }, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($3740f8e8ff9aff46$var$ComboBoxInput, {
        ...props,
        isOpen: state.isOpen,
        loadingState: loadingState,
        inputProps: inputProps,
        inputRef: inputRef,
        triggerProps: buttonProps,
        triggerRef: buttonRef
    })), /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactspectrumoverlays.Popover, {
        isOpen: state.isOpen,
        UNSAFE_style: style,
        UNSAFE_className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'spectrum-InputGroup-popover', {
            'spectrum-InputGroup-popover--quiet': isQuiet
        }),
        ref: popoverRef,
        placement: placement,
        hideArrow: true,
        isNonModal: true,
        isDismissable: false
    }, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactspectrumlistbox.ListBoxBase, {
        ...listBoxProps,
        ref: listBoxRef,
        disallowEmptySelection: true,
        autoFocus: state.focusStrategy,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        shouldUseVirtualFocus: true,
        isLoading: loadingState === 'loadingMore',
        onLoadMore: onLoadMore,
        renderEmptyState: ()=>isAsync && /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement("span", {
                className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ddda48ae832cd17d$exports)), 'no-results')
            }, loadingState === 'loading' ? formatMessage('loading') : formatMessage('noResults'))
    }), /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactariaoverlays.DismissButton, {
        onDismiss: ()=>state.close()
    }))));
});
const $3740f8e8ff9aff46$var$ComboBoxInput = /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).forwardRef(function ComboBoxInput(props, ref) {
    let { isQuiet: isQuiet , isDisabled: isDisabled , validationState: validationState , inputProps: inputProps , inputRef: inputRef , triggerProps: triggerProps , triggerRef: triggerRef , autoFocus: autoFocus , style: style , className: className , loadingState: loadingState , isOpen: isOpen , menuTrigger: menuTrigger  } = props;
    let { hoverProps: hoverProps , isHovered: isHovered  } = $dyK7H$reactariainteractions.useHover({
    });
    let formatMessage = $dyK7H$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($44bf33e45cbed0d5$exports)));
    let timeout = $dyK7H$react.useRef(null);
    let [showLoading, setShowLoading] = $dyK7H$react.useState(false);
    let loadingCircle = /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactspectrumprogress.ProgressCircle, {
        "aria-label": formatMessage('loading'),
        size: "S",
        isIndeterminate: true,
        UNSAFE_className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($484f319272acc7c5$exports)), 'spectrum-Textfield-circleLoader', $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'spectrum-InputGroup-input-circleLoader'))
    });
    let isLoading = loadingState === 'loading' || loadingState === 'filtering';
    let inputValue = inputProps.value;
    let lastInputValue = $dyK7H$react.useRef(inputValue);
    $dyK7H$react.useEffect(()=>{
        if (isLoading && !showLoading) {
            if (timeout.current === null) timeout.current = setTimeout(()=>{
                setShowLoading(true);
            }, 500);
            // If user is typing, clear the timer and restart since it is a new request
            if (inputValue !== lastInputValue.current) {
                clearTimeout(timeout.current);
                timeout.current = setTimeout(()=>{
                    setShowLoading(true);
                }, 500);
            }
        } else if (!isLoading) {
            // If loading is no longer happening, clear any timers and hide the loading circle
            setShowLoading(false);
            clearTimeout(timeout.current);
            timeout.current = null;
        }
        lastInputValue.current = inputValue;
    }, [
        isLoading,
        showLoading,
        inputValue
    ]);
    return(/*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactariafocus.FocusRing, {
        within: true,
        isTextInput: true,
        focusClass: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'is-focused'),
        focusRingClass: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement("div", {
        ...hoverProps,
        ref: ref,
        style: style,
        className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'spectrum-InputGroup', {
            'spectrum-InputGroup--quiet': isQuiet,
            'is-disabled': isDisabled,
            'spectrum-InputGroup--invalid': validationState === 'invalid',
            'is-hovered': isHovered
        }, className)
    }, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactspectrumtextfield.TextFieldBase, {
        inputProps: inputProps,
        inputRef: inputRef,
        UNSAFE_className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'spectrum-InputGroup-field'),
        inputClassName: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'spectrum-InputGroup-input'),
        validationIconClassName: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'spectrum-InputGroup-input-validationIcon'),
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        validationState: validationState,
        // loading circle should only be displayed if menu is open, if menuTrigger is "manual", or first time load (to stop circle from showing up when user selects an option)
        // TODO: add special case for completionMode: complete as well
        isLoading: showLoading && (isOpen || menuTrigger === 'manual' || loadingState === 'loading'),
        loadingIndicator: loadingState != null && loadingCircle
    }), /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactariainteractions.PressResponder, {
        preventFocusOnPress: true,
        isPressed: isOpen
    }, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement($dyK7H$reactspectrumbutton.FieldButton, {
        ...triggerProps,
        ref: triggerRef,
        UNSAFE_className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'spectrum-FieldButton'),
        isQuiet: isQuiet,
        validationState: validationState
    }, /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).createElement(($parcel$interopDefault($dyK7H$spectrumiconsuiChevronDownMedium)), {
        UNSAFE_className: $dyK7H$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ae1ecdab5fc4ef1b$exports)), 'spectrum-Dropdown-chevron')
    }))))));
});
/**
 * ComboBoxes combine a text entry with a picker menu, allowing users to filter longer lists to only the selections matching a query.
 */ const $3740f8e8ff9aff46$export$72b9695b8216309a = /*#__PURE__*/ ($parcel$interopDefault($dyK7H$react)).forwardRef($3740f8e8ff9aff46$var$ComboBox);



$parcel$exportWildcard(module.exports, $3740f8e8ff9aff46$exports);
