// @react-spectrum/combobox/dist/module.js
import "./main.css";
import {Item as $3b54a9f3842b03a6$re_export$Item, Section as $3b54a9f3842b03a6$re_export$Section} from "@react-stately/collections";
import $iW1Cw$spectrumiconsuiChevronDownMedium from "@spectrum-icons/ui/ChevronDownMedium";
import {useIsMobileDevice as $iW1Cw$useIsMobileDevice, useUnwrapDOMRef as $iW1Cw$useUnwrapDOMRef, useFocusableRef as $iW1Cw$useFocusableRef, useResizeObserver as $iW1Cw$useResizeObserver, classNames as $iW1Cw$classNames, unwrapDOMRef as $iW1Cw$unwrapDOMRef} from "@react-spectrum/utils";
import {useOverlayPosition as $iW1Cw$useOverlayPosition, DismissButton as $iW1Cw$DismissButton, useOverlayTrigger as $iW1Cw$useOverlayTrigger} from "@react-aria/overlays";
import {Field as $iW1Cw$Field} from "@react-spectrum/label";
import {FieldButton as $iW1Cw$FieldButton, ClearButton as $iW1Cw$ClearButton} from "@react-spectrum/button";
import {FocusRing as $iW1Cw$FocusRing, focusSafely as $iW1Cw$focusSafely, FocusScope as $iW1Cw$FocusScope} from "@react-aria/focus";
import {useListBoxLayout as $iW1Cw$useListBoxLayout, ListBoxBase as $iW1Cw$ListBoxBase} from "@react-spectrum/listbox";
import {Popover as $iW1Cw$Popover, Tray as $iW1Cw$Tray} from "@react-spectrum/overlays";
import {useHover as $iW1Cw$useHover, PressResponder as $iW1Cw$PressResponder, setInteractionModality as $iW1Cw$setInteractionModality} from "@react-aria/interactions";
import {ProgressCircle as $iW1Cw$ProgressCircle} from "@react-spectrum/progress";
import $iW1Cw$react, {useRef as $iW1Cw$useRef, useState as $iW1Cw$useState, useCallback as $iW1Cw$useCallback, useEffect as $iW1Cw$useEffect} from "react";
import {TextFieldBase as $iW1Cw$TextFieldBase} from "@react-spectrum/textfield";
import {useComboBox as $iW1Cw$useComboBox} from "@react-aria/combobox";
import {useComboBoxState as $iW1Cw$useComboBoxState} from "@react-stately/combobox";
import {useMessageFormatter as $iW1Cw$useMessageFormatter, useFilter as $iW1Cw$useFilter} from "@react-aria/i18n";
import {useLayoutEffect as $iW1Cw$useLayoutEffect, mergeProps as $iW1Cw$mergeProps, useId as $iW1Cw$useId} from "@react-aria/utils";
import {useProviderProps as $iW1Cw$useProviderProps, useProvider as $iW1Cw$useProvider} from "@react-spectrum/provider";
import $iW1Cw$spectrumiconsuiAlertMedium from "@spectrum-icons/ui/AlertMedium";
import $iW1Cw$spectrumiconsuiCheckmarkMedium from "@spectrum-icons/ui/CheckmarkMedium";
import {useButton as $iW1Cw$useButton} from "@react-aria/button";
import {useDialog as $iW1Cw$useDialog} from "@react-aria/dialog";
import {useField as $iW1Cw$useField} from "@react-aria/label";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $d40059e4c82e96bb$exports = {};

$parcel$export($d40059e4c82e96bb$exports, "ComboBox", () => $d40059e4c82e96bb$export$72b9695b8216309a);


var $c1270b011cb3d946$exports = {};

$parcel$export($c1270b011cb3d946$exports, "no-results", () => $c1270b011cb3d946$export$5b1939da91820a47, (v) => $c1270b011cb3d946$export$5b1939da91820a47 = v);
$parcel$export($c1270b011cb3d946$exports, "mobile-combobox", () => $c1270b011cb3d946$export$31f2f19e6c50453d, (v) => $c1270b011cb3d946$export$31f2f19e6c50453d = v);
$parcel$export($c1270b011cb3d946$exports, "mobile-input", () => $c1270b011cb3d946$export$bededa1c0ac414c5, (v) => $c1270b011cb3d946$export$bededa1c0ac414c5 = v);
$parcel$export($c1270b011cb3d946$exports, "mobile-value", () => $c1270b011cb3d946$export$e32ca2bb9f795355, (v) => $c1270b011cb3d946$export$e32ca2bb9f795355 = v);
$parcel$export($c1270b011cb3d946$exports, "tray-dialog", () => $c1270b011cb3d946$export$3d4198eea215577a, (v) => $c1270b011cb3d946$export$3d4198eea215577a = v);
$parcel$export($c1270b011cb3d946$exports, "tray-textfield", () => $c1270b011cb3d946$export$73cc97e04e76a6d0, (v) => $c1270b011cb3d946$export$73cc97e04e76a6d0 = v);
$parcel$export($c1270b011cb3d946$exports, "has-label", () => $c1270b011cb3d946$export$b8e014ff9ad5e5e2, (v) => $c1270b011cb3d946$export$b8e014ff9ad5e5e2 = v);
$parcel$export($c1270b011cb3d946$exports, "tray-textfield-input", () => $c1270b011cb3d946$export$feb70b7bc717e2b8, (v) => $c1270b011cb3d946$export$feb70b7bc717e2b8 = v);
$parcel$export($c1270b011cb3d946$exports, "tray-listbox", () => $c1270b011cb3d946$export$7281dc54d38d4bed, (v) => $c1270b011cb3d946$export$7281dc54d38d4bed = v);
var $c1270b011cb3d946$export$5b1939da91820a47;
var $c1270b011cb3d946$export$31f2f19e6c50453d;
var $c1270b011cb3d946$export$bededa1c0ac414c5;
var $c1270b011cb3d946$export$e32ca2bb9f795355;
var $c1270b011cb3d946$export$3d4198eea215577a;
var $c1270b011cb3d946$export$73cc97e04e76a6d0;
var $c1270b011cb3d946$export$b8e014ff9ad5e5e2;
var $c1270b011cb3d946$export$feb70b7bc717e2b8;
var $c1270b011cb3d946$export$7281dc54d38d4bed;
$c1270b011cb3d946$export$5b1939da91820a47 = "no-results_743018";
$c1270b011cb3d946$export$31f2f19e6c50453d = "mobile-combobox_743018";
$c1270b011cb3d946$export$bededa1c0ac414c5 = "mobile-input_743018";
$c1270b011cb3d946$export$e32ca2bb9f795355 = "mobile-value_743018";
$c1270b011cb3d946$export$3d4198eea215577a = "tray-dialog_743018";
$c1270b011cb3d946$export$73cc97e04e76a6d0 = "tray-textfield_743018";
$c1270b011cb3d946$export$b8e014ff9ad5e5e2 = "has-label_743018";
$c1270b011cb3d946$export$feb70b7bc717e2b8 = "tray-textfield-input_743018";
$c1270b011cb3d946$export$7281dc54d38d4bed = "tray-listbox_743018";






var $2c1aea9b85d23d6d$exports = {};
var $a1b84dc412c93cfb$exports = {};
$a1b84dc412c93cfb$exports = JSON.parse("{\"clear\":\"مسح\",\"invalid\":\"(غير صالح)\",\"loading\":\"جارٍ التحميل...\",\"noResults\":\"لا توجد نتائج\"}");


var $248a115f3872c943$exports = {};
$248a115f3872c943$exports = JSON.parse("{\"clear\":\"Изчисти\",\"invalid\":\"(невалиден)\",\"loading\":\"Зареждане...\",\"noResults\":\"Няма резултати\"}");


var $ac429f0ba4469025$exports = {};
$ac429f0ba4469025$exports = JSON.parse("{\"clear\":\"Vymazat\",\"invalid\":\"(neplatné)\",\"loading\":\"Načítání...\",\"noResults\":\"Žádné výsledky\"}");


var $f2b5ffabd5daa3a8$exports = {};
$f2b5ffabd5daa3a8$exports = JSON.parse("{\"clear\":\"Ryd\",\"invalid\":\"(ugyldig)\",\"loading\":\"Indlæser ...\",\"noResults\":\"Ingen resultater\"}");


var $640a9b1caaa73646$exports = {};
$640a9b1caaa73646$exports = JSON.parse("{\"clear\":\"Löschen\",\"invalid\":\"(ungültig)\",\"loading\":\"Wird geladen...\",\"noResults\":\"Keine Ergebnisse\"}");


var $403a96c245298531$exports = {};
$403a96c245298531$exports = JSON.parse("{\"clear\":\"Καθαρισμός\",\"invalid\":\"(δεν ισχύει)\",\"loading\":\"Φόρτωση...\",\"noResults\":\"Χωρίς αποτέλεσμα\"}");


var $405ae5d42193fd14$exports = {};
$405ae5d42193fd14$exports = JSON.parse("{\"loading\":\"Loading...\",\"noResults\":\"No results\",\"clear\":\"Clear\",\"invalid\":\"(invalid)\"}");


var $e3945bfc6943759d$exports = {};
$e3945bfc6943759d$exports = JSON.parse("{\"clear\":\"Borrar\",\"invalid\":\"(no válido)\",\"loading\":\"Cargando...\",\"noResults\":\"Sin resultados\"}");


var $3993f12ef26c9474$exports = {};
$3993f12ef26c9474$exports = JSON.parse("{\"clear\":\"Puhasta\",\"invalid\":\"(kehtetu)\",\"loading\":\"Laadimine...\",\"noResults\":\"Tulemusi pole\"}");


var $ca50ddf2813f5cd8$exports = {};
$ca50ddf2813f5cd8$exports = JSON.parse("{\"clear\":\"Kirkas\",\"invalid\":\"(epäkelpo)\",\"loading\":\"Ladataan...\",\"noResults\":\"Ei tuloksia\"}");


var $9236e8fe41ef0673$exports = {};
$9236e8fe41ef0673$exports = JSON.parse("{\"clear\":\"Effacer\",\"invalid\":\"(non valide)\",\"loading\":\"Chargement en cours...\",\"noResults\":\"Aucun résultat\"}");


var $8983b35107eb257c$exports = {};
$8983b35107eb257c$exports = JSON.parse("{\"clear\":\"נקי\",\"invalid\":\"(לא חוקי)\",\"loading\":\"טוען...\",\"noResults\":\"אין תוצאות\"}");


var $e2b56791521aa0e6$exports = {};
$e2b56791521aa0e6$exports = JSON.parse("{\"clear\":\"Izbriši\",\"invalid\":\"(nevažeće)\",\"loading\":\"Učitavam...\",\"noResults\":\"Nema rezultata\"}");


var $c256b29a68fc228b$exports = {};
$c256b29a68fc228b$exports = JSON.parse("{\"clear\":\"Törlés\",\"invalid\":\"(érvénytelen)\",\"loading\":\"Betöltés folyamatban…\",\"noResults\":\"Nincsenek találatok\"}");


var $3bc19f29516771c4$exports = {};
$3bc19f29516771c4$exports = JSON.parse("{\"clear\":\"Cancella\",\"invalid\":\"(non valido)\",\"loading\":\"Caricamento in corso...\",\"noResults\":\"Nessun risultato\"}");


var $ef5f999e5e128dd4$exports = {};
$ef5f999e5e128dd4$exports = JSON.parse("{\"clear\":\"クリア\",\"invalid\":\"(無効)\",\"loading\":\"読み込み中...\",\"noResults\":\"結果なし\"}");


var $19f4133762dbcbe4$exports = {};
$19f4133762dbcbe4$exports = JSON.parse("{\"clear\":\"지우기\",\"invalid\":\"(유효하지 않음)\",\"loading\":\"로드 중...\",\"noResults\":\"결과 없음\"}");


var $76d35421939e9654$exports = {};
$76d35421939e9654$exports = JSON.parse("{\"clear\":\"Skaidrus\",\"invalid\":\"(netinkama)\",\"loading\":\"Įkeliama...\",\"noResults\":\"Be rezultatų\"}");


var $fd9ee26bf766dac3$exports = {};
$fd9ee26bf766dac3$exports = JSON.parse("{\"clear\":\"Notīrīt\",\"invalid\":\"(nederīgs)\",\"loading\":\"Notiek ielāde...\",\"noResults\":\"Nav rezultātu\"}");


var $7c5b99c47c9393e9$exports = {};
$7c5b99c47c9393e9$exports = JSON.parse("{\"clear\":\"Tøm\",\"invalid\":\"(ugyldig)\",\"loading\":\"Laster inn ...\",\"noResults\":\"Ingen resultater\"}");


var $0f549f14ad9e7292$exports = {};
$0f549f14ad9e7292$exports = JSON.parse("{\"clear\":\"Helder\",\"invalid\":\"(ongeldig)\",\"loading\":\"Laden...\",\"noResults\":\"Geen resultaten\"}");


var $3a25c20901104098$exports = {};
$3a25c20901104098$exports = JSON.parse("{\"clear\":\"Wyczyść\",\"invalid\":\"(nieprawidłowy)\",\"loading\":\"Trwa ładowanie...\",\"noResults\":\"Brak wyników\"}");


var $34766ceb8f70244f$exports = {};
$34766ceb8f70244f$exports = JSON.parse("{\"clear\":\"Limpar\",\"invalid\":\"(inválido)\",\"loading\":\"Carregando...\",\"noResults\":\"Nenhum resultado\"}");


var $b43a5032fd244fa5$exports = {};
$b43a5032fd244fa5$exports = JSON.parse("{\"clear\":\"Limpar\",\"invalid\":\"(inválido)\",\"loading\":\"A carregar...\",\"noResults\":\"Sem resultados\"}");


var $97238f69e71f5072$exports = {};
$97238f69e71f5072$exports = JSON.parse("{\"clear\":\"Golire\",\"invalid\":\"(nevalid)\",\"loading\":\"Se încarcă...\",\"noResults\":\"Niciun rezultat\"}");


var $3791282942972e11$exports = {};
$3791282942972e11$exports = JSON.parse("{\"clear\":\"Очистить\",\"invalid\":\"(недействительно)\",\"loading\":\"Загрузка...\",\"noResults\":\"Результаты отсутствуют\"}");


var $bfc7cf1290f75031$exports = {};
$bfc7cf1290f75031$exports = JSON.parse("{\"clear\":\"Vymazať\",\"invalid\":\"(neplatné)\",\"loading\":\"Načítava sa...\",\"noResults\":\"Žiadne výsledky\"}");


var $19d0942ecd586976$exports = {};
$19d0942ecd586976$exports = JSON.parse("{\"clear\":\"Jasen\",\"invalid\":\"(neveljavno)\",\"loading\":\"Nalaganje...\",\"noResults\":\"Ni rezultatov\"}");


var $f96d5f117b23caa1$exports = {};
$f96d5f117b23caa1$exports = JSON.parse("{\"clear\":\"Izbriši\",\"invalid\":\"(nevažeće)\",\"loading\":\"Učitavam...\",\"noResults\":\"Nema rezultata\"}");


var $743522e2a45ae607$exports = {};
$743522e2a45ae607$exports = JSON.parse("{\"clear\":\"Rensa\",\"invalid\":\"(ogiltigt)\",\"loading\":\"Läser in...\",\"noResults\":\"Inga resultat\"}");


var $ab288d2b57272403$exports = {};
$ab288d2b57272403$exports = JSON.parse("{\"clear\":\"Temizle\",\"invalid\":\"(geçersiz)\",\"loading\":\"Yükleniyor...\",\"noResults\":\"Sonuç yok\"}");


var $479501acf73d9bc0$exports = {};
$479501acf73d9bc0$exports = JSON.parse("{\"clear\":\"Очистити\",\"invalid\":\"(недійсне)\",\"loading\":\"Завантаження...\",\"noResults\":\"Результатів немає\"}");


var $2bb835bdaea0c448$exports = {};
$2bb835bdaea0c448$exports = JSON.parse("{\"clear\":\"透明\",\"invalid\":\"（无效）\",\"loading\":\"正在加载...\",\"noResults\":\"无结果\"}");


var $39db552d57a27d18$exports = {};
$39db552d57a27d18$exports = JSON.parse("{\"clear\":\"清除\",\"invalid\":\"(無效)\",\"loading\":\"正在載入...\",\"noResults\":\"無任何結果\"}");


$2c1aea9b85d23d6d$exports = {
    "ar-AE": $a1b84dc412c93cfb$exports,
    "bg-BG": $248a115f3872c943$exports,
    "cs-CZ": $ac429f0ba4469025$exports,
    "da-DK": $f2b5ffabd5daa3a8$exports,
    "de-DE": $640a9b1caaa73646$exports,
    "el-GR": $403a96c245298531$exports,
    "en-US": $405ae5d42193fd14$exports,
    "es-ES": $e3945bfc6943759d$exports,
    "et-EE": $3993f12ef26c9474$exports,
    "fi-FI": $ca50ddf2813f5cd8$exports,
    "fr-FR": $9236e8fe41ef0673$exports,
    "he-IL": $8983b35107eb257c$exports,
    "hr-HR": $e2b56791521aa0e6$exports,
    "hu-HU": $c256b29a68fc228b$exports,
    "it-IT": $3bc19f29516771c4$exports,
    "ja-JP": $ef5f999e5e128dd4$exports,
    "ko-KR": $19f4133762dbcbe4$exports,
    "lt-LT": $76d35421939e9654$exports,
    "lv-LV": $fd9ee26bf766dac3$exports,
    "nb-NO": $7c5b99c47c9393e9$exports,
    "nl-NL": $0f549f14ad9e7292$exports,
    "pl-PL": $3a25c20901104098$exports,
    "pt-BR": $34766ceb8f70244f$exports,
    "pt-PT": $b43a5032fd244fa5$exports,
    "ro-RO": $97238f69e71f5072$exports,
    "ru-RU": $3791282942972e11$exports,
    "sk-SK": $bfc7cf1290f75031$exports,
    "sl-SI": $19d0942ecd586976$exports,
    "sr-SP": $f96d5f117b23caa1$exports,
    "sv-SE": $743522e2a45ae607$exports,
    "tr-TR": $ab288d2b57272403$exports,
    "uk-UA": $479501acf73d9bc0$exports,
    "zh-CN": $2bb835bdaea0c448$exports,
    "zh-TW": $39db552d57a27d18$exports
};




var $8682ec7d9494680f$exports = {};

$parcel$export($8682ec7d9494680f$exports, "spectrum-Button", () => $8682ec7d9494680f$export$1db4cca5b4ade39a, (v) => $8682ec7d9494680f$export$1db4cca5b4ade39a = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton", () => $8682ec7d9494680f$export$53da69f51b770d3, (v) => $8682ec7d9494680f$export$53da69f51b770d3 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-LogicButton", () => $8682ec7d9494680f$export$36ae67a131730482, (v) => $8682ec7d9494680f$export$36ae67a131730482 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-FieldButton", () => $8682ec7d9494680f$export$a024c99ecf6b5741, (v) => $8682ec7d9494680f$export$a024c99ecf6b5741 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ClearButton", () => $8682ec7d9494680f$export$b752ce409e5660c, (v) => $8682ec7d9494680f$export$b752ce409e5660c = v);
$parcel$export($8682ec7d9494680f$exports, "focus-ring", () => $8682ec7d9494680f$export$f39a09f249340e2a, (v) => $8682ec7d9494680f$export$f39a09f249340e2a = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Icon", () => $8682ec7d9494680f$export$d374b04f30360026, (v) => $8682ec7d9494680f$export$d374b04f30360026 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton--emphasized", () => $8682ec7d9494680f$export$5aabb61a966e266f, (v) => $8682ec7d9494680f$export$5aabb61a966e266f = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton--staticColor", () => $8682ec7d9494680f$export$29b7b5d6a5a497c5, (v) => $8682ec7d9494680f$export$29b7b5d6a5a497c5 = v);
$parcel$export($8682ec7d9494680f$exports, "is-hovered", () => $8682ec7d9494680f$export$b8813cd5d7824ce7, (v) => $8682ec7d9494680f$export$b8813cd5d7824ce7 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button-label", () => $8682ec7d9494680f$export$c081ba562e204942, (v) => $8682ec7d9494680f$export$c081ba562e204942 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton-label", () => $8682ec7d9494680f$export$ebb4635c66bacdb9, (v) => $8682ec7d9494680f$export$ebb4635c66bacdb9 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton-hold", () => $8682ec7d9494680f$export$4a4f01c443eff992, (v) => $8682ec7d9494680f$export$4a4f01c443eff992 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton--quiet", () => $8682ec7d9494680f$export$40e1f009544bacf3, (v) => $8682ec7d9494680f$export$40e1f009544bacf3 = v);
$parcel$export($8682ec7d9494680f$exports, "is-disabled", () => $8682ec7d9494680f$export$d35bc1e505d1ebbf, (v) => $8682ec7d9494680f$export$d35bc1e505d1ebbf = v);
$parcel$export($8682ec7d9494680f$exports, "is-open", () => $8682ec7d9494680f$export$a9781837241c946d, (v) => $8682ec7d9494680f$export$a9781837241c946d = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-FieldButton--quiet", () => $8682ec7d9494680f$export$5ca7b274e4fff9fc, (v) => $8682ec7d9494680f$export$5ca7b274e4fff9fc = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ClearButton--overBackground", () => $8682ec7d9494680f$export$116f2fccf941d34d, (v) => $8682ec7d9494680f$export$116f2fccf941d34d = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ClearButton--small", () => $8682ec7d9494680f$export$49280ffa5236ba8b, (v) => $8682ec7d9494680f$export$49280ffa5236ba8b = v);
$parcel$export($8682ec7d9494680f$exports, "is-focused", () => $8682ec7d9494680f$export$e7dc768d35940237, (v) => $8682ec7d9494680f$export$e7dc768d35940237 = v);
$parcel$export($8682ec7d9494680f$exports, "is-active", () => $8682ec7d9494680f$export$20fd0f7cd4e6112f, (v) => $8682ec7d9494680f$export$20fd0f7cd4e6112f = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--cta", () => $8682ec7d9494680f$export$b818fdf0052a2cab, (v) => $8682ec7d9494680f$export$b818fdf0052a2cab = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--primary", () => $8682ec7d9494680f$export$c1c3511f5e96e926, (v) => $8682ec7d9494680f$export$c1c3511f5e96e926 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--secondary", () => $8682ec7d9494680f$export$4f1b75526afd8ff3, (v) => $8682ec7d9494680f$export$4f1b75526afd8ff3 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--warning", () => $8682ec7d9494680f$export$6f97cef817ff1cac, (v) => $8682ec7d9494680f$export$6f97cef817ff1cac = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--overBackground", () => $8682ec7d9494680f$export$5cda1128f9112d2e, (v) => $8682ec7d9494680f$export$5cda1128f9112d2e = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--quiet", () => $8682ec7d9494680f$export$25f049c049371040, (v) => $8682ec7d9494680f$export$25f049c049371040 = v);
$parcel$export($8682ec7d9494680f$exports, "is-selected", () => $8682ec7d9494680f$export$1e0fb04f31d3c22a, (v) => $8682ec7d9494680f$export$1e0fb04f31d3c22a = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton--staticWhite", () => $8682ec7d9494680f$export$ccc536e1adbdc059, (v) => $8682ec7d9494680f$export$ccc536e1adbdc059 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton--staticBlack", () => $8682ec7d9494680f$export$8f7c9db66978d20f, (v) => $8682ec7d9494680f$export$8f7c9db66978d20f = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-LogicButton--and", () => $8682ec7d9494680f$export$e52daab43a62c528, (v) => $8682ec7d9494680f$export$e52daab43a62c528 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-LogicButton--or", () => $8682ec7d9494680f$export$da76be6126aaa633, (v) => $8682ec7d9494680f$export$da76be6126aaa633 = v);
$parcel$export($8682ec7d9494680f$exports, "is-placeholder", () => $8682ec7d9494680f$export$e5b2f5233e4e5194, (v) => $8682ec7d9494680f$export$e5b2f5233e4e5194 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-FieldButton--invalid", () => $8682ec7d9494680f$export$c9f503f672e8a3c1, (v) => $8682ec7d9494680f$export$c9f503f672e8a3c1 = v);
var $8682ec7d9494680f$export$1db4cca5b4ade39a;
var $8682ec7d9494680f$export$53da69f51b770d3;
var $8682ec7d9494680f$export$36ae67a131730482;
var $8682ec7d9494680f$export$a024c99ecf6b5741;
var $8682ec7d9494680f$export$b752ce409e5660c;
var $8682ec7d9494680f$export$f39a09f249340e2a;
var $8682ec7d9494680f$export$d374b04f30360026;
var $8682ec7d9494680f$export$5aabb61a966e266f;
var $8682ec7d9494680f$export$29b7b5d6a5a497c5;
var $8682ec7d9494680f$export$b8813cd5d7824ce7;
var $8682ec7d9494680f$export$c081ba562e204942;
var $8682ec7d9494680f$export$ebb4635c66bacdb9;
var $8682ec7d9494680f$export$4a4f01c443eff992;
var $8682ec7d9494680f$export$40e1f009544bacf3;
var $8682ec7d9494680f$export$d35bc1e505d1ebbf;
var $8682ec7d9494680f$export$a9781837241c946d;
var $8682ec7d9494680f$export$5ca7b274e4fff9fc;
var $8682ec7d9494680f$export$116f2fccf941d34d;
var $8682ec7d9494680f$export$49280ffa5236ba8b;
var $8682ec7d9494680f$export$e7dc768d35940237;
var $8682ec7d9494680f$export$20fd0f7cd4e6112f;
var $8682ec7d9494680f$export$b818fdf0052a2cab;
var $8682ec7d9494680f$export$c1c3511f5e96e926;
var $8682ec7d9494680f$export$4f1b75526afd8ff3;
var $8682ec7d9494680f$export$6f97cef817ff1cac;
var $8682ec7d9494680f$export$5cda1128f9112d2e;
var $8682ec7d9494680f$export$25f049c049371040;
var $8682ec7d9494680f$export$1e0fb04f31d3c22a;
var $8682ec7d9494680f$export$ccc536e1adbdc059;
var $8682ec7d9494680f$export$8f7c9db66978d20f;
var $8682ec7d9494680f$export$e52daab43a62c528;
var $8682ec7d9494680f$export$da76be6126aaa633;
var $8682ec7d9494680f$export$e5b2f5233e4e5194;
var $8682ec7d9494680f$export$c9f503f672e8a3c1;
$8682ec7d9494680f$export$1db4cca5b4ade39a = "spectrum-Button_e2d99e";
$8682ec7d9494680f$export$53da69f51b770d3 = "spectrum-ActionButton_e2d99e";
$8682ec7d9494680f$export$36ae67a131730482 = "spectrum-LogicButton_e2d99e";
$8682ec7d9494680f$export$a024c99ecf6b5741 = "spectrum-FieldButton_e2d99e";
$8682ec7d9494680f$export$b752ce409e5660c = "spectrum-ClearButton_e2d99e";
$8682ec7d9494680f$export$f39a09f249340e2a = "focus-ring_e2d99e";
$8682ec7d9494680f$export$d374b04f30360026 = "spectrum-Icon_e2d99e";
$8682ec7d9494680f$export$5aabb61a966e266f = "spectrum-ActionButton--emphasized_e2d99e";
$8682ec7d9494680f$export$29b7b5d6a5a497c5 = "spectrum-ActionButton--staticColor_e2d99e";
$8682ec7d9494680f$export$b8813cd5d7824ce7 = "is-hovered_e2d99e";
$8682ec7d9494680f$export$c081ba562e204942 = "spectrum-Button-label_e2d99e";
$8682ec7d9494680f$export$ebb4635c66bacdb9 = "spectrum-ActionButton-label_e2d99e";
$8682ec7d9494680f$export$4a4f01c443eff992 = "spectrum-ActionButton-hold_e2d99e";
$8682ec7d9494680f$export$40e1f009544bacf3 = "spectrum-ActionButton--quiet_e2d99e";
$8682ec7d9494680f$export$d35bc1e505d1ebbf = "is-disabled_e2d99e";
$8682ec7d9494680f$export$a9781837241c946d = "is-open_e2d99e";
$8682ec7d9494680f$export$5ca7b274e4fff9fc = "spectrum-FieldButton--quiet_e2d99e";
$8682ec7d9494680f$export$116f2fccf941d34d = "spectrum-ClearButton--overBackground_e2d99e";
$8682ec7d9494680f$export$49280ffa5236ba8b = "spectrum-ClearButton--small_e2d99e";
$8682ec7d9494680f$export$e7dc768d35940237 = "is-focused_e2d99e";
$8682ec7d9494680f$export$20fd0f7cd4e6112f = "is-active_e2d99e";
$8682ec7d9494680f$export$b818fdf0052a2cab = "spectrum-Button--cta_e2d99e";
$8682ec7d9494680f$export$c1c3511f5e96e926 = "spectrum-Button--primary_e2d99e";
$8682ec7d9494680f$export$4f1b75526afd8ff3 = "spectrum-Button--secondary_e2d99e";
$8682ec7d9494680f$export$6f97cef817ff1cac = "spectrum-Button--warning_e2d99e";
$8682ec7d9494680f$export$5cda1128f9112d2e = "spectrum-Button--overBackground_e2d99e";
$8682ec7d9494680f$export$25f049c049371040 = "spectrum-Button--quiet_e2d99e";
$8682ec7d9494680f$export$1e0fb04f31d3c22a = "is-selected_e2d99e";
$8682ec7d9494680f$export$ccc536e1adbdc059 = "spectrum-ActionButton--staticWhite_e2d99e";
$8682ec7d9494680f$export$8f7c9db66978d20f = "spectrum-ActionButton--staticBlack_e2d99e";
$8682ec7d9494680f$export$e52daab43a62c528 = "spectrum-LogicButton--and_e2d99e";
$8682ec7d9494680f$export$da76be6126aaa633 = "spectrum-LogicButton--or_e2d99e";
$8682ec7d9494680f$export$e5b2f5233e4e5194 = "is-placeholder_e2d99e";
$8682ec7d9494680f$export$c9f503f672e8a3c1 = "spectrum-FieldButton--invalid_e2d99e";













var $4a84444d5a56ab41$exports = {};

$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel", () => $4a84444d5a56ab41$export$493d18e796ae054, (v) => $4a84444d5a56ab41$export$493d18e796ae054 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel--positionSide", () => $4a84444d5a56ab41$export$9e6f19dc21f22f2e, (v) => $4a84444d5a56ab41$export$9e6f19dc21f22f2e = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel-requiredIcon", () => $4a84444d5a56ab41$export$7c47642c7d46f3c9, (v) => $4a84444d5a56ab41$export$7c47642c7d46f3c9 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel--alignEnd", () => $4a84444d5a56ab41$export$885efcc08143a987, (v) => $4a84444d5a56ab41$export$885efcc08143a987 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field", () => $4a84444d5a56ab41$export$f6d480ae1e56eba0, (v) => $4a84444d5a56ab41$export$f6d480ae1e56eba0 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field--positionTop", () => $4a84444d5a56ab41$export$a4ea780a9064d7f9, (v) => $4a84444d5a56ab41$export$a4ea780a9064d7f9 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field-field", () => $4a84444d5a56ab41$export$3ab8a3dc7f0563d, (v) => $4a84444d5a56ab41$export$3ab8a3dc7f0563d = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field-field--multiline", () => $4a84444d5a56ab41$export$f58a1e966a92ba5e, (v) => $4a84444d5a56ab41$export$f58a1e966a92ba5e = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field--positionSide", () => $4a84444d5a56ab41$export$2963225c91179589, (v) => $4a84444d5a56ab41$export$2963225c91179589 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field-wrapper", () => $4a84444d5a56ab41$export$127df0b7290eb8ef, (v) => $4a84444d5a56ab41$export$127df0b7290eb8ef = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form", () => $4a84444d5a56ab41$export$94d2bcc94afabd89, (v) => $4a84444d5a56ab41$export$94d2bcc94afabd89 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form--positionSide", () => $4a84444d5a56ab41$export$7f44db659563c8f4, (v) => $4a84444d5a56ab41$export$7f44db659563c8f4 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form--positionTop", () => $4a84444d5a56ab41$export$11b8bc81fe551057, (v) => $4a84444d5a56ab41$export$11b8bc81fe551057 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form-itemLabel", () => $4a84444d5a56ab41$export$51ec27881a1e3928, (v) => $4a84444d5a56ab41$export$51ec27881a1e3928 = v);
$parcel$export($4a84444d5a56ab41$exports, "is-disabled", () => $4a84444d5a56ab41$export$d35bc1e505d1ebbf, (v) => $4a84444d5a56ab41$export$d35bc1e505d1ebbf = v);
var $4a84444d5a56ab41$export$493d18e796ae054;
var $4a84444d5a56ab41$export$9e6f19dc21f22f2e;
var $4a84444d5a56ab41$export$7c47642c7d46f3c9;
var $4a84444d5a56ab41$export$885efcc08143a987;
var $4a84444d5a56ab41$export$f6d480ae1e56eba0;
var $4a84444d5a56ab41$export$a4ea780a9064d7f9;
var $4a84444d5a56ab41$export$3ab8a3dc7f0563d;
var $4a84444d5a56ab41$export$f58a1e966a92ba5e;
var $4a84444d5a56ab41$export$2963225c91179589;
var $4a84444d5a56ab41$export$127df0b7290eb8ef;
var $4a84444d5a56ab41$export$94d2bcc94afabd89;
var $4a84444d5a56ab41$export$7f44db659563c8f4;
var $4a84444d5a56ab41$export$11b8bc81fe551057;
var $4a84444d5a56ab41$export$51ec27881a1e3928;
var $4a84444d5a56ab41$export$d35bc1e505d1ebbf;
$4a84444d5a56ab41$export$493d18e796ae054 = "spectrum-FieldLabel_d2db1f";
$4a84444d5a56ab41$export$9e6f19dc21f22f2e = "spectrum-FieldLabel--positionSide_d2db1f";
$4a84444d5a56ab41$export$7c47642c7d46f3c9 = "spectrum-FieldLabel-requiredIcon_d2db1f";
$4a84444d5a56ab41$export$885efcc08143a987 = "spectrum-FieldLabel--alignEnd_d2db1f";
$4a84444d5a56ab41$export$f6d480ae1e56eba0 = "spectrum-Field_d2db1f";
$4a84444d5a56ab41$export$a4ea780a9064d7f9 = "spectrum-Field--positionTop_d2db1f";
$4a84444d5a56ab41$export$3ab8a3dc7f0563d = "spectrum-Field-field_d2db1f";
$4a84444d5a56ab41$export$f58a1e966a92ba5e = "spectrum-Field-field--multiline_d2db1f";
$4a84444d5a56ab41$export$2963225c91179589 = "spectrum-Field--positionSide_d2db1f";
$4a84444d5a56ab41$export$127df0b7290eb8ef = "spectrum-Field-wrapper_d2db1f";
$4a84444d5a56ab41$export$94d2bcc94afabd89 = "spectrum-Form_d2db1f";
$4a84444d5a56ab41$export$7f44db659563c8f4 = "spectrum-Form--positionSide_d2db1f";
$4a84444d5a56ab41$export$11b8bc81fe551057 = "spectrum-Form--positionTop_d2db1f";
$4a84444d5a56ab41$export$51ec27881a1e3928 = "spectrum-Form-itemLabel_d2db1f";
$4a84444d5a56ab41$export$d35bc1e505d1ebbf = "is-disabled_d2db1f";






var $fe88933789dbeaab$exports = {};

$parcel$export($fe88933789dbeaab$exports, "spectrum-Search", () => $fe88933789dbeaab$export$713911d754e1e4ae, (v) => $fe88933789dbeaab$export$713911d754e1e4ae = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-ClearButton", () => $fe88933789dbeaab$export$b752ce409e5660c, (v) => $fe88933789dbeaab$export$b752ce409e5660c = v);
$parcel$export($fe88933789dbeaab$exports, "is-quiet", () => $fe88933789dbeaab$export$6d5c5433f0da4241, (v) => $fe88933789dbeaab$export$6d5c5433f0da4241 = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-Search-input", () => $fe88933789dbeaab$export$50e02603f2d97ae1, (v) => $fe88933789dbeaab$export$50e02603f2d97ae1 = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-Search--invalid", () => $fe88933789dbeaab$export$77a6151a0b46a6a3, (v) => $fe88933789dbeaab$export$77a6151a0b46a6a3 = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-Search--valid", () => $fe88933789dbeaab$export$aa81e3ab40f37567, (v) => $fe88933789dbeaab$export$aa81e3ab40f37567 = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-Search--loadable", () => $fe88933789dbeaab$export$9c74d64afae3b2d4, (v) => $fe88933789dbeaab$export$9c74d64afae3b2d4 = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-Search-circleLoader", () => $fe88933789dbeaab$export$70ae4353602451dc, (v) => $fe88933789dbeaab$export$70ae4353602451dc = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-Search-validationIcon", () => $fe88933789dbeaab$export$938f0981825eae6e, (v) => $fe88933789dbeaab$export$938f0981825eae6e = v);
$parcel$export($fe88933789dbeaab$exports, "spectrum-Textfield", () => $fe88933789dbeaab$export$2c8a10299d8b3418, (v) => $fe88933789dbeaab$export$2c8a10299d8b3418 = v);
var $fe88933789dbeaab$export$713911d754e1e4ae;
var $fe88933789dbeaab$export$b752ce409e5660c;
var $fe88933789dbeaab$export$6d5c5433f0da4241;
var $fe88933789dbeaab$export$50e02603f2d97ae1;
var $fe88933789dbeaab$export$77a6151a0b46a6a3;
var $fe88933789dbeaab$export$aa81e3ab40f37567;
var $fe88933789dbeaab$export$9c74d64afae3b2d4;
var $fe88933789dbeaab$export$70ae4353602451dc;
var $fe88933789dbeaab$export$938f0981825eae6e;
var $fe88933789dbeaab$export$2c8a10299d8b3418;
$fe88933789dbeaab$export$713911d754e1e4ae = "spectrum-Search_d3c0f3";
$fe88933789dbeaab$export$b752ce409e5660c = "spectrum-ClearButton_d3c0f3";
$fe88933789dbeaab$export$6d5c5433f0da4241 = "is-quiet_d3c0f3";
$fe88933789dbeaab$export$50e02603f2d97ae1 = "spectrum-Search-input_d3c0f3";
$fe88933789dbeaab$export$77a6151a0b46a6a3 = "spectrum-Search--invalid_d3c0f3";
$fe88933789dbeaab$export$aa81e3ab40f37567 = "spectrum-Search--valid_d3c0f3";
$fe88933789dbeaab$export$9c74d64afae3b2d4 = "spectrum-Search--loadable_d3c0f3";
$fe88933789dbeaab$export$70ae4353602451dc = "spectrum-Search-circleLoader_d3c0f3";
$fe88933789dbeaab$export$938f0981825eae6e = "spectrum-Search-validationIcon_d3c0f3";
$fe88933789dbeaab$export$2c8a10299d8b3418 = "spectrum-Textfield_d3c0f3";



var $a551245eb06c6862$exports = {};

$parcel$export($a551245eb06c6862$exports, "spectrum-InputGroup", () => $a551245eb06c6862$export$5be359c1227eedcb, (v) => $a551245eb06c6862$export$5be359c1227eedcb = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-FieldButton", () => $a551245eb06c6862$export$a024c99ecf6b5741, (v) => $a551245eb06c6862$export$a024c99ecf6b5741 = v);
$parcel$export($a551245eb06c6862$exports, "is-disabled", () => $a551245eb06c6862$export$d35bc1e505d1ebbf, (v) => $a551245eb06c6862$export$d35bc1e505d1ebbf = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-InputGroup--invalid", () => $a551245eb06c6862$export$3b48509739a9cda0, (v) => $a551245eb06c6862$export$3b48509739a9cda0 = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-InputGroup--quiet", () => $a551245eb06c6862$export$ebf265ffc062029b, (v) => $a551245eb06c6862$export$ebf265ffc062029b = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-InputGroup-field", () => $a551245eb06c6862$export$ea77459595e26bec, (v) => $a551245eb06c6862$export$ea77459595e26bec = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-InputGroup-input-validationIcon", () => $a551245eb06c6862$export$f7ab13ee113f4884, (v) => $a551245eb06c6862$export$f7ab13ee113f4884 = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-InputGroup-input-circleLoader", () => $a551245eb06c6862$export$dcc9e00cdad967fa, (v) => $a551245eb06c6862$export$dcc9e00cdad967fa = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-InputGroup-input", () => $a551245eb06c6862$export$92706f5c6fe4074, (v) => $a551245eb06c6862$export$92706f5c6fe4074 = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-InputGroup-icon", () => $a551245eb06c6862$export$b91b6af799fd93a8, (v) => $a551245eb06c6862$export$b91b6af799fd93a8 = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-InputGroup-popover--quiet", () => $a551245eb06c6862$export$5e116178f9b1556a, (v) => $a551245eb06c6862$export$5e116178f9b1556a = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-Datepicker--range", () => $a551245eb06c6862$export$1d8b0b29e11d7cd, (v) => $a551245eb06c6862$export$1d8b0b29e11d7cd = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-Datepicker--datetimeRange", () => $a551245eb06c6862$export$58dd7cd23c92699f, (v) => $a551245eb06c6862$export$58dd7cd23c92699f = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-Datepicker-startField", () => $a551245eb06c6862$export$80f9c838a8406469, (v) => $a551245eb06c6862$export$80f9c838a8406469 = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-Datepicker-endField", () => $a551245eb06c6862$export$1c72fb41fa4f21ee, (v) => $a551245eb06c6862$export$1c72fb41fa4f21ee = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-Datepicker--rangeDash", () => $a551245eb06c6862$export$202113e883898dc, (v) => $a551245eb06c6862$export$202113e883898dc = v);
$parcel$export($a551245eb06c6862$exports, "is-focused", () => $a551245eb06c6862$export$e7dc768d35940237, (v) => $a551245eb06c6862$export$e7dc768d35940237 = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-Datepicker-focusRing", () => $a551245eb06c6862$export$eb262b15b9261fde, (v) => $a551245eb06c6862$export$eb262b15b9261fde = v);
$parcel$export($a551245eb06c6862$exports, "is-hovered", () => $a551245eb06c6862$export$b8813cd5d7824ce7, (v) => $a551245eb06c6862$export$b8813cd5d7824ce7 = v);
$parcel$export($a551245eb06c6862$exports, "focus-ring", () => $a551245eb06c6862$export$f39a09f249340e2a, (v) => $a551245eb06c6862$export$f39a09f249340e2a = v);
$parcel$export($a551245eb06c6862$exports, "is-selected", () => $a551245eb06c6862$export$1e0fb04f31d3c22a, (v) => $a551245eb06c6862$export$1e0fb04f31d3c22a = v);
$parcel$export($a551245eb06c6862$exports, "spectrum-FieldButton--invalid", () => $a551245eb06c6862$export$c9f503f672e8a3c1, (v) => $a551245eb06c6862$export$c9f503f672e8a3c1 = v);
var $a551245eb06c6862$export$5be359c1227eedcb;
var $a551245eb06c6862$export$a024c99ecf6b5741;
var $a551245eb06c6862$export$d35bc1e505d1ebbf;
var $a551245eb06c6862$export$3b48509739a9cda0;
var $a551245eb06c6862$export$ebf265ffc062029b;
var $a551245eb06c6862$export$ea77459595e26bec;
var $a551245eb06c6862$export$f7ab13ee113f4884;
var $a551245eb06c6862$export$dcc9e00cdad967fa;
var $a551245eb06c6862$export$92706f5c6fe4074;
var $a551245eb06c6862$export$b91b6af799fd93a8;
var $a551245eb06c6862$export$5e116178f9b1556a;
var $a551245eb06c6862$export$1d8b0b29e11d7cd;
var $a551245eb06c6862$export$58dd7cd23c92699f;
var $a551245eb06c6862$export$80f9c838a8406469;
var $a551245eb06c6862$export$1c72fb41fa4f21ee;
var $a551245eb06c6862$export$202113e883898dc;
var $a551245eb06c6862$export$e7dc768d35940237;
var $a551245eb06c6862$export$eb262b15b9261fde;
var $a551245eb06c6862$export$b8813cd5d7824ce7;
var $a551245eb06c6862$export$f39a09f249340e2a;
var $a551245eb06c6862$export$1e0fb04f31d3c22a;
var $a551245eb06c6862$export$c9f503f672e8a3c1;
$a551245eb06c6862$export$5be359c1227eedcb = "spectrum-InputGroup_a0942f";
$a551245eb06c6862$export$a024c99ecf6b5741 = "spectrum-FieldButton_a0942f";
$a551245eb06c6862$export$d35bc1e505d1ebbf = "is-disabled_a0942f";
$a551245eb06c6862$export$3b48509739a9cda0 = "spectrum-InputGroup--invalid_a0942f";
$a551245eb06c6862$export$ebf265ffc062029b = "spectrum-InputGroup--quiet_a0942f";
$a551245eb06c6862$export$ea77459595e26bec = "spectrum-InputGroup-field_a0942f";
$a551245eb06c6862$export$f7ab13ee113f4884 = "spectrum-InputGroup-input-validationIcon_a0942f";
$a551245eb06c6862$export$dcc9e00cdad967fa = "spectrum-InputGroup-input-circleLoader_a0942f";
$a551245eb06c6862$export$92706f5c6fe4074 = "spectrum-InputGroup-input_a0942f";
$a551245eb06c6862$export$b91b6af799fd93a8 = "spectrum-InputGroup-icon_a0942f";
$a551245eb06c6862$export$5e116178f9b1556a = "spectrum-InputGroup-popover--quiet_a0942f";
$a551245eb06c6862$export$1d8b0b29e11d7cd = "spectrum-Datepicker--range_a0942f";
$a551245eb06c6862$export$58dd7cd23c92699f = "spectrum-Datepicker--datetimeRange_a0942f";
$a551245eb06c6862$export$80f9c838a8406469 = "spectrum-Datepicker-startField_a0942f";
$a551245eb06c6862$export$1c72fb41fa4f21ee = "spectrum-Datepicker-endField_a0942f";
$a551245eb06c6862$export$202113e883898dc = "spectrum-Datepicker--rangeDash_a0942f";
$a551245eb06c6862$export$e7dc768d35940237 = "is-focused_a0942f";
$a551245eb06c6862$export$eb262b15b9261fde = "spectrum-Datepicker-focusRing_a0942f";
$a551245eb06c6862$export$b8813cd5d7824ce7 = "is-hovered_a0942f";
$a551245eb06c6862$export$f39a09f249340e2a = "focus-ring_a0942f";
$a551245eb06c6862$export$1e0fb04f31d3c22a = "is-selected_a0942f";
$a551245eb06c6862$export$c9f503f672e8a3c1 = "spectrum-FieldButton--invalid_a0942f";



var $302bd214da97cdcb$exports = {};

$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield-input", () => $302bd214da97cdcb$export$1ac009feb2b0bd0c, (v) => $302bd214da97cdcb$export$1ac009feb2b0bd0c = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield", () => $302bd214da97cdcb$export$2c8a10299d8b3418, (v) => $302bd214da97cdcb$export$2c8a10299d8b3418 = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield--quiet", () => $302bd214da97cdcb$export$bffd5e3d61a81737, (v) => $302bd214da97cdcb$export$bffd5e3d61a81737 = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield--multiline", () => $302bd214da97cdcb$export$9699f1c538a448a0, (v) => $302bd214da97cdcb$export$9699f1c538a448a0 = v);
$parcel$export($302bd214da97cdcb$exports, "is-placeholder", () => $302bd214da97cdcb$export$e5b2f5233e4e5194, (v) => $302bd214da97cdcb$export$e5b2f5233e4e5194 = v);
$parcel$export($302bd214da97cdcb$exports, "is-hovered", () => $302bd214da97cdcb$export$b8813cd5d7824ce7, (v) => $302bd214da97cdcb$export$b8813cd5d7824ce7 = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield--valid", () => $302bd214da97cdcb$export$c2abbc3f970170b7, (v) => $302bd214da97cdcb$export$c2abbc3f970170b7 = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield--invalid", () => $302bd214da97cdcb$export$58e50b10d30123df, (v) => $302bd214da97cdcb$export$58e50b10d30123df = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield--loadable", () => $302bd214da97cdcb$export$14fd30fea6f1348c, (v) => $302bd214da97cdcb$export$14fd30fea6f1348c = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield-validationIcon", () => $302bd214da97cdcb$export$cf0ead5b44db0da3, (v) => $302bd214da97cdcb$export$cf0ead5b44db0da3 = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield-icon", () => $302bd214da97cdcb$export$222d1ccd1870be1d, (v) => $302bd214da97cdcb$export$222d1ccd1870be1d = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield-inputIcon", () => $302bd214da97cdcb$export$c1cdcaa5fe76a871, (v) => $302bd214da97cdcb$export$c1cdcaa5fe76a871 = v);
$parcel$export($302bd214da97cdcb$exports, "spectrum-Textfield-circleLoader", () => $302bd214da97cdcb$export$82a249fb4d6127, (v) => $302bd214da97cdcb$export$82a249fb4d6127 = v);
$parcel$export($302bd214da97cdcb$exports, "is-focused", () => $302bd214da97cdcb$export$e7dc768d35940237, (v) => $302bd214da97cdcb$export$e7dc768d35940237 = v);
$parcel$export($302bd214da97cdcb$exports, "focus-ring", () => $302bd214da97cdcb$export$f39a09f249340e2a, (v) => $302bd214da97cdcb$export$f39a09f249340e2a = v);
$parcel$export($302bd214da97cdcb$exports, "is-disabled", () => $302bd214da97cdcb$export$d35bc1e505d1ebbf, (v) => $302bd214da97cdcb$export$d35bc1e505d1ebbf = v);
var $302bd214da97cdcb$export$1ac009feb2b0bd0c;
var $302bd214da97cdcb$export$2c8a10299d8b3418;
var $302bd214da97cdcb$export$bffd5e3d61a81737;
var $302bd214da97cdcb$export$9699f1c538a448a0;
var $302bd214da97cdcb$export$e5b2f5233e4e5194;
var $302bd214da97cdcb$export$b8813cd5d7824ce7;
var $302bd214da97cdcb$export$c2abbc3f970170b7;
var $302bd214da97cdcb$export$58e50b10d30123df;
var $302bd214da97cdcb$export$14fd30fea6f1348c;
var $302bd214da97cdcb$export$cf0ead5b44db0da3;
var $302bd214da97cdcb$export$222d1ccd1870be1d;
var $302bd214da97cdcb$export$c1cdcaa5fe76a871;
var $302bd214da97cdcb$export$82a249fb4d6127;
var $302bd214da97cdcb$export$e7dc768d35940237;
var $302bd214da97cdcb$export$f39a09f249340e2a;
var $302bd214da97cdcb$export$d35bc1e505d1ebbf;
$302bd214da97cdcb$export$1ac009feb2b0bd0c = "spectrum-Textfield-input_73bc77";
$302bd214da97cdcb$export$2c8a10299d8b3418 = "spectrum-Textfield_73bc77";
$302bd214da97cdcb$export$bffd5e3d61a81737 = "spectrum-Textfield--quiet_73bc77";
$302bd214da97cdcb$export$9699f1c538a448a0 = "spectrum-Textfield--multiline_73bc77";
$302bd214da97cdcb$export$e5b2f5233e4e5194 = "is-placeholder_73bc77";
$302bd214da97cdcb$export$b8813cd5d7824ce7 = "is-hovered_73bc77";
$302bd214da97cdcb$export$c2abbc3f970170b7 = "spectrum-Textfield--valid_73bc77";
$302bd214da97cdcb$export$58e50b10d30123df = "spectrum-Textfield--invalid_73bc77";
$302bd214da97cdcb$export$14fd30fea6f1348c = "spectrum-Textfield--loadable_73bc77";
$302bd214da97cdcb$export$cf0ead5b44db0da3 = "spectrum-Textfield-validationIcon_73bc77";
$302bd214da97cdcb$export$222d1ccd1870be1d = "spectrum-Textfield-icon_73bc77";
$302bd214da97cdcb$export$c1cdcaa5fe76a871 = "spectrum-Textfield-inputIcon_73bc77";
$302bd214da97cdcb$export$82a249fb4d6127 = "spectrum-Textfield-circleLoader_73bc77";
$302bd214da97cdcb$export$e7dc768d35940237 = "is-focused_73bc77";
$302bd214da97cdcb$export$f39a09f249340e2a = "focus-ring_73bc77";
$302bd214da97cdcb$export$d35bc1e505d1ebbf = "is-disabled_73bc77";












const $7e4e78d76d9386c1$export$7637df911c069b4d = /*#__PURE__*/ $iW1Cw$react.forwardRef(function $7e4e78d76d9386c1$export$7637df911c069b4d(props, ref) {
    props = $iW1Cw$useProviderProps(props);
    let { isQuiet: isQuiet , isDisabled: isDisabled , validationState: validationState , isReadOnly: isReadOnly  } = props;
    let { contains: contains  } = $iW1Cw$useFilter({
        sensitivity: 'base'
    });
    let state = $iW1Cw$useComboBoxState({
        ...props,
        defaultFilter: contains,
        allowsEmptyCollection: true,
        // Needs to be false here otherwise we double up on commitSelection/commitCustomValue calls when
        // user taps on underlay (i.e. initial tap will call setFocused(false) -> commitSelection/commitCustomValue via onBlur,
        // then the closing of the tray will call setFocused(false) again due to cleanup effect)
        shouldCloseOnBlur: false
    });
    let buttonRef = $iW1Cw$useRef();
    let domRef = $iW1Cw$useFocusableRef(ref, buttonRef);
    let { triggerProps: triggerProps , overlayProps: overlayProps  } = $iW1Cw$useOverlayTrigger({
        type: 'listbox'
    }, state, buttonRef);
    let { labelProps: labelProps , fieldProps: fieldProps  } = $iW1Cw$useField({
        ...props,
        labelElementType: 'span'
    });
    // Focus the button and show focus ring when clicking on the label
    labelProps.onClick = ()=>{
        if (!props.isDisabled) {
            buttonRef.current.focus();
            $iW1Cw$setInteractionModality('keyboard');
        }
    };
    let onClose = ()=>state.commit()
    ;
    return(/*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$react.Fragment, null, /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$Field, {
        ...props,
        labelProps: labelProps,
        elementType: "span",
        ref: domRef,
        includeNecessityIndicatorInAccessibilityName: true
    }, /*#__PURE__*/ $iW1Cw$react.createElement($7e4e78d76d9386c1$var$ComboBoxButton, {
        ...$iW1Cw$mergeProps(triggerProps, fieldProps, {
            autoFocus: props.autoFocus
        }),
        ref: buttonRef,
        isQuiet: isQuiet,
        isDisabled: isDisabled,
        isPlaceholder: !state.inputValue,
        validationState: validationState,
        onPress: ()=>!isReadOnly && state.open(null, 'manual')
    }, state.inputValue || props.placeholder || '')), /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$Tray, {
        isOpen: state.isOpen,
        onClose: onClose,
        isFixedHeight: true,
        isNonModal: true,
        ...overlayProps
    }, /*#__PURE__*/ $iW1Cw$react.createElement($7e4e78d76d9386c1$var$ComboBoxTray, {
        ...props,
        onClose: onClose,
        overlayProps: overlayProps,
        state: state
    }))));
});
const $7e4e78d76d9386c1$var$ComboBoxButton = /*#__PURE__*/ $iW1Cw$react.forwardRef(function ComboBoxButton(props, ref) {
    let { isQuiet: isQuiet , isDisabled: isDisabled , isPlaceholder: isPlaceholder , validationState: validationState , children: children , style: style , className: className  } = props;
    let formatMessage = $iW1Cw$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($2c1aea9b85d23d6d$exports)));
    let valueId = $iW1Cw$useId();
    let invalidId = $iW1Cw$useId();
    let validationIcon = validationState === 'invalid' ? /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$spectrumiconsuiAlertMedium, {
        id: invalidId,
        "aria-label": formatMessage('invalid')
    }) : /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$spectrumiconsuiCheckmarkMedium, null);
    let validation = /*#__PURE__*/ $iW1Cw$react.cloneElement(validationIcon, {
        UNSAFE_className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($302bd214da97cdcb$exports)), 'spectrum-Textfield-validationIcon', $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'spectrum-InputGroup-input-validationIcon'))
    });
    let { hoverProps: hoverProps , isHovered: isHovered  } = $iW1Cw$useHover({
    });
    let { buttonProps: buttonProps , isPressed: isPressed  } = $iW1Cw$useButton({
        ...props,
        'aria-labelledby': [
            props['aria-labelledby'],
            props['aria-label'] && !props['aria-labelledby'] ? props.id : null,
            valueId,
            validationState === 'invalid' ? invalidId : null
        ].filter(Boolean).join(' '),
        elementType: 'div'
    }, ref);
    return(/*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$FocusRing, {
        focusClass: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'is-focused'),
        focusRingClass: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'focus-ring')
    }, /*#__PURE__*/ $iW1Cw$react.createElement("div", {
        ...$iW1Cw$mergeProps(hoverProps, buttonProps),
        "aria-haspopup": "dialog",
        ref: ref,
        style: {
            ...style,
            outline: 'none'
        },
        className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'spectrum-InputGroup', {
            'spectrum-InputGroup--quiet': isQuiet,
            'is-disabled': isDisabled,
            'spectrum-InputGroup--invalid': validationState === 'invalid',
            'is-hovered': isHovered
        }, $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($c1270b011cb3d946$exports)), 'mobile-combobox'), className)
    }, /*#__PURE__*/ $iW1Cw$react.createElement("div", {
        className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($302bd214da97cdcb$exports)), 'spectrum-Textfield', {
            'spectrum-Textfield--invalid': validationState === 'invalid',
            'spectrum-Textfield--valid': validationState === 'valid',
            'spectrum-Textfield--quiet': isQuiet
        }, $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'spectrum-InputGroup-field'))
    }, /*#__PURE__*/ $iW1Cw$react.createElement("div", {
        className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($302bd214da97cdcb$exports)), 'spectrum-Textfield-input', {
            'is-hovered': isHovered,
            'is-placeholder': isPlaceholder,
            'is-disabled': isDisabled
        }, $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'spectrum-InputGroup-input', $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($4a84444d5a56ab41$exports)), 'spectrum-Field-field')), $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($c1270b011cb3d946$exports)), 'mobile-input'))
    }, /*#__PURE__*/ $iW1Cw$react.createElement("span", {
        id: valueId,
        className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($c1270b011cb3d946$exports)), 'mobile-value')
    }, children)), validationState ? validation : null), /*#__PURE__*/ $iW1Cw$react.createElement("div", {
        className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), 'spectrum-FieldButton', {
            'spectrum-FieldButton--quiet': isQuiet,
            'is-active': isPressed,
            'is-disabled': isDisabled,
            'spectrum-FieldButton--invalid': validationState === 'invalid',
            'is-hovered': isHovered
        }, $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'spectrum-FieldButton'))
    }, /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$spectrumiconsuiChevronDownMedium, {
        UNSAFE_className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'spectrum-Dropdown-chevron')
    })))));
});
function $7e4e78d76d9386c1$var$ComboBoxTray(props) {
    let { state: // completionMode = 'suggest',
    state , isDisabled: isDisabled , validationState: validationState , label: label , overlayProps: overlayProps , loadingState: loadingState , onLoadMore: onLoadMore , onClose: onClose  } = props;
    let timeout = $iW1Cw$useRef(null);
    let [showLoading, setShowLoading] = $iW1Cw$useState(false);
    let inputRef = $iW1Cw$useRef();
    let buttonRef = $iW1Cw$useRef();
    let popoverRef = $iW1Cw$useRef();
    let listBoxRef = $iW1Cw$useRef();
    let layout = $iW1Cw$useListBoxLayout(state);
    let formatMessage = $iW1Cw$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($2c1aea9b85d23d6d$exports)));
    let { inputProps: inputProps , listBoxProps: listBoxProps , labelProps: labelProps  } = $iW1Cw$useComboBox({
        ...props,
        // completionMode,
        keyboardDelegate: layout,
        buttonRef: $iW1Cw$unwrapDOMRef(buttonRef),
        popoverRef: popoverRef,
        listBoxRef: listBoxRef,
        inputRef: inputRef
    }, state);
    $iW1Cw$react.useEffect(()=>{
        $iW1Cw$focusSafely(inputRef.current);
        // When the tray unmounts, set state.isFocused (i.e. the tray input's focus tracker) to false.
        // This is to prevent state.isFocused from being set to true when the tray closes via tapping on the underlay
        // (FocusScope attempts to restore focus to the tray input when tapping outside the tray due to "contain")
        // Have to do this manually since React doesn't call onBlur when a component is unmounted: https://github.com/facebook/react/issues/12363
        return ()=>{
            state.setFocused(false);
        };
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
    let { dialogProps: dialogProps  } = $iW1Cw$useDialog({
        'aria-labelledby': $iW1Cw$useId(labelProps.id)
    }, popoverRef);
    // Override the role of the input to "searchbox" instead of "combobox".
    // Since the listbox is always visible, the combobox role doesn't really give us anything.
    // VoiceOver on iOS reads "double tap to collapse" when focused on the input rather than
    // "double tap to edit text", as with a textbox or searchbox. We'd like double tapping to
    // open the virtual keyboard rather than closing the tray.
    inputProps.role = 'searchbox';
    inputProps['aria-haspopup'] = 'listbox';
    delete inputProps.onTouchEnd;
    let clearButton = /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$ClearButton, {
        preventFocus: true,
        "aria-label": formatMessage('clear'),
        excludeFromTabOrder: true,
        onPress: ()=>{
            state.setInputValue('');
            inputRef.current.focus();
        },
        UNSAFE_className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($fe88933789dbeaab$exports)), 'spectrum-ClearButton'),
        isDisabled: isDisabled
    });
    let loadingCircle = /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$ProgressCircle, {
        "aria-label": formatMessage('loading'),
        size: "S",
        isIndeterminate: true,
        UNSAFE_className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($fe88933789dbeaab$exports)), 'spectrum-Search-circleLoader', $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($302bd214da97cdcb$exports)), 'spectrum-Textfield-circleLoader'))
    });
    // Close the software keyboard on scroll to give the user a bigger area to scroll.
    // But only do this if scrolling with touch, otherwise it can cause issues with touch
    // screen readers.
    let isTouchDown = $iW1Cw$useRef(false);
    let onTouchStart = ()=>{
        isTouchDown.current = true;
    };
    let onTouchEnd = ()=>{
        isTouchDown.current = false;
    };
    let onScroll = $iW1Cw$useCallback(()=>{
        if (!inputRef.current || document.activeElement !== inputRef.current || !isTouchDown.current) return;
        popoverRef.current.focus();
    }, [
        inputRef,
        popoverRef,
        isTouchDown
    ]);
    let inputValue = inputProps.value;
    let lastInputValue = $iW1Cw$useRef(inputValue);
    $iW1Cw$useEffect(()=>{
        if (loadingState === 'filtering' && !showLoading) {
            if (timeout.current === null) timeout.current = setTimeout(()=>{
                setShowLoading(true);
            }, 500);
            // If user is typing, clear the timer and restart since it is a new request
            if (inputValue !== lastInputValue.current) {
                clearTimeout(timeout.current);
                timeout.current = setTimeout(()=>{
                    setShowLoading(true);
                }, 500);
            }
        } else if (loadingState !== 'filtering') {
            // If loading is no longer happening, clear any timers and hide the loading circle
            setShowLoading(false);
            clearTimeout(timeout.current);
            timeout.current = null;
        }
        lastInputValue.current = inputValue;
    }, [
        loadingState,
        inputValue,
        showLoading
    ]);
    let onKeyDown = (e)=>{
        // Close virtual keyboard if user hits Enter w/o any focused options
        if (e.key === 'Enter' && state.selectionManager.focusedKey == null) popoverRef.current.focus();
        else inputProps.onKeyDown(e);
    };
    return(/*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$FocusScope, {
        restoreFocus: true,
        contain: true
    }, /*#__PURE__*/ $iW1Cw$react.createElement("div", {
        ...$iW1Cw$mergeProps(overlayProps, dialogProps),
        ref: popoverRef,
        className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($c1270b011cb3d946$exports)), 'tray-dialog')
    }, /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$DismissButton, {
        onDismiss: onClose
    }), /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$TextFieldBase, {
        label: label,
        labelProps: labelProps,
        inputProps: {
            ...inputProps,
            onKeyDown: onKeyDown
        },
        inputRef: inputRef,
        isDisabled: isDisabled,
        isLoading: showLoading && loadingState === 'filtering',
        loadingIndicator: loadingState != null && loadingCircle,
        validationState: validationState,
        labelAlign: "start",
        labelPosition: "top",
        wrapperChildren: (state.inputValue !== '' || loadingState === 'filtering' || validationState != null) && !props.isReadOnly && clearButton,
        UNSAFE_className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($fe88933789dbeaab$exports)), 'spectrum-Search', 'spectrum-Textfield', 'spectrum-Search--loadable', {
            'spectrum-Search--invalid': validationState === 'invalid',
            'spectrum-Search--valid': validationState === 'valid'
        }, $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($c1270b011cb3d946$exports)), 'tray-textfield', {
            'has-label': !!props.label
        })),
        inputClassName: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($c1270b011cb3d946$exports)), 'tray-textfield-input', $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($fe88933789dbeaab$exports)), 'spectrum-Search-input')),
        validationIconClassName: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($fe88933789dbeaab$exports)), 'spectrum-Search-validationIcon')
    }), /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$ListBoxBase, {
        ...listBoxProps,
        domProps: {
            onTouchStart: onTouchStart,
            onTouchEnd: onTouchEnd
        },
        disallowEmptySelection: true,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        shouldUseVirtualFocus: true,
        renderEmptyState: ()=>loadingState !== 'loading' && /*#__PURE__*/ $iW1Cw$react.createElement("span", {
                className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($c1270b011cb3d946$exports)), 'no-results')
            }, formatMessage('noResults'))
        ,
        UNSAFE_className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($c1270b011cb3d946$exports)), 'tray-listbox'),
        ref: listBoxRef,
        onScroll: onScroll,
        onLoadMore: onLoadMore,
        isLoading: loadingState === 'loading' || loadingState === 'loadingMore'
    }), /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$DismissButton, {
        onDismiss: onClose
    }))));
}















function $d40059e4c82e96bb$var$ComboBox(props, ref) {
    props = $iW1Cw$useProviderProps(props);
    let isMobile = $iW1Cw$useIsMobileDevice();
    if (isMobile) // menuTrigger=focus/manual don't apply to mobile combobox
    return(/*#__PURE__*/ $iW1Cw$react.createElement($7e4e78d76d9386c1$export$7637df911c069b4d, {
        ...props,
        menuTrigger: "input",
        ref: ref
    }));
    else return(/*#__PURE__*/ $iW1Cw$react.createElement($d40059e4c82e96bb$var$ComboBoxBase, {
        ...props,
        ref: ref
    }));
}
const $d40059e4c82e96bb$var$ComboBoxBase = /*#__PURE__*/ $iW1Cw$react.forwardRef(function ComboBoxBase(props, ref) {
    let { menuTrigger: menuTrigger = 'input' , shouldFlip: shouldFlip = true , direction: direction = 'bottom' , isQuiet: isQuiet , loadingState: loadingState , onLoadMore: onLoadMore  } = props;
    let formatMessage = $iW1Cw$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($2c1aea9b85d23d6d$exports)));
    let isAsync = loadingState != null;
    let popoverRef = $iW1Cw$useRef();
    let unwrappedPopoverRef = $iW1Cw$useUnwrapDOMRef(popoverRef);
    let buttonRef = $iW1Cw$useRef();
    let unwrappedButtonRef = $iW1Cw$useUnwrapDOMRef(buttonRef);
    let listBoxRef = $iW1Cw$useRef();
    let inputRef = $iW1Cw$useRef();
    let domRef = $iW1Cw$useFocusableRef(ref, inputRef);
    let { contains: contains  } = $iW1Cw$useFilter({
        sensitivity: 'base'
    });
    let state = $iW1Cw$useComboBoxState({
        ...props,
        defaultFilter: contains,
        allowsEmptyCollection: isAsync
    });
    let layout = $iW1Cw$useListBoxLayout(state);
    let { buttonProps: buttonProps , inputProps: inputProps , listBoxProps: listBoxProps , labelProps: labelProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $iW1Cw$useComboBox({
        ...props,
        keyboardDelegate: layout,
        buttonRef: unwrappedButtonRef,
        popoverRef: unwrappedPopoverRef,
        listBoxRef: listBoxRef,
        inputRef: inputRef,
        menuTrigger: menuTrigger
    }, state);
    let { overlayProps: overlayProps , placement: placement , updatePosition: updatePosition  } = $iW1Cw$useOverlayPosition({
        targetRef: unwrappedButtonRef,
        overlayRef: unwrappedPopoverRef,
        scrollRef: listBoxRef,
        placement: `${direction} end`,
        shouldFlip: shouldFlip,
        isOpen: state.isOpen,
        onClose: state.close
    });
    // Measure the width of the inputfield and the button to inform the width of the menu (below).
    let [menuWidth, setMenuWidth] = $iW1Cw$useState(null);
    let { scale: scale  } = $iW1Cw$useProvider();
    let onResize = $iW1Cw$useCallback(()=>{
        if (unwrappedButtonRef.current && inputRef.current) {
            let buttonWidth = unwrappedButtonRef.current.offsetWidth;
            let inputWidth = inputRef.current.offsetWidth;
            setMenuWidth(buttonWidth + inputWidth);
        }
    }, [
        unwrappedButtonRef,
        inputRef,
        setMenuWidth
    ]);
    $iW1Cw$useResizeObserver({
        ref: domRef,
        onResize: onResize
    });
    $iW1Cw$useLayoutEffect(onResize, [
        scale,
        onResize
    ]);
    // Update position once the ListBox has rendered. This ensures that
    // it flips properly when it doesn't fit in the available space.
    // TODO: add ResizeObserver to useOverlayPosition so we don't need this.
    $iW1Cw$useLayoutEffect(()=>{
        if (state.isOpen) requestAnimationFrame(()=>{
            updatePosition();
        });
    }, [
        state.isOpen,
        updatePosition
    ]);
    let style = {
        ...overlayProps.style,
        width: isQuiet ? null : menuWidth,
        minWidth: isQuiet ? `calc(${menuWidth}px + calc(2 * var(--spectrum-dropdown-quiet-offset)))` : menuWidth
    };
    return(/*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$react.Fragment, null, /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$Field, {
        ...props,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        labelProps: labelProps,
        ref: domRef
    }, /*#__PURE__*/ $iW1Cw$react.createElement($d40059e4c82e96bb$var$ComboBoxInput, {
        ...props,
        isOpen: state.isOpen,
        loadingState: loadingState,
        inputProps: inputProps,
        inputRef: inputRef,
        triggerProps: buttonProps,
        triggerRef: buttonRef
    })), /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$Popover, {
        isOpen: state.isOpen,
        UNSAFE_style: style,
        UNSAFE_className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'spectrum-InputGroup-popover', {
            'spectrum-InputGroup-popover--quiet': isQuiet
        }),
        ref: popoverRef,
        placement: placement,
        hideArrow: true,
        isNonModal: true,
        isDismissable: false
    }, /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$ListBoxBase, {
        ...listBoxProps,
        ref: listBoxRef,
        disallowEmptySelection: true,
        autoFocus: state.focusStrategy,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        shouldUseVirtualFocus: true,
        isLoading: loadingState === 'loadingMore',
        onLoadMore: onLoadMore,
        renderEmptyState: ()=>isAsync && /*#__PURE__*/ $iW1Cw$react.createElement("span", {
                className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($c1270b011cb3d946$exports)), 'no-results')
            }, loadingState === 'loading' ? formatMessage('loading') : formatMessage('noResults'))
    }), /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$DismissButton, {
        onDismiss: ()=>state.close()
    }))));
});
const $d40059e4c82e96bb$var$ComboBoxInput = /*#__PURE__*/ $iW1Cw$react.forwardRef(function ComboBoxInput(props, ref) {
    let { isQuiet: isQuiet , isDisabled: isDisabled , validationState: validationState , inputProps: inputProps , inputRef: inputRef , triggerProps: triggerProps , triggerRef: triggerRef , autoFocus: autoFocus , style: style , className: className , loadingState: loadingState , isOpen: isOpen , menuTrigger: menuTrigger  } = props;
    let { hoverProps: hoverProps , isHovered: isHovered  } = $iW1Cw$useHover({
    });
    let formatMessage = $iW1Cw$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($2c1aea9b85d23d6d$exports)));
    let timeout = $iW1Cw$useRef(null);
    let [showLoading, setShowLoading] = $iW1Cw$useState(false);
    let loadingCircle = /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$ProgressCircle, {
        "aria-label": formatMessage('loading'),
        size: "S",
        isIndeterminate: true,
        UNSAFE_className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($302bd214da97cdcb$exports)), 'spectrum-Textfield-circleLoader', $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'spectrum-InputGroup-input-circleLoader'))
    });
    let isLoading = loadingState === 'loading' || loadingState === 'filtering';
    let inputValue = inputProps.value;
    let lastInputValue = $iW1Cw$useRef(inputValue);
    $iW1Cw$useEffect(()=>{
        if (isLoading && !showLoading) {
            if (timeout.current === null) timeout.current = setTimeout(()=>{
                setShowLoading(true);
            }, 500);
            // If user is typing, clear the timer and restart since it is a new request
            if (inputValue !== lastInputValue.current) {
                clearTimeout(timeout.current);
                timeout.current = setTimeout(()=>{
                    setShowLoading(true);
                }, 500);
            }
        } else if (!isLoading) {
            // If loading is no longer happening, clear any timers and hide the loading circle
            setShowLoading(false);
            clearTimeout(timeout.current);
            timeout.current = null;
        }
        lastInputValue.current = inputValue;
    }, [
        isLoading,
        showLoading,
        inputValue
    ]);
    return(/*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$FocusRing, {
        within: true,
        isTextInput: true,
        focusClass: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'is-focused'),
        focusRingClass: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ $iW1Cw$react.createElement("div", {
        ...hoverProps,
        ref: ref,
        style: style,
        className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'spectrum-InputGroup', {
            'spectrum-InputGroup--quiet': isQuiet,
            'is-disabled': isDisabled,
            'spectrum-InputGroup--invalid': validationState === 'invalid',
            'is-hovered': isHovered
        }, className)
    }, /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$TextFieldBase, {
        inputProps: inputProps,
        inputRef: inputRef,
        UNSAFE_className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'spectrum-InputGroup-field'),
        inputClassName: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'spectrum-InputGroup-input'),
        validationIconClassName: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'spectrum-InputGroup-input-validationIcon'),
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        validationState: validationState,
        // loading circle should only be displayed if menu is open, if menuTrigger is "manual", or first time load (to stop circle from showing up when user selects an option)
        // TODO: add special case for completionMode: complete as well
        isLoading: showLoading && (isOpen || menuTrigger === 'manual' || loadingState === 'loading'),
        loadingIndicator: loadingState != null && loadingCircle
    }), /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$PressResponder, {
        preventFocusOnPress: true,
        isPressed: isOpen
    }, /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$FieldButton, {
        ...triggerProps,
        ref: triggerRef,
        UNSAFE_className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'spectrum-FieldButton'),
        isQuiet: isQuiet,
        validationState: validationState
    }, /*#__PURE__*/ $iW1Cw$react.createElement($iW1Cw$spectrumiconsuiChevronDownMedium, {
        UNSAFE_className: $iW1Cw$classNames((/*@__PURE__*/$parcel$interopDefault($a551245eb06c6862$exports)), 'spectrum-Dropdown-chevron')
    }))))));
});
/**
 * ComboBoxes combine a text entry with a picker menu, allowing users to filter longer lists to only the selections matching a query.
 */ const $d40059e4c82e96bb$export$72b9695b8216309a = /*#__PURE__*/ $iW1Cw$react.forwardRef($d40059e4c82e96bb$var$ComboBox);





export {$3b54a9f3842b03a6$re_export$Item as Item, $3b54a9f3842b03a6$re_export$Section as Section, $d40059e4c82e96bb$export$72b9695b8216309a as ComboBox};
