// @react-spectrum/numberfield/dist/main.js
require("./main.css");
var $4Mw7h$reactspectrumutils = require("@react-spectrum/utils");
var $4Mw7h$reactspectrumlabel = require("@react-spectrum/label");
var $4Mw7h$reactariafocus = require("@react-aria/focus");
var $4Mw7h$reactariautils = require("@react-aria/utils");
var $4Mw7h$react = require("react");
var $4Mw7h$reactspectrumtextfield = require("@react-spectrum/textfield");
var $4Mw7h$reactariainteractions = require("@react-aria/interactions");
var $4Mw7h$reactariai18n = require("@react-aria/i18n");
var $4Mw7h$reactarianumberfield = require("@react-aria/numberfield");
var $4Mw7h$reactstatelynumberfield = require("@react-stately/numberfield");
var $4Mw7h$reactspectrumprovider = require("@react-spectrum/provider");
var $4Mw7h$spectrumiconsworkflowAdd = require("@spectrum-icons/workflow/Add");
var $4Mw7h$spectrumiconsuiChevronDownSmall = require("@spectrum-icons/ui/ChevronDownSmall");
var $4Mw7h$spectrumiconsuiChevronUpSmall = require("@spectrum-icons/ui/ChevronUpSmall");
var $4Mw7h$spectrumiconsworkflowRemove = require("@spectrum-icons/workflow/Remove");
var $4Mw7h$reactariabutton = require("@react-aria/button");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $ccf6ac74f87aa7ec$exports = {};

$parcel$export($ccf6ac74f87aa7ec$exports, "NumberField", () => $ccf6ac74f87aa7ec$export$63c5fa0b2fdccd2e);













var $5c960f27611f75d2$exports = {};

$parcel$export($5c960f27611f75d2$exports, "spectrum-Button", () => $5c960f27611f75d2$export$1db4cca5b4ade39a, (v) => $5c960f27611f75d2$export$1db4cca5b4ade39a = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-ActionButton", () => $5c960f27611f75d2$export$53da69f51b770d3, (v) => $5c960f27611f75d2$export$53da69f51b770d3 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-LogicButton", () => $5c960f27611f75d2$export$36ae67a131730482, (v) => $5c960f27611f75d2$export$36ae67a131730482 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-FieldButton", () => $5c960f27611f75d2$export$a024c99ecf6b5741, (v) => $5c960f27611f75d2$export$a024c99ecf6b5741 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-ClearButton", () => $5c960f27611f75d2$export$b752ce409e5660c, (v) => $5c960f27611f75d2$export$b752ce409e5660c = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-Stepper-button", () => $5c960f27611f75d2$export$4e11596482f95f63, (v) => $5c960f27611f75d2$export$4e11596482f95f63 = v);
$parcel$export($5c960f27611f75d2$exports, "focus-ring", () => $5c960f27611f75d2$export$f39a09f249340e2a, (v) => $5c960f27611f75d2$export$f39a09f249340e2a = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-Icon", () => $5c960f27611f75d2$export$d374b04f30360026, (v) => $5c960f27611f75d2$export$d374b04f30360026 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-ActionButton--emphasized", () => $5c960f27611f75d2$export$5aabb61a966e266f, (v) => $5c960f27611f75d2$export$5aabb61a966e266f = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-ActionButton--staticColor", () => $5c960f27611f75d2$export$29b7b5d6a5a497c5, (v) => $5c960f27611f75d2$export$29b7b5d6a5a497c5 = v);
$parcel$export($5c960f27611f75d2$exports, "is-hovered", () => $5c960f27611f75d2$export$b8813cd5d7824ce7, (v) => $5c960f27611f75d2$export$b8813cd5d7824ce7 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-Button-label", () => $5c960f27611f75d2$export$c081ba562e204942, (v) => $5c960f27611f75d2$export$c081ba562e204942 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-ActionButton-label", () => $5c960f27611f75d2$export$ebb4635c66bacdb9, (v) => $5c960f27611f75d2$export$ebb4635c66bacdb9 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-ActionButton-hold", () => $5c960f27611f75d2$export$4a4f01c443eff992, (v) => $5c960f27611f75d2$export$4a4f01c443eff992 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-ActionButton--quiet", () => $5c960f27611f75d2$export$40e1f009544bacf3, (v) => $5c960f27611f75d2$export$40e1f009544bacf3 = v);
$parcel$export($5c960f27611f75d2$exports, "is-disabled", () => $5c960f27611f75d2$export$d35bc1e505d1ebbf, (v) => $5c960f27611f75d2$export$d35bc1e505d1ebbf = v);
$parcel$export($5c960f27611f75d2$exports, "is-open", () => $5c960f27611f75d2$export$a9781837241c946d, (v) => $5c960f27611f75d2$export$a9781837241c946d = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-FieldButton--quiet", () => $5c960f27611f75d2$export$5ca7b274e4fff9fc, (v) => $5c960f27611f75d2$export$5ca7b274e4fff9fc = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-ClearButton--overBackground", () => $5c960f27611f75d2$export$116f2fccf941d34d, (v) => $5c960f27611f75d2$export$116f2fccf941d34d = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-ClearButton--small", () => $5c960f27611f75d2$export$49280ffa5236ba8b, (v) => $5c960f27611f75d2$export$49280ffa5236ba8b = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-Stepper", () => $5c960f27611f75d2$export$f49c123746e1fb8, (v) => $5c960f27611f75d2$export$f49c123746e1fb8 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-Stepper--isMobile", () => $5c960f27611f75d2$export$75e6f9c4d1d95421, (v) => $5c960f27611f75d2$export$75e6f9c4d1d95421 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-Stepper-button--stepUp", () => $5c960f27611f75d2$export$6c76d609d604c6e5, (v) => $5c960f27611f75d2$export$6c76d609d604c6e5 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-Stepper-button--stepDown", () => $5c960f27611f75d2$export$e4ad52f6ba4e317b, (v) => $5c960f27611f75d2$export$e4ad52f6ba4e317b = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-Stepper-field", () => $5c960f27611f75d2$export$faac6573b15f75d6, (v) => $5c960f27611f75d2$export$faac6573b15f75d6 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-Stepper-icon", () => $5c960f27611f75d2$export$639c56ba6082b615, (v) => $5c960f27611f75d2$export$639c56ba6082b615 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-Stepper--isQuiet", () => $5c960f27611f75d2$export$70ed0eb2ba7512f8, (v) => $5c960f27611f75d2$export$70ed0eb2ba7512f8 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-Stepper--showStepper", () => $5c960f27611f75d2$export$a4e366275982a12f, (v) => $5c960f27611f75d2$export$a4e366275982a12f = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-Stepper-input", () => $5c960f27611f75d2$export$8273fb931edd57d4, (v) => $5c960f27611f75d2$export$8273fb931edd57d4 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-Stepper-container", () => $5c960f27611f75d2$export$f49a6eb8460c3f93, (v) => $5c960f27611f75d2$export$f49a6eb8460c3f93 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-Stepper-container--isMobile", () => $5c960f27611f75d2$export$3486d98f7bb904b1, (v) => $5c960f27611f75d2$export$3486d98f7bb904b1 = v);
$parcel$export($5c960f27611f75d2$exports, "spectrum-Stepper-button-icon", () => $5c960f27611f75d2$export$7d7cb0c4055e98ab, (v) => $5c960f27611f75d2$export$7d7cb0c4055e98ab = v);
$parcel$export($5c960f27611f75d2$exports, "is-active", () => $5c960f27611f75d2$export$20fd0f7cd4e6112f, (v) => $5c960f27611f75d2$export$20fd0f7cd4e6112f = v);
$parcel$export($5c960f27611f75d2$exports, "is-focused", () => $5c960f27611f75d2$export$e7dc768d35940237, (v) => $5c960f27611f75d2$export$e7dc768d35940237 = v);
$parcel$export($5c960f27611f75d2$exports, "is-invalid", () => $5c960f27611f75d2$export$14a30de2866685fa, (v) => $5c960f27611f75d2$export$14a30de2866685fa = v);
var $5c960f27611f75d2$export$1db4cca5b4ade39a;
var $5c960f27611f75d2$export$53da69f51b770d3;
var $5c960f27611f75d2$export$36ae67a131730482;
var $5c960f27611f75d2$export$a024c99ecf6b5741;
var $5c960f27611f75d2$export$b752ce409e5660c;
var $5c960f27611f75d2$export$4e11596482f95f63;
var $5c960f27611f75d2$export$f39a09f249340e2a;
var $5c960f27611f75d2$export$d374b04f30360026;
var $5c960f27611f75d2$export$5aabb61a966e266f;
var $5c960f27611f75d2$export$29b7b5d6a5a497c5;
var $5c960f27611f75d2$export$b8813cd5d7824ce7;
var $5c960f27611f75d2$export$c081ba562e204942;
var $5c960f27611f75d2$export$ebb4635c66bacdb9;
var $5c960f27611f75d2$export$4a4f01c443eff992;
var $5c960f27611f75d2$export$40e1f009544bacf3;
var $5c960f27611f75d2$export$d35bc1e505d1ebbf;
var $5c960f27611f75d2$export$a9781837241c946d;
var $5c960f27611f75d2$export$5ca7b274e4fff9fc;
var $5c960f27611f75d2$export$116f2fccf941d34d;
var $5c960f27611f75d2$export$49280ffa5236ba8b;
var $5c960f27611f75d2$export$f49c123746e1fb8;
var $5c960f27611f75d2$export$75e6f9c4d1d95421;
var $5c960f27611f75d2$export$6c76d609d604c6e5;
var $5c960f27611f75d2$export$e4ad52f6ba4e317b;
var $5c960f27611f75d2$export$faac6573b15f75d6;
var $5c960f27611f75d2$export$639c56ba6082b615;
var $5c960f27611f75d2$export$70ed0eb2ba7512f8;
var $5c960f27611f75d2$export$a4e366275982a12f;
var $5c960f27611f75d2$export$8273fb931edd57d4;
var $5c960f27611f75d2$export$f49a6eb8460c3f93;
var $5c960f27611f75d2$export$3486d98f7bb904b1;
var $5c960f27611f75d2$export$7d7cb0c4055e98ab;
var $5c960f27611f75d2$export$20fd0f7cd4e6112f;
var $5c960f27611f75d2$export$e7dc768d35940237;
var $5c960f27611f75d2$export$14a30de2866685fa;
$5c960f27611f75d2$export$1db4cca5b4ade39a = "spectrum-Button_8890ad";
$5c960f27611f75d2$export$53da69f51b770d3 = "spectrum-ActionButton_8890ad";
$5c960f27611f75d2$export$36ae67a131730482 = "spectrum-LogicButton_8890ad";
$5c960f27611f75d2$export$a024c99ecf6b5741 = "spectrum-FieldButton_8890ad";
$5c960f27611f75d2$export$b752ce409e5660c = "spectrum-ClearButton_8890ad";
$5c960f27611f75d2$export$4e11596482f95f63 = "spectrum-Stepper-button_8890ad";
$5c960f27611f75d2$export$f39a09f249340e2a = "focus-ring_8890ad";
$5c960f27611f75d2$export$d374b04f30360026 = "spectrum-Icon_8890ad";
$5c960f27611f75d2$export$5aabb61a966e266f = "spectrum-ActionButton--emphasized_8890ad";
$5c960f27611f75d2$export$29b7b5d6a5a497c5 = "spectrum-ActionButton--staticColor_8890ad";
$5c960f27611f75d2$export$b8813cd5d7824ce7 = "is-hovered_8890ad";
$5c960f27611f75d2$export$c081ba562e204942 = "spectrum-Button-label_8890ad";
$5c960f27611f75d2$export$ebb4635c66bacdb9 = "spectrum-ActionButton-label_8890ad";
$5c960f27611f75d2$export$4a4f01c443eff992 = "spectrum-ActionButton-hold_8890ad";
$5c960f27611f75d2$export$40e1f009544bacf3 = "spectrum-ActionButton--quiet_8890ad";
$5c960f27611f75d2$export$d35bc1e505d1ebbf = "is-disabled_8890ad";
$5c960f27611f75d2$export$a9781837241c946d = "is-open_8890ad";
$5c960f27611f75d2$export$5ca7b274e4fff9fc = "spectrum-FieldButton--quiet_8890ad";
$5c960f27611f75d2$export$116f2fccf941d34d = "spectrum-ClearButton--overBackground_8890ad";
$5c960f27611f75d2$export$49280ffa5236ba8b = "spectrum-ClearButton--small_8890ad";
$5c960f27611f75d2$export$f49c123746e1fb8 = "spectrum-Stepper_8890ad";
$5c960f27611f75d2$export$75e6f9c4d1d95421 = "spectrum-Stepper--isMobile_8890ad";
$5c960f27611f75d2$export$6c76d609d604c6e5 = "spectrum-Stepper-button--stepUp_8890ad";
$5c960f27611f75d2$export$e4ad52f6ba4e317b = "spectrum-Stepper-button--stepDown_8890ad";
$5c960f27611f75d2$export$faac6573b15f75d6 = "spectrum-Stepper-field_8890ad";
$5c960f27611f75d2$export$639c56ba6082b615 = "spectrum-Stepper-icon_8890ad";
$5c960f27611f75d2$export$70ed0eb2ba7512f8 = "spectrum-Stepper--isQuiet_8890ad";
$5c960f27611f75d2$export$a4e366275982a12f = "spectrum-Stepper--showStepper_8890ad";
$5c960f27611f75d2$export$8273fb931edd57d4 = "spectrum-Stepper-input_8890ad";
$5c960f27611f75d2$export$f49a6eb8460c3f93 = "spectrum-Stepper-container_8890ad";
$5c960f27611f75d2$export$3486d98f7bb904b1 = "spectrum-Stepper-container--isMobile_8890ad";
$5c960f27611f75d2$export$7d7cb0c4055e98ab = "spectrum-Stepper-button-icon_8890ad";
$5c960f27611f75d2$export$20fd0f7cd4e6112f = "is-active_8890ad";
$5c960f27611f75d2$export$e7dc768d35940237 = "is-focused_8890ad";
$5c960f27611f75d2$export$14a30de2866685fa = "is-invalid_8890ad";





function $63a371c16204d9f5$var$StepButton(props, ref) {
    props = $4Mw7h$reactspectrumprovider.useProviderProps(props);
    let { scale: scale  } = $4Mw7h$reactspectrumprovider.useProvider();
    let { direction: direction , isDisabled: isDisabled , isQuiet: isQuiet  } = props;
    /**
   * Must use div for now because Safari pointer event bugs on disabled form elements.
   * Link https://bugs.webkit.org/show_bug.cgi?id=219188.
   */ let { buttonProps: buttonProps , isPressed: isPressed  } = $4Mw7h$reactariabutton.useButton({
        ...props,
        elementType: 'div'
    }, ref);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $4Mw7h$reactariainteractions.useHover(props);
    return(/*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).createElement($4Mw7h$reactariafocus.FocusRing, {
        focusRingClass: $4Mw7h$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($5c960f27611f75d2$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).createElement("div", {
        className: $4Mw7h$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($5c960f27611f75d2$exports)), 'spectrum-Stepper-button', {
            'spectrum-Stepper-button--stepUp': direction === 'up',
            'spectrum-Stepper-button--stepDown': direction === 'down',
            'spectrum-Stepper-button--isQuiet': isQuiet,
            'is-hovered': isHovered,
            'is-active': isPressed,
            'is-disabled': isDisabled
        }),
        ...$4Mw7h$reactariautils.mergeProps(hoverProps, buttonProps),
        ref: ref
    }, direction === 'up' && scale === 'large' && /*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).createElement(($parcel$interopDefault($4Mw7h$spectrumiconsworkflowAdd)), {
        UNSAFE_className: $4Mw7h$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($5c960f27611f75d2$exports)), 'spectrum-Stepper-button-icon', 'spectrum-Stepper-stepUpIcon'),
        size: "S"
    }), direction === 'up' && scale === 'medium' && /*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).createElement(($parcel$interopDefault($4Mw7h$spectrumiconsuiChevronUpSmall)), {
        UNSAFE_className: $4Mw7h$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($5c960f27611f75d2$exports)), 'spectrum-Stepper-button-icon', 'spectrum-Stepper-stepUpIcon')
    }), direction === 'down' && scale === 'large' && /*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).createElement(($parcel$interopDefault($4Mw7h$spectrumiconsworkflowRemove)), {
        UNSAFE_className: $4Mw7h$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($5c960f27611f75d2$exports)), 'spectrum-Stepper-button-icon', 'spectrum-Stepper-stepDownIcon'),
        size: "S"
    }), direction === 'down' && scale === 'medium' && /*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).createElement(($parcel$interopDefault($4Mw7h$spectrumiconsuiChevronDownSmall)), {
        UNSAFE_className: $4Mw7h$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($5c960f27611f75d2$exports)), 'spectrum-Stepper-button-icon', 'spectrum-Stepper-stepDownIcon')
    }))));
}
/**
 * Buttons for NumberField.
 */ let $63a371c16204d9f5$export$b2f6b60c1d32d6aa = /*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).forwardRef($63a371c16204d9f5$var$StepButton);









function $ccf6ac74f87aa7ec$var$NumberField(props, ref) {
    props = $4Mw7h$reactspectrumprovider.useProviderProps(props);
    let provider = $4Mw7h$reactspectrumprovider.useProvider();
    let { isQuiet: isQuiet , isReadOnly: isReadOnly , isDisabled: isDisabled , hideStepper: hideStepper  } = props;
    let { styleProps: style  } = $4Mw7h$reactspectrumutils.useStyleProps(props);
    let { locale: locale  } = $4Mw7h$reactariai18n.useLocale();
    let state = $4Mw7h$reactstatelynumberfield.useNumberFieldState({
        ...props,
        locale: locale
    });
    let inputRef = $4Mw7h$react.useRef();
    let domRef = $4Mw7h$reactspectrumutils.useFocusableRef(ref, inputRef);
    let { groupProps: groupProps , labelProps: labelProps , inputProps: inputProps , incrementButtonProps: incrementButtonProps , decrementButtonProps: decrementButtonProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $4Mw7h$reactarianumberfield.useNumberField(props, state, inputRef);
    let isMobile = provider.scale === 'large';
    let showStepper = !hideStepper;
    let { isHovered: isHovered , hoverProps: hoverProps  } = $4Mw7h$reactariainteractions.useHover({
        isDisabled: isDisabled
    });
    let className = $4Mw7h$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($5c960f27611f75d2$exports)), 'spectrum-Stepper', {
        'spectrum-Stepper--isQuiet': isQuiet,
        'is-disabled': isDisabled,
        'spectrum-Stepper--readonly': isReadOnly,
        'is-invalid': props.validationState === 'invalid',
        'spectrum-Stepper--showStepper': showStepper,
        'spectrum-Stepper--isMobile': isMobile,
        'is-hovered': isHovered,
        // because FocusRing won't pass along the className from Field, we have to handle that ourselves
        [style.className]: !props.label && style.className
    });
    return(/*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).createElement($4Mw7h$reactspectrumlabel.Field, {
        ...props,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        labelProps: labelProps,
        ref: domRef,
        wrapperClassName: $4Mw7h$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($5c960f27611f75d2$exports)), 'spectrum-Stepper-container', {
            'spectrum-Stepper-container--isMobile': isMobile
        })
    }, /*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).createElement($ccf6ac74f87aa7ec$var$NumberFieldInput, {
        ...props,
        groupProps: $4Mw7h$reactariautils.mergeProps(groupProps, hoverProps),
        inputProps: inputProps,
        inputRef: inputRef,
        incrementProps: incrementButtonProps,
        decrementProps: decrementButtonProps,
        className: className,
        style: style
    })));
}
const $ccf6ac74f87aa7ec$var$NumberFieldInput = /*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).forwardRef(function NumberFieldInput(props, ref) {
    let { groupProps: groupProps , inputProps: inputProps , inputRef: inputRef , incrementProps: incrementProps , decrementProps: decrementProps , className: className , style: style , autoFocus: autoFocus , isQuiet: isQuiet , hideStepper: hideStepper , validationState: validationState  } = props;
    let showStepper = !hideStepper;
    return(/*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).createElement($4Mw7h$reactariafocus.FocusRing, {
        within: true,
        isTextInput: true,
        focusClass: $4Mw7h$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($5c960f27611f75d2$exports)), 'is-focused'),
        focusRingClass: $4Mw7h$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($5c960f27611f75d2$exports)), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).createElement("div", {
        ...groupProps,
        ref: ref,
        style: style,
        className: className
    }, /*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).createElement($4Mw7h$reactspectrumtextfield.TextFieldBase, {
        UNSAFE_className: $4Mw7h$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($5c960f27611f75d2$exports)), 'spectrum-Stepper-field'),
        inputClassName: $4Mw7h$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($5c960f27611f75d2$exports)), 'spectrum-Stepper-input'),
        validationIconClassName: $4Mw7h$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($5c960f27611f75d2$exports)), 'spectrum-Stepper-icon'),
        isQuiet: isQuiet,
        inputRef: inputRef,
        validationState: validationState,
        inputProps: inputProps
    }), showStepper && /*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).createElement(($parcel$interopDefault($4Mw7h$react)).Fragment, null, /*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).createElement($63a371c16204d9f5$export$b2f6b60c1d32d6aa, {
        direction: "up",
        isQuiet: isQuiet,
        ...incrementProps
    }), /*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).createElement($63a371c16204d9f5$export$b2f6b60c1d32d6aa, {
        direction: "down",
        isQuiet: isQuiet,
        ...decrementProps
    })))));
});
/**
 * NumberFields allow users to enter a number, and increment or decrement the value using stepper buttons.
 */ let $ccf6ac74f87aa7ec$export$63c5fa0b2fdccd2e = /*#__PURE__*/ ($parcel$interopDefault($4Mw7h$react)).forwardRef($ccf6ac74f87aa7ec$var$NumberField);


$parcel$exportWildcard(module.exports, $ccf6ac74f87aa7ec$exports);
