// @react-spectrum/numberfield/dist/module.js
import "./main.css";
import {useStyleProps as $k93dG$useStyleProps, useFocusableRef as $k93dG$useFocusableRef, classNames as $k93dG$classNames} from "@react-spectrum/utils";
import {Field as $k93dG$Field} from "@react-spectrum/label";
import {FocusRing as $k93dG$FocusRing} from "@react-aria/focus";
import {mergeProps as $k93dG$mergeProps} from "@react-aria/utils";
import $k93dG$react, {useRef as $k93dG$useRef} from "react";
import {TextFieldBase as $k93dG$TextFieldBase} from "@react-spectrum/textfield";
import {useHover as $k93dG$useHover} from "@react-aria/interactions";
import {useLocale as $k93dG$useLocale} from "@react-aria/i18n";
import {useNumberField as $k93dG$useNumberField} from "@react-aria/numberfield";
import {useNumberFieldState as $k93dG$useNumberFieldState} from "@react-stately/numberfield";
import {useProviderProps as $k93dG$useProviderProps, useProvider as $k93dG$useProvider} from "@react-spectrum/provider";
import $k93dG$spectrumiconsworkflowAdd from "@spectrum-icons/workflow/Add";
import $k93dG$spectrumiconsuiChevronDownSmall from "@spectrum-icons/ui/ChevronDownSmall";
import $k93dG$spectrumiconsuiChevronUpSmall from "@spectrum-icons/ui/ChevronUpSmall";
import $k93dG$spectrumiconsworkflowRemove from "@spectrum-icons/workflow/Remove";
import {useButton as $k93dG$useButton} from "@react-aria/button";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $5d632a544d0c6901$exports = {};

$parcel$export($5d632a544d0c6901$exports, "NumberField", () => $5d632a544d0c6901$export$63c5fa0b2fdccd2e);













var $4921fa4317dcd122$exports = {};

$parcel$export($4921fa4317dcd122$exports, "spectrum-Button", () => $4921fa4317dcd122$export$1db4cca5b4ade39a, (v) => $4921fa4317dcd122$export$1db4cca5b4ade39a = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-ActionButton", () => $4921fa4317dcd122$export$53da69f51b770d3, (v) => $4921fa4317dcd122$export$53da69f51b770d3 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-LogicButton", () => $4921fa4317dcd122$export$36ae67a131730482, (v) => $4921fa4317dcd122$export$36ae67a131730482 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-FieldButton", () => $4921fa4317dcd122$export$a024c99ecf6b5741, (v) => $4921fa4317dcd122$export$a024c99ecf6b5741 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-ClearButton", () => $4921fa4317dcd122$export$b752ce409e5660c, (v) => $4921fa4317dcd122$export$b752ce409e5660c = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-Stepper-button", () => $4921fa4317dcd122$export$4e11596482f95f63, (v) => $4921fa4317dcd122$export$4e11596482f95f63 = v);
$parcel$export($4921fa4317dcd122$exports, "focus-ring", () => $4921fa4317dcd122$export$f39a09f249340e2a, (v) => $4921fa4317dcd122$export$f39a09f249340e2a = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-Icon", () => $4921fa4317dcd122$export$d374b04f30360026, (v) => $4921fa4317dcd122$export$d374b04f30360026 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-ActionButton--emphasized", () => $4921fa4317dcd122$export$5aabb61a966e266f, (v) => $4921fa4317dcd122$export$5aabb61a966e266f = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-ActionButton--staticColor", () => $4921fa4317dcd122$export$29b7b5d6a5a497c5, (v) => $4921fa4317dcd122$export$29b7b5d6a5a497c5 = v);
$parcel$export($4921fa4317dcd122$exports, "is-hovered", () => $4921fa4317dcd122$export$b8813cd5d7824ce7, (v) => $4921fa4317dcd122$export$b8813cd5d7824ce7 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-Button-label", () => $4921fa4317dcd122$export$c081ba562e204942, (v) => $4921fa4317dcd122$export$c081ba562e204942 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-ActionButton-label", () => $4921fa4317dcd122$export$ebb4635c66bacdb9, (v) => $4921fa4317dcd122$export$ebb4635c66bacdb9 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-ActionButton-hold", () => $4921fa4317dcd122$export$4a4f01c443eff992, (v) => $4921fa4317dcd122$export$4a4f01c443eff992 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-ActionButton--quiet", () => $4921fa4317dcd122$export$40e1f009544bacf3, (v) => $4921fa4317dcd122$export$40e1f009544bacf3 = v);
$parcel$export($4921fa4317dcd122$exports, "is-disabled", () => $4921fa4317dcd122$export$d35bc1e505d1ebbf, (v) => $4921fa4317dcd122$export$d35bc1e505d1ebbf = v);
$parcel$export($4921fa4317dcd122$exports, "is-open", () => $4921fa4317dcd122$export$a9781837241c946d, (v) => $4921fa4317dcd122$export$a9781837241c946d = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-FieldButton--quiet", () => $4921fa4317dcd122$export$5ca7b274e4fff9fc, (v) => $4921fa4317dcd122$export$5ca7b274e4fff9fc = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-ClearButton--overBackground", () => $4921fa4317dcd122$export$116f2fccf941d34d, (v) => $4921fa4317dcd122$export$116f2fccf941d34d = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-ClearButton--small", () => $4921fa4317dcd122$export$49280ffa5236ba8b, (v) => $4921fa4317dcd122$export$49280ffa5236ba8b = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-Stepper", () => $4921fa4317dcd122$export$f49c123746e1fb8, (v) => $4921fa4317dcd122$export$f49c123746e1fb8 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-Stepper--isMobile", () => $4921fa4317dcd122$export$75e6f9c4d1d95421, (v) => $4921fa4317dcd122$export$75e6f9c4d1d95421 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-Stepper-button--stepUp", () => $4921fa4317dcd122$export$6c76d609d604c6e5, (v) => $4921fa4317dcd122$export$6c76d609d604c6e5 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-Stepper-button--stepDown", () => $4921fa4317dcd122$export$e4ad52f6ba4e317b, (v) => $4921fa4317dcd122$export$e4ad52f6ba4e317b = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-Stepper-field", () => $4921fa4317dcd122$export$faac6573b15f75d6, (v) => $4921fa4317dcd122$export$faac6573b15f75d6 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-Stepper-icon", () => $4921fa4317dcd122$export$639c56ba6082b615, (v) => $4921fa4317dcd122$export$639c56ba6082b615 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-Stepper--isQuiet", () => $4921fa4317dcd122$export$70ed0eb2ba7512f8, (v) => $4921fa4317dcd122$export$70ed0eb2ba7512f8 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-Stepper--showStepper", () => $4921fa4317dcd122$export$a4e366275982a12f, (v) => $4921fa4317dcd122$export$a4e366275982a12f = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-Stepper-input", () => $4921fa4317dcd122$export$8273fb931edd57d4, (v) => $4921fa4317dcd122$export$8273fb931edd57d4 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-Stepper-container", () => $4921fa4317dcd122$export$f49a6eb8460c3f93, (v) => $4921fa4317dcd122$export$f49a6eb8460c3f93 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-Stepper-container--isMobile", () => $4921fa4317dcd122$export$3486d98f7bb904b1, (v) => $4921fa4317dcd122$export$3486d98f7bb904b1 = v);
$parcel$export($4921fa4317dcd122$exports, "spectrum-Stepper-button-icon", () => $4921fa4317dcd122$export$7d7cb0c4055e98ab, (v) => $4921fa4317dcd122$export$7d7cb0c4055e98ab = v);
$parcel$export($4921fa4317dcd122$exports, "is-active", () => $4921fa4317dcd122$export$20fd0f7cd4e6112f, (v) => $4921fa4317dcd122$export$20fd0f7cd4e6112f = v);
$parcel$export($4921fa4317dcd122$exports, "is-focused", () => $4921fa4317dcd122$export$e7dc768d35940237, (v) => $4921fa4317dcd122$export$e7dc768d35940237 = v);
$parcel$export($4921fa4317dcd122$exports, "is-invalid", () => $4921fa4317dcd122$export$14a30de2866685fa, (v) => $4921fa4317dcd122$export$14a30de2866685fa = v);
var $4921fa4317dcd122$export$1db4cca5b4ade39a;
var $4921fa4317dcd122$export$53da69f51b770d3;
var $4921fa4317dcd122$export$36ae67a131730482;
var $4921fa4317dcd122$export$a024c99ecf6b5741;
var $4921fa4317dcd122$export$b752ce409e5660c;
var $4921fa4317dcd122$export$4e11596482f95f63;
var $4921fa4317dcd122$export$f39a09f249340e2a;
var $4921fa4317dcd122$export$d374b04f30360026;
var $4921fa4317dcd122$export$5aabb61a966e266f;
var $4921fa4317dcd122$export$29b7b5d6a5a497c5;
var $4921fa4317dcd122$export$b8813cd5d7824ce7;
var $4921fa4317dcd122$export$c081ba562e204942;
var $4921fa4317dcd122$export$ebb4635c66bacdb9;
var $4921fa4317dcd122$export$4a4f01c443eff992;
var $4921fa4317dcd122$export$40e1f009544bacf3;
var $4921fa4317dcd122$export$d35bc1e505d1ebbf;
var $4921fa4317dcd122$export$a9781837241c946d;
var $4921fa4317dcd122$export$5ca7b274e4fff9fc;
var $4921fa4317dcd122$export$116f2fccf941d34d;
var $4921fa4317dcd122$export$49280ffa5236ba8b;
var $4921fa4317dcd122$export$f49c123746e1fb8;
var $4921fa4317dcd122$export$75e6f9c4d1d95421;
var $4921fa4317dcd122$export$6c76d609d604c6e5;
var $4921fa4317dcd122$export$e4ad52f6ba4e317b;
var $4921fa4317dcd122$export$faac6573b15f75d6;
var $4921fa4317dcd122$export$639c56ba6082b615;
var $4921fa4317dcd122$export$70ed0eb2ba7512f8;
var $4921fa4317dcd122$export$a4e366275982a12f;
var $4921fa4317dcd122$export$8273fb931edd57d4;
var $4921fa4317dcd122$export$f49a6eb8460c3f93;
var $4921fa4317dcd122$export$3486d98f7bb904b1;
var $4921fa4317dcd122$export$7d7cb0c4055e98ab;
var $4921fa4317dcd122$export$20fd0f7cd4e6112f;
var $4921fa4317dcd122$export$e7dc768d35940237;
var $4921fa4317dcd122$export$14a30de2866685fa;
$4921fa4317dcd122$export$1db4cca5b4ade39a = "spectrum-Button_8890ad";
$4921fa4317dcd122$export$53da69f51b770d3 = "spectrum-ActionButton_8890ad";
$4921fa4317dcd122$export$36ae67a131730482 = "spectrum-LogicButton_8890ad";
$4921fa4317dcd122$export$a024c99ecf6b5741 = "spectrum-FieldButton_8890ad";
$4921fa4317dcd122$export$b752ce409e5660c = "spectrum-ClearButton_8890ad";
$4921fa4317dcd122$export$4e11596482f95f63 = "spectrum-Stepper-button_8890ad";
$4921fa4317dcd122$export$f39a09f249340e2a = "focus-ring_8890ad";
$4921fa4317dcd122$export$d374b04f30360026 = "spectrum-Icon_8890ad";
$4921fa4317dcd122$export$5aabb61a966e266f = "spectrum-ActionButton--emphasized_8890ad";
$4921fa4317dcd122$export$29b7b5d6a5a497c5 = "spectrum-ActionButton--staticColor_8890ad";
$4921fa4317dcd122$export$b8813cd5d7824ce7 = "is-hovered_8890ad";
$4921fa4317dcd122$export$c081ba562e204942 = "spectrum-Button-label_8890ad";
$4921fa4317dcd122$export$ebb4635c66bacdb9 = "spectrum-ActionButton-label_8890ad";
$4921fa4317dcd122$export$4a4f01c443eff992 = "spectrum-ActionButton-hold_8890ad";
$4921fa4317dcd122$export$40e1f009544bacf3 = "spectrum-ActionButton--quiet_8890ad";
$4921fa4317dcd122$export$d35bc1e505d1ebbf = "is-disabled_8890ad";
$4921fa4317dcd122$export$a9781837241c946d = "is-open_8890ad";
$4921fa4317dcd122$export$5ca7b274e4fff9fc = "spectrum-FieldButton--quiet_8890ad";
$4921fa4317dcd122$export$116f2fccf941d34d = "spectrum-ClearButton--overBackground_8890ad";
$4921fa4317dcd122$export$49280ffa5236ba8b = "spectrum-ClearButton--small_8890ad";
$4921fa4317dcd122$export$f49c123746e1fb8 = "spectrum-Stepper_8890ad";
$4921fa4317dcd122$export$75e6f9c4d1d95421 = "spectrum-Stepper--isMobile_8890ad";
$4921fa4317dcd122$export$6c76d609d604c6e5 = "spectrum-Stepper-button--stepUp_8890ad";
$4921fa4317dcd122$export$e4ad52f6ba4e317b = "spectrum-Stepper-button--stepDown_8890ad";
$4921fa4317dcd122$export$faac6573b15f75d6 = "spectrum-Stepper-field_8890ad";
$4921fa4317dcd122$export$639c56ba6082b615 = "spectrum-Stepper-icon_8890ad";
$4921fa4317dcd122$export$70ed0eb2ba7512f8 = "spectrum-Stepper--isQuiet_8890ad";
$4921fa4317dcd122$export$a4e366275982a12f = "spectrum-Stepper--showStepper_8890ad";
$4921fa4317dcd122$export$8273fb931edd57d4 = "spectrum-Stepper-input_8890ad";
$4921fa4317dcd122$export$f49a6eb8460c3f93 = "spectrum-Stepper-container_8890ad";
$4921fa4317dcd122$export$3486d98f7bb904b1 = "spectrum-Stepper-container--isMobile_8890ad";
$4921fa4317dcd122$export$7d7cb0c4055e98ab = "spectrum-Stepper-button-icon_8890ad";
$4921fa4317dcd122$export$20fd0f7cd4e6112f = "is-active_8890ad";
$4921fa4317dcd122$export$e7dc768d35940237 = "is-focused_8890ad";
$4921fa4317dcd122$export$14a30de2866685fa = "is-invalid_8890ad";





function $fc8b5aac78824257$var$StepButton(props, ref) {
    props = $k93dG$useProviderProps(props);
    let { scale: scale  } = $k93dG$useProvider();
    let { direction: direction , isDisabled: isDisabled , isQuiet: isQuiet  } = props;
    /**
   * Must use div for now because Safari pointer event bugs on disabled form elements.
   * Link https://bugs.webkit.org/show_bug.cgi?id=219188.
   */ let { buttonProps: buttonProps , isPressed: isPressed  } = $k93dG$useButton({
        ...props,
        elementType: 'div'
    }, ref);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $k93dG$useHover(props);
    return(/*#__PURE__*/ $k93dG$react.createElement($k93dG$FocusRing, {
        focusRingClass: $k93dG$classNames((/*@__PURE__*/$parcel$interopDefault($4921fa4317dcd122$exports)), 'focus-ring')
    }, /*#__PURE__*/ $k93dG$react.createElement("div", {
        className: $k93dG$classNames((/*@__PURE__*/$parcel$interopDefault($4921fa4317dcd122$exports)), 'spectrum-Stepper-button', {
            'spectrum-Stepper-button--stepUp': direction === 'up',
            'spectrum-Stepper-button--stepDown': direction === 'down',
            'spectrum-Stepper-button--isQuiet': isQuiet,
            'is-hovered': isHovered,
            'is-active': isPressed,
            'is-disabled': isDisabled
        }),
        ...$k93dG$mergeProps(hoverProps, buttonProps),
        ref: ref
    }, direction === 'up' && scale === 'large' && /*#__PURE__*/ $k93dG$react.createElement($k93dG$spectrumiconsworkflowAdd, {
        UNSAFE_className: $k93dG$classNames((/*@__PURE__*/$parcel$interopDefault($4921fa4317dcd122$exports)), 'spectrum-Stepper-button-icon', 'spectrum-Stepper-stepUpIcon'),
        size: "S"
    }), direction === 'up' && scale === 'medium' && /*#__PURE__*/ $k93dG$react.createElement($k93dG$spectrumiconsuiChevronUpSmall, {
        UNSAFE_className: $k93dG$classNames((/*@__PURE__*/$parcel$interopDefault($4921fa4317dcd122$exports)), 'spectrum-Stepper-button-icon', 'spectrum-Stepper-stepUpIcon')
    }), direction === 'down' && scale === 'large' && /*#__PURE__*/ $k93dG$react.createElement($k93dG$spectrumiconsworkflowRemove, {
        UNSAFE_className: $k93dG$classNames((/*@__PURE__*/$parcel$interopDefault($4921fa4317dcd122$exports)), 'spectrum-Stepper-button-icon', 'spectrum-Stepper-stepDownIcon'),
        size: "S"
    }), direction === 'down' && scale === 'medium' && /*#__PURE__*/ $k93dG$react.createElement($k93dG$spectrumiconsuiChevronDownSmall, {
        UNSAFE_className: $k93dG$classNames((/*@__PURE__*/$parcel$interopDefault($4921fa4317dcd122$exports)), 'spectrum-Stepper-button-icon', 'spectrum-Stepper-stepDownIcon')
    }))));
}
/**
 * Buttons for NumberField.
 */ let $fc8b5aac78824257$export$b2f6b60c1d32d6aa = /*#__PURE__*/ $k93dG$react.forwardRef($fc8b5aac78824257$var$StepButton);









function $5d632a544d0c6901$var$NumberField(props, ref) {
    props = $k93dG$useProviderProps(props);
    let provider = $k93dG$useProvider();
    let { isQuiet: isQuiet , isReadOnly: isReadOnly , isDisabled: isDisabled , hideStepper: hideStepper  } = props;
    let { styleProps: style  } = $k93dG$useStyleProps(props);
    let { locale: locale  } = $k93dG$useLocale();
    let state = $k93dG$useNumberFieldState({
        ...props,
        locale: locale
    });
    let inputRef = $k93dG$useRef();
    let domRef = $k93dG$useFocusableRef(ref, inputRef);
    let { groupProps: groupProps , labelProps: labelProps , inputProps: inputProps , incrementButtonProps: incrementButtonProps , decrementButtonProps: decrementButtonProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $k93dG$useNumberField(props, state, inputRef);
    let isMobile = provider.scale === 'large';
    let showStepper = !hideStepper;
    let { isHovered: isHovered , hoverProps: hoverProps  } = $k93dG$useHover({
        isDisabled: isDisabled
    });
    let className = $k93dG$classNames((/*@__PURE__*/$parcel$interopDefault($4921fa4317dcd122$exports)), 'spectrum-Stepper', {
        'spectrum-Stepper--isQuiet': isQuiet,
        'is-disabled': isDisabled,
        'spectrum-Stepper--readonly': isReadOnly,
        'is-invalid': props.validationState === 'invalid',
        'spectrum-Stepper--showStepper': showStepper,
        'spectrum-Stepper--isMobile': isMobile,
        'is-hovered': isHovered,
        // because FocusRing won't pass along the className from Field, we have to handle that ourselves
        [style.className]: !props.label && style.className
    });
    return(/*#__PURE__*/ $k93dG$react.createElement($k93dG$Field, {
        ...props,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        labelProps: labelProps,
        ref: domRef,
        wrapperClassName: $k93dG$classNames((/*@__PURE__*/$parcel$interopDefault($4921fa4317dcd122$exports)), 'spectrum-Stepper-container', {
            'spectrum-Stepper-container--isMobile': isMobile
        })
    }, /*#__PURE__*/ $k93dG$react.createElement($5d632a544d0c6901$var$NumberFieldInput, {
        ...props,
        groupProps: $k93dG$mergeProps(groupProps, hoverProps),
        inputProps: inputProps,
        inputRef: inputRef,
        incrementProps: incrementButtonProps,
        decrementProps: decrementButtonProps,
        className: className,
        style: style
    })));
}
const $5d632a544d0c6901$var$NumberFieldInput = /*#__PURE__*/ $k93dG$react.forwardRef(function NumberFieldInput(props, ref) {
    let { groupProps: groupProps , inputProps: inputProps , inputRef: inputRef , incrementProps: incrementProps , decrementProps: decrementProps , className: className , style: style , autoFocus: autoFocus , isQuiet: isQuiet , hideStepper: hideStepper , validationState: validationState  } = props;
    let showStepper = !hideStepper;
    return(/*#__PURE__*/ $k93dG$react.createElement($k93dG$FocusRing, {
        within: true,
        isTextInput: true,
        focusClass: $k93dG$classNames((/*@__PURE__*/$parcel$interopDefault($4921fa4317dcd122$exports)), 'is-focused'),
        focusRingClass: $k93dG$classNames((/*@__PURE__*/$parcel$interopDefault($4921fa4317dcd122$exports)), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ $k93dG$react.createElement("div", {
        ...groupProps,
        ref: ref,
        style: style,
        className: className
    }, /*#__PURE__*/ $k93dG$react.createElement($k93dG$TextFieldBase, {
        UNSAFE_className: $k93dG$classNames((/*@__PURE__*/$parcel$interopDefault($4921fa4317dcd122$exports)), 'spectrum-Stepper-field'),
        inputClassName: $k93dG$classNames((/*@__PURE__*/$parcel$interopDefault($4921fa4317dcd122$exports)), 'spectrum-Stepper-input'),
        validationIconClassName: $k93dG$classNames((/*@__PURE__*/$parcel$interopDefault($4921fa4317dcd122$exports)), 'spectrum-Stepper-icon'),
        isQuiet: isQuiet,
        inputRef: inputRef,
        validationState: validationState,
        inputProps: inputProps
    }), showStepper && /*#__PURE__*/ $k93dG$react.createElement($k93dG$react.Fragment, null, /*#__PURE__*/ $k93dG$react.createElement($fc8b5aac78824257$export$b2f6b60c1d32d6aa, {
        direction: "up",
        isQuiet: isQuiet,
        ...incrementProps
    }), /*#__PURE__*/ $k93dG$react.createElement($fc8b5aac78824257$export$b2f6b60c1d32d6aa, {
        direction: "down",
        isQuiet: isQuiet,
        ...decrementProps
    })))));
});
/**
 * NumberFields allow users to enter a number, and increment or decrement the value using stepper buttons.
 */ let $5d632a544d0c6901$export$63c5fa0b2fdccd2e = /*#__PURE__*/ $k93dG$react.forwardRef($5d632a544d0c6901$var$NumberField);




export {$5d632a544d0c6901$export$63c5fa0b2fdccd2e as NumberField};
