// @react-spectrum/text/dist/main.js
var $cn8pf$reactariautils = require("@react-aria/utils");
var $cn8pf$react = require("react");
var $cn8pf$reactspectrumutils = require("@react-spectrum/utils");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $acd02dd1a8b16753$exports = {};

$parcel$export($acd02dd1a8b16753$exports, "Text", () => $acd02dd1a8b16753$export$5f1af8db9871e1d6);



function $acd02dd1a8b16753$var$Text(props, ref) {
    props = $cn8pf$reactspectrumutils.useSlotProps(props, 'text');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $cn8pf$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $cn8pf$reactspectrumutils.useDOMRef(ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($cn8pf$react)).createElement("span", {
        ...$cn8pf$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, children));
}
/**
 * Text represents text with no specific semantic meaning.
 */ const $acd02dd1a8b16753$export$5f1af8db9871e1d6 = /*#__PURE__*/ $cn8pf$react.forwardRef($acd02dd1a8b16753$var$Text);


var $ba9bdabb573cb821$exports = {};

$parcel$export($ba9bdabb573cb821$exports, "Heading", () => $ba9bdabb573cb821$export$a8a3e93435678ff9);



function $ba9bdabb573cb821$var$Heading(props, ref) {
    props = $cn8pf$reactspectrumutils.useSlotProps(props, 'heading');
    let { children: children , level: level = 3 , ...otherProps } = props;
    let { styleProps: styleProps  } = $cn8pf$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $cn8pf$reactspectrumutils.useDOMRef(ref);
    let HeadingTag = `h${level}`;
    return(/*#__PURE__*/ ($parcel$interopDefault($cn8pf$react)).createElement(HeadingTag, {
        ...$cn8pf$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, children));
}
/**
 * Heading is used to create various levels of typographic hierarchies.
 */ const $ba9bdabb573cb821$export$a8a3e93435678ff9 = /*#__PURE__*/ $cn8pf$react.forwardRef($ba9bdabb573cb821$var$Heading);


var $98098b763ba271fd$exports = {};

$parcel$export($98098b763ba271fd$exports, "Keyboard", () => $98098b763ba271fd$export$16e4d70cc375e707);



function $98098b763ba271fd$var$Keyboard(props, ref) {
    props = $cn8pf$reactspectrumutils.useSlotProps(props, 'keyboard');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $cn8pf$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $cn8pf$reactspectrumutils.useDOMRef(ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($cn8pf$react)).createElement("kbd", {
        ...$cn8pf$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        dir: "ltr",
        ref: domRef
    }, children));
}
/**
 * Keyboard represents text that specifies a keyboard command.
 */ const $98098b763ba271fd$export$16e4d70cc375e707 = /*#__PURE__*/ $cn8pf$react.forwardRef($98098b763ba271fd$var$Keyboard);


$parcel$exportWildcard(module.exports, $acd02dd1a8b16753$exports);
$parcel$exportWildcard(module.exports, $ba9bdabb573cb821$exports);
$parcel$exportWildcard(module.exports, $98098b763ba271fd$exports);
