// @react-spectrum/text/dist/module.js
import {filterDOMProps as $hgL9i$filterDOMProps} from "@react-aria/utils";
import $hgL9i$react, {forwardRef as $hgL9i$forwardRef} from "react";
import {useSlotProps as $hgL9i$useSlotProps, useStyleProps as $hgL9i$useStyleProps, useDOMRef as $hgL9i$useDOMRef} from "@react-spectrum/utils";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $2f505f60703ca8dc$exports = {};

$parcel$export($2f505f60703ca8dc$exports, "Text", () => $2f505f60703ca8dc$export$5f1af8db9871e1d6);



function $2f505f60703ca8dc$var$Text(props, ref) {
    props = $hgL9i$useSlotProps(props, 'text');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $hgL9i$useStyleProps(otherProps);
    let domRef = $hgL9i$useDOMRef(ref);
    return(/*#__PURE__*/ $hgL9i$react.createElement("span", {
        ...$hgL9i$filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, children));
}
/**
 * Text represents text with no specific semantic meaning.
 */ const $2f505f60703ca8dc$export$5f1af8db9871e1d6 = /*#__PURE__*/ $hgL9i$forwardRef($2f505f60703ca8dc$var$Text);


var $f62b4ac2b080fa8f$exports = {};

$parcel$export($f62b4ac2b080fa8f$exports, "Heading", () => $f62b4ac2b080fa8f$export$a8a3e93435678ff9);



function $f62b4ac2b080fa8f$var$Heading(props, ref) {
    props = $hgL9i$useSlotProps(props, 'heading');
    let { children: children , level: level = 3 , ...otherProps } = props;
    let { styleProps: styleProps  } = $hgL9i$useStyleProps(otherProps);
    let domRef = $hgL9i$useDOMRef(ref);
    let HeadingTag = `h${level}`;
    return(/*#__PURE__*/ $hgL9i$react.createElement(HeadingTag, {
        ...$hgL9i$filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, children));
}
/**
 * Heading is used to create various levels of typographic hierarchies.
 */ const $f62b4ac2b080fa8f$export$a8a3e93435678ff9 = /*#__PURE__*/ $hgL9i$forwardRef($f62b4ac2b080fa8f$var$Heading);


var $9db65624fc9b5771$exports = {};

$parcel$export($9db65624fc9b5771$exports, "Keyboard", () => $9db65624fc9b5771$export$16e4d70cc375e707);



function $9db65624fc9b5771$var$Keyboard(props, ref) {
    props = $hgL9i$useSlotProps(props, 'keyboard');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $hgL9i$useStyleProps(otherProps);
    let domRef = $hgL9i$useDOMRef(ref);
    return(/*#__PURE__*/ $hgL9i$react.createElement("kbd", {
        ...$hgL9i$filterDOMProps(otherProps),
        ...styleProps,
        dir: "ltr",
        ref: domRef
    }, children));
}
/**
 * Keyboard represents text that specifies a keyboard command.
 */ const $9db65624fc9b5771$export$16e4d70cc375e707 = /*#__PURE__*/ $hgL9i$forwardRef($9db65624fc9b5771$var$Keyboard);




export {$2f505f60703ca8dc$export$5f1af8db9871e1d6 as Text, $f62b4ac2b080fa8f$export$a8a3e93435678ff9 as Heading, $9db65624fc9b5771$export$16e4d70cc375e707 as Keyboard};
