// @react-spectrum/dialog/dist/main.js
require("./main.css");
var $aGPKm$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $aGPKm$reactspectrumbutton = require("@react-spectrum/button");
var $aGPKm$reactspectrumbuttongroup = require("@react-spectrum/buttongroup");
var $aGPKm$reactariautils = require("@react-aria/utils");
var $aGPKm$reactspectrumutils = require("@react-spectrum/utils");
var $aGPKm$reactspectrumview = require("@react-spectrum/view");
var $aGPKm$reactspectrumdivider = require("@react-spectrum/divider");
var $aGPKm$reactspectrumtext = require("@react-spectrum/text");
var $aGPKm$react = require("react");
var $aGPKm$reactariai18n = require("@react-aria/i18n");
var $aGPKm$spectrumiconsuiCrossLarge = require("@spectrum-icons/ui/CrossLarge");
var $aGPKm$reactariaoverlays = require("@react-aria/overlays");
var $aGPKm$reactariafocus = require("@react-aria/focus");
var $aGPKm$reactspectrumlayout = require("@react-spectrum/layout");
var $aGPKm$reactariadialog = require("@react-aria/dialog");
var $aGPKm$reactspectrumoverlays = require("@react-spectrum/overlays");
var $aGPKm$reactstatelyoverlays = require("@react-stately/overlays");
var $aGPKm$reactariainteractions = require("@react-aria/interactions");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $1bf9453c78052870$exports = {};

$parcel$export($1bf9453c78052870$exports, "AlertDialog", () => $1bf9453c78052870$export$de466dd8317b0b75);






var $962c86794f106020$exports = {};

$parcel$export($962c86794f106020$exports, "Dialog", () => $962c86794f106020$export$3ddf2d174ce01153);




const $59fcf3b6a0bdb727$export$8b93a07348a7730c = ($parcel$interopDefault($aGPKm$react)).createContext(null);





var $7a223c7618933de9$exports = {};
var $65ab63371501fc94$exports = {};
$65ab63371501fc94$exports = JSON.parse("{\"alert\":\"تنبيه\",\"dismiss\":\"تجاهل\"}");


var $9a456c34c8316043$exports = {};
$9a456c34c8316043$exports = JSON.parse("{\"alert\":\"Сигнал\",\"dismiss\":\"Отхвърляне\"}");


var $c7cc7b4686b5bf4d$exports = {};
$c7cc7b4686b5bf4d$exports = JSON.parse("{\"alert\":\"Výstraha\",\"dismiss\":\"Odstranit\"}");


var $f5666c1c51cebc8d$exports = {};
$f5666c1c51cebc8d$exports = JSON.parse("{\"alert\":\"Advarsel\",\"dismiss\":\"Luk\"}");


var $0a5b12eb5fa7ff54$exports = {};
$0a5b12eb5fa7ff54$exports = JSON.parse("{\"alert\":\"Warnhinweis\",\"dismiss\":\"Schließen\"}");


var $9dcd6e164b9b77bf$exports = {};
$9dcd6e164b9b77bf$exports = JSON.parse("{\"alert\":\"Ειδοποίηση\",\"dismiss\":\"Απόρριψη\"}");


var $806b92dcb141b7f9$exports = {};
$806b92dcb141b7f9$exports = JSON.parse("{\"dismiss\":\"Dismiss\",\"alert\":\"Alert\"}");


var $c35f2338054c8a87$exports = {};
$c35f2338054c8a87$exports = JSON.parse("{\"alert\":\"Alerta\",\"dismiss\":\"Descartar\"}");


var $f1c5a1d56c0c0b4c$exports = {};
$f1c5a1d56c0c0b4c$exports = JSON.parse("{\"alert\":\"Teade\",\"dismiss\":\"Lõpeta\"}");


var $5f0eef29036ba1a2$exports = {};
$5f0eef29036ba1a2$exports = JSON.parse("{\"alert\":\"Hälytys\",\"dismiss\":\"Hylkää\"}");


var $2fdae957caba1dd0$exports = {};
$2fdae957caba1dd0$exports = JSON.parse("{\"alert\":\"Alerte\",\"dismiss\":\"Rejeter\"}");


var $a45676ea154dc510$exports = {};
$a45676ea154dc510$exports = JSON.parse("{\"alert\":\"התראה\",\"dismiss\":\"התעלם\"}");


var $ed7ae8acce194f82$exports = {};
$ed7ae8acce194f82$exports = JSON.parse("{\"alert\":\"Upozorenje\",\"dismiss\":\"Odbaci\"}");


var $c231a41de85c9e21$exports = {};
$c231a41de85c9e21$exports = JSON.parse("{\"alert\":\"Figyelmeztetés\",\"dismiss\":\"Elutasítás\"}");


var $5a93dca271af449e$exports = {};
$5a93dca271af449e$exports = JSON.parse("{\"alert\":\"Avviso\",\"dismiss\":\"Ignora\"}");


var $980ee5844dd13e49$exports = {};
$980ee5844dd13e49$exports = JSON.parse("{\"alert\":\"アラート\",\"dismiss\":\"閉じる\"}");


var $8541436f4f0d5417$exports = {};
$8541436f4f0d5417$exports = JSON.parse("{\"alert\":\"경고\",\"dismiss\":\"무시\"}");


var $899fcc0320226522$exports = {};
$899fcc0320226522$exports = JSON.parse("{\"alert\":\"Įspėjimas\",\"dismiss\":\"Atmesti\"}");


var $6d22160168494c97$exports = {};
$6d22160168494c97$exports = JSON.parse("{\"alert\":\"Brīdinājums\",\"dismiss\":\"Nerādīt\"}");


var $6115bcfbc3e7b614$exports = {};
$6115bcfbc3e7b614$exports = JSON.parse("{\"alert\":\"Varsel\",\"dismiss\":\"Lukk\"}");


var $6e8f6056f38e9d69$exports = {};
$6e8f6056f38e9d69$exports = JSON.parse("{\"alert\":\"Melding\",\"dismiss\":\"Negeren\"}");


var $6d7c87d4d14b55a8$exports = {};
$6d7c87d4d14b55a8$exports = JSON.parse("{\"alert\":\"Ostrzeżenie\",\"dismiss\":\"Zignoruj\"}");


var $9ce1a7b94040ddee$exports = {};
$9ce1a7b94040ddee$exports = JSON.parse("{\"alert\":\"Alerta\",\"dismiss\":\"Descartar\"}");


var $5adcb3d147496a35$exports = {};
$5adcb3d147496a35$exports = JSON.parse("{\"alert\":\"Alerta\",\"dismiss\":\"Dispensar\"}");


var $0364081f7d32b7b3$exports = {};
$0364081f7d32b7b3$exports = JSON.parse("{\"alert\":\"Alertă\",\"dismiss\":\"Revocare\"}");


var $c219f19717f7d288$exports = {};
$c219f19717f7d288$exports = JSON.parse("{\"alert\":\"Предупреждение\",\"dismiss\":\"Пропустить\"}");


var $5e33ca1bfcdc0e4d$exports = {};
$5e33ca1bfcdc0e4d$exports = JSON.parse("{\"alert\":\"Upozornenie\",\"dismiss\":\"Zrušiť\"}");


var $6474981714d9c876$exports = {};
$6474981714d9c876$exports = JSON.parse("{\"alert\":\"Opozorilo\",\"dismiss\":\"Opusti\"}");


var $9e0d49d63cf9110a$exports = {};
$9e0d49d63cf9110a$exports = JSON.parse("{\"alert\":\"Upozorenje\",\"dismiss\":\"Odbaci\"}");


var $35e84a06c3231a1c$exports = {};
$35e84a06c3231a1c$exports = JSON.parse("{\"alert\":\"Varning\",\"dismiss\":\"Avvisa\"}");


var $ba9b89c43b3c337b$exports = {};
$ba9b89c43b3c337b$exports = JSON.parse("{\"alert\":\"Uyarı\",\"dismiss\":\"Kapat\"}");


var $e268bf65b03c0881$exports = {};
$e268bf65b03c0881$exports = JSON.parse("{\"alert\":\"Сигнал тривоги\",\"dismiss\":\"Скасувати\"}");


var $4c79597104e1f44c$exports = {};
$4c79597104e1f44c$exports = JSON.parse("{\"alert\":\"警报\",\"dismiss\":\"取消\"}");


var $0e5fd216d30024c9$exports = {};
$0e5fd216d30024c9$exports = JSON.parse("{\"alert\":\"警示\",\"dismiss\":\"關閉\"}");


$7a223c7618933de9$exports = {
    "ar-AE": $65ab63371501fc94$exports,
    "bg-BG": $9a456c34c8316043$exports,
    "cs-CZ": $c7cc7b4686b5bf4d$exports,
    "da-DK": $f5666c1c51cebc8d$exports,
    "de-DE": $0a5b12eb5fa7ff54$exports,
    "el-GR": $9dcd6e164b9b77bf$exports,
    "en-US": $806b92dcb141b7f9$exports,
    "es-ES": $c35f2338054c8a87$exports,
    "et-EE": $f1c5a1d56c0c0b4c$exports,
    "fi-FI": $5f0eef29036ba1a2$exports,
    "fr-FR": $2fdae957caba1dd0$exports,
    "he-IL": $a45676ea154dc510$exports,
    "hr-HR": $ed7ae8acce194f82$exports,
    "hu-HU": $c231a41de85c9e21$exports,
    "it-IT": $5a93dca271af449e$exports,
    "ja-JP": $980ee5844dd13e49$exports,
    "ko-KR": $8541436f4f0d5417$exports,
    "lt-LT": $899fcc0320226522$exports,
    "lv-LV": $6d22160168494c97$exports,
    "nb-NO": $6115bcfbc3e7b614$exports,
    "nl-NL": $6e8f6056f38e9d69$exports,
    "pl-PL": $6d7c87d4d14b55a8$exports,
    "pt-BR": $9ce1a7b94040ddee$exports,
    "pt-PT": $5adcb3d147496a35$exports,
    "ro-RO": $0364081f7d32b7b3$exports,
    "ru-RU": $c219f19717f7d288$exports,
    "sk-SK": $5e33ca1bfcdc0e4d$exports,
    "sl-SI": $6474981714d9c876$exports,
    "sr-SP": $9e0d49d63cf9110a$exports,
    "sv-SE": $35e84a06c3231a1c$exports,
    "tr-TR": $ba9b89c43b3c337b$exports,
    "uk-UA": $e268bf65b03c0881$exports,
    "zh-CN": $4c79597104e1f44c$exports,
    "zh-TW": $0e5fd216d30024c9$exports
};




var $e4ca8feea1f82112$exports = {};

$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog", () => $e4ca8feea1f82112$export$4681f3bcac39b2b1, (v) => $e4ca8feea1f82112$export$4681f3bcac39b2b1 = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog--small", () => $e4ca8feea1f82112$export$5f2d9cc47409bd63, (v) => $e4ca8feea1f82112$export$5f2d9cc47409bd63 = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog--medium", () => $e4ca8feea1f82112$export$b9b8ef443d0a67ca, (v) => $e4ca8feea1f82112$export$b9b8ef443d0a67ca = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog--large", () => $e4ca8feea1f82112$export$b73c0c2fca84be2b, (v) => $e4ca8feea1f82112$export$b73c0c2fca84be2b = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog-hero", () => $e4ca8feea1f82112$export$d405fb0a0231aa1a, (v) => $e4ca8feea1f82112$export$d405fb0a0231aa1a = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog-grid", () => $e4ca8feea1f82112$export$c03cd5e23af67ff5, (v) => $e4ca8feea1f82112$export$c03cd5e23af67ff5 = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog-heading", () => $e4ca8feea1f82112$export$94335d73b7b209ec, (v) => $e4ca8feea1f82112$export$94335d73b7b209ec = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog-heading--noHeader", () => $e4ca8feea1f82112$export$ade8307239736e89, (v) => $e4ca8feea1f82112$export$ade8307239736e89 = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog-header", () => $e4ca8feea1f82112$export$1e158eaa6363e131, (v) => $e4ca8feea1f82112$export$1e158eaa6363e131 = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog-typeIcon", () => $e4ca8feea1f82112$export$5beee9480fc6ff07, (v) => $e4ca8feea1f82112$export$5beee9480fc6ff07 = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog-divider", () => $e4ca8feea1f82112$export$dd7633d642d0bc1c, (v) => $e4ca8feea1f82112$export$dd7633d642d0bc1c = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog--noDivider", () => $e4ca8feea1f82112$export$22425527a8943917, (v) => $e4ca8feea1f82112$export$22425527a8943917 = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog-content", () => $e4ca8feea1f82112$export$cb8eccd0f3639238, (v) => $e4ca8feea1f82112$export$cb8eccd0f3639238 = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog-footer", () => $e4ca8feea1f82112$export$3ecb5025a06fe08f, (v) => $e4ca8feea1f82112$export$3ecb5025a06fe08f = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Button", () => $e4ca8feea1f82112$export$1db4cca5b4ade39a, (v) => $e4ca8feea1f82112$export$1db4cca5b4ade39a = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog-buttonGroup", () => $e4ca8feea1f82112$export$570ae852861555b7, (v) => $e4ca8feea1f82112$export$570ae852861555b7 = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog-buttonGroup--noFooter", () => $e4ca8feea1f82112$export$39dd75369bf2e830, (v) => $e4ca8feea1f82112$export$39dd75369bf2e830 = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog--dismissable", () => $e4ca8feea1f82112$export$9461b285d631d575, (v) => $e4ca8feea1f82112$export$9461b285d631d575 = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog-closeButton", () => $e4ca8feea1f82112$export$a8494287d763a75d, (v) => $e4ca8feea1f82112$export$a8494287d763a75d = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog--error", () => $e4ca8feea1f82112$export$6b5beff56972a3b, (v) => $e4ca8feea1f82112$export$6b5beff56972a3b = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog--fullscreen", () => $e4ca8feea1f82112$export$118506da5d3ee7c4, (v) => $e4ca8feea1f82112$export$118506da5d3ee7c4 = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog--fullscreenTakeover", () => $e4ca8feea1f82112$export$5c061b2d3664fda9, (v) => $e4ca8feea1f82112$export$5c061b2d3664fda9 = v);
$parcel$export($e4ca8feea1f82112$exports, "spectrum-Dialog--warning", () => $e4ca8feea1f82112$export$4e66396adf61534f, (v) => $e4ca8feea1f82112$export$4e66396adf61534f = v);
var $e4ca8feea1f82112$export$4681f3bcac39b2b1;
var $e4ca8feea1f82112$export$5f2d9cc47409bd63;
var $e4ca8feea1f82112$export$b9b8ef443d0a67ca;
var $e4ca8feea1f82112$export$b73c0c2fca84be2b;
var $e4ca8feea1f82112$export$d405fb0a0231aa1a;
var $e4ca8feea1f82112$export$c03cd5e23af67ff5;
var $e4ca8feea1f82112$export$94335d73b7b209ec;
var $e4ca8feea1f82112$export$ade8307239736e89;
var $e4ca8feea1f82112$export$1e158eaa6363e131;
var $e4ca8feea1f82112$export$5beee9480fc6ff07;
var $e4ca8feea1f82112$export$dd7633d642d0bc1c;
var $e4ca8feea1f82112$export$22425527a8943917;
var $e4ca8feea1f82112$export$cb8eccd0f3639238;
var $e4ca8feea1f82112$export$3ecb5025a06fe08f;
var $e4ca8feea1f82112$export$1db4cca5b4ade39a;
var $e4ca8feea1f82112$export$570ae852861555b7;
var $e4ca8feea1f82112$export$39dd75369bf2e830;
var $e4ca8feea1f82112$export$9461b285d631d575;
var $e4ca8feea1f82112$export$a8494287d763a75d;
var $e4ca8feea1f82112$export$6b5beff56972a3b;
var $e4ca8feea1f82112$export$118506da5d3ee7c4;
var $e4ca8feea1f82112$export$5c061b2d3664fda9;
var $e4ca8feea1f82112$export$4e66396adf61534f;
$e4ca8feea1f82112$export$4681f3bcac39b2b1 = "spectrum-Dialog_6d8b48";
$e4ca8feea1f82112$export$5f2d9cc47409bd63 = "spectrum-Dialog--small_6d8b48";
$e4ca8feea1f82112$export$b9b8ef443d0a67ca = "spectrum-Dialog--medium_6d8b48";
$e4ca8feea1f82112$export$b73c0c2fca84be2b = "spectrum-Dialog--large_6d8b48";
$e4ca8feea1f82112$export$d405fb0a0231aa1a = "spectrum-Dialog-hero_6d8b48";
$e4ca8feea1f82112$export$c03cd5e23af67ff5 = "spectrum-Dialog-grid_6d8b48";
$e4ca8feea1f82112$export$94335d73b7b209ec = "spectrum-Dialog-heading_6d8b48";
$e4ca8feea1f82112$export$ade8307239736e89 = "spectrum-Dialog-heading--noHeader_6d8b48";
$e4ca8feea1f82112$export$1e158eaa6363e131 = "spectrum-Dialog-header_6d8b48";
$e4ca8feea1f82112$export$5beee9480fc6ff07 = "spectrum-Dialog-typeIcon_6d8b48";
$e4ca8feea1f82112$export$dd7633d642d0bc1c = "spectrum-Dialog-divider_6d8b48";
$e4ca8feea1f82112$export$22425527a8943917 = "spectrum-Dialog--noDivider_6d8b48";
$e4ca8feea1f82112$export$cb8eccd0f3639238 = "spectrum-Dialog-content_6d8b48";
$e4ca8feea1f82112$export$3ecb5025a06fe08f = "spectrum-Dialog-footer_6d8b48";
$e4ca8feea1f82112$export$1db4cca5b4ade39a = "spectrum-Button_6d8b48";
$e4ca8feea1f82112$export$570ae852861555b7 = "spectrum-Dialog-buttonGroup_6d8b48";
$e4ca8feea1f82112$export$39dd75369bf2e830 = "spectrum-Dialog-buttonGroup--noFooter_6d8b48";
$e4ca8feea1f82112$export$9461b285d631d575 = "spectrum-Dialog--dismissable_6d8b48";
$e4ca8feea1f82112$export$a8494287d763a75d = "spectrum-Dialog-closeButton_6d8b48";
$e4ca8feea1f82112$export$6b5beff56972a3b = "spectrum-Dialog--error_6d8b48";
$e4ca8feea1f82112$export$118506da5d3ee7c4 = "spectrum-Dialog--fullscreen_6d8b48";
$e4ca8feea1f82112$export$5c061b2d3664fda9 = "spectrum-Dialog--fullscreenTakeover_6d8b48";
$e4ca8feea1f82112$export$4e66396adf61534f = "spectrum-Dialog--warning_6d8b48";




let $962c86794f106020$var$sizeMap = {
    S: 'small',
    M: 'medium',
    L: 'large',
    fullscreen: 'fullscreen',
    fullscreenTakeover: 'fullscreenTakeover'
};
function $962c86794f106020$var$Dialog(props, ref) {
    let { type: type = 'modal' , ...contextProps } = $aGPKm$react.useContext($59fcf3b6a0bdb727$export$8b93a07348a7730c) || {
    };
    let { children: children , isDismissable: isDismissable = contextProps.isDismissable , onDismiss: onDismiss = contextProps.onClose , size: size , ...otherProps } = props;
    let formatMessage = $aGPKm$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($7a223c7618933de9$exports)));
    let { styleProps: styleProps  } = $aGPKm$reactspectrumutils.useStyleProps(otherProps);
    size = type === 'popover' ? size || 'S' : size || 'L';
    let domRef = $aGPKm$reactspectrumutils.useDOMRef(ref);
    let gridRef = $aGPKm$react.useRef();
    let sizeVariant = $962c86794f106020$var$sizeMap[type] || $962c86794f106020$var$sizeMap[size];
    let { dialogProps: dialogProps , titleProps: titleProps  } = $aGPKm$reactariadialog.useDialog($aGPKm$reactariautils.mergeProps(contextProps, props), domRef);
    let hasHeader = $aGPKm$reactspectrumutils.useHasChild(`.${(/*@__PURE__*/$parcel$interopDefault($e4ca8feea1f82112$exports))['spectrum-Dialog-header']}`, $aGPKm$reactspectrumutils.unwrapDOMRef(gridRef));
    let hasFooter = $aGPKm$reactspectrumutils.useHasChild(`.${(/*@__PURE__*/$parcel$interopDefault($e4ca8feea1f82112$exports))['spectrum-Dialog-footer']}`, $aGPKm$reactspectrumutils.unwrapDOMRef(gridRef));
    let slots = $aGPKm$react.useMemo(()=>({
            hero: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($e4ca8feea1f82112$exports))['spectrum-Dialog-hero']
            },
            header: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($e4ca8feea1f82112$exports))['spectrum-Dialog-header']
            },
            heading: {
                UNSAFE_className: $aGPKm$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($e4ca8feea1f82112$exports)), 'spectrum-Dialog-heading', {
                    'spectrum-Dialog-heading--noHeader': !hasHeader
                }),
                level: 2,
                ...titleProps
            },
            typeIcon: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($e4ca8feea1f82112$exports))['spectrum-Dialog-typeIcon']
            },
            divider: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($e4ca8feea1f82112$exports))['spectrum-Dialog-divider'],
                size: 'M'
            },
            content: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($e4ca8feea1f82112$exports))['spectrum-Dialog-content']
            },
            footer: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($e4ca8feea1f82112$exports))['spectrum-Dialog-footer']
            },
            buttonGroup: {
                UNSAFE_className: $aGPKm$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($e4ca8feea1f82112$exports)), 'spectrum-Dialog-buttonGroup', {
                    'spectrum-Dialog-buttonGroup--noFooter': !hasFooter
                }),
                align: 'end'
            }
        })
    , [
        (/*@__PURE__*/$parcel$interopDefault($e4ca8feea1f82112$exports)),
        hasFooter,
        hasHeader,
        titleProps
    ]);
    // If rendered in a popover or tray there won't be a visible dismiss button,
    // so we render a hidden one for screen readers.
    let dismissButton;
    if (type === 'popover' || type === 'tray') dismissButton = /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactariaoverlays.DismissButton, {
        onDismiss: onDismiss
    });
    return(/*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactariafocus.FocusScope, {
        contain: true,
        restoreFocus: true
    }, /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement("section", {
        ...styleProps,
        ...dialogProps,
        className: $aGPKm$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($e4ca8feea1f82112$exports)), 'spectrum-Dialog', {
            [`spectrum-Dialog--${sizeVariant}`]: sizeVariant,
            'spectrum-Dialog--dismissable': isDismissable
        }, styleProps.className),
        ref: domRef
    }, /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactspectrumlayout.Grid, {
        ref: gridRef,
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($e4ca8feea1f82112$exports))['spectrum-Dialog-grid']
    }, /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactspectrumutils.SlotProvider, {
        slots: slots
    }, children), isDismissable && /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactspectrumbutton.ActionButton, {
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($e4ca8feea1f82112$exports))['spectrum-Dialog-closeButton'],
        isQuiet: true,
        "aria-label": formatMessage('dismiss'),
        onPress: onDismiss
    }, /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement(($parcel$interopDefault($aGPKm$spectrumiconsuiCrossLarge)), null))), dismissButton)));
}
/**
 * Dialogs are windows containing contextual information, tasks, or workflows that appear over the user interface.
 * Depending on the kind of Dialog, further interactions may be blocked until the Dialog is acknowledged.
 */ let $962c86794f106020$export$3ddf2d174ce01153 = /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).forwardRef($962c86794f106020$var$Dialog);









/**
 * AlertDialogs are a specific type of Dialog. They display important information that users need to acknowledge.
 */ function $1bf9453c78052870$var$AlertDialog(props, ref) {
    let { onClose: onClose = ()=>{
    }  } = $aGPKm$react.useContext($59fcf3b6a0bdb727$export$8b93a07348a7730c) || {
    };
    let { variant: variant , children: children , primaryActionLabel: primaryActionLabel , secondaryActionLabel: secondaryActionLabel , cancelLabel: cancelLabel , autoFocusButton: autoFocusButton , title: title , isPrimaryActionDisabled: isPrimaryActionDisabled , isSecondaryActionDisabled: isSecondaryActionDisabled , onCancel: onCancel = ()=>{
    } , onPrimaryAction: onPrimaryAction = ()=>{
    } , onSecondaryAction: onSecondaryAction = ()=>{
    } , ...otherProps } = props;
    let { styleProps: styleProps  } = $aGPKm$reactspectrumutils.useStyleProps(otherProps);
    let formatMessage = $aGPKm$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($7a223c7618933de9$exports)));
    let confirmVariant = 'primary';
    if (variant) {
        if (variant === 'confirmation') confirmVariant = 'cta';
        else if (variant === 'destructive') confirmVariant = 'negative';
    }
    return(/*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($962c86794f106020$export$3ddf2d174ce01153, {
        UNSAFE_style: styleProps.style,
        UNSAFE_className: $aGPKm$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($e4ca8feea1f82112$exports)), {
            [`spectrum-Dialog--${variant}`]: variant
        }, styleProps.className),
        isHidden: styleProps.hidden,
        size: "M",
        role: "alertdialog",
        ref: ref
    }, /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactspectrumtext.Heading, null, title), (variant === 'error' || variant === 'warning') && /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement(($parcel$interopDefault($aGPKm$spectrumiconsuiAlertMedium)), {
        slot: "typeIcon",
        "aria-label": formatMessage('alert')
    }), /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactspectrumdivider.Divider, null), /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactspectrumview.Content, null, children), /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactspectrumbuttongroup.ButtonGroup, {
        align: "end"
    }, cancelLabel && /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactspectrumbutton.Button, {
        variant: "secondary",
        onPress: ()=>$aGPKm$reactariautils.chain(onClose(), onCancel())
        ,
        autoFocus: autoFocusButton === 'cancel'
    }, cancelLabel), secondaryActionLabel && /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactspectrumbutton.Button, {
        variant: "secondary",
        onPress: ()=>$aGPKm$reactariautils.chain(onClose(), onSecondaryAction())
        ,
        isDisabled: isSecondaryActionDisabled,
        autoFocus: autoFocusButton === 'secondary'
    }, secondaryActionLabel), /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactspectrumbutton.Button, {
        variant: confirmVariant,
        onPress: ()=>$aGPKm$reactariautils.chain(onClose(), onPrimaryAction())
        ,
        isDisabled: isPrimaryActionDisabled,
        autoFocus: autoFocusButton === 'primary'
    }, primaryActionLabel))));
}
/**
 * AlertDialogs are a specific type of Dialog. They display important information that users need to acknowledge.
 */ let $1bf9453c78052870$export$de466dd8317b0b75 = /*#__PURE__*/ $aGPKm$react.forwardRef($1bf9453c78052870$var$AlertDialog);



var $27e3a817835df4bb$exports = {};

$parcel$export($27e3a817835df4bb$exports, "DialogTrigger", () => $27e3a817835df4bb$export$2e1e1122cf0cba88);







function $27e3a817835df4bb$var$DialogTrigger(props) {
    let { children: children , type: type = 'modal' , mobileType: mobileType = type === 'popover' ? 'modal' : type , hideArrow: hideArrow , targetRef: targetRef , isDismissable: isDismissable , isKeyboardDismissDisabled: isKeyboardDismissDisabled , ...positionProps } = props;
    if (!Array.isArray(children) || children.length > 2) throw new Error('DialogTrigger must have exactly 2 children');
    // if a function is passed as the second child, it won't appear in toArray
    let [trigger, content] = children;
    // On small devices, show a modal or tray instead of a popover.
    // TODO: DNA variable?
    let isMobile = $aGPKm$reactspectrumutils.useMediaQuery('(max-width: 700px)');
    if (isMobile) {
        // handle cases where desktop popovers need a close button for the mobile modal view
        if (type !== 'modal' && mobileType === 'modal') isDismissable = true;
        type = mobileType;
    }
    let state = $aGPKm$reactstatelyoverlays.useOverlayTriggerState(props);
    let wasOpen = $aGPKm$react.useRef(false);
    wasOpen.current = state.isOpen;
    let isExiting = $aGPKm$react.useRef(false);
    let onExiting = ()=>isExiting.current = true
    ;
    let onExited = ()=>isExiting.current = false
    ;
    // eslint-disable-next-line arrow-body-style
    $aGPKm$react.useEffect(()=>{
        return ()=>{
            if ((wasOpen.current || isExiting.current) && type !== 'popover' && type !== 'tray') console.warn('A DialogTrigger unmounted while open. This is likely due to being placed within a trigger that unmounts or inside a conditional. Consider using a DialogContainer instead.');
        };
    }, []);
    if (type === 'popover') return(/*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($27e3a817835df4bb$var$PopoverTrigger, {
        ...positionProps,
        state: state,
        targetRef: targetRef,
        trigger: trigger,
        content: content,
        isKeyboardDismissDisabled: isKeyboardDismissDisabled,
        hideArrow: hideArrow
    }));
    let renderOverlay = ()=>{
        switch(type){
            case 'fullscreen':
            case 'fullscreenTakeover':
            case 'modal':
                return(/*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactspectrumoverlays.Modal, {
                    isOpen: state.isOpen,
                    isDismissable: type === 'modal' ? isDismissable : false,
                    onClose: state.close,
                    type: type,
                    isKeyboardDismissDisabled: isKeyboardDismissDisabled,
                    onExiting: onExiting,
                    onExited: onExited
                }, typeof content === 'function' ? content(state.close) : content));
            case 'tray':
                return(/*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactspectrumoverlays.Tray, {
                    isOpen: state.isOpen,
                    onClose: state.close,
                    isKeyboardDismissDisabled: isKeyboardDismissDisabled
                }, typeof content === 'function' ? content(state.close) : content));
        }
    };
    return(/*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($27e3a817835df4bb$var$DialogTriggerBase, {
        type: type,
        state: state,
        isDismissable: isDismissable,
        trigger: trigger,
        overlay: renderOverlay()
    }));
}
// Support DialogTrigger inside components using CollectionBuilder.
$27e3a817835df4bb$var$DialogTrigger.getCollectionNode = function*(props) {
    let [trigger] = ($parcel$interopDefault($aGPKm$react)).Children.toArray(props.children);
    let [, content] = props.children;
    yield {
        element: trigger,
        wrapper: (element)=>/*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($27e3a817835df4bb$var$DialogTrigger, {
                key: element.key,
                ...props
            }, element, content)
    };
};
/**
 * DialogTrigger serves as a wrapper around a Dialog and its associated trigger, linking the Dialog's
 * open state with the trigger's press state. Additionally, it allows you to customize the type and
 * positioning of the Dialog.
 */ // We don't want getCollectionNode to show up in the type definition
let $27e3a817835df4bb$export$2e1e1122cf0cba88 = $27e3a817835df4bb$var$DialogTrigger;
function $27e3a817835df4bb$var$PopoverTrigger({ state: state , targetRef: targetRef , trigger: trigger , content: content , hideArrow: hideArrow , isKeyboardDismissDisabled: isKeyboardDismissDisabled , ...props }) {
    let triggerRef = $aGPKm$react.useRef();
    let overlayRef = $aGPKm$react.useRef();
    let { overlayProps: popoverProps , placement: placement , arrowProps: arrowProps  } = $aGPKm$reactariaoverlays.useOverlayPosition({
        targetRef: targetRef || triggerRef,
        overlayRef: $aGPKm$reactspectrumutils.unwrapDOMRef(overlayRef),
        placement: props.placement,
        containerPadding: props.containerPadding,
        offset: props.offset,
        crossOffset: props.crossOffset,
        shouldFlip: props.shouldFlip,
        isOpen: state.isOpen
    });
    let { triggerProps: triggerProps , overlayProps: overlayProps  } = $aGPKm$reactariaoverlays.useOverlayTrigger({
        type: 'dialog'
    }, state, triggerRef);
    let triggerPropsWithRef = {
        ...triggerProps,
        ref: targetRef ? undefined : triggerRef
    };
    let overlay = /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactspectrumoverlays.Popover, {
        isOpen: state.isOpen,
        UNSAFE_style: popoverProps.style,
        ref: overlayRef,
        onClose: state.close,
        placement: placement,
        arrowProps: arrowProps,
        isKeyboardDismissDisabled: isKeyboardDismissDisabled,
        hideArrow: hideArrow
    }, typeof content === 'function' ? content(state.close) : content);
    return(/*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($27e3a817835df4bb$var$DialogTriggerBase, {
        type: "popover",
        state: state,
        triggerProps: triggerPropsWithRef,
        dialogProps: overlayProps,
        trigger: trigger,
        overlay: overlay
    }));
}
function $27e3a817835df4bb$var$DialogTriggerBase({ type: type , state: state , isDismissable: isDismissable , dialogProps: dialogProps = {
} , triggerProps: triggerProps = {
} , overlay: overlay , trigger: trigger  }) {
    let context = {
        type: type,
        onClose: state.close,
        isDismissable: isDismissable,
        ...dialogProps
    };
    return(/*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$react.Fragment, null, /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactariainteractions.PressResponder, {
        ...triggerProps,
        onPress: state.toggle,
        isPressed: state.isOpen && type !== 'modal' && type !== 'fullscreen' && type !== 'fullscreenTakeover'
    }, trigger), /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($59fcf3b6a0bdb727$export$8b93a07348a7730c.Provider, {
        value: context
    }, overlay)));
}


var $fc9acb28860f6f19$exports = {};

$parcel$export($fc9acb28860f6f19$exports, "DialogContainer", () => $fc9acb28860f6f19$export$547754aada6e339f);



function $fc9acb28860f6f19$export$547754aada6e339f(props) {
    let { children: children , type: type = 'modal' , onDismiss: onDismiss , isDismissable: isDismissable , isKeyboardDismissDisabled: isKeyboardDismissDisabled  } = props;
    let childArray = ($parcel$interopDefault($aGPKm$react)).Children.toArray(children);
    if (childArray.length > 1) throw new Error('Only a single child can be passed to DialogContainer.');
    let lastChild = $aGPKm$react.useRef(null);
    let child = /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).isValidElement(childArray[0]) ? childArray[0] : null;
    if (child) lastChild.current = child;
    let context = {
        type: type,
        onClose: onDismiss,
        isDismissable: isDismissable
    };
    return(/*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($aGPKm$reactspectrumoverlays.Modal, {
        isOpen: !!child,
        onClose: onDismiss,
        type: type,
        isDismissable: isDismissable,
        isKeyboardDismissDisabled: isKeyboardDismissDisabled
    }, /*#__PURE__*/ ($parcel$interopDefault($aGPKm$react)).createElement($59fcf3b6a0bdb727$export$8b93a07348a7730c.Provider, {
        value: context
    }, lastChild.current)));
}


var $c69180d7765a688e$exports = {};

$parcel$export($c69180d7765a688e$exports, "useDialogContainer", () => $c69180d7765a688e$export$a2f2d8fa6720dab1);


function $c69180d7765a688e$export$a2f2d8fa6720dab1() {
    let context = $aGPKm$react.useContext($59fcf3b6a0bdb727$export$8b93a07348a7730c);
    if (!context) throw new Error('Cannot call useDialogContext outside a <DialogTrigger> or <DialogContainer>.');
    return {
        type: context.type,
        dismiss () {
            context.onClose();
        }
    };
}


$parcel$exportWildcard(module.exports, $1bf9453c78052870$exports);
$parcel$exportWildcard(module.exports, $962c86794f106020$exports);
$parcel$exportWildcard(module.exports, $27e3a817835df4bb$exports);
$parcel$exportWildcard(module.exports, $fc9acb28860f6f19$exports);
$parcel$exportWildcard(module.exports, $c69180d7765a688e$exports);
