// @react-spectrum/dialog/dist/module.js
import "./main.css";
import $a3vs5$spectrumiconsuiAlertMedium from "@spectrum-icons/ui/AlertMedium";
import {Button as $a3vs5$Button, ActionButton as $a3vs5$ActionButton} from "@react-spectrum/button";
import {ButtonGroup as $a3vs5$ButtonGroup} from "@react-spectrum/buttongroup";
import {chain as $a3vs5$chain, mergeProps as $a3vs5$mergeProps} from "@react-aria/utils";
import {useStyleProps as $a3vs5$useStyleProps, classNames as $a3vs5$classNames, useDOMRef as $a3vs5$useDOMRef, useHasChild as $a3vs5$useHasChild, unwrapDOMRef as $a3vs5$unwrapDOMRef, SlotProvider as $a3vs5$SlotProvider, useMediaQuery as $a3vs5$useMediaQuery} from "@react-spectrum/utils";
import {Content as $a3vs5$Content} from "@react-spectrum/view";
import {Divider as $a3vs5$Divider} from "@react-spectrum/divider";
import {Heading as $a3vs5$Heading} from "@react-spectrum/text";
import $a3vs5$react, {useContext as $a3vs5$useContext, forwardRef as $a3vs5$forwardRef, useRef as $a3vs5$useRef, useMemo as $a3vs5$useMemo, useEffect as $a3vs5$useEffect, Fragment as $a3vs5$Fragment} from "react";
import {useMessageFormatter as $a3vs5$useMessageFormatter} from "@react-aria/i18n";
import $a3vs5$spectrumiconsuiCrossLarge from "@spectrum-icons/ui/CrossLarge";
import {DismissButton as $a3vs5$DismissButton, useOverlayPosition as $a3vs5$useOverlayPosition, useOverlayTrigger as $a3vs5$useOverlayTrigger} from "@react-aria/overlays";
import {FocusScope as $a3vs5$FocusScope} from "@react-aria/focus";
import {Grid as $a3vs5$Grid} from "@react-spectrum/layout";
import {useDialog as $a3vs5$useDialog} from "@react-aria/dialog";
import {Modal as $a3vs5$Modal, Tray as $a3vs5$Tray, Popover as $a3vs5$Popover} from "@react-spectrum/overlays";
import {useOverlayTriggerState as $a3vs5$useOverlayTriggerState} from "@react-stately/overlays";
import {PressResponder as $a3vs5$PressResponder} from "@react-aria/interactions";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $d1eec556c7100bf3$exports = {};

$parcel$export($d1eec556c7100bf3$exports, "AlertDialog", () => $d1eec556c7100bf3$export$de466dd8317b0b75);






var $e417d6c4dc107c51$exports = {};

$parcel$export($e417d6c4dc107c51$exports, "Dialog", () => $e417d6c4dc107c51$export$3ddf2d174ce01153);




const $12622d8e985adaeb$export$8b93a07348a7730c = $a3vs5$react.createContext(null);





var $a8506252fb8fadd7$exports = {};
var $746a7e27425387cd$exports = {};
$746a7e27425387cd$exports = JSON.parse("{\"alert\":\"تنبيه\",\"dismiss\":\"تجاهل\"}");


var $9b92f63ff847a222$exports = {};
$9b92f63ff847a222$exports = JSON.parse("{\"alert\":\"Сигнал\",\"dismiss\":\"Отхвърляне\"}");


var $20cf6328f6218166$exports = {};
$20cf6328f6218166$exports = JSON.parse("{\"alert\":\"Výstraha\",\"dismiss\":\"Odstranit\"}");


var $feae5726cb1a9ad6$exports = {};
$feae5726cb1a9ad6$exports = JSON.parse("{\"alert\":\"Advarsel\",\"dismiss\":\"Luk\"}");


var $8783af492db96afd$exports = {};
$8783af492db96afd$exports = JSON.parse("{\"alert\":\"Warnhinweis\",\"dismiss\":\"Schließen\"}");


var $0fa3badf58ed8b26$exports = {};
$0fa3badf58ed8b26$exports = JSON.parse("{\"alert\":\"Ειδοποίηση\",\"dismiss\":\"Απόρριψη\"}");


var $cd542c896a7eb5db$exports = {};
$cd542c896a7eb5db$exports = JSON.parse("{\"dismiss\":\"Dismiss\",\"alert\":\"Alert\"}");


var $c490ca1f2ec31c6f$exports = {};
$c490ca1f2ec31c6f$exports = JSON.parse("{\"alert\":\"Alerta\",\"dismiss\":\"Descartar\"}");


var $1a8bff4bba863b01$exports = {};
$1a8bff4bba863b01$exports = JSON.parse("{\"alert\":\"Teade\",\"dismiss\":\"Lõpeta\"}");


var $490762b5085e1865$exports = {};
$490762b5085e1865$exports = JSON.parse("{\"alert\":\"Hälytys\",\"dismiss\":\"Hylkää\"}");


var $ecf99f93d8f74c7e$exports = {};
$ecf99f93d8f74c7e$exports = JSON.parse("{\"alert\":\"Alerte\",\"dismiss\":\"Rejeter\"}");


var $208b633fd31536c4$exports = {};
$208b633fd31536c4$exports = JSON.parse("{\"alert\":\"התראה\",\"dismiss\":\"התעלם\"}");


var $2f21ba5e1c145a4b$exports = {};
$2f21ba5e1c145a4b$exports = JSON.parse("{\"alert\":\"Upozorenje\",\"dismiss\":\"Odbaci\"}");


var $d60c8f2474b5bc66$exports = {};
$d60c8f2474b5bc66$exports = JSON.parse("{\"alert\":\"Figyelmeztetés\",\"dismiss\":\"Elutasítás\"}");


var $4450aa33843982d5$exports = {};
$4450aa33843982d5$exports = JSON.parse("{\"alert\":\"Avviso\",\"dismiss\":\"Ignora\"}");


var $52ecd0b20e94a497$exports = {};
$52ecd0b20e94a497$exports = JSON.parse("{\"alert\":\"アラート\",\"dismiss\":\"閉じる\"}");


var $a840f8ab1d6e3059$exports = {};
$a840f8ab1d6e3059$exports = JSON.parse("{\"alert\":\"경고\",\"dismiss\":\"무시\"}");


var $5c53a36707aecc71$exports = {};
$5c53a36707aecc71$exports = JSON.parse("{\"alert\":\"Įspėjimas\",\"dismiss\":\"Atmesti\"}");


var $2d8d6fb42de45dee$exports = {};
$2d8d6fb42de45dee$exports = JSON.parse("{\"alert\":\"Brīdinājums\",\"dismiss\":\"Nerādīt\"}");


var $88b4000a39864772$exports = {};
$88b4000a39864772$exports = JSON.parse("{\"alert\":\"Varsel\",\"dismiss\":\"Lukk\"}");


var $496dbd5dce0195a1$exports = {};
$496dbd5dce0195a1$exports = JSON.parse("{\"alert\":\"Melding\",\"dismiss\":\"Negeren\"}");


var $83f5ba9404561e21$exports = {};
$83f5ba9404561e21$exports = JSON.parse("{\"alert\":\"Ostrzeżenie\",\"dismiss\":\"Zignoruj\"}");


var $be9e9fc1ddcd1e9f$exports = {};
$be9e9fc1ddcd1e9f$exports = JSON.parse("{\"alert\":\"Alerta\",\"dismiss\":\"Descartar\"}");


var $ba2036b31a09e341$exports = {};
$ba2036b31a09e341$exports = JSON.parse("{\"alert\":\"Alerta\",\"dismiss\":\"Dispensar\"}");


var $f532dc5a774abced$exports = {};
$f532dc5a774abced$exports = JSON.parse("{\"alert\":\"Alertă\",\"dismiss\":\"Revocare\"}");


var $f859c17c1efa49ad$exports = {};
$f859c17c1efa49ad$exports = JSON.parse("{\"alert\":\"Предупреждение\",\"dismiss\":\"Пропустить\"}");


var $b2078ccf7f9d4f5a$exports = {};
$b2078ccf7f9d4f5a$exports = JSON.parse("{\"alert\":\"Upozornenie\",\"dismiss\":\"Zrušiť\"}");


var $72e7e998316cc8cd$exports = {};
$72e7e998316cc8cd$exports = JSON.parse("{\"alert\":\"Opozorilo\",\"dismiss\":\"Opusti\"}");


var $b93babbfeec77b8b$exports = {};
$b93babbfeec77b8b$exports = JSON.parse("{\"alert\":\"Upozorenje\",\"dismiss\":\"Odbaci\"}");


var $14c1ee30bef2d6c4$exports = {};
$14c1ee30bef2d6c4$exports = JSON.parse("{\"alert\":\"Varning\",\"dismiss\":\"Avvisa\"}");


var $6c1d9f3e37a33bcb$exports = {};
$6c1d9f3e37a33bcb$exports = JSON.parse("{\"alert\":\"Uyarı\",\"dismiss\":\"Kapat\"}");


var $b3c51d8eb840043d$exports = {};
$b3c51d8eb840043d$exports = JSON.parse("{\"alert\":\"Сигнал тривоги\",\"dismiss\":\"Скасувати\"}");


var $c150228bcb0342da$exports = {};
$c150228bcb0342da$exports = JSON.parse("{\"alert\":\"警报\",\"dismiss\":\"取消\"}");


var $8dd75f36a163eea4$exports = {};
$8dd75f36a163eea4$exports = JSON.parse("{\"alert\":\"警示\",\"dismiss\":\"關閉\"}");


$a8506252fb8fadd7$exports = {
    "ar-AE": $746a7e27425387cd$exports,
    "bg-BG": $9b92f63ff847a222$exports,
    "cs-CZ": $20cf6328f6218166$exports,
    "da-DK": $feae5726cb1a9ad6$exports,
    "de-DE": $8783af492db96afd$exports,
    "el-GR": $0fa3badf58ed8b26$exports,
    "en-US": $cd542c896a7eb5db$exports,
    "es-ES": $c490ca1f2ec31c6f$exports,
    "et-EE": $1a8bff4bba863b01$exports,
    "fi-FI": $490762b5085e1865$exports,
    "fr-FR": $ecf99f93d8f74c7e$exports,
    "he-IL": $208b633fd31536c4$exports,
    "hr-HR": $2f21ba5e1c145a4b$exports,
    "hu-HU": $d60c8f2474b5bc66$exports,
    "it-IT": $4450aa33843982d5$exports,
    "ja-JP": $52ecd0b20e94a497$exports,
    "ko-KR": $a840f8ab1d6e3059$exports,
    "lt-LT": $5c53a36707aecc71$exports,
    "lv-LV": $2d8d6fb42de45dee$exports,
    "nb-NO": $88b4000a39864772$exports,
    "nl-NL": $496dbd5dce0195a1$exports,
    "pl-PL": $83f5ba9404561e21$exports,
    "pt-BR": $be9e9fc1ddcd1e9f$exports,
    "pt-PT": $ba2036b31a09e341$exports,
    "ro-RO": $f532dc5a774abced$exports,
    "ru-RU": $f859c17c1efa49ad$exports,
    "sk-SK": $b2078ccf7f9d4f5a$exports,
    "sl-SI": $72e7e998316cc8cd$exports,
    "sr-SP": $b93babbfeec77b8b$exports,
    "sv-SE": $14c1ee30bef2d6c4$exports,
    "tr-TR": $6c1d9f3e37a33bcb$exports,
    "uk-UA": $b3c51d8eb840043d$exports,
    "zh-CN": $c150228bcb0342da$exports,
    "zh-TW": $8dd75f36a163eea4$exports
};




var $526b49150b65bcba$exports = {};

$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog", () => $526b49150b65bcba$export$4681f3bcac39b2b1, (v) => $526b49150b65bcba$export$4681f3bcac39b2b1 = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog--small", () => $526b49150b65bcba$export$5f2d9cc47409bd63, (v) => $526b49150b65bcba$export$5f2d9cc47409bd63 = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog--medium", () => $526b49150b65bcba$export$b9b8ef443d0a67ca, (v) => $526b49150b65bcba$export$b9b8ef443d0a67ca = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog--large", () => $526b49150b65bcba$export$b73c0c2fca84be2b, (v) => $526b49150b65bcba$export$b73c0c2fca84be2b = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog-hero", () => $526b49150b65bcba$export$d405fb0a0231aa1a, (v) => $526b49150b65bcba$export$d405fb0a0231aa1a = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog-grid", () => $526b49150b65bcba$export$c03cd5e23af67ff5, (v) => $526b49150b65bcba$export$c03cd5e23af67ff5 = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog-heading", () => $526b49150b65bcba$export$94335d73b7b209ec, (v) => $526b49150b65bcba$export$94335d73b7b209ec = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog-heading--noHeader", () => $526b49150b65bcba$export$ade8307239736e89, (v) => $526b49150b65bcba$export$ade8307239736e89 = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog-header", () => $526b49150b65bcba$export$1e158eaa6363e131, (v) => $526b49150b65bcba$export$1e158eaa6363e131 = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog-typeIcon", () => $526b49150b65bcba$export$5beee9480fc6ff07, (v) => $526b49150b65bcba$export$5beee9480fc6ff07 = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog-divider", () => $526b49150b65bcba$export$dd7633d642d0bc1c, (v) => $526b49150b65bcba$export$dd7633d642d0bc1c = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog--noDivider", () => $526b49150b65bcba$export$22425527a8943917, (v) => $526b49150b65bcba$export$22425527a8943917 = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog-content", () => $526b49150b65bcba$export$cb8eccd0f3639238, (v) => $526b49150b65bcba$export$cb8eccd0f3639238 = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog-footer", () => $526b49150b65bcba$export$3ecb5025a06fe08f, (v) => $526b49150b65bcba$export$3ecb5025a06fe08f = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Button", () => $526b49150b65bcba$export$1db4cca5b4ade39a, (v) => $526b49150b65bcba$export$1db4cca5b4ade39a = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog-buttonGroup", () => $526b49150b65bcba$export$570ae852861555b7, (v) => $526b49150b65bcba$export$570ae852861555b7 = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog-buttonGroup--noFooter", () => $526b49150b65bcba$export$39dd75369bf2e830, (v) => $526b49150b65bcba$export$39dd75369bf2e830 = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog--dismissable", () => $526b49150b65bcba$export$9461b285d631d575, (v) => $526b49150b65bcba$export$9461b285d631d575 = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog-closeButton", () => $526b49150b65bcba$export$a8494287d763a75d, (v) => $526b49150b65bcba$export$a8494287d763a75d = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog--error", () => $526b49150b65bcba$export$6b5beff56972a3b, (v) => $526b49150b65bcba$export$6b5beff56972a3b = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog--fullscreen", () => $526b49150b65bcba$export$118506da5d3ee7c4, (v) => $526b49150b65bcba$export$118506da5d3ee7c4 = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog--fullscreenTakeover", () => $526b49150b65bcba$export$5c061b2d3664fda9, (v) => $526b49150b65bcba$export$5c061b2d3664fda9 = v);
$parcel$export($526b49150b65bcba$exports, "spectrum-Dialog--warning", () => $526b49150b65bcba$export$4e66396adf61534f, (v) => $526b49150b65bcba$export$4e66396adf61534f = v);
var $526b49150b65bcba$export$4681f3bcac39b2b1;
var $526b49150b65bcba$export$5f2d9cc47409bd63;
var $526b49150b65bcba$export$b9b8ef443d0a67ca;
var $526b49150b65bcba$export$b73c0c2fca84be2b;
var $526b49150b65bcba$export$d405fb0a0231aa1a;
var $526b49150b65bcba$export$c03cd5e23af67ff5;
var $526b49150b65bcba$export$94335d73b7b209ec;
var $526b49150b65bcba$export$ade8307239736e89;
var $526b49150b65bcba$export$1e158eaa6363e131;
var $526b49150b65bcba$export$5beee9480fc6ff07;
var $526b49150b65bcba$export$dd7633d642d0bc1c;
var $526b49150b65bcba$export$22425527a8943917;
var $526b49150b65bcba$export$cb8eccd0f3639238;
var $526b49150b65bcba$export$3ecb5025a06fe08f;
var $526b49150b65bcba$export$1db4cca5b4ade39a;
var $526b49150b65bcba$export$570ae852861555b7;
var $526b49150b65bcba$export$39dd75369bf2e830;
var $526b49150b65bcba$export$9461b285d631d575;
var $526b49150b65bcba$export$a8494287d763a75d;
var $526b49150b65bcba$export$6b5beff56972a3b;
var $526b49150b65bcba$export$118506da5d3ee7c4;
var $526b49150b65bcba$export$5c061b2d3664fda9;
var $526b49150b65bcba$export$4e66396adf61534f;
$526b49150b65bcba$export$4681f3bcac39b2b1 = "spectrum-Dialog_6d8b48";
$526b49150b65bcba$export$5f2d9cc47409bd63 = "spectrum-Dialog--small_6d8b48";
$526b49150b65bcba$export$b9b8ef443d0a67ca = "spectrum-Dialog--medium_6d8b48";
$526b49150b65bcba$export$b73c0c2fca84be2b = "spectrum-Dialog--large_6d8b48";
$526b49150b65bcba$export$d405fb0a0231aa1a = "spectrum-Dialog-hero_6d8b48";
$526b49150b65bcba$export$c03cd5e23af67ff5 = "spectrum-Dialog-grid_6d8b48";
$526b49150b65bcba$export$94335d73b7b209ec = "spectrum-Dialog-heading_6d8b48";
$526b49150b65bcba$export$ade8307239736e89 = "spectrum-Dialog-heading--noHeader_6d8b48";
$526b49150b65bcba$export$1e158eaa6363e131 = "spectrum-Dialog-header_6d8b48";
$526b49150b65bcba$export$5beee9480fc6ff07 = "spectrum-Dialog-typeIcon_6d8b48";
$526b49150b65bcba$export$dd7633d642d0bc1c = "spectrum-Dialog-divider_6d8b48";
$526b49150b65bcba$export$22425527a8943917 = "spectrum-Dialog--noDivider_6d8b48";
$526b49150b65bcba$export$cb8eccd0f3639238 = "spectrum-Dialog-content_6d8b48";
$526b49150b65bcba$export$3ecb5025a06fe08f = "spectrum-Dialog-footer_6d8b48";
$526b49150b65bcba$export$1db4cca5b4ade39a = "spectrum-Button_6d8b48";
$526b49150b65bcba$export$570ae852861555b7 = "spectrum-Dialog-buttonGroup_6d8b48";
$526b49150b65bcba$export$39dd75369bf2e830 = "spectrum-Dialog-buttonGroup--noFooter_6d8b48";
$526b49150b65bcba$export$9461b285d631d575 = "spectrum-Dialog--dismissable_6d8b48";
$526b49150b65bcba$export$a8494287d763a75d = "spectrum-Dialog-closeButton_6d8b48";
$526b49150b65bcba$export$6b5beff56972a3b = "spectrum-Dialog--error_6d8b48";
$526b49150b65bcba$export$118506da5d3ee7c4 = "spectrum-Dialog--fullscreen_6d8b48";
$526b49150b65bcba$export$5c061b2d3664fda9 = "spectrum-Dialog--fullscreenTakeover_6d8b48";
$526b49150b65bcba$export$4e66396adf61534f = "spectrum-Dialog--warning_6d8b48";




let $e417d6c4dc107c51$var$sizeMap = {
    S: 'small',
    M: 'medium',
    L: 'large',
    fullscreen: 'fullscreen',
    fullscreenTakeover: 'fullscreenTakeover'
};
function $e417d6c4dc107c51$var$Dialog(props, ref) {
    let { type: type = 'modal' , ...contextProps } = $a3vs5$useContext($12622d8e985adaeb$export$8b93a07348a7730c) || {
    };
    let { children: children , isDismissable: isDismissable = contextProps.isDismissable , onDismiss: onDismiss = contextProps.onClose , size: size , ...otherProps } = props;
    let formatMessage = $a3vs5$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($a8506252fb8fadd7$exports)));
    let { styleProps: styleProps  } = $a3vs5$useStyleProps(otherProps);
    size = type === 'popover' ? size || 'S' : size || 'L';
    let domRef = $a3vs5$useDOMRef(ref);
    let gridRef = $a3vs5$useRef();
    let sizeVariant = $e417d6c4dc107c51$var$sizeMap[type] || $e417d6c4dc107c51$var$sizeMap[size];
    let { dialogProps: dialogProps , titleProps: titleProps  } = $a3vs5$useDialog($a3vs5$mergeProps(contextProps, props), domRef);
    let hasHeader = $a3vs5$useHasChild(`.${(/*@__PURE__*/$parcel$interopDefault($526b49150b65bcba$exports))['spectrum-Dialog-header']}`, $a3vs5$unwrapDOMRef(gridRef));
    let hasFooter = $a3vs5$useHasChild(`.${(/*@__PURE__*/$parcel$interopDefault($526b49150b65bcba$exports))['spectrum-Dialog-footer']}`, $a3vs5$unwrapDOMRef(gridRef));
    let slots = $a3vs5$useMemo(()=>({
            hero: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($526b49150b65bcba$exports))['spectrum-Dialog-hero']
            },
            header: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($526b49150b65bcba$exports))['spectrum-Dialog-header']
            },
            heading: {
                UNSAFE_className: $a3vs5$classNames((/*@__PURE__*/$parcel$interopDefault($526b49150b65bcba$exports)), 'spectrum-Dialog-heading', {
                    'spectrum-Dialog-heading--noHeader': !hasHeader
                }),
                level: 2,
                ...titleProps
            },
            typeIcon: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($526b49150b65bcba$exports))['spectrum-Dialog-typeIcon']
            },
            divider: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($526b49150b65bcba$exports))['spectrum-Dialog-divider'],
                size: 'M'
            },
            content: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($526b49150b65bcba$exports))['spectrum-Dialog-content']
            },
            footer: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($526b49150b65bcba$exports))['spectrum-Dialog-footer']
            },
            buttonGroup: {
                UNSAFE_className: $a3vs5$classNames((/*@__PURE__*/$parcel$interopDefault($526b49150b65bcba$exports)), 'spectrum-Dialog-buttonGroup', {
                    'spectrum-Dialog-buttonGroup--noFooter': !hasFooter
                }),
                align: 'end'
            }
        })
    , [
        (/*@__PURE__*/$parcel$interopDefault($526b49150b65bcba$exports)),
        hasFooter,
        hasHeader,
        titleProps
    ]);
    // If rendered in a popover or tray there won't be a visible dismiss button,
    // so we render a hidden one for screen readers.
    let dismissButton;
    if (type === 'popover' || type === 'tray') dismissButton = /*#__PURE__*/ $a3vs5$react.createElement($a3vs5$DismissButton, {
        onDismiss: onDismiss
    });
    return(/*#__PURE__*/ $a3vs5$react.createElement($a3vs5$FocusScope, {
        contain: true,
        restoreFocus: true
    }, /*#__PURE__*/ $a3vs5$react.createElement("section", {
        ...styleProps,
        ...dialogProps,
        className: $a3vs5$classNames((/*@__PURE__*/$parcel$interopDefault($526b49150b65bcba$exports)), 'spectrum-Dialog', {
            [`spectrum-Dialog--${sizeVariant}`]: sizeVariant,
            'spectrum-Dialog--dismissable': isDismissable
        }, styleProps.className),
        ref: domRef
    }, /*#__PURE__*/ $a3vs5$react.createElement($a3vs5$Grid, {
        ref: gridRef,
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($526b49150b65bcba$exports))['spectrum-Dialog-grid']
    }, /*#__PURE__*/ $a3vs5$react.createElement($a3vs5$SlotProvider, {
        slots: slots
    }, children), isDismissable && /*#__PURE__*/ $a3vs5$react.createElement($a3vs5$ActionButton, {
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($526b49150b65bcba$exports))['spectrum-Dialog-closeButton'],
        isQuiet: true,
        "aria-label": formatMessage('dismiss'),
        onPress: onDismiss
    }, /*#__PURE__*/ $a3vs5$react.createElement($a3vs5$spectrumiconsuiCrossLarge, null))), dismissButton)));
}
/**
 * Dialogs are windows containing contextual information, tasks, or workflows that appear over the user interface.
 * Depending on the kind of Dialog, further interactions may be blocked until the Dialog is acknowledged.
 */ let $e417d6c4dc107c51$export$3ddf2d174ce01153 = /*#__PURE__*/ $a3vs5$react.forwardRef($e417d6c4dc107c51$var$Dialog);









/**
 * AlertDialogs are a specific type of Dialog. They display important information that users need to acknowledge.
 */ function $d1eec556c7100bf3$var$AlertDialog(props, ref) {
    let { onClose: onClose = ()=>{
    }  } = $a3vs5$useContext($12622d8e985adaeb$export$8b93a07348a7730c) || {
    };
    let { variant: variant , children: children , primaryActionLabel: primaryActionLabel , secondaryActionLabel: secondaryActionLabel , cancelLabel: cancelLabel , autoFocusButton: autoFocusButton , title: title , isPrimaryActionDisabled: isPrimaryActionDisabled , isSecondaryActionDisabled: isSecondaryActionDisabled , onCancel: onCancel = ()=>{
    } , onPrimaryAction: onPrimaryAction = ()=>{
    } , onSecondaryAction: onSecondaryAction = ()=>{
    } , ...otherProps } = props;
    let { styleProps: styleProps  } = $a3vs5$useStyleProps(otherProps);
    let formatMessage = $a3vs5$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($a8506252fb8fadd7$exports)));
    let confirmVariant = 'primary';
    if (variant) {
        if (variant === 'confirmation') confirmVariant = 'cta';
        else if (variant === 'destructive') confirmVariant = 'negative';
    }
    return(/*#__PURE__*/ $a3vs5$react.createElement($e417d6c4dc107c51$export$3ddf2d174ce01153, {
        UNSAFE_style: styleProps.style,
        UNSAFE_className: $a3vs5$classNames((/*@__PURE__*/$parcel$interopDefault($526b49150b65bcba$exports)), {
            [`spectrum-Dialog--${variant}`]: variant
        }, styleProps.className),
        isHidden: styleProps.hidden,
        size: "M",
        role: "alertdialog",
        ref: ref
    }, /*#__PURE__*/ $a3vs5$react.createElement($a3vs5$Heading, null, title), (variant === 'error' || variant === 'warning') && /*#__PURE__*/ $a3vs5$react.createElement($a3vs5$spectrumiconsuiAlertMedium, {
        slot: "typeIcon",
        "aria-label": formatMessage('alert')
    }), /*#__PURE__*/ $a3vs5$react.createElement($a3vs5$Divider, null), /*#__PURE__*/ $a3vs5$react.createElement($a3vs5$Content, null, children), /*#__PURE__*/ $a3vs5$react.createElement($a3vs5$ButtonGroup, {
        align: "end"
    }, cancelLabel && /*#__PURE__*/ $a3vs5$react.createElement($a3vs5$Button, {
        variant: "secondary",
        onPress: ()=>$a3vs5$chain(onClose(), onCancel())
        ,
        autoFocus: autoFocusButton === 'cancel'
    }, cancelLabel), secondaryActionLabel && /*#__PURE__*/ $a3vs5$react.createElement($a3vs5$Button, {
        variant: "secondary",
        onPress: ()=>$a3vs5$chain(onClose(), onSecondaryAction())
        ,
        isDisabled: isSecondaryActionDisabled,
        autoFocus: autoFocusButton === 'secondary'
    }, secondaryActionLabel), /*#__PURE__*/ $a3vs5$react.createElement($a3vs5$Button, {
        variant: confirmVariant,
        onPress: ()=>$a3vs5$chain(onClose(), onPrimaryAction())
        ,
        isDisabled: isPrimaryActionDisabled,
        autoFocus: autoFocusButton === 'primary'
    }, primaryActionLabel))));
}
/**
 * AlertDialogs are a specific type of Dialog. They display important information that users need to acknowledge.
 */ let $d1eec556c7100bf3$export$de466dd8317b0b75 = /*#__PURE__*/ $a3vs5$forwardRef($d1eec556c7100bf3$var$AlertDialog);



var $64fc83526e627438$exports = {};

$parcel$export($64fc83526e627438$exports, "DialogTrigger", () => $64fc83526e627438$export$2e1e1122cf0cba88);







function $64fc83526e627438$var$DialogTrigger(props) {
    let { children: children , type: type = 'modal' , mobileType: mobileType = type === 'popover' ? 'modal' : type , hideArrow: hideArrow , targetRef: targetRef , isDismissable: isDismissable , isKeyboardDismissDisabled: isKeyboardDismissDisabled , ...positionProps } = props;
    if (!Array.isArray(children) || children.length > 2) throw new Error('DialogTrigger must have exactly 2 children');
    // if a function is passed as the second child, it won't appear in toArray
    let [trigger, content] = children;
    // On small devices, show a modal or tray instead of a popover.
    // TODO: DNA variable?
    let isMobile = $a3vs5$useMediaQuery('(max-width: 700px)');
    if (isMobile) {
        // handle cases where desktop popovers need a close button for the mobile modal view
        if (type !== 'modal' && mobileType === 'modal') isDismissable = true;
        type = mobileType;
    }
    let state = $a3vs5$useOverlayTriggerState(props);
    let wasOpen = $a3vs5$useRef(false);
    wasOpen.current = state.isOpen;
    let isExiting = $a3vs5$useRef(false);
    let onExiting = ()=>isExiting.current = true
    ;
    let onExited = ()=>isExiting.current = false
    ;
    // eslint-disable-next-line arrow-body-style
    $a3vs5$useEffect(()=>{
        return ()=>{
            if ((wasOpen.current || isExiting.current) && type !== 'popover' && type !== 'tray') console.warn('A DialogTrigger unmounted while open. This is likely due to being placed within a trigger that unmounts or inside a conditional. Consider using a DialogContainer instead.');
        };
    }, []);
    if (type === 'popover') return(/*#__PURE__*/ $a3vs5$react.createElement($64fc83526e627438$var$PopoverTrigger, {
        ...positionProps,
        state: state,
        targetRef: targetRef,
        trigger: trigger,
        content: content,
        isKeyboardDismissDisabled: isKeyboardDismissDisabled,
        hideArrow: hideArrow
    }));
    let renderOverlay = ()=>{
        switch(type){
            case 'fullscreen':
            case 'fullscreenTakeover':
            case 'modal':
                return(/*#__PURE__*/ $a3vs5$react.createElement($a3vs5$Modal, {
                    isOpen: state.isOpen,
                    isDismissable: type === 'modal' ? isDismissable : false,
                    onClose: state.close,
                    type: type,
                    isKeyboardDismissDisabled: isKeyboardDismissDisabled,
                    onExiting: onExiting,
                    onExited: onExited
                }, typeof content === 'function' ? content(state.close) : content));
            case 'tray':
                return(/*#__PURE__*/ $a3vs5$react.createElement($a3vs5$Tray, {
                    isOpen: state.isOpen,
                    onClose: state.close,
                    isKeyboardDismissDisabled: isKeyboardDismissDisabled
                }, typeof content === 'function' ? content(state.close) : content));
        }
    };
    return(/*#__PURE__*/ $a3vs5$react.createElement($64fc83526e627438$var$DialogTriggerBase, {
        type: type,
        state: state,
        isDismissable: isDismissable,
        trigger: trigger,
        overlay: renderOverlay()
    }));
}
// Support DialogTrigger inside components using CollectionBuilder.
$64fc83526e627438$var$DialogTrigger.getCollectionNode = function*(props) {
    let [trigger] = $a3vs5$react.Children.toArray(props.children);
    let [, content] = props.children;
    yield {
        element: trigger,
        wrapper: (element)=>/*#__PURE__*/ $a3vs5$react.createElement($64fc83526e627438$var$DialogTrigger, {
                key: element.key,
                ...props
            }, element, content)
    };
};
/**
 * DialogTrigger serves as a wrapper around a Dialog and its associated trigger, linking the Dialog's
 * open state with the trigger's press state. Additionally, it allows you to customize the type and
 * positioning of the Dialog.
 */ // We don't want getCollectionNode to show up in the type definition
let $64fc83526e627438$export$2e1e1122cf0cba88 = $64fc83526e627438$var$DialogTrigger;
function $64fc83526e627438$var$PopoverTrigger({ state: state , targetRef: targetRef , trigger: trigger , content: content , hideArrow: hideArrow , isKeyboardDismissDisabled: isKeyboardDismissDisabled , ...props }) {
    let triggerRef = $a3vs5$useRef();
    let overlayRef = $a3vs5$useRef();
    let { overlayProps: popoverProps , placement: placement , arrowProps: arrowProps  } = $a3vs5$useOverlayPosition({
        targetRef: targetRef || triggerRef,
        overlayRef: $a3vs5$unwrapDOMRef(overlayRef),
        placement: props.placement,
        containerPadding: props.containerPadding,
        offset: props.offset,
        crossOffset: props.crossOffset,
        shouldFlip: props.shouldFlip,
        isOpen: state.isOpen
    });
    let { triggerProps: triggerProps , overlayProps: overlayProps  } = $a3vs5$useOverlayTrigger({
        type: 'dialog'
    }, state, triggerRef);
    let triggerPropsWithRef = {
        ...triggerProps,
        ref: targetRef ? undefined : triggerRef
    };
    let overlay = /*#__PURE__*/ $a3vs5$react.createElement($a3vs5$Popover, {
        isOpen: state.isOpen,
        UNSAFE_style: popoverProps.style,
        ref: overlayRef,
        onClose: state.close,
        placement: placement,
        arrowProps: arrowProps,
        isKeyboardDismissDisabled: isKeyboardDismissDisabled,
        hideArrow: hideArrow
    }, typeof content === 'function' ? content(state.close) : content);
    return(/*#__PURE__*/ $a3vs5$react.createElement($64fc83526e627438$var$DialogTriggerBase, {
        type: "popover",
        state: state,
        triggerProps: triggerPropsWithRef,
        dialogProps: overlayProps,
        trigger: trigger,
        overlay: overlay
    }));
}
function $64fc83526e627438$var$DialogTriggerBase({ type: type , state: state , isDismissable: isDismissable , dialogProps: dialogProps = {
} , triggerProps: triggerProps = {
} , overlay: overlay , trigger: trigger  }) {
    let context = {
        type: type,
        onClose: state.close,
        isDismissable: isDismissable,
        ...dialogProps
    };
    return(/*#__PURE__*/ $a3vs5$react.createElement($a3vs5$Fragment, null, /*#__PURE__*/ $a3vs5$react.createElement($a3vs5$PressResponder, {
        ...triggerProps,
        onPress: state.toggle,
        isPressed: state.isOpen && type !== 'modal' && type !== 'fullscreen' && type !== 'fullscreenTakeover'
    }, trigger), /*#__PURE__*/ $a3vs5$react.createElement($12622d8e985adaeb$export$8b93a07348a7730c.Provider, {
        value: context
    }, overlay)));
}


var $9118838491d90f43$exports = {};

$parcel$export($9118838491d90f43$exports, "DialogContainer", () => $9118838491d90f43$export$547754aada6e339f);



function $9118838491d90f43$export$547754aada6e339f(props) {
    let { children: children , type: type = 'modal' , onDismiss: onDismiss , isDismissable: isDismissable , isKeyboardDismissDisabled: isKeyboardDismissDisabled  } = props;
    let childArray = $a3vs5$react.Children.toArray(children);
    if (childArray.length > 1) throw new Error('Only a single child can be passed to DialogContainer.');
    let lastChild = $a3vs5$useRef(null);
    let child = /*#__PURE__*/ $a3vs5$react.isValidElement(childArray[0]) ? childArray[0] : null;
    if (child) lastChild.current = child;
    let context = {
        type: type,
        onClose: onDismiss,
        isDismissable: isDismissable
    };
    return(/*#__PURE__*/ $a3vs5$react.createElement($a3vs5$Modal, {
        isOpen: !!child,
        onClose: onDismiss,
        type: type,
        isDismissable: isDismissable,
        isKeyboardDismissDisabled: isKeyboardDismissDisabled
    }, /*#__PURE__*/ $a3vs5$react.createElement($12622d8e985adaeb$export$8b93a07348a7730c.Provider, {
        value: context
    }, lastChild.current)));
}


var $fd952a49be617d20$exports = {};

$parcel$export($fd952a49be617d20$exports, "useDialogContainer", () => $fd952a49be617d20$export$a2f2d8fa6720dab1);


function $fd952a49be617d20$export$a2f2d8fa6720dab1() {
    let context = $a3vs5$useContext($12622d8e985adaeb$export$8b93a07348a7730c);
    if (!context) throw new Error('Cannot call useDialogContext outside a <DialogTrigger> or <DialogContainer>.');
    return {
        type: context.type,
        dismiss () {
            context.onClose();
        }
    };
}




export {$d1eec556c7100bf3$export$de466dd8317b0b75 as AlertDialog, $e417d6c4dc107c51$export$3ddf2d174ce01153 as Dialog, $64fc83526e627438$export$2e1e1122cf0cba88 as DialogTrigger, $9118838491d90f43$export$547754aada6e339f as DialogContainer, $fd952a49be617d20$export$a2f2d8fa6720dab1 as useDialogContainer};
