// @react-spectrum/switch/dist/main.js
require("./main.css");
var $iaZbI$reactspectrumutils = require("@react-spectrum/utils");
var $iaZbI$reactariafocus = require("@react-aria/focus");
var $iaZbI$react = require("react");
var $iaZbI$reactariainteractions = require("@react-aria/interactions");
var $iaZbI$reactspectrumprovider = require("@react-spectrum/provider");
var $iaZbI$reactariaswitch = require("@react-aria/switch");
var $iaZbI$reactstatelytoggle = require("@react-stately/toggle");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $f878173124962684$exports = {};

$parcel$export($f878173124962684$exports, "Switch", () => $f878173124962684$export$b5d5cf8927ab7262);



var $e3310a3d26e16aaf$exports = {};

$parcel$export($e3310a3d26e16aaf$exports, "spectrum-ToggleSwitch", () => $e3310a3d26e16aaf$export$89a9162656c0c38c, (v) => $e3310a3d26e16aaf$export$89a9162656c0c38c = v);
$parcel$export($e3310a3d26e16aaf$exports, "spectrum-ToggleSwitch-input", () => $e3310a3d26e16aaf$export$38df7ea4ffef819a, (v) => $e3310a3d26e16aaf$export$38df7ea4ffef819a = v);
$parcel$export($e3310a3d26e16aaf$exports, "spectrum-ToggleSwitch-switch", () => $e3310a3d26e16aaf$export$b82d06d1b520a22f, (v) => $e3310a3d26e16aaf$export$b82d06d1b520a22f = v);
$parcel$export($e3310a3d26e16aaf$exports, "focus-ring", () => $e3310a3d26e16aaf$export$f39a09f249340e2a, (v) => $e3310a3d26e16aaf$export$f39a09f249340e2a = v);
$parcel$export($e3310a3d26e16aaf$exports, "spectrum-ToggleSwitch-label", () => $e3310a3d26e16aaf$export$8f1a69021a3897b2, (v) => $e3310a3d26e16aaf$export$8f1a69021a3897b2 = v);
$parcel$export($e3310a3d26e16aaf$exports, "is-hovered", () => $e3310a3d26e16aaf$export$b8813cd5d7824ce7, (v) => $e3310a3d26e16aaf$export$b8813cd5d7824ce7 = v);
$parcel$export($e3310a3d26e16aaf$exports, "spectrum-ToggleSwitch--quiet", () => $e3310a3d26e16aaf$export$ce1b1dcfc10f3df4, (v) => $e3310a3d26e16aaf$export$ce1b1dcfc10f3df4 = v);
$parcel$export($e3310a3d26e16aaf$exports, "spectrum-ToggleSwitch--ab", () => $e3310a3d26e16aaf$export$dc5a33fddabf5882, (v) => $e3310a3d26e16aaf$export$dc5a33fddabf5882 = v);
var $e3310a3d26e16aaf$export$89a9162656c0c38c;
var $e3310a3d26e16aaf$export$38df7ea4ffef819a;
var $e3310a3d26e16aaf$export$b82d06d1b520a22f;
var $e3310a3d26e16aaf$export$f39a09f249340e2a;
var $e3310a3d26e16aaf$export$8f1a69021a3897b2;
var $e3310a3d26e16aaf$export$b8813cd5d7824ce7;
var $e3310a3d26e16aaf$export$ce1b1dcfc10f3df4;
var $e3310a3d26e16aaf$export$dc5a33fddabf5882;
$e3310a3d26e16aaf$export$89a9162656c0c38c = "spectrum-ToggleSwitch_3526dd";
$e3310a3d26e16aaf$export$38df7ea4ffef819a = "spectrum-ToggleSwitch-input_3526dd";
$e3310a3d26e16aaf$export$b82d06d1b520a22f = "spectrum-ToggleSwitch-switch_3526dd";
$e3310a3d26e16aaf$export$f39a09f249340e2a = "focus-ring_3526dd";
$e3310a3d26e16aaf$export$8f1a69021a3897b2 = "spectrum-ToggleSwitch-label_3526dd";
$e3310a3d26e16aaf$export$b8813cd5d7824ce7 = "is-hovered_3526dd";
$e3310a3d26e16aaf$export$ce1b1dcfc10f3df4 = "spectrum-ToggleSwitch--quiet_3526dd";
$e3310a3d26e16aaf$export$dc5a33fddabf5882 = "spectrum-ToggleSwitch--ab_3526dd";






function $f878173124962684$var$Switch(props, ref) {
    props = $iaZbI$reactspectrumprovider.useProviderProps(props);
    let { isEmphasized: isEmphasized = false , isDisabled: isDisabled = false , autoFocus: autoFocus , children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $iaZbI$reactspectrumutils.useStyleProps(otherProps);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $iaZbI$reactariainteractions.useHover({
        isDisabled: isDisabled
    });
    let inputRef = $iaZbI$react.useRef(null);
    let domRef = $iaZbI$reactspectrumutils.useFocusableRef(ref, inputRef);
    let state = $iaZbI$reactstatelytoggle.useToggleState(props);
    let { inputProps: inputProps  } = $iaZbI$reactariaswitch.useSwitch(props, state, inputRef);
    return(/*#__PURE__*/ ($parcel$interopDefault($iaZbI$react)).createElement("label", {
        ...styleProps,
        ...hoverProps,
        ref: domRef,
        className: $iaZbI$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($e3310a3d26e16aaf$exports)), 'spectrum-ToggleSwitch', {
            'spectrum-ToggleSwitch--quiet': !isEmphasized,
            'is-disabled': isDisabled,
            'is-hovered': isHovered
        }, styleProps.className)
    }, /*#__PURE__*/ ($parcel$interopDefault($iaZbI$react)).createElement($iaZbI$reactariafocus.FocusRing, {
        focusRingClass: $iaZbI$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($e3310a3d26e16aaf$exports)), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ ($parcel$interopDefault($iaZbI$react)).createElement("input", {
        ...inputProps,
        ref: inputRef,
        className: $iaZbI$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($e3310a3d26e16aaf$exports)), 'spectrum-ToggleSwitch-input')
    })), /*#__PURE__*/ ($parcel$interopDefault($iaZbI$react)).createElement("span", {
        className: $iaZbI$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($e3310a3d26e16aaf$exports)), 'spectrum-ToggleSwitch-switch')
    }), children && /*#__PURE__*/ ($parcel$interopDefault($iaZbI$react)).createElement("span", {
        className: $iaZbI$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($e3310a3d26e16aaf$exports)), 'spectrum-ToggleSwitch-label')
    }, children)));
}
/**
 * Switches allow users to turn an individual option on or off.
 * They are usually used to activate or deactivate a specific setting.
 */ const $f878173124962684$export$b5d5cf8927ab7262 = /*#__PURE__*/ $iaZbI$react.forwardRef($f878173124962684$var$Switch);


$parcel$exportWildcard(module.exports, $f878173124962684$exports);
