// @react-spectrum/switch/dist/module.js
import "./main.css";
import {useStyleProps as $eGwyD$useStyleProps, useFocusableRef as $eGwyD$useFocusableRef, classNames as $eGwyD$classNames} from "@react-spectrum/utils";
import {FocusRing as $eGwyD$FocusRing} from "@react-aria/focus";
import $eGwyD$react, {useRef as $eGwyD$useRef, forwardRef as $eGwyD$forwardRef} from "react";
import {useHover as $eGwyD$useHover} from "@react-aria/interactions";
import {useProviderProps as $eGwyD$useProviderProps} from "@react-spectrum/provider";
import {useSwitch as $eGwyD$useSwitch} from "@react-aria/switch";
import {useToggleState as $eGwyD$useToggleState} from "@react-stately/toggle";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $ceb92c6cc404b1a5$exports = {};

$parcel$export($ceb92c6cc404b1a5$exports, "Switch", () => $ceb92c6cc404b1a5$export$b5d5cf8927ab7262);



var $f57785370c8eb19b$exports = {};

$parcel$export($f57785370c8eb19b$exports, "spectrum-ToggleSwitch", () => $f57785370c8eb19b$export$89a9162656c0c38c, (v) => $f57785370c8eb19b$export$89a9162656c0c38c = v);
$parcel$export($f57785370c8eb19b$exports, "spectrum-ToggleSwitch-input", () => $f57785370c8eb19b$export$38df7ea4ffef819a, (v) => $f57785370c8eb19b$export$38df7ea4ffef819a = v);
$parcel$export($f57785370c8eb19b$exports, "spectrum-ToggleSwitch-switch", () => $f57785370c8eb19b$export$b82d06d1b520a22f, (v) => $f57785370c8eb19b$export$b82d06d1b520a22f = v);
$parcel$export($f57785370c8eb19b$exports, "focus-ring", () => $f57785370c8eb19b$export$f39a09f249340e2a, (v) => $f57785370c8eb19b$export$f39a09f249340e2a = v);
$parcel$export($f57785370c8eb19b$exports, "spectrum-ToggleSwitch-label", () => $f57785370c8eb19b$export$8f1a69021a3897b2, (v) => $f57785370c8eb19b$export$8f1a69021a3897b2 = v);
$parcel$export($f57785370c8eb19b$exports, "is-hovered", () => $f57785370c8eb19b$export$b8813cd5d7824ce7, (v) => $f57785370c8eb19b$export$b8813cd5d7824ce7 = v);
$parcel$export($f57785370c8eb19b$exports, "spectrum-ToggleSwitch--quiet", () => $f57785370c8eb19b$export$ce1b1dcfc10f3df4, (v) => $f57785370c8eb19b$export$ce1b1dcfc10f3df4 = v);
$parcel$export($f57785370c8eb19b$exports, "spectrum-ToggleSwitch--ab", () => $f57785370c8eb19b$export$dc5a33fddabf5882, (v) => $f57785370c8eb19b$export$dc5a33fddabf5882 = v);
var $f57785370c8eb19b$export$89a9162656c0c38c;
var $f57785370c8eb19b$export$38df7ea4ffef819a;
var $f57785370c8eb19b$export$b82d06d1b520a22f;
var $f57785370c8eb19b$export$f39a09f249340e2a;
var $f57785370c8eb19b$export$8f1a69021a3897b2;
var $f57785370c8eb19b$export$b8813cd5d7824ce7;
var $f57785370c8eb19b$export$ce1b1dcfc10f3df4;
var $f57785370c8eb19b$export$dc5a33fddabf5882;
$f57785370c8eb19b$export$89a9162656c0c38c = "spectrum-ToggleSwitch_3526dd";
$f57785370c8eb19b$export$38df7ea4ffef819a = "spectrum-ToggleSwitch-input_3526dd";
$f57785370c8eb19b$export$b82d06d1b520a22f = "spectrum-ToggleSwitch-switch_3526dd";
$f57785370c8eb19b$export$f39a09f249340e2a = "focus-ring_3526dd";
$f57785370c8eb19b$export$8f1a69021a3897b2 = "spectrum-ToggleSwitch-label_3526dd";
$f57785370c8eb19b$export$b8813cd5d7824ce7 = "is-hovered_3526dd";
$f57785370c8eb19b$export$ce1b1dcfc10f3df4 = "spectrum-ToggleSwitch--quiet_3526dd";
$f57785370c8eb19b$export$dc5a33fddabf5882 = "spectrum-ToggleSwitch--ab_3526dd";






function $ceb92c6cc404b1a5$var$Switch(props, ref) {
    props = $eGwyD$useProviderProps(props);
    let { isEmphasized: isEmphasized = false , isDisabled: isDisabled = false , autoFocus: autoFocus , children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $eGwyD$useStyleProps(otherProps);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $eGwyD$useHover({
        isDisabled: isDisabled
    });
    let inputRef = $eGwyD$useRef(null);
    let domRef = $eGwyD$useFocusableRef(ref, inputRef);
    let state = $eGwyD$useToggleState(props);
    let { inputProps: inputProps  } = $eGwyD$useSwitch(props, state, inputRef);
    return(/*#__PURE__*/ $eGwyD$react.createElement("label", {
        ...styleProps,
        ...hoverProps,
        ref: domRef,
        className: $eGwyD$classNames((/*@__PURE__*/$parcel$interopDefault($f57785370c8eb19b$exports)), 'spectrum-ToggleSwitch', {
            'spectrum-ToggleSwitch--quiet': !isEmphasized,
            'is-disabled': isDisabled,
            'is-hovered': isHovered
        }, styleProps.className)
    }, /*#__PURE__*/ $eGwyD$react.createElement($eGwyD$FocusRing, {
        focusRingClass: $eGwyD$classNames((/*@__PURE__*/$parcel$interopDefault($f57785370c8eb19b$exports)), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ $eGwyD$react.createElement("input", {
        ...inputProps,
        ref: inputRef,
        className: $eGwyD$classNames((/*@__PURE__*/$parcel$interopDefault($f57785370c8eb19b$exports)), 'spectrum-ToggleSwitch-input')
    })), /*#__PURE__*/ $eGwyD$react.createElement("span", {
        className: $eGwyD$classNames((/*@__PURE__*/$parcel$interopDefault($f57785370c8eb19b$exports)), 'spectrum-ToggleSwitch-switch')
    }), children && /*#__PURE__*/ $eGwyD$react.createElement("span", {
        className: $eGwyD$classNames((/*@__PURE__*/$parcel$interopDefault($f57785370c8eb19b$exports)), 'spectrum-ToggleSwitch-label')
    }, children)));
}
/**
 * Switches allow users to turn an individual option on or off.
 * They are usually used to activate or deactivate a specific setting.
 */ const $ceb92c6cc404b1a5$export$b5d5cf8927ab7262 = /*#__PURE__*/ $eGwyD$forwardRef($ceb92c6cc404b1a5$var$Switch);




export {$ceb92c6cc404b1a5$export$b5d5cf8927ab7262 as Switch};
