// @react-spectrum/progress/dist/module.js
import "./main.css";
import {classNames as $byFKM$classNames, useDOMRef as $byFKM$useDOMRef, useStyleProps as $byFKM$useStyleProps} from "@react-spectrum/utils";
import $byFKM$react from "react";
import {useProgressBar as $byFKM$useProgressBar} from "@react-aria/progress";
import {clamp as $byFKM$clamp} from "@react-aria/utils";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $3e2fc00d5e01478b$exports = {};

$parcel$export($3e2fc00d5e01478b$exports, "ProgressBar", () => $3e2fc00d5e01478b$export$c17561cb55d4db30);

var $287d3431be7af3d0$exports = {};

$parcel$export($287d3431be7af3d0$exports, "ProgressBarBase", () => $287d3431be7af3d0$export$7c6ed87244065f3a);



var $0cd8e49fa97e4664$exports = {};

$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader", () => $0cd8e49fa97e4664$export$162ba336fb0c239f, (v) => $0cd8e49fa97e4664$export$162ba336fb0c239f = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader--indeterminate", () => $0cd8e49fa97e4664$export$1249dfbb98223794, (v) => $0cd8e49fa97e4664$export$1249dfbb98223794 = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader--sideLabel", () => $0cd8e49fa97e4664$export$99af8802cc2a10dc, (v) => $0cd8e49fa97e4664$export$99af8802cc2a10dc = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader-track", () => $0cd8e49fa97e4664$export$3e2e13fb57b5c74b, (v) => $0cd8e49fa97e4664$export$3e2e13fb57b5c74b = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader-fill", () => $0cd8e49fa97e4664$export$84c9ee7bc724b71b, (v) => $0cd8e49fa97e4664$export$84c9ee7bc724b71b = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader-label", () => $0cd8e49fa97e4664$export$5379ba198163be86, (v) => $0cd8e49fa97e4664$export$5379ba198163be86 = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader-percentage", () => $0cd8e49fa97e4664$export$ecc9c825de9d1a18, (v) => $0cd8e49fa97e4664$export$ecc9c825de9d1a18 = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader--small", () => $0cd8e49fa97e4664$export$b945305960ad74a7, (v) => $0cd8e49fa97e4664$export$b945305960ad74a7 = v);
$parcel$export($0cd8e49fa97e4664$exports, "indeterminate-loop-ltr", () => $0cd8e49fa97e4664$export$a889038a25113dd4, (v) => $0cd8e49fa97e4664$export$a889038a25113dd4 = v);
$parcel$export($0cd8e49fa97e4664$exports, "indeterminate-loop-rtl", () => $0cd8e49fa97e4664$export$aeff243cf3356f00, (v) => $0cd8e49fa97e4664$export$aeff243cf3356f00 = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader--overBackground", () => $0cd8e49fa97e4664$export$55612d0bd9b5d41, (v) => $0cd8e49fa97e4664$export$55612d0bd9b5d41 = v);
$parcel$export($0cd8e49fa97e4664$exports, "is-positive", () => $0cd8e49fa97e4664$export$aea9fd919467f03a, (v) => $0cd8e49fa97e4664$export$aea9fd919467f03a = v);
$parcel$export($0cd8e49fa97e4664$exports, "is-warning", () => $0cd8e49fa97e4664$export$d4ebda6218901908, (v) => $0cd8e49fa97e4664$export$d4ebda6218901908 = v);
$parcel$export($0cd8e49fa97e4664$exports, "is-critical", () => $0cd8e49fa97e4664$export$a1f608a9810da50f, (v) => $0cd8e49fa97e4664$export$a1f608a9810da50f = v);
var $0cd8e49fa97e4664$export$162ba336fb0c239f;
var $0cd8e49fa97e4664$export$1249dfbb98223794;
var $0cd8e49fa97e4664$export$99af8802cc2a10dc;
var $0cd8e49fa97e4664$export$3e2e13fb57b5c74b;
var $0cd8e49fa97e4664$export$84c9ee7bc724b71b;
var $0cd8e49fa97e4664$export$5379ba198163be86;
var $0cd8e49fa97e4664$export$ecc9c825de9d1a18;
var $0cd8e49fa97e4664$export$b945305960ad74a7;
var $0cd8e49fa97e4664$export$a889038a25113dd4;
var $0cd8e49fa97e4664$export$aeff243cf3356f00;
var $0cd8e49fa97e4664$export$55612d0bd9b5d41;
var $0cd8e49fa97e4664$export$aea9fd919467f03a;
var $0cd8e49fa97e4664$export$d4ebda6218901908;
var $0cd8e49fa97e4664$export$a1f608a9810da50f;
$0cd8e49fa97e4664$export$162ba336fb0c239f = "spectrum-BarLoader_48541f";
$0cd8e49fa97e4664$export$1249dfbb98223794 = "spectrum-BarLoader--indeterminate_48541f";
$0cd8e49fa97e4664$export$99af8802cc2a10dc = "spectrum-BarLoader--sideLabel_48541f";
$0cd8e49fa97e4664$export$3e2e13fb57b5c74b = "spectrum-BarLoader-track_48541f";
$0cd8e49fa97e4664$export$84c9ee7bc724b71b = "spectrum-BarLoader-fill_48541f";
$0cd8e49fa97e4664$export$5379ba198163be86 = "spectrum-BarLoader-label_48541f";
$0cd8e49fa97e4664$export$ecc9c825de9d1a18 = "spectrum-BarLoader-percentage_48541f";
$0cd8e49fa97e4664$export$b945305960ad74a7 = "spectrum-BarLoader--small_48541f";
$0cd8e49fa97e4664$export$a889038a25113dd4 = "indeterminate-loop-ltr_48541f";
$0cd8e49fa97e4664$export$aeff243cf3356f00 = "indeterminate-loop-rtl_48541f";
$0cd8e49fa97e4664$export$55612d0bd9b5d41 = "spectrum-BarLoader--overBackground_48541f";
$0cd8e49fa97e4664$export$aea9fd919467f03a = "is-positive_48541f";
$0cd8e49fa97e4664$export$d4ebda6218901908 = "is-warning_48541f";
$0cd8e49fa97e4664$export$a1f608a9810da50f = "is-critical_48541f";


// Base ProgressBar component shared with Meter.
function $287d3431be7af3d0$var$ProgressBarBase(props, ref) {
    let { value: value = 0 , minValue: minValue = 0 , maxValue: maxValue = 100 , size: size = 'L' , label: label , barClassName: barClassName , showValueLabel: showValueLabel = !!label , labelPosition: labelPosition = 'top' , isIndeterminate: isIndeterminate = false , barProps: barProps , labelProps: labelProps , 'aria-label': ariaLabel , 'aria-labelledby': ariaLabelledby , ...otherProps } = props;
    let domRef = $byFKM$useDOMRef(ref);
    let { styleProps: styleProps  } = $byFKM$useStyleProps(otherProps);
    value = $byFKM$clamp(value, minValue, maxValue);
    let barStyle = {
    };
    if (!isIndeterminate) {
        let percentage = (value - minValue) / (maxValue - minValue);
        barStyle.width = `${Math.round(percentage * 100)}%`;
    }
    // Ideally this should be in useProgressBar, but children
    // are not supported in ProgressCircle which shares that hook...
    if (!label && !ariaLabel && !ariaLabelledby) console.warn('If you do not provide a visible label via children, you must specify an aria-label or aria-labelledby attribute for accessibility');
    // use inline style for fit-content because cssnano is too smart for us and will strip out the -moz prefix in css files
    return(/*#__PURE__*/ $byFKM$react.createElement("div", {
        ...barProps,
        ref: domRef,
        className: $byFKM$classNames((/*@__PURE__*/$parcel$interopDefault($0cd8e49fa97e4664$exports)), 'spectrum-BarLoader', {
            'spectrum-BarLoader--small': size === 'S',
            'spectrum-BarLoader--large': size === 'L',
            'spectrum-BarLoader--indeterminate': isIndeterminate,
            'spectrum-BarLoader--sideLabel': labelPosition === 'side'
        }, barClassName, styleProps.className),
        style: {
            minWidth: '-moz-fit-content',
            ...styleProps.style
        }
    }, label && /*#__PURE__*/ $byFKM$react.createElement("span", {
        ...labelProps,
        className: $byFKM$classNames((/*@__PURE__*/$parcel$interopDefault($0cd8e49fa97e4664$exports)), 'spectrum-BarLoader-label')
    }, label), showValueLabel && /*#__PURE__*/ $byFKM$react.createElement("div", {
        className: $byFKM$classNames((/*@__PURE__*/$parcel$interopDefault($0cd8e49fa97e4664$exports)), 'spectrum-BarLoader-percentage')
    }, barProps['aria-valuetext']), /*#__PURE__*/ $byFKM$react.createElement("div", {
        className: $byFKM$classNames((/*@__PURE__*/$parcel$interopDefault($0cd8e49fa97e4664$exports)), 'spectrum-BarLoader-track')
    }, /*#__PURE__*/ $byFKM$react.createElement("div", {
        className: $byFKM$classNames((/*@__PURE__*/$parcel$interopDefault($0cd8e49fa97e4664$exports)), 'spectrum-BarLoader-fill'),
        style: barStyle
    }))));
}
let $287d3431be7af3d0$export$7c6ed87244065f3a = /*#__PURE__*/ $byFKM$react.forwardRef($287d3431be7af3d0$var$ProgressBarBase);





function $3e2fc00d5e01478b$var$ProgressBar(props, ref) {
    let { variant: variant , ...otherProps } = props;
    const { progressBarProps: progressBarProps , labelProps: labelProps  } = $byFKM$useProgressBar(props);
    return(/*#__PURE__*/ $byFKM$react.createElement($287d3431be7af3d0$export$7c6ed87244065f3a, {
        ...otherProps,
        ref: ref,
        barProps: progressBarProps,
        labelProps: labelProps,
        barClassName: $byFKM$classNames((/*@__PURE__*/$parcel$interopDefault($0cd8e49fa97e4664$exports)), {
            'spectrum-BarLoader--overBackground': variant === 'overBackground'
        })
    }));
}
/**
 * ProgressBars show the progression of a system operation: downloading, uploading, processing, etc., in a visual way.
 * They can represent either determinate or indeterminate progress.
 */ let $3e2fc00d5e01478b$export$c17561cb55d4db30 = /*#__PURE__*/ $byFKM$react.forwardRef($3e2fc00d5e01478b$var$ProgressBar);


var $9890faa22af21f4f$exports = {};

$parcel$export($9890faa22af21f4f$exports, "ProgressCircle", () => $9890faa22af21f4f$export$c79b9d6b4cc92af7);



var $5e9896e6052c3bbb$exports = {};

$parcel$export($5e9896e6052c3bbb$exports, "spectrum-CircleLoader--indeterminate-fill-submask-2", () => $5e9896e6052c3bbb$export$f7083dbeadcbb2bd, (v) => $5e9896e6052c3bbb$export$f7083dbeadcbb2bd = v);
$parcel$export($5e9896e6052c3bbb$exports, "spectrum-fill-mask-2", () => $5e9896e6052c3bbb$export$7219832bea80eb54, (v) => $5e9896e6052c3bbb$export$7219832bea80eb54 = v);
$parcel$export($5e9896e6052c3bbb$exports, "spectrum-CircleLoader", () => $5e9896e6052c3bbb$export$9633d311d6d6830b, (v) => $5e9896e6052c3bbb$export$9633d311d6d6830b = v);
$parcel$export($5e9896e6052c3bbb$exports, "spectrum-CircleLoader-track", () => $5e9896e6052c3bbb$export$278efdb85ed802da, (v) => $5e9896e6052c3bbb$export$278efdb85ed802da = v);
$parcel$export($5e9896e6052c3bbb$exports, "spectrum-CircleLoader-fills", () => $5e9896e6052c3bbb$export$3038f4c0c86f539b, (v) => $5e9896e6052c3bbb$export$3038f4c0c86f539b = v);
$parcel$export($5e9896e6052c3bbb$exports, "spectrum-CircleLoader-fill", () => $5e9896e6052c3bbb$export$58fb9ac68cf8e784, (v) => $5e9896e6052c3bbb$export$58fb9ac68cf8e784 = v);
$parcel$export($5e9896e6052c3bbb$exports, "spectrum-CircleLoader-fillMask1", () => $5e9896e6052c3bbb$export$b676cd958401a736, (v) => $5e9896e6052c3bbb$export$b676cd958401a736 = v);
$parcel$export($5e9896e6052c3bbb$exports, "spectrum-CircleLoader-fillMask2", () => $5e9896e6052c3bbb$export$8c428f4f13e424b, (v) => $5e9896e6052c3bbb$export$8c428f4f13e424b = v);
$parcel$export($5e9896e6052c3bbb$exports, "spectrum-CircleLoader-fillSubMask1", () => $5e9896e6052c3bbb$export$98ba5ef650aeb8ca, (v) => $5e9896e6052c3bbb$export$98ba5ef650aeb8ca = v);
$parcel$export($5e9896e6052c3bbb$exports, "spectrum-CircleLoader-fillSubMask2", () => $5e9896e6052c3bbb$export$dc70b48c897b0f1a, (v) => $5e9896e6052c3bbb$export$dc70b48c897b0f1a = v);
$parcel$export($5e9896e6052c3bbb$exports, "spectrum-CircleLoader--small", () => $5e9896e6052c3bbb$export$79819ab9844dbe9b, (v) => $5e9896e6052c3bbb$export$79819ab9844dbe9b = v);
$parcel$export($5e9896e6052c3bbb$exports, "spectrum-CircleLoader--large", () => $5e9896e6052c3bbb$export$c8c34e5e37bf511c, (v) => $5e9896e6052c3bbb$export$c8c34e5e37bf511c = v);
$parcel$export($5e9896e6052c3bbb$exports, "spectrum-CircleLoader--indeterminate", () => $5e9896e6052c3bbb$export$79b7e5abeef5134f, (v) => $5e9896e6052c3bbb$export$79b7e5abeef5134f = v);
$parcel$export($5e9896e6052c3bbb$exports, "spectrum-fills-rotate", () => $5e9896e6052c3bbb$export$3663cbdb240663ec, (v) => $5e9896e6052c3bbb$export$3663cbdb240663ec = v);
$parcel$export($5e9896e6052c3bbb$exports, "spectrum-fill-mask-1", () => $5e9896e6052c3bbb$export$524c81e63e1d64f5, (v) => $5e9896e6052c3bbb$export$524c81e63e1d64f5 = v);
$parcel$export($5e9896e6052c3bbb$exports, "spectrum-CircleLoader--overBackground", () => $5e9896e6052c3bbb$export$6668708cb539f403, (v) => $5e9896e6052c3bbb$export$6668708cb539f403 = v);
var $5e9896e6052c3bbb$export$f7083dbeadcbb2bd;
var $5e9896e6052c3bbb$export$7219832bea80eb54;
var $5e9896e6052c3bbb$export$9633d311d6d6830b;
var $5e9896e6052c3bbb$export$278efdb85ed802da;
var $5e9896e6052c3bbb$export$3038f4c0c86f539b;
var $5e9896e6052c3bbb$export$58fb9ac68cf8e784;
var $5e9896e6052c3bbb$export$b676cd958401a736;
var $5e9896e6052c3bbb$export$8c428f4f13e424b;
var $5e9896e6052c3bbb$export$98ba5ef650aeb8ca;
var $5e9896e6052c3bbb$export$dc70b48c897b0f1a;
var $5e9896e6052c3bbb$export$79819ab9844dbe9b;
var $5e9896e6052c3bbb$export$c8c34e5e37bf511c;
var $5e9896e6052c3bbb$export$79b7e5abeef5134f;
var $5e9896e6052c3bbb$export$3663cbdb240663ec;
var $5e9896e6052c3bbb$export$524c81e63e1d64f5;
var $5e9896e6052c3bbb$export$6668708cb539f403;
$5e9896e6052c3bbb$export$f7083dbeadcbb2bd = "spectrum-CircleLoader--indeterminate-fill-submask-2_7c1153";
$5e9896e6052c3bbb$export$7219832bea80eb54 = "spectrum-fill-mask-2_7c1153";
$5e9896e6052c3bbb$export$9633d311d6d6830b = "spectrum-CircleLoader_7c1153";
$5e9896e6052c3bbb$export$278efdb85ed802da = "spectrum-CircleLoader-track_7c1153";
$5e9896e6052c3bbb$export$3038f4c0c86f539b = "spectrum-CircleLoader-fills_7c1153";
$5e9896e6052c3bbb$export$58fb9ac68cf8e784 = "spectrum-CircleLoader-fill_7c1153";
$5e9896e6052c3bbb$export$b676cd958401a736 = "spectrum-CircleLoader-fillMask1_7c1153";
$5e9896e6052c3bbb$export$8c428f4f13e424b = "spectrum-CircleLoader-fillMask2_7c1153";
$5e9896e6052c3bbb$export$98ba5ef650aeb8ca = "spectrum-CircleLoader-fillSubMask1_7c1153";
$5e9896e6052c3bbb$export$dc70b48c897b0f1a = "spectrum-CircleLoader-fillSubMask2_7c1153";
$5e9896e6052c3bbb$export$79819ab9844dbe9b = "spectrum-CircleLoader--small_7c1153";
$5e9896e6052c3bbb$export$c8c34e5e37bf511c = "spectrum-CircleLoader--large_7c1153";
$5e9896e6052c3bbb$export$79b7e5abeef5134f = "spectrum-CircleLoader--indeterminate_7c1153";
$5e9896e6052c3bbb$export$3663cbdb240663ec = "spectrum-fills-rotate_7c1153";
$5e9896e6052c3bbb$export$524c81e63e1d64f5 = "spectrum-fill-mask-1_7c1153";
$5e9896e6052c3bbb$export$6668708cb539f403 = "spectrum-CircleLoader--overBackground_7c1153";



function $9890faa22af21f4f$var$ProgressCircle(props, ref) {
    let { value: value = 0 , minValue: minValue = 0 , maxValue: maxValue = 100 , size: size = 'M' , variant: variant , isIndeterminate: isIndeterminate = false , 'aria-label': ariaLabel , 'aria-labelledby': ariaLabelledby , ...otherProps } = props;
    let domRef = $byFKM$useDOMRef(ref);
    let { styleProps: styleProps  } = $byFKM$useStyleProps(otherProps);
    value = $byFKM$clamp(value, minValue, maxValue);
    let { progressBarProps: progressBarProps  } = $byFKM$useProgressBar({
        ...props,
        value: value
    });
    let subMask1Style = {
    };
    let subMask2Style = {
    };
    if (!isIndeterminate) {
        let percentage = (value - minValue) / (maxValue - minValue) * 100;
        let angle;
        if (percentage > 0 && percentage <= 50) {
            angle = -180 + percentage / 50 * 180;
            subMask1Style.transform = `rotate(${angle}deg)`;
            subMask2Style.transform = 'rotate(-180deg)';
        } else if (percentage > 50) {
            angle = -180 + (percentage - 50) / 50 * 180;
            subMask1Style.transform = 'rotate(0deg)';
            subMask2Style.transform = `rotate(${angle}deg)`;
        }
    }
    if (!ariaLabel && !ariaLabelledby) console.warn('ProgressCircle requires an aria-label or aria-labelledby attribute for accessibility');
    return(/*#__PURE__*/ $byFKM$react.createElement("div", {
        ...styleProps,
        ...progressBarProps,
        ref: domRef,
        className: $byFKM$classNames((/*@__PURE__*/$parcel$interopDefault($5e9896e6052c3bbb$exports)), 'spectrum-CircleLoader', {
            'spectrum-CircleLoader--indeterminate': isIndeterminate,
            'spectrum-CircleLoader--small': size === 'S',
            'spectrum-CircleLoader--large': size === 'L',
            'spectrum-CircleLoader--overBackground': variant === 'overBackground'
        }, styleProps.className)
    }, /*#__PURE__*/ $byFKM$react.createElement("div", {
        className: $byFKM$classNames((/*@__PURE__*/$parcel$interopDefault($5e9896e6052c3bbb$exports)), 'spectrum-CircleLoader-track')
    }), /*#__PURE__*/ $byFKM$react.createElement("div", {
        className: $byFKM$classNames((/*@__PURE__*/$parcel$interopDefault($5e9896e6052c3bbb$exports)), 'spectrum-CircleLoader-fills')
    }, /*#__PURE__*/ $byFKM$react.createElement("div", {
        className: $byFKM$classNames((/*@__PURE__*/$parcel$interopDefault($5e9896e6052c3bbb$exports)), 'spectrum-CircleLoader-fillMask1')
    }, /*#__PURE__*/ $byFKM$react.createElement("div", {
        className: $byFKM$classNames((/*@__PURE__*/$parcel$interopDefault($5e9896e6052c3bbb$exports)), 'spectrum-CircleLoader-fillSubMask1'),
        "data-testid": "fillSubMask1",
        style: subMask1Style
    }, /*#__PURE__*/ $byFKM$react.createElement("div", {
        className: $byFKM$classNames((/*@__PURE__*/$parcel$interopDefault($5e9896e6052c3bbb$exports)), 'spectrum-CircleLoader-fill')
    }))), /*#__PURE__*/ $byFKM$react.createElement("div", {
        className: $byFKM$classNames((/*@__PURE__*/$parcel$interopDefault($5e9896e6052c3bbb$exports)), 'spectrum-CircleLoader-fillMask2')
    }, /*#__PURE__*/ $byFKM$react.createElement("div", {
        className: $byFKM$classNames((/*@__PURE__*/$parcel$interopDefault($5e9896e6052c3bbb$exports)), 'spectrum-CircleLoader-fillSubMask2'),
        "data-testid": "fillSubMask2",
        style: subMask2Style
    }, /*#__PURE__*/ $byFKM$react.createElement("div", {
        className: $byFKM$classNames((/*@__PURE__*/$parcel$interopDefault($5e9896e6052c3bbb$exports)), 'spectrum-CircleLoader-fill')
    }))))));
}
/**
 * ProgressCircles show the progression of a system operation such as downloading, uploading, processing, etc. in a visual way.
 * They can represent determinate or indeterminate progress.
 */ let $9890faa22af21f4f$export$c79b9d6b4cc92af7 = /*#__PURE__*/ $byFKM$react.forwardRef($9890faa22af21f4f$var$ProgressCircle);





export {$3e2fc00d5e01478b$export$c17561cb55d4db30 as ProgressBar, $9890faa22af21f4f$export$c79b9d6b4cc92af7 as ProgressCircle, $287d3431be7af3d0$export$7c6ed87244065f3a as ProgressBarBase};
