// @react-spectrum/checkbox/dist/main.js
require("./main.css");
var $15H5o$spectrumiconsuiCheckmarkSmall = require("@spectrum-icons/ui/CheckmarkSmall");
var $15H5o$reactspectrumutils = require("@react-spectrum/utils");
var $15H5o$spectrumiconsuiDashSmall = require("@spectrum-icons/ui/DashSmall");
var $15H5o$reactariafocus = require("@react-aria/focus");
var $15H5o$react = require("react");
var $15H5o$reactariacheckbox = require("@react-aria/checkbox");
var $15H5o$reactariainteractions = require("@react-aria/interactions");
var $15H5o$reactspectrumprovider = require("@react-spectrum/provider");
var $15H5o$reactstatelytoggle = require("@react-stately/toggle");
var $15H5o$reactspectrumlabel = require("@react-spectrum/label");
var $15H5o$reactstatelycheckbox = require("@react-stately/checkbox");
var $15H5o$reactspectrumform = require("@react-spectrum/form");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $654f6cb72ad0c9b1$exports = {};

$parcel$export($654f6cb72ad0c9b1$exports, "Checkbox", () => $654f6cb72ad0c9b1$export$48513f6b9f8ce62d);

const $6c871030f38bcff1$export$baf37c4be89255b8 = ($parcel$interopDefault($15H5o$react)).createContext(null);







var $56db5d48959d3945$exports = {};

$parcel$export($56db5d48959d3945$exports, "spectrum-Checkbox", () => $56db5d48959d3945$export$f8ee3e99d637ccac, (v) => $56db5d48959d3945$export$f8ee3e99d637ccac = v);
$parcel$export($56db5d48959d3945$exports, "spectrum-Checkbox-input", () => $56db5d48959d3945$export$fbf255e91152b086, (v) => $56db5d48959d3945$export$fbf255e91152b086 = v);
$parcel$export($56db5d48959d3945$exports, "focus-ring", () => $56db5d48959d3945$export$f39a09f249340e2a, (v) => $56db5d48959d3945$export$f39a09f249340e2a = v);
$parcel$export($56db5d48959d3945$exports, "spectrum-Checkbox-box", () => $56db5d48959d3945$export$b07bd551df410935, (v) => $56db5d48959d3945$export$b07bd551df410935 = v);
$parcel$export($56db5d48959d3945$exports, "is-indeterminate", () => $56db5d48959d3945$export$682c8615600f6ee, (v) => $56db5d48959d3945$export$682c8615600f6ee = v);
$parcel$export($56db5d48959d3945$exports, "spectrum-Checkbox-checkmark", () => $56db5d48959d3945$export$e152b25f70350d85, (v) => $56db5d48959d3945$export$e152b25f70350d85 = v);
$parcel$export($56db5d48959d3945$exports, "spectrum-Checkbox-partialCheckmark", () => $56db5d48959d3945$export$7f6ea68ad4b1fa69, (v) => $56db5d48959d3945$export$7f6ea68ad4b1fa69 = v);
$parcel$export($56db5d48959d3945$exports, "is-checked", () => $56db5d48959d3945$export$3f46c891456641e7, (v) => $56db5d48959d3945$export$3f46c891456641e7 = v);
$parcel$export($56db5d48959d3945$exports, "spectrum-Checkbox-label", () => $56db5d48959d3945$export$9244b14cfa532d4f, (v) => $56db5d48959d3945$export$9244b14cfa532d4f = v);
$parcel$export($56db5d48959d3945$exports, "is-hovered", () => $56db5d48959d3945$export$b8813cd5d7824ce7, (v) => $56db5d48959d3945$export$b8813cd5d7824ce7 = v);
$parcel$export($56db5d48959d3945$exports, "spectrum-Checkbox--quiet", () => $56db5d48959d3945$export$3a2eb50fe5450439, (v) => $56db5d48959d3945$export$3a2eb50fe5450439 = v);
$parcel$export($56db5d48959d3945$exports, "is-invalid", () => $56db5d48959d3945$export$14a30de2866685fa, (v) => $56db5d48959d3945$export$14a30de2866685fa = v);
var $56db5d48959d3945$export$f8ee3e99d637ccac;
var $56db5d48959d3945$export$fbf255e91152b086;
var $56db5d48959d3945$export$f39a09f249340e2a;
var $56db5d48959d3945$export$b07bd551df410935;
var $56db5d48959d3945$export$682c8615600f6ee;
var $56db5d48959d3945$export$e152b25f70350d85;
var $56db5d48959d3945$export$7f6ea68ad4b1fa69;
var $56db5d48959d3945$export$3f46c891456641e7;
var $56db5d48959d3945$export$9244b14cfa532d4f;
var $56db5d48959d3945$export$b8813cd5d7824ce7;
var $56db5d48959d3945$export$3a2eb50fe5450439;
var $56db5d48959d3945$export$14a30de2866685fa;
$56db5d48959d3945$export$f8ee3e99d637ccac = "spectrum-Checkbox_4870fc";
$56db5d48959d3945$export$fbf255e91152b086 = "spectrum-Checkbox-input_4870fc";
$56db5d48959d3945$export$f39a09f249340e2a = "focus-ring_4870fc";
$56db5d48959d3945$export$b07bd551df410935 = "spectrum-Checkbox-box_4870fc";
$56db5d48959d3945$export$682c8615600f6ee = "is-indeterminate_4870fc";
$56db5d48959d3945$export$e152b25f70350d85 = "spectrum-Checkbox-checkmark_4870fc";
$56db5d48959d3945$export$7f6ea68ad4b1fa69 = "spectrum-Checkbox-partialCheckmark_4870fc";
$56db5d48959d3945$export$3f46c891456641e7 = "is-checked_4870fc";
$56db5d48959d3945$export$9244b14cfa532d4f = "spectrum-Checkbox-label_4870fc";
$56db5d48959d3945$export$b8813cd5d7824ce7 = "is-hovered_4870fc";
$56db5d48959d3945$export$3a2eb50fe5450439 = "spectrum-Checkbox--quiet_4870fc";
$56db5d48959d3945$export$14a30de2866685fa = "is-invalid_4870fc";






function $654f6cb72ad0c9b1$var$Checkbox(props, ref) {
    let originalProps = props;
    props = $15H5o$reactspectrumprovider.useProviderProps(props);
    let { isIndeterminate: isIndeterminate = false , isEmphasized: isEmphasized = false , isDisabled: isDisabled = false , autoFocus: autoFocus , children: children , validationState: validationState , ...otherProps } = props;
    let { styleProps: styleProps  } = $15H5o$reactspectrumutils.useStyleProps(otherProps);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $15H5o$reactariainteractions.useHover({
        isDisabled: isDisabled
    });
    let inputRef = $15H5o$react.useRef(null);
    let domRef = $15H5o$reactspectrumutils.useFocusableRef(ref, inputRef);
    // Swap hooks depending on whether this checkbox is inside a CheckboxGroup.
    // This is a bit unorthodox. Typically, hooks cannot be called in a conditional,
    // but since the checkbox won't move in and out of a group, it should be safe.
    let groupState = $15H5o$react.useContext($6c871030f38bcff1$export$baf37c4be89255b8);
    let { inputProps: inputProps  } = groupState ? $15H5o$reactariacheckbox.useCheckboxGroupItem({
        ...props,
        // Value is optional for standalone checkboxes, but required for CheckboxGroup items;
        // it's passed explicitly here to avoid typescript error (requires strictNullChecks disabled).
        value: props.value,
        // Only pass isRequired and validationState to react-aria if they came from
        // the props for this individual checkbox, and not from the group via context.
        isRequired: originalProps.isRequired,
        validationState: originalProps.validationState
    }, groupState, inputRef) : $15H5o$reactariacheckbox.useCheckbox(props, $15H5o$reactstatelytoggle.useToggleState(props), inputRef);
    let markIcon = isIndeterminate ? /*#__PURE__*/ ($parcel$interopDefault($15H5o$react)).createElement(($parcel$interopDefault($15H5o$spectrumiconsuiDashSmall)), {
        UNSAFE_className: $15H5o$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($56db5d48959d3945$exports)), 'spectrum-Checkbox-partialCheckmark')
    }) : /*#__PURE__*/ ($parcel$interopDefault($15H5o$react)).createElement(($parcel$interopDefault($15H5o$spectrumiconsuiCheckmarkSmall)), {
        UNSAFE_className: $15H5o$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($56db5d48959d3945$exports)), 'spectrum-Checkbox-checkmark')
    });
    if (groupState) {
        for (let key of [
            'isSelected',
            'defaultSelected',
            'isEmphasized'
        ])if (originalProps[key] != null) console.warn(`${key} is unsupported on individual <Checkbox> elements within a <CheckboxGroup>. Please apply these props to the group instead.`);
        if (props.value == null) console.warn('A <Checkbox> element within a <CheckboxGroup> requires a `value` property.');
    }
    return(/*#__PURE__*/ ($parcel$interopDefault($15H5o$react)).createElement("label", {
        ...styleProps,
        ...hoverProps,
        ref: domRef,
        className: $15H5o$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($56db5d48959d3945$exports)), 'spectrum-Checkbox', {
            'is-checked': inputProps.checked,
            'is-indeterminate': isIndeterminate,
            'spectrum-Checkbox--quiet': !isEmphasized,
            'is-invalid': validationState === 'invalid',
            'is-disabled': isDisabled,
            'is-hovered': isHovered
        }, styleProps.className)
    }, /*#__PURE__*/ ($parcel$interopDefault($15H5o$react)).createElement($15H5o$reactariafocus.FocusRing, {
        focusRingClass: $15H5o$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($56db5d48959d3945$exports)), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ ($parcel$interopDefault($15H5o$react)).createElement("input", {
        ...inputProps,
        ref: inputRef,
        className: $15H5o$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($56db5d48959d3945$exports)), 'spectrum-Checkbox-input')
    })), /*#__PURE__*/ ($parcel$interopDefault($15H5o$react)).createElement("span", {
        className: $15H5o$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($56db5d48959d3945$exports)), 'spectrum-Checkbox-box')
    }, markIcon), children && /*#__PURE__*/ ($parcel$interopDefault($15H5o$react)).createElement("span", {
        className: $15H5o$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($56db5d48959d3945$exports)), 'spectrum-Checkbox-label')
    }, children)));
}
/**
 * Checkboxes allow users to select multiple items from a list of individual items,
 * or to mark one individual item as selected.
 */ let $654f6cb72ad0c9b1$export$48513f6b9f8ce62d = /*#__PURE__*/ $15H5o$react.forwardRef($654f6cb72ad0c9b1$var$Checkbox);


var $81d9ca8553250823$exports = {};

$parcel$export($81d9ca8553250823$exports, "CheckboxGroup", () => $81d9ca8553250823$export$4aa08d5625cb8ead);



var $0043d8d062ef413c$exports = {};

$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel", () => $0043d8d062ef413c$export$493d18e796ae054, (v) => $0043d8d062ef413c$export$493d18e796ae054 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel--positionSide", () => $0043d8d062ef413c$export$9e6f19dc21f22f2e, (v) => $0043d8d062ef413c$export$9e6f19dc21f22f2e = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel-requiredIcon", () => $0043d8d062ef413c$export$7c47642c7d46f3c9, (v) => $0043d8d062ef413c$export$7c47642c7d46f3c9 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-FieldLabel--alignEnd", () => $0043d8d062ef413c$export$885efcc08143a987, (v) => $0043d8d062ef413c$export$885efcc08143a987 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field", () => $0043d8d062ef413c$export$f6d480ae1e56eba0, (v) => $0043d8d062ef413c$export$f6d480ae1e56eba0 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field--positionTop", () => $0043d8d062ef413c$export$a4ea780a9064d7f9, (v) => $0043d8d062ef413c$export$a4ea780a9064d7f9 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field-field", () => $0043d8d062ef413c$export$3ab8a3dc7f0563d, (v) => $0043d8d062ef413c$export$3ab8a3dc7f0563d = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field-field--multiline", () => $0043d8d062ef413c$export$f58a1e966a92ba5e, (v) => $0043d8d062ef413c$export$f58a1e966a92ba5e = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field--positionSide", () => $0043d8d062ef413c$export$2963225c91179589, (v) => $0043d8d062ef413c$export$2963225c91179589 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Field-wrapper", () => $0043d8d062ef413c$export$127df0b7290eb8ef, (v) => $0043d8d062ef413c$export$127df0b7290eb8ef = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form", () => $0043d8d062ef413c$export$94d2bcc94afabd89, (v) => $0043d8d062ef413c$export$94d2bcc94afabd89 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form--positionSide", () => $0043d8d062ef413c$export$7f44db659563c8f4, (v) => $0043d8d062ef413c$export$7f44db659563c8f4 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form--positionTop", () => $0043d8d062ef413c$export$11b8bc81fe551057, (v) => $0043d8d062ef413c$export$11b8bc81fe551057 = v);
$parcel$export($0043d8d062ef413c$exports, "spectrum-Form-itemLabel", () => $0043d8d062ef413c$export$51ec27881a1e3928, (v) => $0043d8d062ef413c$export$51ec27881a1e3928 = v);
$parcel$export($0043d8d062ef413c$exports, "is-disabled", () => $0043d8d062ef413c$export$d35bc1e505d1ebbf, (v) => $0043d8d062ef413c$export$d35bc1e505d1ebbf = v);
var $0043d8d062ef413c$export$493d18e796ae054;
var $0043d8d062ef413c$export$9e6f19dc21f22f2e;
var $0043d8d062ef413c$export$7c47642c7d46f3c9;
var $0043d8d062ef413c$export$885efcc08143a987;
var $0043d8d062ef413c$export$f6d480ae1e56eba0;
var $0043d8d062ef413c$export$a4ea780a9064d7f9;
var $0043d8d062ef413c$export$3ab8a3dc7f0563d;
var $0043d8d062ef413c$export$f58a1e966a92ba5e;
var $0043d8d062ef413c$export$2963225c91179589;
var $0043d8d062ef413c$export$127df0b7290eb8ef;
var $0043d8d062ef413c$export$94d2bcc94afabd89;
var $0043d8d062ef413c$export$7f44db659563c8f4;
var $0043d8d062ef413c$export$11b8bc81fe551057;
var $0043d8d062ef413c$export$51ec27881a1e3928;
var $0043d8d062ef413c$export$d35bc1e505d1ebbf;
$0043d8d062ef413c$export$493d18e796ae054 = "spectrum-FieldLabel_d2db1f";
$0043d8d062ef413c$export$9e6f19dc21f22f2e = "spectrum-FieldLabel--positionSide_d2db1f";
$0043d8d062ef413c$export$7c47642c7d46f3c9 = "spectrum-FieldLabel-requiredIcon_d2db1f";
$0043d8d062ef413c$export$885efcc08143a987 = "spectrum-FieldLabel--alignEnd_d2db1f";
$0043d8d062ef413c$export$f6d480ae1e56eba0 = "spectrum-Field_d2db1f";
$0043d8d062ef413c$export$a4ea780a9064d7f9 = "spectrum-Field--positionTop_d2db1f";
$0043d8d062ef413c$export$3ab8a3dc7f0563d = "spectrum-Field-field_d2db1f";
$0043d8d062ef413c$export$f58a1e966a92ba5e = "spectrum-Field-field--multiline_d2db1f";
$0043d8d062ef413c$export$2963225c91179589 = "spectrum-Field--positionSide_d2db1f";
$0043d8d062ef413c$export$127df0b7290eb8ef = "spectrum-Field-wrapper_d2db1f";
$0043d8d062ef413c$export$94d2bcc94afabd89 = "spectrum-Form_d2db1f";
$0043d8d062ef413c$export$7f44db659563c8f4 = "spectrum-Form--positionSide_d2db1f";
$0043d8d062ef413c$export$11b8bc81fe551057 = "spectrum-Form--positionTop_d2db1f";
$0043d8d062ef413c$export$51ec27881a1e3928 = "spectrum-Form-itemLabel_d2db1f";
$0043d8d062ef413c$export$d35bc1e505d1ebbf = "is-disabled_d2db1f";




var $d2824def51242b29$exports = {};

$parcel$export($d2824def51242b29$exports, "spectrum-FieldGroup", () => $d2824def51242b29$export$75eb16cbfc18279c, (v) => $d2824def51242b29$export$75eb16cbfc18279c = v);
$parcel$export($d2824def51242b29$exports, "spectrum-FieldGroup--positionSide", () => $d2824def51242b29$export$859b606bd9f45f81, (v) => $d2824def51242b29$export$859b606bd9f45f81 = v);
$parcel$export($d2824def51242b29$exports, "spectrum-FieldGroup-group", () => $d2824def51242b29$export$984c7bbbfc7ddffc, (v) => $d2824def51242b29$export$984c7bbbfc7ddffc = v);
$parcel$export($d2824def51242b29$exports, "spectrum-FieldGroup-group--horizontal", () => $d2824def51242b29$export$1ad8646bcba21c0e, (v) => $d2824def51242b29$export$1ad8646bcba21c0e = v);
var $d2824def51242b29$export$75eb16cbfc18279c;
var $d2824def51242b29$export$859b606bd9f45f81;
var $d2824def51242b29$export$984c7bbbfc7ddffc;
var $d2824def51242b29$export$1ad8646bcba21c0e;
$d2824def51242b29$export$75eb16cbfc18279c = "spectrum-FieldGroup_41d350";
$d2824def51242b29$export$859b606bd9f45f81 = "spectrum-FieldGroup--positionSide_41d350";
$d2824def51242b29$export$984c7bbbfc7ddffc = "spectrum-FieldGroup-group_41d350";
$d2824def51242b29$export$1ad8646bcba21c0e = "spectrum-FieldGroup-group--horizontal_41d350";





function $81d9ca8553250823$var$CheckboxGroup(props, ref) {
    props = $15H5o$reactspectrumprovider.useProviderProps(props);
    props = $15H5o$reactspectrumform.useFormProps(props);
    let { isEmphasized: isEmphasized , isRequired: isRequired , necessityIndicator: necessityIndicator , label: label , labelPosition: labelPosition = 'top' , labelAlign: labelAlign , children: children , orientation: orientation = 'vertical' , validationState: validationState , ...otherProps } = props;
    let domRef = $15H5o$reactspectrumutils.useDOMRef(ref);
    let { styleProps: styleProps  } = $15H5o$reactspectrumutils.useStyleProps(otherProps);
    let state = $15H5o$reactstatelycheckbox.useCheckboxGroupState(props);
    let { labelProps: labelProps , groupProps: groupProps  } = $15H5o$reactariacheckbox.useCheckboxGroup(props, state);
    return(/*#__PURE__*/ ($parcel$interopDefault($15H5o$react)).createElement("div", {
        ...styleProps,
        ...groupProps,
        className: $15H5o$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d2824def51242b29$exports)), 'spectrum-FieldGroup', {
            'spectrum-FieldGroup--positionSide': labelPosition === 'side'
        }, // This is so checkbox group works inside a <Form>
        $15H5o$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($0043d8d062ef413c$exports)), 'spectrum-Field'), styleProps.className),
        ref: domRef
    }, label && /*#__PURE__*/ ($parcel$interopDefault($15H5o$react)).createElement($15H5o$reactspectrumlabel.Label, {
        ...labelProps,
        elementType: "span",
        labelPosition: labelPosition,
        labelAlign: labelAlign,
        isRequired: isRequired,
        necessityIndicator: necessityIndicator,
        includeNecessityIndicatorInAccessibilityName: true
    }, label), /*#__PURE__*/ ($parcel$interopDefault($15H5o$react)).createElement("div", {
        role: "presentation",
        className: $15H5o$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d2824def51242b29$exports)), 'spectrum-FieldGroup-group', {
            'spectrum-FieldGroup-group--horizontal': orientation === 'horizontal'
        })
    }, /*#__PURE__*/ ($parcel$interopDefault($15H5o$react)).createElement($15H5o$reactspectrumprovider.Provider, {
        isEmphasized: isEmphasized,
        validationState: validationState
    }, /*#__PURE__*/ ($parcel$interopDefault($15H5o$react)).createElement($6c871030f38bcff1$export$baf37c4be89255b8.Provider, {
        value: state
    }, children)))));
}
/**
 * A CheckboxGroup allows users to select one or more items from a list of choices.
 */ const $81d9ca8553250823$export$4aa08d5625cb8ead = /*#__PURE__*/ ($parcel$interopDefault($15H5o$react)).forwardRef($81d9ca8553250823$var$CheckboxGroup);


$parcel$exportWildcard(module.exports, $654f6cb72ad0c9b1$exports);
$parcel$exportWildcard(module.exports, $81d9ca8553250823$exports);
