// @react-spectrum/checkbox/dist/module.js
import "./main.css";
import $62FUl$spectrumiconsuiCheckmarkSmall from "@spectrum-icons/ui/CheckmarkSmall";
import {useStyleProps as $62FUl$useStyleProps, useFocusableRef as $62FUl$useFocusableRef, classNames as $62FUl$classNames, useDOMRef as $62FUl$useDOMRef} from "@react-spectrum/utils";
import $62FUl$spectrumiconsuiDashSmall from "@spectrum-icons/ui/DashSmall";
import {FocusRing as $62FUl$FocusRing} from "@react-aria/focus";
import $62FUl$react, {useRef as $62FUl$useRef, useContext as $62FUl$useContext, forwardRef as $62FUl$forwardRef} from "react";
import {useCheckboxGroupItem as $62FUl$useCheckboxGroupItem, useCheckbox as $62FUl$useCheckbox, useCheckboxGroup as $62FUl$useCheckboxGroup} from "@react-aria/checkbox";
import {useHover as $62FUl$useHover} from "@react-aria/interactions";
import {useProviderProps as $62FUl$useProviderProps, Provider as $62FUl$Provider} from "@react-spectrum/provider";
import {useToggleState as $62FUl$useToggleState} from "@react-stately/toggle";
import {Label as $62FUl$Label} from "@react-spectrum/label";
import {useCheckboxGroupState as $62FUl$useCheckboxGroupState} from "@react-stately/checkbox";
import {useFormProps as $62FUl$useFormProps} from "@react-spectrum/form";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $0a4d54ef74031de7$exports = {};

$parcel$export($0a4d54ef74031de7$exports, "Checkbox", () => $0a4d54ef74031de7$export$48513f6b9f8ce62d);

const $6ea9f741bd70fd45$export$baf37c4be89255b8 = $62FUl$react.createContext(null);







var $0d6b589493ad0e85$exports = {};

$parcel$export($0d6b589493ad0e85$exports, "spectrum-Checkbox", () => $0d6b589493ad0e85$export$f8ee3e99d637ccac, (v) => $0d6b589493ad0e85$export$f8ee3e99d637ccac = v);
$parcel$export($0d6b589493ad0e85$exports, "spectrum-Checkbox-input", () => $0d6b589493ad0e85$export$fbf255e91152b086, (v) => $0d6b589493ad0e85$export$fbf255e91152b086 = v);
$parcel$export($0d6b589493ad0e85$exports, "focus-ring", () => $0d6b589493ad0e85$export$f39a09f249340e2a, (v) => $0d6b589493ad0e85$export$f39a09f249340e2a = v);
$parcel$export($0d6b589493ad0e85$exports, "spectrum-Checkbox-box", () => $0d6b589493ad0e85$export$b07bd551df410935, (v) => $0d6b589493ad0e85$export$b07bd551df410935 = v);
$parcel$export($0d6b589493ad0e85$exports, "is-indeterminate", () => $0d6b589493ad0e85$export$682c8615600f6ee, (v) => $0d6b589493ad0e85$export$682c8615600f6ee = v);
$parcel$export($0d6b589493ad0e85$exports, "spectrum-Checkbox-checkmark", () => $0d6b589493ad0e85$export$e152b25f70350d85, (v) => $0d6b589493ad0e85$export$e152b25f70350d85 = v);
$parcel$export($0d6b589493ad0e85$exports, "spectrum-Checkbox-partialCheckmark", () => $0d6b589493ad0e85$export$7f6ea68ad4b1fa69, (v) => $0d6b589493ad0e85$export$7f6ea68ad4b1fa69 = v);
$parcel$export($0d6b589493ad0e85$exports, "is-checked", () => $0d6b589493ad0e85$export$3f46c891456641e7, (v) => $0d6b589493ad0e85$export$3f46c891456641e7 = v);
$parcel$export($0d6b589493ad0e85$exports, "spectrum-Checkbox-label", () => $0d6b589493ad0e85$export$9244b14cfa532d4f, (v) => $0d6b589493ad0e85$export$9244b14cfa532d4f = v);
$parcel$export($0d6b589493ad0e85$exports, "is-hovered", () => $0d6b589493ad0e85$export$b8813cd5d7824ce7, (v) => $0d6b589493ad0e85$export$b8813cd5d7824ce7 = v);
$parcel$export($0d6b589493ad0e85$exports, "spectrum-Checkbox--quiet", () => $0d6b589493ad0e85$export$3a2eb50fe5450439, (v) => $0d6b589493ad0e85$export$3a2eb50fe5450439 = v);
$parcel$export($0d6b589493ad0e85$exports, "is-invalid", () => $0d6b589493ad0e85$export$14a30de2866685fa, (v) => $0d6b589493ad0e85$export$14a30de2866685fa = v);
var $0d6b589493ad0e85$export$f8ee3e99d637ccac;
var $0d6b589493ad0e85$export$fbf255e91152b086;
var $0d6b589493ad0e85$export$f39a09f249340e2a;
var $0d6b589493ad0e85$export$b07bd551df410935;
var $0d6b589493ad0e85$export$682c8615600f6ee;
var $0d6b589493ad0e85$export$e152b25f70350d85;
var $0d6b589493ad0e85$export$7f6ea68ad4b1fa69;
var $0d6b589493ad0e85$export$3f46c891456641e7;
var $0d6b589493ad0e85$export$9244b14cfa532d4f;
var $0d6b589493ad0e85$export$b8813cd5d7824ce7;
var $0d6b589493ad0e85$export$3a2eb50fe5450439;
var $0d6b589493ad0e85$export$14a30de2866685fa;
$0d6b589493ad0e85$export$f8ee3e99d637ccac = "spectrum-Checkbox_4870fc";
$0d6b589493ad0e85$export$fbf255e91152b086 = "spectrum-Checkbox-input_4870fc";
$0d6b589493ad0e85$export$f39a09f249340e2a = "focus-ring_4870fc";
$0d6b589493ad0e85$export$b07bd551df410935 = "spectrum-Checkbox-box_4870fc";
$0d6b589493ad0e85$export$682c8615600f6ee = "is-indeterminate_4870fc";
$0d6b589493ad0e85$export$e152b25f70350d85 = "spectrum-Checkbox-checkmark_4870fc";
$0d6b589493ad0e85$export$7f6ea68ad4b1fa69 = "spectrum-Checkbox-partialCheckmark_4870fc";
$0d6b589493ad0e85$export$3f46c891456641e7 = "is-checked_4870fc";
$0d6b589493ad0e85$export$9244b14cfa532d4f = "spectrum-Checkbox-label_4870fc";
$0d6b589493ad0e85$export$b8813cd5d7824ce7 = "is-hovered_4870fc";
$0d6b589493ad0e85$export$3a2eb50fe5450439 = "spectrum-Checkbox--quiet_4870fc";
$0d6b589493ad0e85$export$14a30de2866685fa = "is-invalid_4870fc";






function $0a4d54ef74031de7$var$Checkbox(props, ref) {
    let originalProps = props;
    props = $62FUl$useProviderProps(props);
    let { isIndeterminate: isIndeterminate = false , isEmphasized: isEmphasized = false , isDisabled: isDisabled = false , autoFocus: autoFocus , children: children , validationState: validationState , ...otherProps } = props;
    let { styleProps: styleProps  } = $62FUl$useStyleProps(otherProps);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $62FUl$useHover({
        isDisabled: isDisabled
    });
    let inputRef = $62FUl$useRef(null);
    let domRef = $62FUl$useFocusableRef(ref, inputRef);
    // Swap hooks depending on whether this checkbox is inside a CheckboxGroup.
    // This is a bit unorthodox. Typically, hooks cannot be called in a conditional,
    // but since the checkbox won't move in and out of a group, it should be safe.
    let groupState = $62FUl$useContext($6ea9f741bd70fd45$export$baf37c4be89255b8);
    let { inputProps: inputProps  } = groupState ? $62FUl$useCheckboxGroupItem({
        ...props,
        // Value is optional for standalone checkboxes, but required for CheckboxGroup items;
        // it's passed explicitly here to avoid typescript error (requires strictNullChecks disabled).
        value: props.value,
        // Only pass isRequired and validationState to react-aria if they came from
        // the props for this individual checkbox, and not from the group via context.
        isRequired: originalProps.isRequired,
        validationState: originalProps.validationState
    }, groupState, inputRef) : $62FUl$useCheckbox(props, $62FUl$useToggleState(props), inputRef);
    let markIcon = isIndeterminate ? /*#__PURE__*/ $62FUl$react.createElement($62FUl$spectrumiconsuiDashSmall, {
        UNSAFE_className: $62FUl$classNames((/*@__PURE__*/$parcel$interopDefault($0d6b589493ad0e85$exports)), 'spectrum-Checkbox-partialCheckmark')
    }) : /*#__PURE__*/ $62FUl$react.createElement($62FUl$spectrumiconsuiCheckmarkSmall, {
        UNSAFE_className: $62FUl$classNames((/*@__PURE__*/$parcel$interopDefault($0d6b589493ad0e85$exports)), 'spectrum-Checkbox-checkmark')
    });
    if (groupState) {
        for (let key of [
            'isSelected',
            'defaultSelected',
            'isEmphasized'
        ])if (originalProps[key] != null) console.warn(`${key} is unsupported on individual <Checkbox> elements within a <CheckboxGroup>. Please apply these props to the group instead.`);
        if (props.value == null) console.warn('A <Checkbox> element within a <CheckboxGroup> requires a `value` property.');
    }
    return(/*#__PURE__*/ $62FUl$react.createElement("label", {
        ...styleProps,
        ...hoverProps,
        ref: domRef,
        className: $62FUl$classNames((/*@__PURE__*/$parcel$interopDefault($0d6b589493ad0e85$exports)), 'spectrum-Checkbox', {
            'is-checked': inputProps.checked,
            'is-indeterminate': isIndeterminate,
            'spectrum-Checkbox--quiet': !isEmphasized,
            'is-invalid': validationState === 'invalid',
            'is-disabled': isDisabled,
            'is-hovered': isHovered
        }, styleProps.className)
    }, /*#__PURE__*/ $62FUl$react.createElement($62FUl$FocusRing, {
        focusRingClass: $62FUl$classNames((/*@__PURE__*/$parcel$interopDefault($0d6b589493ad0e85$exports)), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ $62FUl$react.createElement("input", {
        ...inputProps,
        ref: inputRef,
        className: $62FUl$classNames((/*@__PURE__*/$parcel$interopDefault($0d6b589493ad0e85$exports)), 'spectrum-Checkbox-input')
    })), /*#__PURE__*/ $62FUl$react.createElement("span", {
        className: $62FUl$classNames((/*@__PURE__*/$parcel$interopDefault($0d6b589493ad0e85$exports)), 'spectrum-Checkbox-box')
    }, markIcon), children && /*#__PURE__*/ $62FUl$react.createElement("span", {
        className: $62FUl$classNames((/*@__PURE__*/$parcel$interopDefault($0d6b589493ad0e85$exports)), 'spectrum-Checkbox-label')
    }, children)));
}
/**
 * Checkboxes allow users to select multiple items from a list of individual items,
 * or to mark one individual item as selected.
 */ let $0a4d54ef74031de7$export$48513f6b9f8ce62d = /*#__PURE__*/ $62FUl$forwardRef($0a4d54ef74031de7$var$Checkbox);


var $9a9832fe4c651f53$exports = {};

$parcel$export($9a9832fe4c651f53$exports, "CheckboxGroup", () => $9a9832fe4c651f53$export$4aa08d5625cb8ead);



var $4a84444d5a56ab41$exports = {};

$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel", () => $4a84444d5a56ab41$export$493d18e796ae054, (v) => $4a84444d5a56ab41$export$493d18e796ae054 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel--positionSide", () => $4a84444d5a56ab41$export$9e6f19dc21f22f2e, (v) => $4a84444d5a56ab41$export$9e6f19dc21f22f2e = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel-requiredIcon", () => $4a84444d5a56ab41$export$7c47642c7d46f3c9, (v) => $4a84444d5a56ab41$export$7c47642c7d46f3c9 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-FieldLabel--alignEnd", () => $4a84444d5a56ab41$export$885efcc08143a987, (v) => $4a84444d5a56ab41$export$885efcc08143a987 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field", () => $4a84444d5a56ab41$export$f6d480ae1e56eba0, (v) => $4a84444d5a56ab41$export$f6d480ae1e56eba0 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field--positionTop", () => $4a84444d5a56ab41$export$a4ea780a9064d7f9, (v) => $4a84444d5a56ab41$export$a4ea780a9064d7f9 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field-field", () => $4a84444d5a56ab41$export$3ab8a3dc7f0563d, (v) => $4a84444d5a56ab41$export$3ab8a3dc7f0563d = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field-field--multiline", () => $4a84444d5a56ab41$export$f58a1e966a92ba5e, (v) => $4a84444d5a56ab41$export$f58a1e966a92ba5e = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field--positionSide", () => $4a84444d5a56ab41$export$2963225c91179589, (v) => $4a84444d5a56ab41$export$2963225c91179589 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Field-wrapper", () => $4a84444d5a56ab41$export$127df0b7290eb8ef, (v) => $4a84444d5a56ab41$export$127df0b7290eb8ef = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form", () => $4a84444d5a56ab41$export$94d2bcc94afabd89, (v) => $4a84444d5a56ab41$export$94d2bcc94afabd89 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form--positionSide", () => $4a84444d5a56ab41$export$7f44db659563c8f4, (v) => $4a84444d5a56ab41$export$7f44db659563c8f4 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form--positionTop", () => $4a84444d5a56ab41$export$11b8bc81fe551057, (v) => $4a84444d5a56ab41$export$11b8bc81fe551057 = v);
$parcel$export($4a84444d5a56ab41$exports, "spectrum-Form-itemLabel", () => $4a84444d5a56ab41$export$51ec27881a1e3928, (v) => $4a84444d5a56ab41$export$51ec27881a1e3928 = v);
$parcel$export($4a84444d5a56ab41$exports, "is-disabled", () => $4a84444d5a56ab41$export$d35bc1e505d1ebbf, (v) => $4a84444d5a56ab41$export$d35bc1e505d1ebbf = v);
var $4a84444d5a56ab41$export$493d18e796ae054;
var $4a84444d5a56ab41$export$9e6f19dc21f22f2e;
var $4a84444d5a56ab41$export$7c47642c7d46f3c9;
var $4a84444d5a56ab41$export$885efcc08143a987;
var $4a84444d5a56ab41$export$f6d480ae1e56eba0;
var $4a84444d5a56ab41$export$a4ea780a9064d7f9;
var $4a84444d5a56ab41$export$3ab8a3dc7f0563d;
var $4a84444d5a56ab41$export$f58a1e966a92ba5e;
var $4a84444d5a56ab41$export$2963225c91179589;
var $4a84444d5a56ab41$export$127df0b7290eb8ef;
var $4a84444d5a56ab41$export$94d2bcc94afabd89;
var $4a84444d5a56ab41$export$7f44db659563c8f4;
var $4a84444d5a56ab41$export$11b8bc81fe551057;
var $4a84444d5a56ab41$export$51ec27881a1e3928;
var $4a84444d5a56ab41$export$d35bc1e505d1ebbf;
$4a84444d5a56ab41$export$493d18e796ae054 = "spectrum-FieldLabel_d2db1f";
$4a84444d5a56ab41$export$9e6f19dc21f22f2e = "spectrum-FieldLabel--positionSide_d2db1f";
$4a84444d5a56ab41$export$7c47642c7d46f3c9 = "spectrum-FieldLabel-requiredIcon_d2db1f";
$4a84444d5a56ab41$export$885efcc08143a987 = "spectrum-FieldLabel--alignEnd_d2db1f";
$4a84444d5a56ab41$export$f6d480ae1e56eba0 = "spectrum-Field_d2db1f";
$4a84444d5a56ab41$export$a4ea780a9064d7f9 = "spectrum-Field--positionTop_d2db1f";
$4a84444d5a56ab41$export$3ab8a3dc7f0563d = "spectrum-Field-field_d2db1f";
$4a84444d5a56ab41$export$f58a1e966a92ba5e = "spectrum-Field-field--multiline_d2db1f";
$4a84444d5a56ab41$export$2963225c91179589 = "spectrum-Field--positionSide_d2db1f";
$4a84444d5a56ab41$export$127df0b7290eb8ef = "spectrum-Field-wrapper_d2db1f";
$4a84444d5a56ab41$export$94d2bcc94afabd89 = "spectrum-Form_d2db1f";
$4a84444d5a56ab41$export$7f44db659563c8f4 = "spectrum-Form--positionSide_d2db1f";
$4a84444d5a56ab41$export$11b8bc81fe551057 = "spectrum-Form--positionTop_d2db1f";
$4a84444d5a56ab41$export$51ec27881a1e3928 = "spectrum-Form-itemLabel_d2db1f";
$4a84444d5a56ab41$export$d35bc1e505d1ebbf = "is-disabled_d2db1f";




var $003163fb2d58eced$exports = {};

$parcel$export($003163fb2d58eced$exports, "spectrum-FieldGroup", () => $003163fb2d58eced$export$75eb16cbfc18279c, (v) => $003163fb2d58eced$export$75eb16cbfc18279c = v);
$parcel$export($003163fb2d58eced$exports, "spectrum-FieldGroup--positionSide", () => $003163fb2d58eced$export$859b606bd9f45f81, (v) => $003163fb2d58eced$export$859b606bd9f45f81 = v);
$parcel$export($003163fb2d58eced$exports, "spectrum-FieldGroup-group", () => $003163fb2d58eced$export$984c7bbbfc7ddffc, (v) => $003163fb2d58eced$export$984c7bbbfc7ddffc = v);
$parcel$export($003163fb2d58eced$exports, "spectrum-FieldGroup-group--horizontal", () => $003163fb2d58eced$export$1ad8646bcba21c0e, (v) => $003163fb2d58eced$export$1ad8646bcba21c0e = v);
var $003163fb2d58eced$export$75eb16cbfc18279c;
var $003163fb2d58eced$export$859b606bd9f45f81;
var $003163fb2d58eced$export$984c7bbbfc7ddffc;
var $003163fb2d58eced$export$1ad8646bcba21c0e;
$003163fb2d58eced$export$75eb16cbfc18279c = "spectrum-FieldGroup_41d350";
$003163fb2d58eced$export$859b606bd9f45f81 = "spectrum-FieldGroup--positionSide_41d350";
$003163fb2d58eced$export$984c7bbbfc7ddffc = "spectrum-FieldGroup-group_41d350";
$003163fb2d58eced$export$1ad8646bcba21c0e = "spectrum-FieldGroup-group--horizontal_41d350";





function $9a9832fe4c651f53$var$CheckboxGroup(props, ref) {
    props = $62FUl$useProviderProps(props);
    props = $62FUl$useFormProps(props);
    let { isEmphasized: isEmphasized , isRequired: isRequired , necessityIndicator: necessityIndicator , label: label , labelPosition: labelPosition = 'top' , labelAlign: labelAlign , children: children , orientation: orientation = 'vertical' , validationState: validationState , ...otherProps } = props;
    let domRef = $62FUl$useDOMRef(ref);
    let { styleProps: styleProps  } = $62FUl$useStyleProps(otherProps);
    let state = $62FUl$useCheckboxGroupState(props);
    let { labelProps: labelProps , groupProps: groupProps  } = $62FUl$useCheckboxGroup(props, state);
    return(/*#__PURE__*/ $62FUl$react.createElement("div", {
        ...styleProps,
        ...groupProps,
        className: $62FUl$classNames((/*@__PURE__*/$parcel$interopDefault($003163fb2d58eced$exports)), 'spectrum-FieldGroup', {
            'spectrum-FieldGroup--positionSide': labelPosition === 'side'
        }, // This is so checkbox group works inside a <Form>
        $62FUl$classNames((/*@__PURE__*/$parcel$interopDefault($4a84444d5a56ab41$exports)), 'spectrum-Field'), styleProps.className),
        ref: domRef
    }, label && /*#__PURE__*/ $62FUl$react.createElement($62FUl$Label, {
        ...labelProps,
        elementType: "span",
        labelPosition: labelPosition,
        labelAlign: labelAlign,
        isRequired: isRequired,
        necessityIndicator: necessityIndicator,
        includeNecessityIndicatorInAccessibilityName: true
    }, label), /*#__PURE__*/ $62FUl$react.createElement("div", {
        role: "presentation",
        className: $62FUl$classNames((/*@__PURE__*/$parcel$interopDefault($003163fb2d58eced$exports)), 'spectrum-FieldGroup-group', {
            'spectrum-FieldGroup-group--horizontal': orientation === 'horizontal'
        })
    }, /*#__PURE__*/ $62FUl$react.createElement($62FUl$Provider, {
        isEmphasized: isEmphasized,
        validationState: validationState
    }, /*#__PURE__*/ $62FUl$react.createElement($6ea9f741bd70fd45$export$baf37c4be89255b8.Provider, {
        value: state
    }, children)))));
}
/**
 * A CheckboxGroup allows users to select one or more items from a list of choices.
 */ const $9a9832fe4c651f53$export$4aa08d5625cb8ead = /*#__PURE__*/ $62FUl$react.forwardRef($9a9832fe4c651f53$var$CheckboxGroup);




export {$0a4d54ef74031de7$export$48513f6b9f8ce62d as Checkbox, $9a9832fe4c651f53$export$4aa08d5625cb8ead as CheckboxGroup};
