// @react-spectrum/tabs/dist/main.js
require("./main.css");
var $e7A5M$reactstatelycollections = require("@react-stately/collections");
var $e7A5M$reactspectrumutils = require("@react-spectrum/utils");
var $e7A5M$reactariautils = require("@react-aria/utils");
var $e7A5M$reactariafocus = require("@react-aria/focus");
var $e7A5M$reactspectrumpicker = require("@react-spectrum/picker");
var $e7A5M$reactstatelylist = require("@react-stately/list");
var $e7A5M$react = require("react");
var $e7A5M$reactstatelytabs = require("@react-stately/tabs");
var $e7A5M$reactspectrumtext = require("@react-spectrum/text");
var $e7A5M$reactariainteractions = require("@react-aria/interactions");
var $e7A5M$reactariai18n = require("@react-aria/i18n");
var $e7A5M$reactspectrumprovider = require("@react-spectrum/provider");
var $e7A5M$reactariatabs = require("@react-aria/tabs");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Item", () => $e7A5M$reactstatelycollections.Item);
var $4b5f7944b4795980$exports = {};

$parcel$export($4b5f7944b4795980$exports, "TabList", () => $4b5f7944b4795980$export$e51a686c67fdaa2d);
$parcel$export($4b5f7944b4795980$exports, "TabPanels", () => $4b5f7944b4795980$export$5dae8d435677f210);
$parcel$export($4b5f7944b4795980$exports, "Tabs", () => $4b5f7944b4795980$export$b2539bed5023c21c);







var $652630c2aefd3d05$exports = {};

$parcel$export($652630c2aefd3d05$exports, "spectrum-TabsPanel", () => $652630c2aefd3d05$export$34f1a65c57bb7bb, (v) => $652630c2aefd3d05$export$34f1a65c57bb7bb = v);
$parcel$export($652630c2aefd3d05$exports, "spectrum-TabsPanel--horizontal", () => $652630c2aefd3d05$export$778fb839eaa0528c, (v) => $652630c2aefd3d05$export$778fb839eaa0528c = v);
$parcel$export($652630c2aefd3d05$exports, "spectrum-TabsPanel--vertical", () => $652630c2aefd3d05$export$9a6e2fb72c648072, (v) => $652630c2aefd3d05$export$9a6e2fb72c648072 = v);
$parcel$export($652630c2aefd3d05$exports, "spectrum-Tabs", () => $652630c2aefd3d05$export$35c105c0d9c8e56f, (v) => $652630c2aefd3d05$export$35c105c0d9c8e56f = v);
$parcel$export($652630c2aefd3d05$exports, "spectrum-Tabs-item", () => $652630c2aefd3d05$export$123e4c01a5440a3e, (v) => $652630c2aefd3d05$export$123e4c01a5440a3e = v);
$parcel$export($652630c2aefd3d05$exports, "is-disabled", () => $652630c2aefd3d05$export$d35bc1e505d1ebbf, (v) => $652630c2aefd3d05$export$d35bc1e505d1ebbf = v);
$parcel$export($652630c2aefd3d05$exports, "spectrum-Tabs-itemLabel", () => $652630c2aefd3d05$export$1f253edac4b98526, (v) => $652630c2aefd3d05$export$1f253edac4b98526 = v);
$parcel$export($652630c2aefd3d05$exports, "spectrum-Icon", () => $652630c2aefd3d05$export$d374b04f30360026, (v) => $652630c2aefd3d05$export$d374b04f30360026 = v);
$parcel$export($652630c2aefd3d05$exports, "spectrum-Tabs-selectionIndicator", () => $652630c2aefd3d05$export$a115f9117a19563e, (v) => $652630c2aefd3d05$export$a115f9117a19563e = v);
$parcel$export($652630c2aefd3d05$exports, "spectrum-Tabs--horizontal", () => $652630c2aefd3d05$export$229f1ac4d3d38f0e, (v) => $652630c2aefd3d05$export$229f1ac4d3d38f0e = v);
$parcel$export($652630c2aefd3d05$exports, "spectrum-Tabs--compact", () => $652630c2aefd3d05$export$e15c78184ee03abe, (v) => $652630c2aefd3d05$export$e15c78184ee03abe = v);
$parcel$export($652630c2aefd3d05$exports, "spectrum-Tabs--isCollapsed", () => $652630c2aefd3d05$export$ea514cd060922d6b, (v) => $652630c2aefd3d05$export$ea514cd060922d6b = v);
$parcel$export($652630c2aefd3d05$exports, "spectrum-Tabs--quiet", () => $652630c2aefd3d05$export$831e830d5828d42b, (v) => $652630c2aefd3d05$export$831e830d5828d42b = v);
$parcel$export($652630c2aefd3d05$exports, "spectrum-Tabs--vertical", () => $652630c2aefd3d05$export$e415005be3957065, (v) => $652630c2aefd3d05$export$e415005be3957065 = v);
$parcel$export($652630c2aefd3d05$exports, "spectrum-TabsPanel-collapseWrapper", () => $652630c2aefd3d05$export$e09869956b4c96f2, (v) => $652630c2aefd3d05$export$e09869956b4c96f2 = v);
$parcel$export($652630c2aefd3d05$exports, "spectrum-TabsPanel-tabs", () => $652630c2aefd3d05$export$4a157a691c854e23, (v) => $652630c2aefd3d05$export$4a157a691c854e23 = v);
$parcel$export($652630c2aefd3d05$exports, "spectrum-TabsPanel-tabpanel", () => $652630c2aefd3d05$export$acb5da0970288266, (v) => $652630c2aefd3d05$export$acb5da0970288266 = v);
$parcel$export($652630c2aefd3d05$exports, "focus-ring", () => $652630c2aefd3d05$export$f39a09f249340e2a, (v) => $652630c2aefd3d05$export$f39a09f249340e2a = v);
$parcel$export($652630c2aefd3d05$exports, "is-hovered", () => $652630c2aefd3d05$export$b8813cd5d7824ce7, (v) => $652630c2aefd3d05$export$b8813cd5d7824ce7 = v);
$parcel$export($652630c2aefd3d05$exports, "is-selected", () => $652630c2aefd3d05$export$1e0fb04f31d3c22a, (v) => $652630c2aefd3d05$export$1e0fb04f31d3c22a = v);
var $652630c2aefd3d05$export$34f1a65c57bb7bb;
var $652630c2aefd3d05$export$778fb839eaa0528c;
var $652630c2aefd3d05$export$9a6e2fb72c648072;
var $652630c2aefd3d05$export$35c105c0d9c8e56f;
var $652630c2aefd3d05$export$123e4c01a5440a3e;
var $652630c2aefd3d05$export$d35bc1e505d1ebbf;
var $652630c2aefd3d05$export$1f253edac4b98526;
var $652630c2aefd3d05$export$d374b04f30360026;
var $652630c2aefd3d05$export$a115f9117a19563e;
var $652630c2aefd3d05$export$229f1ac4d3d38f0e;
var $652630c2aefd3d05$export$e15c78184ee03abe;
var $652630c2aefd3d05$export$ea514cd060922d6b;
var $652630c2aefd3d05$export$831e830d5828d42b;
var $652630c2aefd3d05$export$e415005be3957065;
var $652630c2aefd3d05$export$e09869956b4c96f2;
var $652630c2aefd3d05$export$4a157a691c854e23;
var $652630c2aefd3d05$export$acb5da0970288266;
var $652630c2aefd3d05$export$f39a09f249340e2a;
var $652630c2aefd3d05$export$b8813cd5d7824ce7;
var $652630c2aefd3d05$export$1e0fb04f31d3c22a;
$652630c2aefd3d05$export$34f1a65c57bb7bb = "spectrum-TabsPanel_3e140b";
$652630c2aefd3d05$export$778fb839eaa0528c = "spectrum-TabsPanel--horizontal_3e140b";
$652630c2aefd3d05$export$9a6e2fb72c648072 = "spectrum-TabsPanel--vertical_3e140b";
$652630c2aefd3d05$export$35c105c0d9c8e56f = "spectrum-Tabs_3e140b";
$652630c2aefd3d05$export$123e4c01a5440a3e = "spectrum-Tabs-item_3e140b";
$652630c2aefd3d05$export$d35bc1e505d1ebbf = "is-disabled_3e140b";
$652630c2aefd3d05$export$1f253edac4b98526 = "spectrum-Tabs-itemLabel_3e140b";
$652630c2aefd3d05$export$d374b04f30360026 = "spectrum-Icon_3e140b";
$652630c2aefd3d05$export$a115f9117a19563e = "spectrum-Tabs-selectionIndicator_3e140b";
$652630c2aefd3d05$export$229f1ac4d3d38f0e = "spectrum-Tabs--horizontal_3e140b";
$652630c2aefd3d05$export$e15c78184ee03abe = "spectrum-Tabs--compact_3e140b";
$652630c2aefd3d05$export$ea514cd060922d6b = "spectrum-Tabs--isCollapsed_3e140b";
$652630c2aefd3d05$export$831e830d5828d42b = "spectrum-Tabs--quiet_3e140b";
$652630c2aefd3d05$export$e415005be3957065 = "spectrum-Tabs--vertical_3e140b";
$652630c2aefd3d05$export$e09869956b4c96f2 = "spectrum-TabsPanel-collapseWrapper_3e140b";
$652630c2aefd3d05$export$4a157a691c854e23 = "spectrum-TabsPanel-tabs_3e140b";
$652630c2aefd3d05$export$acb5da0970288266 = "spectrum-TabsPanel-tabpanel_3e140b";
$652630c2aefd3d05$export$f39a09f249340e2a = "focus-ring_3e140b";
$652630c2aefd3d05$export$b8813cd5d7824ce7 = "is-hovered_3e140b";
$652630c2aefd3d05$export$1e0fb04f31d3c22a = "is-selected_3e140b";










const $4b5f7944b4795980$var$TabContext = /*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createContext(null);
function $4b5f7944b4795980$var$Tabs(props, ref) {
    props = $e7A5M$reactspectrumprovider.useProviderProps(props);
    let { orientation: orientation = 'horizontal' , density: density = 'regular' , children: children , ...otherProps } = props;
    let domRef = $e7A5M$reactspectrumutils.useDOMRef(ref);
    let tablistRef = $e7A5M$react.useRef();
    let wrapperRef = $e7A5M$react.useRef();
    let { direction: direction  } = $e7A5M$reactariai18n.useLocale();
    let { styleProps: styleProps  } = $e7A5M$reactspectrumutils.useStyleProps(otherProps);
    let [collapse, setCollapse] = $e7A5M$reactariautils.useValueEffect(false);
    let [selectedTab1, setSelectedTab] = $e7A5M$react.useState();
    const [tabListState, setTabListState] = $e7A5M$react.useState(null);
    $e7A5M$react.useEffect(()=>{
        if (tablistRef.current) {
            let selectedTab = tablistRef.current.querySelector(`[data-key="${tabListState === null || tabListState === void 0 ? void 0 : tabListState.selectedKey}"]`);
            if (selectedTab != null) setSelectedTab(selectedTab);
        }
    // collapse is in the dep array so selectedTab can be updated for TabLine positioning
    }, [
        children,
        tabListState === null || tabListState === void 0 ? void 0 : tabListState.selectedKey,
        collapse,
        tablistRef
    ]);
    let checkShouldCollapse = $e7A5M$react.useCallback(()=>{
        let computeShouldCollapse = ()=>{
            if (wrapperRef.current) {
                let tabsComponent = wrapperRef.current;
                let tabs = tablistRef.current.querySelectorAll('[role="tab"]');
                let lastTab = tabs[tabs.length - 1];
                let end = direction === 'rtl' ? 'left' : 'right';
                let farEdgeTabList = tabsComponent.getBoundingClientRect()[end];
                let farEdgeLastTab = lastTab === null || lastTab === void 0 ? void 0 : lastTab.getBoundingClientRect()[end];
                let shouldCollapse = direction === 'rtl' ? farEdgeLastTab < farEdgeTabList : farEdgeTabList < farEdgeLastTab;
                return shouldCollapse;
            }
        };
        if (orientation !== 'vertical') setCollapse(function*() {
            // Make Tabs render in non-collapsed state
            yield false;
            // Compute if Tabs should collapse and update
            yield computeShouldCollapse();
        });
    }, [
        tablistRef,
        wrapperRef,
        direction,
        orientation,
        setCollapse
    ]);
    $e7A5M$react.useEffect(()=>{
        checkShouldCollapse();
    }, [
        children,
        checkShouldCollapse
    ]);
    $e7A5M$reactariautils.useResizeObserver({
        ref: wrapperRef,
        onResize: checkShouldCollapse
    });
    let tabPanelProps = {
        'aria-labelledby': undefined
    };
    // When the tabs are collapsed, the tabPanel should be labelled by the Picker button element.
    let collapsibleTabListId = $e7A5M$reactariautils.useId();
    if (collapse && orientation !== 'vertical') tabPanelProps['aria-labelledby'] = collapsibleTabListId;
    return(/*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement($4b5f7944b4795980$var$TabContext.Provider, {
        value: {
            tabProps: {
                ...props,
                orientation: orientation,
                density: density
            },
            tabState: {
                tabListState: tabListState,
                setTabListState: setTabListState,
                selectedTab: selectedTab1,
                collapse: collapse
            },
            refs: {
                tablistRef: tablistRef,
                wrapperRef: wrapperRef
            },
            tabPanelProps: tabPanelProps
        }
    }, /*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement("div", {
        ...$e7A5M$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef,
        className: $e7A5M$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($652630c2aefd3d05$exports)), 'spectrum-TabsPanel', `spectrum-TabsPanel--${orientation}`, styleProps.className)
    }, props.children)));
}
// @private
function $4b5f7944b4795980$var$Tab(props) {
    let { item: item , state: state , isDisabled: propsDisabled  } = props;
    let { key: key , rendered: rendered  } = item;
    let isDisabled = propsDisabled || state.disabledKeys.has(key);
    let ref = $e7A5M$react.useRef();
    let { tabProps: tabProps  } = $e7A5M$reactariatabs.useTab({
        key: key,
        isDisabled: isDisabled
    }, state, ref);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $e7A5M$reactariainteractions.useHover({
        ...props
    });
    let isSelected = state.selectedKey === key;
    let domProps = $e7A5M$reactariautils.filterDOMProps(item.props);
    delete domProps.id;
    return(/*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement($e7A5M$reactariafocus.FocusRing, {
        focusRingClass: $e7A5M$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($652630c2aefd3d05$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement("div", {
        ...$e7A5M$reactariautils.mergeProps(tabProps, hoverProps, domProps),
        ref: ref,
        className: $e7A5M$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($652630c2aefd3d05$exports)), 'spectrum-Tabs-item', {
            'is-selected': isSelected,
            'is-disabled': isDisabled,
            'is-hovered': isHovered
        })
    }, /*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement($e7A5M$reactspectrumutils.SlotProvider, {
        slots: {
            icon: {
                size: 'S',
                UNSAFE_className: $e7A5M$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($652630c2aefd3d05$exports)), 'spectrum-Icon')
            },
            text: {
                UNSAFE_className: $e7A5M$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($652630c2aefd3d05$exports)), 'spectrum-Tabs-itemLabel')
            }
        }
    }, typeof rendered === 'string' ? /*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement($e7A5M$reactspectrumtext.Text, null, rendered) : rendered))));
}
// @private
function $4b5f7944b4795980$var$TabLine(props) {
    let { orientation: orientation , selectedTab: // Is either the tab node (non-collapsed) or the picker node (collapsed)
    selectedTab , selectedKey: // selectedKey is provided so that the TabLine styles are updated when the TabPicker's width updates from a selection change
    selectedKey  } = props;
    let { direction: direction  } = $e7A5M$reactariai18n.useLocale();
    let { scale: scale  } = $e7A5M$reactspectrumprovider.useProvider();
    let [style, setStyle] = $e7A5M$react.useState({
        width: undefined,
        height: undefined
    });
    $e7A5M$reactariautils.useLayoutEffect(()=>{
        if (selectedTab) {
            var ref;
            let styleObj = {
                transform: undefined,
                width: undefined,
                height: undefined
            };
            // In RTL, calculate the transform from the right edge of the tablist so that resizing the window doesn't break the Tabline position due to offsetLeft changes
            let offset = direction === 'rtl' ? -1 * (((ref = selectedTab.offsetParent) === null || ref === void 0 ? void 0 : ref.offsetWidth) - selectedTab.offsetWidth - selectedTab.offsetLeft) : selectedTab.offsetLeft;
            styleObj.transform = orientation === 'vertical' ? `translateY(${selectedTab.offsetTop}px)` : `translateX(${offset}px)`;
            if (orientation === 'horizontal') styleObj.width = `${selectedTab.offsetWidth}px`;
            else styleObj.height = `${selectedTab.offsetHeight}px`;
            setStyle(styleObj);
        }
    }, [
        direction,
        setStyle,
        selectedTab,
        orientation,
        scale,
        selectedKey
    ]);
    return(/*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement("div", {
        className: $e7A5M$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($652630c2aefd3d05$exports)), 'spectrum-Tabs-selectionIndicator'),
        role: "presentation",
        style: style
    }));
}
function $4b5f7944b4795980$export$e51a686c67fdaa2d(props) {
    const tabContext = $e7A5M$react.useContext($4b5f7944b4795980$var$TabContext);
    const { refs: refs , tabState: tabState , tabProps: tabProps , tabPanelProps: tabPanelProps  } = tabContext;
    const { isQuiet: isQuiet , density: density , isDisabled: isDisabled , orientation: orientation  } = tabProps;
    const { selectedTab: selectedTab , collapse: collapse , setTabListState: setTabListState  } = tabState;
    const { tablistRef: tablistRef , wrapperRef: wrapperRef  } = refs;
    // Pass original Tab props but override children to create the collection.
    const state = $e7A5M$reactstatelytabs.useTabListState({
        ...tabProps,
        children: props.children
    });
    let { styleProps: styleProps  } = $e7A5M$reactspectrumutils.useStyleProps(props);
    const { tabListProps: tabListProps  } = $e7A5M$reactariatabs.useTabList({
        ...tabProps,
        ...props
    }, state, tablistRef);
    $e7A5M$react.useEffect(()=>{
        // Passing back to root as useTabPanel needs the TabListState
        setTabListState(state);
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [
        state.disabledKeys,
        state.selectedItem,
        state.selectedKey,
        props.children
    ]);
    let stylePropsForVertical = orientation === 'vertical' ? styleProps : {
    };
    let tabListclassName = $e7A5M$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($652630c2aefd3d05$exports)), 'spectrum-TabsPanel-tabs');
    const tabContent = /*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement("div", {
        ...stylePropsForVertical,
        ...tabListProps,
        ref: tablistRef,
        className: $e7A5M$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($652630c2aefd3d05$exports)), 'spectrum-Tabs', `spectrum-Tabs--${orientation}`, tabListclassName, {
            'spectrum-Tabs--quiet': isQuiet,
            ['spectrum-Tabs--compact']: density === 'compact'
        }, orientation === 'vertical' && styleProps.className)
    }, [
        ...state.collection
    ].map((item)=>/*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement($4b5f7944b4795980$var$Tab, {
            key: item.key,
            item: item,
            state: state,
            isDisabled: isDisabled,
            orientation: orientation
        })
    ), /*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement($4b5f7944b4795980$var$TabLine, {
        orientation: orientation,
        selectedTab: selectedTab
    }));
    if (orientation === 'vertical') return tabContent;
    else return(/*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement("div", {
        ...styleProps,
        ref: wrapperRef,
        className: $e7A5M$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($652630c2aefd3d05$exports)), 'spectrum-TabsPanel-collapseWrapper', styleProps.className)
    }, collapse ? /*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement($4b5f7944b4795980$var$TabPicker, {
        ...props,
        ...tabProps,
        id: tabPanelProps['aria-labelledby'],
        state: state,
        className: tabListclassName
    }) : tabContent));
}
function $4b5f7944b4795980$export$5dae8d435677f210(props) {
    const { tabState: tabState , tabProps: tabProps  } = $e7A5M$react.useContext($4b5f7944b4795980$var$TabContext);
    const { tabListState: tabListState  } = tabState;
    const factory = (nodes)=>new $e7A5M$reactstatelylist.ListCollection(nodes)
    ;
    const collection = $e7A5M$reactstatelycollections.useCollection({
        items: tabProps.items,
        ...props
    }, factory, {
        suppressTextValueWarning: true
    });
    const selectedItem = tabListState ? collection.getItem(tabListState.selectedKey) : null;
    return(/*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement($4b5f7944b4795980$var$TabPanel, {
        ...props,
        key: tabListState === null || tabListState === void 0 ? void 0 : tabListState.selectedKey
    }, selectedItem && selectedItem.props.children));
}
// @private
function $4b5f7944b4795980$var$TabPanel(props) {
    const { tabState: tabState , tabPanelProps: ctxTabPanelProps  } = $e7A5M$react.useContext($4b5f7944b4795980$var$TabContext);
    const { tabListState: tabListState  } = tabState;
    let ref = $e7A5M$react.useRef();
    const { tabPanelProps: tabPanelProps  } = $e7A5M$reactariatabs.useTabPanel(props, tabListState, ref);
    let { styleProps: styleProps  } = $e7A5M$reactspectrumutils.useStyleProps(props);
    if (ctxTabPanelProps['aria-labelledby']) tabPanelProps['aria-labelledby'] = ctxTabPanelProps['aria-labelledby'];
    return(/*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement($e7A5M$reactariafocus.FocusRing, {
        focusRingClass: $e7A5M$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($652630c2aefd3d05$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement("div", {
        ...styleProps,
        ...tabPanelProps,
        ref: ref,
        className: $e7A5M$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($652630c2aefd3d05$exports)), 'spectrum-TabsPanel-tabpanel', styleProps.className)
    }, props.children)));
}
function $4b5f7944b4795980$var$TabPicker(props) {
    let { isDisabled: isDisabled , isQuiet: isQuiet , state: state , 'aria-labelledby': ariaLabeledBy , 'aria-label': ariaLabel , density: density , className: className , id: id  } = props;
    let ref = $e7A5M$react.useRef();
    let [pickerNode, setPickerNode] = $e7A5M$react.useState(null);
    $e7A5M$react.useEffect(()=>{
        let node = $e7A5M$reactspectrumutils.unwrapDOMRef(ref);
        setPickerNode(node.current);
    }, [
        ref
    ]);
    let items = [
        ...state.collection
    ].map((item)=>({
            rendered: item.rendered,
            textValue: item.textValue,
            id: item.key
        })
    );
    let pickerProps = {
        'aria-labelledby': ariaLabeledBy,
        'aria-label': ariaLabel
    };
    // TODO: Figure out if tabListProps should go onto the div here, v2 doesn't do it
    return(/*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement("div", {
        className: $e7A5M$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($652630c2aefd3d05$exports)), 'spectrum-Tabs', 'spectrum-Tabs--horizontal', 'spectrum-Tabs--isCollapsed', {
            'spectrum-Tabs--quiet': isQuiet,
            ['spectrum-Tabs--compact']: density === 'compact'
        }, className)
    }, /*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement($e7A5M$reactspectrumutils.SlotProvider, {
        slots: {
            icon: {
                size: 'S',
                UNSAFE_className: $e7A5M$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($652630c2aefd3d05$exports)), 'spectrum-Icon')
            },
            button: {
                focusRingClass: $e7A5M$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($652630c2aefd3d05$exports)), 'focus-ring')
            }
        }
    }, /*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement($e7A5M$reactspectrumpicker.Picker, {
        ...pickerProps,
        id: id,
        items: items,
        ref: ref,
        isQuiet: true,
        isDisabled: isDisabled,
        selectedKey: state.selectedKey,
        disabledKeys: state.disabledKeys,
        onSelectionChange: state.setSelectedKey
    }, (item)=>/*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement($e7A5M$reactspectrumpicker.Item, {
            textValue: item.textValue
        }, item.rendered)
    ), pickerNode && /*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).createElement($4b5f7944b4795980$var$TabLine, {
        orientation: "horizontal",
        selectedTab: pickerNode,
        selectedKey: state.selectedKey
    }))));
}
/**
 * Tabs organize content into multiple sections and allow users to navigate between them. The content under the set of tabs should be related and form a coherent unit.
 */ // forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
const $4b5f7944b4795980$export$b2539bed5023c21c = /*#__PURE__*/ ($parcel$interopDefault($e7A5M$react)).forwardRef($4b5f7944b4795980$var$Tabs);



$parcel$exportWildcard(module.exports, $4b5f7944b4795980$exports);
