// @react-spectrum/theme-default/dist/main.js
require("./main.css");

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "theme", () => $5ea826602639494b$export$bca14c5b3b88a9c9);
var $4025619061c54f77$exports = {};

$parcel$export($4025619061c54f77$exports, "spectrum--darkest", () => $4025619061c54f77$export$4ecdba604f5f1f44, (v) => $4025619061c54f77$export$4ecdba604f5f1f44 = v);
var $4025619061c54f77$export$4ecdba604f5f1f44;
$4025619061c54f77$export$4ecdba604f5f1f44 = "spectrum--darkest_256eeb";


var $af8bc55c9d966366$exports = {};

$parcel$export($af8bc55c9d966366$exports, "spectrum", () => $af8bc55c9d966366$export$3311ab3a441bc141, (v) => $af8bc55c9d966366$export$3311ab3a441bc141 = v);
$parcel$export($af8bc55c9d966366$exports, "spectrum--medium", () => $af8bc55c9d966366$export$4c0c83b3f4303ef8, (v) => $af8bc55c9d966366$export$4c0c83b3f4303ef8 = v);
$parcel$export($af8bc55c9d966366$exports, "spectrum--large", () => $af8bc55c9d966366$export$a88a8dbe29386d31, (v) => $af8bc55c9d966366$export$a88a8dbe29386d31 = v);
$parcel$export($af8bc55c9d966366$exports, "spectrum--darkest", () => $af8bc55c9d966366$export$4ecdba604f5f1f44, (v) => $af8bc55c9d966366$export$4ecdba604f5f1f44 = v);
$parcel$export($af8bc55c9d966366$exports, "spectrum--dark", () => $af8bc55c9d966366$export$68dc111a79481afd, (v) => $af8bc55c9d966366$export$68dc111a79481afd = v);
$parcel$export($af8bc55c9d966366$exports, "spectrum--light", () => $af8bc55c9d966366$export$efb37c2f79da8163, (v) => $af8bc55c9d966366$export$efb37c2f79da8163 = v);
$parcel$export($af8bc55c9d966366$exports, "spectrum--lightest", () => $af8bc55c9d966366$export$baaa804dc80cce18, (v) => $af8bc55c9d966366$export$baaa804dc80cce18 = v);
var $af8bc55c9d966366$export$3311ab3a441bc141;
var $af8bc55c9d966366$export$4c0c83b3f4303ef8;
var $af8bc55c9d966366$export$a88a8dbe29386d31;
var $af8bc55c9d966366$export$4ecdba604f5f1f44;
var $af8bc55c9d966366$export$68dc111a79481afd;
var $af8bc55c9d966366$export$efb37c2f79da8163;
var $af8bc55c9d966366$export$baaa804dc80cce18;
$af8bc55c9d966366$export$3311ab3a441bc141 = "spectrum_9e130c";
$af8bc55c9d966366$export$4c0c83b3f4303ef8 = "spectrum--medium_9e130c";
$af8bc55c9d966366$export$a88a8dbe29386d31 = "spectrum--large_9e130c";
$af8bc55c9d966366$export$4ecdba604f5f1f44 = "spectrum--darkest_9e130c";
$af8bc55c9d966366$export$68dc111a79481afd = "spectrum--dark_9e130c";
$af8bc55c9d966366$export$efb37c2f79da8163 = "spectrum--light_9e130c";
$af8bc55c9d966366$export$baaa804dc80cce18 = "spectrum--lightest_9e130c";


var $193d6c718c1151ae$exports = {};

$parcel$export($193d6c718c1151ae$exports, "spectrum--large", () => $193d6c718c1151ae$export$a88a8dbe29386d31, (v) => $193d6c718c1151ae$export$a88a8dbe29386d31 = v);
var $193d6c718c1151ae$export$a88a8dbe29386d31;
$193d6c718c1151ae$export$a88a8dbe29386d31 = "spectrum--large_c40598";


var $b61b6432caa20860$exports = {};

$parcel$export($b61b6432caa20860$exports, "spectrum--light", () => $b61b6432caa20860$export$efb37c2f79da8163, (v) => $b61b6432caa20860$export$efb37c2f79da8163 = v);
var $b61b6432caa20860$export$efb37c2f79da8163;
$b61b6432caa20860$export$efb37c2f79da8163 = "spectrum--light_a40724";


var $30f30767af400b7c$exports = {};

$parcel$export($30f30767af400b7c$exports, "spectrum--medium", () => $30f30767af400b7c$export$4c0c83b3f4303ef8, (v) => $30f30767af400b7c$export$4c0c83b3f4303ef8 = v);
var $30f30767af400b7c$export$4c0c83b3f4303ef8;
$30f30767af400b7c$export$4c0c83b3f4303ef8 = "spectrum--medium_4b172c";


let $5ea826602639494b$export$bca14c5b3b88a9c9 = {
    global: (/*@__PURE__*/$parcel$interopDefault($af8bc55c9d966366$exports)),
    light: (/*@__PURE__*/$parcel$interopDefault($b61b6432caa20860$exports)),
    dark: (/*@__PURE__*/$parcel$interopDefault($4025619061c54f77$exports)),
    medium: (/*@__PURE__*/$parcel$interopDefault($30f30767af400b7c$exports)),
    large: (/*@__PURE__*/$parcel$interopDefault($193d6c718c1151ae$exports))
};
