// @react-spectrum/theme-default/dist/module.js
import "./main.css";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $c3a89c526a7413ef$exports = {};

$parcel$export($c3a89c526a7413ef$exports, "spectrum--darkest", () => $c3a89c526a7413ef$export$4ecdba604f5f1f44, (v) => $c3a89c526a7413ef$export$4ecdba604f5f1f44 = v);
var $c3a89c526a7413ef$export$4ecdba604f5f1f44;
$c3a89c526a7413ef$export$4ecdba604f5f1f44 = "spectrum--darkest_256eeb";


var $2a743218f11c9e8a$exports = {};

$parcel$export($2a743218f11c9e8a$exports, "spectrum", () => $2a743218f11c9e8a$export$3311ab3a441bc141, (v) => $2a743218f11c9e8a$export$3311ab3a441bc141 = v);
$parcel$export($2a743218f11c9e8a$exports, "spectrum--medium", () => $2a743218f11c9e8a$export$4c0c83b3f4303ef8, (v) => $2a743218f11c9e8a$export$4c0c83b3f4303ef8 = v);
$parcel$export($2a743218f11c9e8a$exports, "spectrum--large", () => $2a743218f11c9e8a$export$a88a8dbe29386d31, (v) => $2a743218f11c9e8a$export$a88a8dbe29386d31 = v);
$parcel$export($2a743218f11c9e8a$exports, "spectrum--darkest", () => $2a743218f11c9e8a$export$4ecdba604f5f1f44, (v) => $2a743218f11c9e8a$export$4ecdba604f5f1f44 = v);
$parcel$export($2a743218f11c9e8a$exports, "spectrum--dark", () => $2a743218f11c9e8a$export$68dc111a79481afd, (v) => $2a743218f11c9e8a$export$68dc111a79481afd = v);
$parcel$export($2a743218f11c9e8a$exports, "spectrum--light", () => $2a743218f11c9e8a$export$efb37c2f79da8163, (v) => $2a743218f11c9e8a$export$efb37c2f79da8163 = v);
$parcel$export($2a743218f11c9e8a$exports, "spectrum--lightest", () => $2a743218f11c9e8a$export$baaa804dc80cce18, (v) => $2a743218f11c9e8a$export$baaa804dc80cce18 = v);
var $2a743218f11c9e8a$export$3311ab3a441bc141;
var $2a743218f11c9e8a$export$4c0c83b3f4303ef8;
var $2a743218f11c9e8a$export$a88a8dbe29386d31;
var $2a743218f11c9e8a$export$4ecdba604f5f1f44;
var $2a743218f11c9e8a$export$68dc111a79481afd;
var $2a743218f11c9e8a$export$efb37c2f79da8163;
var $2a743218f11c9e8a$export$baaa804dc80cce18;
$2a743218f11c9e8a$export$3311ab3a441bc141 = "spectrum_9e130c";
$2a743218f11c9e8a$export$4c0c83b3f4303ef8 = "spectrum--medium_9e130c";
$2a743218f11c9e8a$export$a88a8dbe29386d31 = "spectrum--large_9e130c";
$2a743218f11c9e8a$export$4ecdba604f5f1f44 = "spectrum--darkest_9e130c";
$2a743218f11c9e8a$export$68dc111a79481afd = "spectrum--dark_9e130c";
$2a743218f11c9e8a$export$efb37c2f79da8163 = "spectrum--light_9e130c";
$2a743218f11c9e8a$export$baaa804dc80cce18 = "spectrum--lightest_9e130c";


var $f2234261c99e5ffe$exports = {};

$parcel$export($f2234261c99e5ffe$exports, "spectrum--large", () => $f2234261c99e5ffe$export$a88a8dbe29386d31, (v) => $f2234261c99e5ffe$export$a88a8dbe29386d31 = v);
var $f2234261c99e5ffe$export$a88a8dbe29386d31;
$f2234261c99e5ffe$export$a88a8dbe29386d31 = "spectrum--large_c40598";


var $80173824a116cc16$exports = {};

$parcel$export($80173824a116cc16$exports, "spectrum--light", () => $80173824a116cc16$export$efb37c2f79da8163, (v) => $80173824a116cc16$export$efb37c2f79da8163 = v);
var $80173824a116cc16$export$efb37c2f79da8163;
$80173824a116cc16$export$efb37c2f79da8163 = "spectrum--light_a40724";


var $2c1245723b3c6193$exports = {};

$parcel$export($2c1245723b3c6193$exports, "spectrum--medium", () => $2c1245723b3c6193$export$4c0c83b3f4303ef8, (v) => $2c1245723b3c6193$export$4c0c83b3f4303ef8 = v);
var $2c1245723b3c6193$export$4c0c83b3f4303ef8;
$2c1245723b3c6193$export$4c0c83b3f4303ef8 = "spectrum--medium_4b172c";


let $6387683576a4c598$export$bca14c5b3b88a9c9 = {
    global: (/*@__PURE__*/$parcel$interopDefault($2a743218f11c9e8a$exports)),
    light: (/*@__PURE__*/$parcel$interopDefault($80173824a116cc16$exports)),
    dark: (/*@__PURE__*/$parcel$interopDefault($c3a89c526a7413ef$exports)),
    medium: (/*@__PURE__*/$parcel$interopDefault($2c1245723b3c6193$exports)),
    large: (/*@__PURE__*/$parcel$interopDefault($f2234261c99e5ffe$exports))
};


export {$6387683576a4c598$export$bca14c5b3b88a9c9 as theme};
