// @react-aria/searchfield/dist/main.js
var $8vLER$reactariautils = require("@react-aria/utils");
var $8vLER$reactariai18n = require("@react-aria/i18n");
var $8vLER$reactariatextfield = require("@react-aria/textfield");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $71c53ff5c2173186$exports = {};

$parcel$export($71c53ff5c2173186$exports, "useSearchField", () => $71c53ff5c2173186$export$9bb30bbe003b82e0);

var $b10fe5d4b20a1b77$exports = {};
var $a7fccca2fdf21ca1$exports = {};
$a7fccca2fdf21ca1$exports = JSON.parse("{\"Clear search\":\"مسح البحث\"}");


var $0fca214a95313a21$exports = {};
$0fca214a95313a21$exports = JSON.parse("{\"Clear search\":\"Изчистване на търсене\"}");


var $f6b13912eb642f14$exports = {};
$f6b13912eb642f14$exports = JSON.parse("{\"Clear search\":\"Vymazat hledání\"}");


var $efa33f31596acc74$exports = {};
$efa33f31596acc74$exports = JSON.parse("{\"Clear search\":\"Ryd søgning\"}");


var $40abd42f36d53f3c$exports = {};
$40abd42f36d53f3c$exports = JSON.parse("{\"Clear search\":\"Suche zurücksetzen\"}");


var $7ab64b571bea1c65$exports = {};
$7ab64b571bea1c65$exports = JSON.parse("{\"Clear search\":\"Απαλοιφή αναζήτησης\"}");


var $9966466219093c3b$exports = {};
$9966466219093c3b$exports = JSON.parse("{\"Clear search\":\"Clear search\"}");


var $fc21126eb3f29986$exports = {};
$fc21126eb3f29986$exports = JSON.parse("{\"Clear search\":\"Borrar búsqueda\"}");


var $66df07c5ce1139e0$exports = {};
$66df07c5ce1139e0$exports = JSON.parse("{\"Clear search\":\"Tühjenda otsing\"}");


var $954a6e73b1073a21$exports = {};
$954a6e73b1073a21$exports = JSON.parse("{\"Clear search\":\"Tyhjennä haku\"}");


var $59f37ced70618ce7$exports = {};
$59f37ced70618ce7$exports = JSON.parse("{\"Clear search\":\"Effacer la recherche\"}");


var $7ded182beb993735$exports = {};
$7ded182beb993735$exports = JSON.parse("{\"Clear search\":\"נקה חיפוש\"}");


var $133635c4fb1243bc$exports = {};
$133635c4fb1243bc$exports = JSON.parse("{\"Clear search\":\"Obriši pretragu\"}");


var $3d69da9590ce23f2$exports = {};
$3d69da9590ce23f2$exports = JSON.parse("{\"Clear search\":\"Keresés törlése\"}");


var $141877d98cd0f121$exports = {};
$141877d98cd0f121$exports = JSON.parse("{\"Clear search\":\"Cancella ricerca\"}");


var $371b187302b0633c$exports = {};
$371b187302b0633c$exports = JSON.parse("{\"Clear search\":\"検索をクリア\"}");


var $6499c5fb0ff08056$exports = {};
$6499c5fb0ff08056$exports = JSON.parse("{\"Clear search\":\"검색 지우기\"}");


var $8b556ea59f1b48b7$exports = {};
$8b556ea59f1b48b7$exports = JSON.parse("{\"Clear search\":\"Išvalyti iešką\"}");


var $2a0ede34713f9e59$exports = {};
$2a0ede34713f9e59$exports = JSON.parse("{\"Clear search\":\"Notīrīt meklēšanu\"}");


var $0b85c71f9759d72c$exports = {};
$0b85c71f9759d72c$exports = JSON.parse("{\"Clear search\":\"Tøm søk\"}");


var $485956473f3f1d6d$exports = {};
$485956473f3f1d6d$exports = JSON.parse("{\"Clear search\":\"Zoekactie wissen\"}");


var $ae55f3c9c5c9de59$exports = {};
$ae55f3c9c5c9de59$exports = JSON.parse("{\"Clear search\":\"Wyczyść zawartość wyszukiwania\"}");


var $550a74e096165e22$exports = {};
$550a74e096165e22$exports = JSON.parse("{\"Clear search\":\"Limpar pesquisa\"}");


var $718a2369b9916ffa$exports = {};
$718a2369b9916ffa$exports = JSON.parse("{\"Clear search\":\"Limpar pesquisa\"}");


var $d07f7acbc361663e$exports = {};
$d07f7acbc361663e$exports = JSON.parse("{\"Clear search\":\"Ştergeţi căutarea\"}");


var $ebfea54406d2d418$exports = {};
$ebfea54406d2d418$exports = JSON.parse("{\"Clear search\":\"Очистить поиск\"}");


var $ec77d51d7e7f2515$exports = {};
$ec77d51d7e7f2515$exports = JSON.parse("{\"Clear search\":\"Vymazať vyhľadávanie\"}");


var $cb75bdcbfb29b325$exports = {};
$cb75bdcbfb29b325$exports = JSON.parse("{\"Clear search\":\"Počisti iskanje\"}");


var $23cfbc9f6686af10$exports = {};
$23cfbc9f6686af10$exports = JSON.parse("{\"Clear search\":\"Obriši pretragu\"}");


var $dffd9ab409b646f2$exports = {};
$dffd9ab409b646f2$exports = JSON.parse("{\"Clear search\":\"Rensa sökning\"}");


var $44fffce9ce97a6b7$exports = {};
$44fffce9ce97a6b7$exports = JSON.parse("{\"Clear search\":\"Aramayı temizle\"}");


var $8b457dd52ff2340d$exports = {};
$8b457dd52ff2340d$exports = JSON.parse("{\"Clear search\":\"Очистити пошук\"}");


var $2c107bd4768063fe$exports = {};
$2c107bd4768063fe$exports = JSON.parse("{\"Clear search\":\"清除搜索\"}");


var $39c310451cc97816$exports = {};
$39c310451cc97816$exports = JSON.parse("{\"Clear search\":\"清除搜尋條件\"}");


$b10fe5d4b20a1b77$exports = {
    "ar-AE": $a7fccca2fdf21ca1$exports,
    "bg-BG": $0fca214a95313a21$exports,
    "cs-CZ": $f6b13912eb642f14$exports,
    "da-DK": $efa33f31596acc74$exports,
    "de-DE": $40abd42f36d53f3c$exports,
    "el-GR": $7ab64b571bea1c65$exports,
    "en-US": $9966466219093c3b$exports,
    "es-ES": $fc21126eb3f29986$exports,
    "et-EE": $66df07c5ce1139e0$exports,
    "fi-FI": $954a6e73b1073a21$exports,
    "fr-FR": $59f37ced70618ce7$exports,
    "he-IL": $7ded182beb993735$exports,
    "hr-HR": $133635c4fb1243bc$exports,
    "hu-HU": $3d69da9590ce23f2$exports,
    "it-IT": $141877d98cd0f121$exports,
    "ja-JP": $371b187302b0633c$exports,
    "ko-KR": $6499c5fb0ff08056$exports,
    "lt-LT": $8b556ea59f1b48b7$exports,
    "lv-LV": $2a0ede34713f9e59$exports,
    "nb-NO": $0b85c71f9759d72c$exports,
    "nl-NL": $485956473f3f1d6d$exports,
    "pl-PL": $ae55f3c9c5c9de59$exports,
    "pt-BR": $550a74e096165e22$exports,
    "pt-PT": $718a2369b9916ffa$exports,
    "ro-RO": $d07f7acbc361663e$exports,
    "ru-RU": $ebfea54406d2d418$exports,
    "sk-SK": $ec77d51d7e7f2515$exports,
    "sl-SI": $cb75bdcbfb29b325$exports,
    "sr-SP": $23cfbc9f6686af10$exports,
    "sv-SE": $dffd9ab409b646f2$exports,
    "tr-TR": $44fffce9ce97a6b7$exports,
    "uk-UA": $8b457dd52ff2340d$exports,
    "zh-CN": $2c107bd4768063fe$exports,
    "zh-TW": $39c310451cc97816$exports
};




function $71c53ff5c2173186$export$9bb30bbe003b82e0(props, state, inputRef) {
    let formatMessage = $8vLER$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($b10fe5d4b20a1b77$exports)));
    let { isDisabled: isDisabled , onSubmit: onSubmit = ()=>{
    } , onClear: onClear , type: type = 'search'  } = props;
    let onKeyDown = (e)=>{
        const key = e.key;
        if (key === 'Enter' || key === 'Escape') e.preventDefault();
        if (isDisabled) return;
        if (key === 'Enter') onSubmit(state.value);
        if (key === 'Escape') {
            state.setValue('');
            if (onClear) onClear();
        }
    };
    let onClearButtonClick = ()=>{
        state.setValue('');
        if (onClear) onClear();
    };
    let onPressStart = ()=>{
        // this is in PressStart for mobile so that touching the clear button doesn't remove focus from
        // the input and close the keyboard
        inputRef.current.focus();
    };
    let { labelProps: labelProps , inputProps: inputProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $8vLER$reactariatextfield.useTextField({
        ...props,
        value: state.value,
        onChange: state.setValue,
        onKeyDown: $8vLER$reactariautils.chain(onKeyDown, props.onKeyDown),
        type: type
    }, inputRef);
    return {
        labelProps: labelProps,
        inputProps: {
            ...inputProps,
            // already handled by useSearchFieldState
            defaultValue: undefined
        },
        clearButtonProps: {
            'aria-label': formatMessage('Clear search'),
            excludeFromTabOrder: true,
            // @ts-ignore
            preventFocusOnPress: true,
            onPress: onClearButtonClick,
            onPressStart: onPressStart
        },
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps
    };
}


$parcel$exportWildcard(module.exports, $71c53ff5c2173186$exports);
