// @react-aria/searchfield/dist/module.js
import {chain as $9cxG6$chain} from "@react-aria/utils";
import {useMessageFormatter as $9cxG6$useMessageFormatter} from "@react-aria/i18n";
import {useTextField as $9cxG6$useTextField} from "@react-aria/textfield";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $aa8e55a3584c38e6$exports = {};

$parcel$export($aa8e55a3584c38e6$exports, "useSearchField", () => $aa8e55a3584c38e6$export$9bb30bbe003b82e0);

var $6b44e231e1a319cd$exports = {};
var $0bb15c7a71b7c5f6$exports = {};
$0bb15c7a71b7c5f6$exports = JSON.parse("{\"Clear search\":\"مسح البحث\"}");


var $92d32378d44dbb09$exports = {};
$92d32378d44dbb09$exports = JSON.parse("{\"Clear search\":\"Изчистване на търсене\"}");


var $8b53b64df650ed59$exports = {};
$8b53b64df650ed59$exports = JSON.parse("{\"Clear search\":\"Vymazat hledání\"}");


var $6e26a098d0e52588$exports = {};
$6e26a098d0e52588$exports = JSON.parse("{\"Clear search\":\"Ryd søgning\"}");


var $a4ffa820395601cf$exports = {};
$a4ffa820395601cf$exports = JSON.parse("{\"Clear search\":\"Suche zurücksetzen\"}");


var $824108fb29b6eb42$exports = {};
$824108fb29b6eb42$exports = JSON.parse("{\"Clear search\":\"Απαλοιφή αναζήτησης\"}");


var $026d6db9ec48264a$exports = {};
$026d6db9ec48264a$exports = JSON.parse("{\"Clear search\":\"Clear search\"}");


var $3c137c448b6af362$exports = {};
$3c137c448b6af362$exports = JSON.parse("{\"Clear search\":\"Borrar búsqueda\"}");


var $aa3a9e02560bd5dc$exports = {};
$aa3a9e02560bd5dc$exports = JSON.parse("{\"Clear search\":\"Tühjenda otsing\"}");


var $c8462fe7a5cf7e4b$exports = {};
$c8462fe7a5cf7e4b$exports = JSON.parse("{\"Clear search\":\"Tyhjennä haku\"}");


var $12e8af98aa3094fe$exports = {};
$12e8af98aa3094fe$exports = JSON.parse("{\"Clear search\":\"Effacer la recherche\"}");


var $0df95379721dc393$exports = {};
$0df95379721dc393$exports = JSON.parse("{\"Clear search\":\"נקה חיפוש\"}");


var $f0e2a2cf03464987$exports = {};
$f0e2a2cf03464987$exports = JSON.parse("{\"Clear search\":\"Obriši pretragu\"}");


var $d0eefa8a0b3b6d0d$exports = {};
$d0eefa8a0b3b6d0d$exports = JSON.parse("{\"Clear search\":\"Keresés törlése\"}");


var $82eba93f490e1076$exports = {};
$82eba93f490e1076$exports = JSON.parse("{\"Clear search\":\"Cancella ricerca\"}");


var $dcdcc1ae2fd7acac$exports = {};
$dcdcc1ae2fd7acac$exports = JSON.parse("{\"Clear search\":\"検索をクリア\"}");


var $bd2e60125ba0181a$exports = {};
$bd2e60125ba0181a$exports = JSON.parse("{\"Clear search\":\"검색 지우기\"}");


var $68ef1a4f55971a8e$exports = {};
$68ef1a4f55971a8e$exports = JSON.parse("{\"Clear search\":\"Išvalyti iešką\"}");


var $65b66acfb905e67e$exports = {};
$65b66acfb905e67e$exports = JSON.parse("{\"Clear search\":\"Notīrīt meklēšanu\"}");


var $de888e4dcc7d5904$exports = {};
$de888e4dcc7d5904$exports = JSON.parse("{\"Clear search\":\"Tøm søk\"}");


var $a4b5b26c1a191114$exports = {};
$a4b5b26c1a191114$exports = JSON.parse("{\"Clear search\":\"Zoekactie wissen\"}");


var $54831e302c8bd080$exports = {};
$54831e302c8bd080$exports = JSON.parse("{\"Clear search\":\"Wyczyść zawartość wyszukiwania\"}");


var $95b60b6b05b70049$exports = {};
$95b60b6b05b70049$exports = JSON.parse("{\"Clear search\":\"Limpar pesquisa\"}");


var $0c25f17944e83e5d$exports = {};
$0c25f17944e83e5d$exports = JSON.parse("{\"Clear search\":\"Limpar pesquisa\"}");


var $d6e291f4309cc99f$exports = {};
$d6e291f4309cc99f$exports = JSON.parse("{\"Clear search\":\"Ştergeţi căutarea\"}");


var $a01cb1dc97100220$exports = {};
$a01cb1dc97100220$exports = JSON.parse("{\"Clear search\":\"Очистить поиск\"}");


var $3f8fcc8fc099f0ee$exports = {};
$3f8fcc8fc099f0ee$exports = JSON.parse("{\"Clear search\":\"Vymazať vyhľadávanie\"}");


var $3c4c1f2f80f16d72$exports = {};
$3c4c1f2f80f16d72$exports = JSON.parse("{\"Clear search\":\"Počisti iskanje\"}");


var $c0d795a2e2e76c49$exports = {};
$c0d795a2e2e76c49$exports = JSON.parse("{\"Clear search\":\"Obriši pretragu\"}");


var $6582f39062699ffb$exports = {};
$6582f39062699ffb$exports = JSON.parse("{\"Clear search\":\"Rensa sökning\"}");


var $a10776dd7c505254$exports = {};
$a10776dd7c505254$exports = JSON.parse("{\"Clear search\":\"Aramayı temizle\"}");


var $39f88a725f0c8bae$exports = {};
$39f88a725f0c8bae$exports = JSON.parse("{\"Clear search\":\"Очистити пошук\"}");


var $f532a6e5a32b7893$exports = {};
$f532a6e5a32b7893$exports = JSON.parse("{\"Clear search\":\"清除搜索\"}");


var $2c3be1f93d9b301e$exports = {};
$2c3be1f93d9b301e$exports = JSON.parse("{\"Clear search\":\"清除搜尋條件\"}");


$6b44e231e1a319cd$exports = {
    "ar-AE": $0bb15c7a71b7c5f6$exports,
    "bg-BG": $92d32378d44dbb09$exports,
    "cs-CZ": $8b53b64df650ed59$exports,
    "da-DK": $6e26a098d0e52588$exports,
    "de-DE": $a4ffa820395601cf$exports,
    "el-GR": $824108fb29b6eb42$exports,
    "en-US": $026d6db9ec48264a$exports,
    "es-ES": $3c137c448b6af362$exports,
    "et-EE": $aa3a9e02560bd5dc$exports,
    "fi-FI": $c8462fe7a5cf7e4b$exports,
    "fr-FR": $12e8af98aa3094fe$exports,
    "he-IL": $0df95379721dc393$exports,
    "hr-HR": $f0e2a2cf03464987$exports,
    "hu-HU": $d0eefa8a0b3b6d0d$exports,
    "it-IT": $82eba93f490e1076$exports,
    "ja-JP": $dcdcc1ae2fd7acac$exports,
    "ko-KR": $bd2e60125ba0181a$exports,
    "lt-LT": $68ef1a4f55971a8e$exports,
    "lv-LV": $65b66acfb905e67e$exports,
    "nb-NO": $de888e4dcc7d5904$exports,
    "nl-NL": $a4b5b26c1a191114$exports,
    "pl-PL": $54831e302c8bd080$exports,
    "pt-BR": $95b60b6b05b70049$exports,
    "pt-PT": $0c25f17944e83e5d$exports,
    "ro-RO": $d6e291f4309cc99f$exports,
    "ru-RU": $a01cb1dc97100220$exports,
    "sk-SK": $3f8fcc8fc099f0ee$exports,
    "sl-SI": $3c4c1f2f80f16d72$exports,
    "sr-SP": $c0d795a2e2e76c49$exports,
    "sv-SE": $6582f39062699ffb$exports,
    "tr-TR": $a10776dd7c505254$exports,
    "uk-UA": $39f88a725f0c8bae$exports,
    "zh-CN": $f532a6e5a32b7893$exports,
    "zh-TW": $2c3be1f93d9b301e$exports
};




function $aa8e55a3584c38e6$export$9bb30bbe003b82e0(props, state, inputRef) {
    let formatMessage = $9cxG6$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($6b44e231e1a319cd$exports)));
    let { isDisabled: isDisabled , onSubmit: onSubmit = ()=>{
    } , onClear: onClear , type: type = 'search'  } = props;
    let onKeyDown = (e)=>{
        const key = e.key;
        if (key === 'Enter' || key === 'Escape') e.preventDefault();
        if (isDisabled) return;
        if (key === 'Enter') onSubmit(state.value);
        if (key === 'Escape') {
            state.setValue('');
            if (onClear) onClear();
        }
    };
    let onClearButtonClick = ()=>{
        state.setValue('');
        if (onClear) onClear();
    };
    let onPressStart = ()=>{
        // this is in PressStart for mobile so that touching the clear button doesn't remove focus from
        // the input and close the keyboard
        inputRef.current.focus();
    };
    let { labelProps: labelProps , inputProps: inputProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $9cxG6$useTextField({
        ...props,
        value: state.value,
        onChange: state.setValue,
        onKeyDown: $9cxG6$chain(onKeyDown, props.onKeyDown),
        type: type
    }, inputRef);
    return {
        labelProps: labelProps,
        inputProps: {
            ...inputProps,
            // already handled by useSearchFieldState
            defaultValue: undefined
        },
        clearButtonProps: {
            'aria-label': formatMessage('Clear search'),
            excludeFromTabOrder: true,
            // @ts-ignore
            preventFocusOnPress: true,
            onPress: onClearButtonClick,
            onPressStart: onPressStart
        },
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps
    };
}




export {$aa8e55a3584c38e6$export$9bb30bbe003b82e0 as useSearchField};
