// @react-spectrum/listbox/dist/main.js
require("./main.css");
var $l8Zb1$reactstatelycollections = require("@react-stately/collections");
var $l8Zb1$react = require("react");
var $l8Zb1$reactspectrumutils = require("@react-spectrum/utils");
var $l8Zb1$reactstatelylist = require("@react-stately/list");
var $l8Zb1$reactarialistbox = require("@react-aria/listbox");
var $l8Zb1$reactstatelylayout = require("@react-stately/layout");
var $l8Zb1$reactariautils = require("@react-aria/utils");
var $l8Zb1$reactspectrumprogress = require("@react-spectrum/progress");
var $l8Zb1$reactariai18n = require("@react-aria/i18n");
var $l8Zb1$reactspectrumprovider = require("@react-spectrum/provider");
var $l8Zb1$reactariavirtualizer = require("@react-aria/virtualizer");
var $l8Zb1$spectrumiconsuiCheckmarkMedium = require("@spectrum-icons/ui/CheckmarkMedium");
var $l8Zb1$reactariafocus = require("@react-aria/focus");
var $l8Zb1$reactspectrumlayout = require("@react-spectrum/layout");
var $l8Zb1$reactariainteractions = require("@react-aria/interactions");
var $l8Zb1$reactspectrumtext = require("@react-spectrum/text");
var $l8Zb1$reactariaseparator = require("@react-aria/separator");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Item", () => $l8Zb1$reactstatelycollections.Item);
$parcel$export(module.exports, "Section", () => $l8Zb1$reactstatelycollections.Section);
var $5cde28c545cc4211$exports = {};

$parcel$export($5cde28c545cc4211$exports, "ListBox", () => $5cde28c545cc4211$export$41f133550aa26f48);
var $24b64fc51b3c824b$exports = {};

$parcel$export($24b64fc51b3c824b$exports, "useListBoxLayout", () => $24b64fc51b3c824b$export$25768ea656ae32a7);
$parcel$export($24b64fc51b3c824b$exports, "ListBoxBase", () => $24b64fc51b3c824b$export$1afdcf349979fb7e);


var $97a5c01ed556d8f5$exports = {};
var $c9c03464d18bae68$exports = {};
$c9c03464d18bae68$exports = JSON.parse("{\"loading\":\"جارٍ التحميل...\",\"loadingMore\":\"جارٍ تحميل المزيد...\"}");


var $57766d982c55bc35$exports = {};
$57766d982c55bc35$exports = JSON.parse("{\"loading\":\"Зареждане...\",\"loadingMore\":\"Зареждане на още...\"}");


var $a4963ce26b58d483$exports = {};
$a4963ce26b58d483$exports = JSON.parse("{\"loading\":\"Načítání...\",\"loadingMore\":\"Načítání dalších...\"}");


var $5b711ac7321b2584$exports = {};
$5b711ac7321b2584$exports = JSON.parse("{\"loading\":\"Indlæser ...\",\"loadingMore\":\"Indlæser flere ...\"}");


var $aa07c3f6a1018646$exports = {};
$aa07c3f6a1018646$exports = JSON.parse("{\"loading\":\"Laden...\",\"loadingMore\":\"Mehr laden ...\"}");


var $ea5be7d4c19828c9$exports = {};
$ea5be7d4c19828c9$exports = JSON.parse("{\"loading\":\"Φόρτωση...\",\"loadingMore\":\"Φόρτωση περισσότερων...\"}");


var $c5f68a610fa61271$exports = {};
$c5f68a610fa61271$exports = JSON.parse("{\"loading\":\"Loading…\",\"loadingMore\":\"Loading more…\"}");


var $559cd2cba2dd3d08$exports = {};
$559cd2cba2dd3d08$exports = JSON.parse("{\"loading\":\"Cargando…\",\"loadingMore\":\"Cargando más…\"}");


var $27924e5c3a870809$exports = {};
$27924e5c3a870809$exports = JSON.parse("{\"loading\":\"Laadimine...\",\"loadingMore\":\"Laadi rohkem...\"}");


var $0525a86bd2391b12$exports = {};
$0525a86bd2391b12$exports = JSON.parse("{\"loading\":\"Ladataan…\",\"loadingMore\":\"Ladataan lisää…\"}");


var $01a07641cecd76d3$exports = {};
$01a07641cecd76d3$exports = JSON.parse("{\"loading\":\"Chargement...\",\"loadingMore\":\"Chargement supplémentaire...\"}");


var $274e460a69ccfd02$exports = {};
$274e460a69ccfd02$exports = JSON.parse("{\"loading\":\"טוען...\",\"loadingMore\":\"טוען עוד...\"}");


var $c9b1de60e6bc1997$exports = {};
$c9b1de60e6bc1997$exports = JSON.parse("{\"loading\":\"Učitavam...\",\"loadingMore\":\"Učitavam još...\"}");


var $77105a0939477256$exports = {};
$77105a0939477256$exports = JSON.parse("{\"loading\":\"Betöltés folyamatban…\",\"loadingMore\":\"Továbbiak betöltése folyamatban…\"}");


var $6e575fe494b41776$exports = {};
$6e575fe494b41776$exports = JSON.parse("{\"loading\":\"Caricamento...\",\"loadingMore\":\"Caricamento altri...\"}");


var $aad0256b25e45af2$exports = {};
$aad0256b25e45af2$exports = JSON.parse("{\"loading\":\"読み込み中...\",\"loadingMore\":\"さらに読み込み中...\"}");


var $f65606357053eec3$exports = {};
$f65606357053eec3$exports = JSON.parse("{\"loading\":\"로드 중\",\"loadingMore\":\"추가 로드 중\"}");


var $a10614508fa31d09$exports = {};
$a10614508fa31d09$exports = JSON.parse("{\"loading\":\"Įkeliama...\",\"loadingMore\":\"Įkeliama daugiau...\"}");


var $808bedf55397eb6b$exports = {};
$808bedf55397eb6b$exports = JSON.parse("{\"loading\":\"Notiek ielāde...\",\"loadingMore\":\"Tiek ielādēts vēl...\"}");


var $990a2ae6fe06f5b1$exports = {};
$990a2ae6fe06f5b1$exports = JSON.parse("{\"loading\":\"Laster inn ...\",\"loadingMore\":\"Laster inn flere ...\"}");


var $49b0e47d31fcf021$exports = {};
$49b0e47d31fcf021$exports = JSON.parse("{\"loading\":\"Laden...\",\"loadingMore\":\"Meer laden...\"}");


var $2bfeea234fe27f40$exports = {};
$2bfeea234fe27f40$exports = JSON.parse("{\"loading\":\"Ładowanie...\",\"loadingMore\":\"Wczytywanie większej liczby...\"}");


var $74f5c6463b828a97$exports = {};
$74f5c6463b828a97$exports = JSON.parse("{\"loading\":\"Carregando...\",\"loadingMore\":\"Carregando mais...\"}");


var $4b4861bb06537fc4$exports = {};
$4b4861bb06537fc4$exports = JSON.parse("{\"loading\":\"A carregar...\",\"loadingMore\":\"A carregar mais...\"}");


var $d6e02ca9b21933f7$exports = {};
$d6e02ca9b21933f7$exports = JSON.parse("{\"loading\":\"Se încarcă...\",\"loadingMore\":\"Se încarcă mai multe...\"}");


var $7a06014b269909e1$exports = {};
$7a06014b269909e1$exports = JSON.parse("{\"loading\":\"Загрузка...\",\"loadingMore\":\"Дополнительная загрузка...\"}");


var $83c989bdec33b39e$exports = {};
$83c989bdec33b39e$exports = JSON.parse("{\"loading\":\"Načítava sa...\",\"loadingMore\":\"Načítava sa viac...\"}");


var $25dd79cec0950651$exports = {};
$25dd79cec0950651$exports = JSON.parse("{\"loading\":\"Nalaganje...\",\"loadingMore\":\"Nalaganje več vsebine...\"}");


var $fdd00e19b6ea86eb$exports = {};
$fdd00e19b6ea86eb$exports = JSON.parse("{\"loading\":\"Učitavam...\",\"loadingMore\":\"Učitavam još...\"}");


var $924109ca416a45df$exports = {};
$924109ca416a45df$exports = JSON.parse("{\"loading\":\"Läser in...\",\"loadingMore\":\"Läser in mer...\"}");


var $2bf08f7ed0478523$exports = {};
$2bf08f7ed0478523$exports = JSON.parse("{\"loading\":\"Yükleniyor...\",\"loadingMore\":\"Daha fazla yükleniyor...\"}");


var $f53b1e171999b255$exports = {};
$f53b1e171999b255$exports = JSON.parse("{\"loading\":\"Завантаження…\",\"loadingMore\":\"Завантаження інших об’єктів...\"}");


var $30d15af0057cf712$exports = {};
$30d15af0057cf712$exports = JSON.parse("{\"loading\":\"正在加载...\",\"loadingMore\":\"正在加载更多...\"}");


var $52543fad9ebcaf9d$exports = {};
$52543fad9ebcaf9d$exports = JSON.parse("{\"loading\":\"正在載入\",\"loadingMore\":\"正在載入更多…\"}");


$97a5c01ed556d8f5$exports = {
    "ar-AE": $c9c03464d18bae68$exports,
    "bg-BG": $57766d982c55bc35$exports,
    "cs-CZ": $a4963ce26b58d483$exports,
    "da-DK": $5b711ac7321b2584$exports,
    "de-DE": $aa07c3f6a1018646$exports,
    "el-GR": $ea5be7d4c19828c9$exports,
    "en-US": $c5f68a610fa61271$exports,
    "es-ES": $559cd2cba2dd3d08$exports,
    "et-EE": $27924e5c3a870809$exports,
    "fi-FI": $0525a86bd2391b12$exports,
    "fr-FR": $01a07641cecd76d3$exports,
    "he-IL": $274e460a69ccfd02$exports,
    "hr-HR": $c9b1de60e6bc1997$exports,
    "hu-HU": $77105a0939477256$exports,
    "it-IT": $6e575fe494b41776$exports,
    "ja-JP": $aad0256b25e45af2$exports,
    "ko-KR": $f65606357053eec3$exports,
    "lt-LT": $a10614508fa31d09$exports,
    "lv-LV": $808bedf55397eb6b$exports,
    "nb-NO": $990a2ae6fe06f5b1$exports,
    "nl-NL": $49b0e47d31fcf021$exports,
    "pl-PL": $2bfeea234fe27f40$exports,
    "pt-BR": $74f5c6463b828a97$exports,
    "pt-PT": $4b4861bb06537fc4$exports,
    "ro-RO": $d6e02ca9b21933f7$exports,
    "ru-RU": $7a06014b269909e1$exports,
    "sk-SK": $83c989bdec33b39e$exports,
    "sl-SI": $25dd79cec0950651$exports,
    "sr-SP": $fdd00e19b6ea86eb$exports,
    "sv-SE": $924109ca416a45df$exports,
    "tr-TR": $2bf08f7ed0478523$exports,
    "uk-UA": $f53b1e171999b255$exports,
    "zh-CN": $30d15af0057cf712$exports,
    "zh-TW": $52543fad9ebcaf9d$exports
};



const $621ef71b47012af1$export$7ff8f37d2d81a48d = ($parcel$interopDefault($l8Zb1$react)).createContext(null);










var $d53361c88b80133d$exports = {};

$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-popover", () => $d53361c88b80133d$export$d959f30786125b, (v) => $d53361c88b80133d$export$d959f30786125b = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu", () => $d53361c88b80133d$export$fae6af6decd50ac5, (v) => $d53361c88b80133d$export$fae6af6decd50ac5 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-sectionHeading", () => $d53361c88b80133d$export$c20ea4b35020fbdd, (v) => $d53361c88b80133d$export$c20ea4b35020fbdd = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-checkmark", () => $d53361c88b80133d$export$8e15a726e4f7e62d, (v) => $d53361c88b80133d$export$8e15a726e4f7e62d = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-item", () => $d53361c88b80133d$export$45faad328e6c532b, (v) => $d53361c88b80133d$export$45faad328e6c532b = v);
$parcel$export($d53361c88b80133d$exports, "is-selected", () => $d53361c88b80133d$export$1e0fb04f31d3c22a, (v) => $d53361c88b80133d$export$1e0fb04f31d3c22a = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Icon", () => $d53361c88b80133d$export$d374b04f30360026, (v) => $d53361c88b80133d$export$d374b04f30360026 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-itemLabel", () => $d53361c88b80133d$export$1aa6419c5c650567, (v) => $d53361c88b80133d$export$1aa6419c5c650567 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-itemIcon", () => $d53361c88b80133d$export$f431001e1bb54633, (v) => $d53361c88b80133d$export$f431001e1bb54633 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-itemLabel--wrapping", () => $d53361c88b80133d$export$465092164a8e1d98, (v) => $d53361c88b80133d$export$465092164a8e1d98 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-divider", () => $d53361c88b80133d$export$e892026f534efed8, (v) => $d53361c88b80133d$export$e892026f534efed8 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-itemGrid", () => $d53361c88b80133d$export$ee5f52784f57f9c8, (v) => $d53361c88b80133d$export$ee5f52784f57f9c8 = v);
$parcel$export($d53361c88b80133d$exports, "is-selectable", () => $d53361c88b80133d$export$fbd22ba224473193, (v) => $d53361c88b80133d$export$fbd22ba224473193 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-end", () => $d53361c88b80133d$export$926a1bb74ed65667, (v) => $d53361c88b80133d$export$926a1bb74ed65667 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-icon", () => $d53361c88b80133d$export$cbbae63ef9e44c7a, (v) => $d53361c88b80133d$export$cbbae63ef9e44c7a = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-description", () => $d53361c88b80133d$export$36f1275051e39ead, (v) => $d53361c88b80133d$export$36f1275051e39ead = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-keyboard", () => $d53361c88b80133d$export$4f9817a7dff6ba0f, (v) => $d53361c88b80133d$export$4f9817a7dff6ba0f = v);
$parcel$export($d53361c88b80133d$exports, "focus-ring", () => $d53361c88b80133d$export$f39a09f249340e2a, (v) => $d53361c88b80133d$export$f39a09f249340e2a = v);
$parcel$export($d53361c88b80133d$exports, "is-focused", () => $d53361c88b80133d$export$e7dc768d35940237, (v) => $d53361c88b80133d$export$e7dc768d35940237 = v);
$parcel$export($d53361c88b80133d$exports, "is-hovered", () => $d53361c88b80133d$export$b8813cd5d7824ce7, (v) => $d53361c88b80133d$export$b8813cd5d7824ce7 = v);
$parcel$export($d53361c88b80133d$exports, "is-highlighted", () => $d53361c88b80133d$export$451c613d0772eb99, (v) => $d53361c88b80133d$export$451c613d0772eb99 = v);
$parcel$export($d53361c88b80133d$exports, "is-open", () => $d53361c88b80133d$export$a9781837241c946d, (v) => $d53361c88b80133d$export$a9781837241c946d = v);
$parcel$export($d53361c88b80133d$exports, "is-active", () => $d53361c88b80133d$export$20fd0f7cd4e6112f, (v) => $d53361c88b80133d$export$20fd0f7cd4e6112f = v);
$parcel$export($d53361c88b80133d$exports, "is-disabled", () => $d53361c88b80133d$export$d35bc1e505d1ebbf, (v) => $d53361c88b80133d$export$d35bc1e505d1ebbf = v);
var $d53361c88b80133d$export$d959f30786125b;
var $d53361c88b80133d$export$fae6af6decd50ac5;
var $d53361c88b80133d$export$c20ea4b35020fbdd;
var $d53361c88b80133d$export$8e15a726e4f7e62d;
var $d53361c88b80133d$export$45faad328e6c532b;
var $d53361c88b80133d$export$1e0fb04f31d3c22a;
var $d53361c88b80133d$export$d374b04f30360026;
var $d53361c88b80133d$export$1aa6419c5c650567;
var $d53361c88b80133d$export$f431001e1bb54633;
var $d53361c88b80133d$export$465092164a8e1d98;
var $d53361c88b80133d$export$e892026f534efed8;
var $d53361c88b80133d$export$ee5f52784f57f9c8;
var $d53361c88b80133d$export$fbd22ba224473193;
var $d53361c88b80133d$export$926a1bb74ed65667;
var $d53361c88b80133d$export$cbbae63ef9e44c7a;
var $d53361c88b80133d$export$36f1275051e39ead;
var $d53361c88b80133d$export$4f9817a7dff6ba0f;
var $d53361c88b80133d$export$f39a09f249340e2a;
var $d53361c88b80133d$export$e7dc768d35940237;
var $d53361c88b80133d$export$b8813cd5d7824ce7;
var $d53361c88b80133d$export$451c613d0772eb99;
var $d53361c88b80133d$export$a9781837241c946d;
var $d53361c88b80133d$export$20fd0f7cd4e6112f;
var $d53361c88b80133d$export$d35bc1e505d1ebbf;
$d53361c88b80133d$export$d959f30786125b = "spectrum-Menu-popover_7ea47c";
$d53361c88b80133d$export$fae6af6decd50ac5 = "spectrum-Menu_7ea47c";
$d53361c88b80133d$export$c20ea4b35020fbdd = "spectrum-Menu-sectionHeading_7ea47c";
$d53361c88b80133d$export$8e15a726e4f7e62d = "spectrum-Menu-checkmark_7ea47c";
$d53361c88b80133d$export$45faad328e6c532b = "spectrum-Menu-item_7ea47c";
$d53361c88b80133d$export$1e0fb04f31d3c22a = "is-selected_7ea47c";
$d53361c88b80133d$export$d374b04f30360026 = "spectrum-Icon_7ea47c";
$d53361c88b80133d$export$1aa6419c5c650567 = "spectrum-Menu-itemLabel_7ea47c";
$d53361c88b80133d$export$f431001e1bb54633 = "spectrum-Menu-itemIcon_7ea47c";
$d53361c88b80133d$export$465092164a8e1d98 = "spectrum-Menu-itemLabel--wrapping_7ea47c";
$d53361c88b80133d$export$e892026f534efed8 = "spectrum-Menu-divider_7ea47c";
$d53361c88b80133d$export$ee5f52784f57f9c8 = "spectrum-Menu-itemGrid_7ea47c";
$d53361c88b80133d$export$fbd22ba224473193 = "is-selectable_7ea47c";
$d53361c88b80133d$export$926a1bb74ed65667 = "spectrum-Menu-end_7ea47c";
$d53361c88b80133d$export$cbbae63ef9e44c7a = "spectrum-Menu-icon_7ea47c";
$d53361c88b80133d$export$36f1275051e39ead = "spectrum-Menu-description_7ea47c";
$d53361c88b80133d$export$4f9817a7dff6ba0f = "spectrum-Menu-keyboard_7ea47c";
$d53361c88b80133d$export$f39a09f249340e2a = "focus-ring_7ea47c";
$d53361c88b80133d$export$e7dc768d35940237 = "is-focused_7ea47c";
$d53361c88b80133d$export$b8813cd5d7824ce7 = "is-hovered_7ea47c";
$d53361c88b80133d$export$451c613d0772eb99 = "is-highlighted_7ea47c";
$d53361c88b80133d$export$a9781837241c946d = "is-open_7ea47c";
$d53361c88b80133d$export$20fd0f7cd4e6112f = "is-active_7ea47c";
$d53361c88b80133d$export$d35bc1e505d1ebbf = "is-disabled_7ea47c";





function $c9c61f2441136cf6$export$feb3b6b552c14a12(props) {
    let { item: item , shouldSelectOnPressUp: shouldSelectOnPressUp , shouldFocusOnHover: shouldFocusOnHover , shouldUseVirtualFocus: shouldUseVirtualFocus  } = props;
    let { rendered: rendered , key: key  } = item;
    let state = $l8Zb1$react.useContext($621ef71b47012af1$export$7ff8f37d2d81a48d);
    let ref = $l8Zb1$react.useRef();
    let { optionProps: optionProps , labelProps: labelProps , descriptionProps: descriptionProps , isSelected: isSelected , isDisabled: isDisabled , isFocused: isFocused  } = $l8Zb1$reactarialistbox.useOption({
        'aria-label': item['aria-label'],
        key: key,
        shouldSelectOnPressUp: shouldSelectOnPressUp,
        shouldFocusOnHover: shouldFocusOnHover,
        isVirtualized: true,
        shouldUseVirtualFocus: shouldUseVirtualFocus
    }, state, ref);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $l8Zb1$reactariainteractions.useHover({
        ...props,
        isDisabled: isDisabled
    });
    let contents = typeof rendered === 'string' ? /*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement($l8Zb1$reactspectrumtext.Text, null, rendered) : rendered;
    let isKeyboardModality = $l8Zb1$reactariainteractions.isFocusVisible();
    return(/*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement($l8Zb1$reactariafocus.FocusRing, {
        focusRingClass: $l8Zb1$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement("div", {
        ...$l8Zb1$reactariautils.mergeProps(optionProps, shouldFocusOnHover ? {
        } : hoverProps),
        ref: ref,
        className: $l8Zb1$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'spectrum-Menu-item', {
            // If using virtual focus, apply focused styles to the item when the user is interacting with keyboard modality
            'is-focused': shouldUseVirtualFocus && isFocused && isKeyboardModality,
            'is-disabled': isDisabled,
            'is-selected': isSelected,
            'is-selectable': state.selectionManager.selectionMode !== 'none',
            // When shouldFocusOnHover is false, apply hover styles both when hovered with the mouse.
            // Otherwise, apply hover styles when focused using non-keyboard modality.
            'is-hovered': isHovered && !shouldFocusOnHover || isFocused && !isKeyboardModality
        })
    }, /*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement($l8Zb1$reactspectrumlayout.Grid, {
        UNSAFE_className: $l8Zb1$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'spectrum-Menu-itemGrid')
    }, /*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement($l8Zb1$reactspectrumutils.ClearSlots, null, /*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement($l8Zb1$reactspectrumutils.SlotProvider, {
        slots: {
            text: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports))['spectrum-Menu-itemLabel'],
                ...labelProps
            },
            icon: {
                size: 'S',
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports))['spectrum-Menu-icon']
            },
            description: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports))['spectrum-Menu-description'],
                ...descriptionProps
            }
        }
    }, contents, isSelected && /*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement(($parcel$interopDefault($l8Zb1$spectrumiconsuiCheckmarkMedium)), {
        slot: "checkmark",
        UNSAFE_className: $l8Zb1$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'spectrum-Menu-checkmark')
    })))))));
}










function $40f45351b5409a2d$export$dca12b0bb56e4fc(props) {
    let { children: children , reusableView: reusableView , header: header  } = props;
    let item = reusableView.content;
    let { headingProps: headingProps , groupProps: groupProps  } = $l8Zb1$reactarialistbox.useListBoxSection({
        heading: item.rendered,
        'aria-label': item['aria-label']
    });
    let { separatorProps: separatorProps  } = $l8Zb1$reactariaseparator.useSeparator({
        elementType: 'li'
    });
    let headerRef = $l8Zb1$react.useRef();
    $l8Zb1$reactariavirtualizer.useVirtualizerItem({
        reusableView: header,
        ref: headerRef
    });
    let { direction: direction  } = $l8Zb1$reactariai18n.useLocale();
    let state = $l8Zb1$react.useContext($621ef71b47012af1$export$7ff8f37d2d81a48d);
    return(/*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement($l8Zb1$react.Fragment, null, /*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement("div", {
        role: "presentation",
        ref: headerRef,
        style: $l8Zb1$reactariavirtualizer.layoutInfoToStyle(header.layoutInfo, direction)
    }, item.key !== state.collection.getFirstKey() && /*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement("div", {
        ...separatorProps,
        className: $l8Zb1$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'spectrum-Menu-divider')
    }), item.rendered && /*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement("div", {
        ...headingProps,
        className: $l8Zb1$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'spectrum-Menu-sectionHeading')
    }, item.rendered)), /*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement("div", {
        ...groupProps,
        style: $l8Zb1$reactariavirtualizer.layoutInfoToStyle(reusableView.layoutInfo, direction),
        className: $l8Zb1$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'spectrum-Menu')
    }, children)));
}










function $24b64fc51b3c824b$export$25768ea656ae32a7(state) {
    let { scale: scale  } = $l8Zb1$reactspectrumprovider.useProvider();
    let collator = $l8Zb1$reactariai18n.useCollator({
        usage: 'search',
        sensitivity: 'base'
    });
    let layout = $l8Zb1$react.useMemo(()=>new $l8Zb1$reactstatelylayout.ListLayout({
            estimatedRowHeight: scale === 'large' ? 48 : 32,
            estimatedHeadingHeight: scale === 'large' ? 33 : 26,
            padding: scale === 'large' ? 5 : 4,
            loaderHeight: 40,
            placeholderHeight: scale === 'large' ? 48 : 32,
            collator: collator
        })
    , [
        collator,
        scale
    ]);
    layout.collection = state.collection;
    layout.disabledKeys = state.disabledKeys;
    return layout;
}
/** @private */ function $24b64fc51b3c824b$var$ListBoxBase(props, ref) {
    let { layout: layout , state: state , shouldSelectOnPressUp: shouldSelectOnPressUp , focusOnPointerEnter: focusOnPointerEnter , shouldUseVirtualFocus: shouldUseVirtualFocus , domProps: domProps = {
    } , transitionDuration: transitionDuration = 0 , onScroll: onScroll  } = props;
    let { listBoxProps: listBoxProps  } = $l8Zb1$reactarialistbox.useListBox({
        ...props,
        keyboardDelegate: layout,
        isVirtualized: true
    }, state, ref);
    let { styleProps: styleProps  } = $l8Zb1$reactspectrumutils.useStyleProps(props);
    let formatMessage = $l8Zb1$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($97a5c01ed556d8f5$exports)));
    // Sync loading state into the layout.
    layout.isLoading = props.isLoading;
    let renderWrapper = (parent, reusableView, children, renderChildren)=>{
        if (reusableView.viewType === 'section') return(/*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement($40f45351b5409a2d$export$dca12b0bb56e4fc, {
            key: reusableView.key,
            reusableView: reusableView,
            header: children.find((c)=>c.viewType === 'header'
            )
        }, renderChildren(children.filter((c)=>c.viewType === 'item'
        ))));
        return(/*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement($l8Zb1$reactariavirtualizer.VirtualizerItem, {
            key: reusableView.key,
            reusableView: reusableView,
            parent: parent
        }));
    };
    return(/*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement($621ef71b47012af1$export$7ff8f37d2d81a48d.Provider, {
        value: state
    }, /*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement($l8Zb1$reactariavirtualizer.Virtualizer, {
        ...styleProps,
        ...$l8Zb1$reactariautils.mergeProps(listBoxProps, domProps),
        ref: ref,
        focusedKey: state.selectionManager.focusedKey,
        sizeToFit: "height",
        scrollDirection: "vertical",
        className: $l8Zb1$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'spectrum-Menu', styleProps.className),
        layout: layout,
        collection: state.collection,
        renderWrapper: renderWrapper,
        transitionDuration: transitionDuration,
        isLoading: props.isLoading,
        onLoadMore: props.onLoadMore,
        shouldUseVirtualFocus: shouldUseVirtualFocus,
        onScroll: onScroll
    }, (type, item)=>{
        if (type === 'item') return(/*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement($c9c61f2441136cf6$export$feb3b6b552c14a12, {
            item: item,
            shouldSelectOnPressUp: shouldSelectOnPressUp,
            shouldFocusOnHover: focusOnPointerEnter,
            shouldUseVirtualFocus: shouldUseVirtualFocus
        }));
        else if (type === 'loader') return(// aria-selected isn't needed here since this option is not selectable.
        // eslint-disable-next-line jsx-a11y/role-has-required-aria-props
        /*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement("div", {
            role: "option",
            style: {
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center',
                height: '100%'
            }
        }, /*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement($l8Zb1$reactspectrumprogress.ProgressCircle, {
            isIndeterminate: true,
            size: "S",
            "aria-label": state.collection.size > 0 ? formatMessage('loadingMore') : formatMessage('loading'),
            UNSAFE_className: $l8Zb1$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'spectrum-Dropdown-progressCircle')
        })));
        else if (type === 'placeholder') {
            let emptyState = props.renderEmptyState ? props.renderEmptyState() : null;
            if (emptyState == null) return null;
            return(/*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement("div", {
                // aria-selected isn't needed here since this option is not selectable.
                // eslint-disable-next-line jsx-a11y/role-has-required-aria-props
                role: "option"
            }, emptyState));
        }
    })));
}
// forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
const $24b64fc51b3c824b$export$1afdcf349979fb7e = /*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).forwardRef($24b64fc51b3c824b$var$ListBoxBase);





function $5cde28c545cc4211$var$ListBox(props, ref) {
    let state = $l8Zb1$reactstatelylist.useListState(props);
    let layout = $24b64fc51b3c824b$export$25768ea656ae32a7(state);
    let domRef = $l8Zb1$reactspectrumutils.useDOMRef(ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).createElement($24b64fc51b3c824b$export$1afdcf349979fb7e, {
        ...props,
        ref: domRef,
        state: state,
        layout: layout
    }));
}
// forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
/**
 * A list of options that can allow selection of one or more.
 */ const $5cde28c545cc4211$export$41f133550aa26f48 = /*#__PURE__*/ ($parcel$interopDefault($l8Zb1$react)).forwardRef($5cde28c545cc4211$var$ListBox);




$parcel$exportWildcard(module.exports, $5cde28c545cc4211$exports);
$parcel$exportWildcard(module.exports, $24b64fc51b3c824b$exports);
