// @react-spectrum/listbox/dist/module.js
import "./main.css";
import {Item as $09647559903a45ec$re_export$Item, Section as $09647559903a45ec$re_export$Section} from "@react-stately/collections";
import $dbUYi$react, {useMemo as $dbUYi$useMemo, useContext as $dbUYi$useContext, useRef as $dbUYi$useRef, Fragment as $dbUYi$Fragment} from "react";
import {useDOMRef as $dbUYi$useDOMRef, useStyleProps as $dbUYi$useStyleProps, classNames as $dbUYi$classNames, ClearSlots as $dbUYi$ClearSlots, SlotProvider as $dbUYi$SlotProvider} from "@react-spectrum/utils";
import {useListState as $dbUYi$useListState} from "@react-stately/list";
import {useListBox as $dbUYi$useListBox, useOption as $dbUYi$useOption, useListBoxSection as $dbUYi$useListBoxSection} from "@react-aria/listbox";
import {ListLayout as $dbUYi$ListLayout} from "@react-stately/layout";
import {mergeProps as $dbUYi$mergeProps} from "@react-aria/utils";
import {ProgressCircle as $dbUYi$ProgressCircle} from "@react-spectrum/progress";
import {useCollator as $dbUYi$useCollator, useMessageFormatter as $dbUYi$useMessageFormatter, useLocale as $dbUYi$useLocale} from "@react-aria/i18n";
import {useProvider as $dbUYi$useProvider} from "@react-spectrum/provider";
import {VirtualizerItem as $dbUYi$VirtualizerItem, Virtualizer as $dbUYi$Virtualizer, useVirtualizerItem as $dbUYi$useVirtualizerItem, layoutInfoToStyle as $dbUYi$layoutInfoToStyle} from "@react-aria/virtualizer";
import $dbUYi$spectrumiconsuiCheckmarkMedium from "@spectrum-icons/ui/CheckmarkMedium";
import {FocusRing as $dbUYi$FocusRing} from "@react-aria/focus";
import {Grid as $dbUYi$Grid} from "@react-spectrum/layout";
import {useHover as $dbUYi$useHover, isFocusVisible as $dbUYi$isFocusVisible} from "@react-aria/interactions";
import {Text as $dbUYi$Text} from "@react-spectrum/text";
import {useSeparator as $dbUYi$useSeparator} from "@react-aria/separator";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
var $c6e964ce023e5a0b$exports = {};

$parcel$export($c6e964ce023e5a0b$exports, "ListBox", () => $c6e964ce023e5a0b$export$41f133550aa26f48);
var $0609d2e227383a47$exports = {};

$parcel$export($0609d2e227383a47$exports, "useListBoxLayout", () => $0609d2e227383a47$export$25768ea656ae32a7);
$parcel$export($0609d2e227383a47$exports, "ListBoxBase", () => $0609d2e227383a47$export$1afdcf349979fb7e);


var $584fb08d5758edb3$exports = {};
var $96119f826647432b$exports = {};
$96119f826647432b$exports = JSON.parse("{\"loading\":\"جارٍ التحميل...\",\"loadingMore\":\"جارٍ تحميل المزيد...\"}");


var $2262f2ffa94e5584$exports = {};
$2262f2ffa94e5584$exports = JSON.parse("{\"loading\":\"Зареждане...\",\"loadingMore\":\"Зареждане на още...\"}");


var $e4deb0d0bd5ed450$exports = {};
$e4deb0d0bd5ed450$exports = JSON.parse("{\"loading\":\"Načítání...\",\"loadingMore\":\"Načítání dalších...\"}");


var $05d575aeef470e8c$exports = {};
$05d575aeef470e8c$exports = JSON.parse("{\"loading\":\"Indlæser ...\",\"loadingMore\":\"Indlæser flere ...\"}");


var $e5b34536ce019e77$exports = {};
$e5b34536ce019e77$exports = JSON.parse("{\"loading\":\"Laden...\",\"loadingMore\":\"Mehr laden ...\"}");


var $4e8209a0cd219005$exports = {};
$4e8209a0cd219005$exports = JSON.parse("{\"loading\":\"Φόρτωση...\",\"loadingMore\":\"Φόρτωση περισσότερων...\"}");


var $65df8759c2714b6d$exports = {};
$65df8759c2714b6d$exports = JSON.parse("{\"loading\":\"Loading…\",\"loadingMore\":\"Loading more…\"}");


var $c7d47c4f53f06176$exports = {};
$c7d47c4f53f06176$exports = JSON.parse("{\"loading\":\"Cargando…\",\"loadingMore\":\"Cargando más…\"}");


var $23b285a71c3ed5e9$exports = {};
$23b285a71c3ed5e9$exports = JSON.parse("{\"loading\":\"Laadimine...\",\"loadingMore\":\"Laadi rohkem...\"}");


var $a25b843ae983ff7c$exports = {};
$a25b843ae983ff7c$exports = JSON.parse("{\"loading\":\"Ladataan…\",\"loadingMore\":\"Ladataan lisää…\"}");


var $848d0b0e6a703660$exports = {};
$848d0b0e6a703660$exports = JSON.parse("{\"loading\":\"Chargement...\",\"loadingMore\":\"Chargement supplémentaire...\"}");


var $71ce00c5fee7476a$exports = {};
$71ce00c5fee7476a$exports = JSON.parse("{\"loading\":\"טוען...\",\"loadingMore\":\"טוען עוד...\"}");


var $afb5e89fc08d75b3$exports = {};
$afb5e89fc08d75b3$exports = JSON.parse("{\"loading\":\"Učitavam...\",\"loadingMore\":\"Učitavam još...\"}");


var $1cf96846939a2ff1$exports = {};
$1cf96846939a2ff1$exports = JSON.parse("{\"loading\":\"Betöltés folyamatban…\",\"loadingMore\":\"Továbbiak betöltése folyamatban…\"}");


var $59ce0c68047b8664$exports = {};
$59ce0c68047b8664$exports = JSON.parse("{\"loading\":\"Caricamento...\",\"loadingMore\":\"Caricamento altri...\"}");


var $626626492ce4d5d8$exports = {};
$626626492ce4d5d8$exports = JSON.parse("{\"loading\":\"読み込み中...\",\"loadingMore\":\"さらに読み込み中...\"}");


var $e348c2ea9bba378d$exports = {};
$e348c2ea9bba378d$exports = JSON.parse("{\"loading\":\"로드 중\",\"loadingMore\":\"추가 로드 중\"}");


var $74ebd9514976ff6d$exports = {};
$74ebd9514976ff6d$exports = JSON.parse("{\"loading\":\"Įkeliama...\",\"loadingMore\":\"Įkeliama daugiau...\"}");


var $ad11160c2fa3e897$exports = {};
$ad11160c2fa3e897$exports = JSON.parse("{\"loading\":\"Notiek ielāde...\",\"loadingMore\":\"Tiek ielādēts vēl...\"}");


var $934487f594e1c5f5$exports = {};
$934487f594e1c5f5$exports = JSON.parse("{\"loading\":\"Laster inn ...\",\"loadingMore\":\"Laster inn flere ...\"}");


var $1a7e85668de4ca78$exports = {};
$1a7e85668de4ca78$exports = JSON.parse("{\"loading\":\"Laden...\",\"loadingMore\":\"Meer laden...\"}");


var $97a0e3c6aec96a3e$exports = {};
$97a0e3c6aec96a3e$exports = JSON.parse("{\"loading\":\"Ładowanie...\",\"loadingMore\":\"Wczytywanie większej liczby...\"}");


var $f556d053a9601d5d$exports = {};
$f556d053a9601d5d$exports = JSON.parse("{\"loading\":\"Carregando...\",\"loadingMore\":\"Carregando mais...\"}");


var $218091483888e39b$exports = {};
$218091483888e39b$exports = JSON.parse("{\"loading\":\"A carregar...\",\"loadingMore\":\"A carregar mais...\"}");


var $7b4552987c50b1b9$exports = {};
$7b4552987c50b1b9$exports = JSON.parse("{\"loading\":\"Se încarcă...\",\"loadingMore\":\"Se încarcă mai multe...\"}");


var $ce2be94d18fc717d$exports = {};
$ce2be94d18fc717d$exports = JSON.parse("{\"loading\":\"Загрузка...\",\"loadingMore\":\"Дополнительная загрузка...\"}");


var $0decb5d11fda55e5$exports = {};
$0decb5d11fda55e5$exports = JSON.parse("{\"loading\":\"Načítava sa...\",\"loadingMore\":\"Načítava sa viac...\"}");


var $122d69cf4848e89c$exports = {};
$122d69cf4848e89c$exports = JSON.parse("{\"loading\":\"Nalaganje...\",\"loadingMore\":\"Nalaganje več vsebine...\"}");


var $ade91f75c645575b$exports = {};
$ade91f75c645575b$exports = JSON.parse("{\"loading\":\"Učitavam...\",\"loadingMore\":\"Učitavam još...\"}");


var $6ad213cb0b4801f8$exports = {};
$6ad213cb0b4801f8$exports = JSON.parse("{\"loading\":\"Läser in...\",\"loadingMore\":\"Läser in mer...\"}");


var $e84c6cc2712f41ec$exports = {};
$e84c6cc2712f41ec$exports = JSON.parse("{\"loading\":\"Yükleniyor...\",\"loadingMore\":\"Daha fazla yükleniyor...\"}");


var $e959ef735ef82e3a$exports = {};
$e959ef735ef82e3a$exports = JSON.parse("{\"loading\":\"Завантаження…\",\"loadingMore\":\"Завантаження інших об’єктів...\"}");


var $27448465f6e7c61d$exports = {};
$27448465f6e7c61d$exports = JSON.parse("{\"loading\":\"正在加载...\",\"loadingMore\":\"正在加载更多...\"}");


var $32e663b06708e9ed$exports = {};
$32e663b06708e9ed$exports = JSON.parse("{\"loading\":\"正在載入\",\"loadingMore\":\"正在載入更多…\"}");


$584fb08d5758edb3$exports = {
    "ar-AE": $96119f826647432b$exports,
    "bg-BG": $2262f2ffa94e5584$exports,
    "cs-CZ": $e4deb0d0bd5ed450$exports,
    "da-DK": $05d575aeef470e8c$exports,
    "de-DE": $e5b34536ce019e77$exports,
    "el-GR": $4e8209a0cd219005$exports,
    "en-US": $65df8759c2714b6d$exports,
    "es-ES": $c7d47c4f53f06176$exports,
    "et-EE": $23b285a71c3ed5e9$exports,
    "fi-FI": $a25b843ae983ff7c$exports,
    "fr-FR": $848d0b0e6a703660$exports,
    "he-IL": $71ce00c5fee7476a$exports,
    "hr-HR": $afb5e89fc08d75b3$exports,
    "hu-HU": $1cf96846939a2ff1$exports,
    "it-IT": $59ce0c68047b8664$exports,
    "ja-JP": $626626492ce4d5d8$exports,
    "ko-KR": $e348c2ea9bba378d$exports,
    "lt-LT": $74ebd9514976ff6d$exports,
    "lv-LV": $ad11160c2fa3e897$exports,
    "nb-NO": $934487f594e1c5f5$exports,
    "nl-NL": $1a7e85668de4ca78$exports,
    "pl-PL": $97a0e3c6aec96a3e$exports,
    "pt-BR": $f556d053a9601d5d$exports,
    "pt-PT": $218091483888e39b$exports,
    "ro-RO": $7b4552987c50b1b9$exports,
    "ru-RU": $ce2be94d18fc717d$exports,
    "sk-SK": $0decb5d11fda55e5$exports,
    "sl-SI": $122d69cf4848e89c$exports,
    "sr-SP": $ade91f75c645575b$exports,
    "sv-SE": $6ad213cb0b4801f8$exports,
    "tr-TR": $e84c6cc2712f41ec$exports,
    "uk-UA": $e959ef735ef82e3a$exports,
    "zh-CN": $27448465f6e7c61d$exports,
    "zh-TW": $32e663b06708e9ed$exports
};



const $ca44d58a4bb9be3f$export$7ff8f37d2d81a48d = $dbUYi$react.createContext(null);










var $2a670e3d20f05637$exports = {};

$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-popover", () => $2a670e3d20f05637$export$d959f30786125b, (v) => $2a670e3d20f05637$export$d959f30786125b = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu", () => $2a670e3d20f05637$export$fae6af6decd50ac5, (v) => $2a670e3d20f05637$export$fae6af6decd50ac5 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-sectionHeading", () => $2a670e3d20f05637$export$c20ea4b35020fbdd, (v) => $2a670e3d20f05637$export$c20ea4b35020fbdd = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-checkmark", () => $2a670e3d20f05637$export$8e15a726e4f7e62d, (v) => $2a670e3d20f05637$export$8e15a726e4f7e62d = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-item", () => $2a670e3d20f05637$export$45faad328e6c532b, (v) => $2a670e3d20f05637$export$45faad328e6c532b = v);
$parcel$export($2a670e3d20f05637$exports, "is-selected", () => $2a670e3d20f05637$export$1e0fb04f31d3c22a, (v) => $2a670e3d20f05637$export$1e0fb04f31d3c22a = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Icon", () => $2a670e3d20f05637$export$d374b04f30360026, (v) => $2a670e3d20f05637$export$d374b04f30360026 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-itemLabel", () => $2a670e3d20f05637$export$1aa6419c5c650567, (v) => $2a670e3d20f05637$export$1aa6419c5c650567 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-itemIcon", () => $2a670e3d20f05637$export$f431001e1bb54633, (v) => $2a670e3d20f05637$export$f431001e1bb54633 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-itemLabel--wrapping", () => $2a670e3d20f05637$export$465092164a8e1d98, (v) => $2a670e3d20f05637$export$465092164a8e1d98 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-divider", () => $2a670e3d20f05637$export$e892026f534efed8, (v) => $2a670e3d20f05637$export$e892026f534efed8 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-itemGrid", () => $2a670e3d20f05637$export$ee5f52784f57f9c8, (v) => $2a670e3d20f05637$export$ee5f52784f57f9c8 = v);
$parcel$export($2a670e3d20f05637$exports, "is-selectable", () => $2a670e3d20f05637$export$fbd22ba224473193, (v) => $2a670e3d20f05637$export$fbd22ba224473193 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-end", () => $2a670e3d20f05637$export$926a1bb74ed65667, (v) => $2a670e3d20f05637$export$926a1bb74ed65667 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-icon", () => $2a670e3d20f05637$export$cbbae63ef9e44c7a, (v) => $2a670e3d20f05637$export$cbbae63ef9e44c7a = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-description", () => $2a670e3d20f05637$export$36f1275051e39ead, (v) => $2a670e3d20f05637$export$36f1275051e39ead = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-keyboard", () => $2a670e3d20f05637$export$4f9817a7dff6ba0f, (v) => $2a670e3d20f05637$export$4f9817a7dff6ba0f = v);
$parcel$export($2a670e3d20f05637$exports, "focus-ring", () => $2a670e3d20f05637$export$f39a09f249340e2a, (v) => $2a670e3d20f05637$export$f39a09f249340e2a = v);
$parcel$export($2a670e3d20f05637$exports, "is-focused", () => $2a670e3d20f05637$export$e7dc768d35940237, (v) => $2a670e3d20f05637$export$e7dc768d35940237 = v);
$parcel$export($2a670e3d20f05637$exports, "is-hovered", () => $2a670e3d20f05637$export$b8813cd5d7824ce7, (v) => $2a670e3d20f05637$export$b8813cd5d7824ce7 = v);
$parcel$export($2a670e3d20f05637$exports, "is-highlighted", () => $2a670e3d20f05637$export$451c613d0772eb99, (v) => $2a670e3d20f05637$export$451c613d0772eb99 = v);
$parcel$export($2a670e3d20f05637$exports, "is-open", () => $2a670e3d20f05637$export$a9781837241c946d, (v) => $2a670e3d20f05637$export$a9781837241c946d = v);
$parcel$export($2a670e3d20f05637$exports, "is-active", () => $2a670e3d20f05637$export$20fd0f7cd4e6112f, (v) => $2a670e3d20f05637$export$20fd0f7cd4e6112f = v);
$parcel$export($2a670e3d20f05637$exports, "is-disabled", () => $2a670e3d20f05637$export$d35bc1e505d1ebbf, (v) => $2a670e3d20f05637$export$d35bc1e505d1ebbf = v);
var $2a670e3d20f05637$export$d959f30786125b;
var $2a670e3d20f05637$export$fae6af6decd50ac5;
var $2a670e3d20f05637$export$c20ea4b35020fbdd;
var $2a670e3d20f05637$export$8e15a726e4f7e62d;
var $2a670e3d20f05637$export$45faad328e6c532b;
var $2a670e3d20f05637$export$1e0fb04f31d3c22a;
var $2a670e3d20f05637$export$d374b04f30360026;
var $2a670e3d20f05637$export$1aa6419c5c650567;
var $2a670e3d20f05637$export$f431001e1bb54633;
var $2a670e3d20f05637$export$465092164a8e1d98;
var $2a670e3d20f05637$export$e892026f534efed8;
var $2a670e3d20f05637$export$ee5f52784f57f9c8;
var $2a670e3d20f05637$export$fbd22ba224473193;
var $2a670e3d20f05637$export$926a1bb74ed65667;
var $2a670e3d20f05637$export$cbbae63ef9e44c7a;
var $2a670e3d20f05637$export$36f1275051e39ead;
var $2a670e3d20f05637$export$4f9817a7dff6ba0f;
var $2a670e3d20f05637$export$f39a09f249340e2a;
var $2a670e3d20f05637$export$e7dc768d35940237;
var $2a670e3d20f05637$export$b8813cd5d7824ce7;
var $2a670e3d20f05637$export$451c613d0772eb99;
var $2a670e3d20f05637$export$a9781837241c946d;
var $2a670e3d20f05637$export$20fd0f7cd4e6112f;
var $2a670e3d20f05637$export$d35bc1e505d1ebbf;
$2a670e3d20f05637$export$d959f30786125b = "spectrum-Menu-popover_7ea47c";
$2a670e3d20f05637$export$fae6af6decd50ac5 = "spectrum-Menu_7ea47c";
$2a670e3d20f05637$export$c20ea4b35020fbdd = "spectrum-Menu-sectionHeading_7ea47c";
$2a670e3d20f05637$export$8e15a726e4f7e62d = "spectrum-Menu-checkmark_7ea47c";
$2a670e3d20f05637$export$45faad328e6c532b = "spectrum-Menu-item_7ea47c";
$2a670e3d20f05637$export$1e0fb04f31d3c22a = "is-selected_7ea47c";
$2a670e3d20f05637$export$d374b04f30360026 = "spectrum-Icon_7ea47c";
$2a670e3d20f05637$export$1aa6419c5c650567 = "spectrum-Menu-itemLabel_7ea47c";
$2a670e3d20f05637$export$f431001e1bb54633 = "spectrum-Menu-itemIcon_7ea47c";
$2a670e3d20f05637$export$465092164a8e1d98 = "spectrum-Menu-itemLabel--wrapping_7ea47c";
$2a670e3d20f05637$export$e892026f534efed8 = "spectrum-Menu-divider_7ea47c";
$2a670e3d20f05637$export$ee5f52784f57f9c8 = "spectrum-Menu-itemGrid_7ea47c";
$2a670e3d20f05637$export$fbd22ba224473193 = "is-selectable_7ea47c";
$2a670e3d20f05637$export$926a1bb74ed65667 = "spectrum-Menu-end_7ea47c";
$2a670e3d20f05637$export$cbbae63ef9e44c7a = "spectrum-Menu-icon_7ea47c";
$2a670e3d20f05637$export$36f1275051e39ead = "spectrum-Menu-description_7ea47c";
$2a670e3d20f05637$export$4f9817a7dff6ba0f = "spectrum-Menu-keyboard_7ea47c";
$2a670e3d20f05637$export$f39a09f249340e2a = "focus-ring_7ea47c";
$2a670e3d20f05637$export$e7dc768d35940237 = "is-focused_7ea47c";
$2a670e3d20f05637$export$b8813cd5d7824ce7 = "is-hovered_7ea47c";
$2a670e3d20f05637$export$451c613d0772eb99 = "is-highlighted_7ea47c";
$2a670e3d20f05637$export$a9781837241c946d = "is-open_7ea47c";
$2a670e3d20f05637$export$20fd0f7cd4e6112f = "is-active_7ea47c";
$2a670e3d20f05637$export$d35bc1e505d1ebbf = "is-disabled_7ea47c";





function $cab3e898fb78c326$export$feb3b6b552c14a12(props) {
    let { item: item , shouldSelectOnPressUp: shouldSelectOnPressUp , shouldFocusOnHover: shouldFocusOnHover , shouldUseVirtualFocus: shouldUseVirtualFocus  } = props;
    let { rendered: rendered , key: key  } = item;
    let state = $dbUYi$useContext($ca44d58a4bb9be3f$export$7ff8f37d2d81a48d);
    let ref = $dbUYi$useRef();
    let { optionProps: optionProps , labelProps: labelProps , descriptionProps: descriptionProps , isSelected: isSelected , isDisabled: isDisabled , isFocused: isFocused  } = $dbUYi$useOption({
        'aria-label': item['aria-label'],
        key: key,
        shouldSelectOnPressUp: shouldSelectOnPressUp,
        shouldFocusOnHover: shouldFocusOnHover,
        isVirtualized: true,
        shouldUseVirtualFocus: shouldUseVirtualFocus
    }, state, ref);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $dbUYi$useHover({
        ...props,
        isDisabled: isDisabled
    });
    let contents = typeof rendered === 'string' ? /*#__PURE__*/ $dbUYi$react.createElement($dbUYi$Text, null, rendered) : rendered;
    let isKeyboardModality = $dbUYi$isFocusVisible();
    return(/*#__PURE__*/ $dbUYi$react.createElement($dbUYi$FocusRing, {
        focusRingClass: $dbUYi$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'focus-ring')
    }, /*#__PURE__*/ $dbUYi$react.createElement("div", {
        ...$dbUYi$mergeProps(optionProps, shouldFocusOnHover ? {
        } : hoverProps),
        ref: ref,
        className: $dbUYi$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'spectrum-Menu-item', {
            // If using virtual focus, apply focused styles to the item when the user is interacting with keyboard modality
            'is-focused': shouldUseVirtualFocus && isFocused && isKeyboardModality,
            'is-disabled': isDisabled,
            'is-selected': isSelected,
            'is-selectable': state.selectionManager.selectionMode !== 'none',
            // When shouldFocusOnHover is false, apply hover styles both when hovered with the mouse.
            // Otherwise, apply hover styles when focused using non-keyboard modality.
            'is-hovered': isHovered && !shouldFocusOnHover || isFocused && !isKeyboardModality
        })
    }, /*#__PURE__*/ $dbUYi$react.createElement($dbUYi$Grid, {
        UNSAFE_className: $dbUYi$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'spectrum-Menu-itemGrid')
    }, /*#__PURE__*/ $dbUYi$react.createElement($dbUYi$ClearSlots, null, /*#__PURE__*/ $dbUYi$react.createElement($dbUYi$SlotProvider, {
        slots: {
            text: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports))['spectrum-Menu-itemLabel'],
                ...labelProps
            },
            icon: {
                size: 'S',
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports))['spectrum-Menu-icon']
            },
            description: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports))['spectrum-Menu-description'],
                ...descriptionProps
            }
        }
    }, contents, isSelected && /*#__PURE__*/ $dbUYi$react.createElement($dbUYi$spectrumiconsuiCheckmarkMedium, {
        slot: "checkmark",
        UNSAFE_className: $dbUYi$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'spectrum-Menu-checkmark')
    })))))));
}










function $907f4100c33a6c2f$export$dca12b0bb56e4fc(props) {
    let { children: children , reusableView: reusableView , header: header  } = props;
    let item = reusableView.content;
    let { headingProps: headingProps , groupProps: groupProps  } = $dbUYi$useListBoxSection({
        heading: item.rendered,
        'aria-label': item['aria-label']
    });
    let { separatorProps: separatorProps  } = $dbUYi$useSeparator({
        elementType: 'li'
    });
    let headerRef = $dbUYi$useRef();
    $dbUYi$useVirtualizerItem({
        reusableView: header,
        ref: headerRef
    });
    let { direction: direction  } = $dbUYi$useLocale();
    let state = $dbUYi$useContext($ca44d58a4bb9be3f$export$7ff8f37d2d81a48d);
    return(/*#__PURE__*/ $dbUYi$react.createElement($dbUYi$Fragment, null, /*#__PURE__*/ $dbUYi$react.createElement("div", {
        role: "presentation",
        ref: headerRef,
        style: $dbUYi$layoutInfoToStyle(header.layoutInfo, direction)
    }, item.key !== state.collection.getFirstKey() && /*#__PURE__*/ $dbUYi$react.createElement("div", {
        ...separatorProps,
        className: $dbUYi$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'spectrum-Menu-divider')
    }), item.rendered && /*#__PURE__*/ $dbUYi$react.createElement("div", {
        ...headingProps,
        className: $dbUYi$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'spectrum-Menu-sectionHeading')
    }, item.rendered)), /*#__PURE__*/ $dbUYi$react.createElement("div", {
        ...groupProps,
        style: $dbUYi$layoutInfoToStyle(reusableView.layoutInfo, direction),
        className: $dbUYi$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'spectrum-Menu')
    }, children)));
}










function $0609d2e227383a47$export$25768ea656ae32a7(state) {
    let { scale: scale  } = $dbUYi$useProvider();
    let collator = $dbUYi$useCollator({
        usage: 'search',
        sensitivity: 'base'
    });
    let layout = $dbUYi$useMemo(()=>new $dbUYi$ListLayout({
            estimatedRowHeight: scale === 'large' ? 48 : 32,
            estimatedHeadingHeight: scale === 'large' ? 33 : 26,
            padding: scale === 'large' ? 5 : 4,
            loaderHeight: 40,
            placeholderHeight: scale === 'large' ? 48 : 32,
            collator: collator
        })
    , [
        collator,
        scale
    ]);
    layout.collection = state.collection;
    layout.disabledKeys = state.disabledKeys;
    return layout;
}
/** @private */ function $0609d2e227383a47$var$ListBoxBase(props, ref) {
    let { layout: layout , state: state , shouldSelectOnPressUp: shouldSelectOnPressUp , focusOnPointerEnter: focusOnPointerEnter , shouldUseVirtualFocus: shouldUseVirtualFocus , domProps: domProps = {
    } , transitionDuration: transitionDuration = 0 , onScroll: onScroll  } = props;
    let { listBoxProps: listBoxProps  } = $dbUYi$useListBox({
        ...props,
        keyboardDelegate: layout,
        isVirtualized: true
    }, state, ref);
    let { styleProps: styleProps  } = $dbUYi$useStyleProps(props);
    let formatMessage = $dbUYi$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($584fb08d5758edb3$exports)));
    // Sync loading state into the layout.
    layout.isLoading = props.isLoading;
    let renderWrapper = (parent, reusableView, children, renderChildren)=>{
        if (reusableView.viewType === 'section') return(/*#__PURE__*/ $dbUYi$react.createElement($907f4100c33a6c2f$export$dca12b0bb56e4fc, {
            key: reusableView.key,
            reusableView: reusableView,
            header: children.find((c)=>c.viewType === 'header'
            )
        }, renderChildren(children.filter((c)=>c.viewType === 'item'
        ))));
        return(/*#__PURE__*/ $dbUYi$react.createElement($dbUYi$VirtualizerItem, {
            key: reusableView.key,
            reusableView: reusableView,
            parent: parent
        }));
    };
    return(/*#__PURE__*/ $dbUYi$react.createElement($ca44d58a4bb9be3f$export$7ff8f37d2d81a48d.Provider, {
        value: state
    }, /*#__PURE__*/ $dbUYi$react.createElement($dbUYi$Virtualizer, {
        ...styleProps,
        ...$dbUYi$mergeProps(listBoxProps, domProps),
        ref: ref,
        focusedKey: state.selectionManager.focusedKey,
        sizeToFit: "height",
        scrollDirection: "vertical",
        className: $dbUYi$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'spectrum-Menu', styleProps.className),
        layout: layout,
        collection: state.collection,
        renderWrapper: renderWrapper,
        transitionDuration: transitionDuration,
        isLoading: props.isLoading,
        onLoadMore: props.onLoadMore,
        shouldUseVirtualFocus: shouldUseVirtualFocus,
        onScroll: onScroll
    }, (type, item)=>{
        if (type === 'item') return(/*#__PURE__*/ $dbUYi$react.createElement($cab3e898fb78c326$export$feb3b6b552c14a12, {
            item: item,
            shouldSelectOnPressUp: shouldSelectOnPressUp,
            shouldFocusOnHover: focusOnPointerEnter,
            shouldUseVirtualFocus: shouldUseVirtualFocus
        }));
        else if (type === 'loader') return(// aria-selected isn't needed here since this option is not selectable.
        // eslint-disable-next-line jsx-a11y/role-has-required-aria-props
        /*#__PURE__*/ $dbUYi$react.createElement("div", {
            role: "option",
            style: {
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center',
                height: '100%'
            }
        }, /*#__PURE__*/ $dbUYi$react.createElement($dbUYi$ProgressCircle, {
            isIndeterminate: true,
            size: "S",
            "aria-label": state.collection.size > 0 ? formatMessage('loadingMore') : formatMessage('loading'),
            UNSAFE_className: $dbUYi$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'spectrum-Dropdown-progressCircle')
        })));
        else if (type === 'placeholder') {
            let emptyState = props.renderEmptyState ? props.renderEmptyState() : null;
            if (emptyState == null) return null;
            return(/*#__PURE__*/ $dbUYi$react.createElement("div", {
                // aria-selected isn't needed here since this option is not selectable.
                // eslint-disable-next-line jsx-a11y/role-has-required-aria-props
                role: "option"
            }, emptyState));
        }
    })));
}
// forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
const $0609d2e227383a47$export$1afdcf349979fb7e = /*#__PURE__*/ $dbUYi$react.forwardRef($0609d2e227383a47$var$ListBoxBase);





function $c6e964ce023e5a0b$var$ListBox(props, ref) {
    let state = $dbUYi$useListState(props);
    let layout = $0609d2e227383a47$export$25768ea656ae32a7(state);
    let domRef = $dbUYi$useDOMRef(ref);
    return(/*#__PURE__*/ $dbUYi$react.createElement($0609d2e227383a47$export$1afdcf349979fb7e, {
        ...props,
        ref: domRef,
        state: state,
        layout: layout
    }));
}
// forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
/**
 * A list of options that can allow selection of one or more.
 */ const $c6e964ce023e5a0b$export$41f133550aa26f48 = /*#__PURE__*/ $dbUYi$react.forwardRef($c6e964ce023e5a0b$var$ListBox);






export {$09647559903a45ec$re_export$Item as Item, $09647559903a45ec$re_export$Section as Section, $c6e964ce023e5a0b$export$41f133550aa26f48 as ListBox, $0609d2e227383a47$export$25768ea656ae32a7 as useListBoxLayout, $0609d2e227383a47$export$1afdcf349979fb7e as ListBoxBase};
