// @react-spectrum/meter/dist/module.js
import "./main.css";
import {classNames as $dG2V2$classNames} from "@react-spectrum/utils";
import {ProgressBarBase as $dG2V2$ProgressBarBase} from "@react-spectrum/progress";
import $dG2V2$react from "react";
import {useMeter as $dG2V2$useMeter} from "@react-aria/meter";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $54c759f064724756$exports = {};

$parcel$export($54c759f064724756$exports, "Meter", () => $54c759f064724756$export$62e3ae2a4090b879);



var $0cd8e49fa97e4664$exports = {};

$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader", () => $0cd8e49fa97e4664$export$162ba336fb0c239f, (v) => $0cd8e49fa97e4664$export$162ba336fb0c239f = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader--indeterminate", () => $0cd8e49fa97e4664$export$1249dfbb98223794, (v) => $0cd8e49fa97e4664$export$1249dfbb98223794 = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader--sideLabel", () => $0cd8e49fa97e4664$export$99af8802cc2a10dc, (v) => $0cd8e49fa97e4664$export$99af8802cc2a10dc = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader-track", () => $0cd8e49fa97e4664$export$3e2e13fb57b5c74b, (v) => $0cd8e49fa97e4664$export$3e2e13fb57b5c74b = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader-fill", () => $0cd8e49fa97e4664$export$84c9ee7bc724b71b, (v) => $0cd8e49fa97e4664$export$84c9ee7bc724b71b = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader-label", () => $0cd8e49fa97e4664$export$5379ba198163be86, (v) => $0cd8e49fa97e4664$export$5379ba198163be86 = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader-percentage", () => $0cd8e49fa97e4664$export$ecc9c825de9d1a18, (v) => $0cd8e49fa97e4664$export$ecc9c825de9d1a18 = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader--small", () => $0cd8e49fa97e4664$export$b945305960ad74a7, (v) => $0cd8e49fa97e4664$export$b945305960ad74a7 = v);
$parcel$export($0cd8e49fa97e4664$exports, "indeterminate-loop-ltr", () => $0cd8e49fa97e4664$export$a889038a25113dd4, (v) => $0cd8e49fa97e4664$export$a889038a25113dd4 = v);
$parcel$export($0cd8e49fa97e4664$exports, "indeterminate-loop-rtl", () => $0cd8e49fa97e4664$export$aeff243cf3356f00, (v) => $0cd8e49fa97e4664$export$aeff243cf3356f00 = v);
$parcel$export($0cd8e49fa97e4664$exports, "spectrum-BarLoader--overBackground", () => $0cd8e49fa97e4664$export$55612d0bd9b5d41, (v) => $0cd8e49fa97e4664$export$55612d0bd9b5d41 = v);
$parcel$export($0cd8e49fa97e4664$exports, "is-positive", () => $0cd8e49fa97e4664$export$aea9fd919467f03a, (v) => $0cd8e49fa97e4664$export$aea9fd919467f03a = v);
$parcel$export($0cd8e49fa97e4664$exports, "is-warning", () => $0cd8e49fa97e4664$export$d4ebda6218901908, (v) => $0cd8e49fa97e4664$export$d4ebda6218901908 = v);
$parcel$export($0cd8e49fa97e4664$exports, "is-critical", () => $0cd8e49fa97e4664$export$a1f608a9810da50f, (v) => $0cd8e49fa97e4664$export$a1f608a9810da50f = v);
var $0cd8e49fa97e4664$export$162ba336fb0c239f;
var $0cd8e49fa97e4664$export$1249dfbb98223794;
var $0cd8e49fa97e4664$export$99af8802cc2a10dc;
var $0cd8e49fa97e4664$export$3e2e13fb57b5c74b;
var $0cd8e49fa97e4664$export$84c9ee7bc724b71b;
var $0cd8e49fa97e4664$export$5379ba198163be86;
var $0cd8e49fa97e4664$export$ecc9c825de9d1a18;
var $0cd8e49fa97e4664$export$b945305960ad74a7;
var $0cd8e49fa97e4664$export$a889038a25113dd4;
var $0cd8e49fa97e4664$export$aeff243cf3356f00;
var $0cd8e49fa97e4664$export$55612d0bd9b5d41;
var $0cd8e49fa97e4664$export$aea9fd919467f03a;
var $0cd8e49fa97e4664$export$d4ebda6218901908;
var $0cd8e49fa97e4664$export$a1f608a9810da50f;
$0cd8e49fa97e4664$export$162ba336fb0c239f = "spectrum-BarLoader_48541f";
$0cd8e49fa97e4664$export$1249dfbb98223794 = "spectrum-BarLoader--indeterminate_48541f";
$0cd8e49fa97e4664$export$99af8802cc2a10dc = "spectrum-BarLoader--sideLabel_48541f";
$0cd8e49fa97e4664$export$3e2e13fb57b5c74b = "spectrum-BarLoader-track_48541f";
$0cd8e49fa97e4664$export$84c9ee7bc724b71b = "spectrum-BarLoader-fill_48541f";
$0cd8e49fa97e4664$export$5379ba198163be86 = "spectrum-BarLoader-label_48541f";
$0cd8e49fa97e4664$export$ecc9c825de9d1a18 = "spectrum-BarLoader-percentage_48541f";
$0cd8e49fa97e4664$export$b945305960ad74a7 = "spectrum-BarLoader--small_48541f";
$0cd8e49fa97e4664$export$a889038a25113dd4 = "indeterminate-loop-ltr_48541f";
$0cd8e49fa97e4664$export$aeff243cf3356f00 = "indeterminate-loop-rtl_48541f";
$0cd8e49fa97e4664$export$55612d0bd9b5d41 = "spectrum-BarLoader--overBackground_48541f";
$0cd8e49fa97e4664$export$aea9fd919467f03a = "is-positive_48541f";
$0cd8e49fa97e4664$export$d4ebda6218901908 = "is-warning_48541f";
$0cd8e49fa97e4664$export$a1f608a9810da50f = "is-critical_48541f";



function $54c759f064724756$var$Meter(props, ref) {
    let { variant: variant , ...otherProps } = props;
    const { meterProps: meterProps , labelProps: labelProps  } = $dG2V2$useMeter(props);
    return(/*#__PURE__*/ $dG2V2$react.createElement($dG2V2$ProgressBarBase, {
        ...otherProps,
        ref: ref,
        barProps: meterProps,
        labelProps: labelProps,
        barClassName: $dG2V2$classNames((/*@__PURE__*/$parcel$interopDefault($0cd8e49fa97e4664$exports)), {
            'is-positive': variant === 'positive',
            'is-warning': variant === 'warning',
            'is-critical': variant === 'critical'
        })
    }));
}
/**
 * Meters are visual representations of a quantity or an achievement.
 * Their progress is determined by user actions, rather than system actions.
 */ let $54c759f064724756$export$62e3ae2a4090b879 = /*#__PURE__*/ $dG2V2$react.forwardRef($54c759f064724756$var$Meter);




export {$54c759f064724756$export$62e3ae2a4090b879 as Meter};
