// @react-aria/actiongroup/dist/main.js
var $i3cSM$reactariafocus = require("@react-aria/focus");
var $i3cSM$reactariautils = require("@react-aria/utils");
var $i3cSM$reactariai18n = require("@react-aria/i18n");
var $i3cSM$react = require("react");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $beb108c754cce68d$exports = {};

$parcel$export($beb108c754cce68d$exports, "useActionGroup", () => $beb108c754cce68d$export$f4bf8d43c16de704);



const $beb108c754cce68d$var$BUTTON_GROUP_ROLES = {
    'none': 'toolbar',
    'single': 'radiogroup',
    'multiple': 'toolbar'
};
function $beb108c754cce68d$export$f4bf8d43c16de704(props, state, ref) {
    let { isDisabled: isDisabled , orientation: orientation = 'horizontal'  } = props;
    let allKeys = [
        ...state.collection.getKeys()
    ];
    if (!allKeys.some((key)=>!state.disabledKeys.has(key)
    )) isDisabled = true;
    let { direction: direction  } = $i3cSM$reactariai18n.useLocale();
    let focusManager = $i3cSM$reactariafocus.createFocusManager(ref);
    let flipDirection = direction === 'rtl' && orientation === 'horizontal';
    let onKeyDown = (e)=>{
        if (!e.currentTarget.contains(e.target)) return;
        switch(e.key){
            case 'ArrowRight':
            case 'ArrowDown':
                e.preventDefault();
                e.stopPropagation();
                if (e.key === 'ArrowRight' && flipDirection) focusManager.focusPrevious({
                    wrap: true
                });
                else focusManager.focusNext({
                    wrap: true
                });
                break;
            case 'ArrowLeft':
            case 'ArrowUp':
                e.preventDefault();
                e.stopPropagation();
                if (e.key === 'ArrowLeft' && flipDirection) focusManager.focusNext({
                    wrap: true
                });
                else focusManager.focusPrevious({
                    wrap: true
                });
                break;
        }
    };
    let role = $beb108c754cce68d$var$BUTTON_GROUP_ROLES[state.selectionManager.selectionMode];
    return {
        actionGroupProps: {
            ...$i3cSM$reactariautils.filterDOMProps(props, {
                labelable: true
            }),
            role: role,
            'aria-orientation': role === 'toolbar' ? orientation : null,
            'aria-disabled': isDisabled,
            onKeyDown: onKeyDown
        }
    };
}


var $a9279da88ed3f26d$exports = {};

$parcel$export($a9279da88ed3f26d$exports, "useActionGroupItem", () => $a9279da88ed3f26d$export$9597202bd3099a29);


const $a9279da88ed3f26d$var$BUTTON_ROLES = {
    'none': null,
    'single': 'radio',
    'multiple': 'checkbox'
};
function $a9279da88ed3f26d$export$9597202bd3099a29(props, state, ref) {
    let selectionMode = state.selectionManager.selectionMode;
    let buttonProps = {
        role: $a9279da88ed3f26d$var$BUTTON_ROLES[selectionMode]
    };
    if (selectionMode !== 'none') {
        let isSelected = state.selectionManager.isSelected(props.key);
        buttonProps['aria-checked'] = isSelected;
    }
    let isFocused = props.key === state.selectionManager.focusedKey;
    let lastRender = $i3cSM$react.useRef({
        isFocused: isFocused,
        state: state
    });
    lastRender.current = {
        isFocused: isFocused,
        state: state
    };
    // If the focused item is removed from the DOM, reset the focused key to null.
    // eslint-disable-next-line arrow-body-style
    $i3cSM$react.useEffect(()=>{
        return ()=>{
            if (lastRender.current.isFocused) lastRender.current.state.selectionManager.setFocusedKey(null);
        };
    }, []);
    return {
        buttonProps: $i3cSM$reactariautils.mergeProps(buttonProps, {
            tabIndex: isFocused || state.selectionManager.focusedKey == null ? 0 : -1,
            onFocus () {
                state.selectionManager.setFocusedKey(props.key);
            },
            onPress () {
                state.selectionManager.select(props.key);
            }
        })
    };
}


$parcel$exportWildcard(module.exports, $beb108c754cce68d$exports);
$parcel$exportWildcard(module.exports, $a9279da88ed3f26d$exports);
