// @react-aria/actiongroup/dist/module.js
import {createFocusManager as $guTId$createFocusManager} from "@react-aria/focus";
import {filterDOMProps as $guTId$filterDOMProps, mergeProps as $guTId$mergeProps} from "@react-aria/utils";
import {useLocale as $guTId$useLocale} from "@react-aria/i18n";
import {useRef as $guTId$useRef, useEffect as $guTId$useEffect} from "react";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $878c90050cf0d952$exports = {};

$parcel$export($878c90050cf0d952$exports, "useActionGroup", () => $878c90050cf0d952$export$f4bf8d43c16de704);



const $878c90050cf0d952$var$BUTTON_GROUP_ROLES = {
    'none': 'toolbar',
    'single': 'radiogroup',
    'multiple': 'toolbar'
};
function $878c90050cf0d952$export$f4bf8d43c16de704(props, state, ref) {
    let { isDisabled: isDisabled , orientation: orientation = 'horizontal'  } = props;
    let allKeys = [
        ...state.collection.getKeys()
    ];
    if (!allKeys.some((key)=>!state.disabledKeys.has(key)
    )) isDisabled = true;
    let { direction: direction  } = $guTId$useLocale();
    let focusManager = $guTId$createFocusManager(ref);
    let flipDirection = direction === 'rtl' && orientation === 'horizontal';
    let onKeyDown = (e)=>{
        if (!e.currentTarget.contains(e.target)) return;
        switch(e.key){
            case 'ArrowRight':
            case 'ArrowDown':
                e.preventDefault();
                e.stopPropagation();
                if (e.key === 'ArrowRight' && flipDirection) focusManager.focusPrevious({
                    wrap: true
                });
                else focusManager.focusNext({
                    wrap: true
                });
                break;
            case 'ArrowLeft':
            case 'ArrowUp':
                e.preventDefault();
                e.stopPropagation();
                if (e.key === 'ArrowLeft' && flipDirection) focusManager.focusNext({
                    wrap: true
                });
                else focusManager.focusPrevious({
                    wrap: true
                });
                break;
        }
    };
    let role = $878c90050cf0d952$var$BUTTON_GROUP_ROLES[state.selectionManager.selectionMode];
    return {
        actionGroupProps: {
            ...$guTId$filterDOMProps(props, {
                labelable: true
            }),
            role: role,
            'aria-orientation': role === 'toolbar' ? orientation : null,
            'aria-disabled': isDisabled,
            onKeyDown: onKeyDown
        }
    };
}


var $1ffd55f520044688$exports = {};

$parcel$export($1ffd55f520044688$exports, "useActionGroupItem", () => $1ffd55f520044688$export$9597202bd3099a29);


const $1ffd55f520044688$var$BUTTON_ROLES = {
    'none': null,
    'single': 'radio',
    'multiple': 'checkbox'
};
function $1ffd55f520044688$export$9597202bd3099a29(props, state, ref) {
    let selectionMode = state.selectionManager.selectionMode;
    let buttonProps = {
        role: $1ffd55f520044688$var$BUTTON_ROLES[selectionMode]
    };
    if (selectionMode !== 'none') {
        let isSelected = state.selectionManager.isSelected(props.key);
        buttonProps['aria-checked'] = isSelected;
    }
    let isFocused = props.key === state.selectionManager.focusedKey;
    let lastRender = $guTId$useRef({
        isFocused: isFocused,
        state: state
    });
    lastRender.current = {
        isFocused: isFocused,
        state: state
    };
    // If the focused item is removed from the DOM, reset the focused key to null.
    // eslint-disable-next-line arrow-body-style
    $guTId$useEffect(()=>{
        return ()=>{
            if (lastRender.current.isFocused) lastRender.current.state.selectionManager.setFocusedKey(null);
        };
    }, []);
    return {
        buttonProps: $guTId$mergeProps(buttonProps, {
            tabIndex: isFocused || state.selectionManager.focusedKey == null ? 0 : -1,
            onFocus () {
                state.selectionManager.setFocusedKey(props.key);
            },
            onPress () {
                state.selectionManager.select(props.key);
            }
        })
    };
}




export {$878c90050cf0d952$export$f4bf8d43c16de704 as useActionGroup, $1ffd55f520044688$export$9597202bd3099a29 as useActionGroupItem};
