// @react-aria/spinbutton/dist/main.js
var $6zCO4$reactarialiveannouncer = require("@react-aria/live-announcer");
var $6zCO4$react = require("react");
var $6zCO4$reactariautils = require("@react-aria/utils");
var $6zCO4$reactariai18n = require("@react-aria/i18n");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $e968a837e1097a74$exports = {};

$parcel$export($e968a837e1097a74$exports, "useSpinButton", () => $e968a837e1097a74$export$e908e06f4b8e3402);


var $fe86a6380e33c053$exports = {};
var $fbbbe46458541aec$exports = {};
$fbbbe46458541aec$exports = JSON.parse("{\"Empty\":\"فارغ\"}");


var $0f3703b50aa74823$exports = {};
$0f3703b50aa74823$exports = JSON.parse("{\"Empty\":\"Изпразни\"}");


var $7af4f9e9ea420faa$exports = {};
$7af4f9e9ea420faa$exports = JSON.parse("{\"Empty\":\"Prázdné\"}");


var $baa41c4063617676$exports = {};
$baa41c4063617676$exports = JSON.parse("{\"Empty\":\"Tom\"}");


var $838761f721704d02$exports = {};
$838761f721704d02$exports = JSON.parse("{\"Empty\":\"Leer\"}");


var $a413eacfc208d421$exports = {};
$a413eacfc208d421$exports = JSON.parse("{\"Empty\":\"Άδειο\"}");


var $12355e0c0ae5aacd$exports = {};
$12355e0c0ae5aacd$exports = JSON.parse("{\"Empty\":\"Empty\"}");


var $01eb02e705d8a81a$exports = {};
$01eb02e705d8a81a$exports = JSON.parse("{\"Empty\":\"Vacío\"}");


var $4f12dba61481ebb9$exports = {};
$4f12dba61481ebb9$exports = JSON.parse("{\"Empty\":\"Tühjenda\"}");


var $3afb5cdb43b0cfc3$exports = {};
$3afb5cdb43b0cfc3$exports = JSON.parse("{\"Empty\":\"Tyhjä\"}");


var $d635d4486705a374$exports = {};
$d635d4486705a374$exports = JSON.parse("{\"Empty\":\"Vide\"}");


var $b0ed2f3b2e38d15a$exports = {};
$b0ed2f3b2e38d15a$exports = JSON.parse("{\"Empty\":\"ריק\"}");


var $ae71581dc00ac67e$exports = {};
$ae71581dc00ac67e$exports = JSON.parse("{\"Empty\":\"Prazno\"}");


var $da76030387fc33a4$exports = {};
$da76030387fc33a4$exports = JSON.parse("{\"Empty\":\"Üres\"}");


var $cd614d829e34bf35$exports = {};
$cd614d829e34bf35$exports = JSON.parse("{\"Empty\":\"Vuoto\"}");


var $9c9da26bcb9c6085$exports = {};
$9c9da26bcb9c6085$exports = JSON.parse("{\"Empty\":\"空\"}");


var $82b97fcd291ed492$exports = {};
$82b97fcd291ed492$exports = JSON.parse("{\"Empty\":\"비어 있음\"}");


var $9be3339dafd7bd5e$exports = {};
$9be3339dafd7bd5e$exports = JSON.parse("{\"Empty\":\"Tuščias\"}");


var $c8054fd4ac95c5d7$exports = {};
$c8054fd4ac95c5d7$exports = JSON.parse("{\"Empty\":\"Tukšs\"}");


var $a77697c661c215d6$exports = {};
$a77697c661c215d6$exports = JSON.parse("{\"Empty\":\"Tom\"}");


var $e9bb9556c489f53c$exports = {};
$e9bb9556c489f53c$exports = JSON.parse("{\"Empty\":\"Leeg\"}");


var $ba054ddbb7b59773$exports = {};
$ba054ddbb7b59773$exports = JSON.parse("{\"Empty\":\"Pusty\"}");


var $dae54d9824dc1188$exports = {};
$dae54d9824dc1188$exports = JSON.parse("{\"Empty\":\"Vazio\"}");


var $7e29fe7232a32b17$exports = {};
$7e29fe7232a32b17$exports = JSON.parse("{\"Empty\":\"Vazio\"}");


var $70dcac2ba5f59313$exports = {};
$70dcac2ba5f59313$exports = JSON.parse("{\"Empty\":\"Gol\"}");


var $ab018df4034ae4dc$exports = {};
$ab018df4034ae4dc$exports = JSON.parse("{\"Empty\":\"Не заполнено\"}");


var $63e03c9ebf40d08b$exports = {};
$63e03c9ebf40d08b$exports = JSON.parse("{\"Empty\":\"Prázdne\"}");


var $24779fe8276efc7e$exports = {};
$24779fe8276efc7e$exports = JSON.parse("{\"Empty\":\"Prazen\"}");


var $58e7a92bfea66474$exports = {};
$58e7a92bfea66474$exports = JSON.parse("{\"Empty\":\"Prazno\"}");


var $58c9f17f87f1eea7$exports = {};
$58c9f17f87f1eea7$exports = JSON.parse("{\"Empty\":\"Tomt\"}");


var $3e11a46234157763$exports = {};
$3e11a46234157763$exports = JSON.parse("{\"Empty\":\"Boş\"}");


var $dbc3859b1c952d88$exports = {};
$dbc3859b1c952d88$exports = JSON.parse("{\"Empty\":\"Пусто\"}");


var $c9582137fb95cd9f$exports = {};
$c9582137fb95cd9f$exports = JSON.parse("{\"Empty\":\"空\"}");


var $23a291fd13b1d998$exports = {};
$23a291fd13b1d998$exports = JSON.parse("{\"Empty\":\"空白\"}");


$fe86a6380e33c053$exports = {
    "ar-AE": $fbbbe46458541aec$exports,
    "bg-BG": $0f3703b50aa74823$exports,
    "cs-CZ": $7af4f9e9ea420faa$exports,
    "da-DK": $baa41c4063617676$exports,
    "de-DE": $838761f721704d02$exports,
    "el-GR": $a413eacfc208d421$exports,
    "en-US": $12355e0c0ae5aacd$exports,
    "es-ES": $01eb02e705d8a81a$exports,
    "et-EE": $4f12dba61481ebb9$exports,
    "fi-FI": $3afb5cdb43b0cfc3$exports,
    "fr-FR": $d635d4486705a374$exports,
    "he-IL": $b0ed2f3b2e38d15a$exports,
    "hr-HR": $ae71581dc00ac67e$exports,
    "hu-HU": $da76030387fc33a4$exports,
    "it-IT": $cd614d829e34bf35$exports,
    "ja-JP": $9c9da26bcb9c6085$exports,
    "ko-KR": $82b97fcd291ed492$exports,
    "lt-LT": $9be3339dafd7bd5e$exports,
    "lv-LV": $c8054fd4ac95c5d7$exports,
    "nb-NO": $a77697c661c215d6$exports,
    "nl-NL": $e9bb9556c489f53c$exports,
    "pl-PL": $ba054ddbb7b59773$exports,
    "pt-BR": $dae54d9824dc1188$exports,
    "pt-PT": $7e29fe7232a32b17$exports,
    "ro-RO": $70dcac2ba5f59313$exports,
    "ru-RU": $ab018df4034ae4dc$exports,
    "sk-SK": $63e03c9ebf40d08b$exports,
    "sl-SI": $24779fe8276efc7e$exports,
    "sr-SP": $58e7a92bfea66474$exports,
    "sv-SE": $58c9f17f87f1eea7$exports,
    "tr-TR": $3e11a46234157763$exports,
    "uk-UA": $dbc3859b1c952d88$exports,
    "zh-CN": $c9582137fb95cd9f$exports,
    "zh-TW": $23a291fd13b1d998$exports
};




function $e968a837e1097a74$export$e908e06f4b8e3402(props) {
    const _async = $6zCO4$react.useRef();
    let { value: value , textValue: textValue , minValue: minValue , maxValue: maxValue , isDisabled: isDisabled , isReadOnly: isReadOnly , isRequired: isRequired , onIncrement: onIncrement , onIncrementPage: onIncrementPage , onDecrement: onDecrement , onDecrementPage: onDecrementPage , onDecrementToMin: onDecrementToMin , onIncrementToMax: onIncrementToMax  } = props;
    const formatMessage = $6zCO4$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($fe86a6380e33c053$exports)));
    const propsRef = $6zCO4$react.useRef(props);
    propsRef.current = props;
    const clearAsync = ()=>clearTimeout(_async.current)
    ;
    // eslint-disable-next-line arrow-body-style
    $6zCO4$react.useEffect(()=>{
        return ()=>clearAsync()
        ;
    }, []);
    let onKeyDown = (e)=>{
        if (e.ctrlKey || e.metaKey || e.shiftKey || e.altKey || isReadOnly) return;
        switch(e.key){
            case 'PageUp':
                if (onIncrementPage) {
                    e.preventDefault();
                    onIncrementPage();
                    break;
                }
            // fallthrough!
            case 'ArrowUp':
            case 'Up':
                if (onIncrement) {
                    e.preventDefault();
                    onIncrement();
                }
                break;
            case 'PageDown':
                if (onDecrementPage) {
                    e.preventDefault();
                    onDecrementPage();
                    break;
                }
            // fallthrough
            case 'ArrowDown':
            case 'Down':
                if (onDecrement) {
                    e.preventDefault();
                    onDecrement();
                }
                break;
            case 'Home':
                if (onDecrementToMin) {
                    e.preventDefault();
                    onDecrementToMin();
                }
                break;
            case 'End':
                if (onIncrementToMax) {
                    e.preventDefault();
                    onIncrementToMax();
                }
                break;
        }
    };
    let isFocused = $6zCO4$react.useRef(false);
    let onFocus = ()=>{
        isFocused.current = true;
    };
    let onBlur = ()=>{
        isFocused.current = false;
    };
    // Replace Unicode hyphen-minus (U+002D) with minus sign (U+2212).
    // This ensures that macOS VoiceOver announces it as "minus" even with other characters between the minus sign
    // and the number (e.g. currency symbol). Otherwise it announces nothing because it assumes the character is a hyphen.
    // In addition, replace the empty string with the word "Empty" so that iOS VoiceOver does not read "50%" for an empty field.
    textValue = textValue === '' ? formatMessage('Empty') : (textValue || `${value}`).replace('-', '\u2212');
    $6zCO4$react.useEffect(()=>{
        if (isFocused.current) $6zCO4$reactarialiveannouncer.announce(textValue, 'assertive');
    }, [
        textValue
    ]);
    const onIncrementPressStart = $6zCO4$react.useCallback((initialStepDelay)=>{
        clearAsync();
        propsRef.current.onIncrement();
        // Start spinning after initial delay
        _async.current = window.setTimeout(()=>{
            if (isNaN(maxValue) || isNaN(value) || value < maxValue) onIncrementPressStart(60);
        }, initialStepDelay);
    }, [
        onIncrement,
        maxValue,
        value
    ]);
    const onDecrementPressStart = $6zCO4$react.useCallback((initialStepDelay)=>{
        clearAsync();
        propsRef.current.onDecrement();
        // Start spinning after initial delay
        _async.current = window.setTimeout(()=>{
            if (isNaN(minValue) || isNaN(value) || value > minValue) onDecrementPressStart(60);
        }, initialStepDelay);
    }, [
        onDecrement,
        minValue,
        value
    ]);
    let cancelContextMenu = (e)=>{
        e.preventDefault();
    };
    let { addGlobalListener: addGlobalListener , removeAllGlobalListeners: removeAllGlobalListeners  } = $6zCO4$reactariautils.useGlobalListeners();
    return {
        spinButtonProps: {
            role: 'spinbutton',
            'aria-valuenow': !isNaN(value) ? value : null,
            'aria-valuetext': textValue,
            'aria-valuemin': minValue,
            'aria-valuemax': maxValue,
            'aria-disabled': isDisabled || null,
            'aria-readonly': isReadOnly || null,
            'aria-required': isRequired || null,
            onKeyDown: onKeyDown,
            onFocus: onFocus,
            onBlur: onBlur
        },
        incrementButtonProps: {
            onPressStart: ()=>{
                onIncrementPressStart(400);
                addGlobalListener(window, 'contextmenu', cancelContextMenu);
            },
            onPressEnd: ()=>{
                clearAsync();
                removeAllGlobalListeners();
            },
            onFocus: onFocus,
            onBlur: onBlur
        },
        decrementButtonProps: {
            onPressStart: ()=>{
                onDecrementPressStart(400);
                addGlobalListener(window, 'contextmenu', cancelContextMenu);
            },
            onPressEnd: ()=>{
                clearAsync();
                removeAllGlobalListeners();
            },
            onFocus: onFocus,
            onBlur: onBlur
        }
    };
}


$parcel$exportWildcard(module.exports, $e968a837e1097a74$exports);
