// @react-aria/separator/dist/module.js
import {filterDOMProps as $1uahB$filterDOMProps} from "@react-aria/utils";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $c7a318c69e40f5a5$exports = {};

$parcel$export($c7a318c69e40f5a5$exports, "useSeparator", () => $c7a318c69e40f5a5$export$52210f68a14655d0);

function $c7a318c69e40f5a5$export$52210f68a14655d0(props) {
    let domProps = $1uahB$filterDOMProps(props, {
        labelable: true
    });
    let ariaOrientation;
    // if orientation is horizontal, aria-orientation default is horizontal, so we leave it undefined
    // if it's vertical, we need to specify it
    if (props.orientation === 'vertical') ariaOrientation = 'vertical';
    // hr elements implicitly have role = separator and a horizontal orientation
    if (props.elementType !== 'hr') return {
        separatorProps: {
            ...domProps,
            role: 'separator',
            'aria-orientation': ariaOrientation
        }
    };
    return {
        separatorProps: domProps
    };
}




export {$c7a318c69e40f5a5$export$52210f68a14655d0 as useSeparator};
