// @react-spectrum/menu/dist/main.js
require("./main.css");
var $lJf9X$reactstatelycollections = require("@react-stately/collections");
var $lJf9X$reactspectrumutils = require("@react-spectrum/utils");
var $lJf9X$reactariaoverlays = require("@react-aria/overlays");
var $lJf9X$reactariafocus = require("@react-aria/focus");
var $lJf9X$reactspectrumoverlays = require("@react-spectrum/overlays");
var $lJf9X$reactariainteractions = require("@react-aria/interactions");
var $lJf9X$react = require("react");
var $lJf9X$reactariamenu = require("@react-aria/menu");
var $lJf9X$reactstatelymenu = require("@react-stately/menu");
var $lJf9X$reactariautils = require("@react-aria/utils");
var $lJf9X$reactstatelytree = require("@react-stately/tree");
var $lJf9X$spectrumiconsuiCheckmarkMedium = require("@spectrum-icons/ui/CheckmarkMedium");
var $lJf9X$reactspectrumlayout = require("@react-spectrum/layout");
var $lJf9X$reactspectrumtext = require("@react-spectrum/text");
var $lJf9X$reactariaseparator = require("@react-aria/separator");
var $lJf9X$reactspectrumbutton = require("@react-spectrum/button");
var $lJf9X$spectrumiconsworkflowMore = require("@spectrum-icons/workflow/More");
var $lJf9X$reactariai18n = require("@react-aria/i18n");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Item", () => $lJf9X$reactstatelycollections.Item);
$parcel$export(module.exports, "Section", () => $lJf9X$reactstatelycollections.Section);
var $0a5a3ef3fb6afa6f$exports = {};

$parcel$export($0a5a3ef3fb6afa6f$exports, "MenuTrigger", () => $0a5a3ef3fb6afa6f$export$27d2ad3c5815583e);




const $185e8220149307d8$export$c7e742effb1c51e2 = ($parcel$interopDefault($lJf9X$react)).createContext({
});
function $185e8220149307d8$export$21c7ab35b39f78ec() {
    return $lJf9X$react.useContext($185e8220149307d8$export$c7e742effb1c51e2);
}





var $d53361c88b80133d$exports = {};

$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-popover", () => $d53361c88b80133d$export$d959f30786125b, (v) => $d53361c88b80133d$export$d959f30786125b = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu", () => $d53361c88b80133d$export$fae6af6decd50ac5, (v) => $d53361c88b80133d$export$fae6af6decd50ac5 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-sectionHeading", () => $d53361c88b80133d$export$c20ea4b35020fbdd, (v) => $d53361c88b80133d$export$c20ea4b35020fbdd = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-checkmark", () => $d53361c88b80133d$export$8e15a726e4f7e62d, (v) => $d53361c88b80133d$export$8e15a726e4f7e62d = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-item", () => $d53361c88b80133d$export$45faad328e6c532b, (v) => $d53361c88b80133d$export$45faad328e6c532b = v);
$parcel$export($d53361c88b80133d$exports, "is-selected", () => $d53361c88b80133d$export$1e0fb04f31d3c22a, (v) => $d53361c88b80133d$export$1e0fb04f31d3c22a = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Icon", () => $d53361c88b80133d$export$d374b04f30360026, (v) => $d53361c88b80133d$export$d374b04f30360026 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-itemLabel", () => $d53361c88b80133d$export$1aa6419c5c650567, (v) => $d53361c88b80133d$export$1aa6419c5c650567 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-itemIcon", () => $d53361c88b80133d$export$f431001e1bb54633, (v) => $d53361c88b80133d$export$f431001e1bb54633 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-itemLabel--wrapping", () => $d53361c88b80133d$export$465092164a8e1d98, (v) => $d53361c88b80133d$export$465092164a8e1d98 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-divider", () => $d53361c88b80133d$export$e892026f534efed8, (v) => $d53361c88b80133d$export$e892026f534efed8 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-itemGrid", () => $d53361c88b80133d$export$ee5f52784f57f9c8, (v) => $d53361c88b80133d$export$ee5f52784f57f9c8 = v);
$parcel$export($d53361c88b80133d$exports, "is-selectable", () => $d53361c88b80133d$export$fbd22ba224473193, (v) => $d53361c88b80133d$export$fbd22ba224473193 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-end", () => $d53361c88b80133d$export$926a1bb74ed65667, (v) => $d53361c88b80133d$export$926a1bb74ed65667 = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-icon", () => $d53361c88b80133d$export$cbbae63ef9e44c7a, (v) => $d53361c88b80133d$export$cbbae63ef9e44c7a = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-description", () => $d53361c88b80133d$export$36f1275051e39ead, (v) => $d53361c88b80133d$export$36f1275051e39ead = v);
$parcel$export($d53361c88b80133d$exports, "spectrum-Menu-keyboard", () => $d53361c88b80133d$export$4f9817a7dff6ba0f, (v) => $d53361c88b80133d$export$4f9817a7dff6ba0f = v);
$parcel$export($d53361c88b80133d$exports, "focus-ring", () => $d53361c88b80133d$export$f39a09f249340e2a, (v) => $d53361c88b80133d$export$f39a09f249340e2a = v);
$parcel$export($d53361c88b80133d$exports, "is-focused", () => $d53361c88b80133d$export$e7dc768d35940237, (v) => $d53361c88b80133d$export$e7dc768d35940237 = v);
$parcel$export($d53361c88b80133d$exports, "is-hovered", () => $d53361c88b80133d$export$b8813cd5d7824ce7, (v) => $d53361c88b80133d$export$b8813cd5d7824ce7 = v);
$parcel$export($d53361c88b80133d$exports, "is-highlighted", () => $d53361c88b80133d$export$451c613d0772eb99, (v) => $d53361c88b80133d$export$451c613d0772eb99 = v);
$parcel$export($d53361c88b80133d$exports, "is-open", () => $d53361c88b80133d$export$a9781837241c946d, (v) => $d53361c88b80133d$export$a9781837241c946d = v);
$parcel$export($d53361c88b80133d$exports, "is-active", () => $d53361c88b80133d$export$20fd0f7cd4e6112f, (v) => $d53361c88b80133d$export$20fd0f7cd4e6112f = v);
$parcel$export($d53361c88b80133d$exports, "is-disabled", () => $d53361c88b80133d$export$d35bc1e505d1ebbf, (v) => $d53361c88b80133d$export$d35bc1e505d1ebbf = v);
var $d53361c88b80133d$export$d959f30786125b;
var $d53361c88b80133d$export$fae6af6decd50ac5;
var $d53361c88b80133d$export$c20ea4b35020fbdd;
var $d53361c88b80133d$export$8e15a726e4f7e62d;
var $d53361c88b80133d$export$45faad328e6c532b;
var $d53361c88b80133d$export$1e0fb04f31d3c22a;
var $d53361c88b80133d$export$d374b04f30360026;
var $d53361c88b80133d$export$1aa6419c5c650567;
var $d53361c88b80133d$export$f431001e1bb54633;
var $d53361c88b80133d$export$465092164a8e1d98;
var $d53361c88b80133d$export$e892026f534efed8;
var $d53361c88b80133d$export$ee5f52784f57f9c8;
var $d53361c88b80133d$export$fbd22ba224473193;
var $d53361c88b80133d$export$926a1bb74ed65667;
var $d53361c88b80133d$export$cbbae63ef9e44c7a;
var $d53361c88b80133d$export$36f1275051e39ead;
var $d53361c88b80133d$export$4f9817a7dff6ba0f;
var $d53361c88b80133d$export$f39a09f249340e2a;
var $d53361c88b80133d$export$e7dc768d35940237;
var $d53361c88b80133d$export$b8813cd5d7824ce7;
var $d53361c88b80133d$export$451c613d0772eb99;
var $d53361c88b80133d$export$a9781837241c946d;
var $d53361c88b80133d$export$20fd0f7cd4e6112f;
var $d53361c88b80133d$export$d35bc1e505d1ebbf;
$d53361c88b80133d$export$d959f30786125b = "spectrum-Menu-popover_7ea47c";
$d53361c88b80133d$export$fae6af6decd50ac5 = "spectrum-Menu_7ea47c";
$d53361c88b80133d$export$c20ea4b35020fbdd = "spectrum-Menu-sectionHeading_7ea47c";
$d53361c88b80133d$export$8e15a726e4f7e62d = "spectrum-Menu-checkmark_7ea47c";
$d53361c88b80133d$export$45faad328e6c532b = "spectrum-Menu-item_7ea47c";
$d53361c88b80133d$export$1e0fb04f31d3c22a = "is-selected_7ea47c";
$d53361c88b80133d$export$d374b04f30360026 = "spectrum-Icon_7ea47c";
$d53361c88b80133d$export$1aa6419c5c650567 = "spectrum-Menu-itemLabel_7ea47c";
$d53361c88b80133d$export$f431001e1bb54633 = "spectrum-Menu-itemIcon_7ea47c";
$d53361c88b80133d$export$465092164a8e1d98 = "spectrum-Menu-itemLabel--wrapping_7ea47c";
$d53361c88b80133d$export$e892026f534efed8 = "spectrum-Menu-divider_7ea47c";
$d53361c88b80133d$export$ee5f52784f57f9c8 = "spectrum-Menu-itemGrid_7ea47c";
$d53361c88b80133d$export$fbd22ba224473193 = "is-selectable_7ea47c";
$d53361c88b80133d$export$926a1bb74ed65667 = "spectrum-Menu-end_7ea47c";
$d53361c88b80133d$export$cbbae63ef9e44c7a = "spectrum-Menu-icon_7ea47c";
$d53361c88b80133d$export$36f1275051e39ead = "spectrum-Menu-description_7ea47c";
$d53361c88b80133d$export$4f9817a7dff6ba0f = "spectrum-Menu-keyboard_7ea47c";
$d53361c88b80133d$export$f39a09f249340e2a = "focus-ring_7ea47c";
$d53361c88b80133d$export$e7dc768d35940237 = "is-focused_7ea47c";
$d53361c88b80133d$export$b8813cd5d7824ce7 = "is-hovered_7ea47c";
$d53361c88b80133d$export$451c613d0772eb99 = "is-highlighted_7ea47c";
$d53361c88b80133d$export$a9781837241c946d = "is-open_7ea47c";
$d53361c88b80133d$export$20fd0f7cd4e6112f = "is-active_7ea47c";
$d53361c88b80133d$export$d35bc1e505d1ebbf = "is-disabled_7ea47c";




function $0a5a3ef3fb6afa6f$var$MenuTrigger(props, ref) {
    let menuPopoverRef = $lJf9X$react.useRef();
    let triggerRef = $lJf9X$react.useRef();
    let domRef = $lJf9X$reactspectrumutils.useDOMRef(ref);
    let menuTriggerRef = domRef || triggerRef;
    let menuRef = $lJf9X$react.useRef();
    let { children: children , align: align = 'start' , shouldFlip: shouldFlip = true , direction: direction = 'bottom' , closeOnSelect: closeOnSelect , trigger: trigger = 'press'  } = props;
    let [menuTrigger, menu] = ($parcel$interopDefault($lJf9X$react)).Children.toArray(children);
    let state = $lJf9X$reactstatelymenu.useMenuTriggerState(props);
    let { menuTriggerProps: menuTriggerProps , menuProps: menuProps  } = $lJf9X$reactariamenu.useMenuTrigger({
        trigger: trigger
    }, state, menuTriggerRef);
    let initialPlacement;
    switch(direction){
        case 'left':
        case 'right':
        case 'start':
        case 'end':
            initialPlacement = `${direction} ${align === 'end' ? 'bottom' : 'top'}`;
            break;
        case 'bottom':
        case 'top':
        default:
            initialPlacement = `${direction} ${align}`;
    }
    let isMobile = $lJf9X$reactspectrumutils.useIsMobileDevice();
    let { overlayProps: positionProps , placement: placement  } = $lJf9X$reactariaoverlays.useOverlayPosition({
        targetRef: menuTriggerRef,
        overlayRef: $lJf9X$reactspectrumutils.unwrapDOMRef(menuPopoverRef),
        scrollRef: menuRef,
        placement: initialPlacement,
        shouldFlip: shouldFlip,
        isOpen: state.isOpen && !isMobile,
        onClose: state.close
    });
    let menuContext = {
        ...menuProps,
        ref: menuRef,
        onClose: state.close,
        closeOnSelect: closeOnSelect,
        autoFocus: state.focusStrategy || true,
        UNSAFE_style: isMobile ? {
            width: '100%',
            maxHeight: 'inherit'
        } : undefined,
        UNSAFE_className: $lJf9X$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), {
            'spectrum-Menu-popover': !isMobile
        })
    };
    let contents = /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($lJf9X$reactariafocus.FocusScope, {
        restoreFocus: true,
        contain: isMobile
    }, /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($lJf9X$reactariaoverlays.DismissButton, {
        onDismiss: state.close
    }), menu, /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($lJf9X$reactariaoverlays.DismissButton, {
        onDismiss: state.close
    }));
    // On small screen devices, the menu is rendered in a tray, otherwise a popover.
    let overlay;
    if (isMobile) overlay = /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($lJf9X$reactspectrumoverlays.Tray, {
        isOpen: state.isOpen,
        onClose: state.close
    }, contents);
    else overlay = /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($lJf9X$reactspectrumoverlays.Popover, {
        isOpen: state.isOpen,
        UNSAFE_style: positionProps.style,
        ref: menuPopoverRef,
        placement: placement,
        hideArrow: true,
        onClose: state.close,
        shouldCloseOnBlur: true
    }, contents);
    return(/*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($lJf9X$react.Fragment, null, /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($lJf9X$reactspectrumutils.SlotProvider, {
        slots: {
            actionButton: {
                holdAffordance: trigger === 'longPress'
            }
        }
    }, /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($lJf9X$reactariainteractions.PressResponder, {
        ...menuTriggerProps,
        ref: menuTriggerRef,
        isPressed: state.isOpen
    }, menuTrigger)), /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($185e8220149307d8$export$c7e742effb1c51e2.Provider, {
        value: menuContext
    }, overlay)));
}
/**
 * The MenuTrigger serves as a wrapper around a Menu and its associated trigger,
 * linking the Menu's open state with the trigger's press state.
 */ let $0a5a3ef3fb6afa6f$export$27d2ad3c5815583e = /*#__PURE__*/ $lJf9X$react.forwardRef($0a5a3ef3fb6afa6f$var$MenuTrigger);


var $e2bade5edf19d13a$exports = {};

$parcel$export($e2bade5edf19d13a$exports, "Menu", () => $e2bade5edf19d13a$export$d9b273488cd8ce6f);













function $ee23fb1f98cce92a$export$2ce376c2cc3355c8(props) {
    let { item: item , state: state , isVirtualized: isVirtualized , onAction: onAction  } = props;
    let { onClose: onClose , closeOnSelect: closeOnSelect  } = $185e8220149307d8$export$21c7ab35b39f78ec();
    let { rendered: rendered , key: key  } = item;
    let isSelected = state.selectionManager.isSelected(key);
    let isDisabled = state.disabledKeys.has(key);
    let ref = $lJf9X$react.useRef();
    let { menuItemProps: menuItemProps , labelProps: labelProps , descriptionProps: descriptionProps , keyboardShortcutProps: keyboardShortcutProps  } = $lJf9X$reactariamenu.useMenuItem({
        isSelected: isSelected,
        isDisabled: isDisabled,
        'aria-label': item['aria-label'],
        key: key,
        onClose: onClose,
        closeOnSelect: closeOnSelect,
        isVirtualized: isVirtualized,
        onAction: onAction
    }, state, ref);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $lJf9X$reactariainteractions.useHover({
        isDisabled: isDisabled
    });
    let contents = typeof rendered === 'string' ? /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($lJf9X$reactspectrumtext.Text, null, rendered) : rendered;
    return(/*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($lJf9X$reactariafocus.FocusRing, {
        focusRingClass: $lJf9X$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement("li", {
        ...$lJf9X$reactariautils.mergeProps(menuItemProps, hoverProps),
        ref: ref,
        className: $lJf9X$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'spectrum-Menu-item', {
            'is-disabled': isDisabled,
            'is-selected': isSelected,
            'is-selectable': state.selectionManager.selectionMode !== 'none',
            'is-hovered': isHovered
        })
    }, /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($lJf9X$reactspectrumlayout.Grid, {
        UNSAFE_className: $lJf9X$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'spectrum-Menu-itemGrid')
    }, /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($lJf9X$reactspectrumutils.ClearSlots, null, /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($lJf9X$reactspectrumutils.SlotProvider, {
        slots: {
            text: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports))['spectrum-Menu-itemLabel'],
                ...labelProps
            },
            end: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports))['spectrum-Menu-end'],
                ...descriptionProps
            },
            icon: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports))['spectrum-Menu-icon'],
                size: 'S'
            },
            description: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports))['spectrum-Menu-description'],
                ...descriptionProps
            },
            keyboard: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports))['spectrum-Menu-keyboard'],
                ...keyboardShortcutProps
            }
        }
    }, contents, isSelected && /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement(($parcel$interopDefault($lJf9X$spectrumiconsuiCheckmarkMedium)), {
        slot: "checkmark",
        UNSAFE_className: $lJf9X$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'spectrum-Menu-checkmark')
    })))))));
}








function $64fca97e79e27a6f$export$4b1545b4f2016d26(props) {
    let { item: item1 , state: state , onAction: onAction  } = props;
    let { itemProps: itemProps , headingProps: headingProps , groupProps: groupProps  } = $lJf9X$reactariamenu.useMenuSection({
        heading: item1.rendered,
        'aria-label': item1['aria-label']
    });
    let { separatorProps: separatorProps  } = $lJf9X$reactariaseparator.useSeparator({
        elementType: 'li'
    });
    return(/*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($lJf9X$react.Fragment, null, item1.key !== state.collection.getFirstKey() && /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement("li", {
        ...separatorProps,
        className: $lJf9X$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'spectrum-Menu-divider')
    }), /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement("li", itemProps, item1.rendered && /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement("span", {
        ...headingProps,
        className: $lJf9X$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'spectrum-Menu-sectionHeading')
    }, item1.rendered), /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement("ul", {
        ...groupProps,
        className: $lJf9X$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'spectrum-Menu')
    }, [
        ...item1.childNodes
    ].map((node)=>{
        let item = /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($ee23fb1f98cce92a$export$2ce376c2cc3355c8, {
            key: node.key,
            item: node,
            state: state,
            onAction: onAction
        });
        if (node.wrapper) item = node.wrapper(item);
        return item;
    })))));
}







function $e2bade5edf19d13a$var$Menu(props, ref) {
    let contextProps = $lJf9X$react.useContext($185e8220149307d8$export$c7e742effb1c51e2);
    let completeProps = {
        ...$lJf9X$reactariautils.mergeProps(contextProps, props)
    };
    let domRef = $lJf9X$reactspectrumutils.useDOMRef(ref);
    let state = $lJf9X$reactstatelytree.useTreeState(completeProps);
    let { menuProps: menuProps  } = $lJf9X$reactariamenu.useMenu(completeProps, state, domRef);
    let { styleProps: styleProps  } = $lJf9X$reactspectrumutils.useStyleProps(completeProps);
    $lJf9X$reactariautils.useSyncRef(contextProps, domRef);
    return(/*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement("ul", {
        ...menuProps,
        ...styleProps,
        ref: domRef,
        className: $lJf9X$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($d53361c88b80133d$exports)), 'spectrum-Menu', styleProps.className)
    }, [
        ...state.collection
    ].map((item)=>{
        if (item.type === 'section') return(/*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($64fca97e79e27a6f$export$4b1545b4f2016d26, {
            key: item.key,
            item: item,
            state: state,
            onAction: completeProps.onAction
        }));
        let menuItem = /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($ee23fb1f98cce92a$export$2ce376c2cc3355c8, {
            key: item.key,
            item: item,
            state: state,
            onAction: completeProps.onAction
        });
        if (item.wrapper) menuItem = item.wrapper(menuItem);
        return menuItem;
    })));
}
/**
 * Menus display a list of actions or options that a user can choose.
 */ // forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
const $e2bade5edf19d13a$export$d9b273488cd8ce6f = /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).forwardRef($e2bade5edf19d13a$var$Menu);


var $9a1dbc987be70e85$exports = {};

$parcel$export($9a1dbc987be70e85$exports, "ActionMenu", () => $9a1dbc987be70e85$export$ed57a210b9d13bb6);


var $48164cbfe2ce83bd$exports = {};
var $726a06c071f32234$exports = {};
$726a06c071f32234$exports = JSON.parse("{\"moreActions\":\"المزيد من الإجراءات\"}");


var $ecae60ae6ee84aa7$exports = {};
$ecae60ae6ee84aa7$exports = JSON.parse("{\"moreActions\":\"Повече действия\"}");


var $2933a59d444e7bbb$exports = {};
$2933a59d444e7bbb$exports = JSON.parse("{\"moreActions\":\"Další akce\"}");


var $95d3753a0ba6a121$exports = {};
$95d3753a0ba6a121$exports = JSON.parse("{\"moreActions\":\"Flere handlinger\"}");


var $159901cacf99199a$exports = {};
$159901cacf99199a$exports = JSON.parse("{\"moreActions\":\"Mehr Aktionen\"}");


var $3fbbac7ebbe14807$exports = {};
$3fbbac7ebbe14807$exports = JSON.parse("{\"moreActions\":\"Περισσότερες ενέργειες\"}");


var $9d5866f60add7d5c$exports = {};
$9d5866f60add7d5c$exports = JSON.parse("{\"moreActions\":\"More actions\"}");


var $f62647e860091599$exports = {};
$f62647e860091599$exports = JSON.parse("{\"moreActions\":\"Más acciones\"}");


var $bb3e9a56c2efea4a$exports = {};
$bb3e9a56c2efea4a$exports = JSON.parse("{\"moreActions\":\"Veel toiminguid\"}");


var $ce0657fe86b603d4$exports = {};
$ce0657fe86b603d4$exports = JSON.parse("{\"moreActions\":\"Lisää toimintoja\"}");


var $e831bc0a21a6f5ef$exports = {};
$e831bc0a21a6f5ef$exports = JSON.parse("{\"moreActions\":\"Autres actions\"}");


var $f7b31ab2aea5599c$exports = {};
$f7b31ab2aea5599c$exports = JSON.parse("{\"moreActions\":\"פעולות נוספות\"}");


var $b9f8417294bd9217$exports = {};
$b9f8417294bd9217$exports = JSON.parse("{\"moreActions\":\"Dodatne radnje\"}");


var $ae389e8071587137$exports = {};
$ae389e8071587137$exports = JSON.parse("{\"moreActions\":\"További lehetőségek\"}");


var $b3c437be83c5b43c$exports = {};
$b3c437be83c5b43c$exports = JSON.parse("{\"moreActions\":\"Altre azioni\"}");


var $03db1b783f145fc7$exports = {};
$03db1b783f145fc7$exports = JSON.parse("{\"moreActions\":\"その他のアクション\"}");


var $f11d8c43ea9f7ba1$exports = {};
$f11d8c43ea9f7ba1$exports = JSON.parse("{\"moreActions\":\"기타 작업\"}");


var $ef719e9091be7abe$exports = {};
$ef719e9091be7abe$exports = JSON.parse("{\"moreActions\":\"Daugiau veiksmų\"}");


var $be45afc245c4ac9f$exports = {};
$be45afc245c4ac9f$exports = JSON.parse("{\"moreActions\":\"Citas darbības\"}");


var $aecddea55c29f6ce$exports = {};
$aecddea55c29f6ce$exports = JSON.parse("{\"moreActions\":\"Flere handlinger\"}");


var $0bbc5349abf42763$exports = {};
$0bbc5349abf42763$exports = JSON.parse("{\"moreActions\":\"Meer handelingen\"}");


var $274b8a0ede7a4124$exports = {};
$274b8a0ede7a4124$exports = JSON.parse("{\"moreActions\":\"Więcej akcji\"}");


var $b6cde9639b90d25e$exports = {};
$b6cde9639b90d25e$exports = JSON.parse("{\"moreActions\":\"Mais ações\"}");


var $4f87283b5db19abe$exports = {};
$4f87283b5db19abe$exports = JSON.parse("{\"moreActions\":\"Mais ações\"}");


var $0999abe7beac7999$exports = {};
$0999abe7beac7999$exports = JSON.parse("{\"moreActions\":\"Mai multe acțiuni\"}");


var $ba473569082afa8d$exports = {};
$ba473569082afa8d$exports = JSON.parse("{\"moreActions\":\"Дополнительные действия\"}");


var $baefb2493240a5c0$exports = {};
$baefb2493240a5c0$exports = JSON.parse("{\"moreActions\":\"Ďalšie akcie\"}");


var $fb52009015f93f39$exports = {};
$fb52009015f93f39$exports = JSON.parse("{\"moreActions\":\"Več možnosti\"}");


var $c9b959ceb4b3c3a4$exports = {};
$c9b959ceb4b3c3a4$exports = JSON.parse("{\"moreActions\":\"Dodatne radnje\"}");


var $8598b018e9660aee$exports = {};
$8598b018e9660aee$exports = JSON.parse("{\"moreActions\":\"Fler åtgärder\"}");


var $7c48445f4eaf06f4$exports = {};
$7c48445f4eaf06f4$exports = JSON.parse("{\"moreActions\":\"Daha fazla eylem\"}");


var $4e27050ebf56cdd9$exports = {};
$4e27050ebf56cdd9$exports = JSON.parse("{\"moreActions\":\"Більше дій\"}");


var $06ea93cbb3142213$exports = {};
$06ea93cbb3142213$exports = JSON.parse("{\"moreActions\":\"更多操作\"}");


var $9c6a5108952aa436$exports = {};
$9c6a5108952aa436$exports = JSON.parse("{\"moreActions\":\"更多動作\"}");


$48164cbfe2ce83bd$exports = {
    "ar-AE": $726a06c071f32234$exports,
    "bg-BG": $ecae60ae6ee84aa7$exports,
    "cs-CZ": $2933a59d444e7bbb$exports,
    "da-DK": $95d3753a0ba6a121$exports,
    "de-DE": $159901cacf99199a$exports,
    "el-GR": $3fbbac7ebbe14807$exports,
    "en-US": $9d5866f60add7d5c$exports,
    "es-ES": $f62647e860091599$exports,
    "et-EE": $bb3e9a56c2efea4a$exports,
    "fi-FI": $ce0657fe86b603d4$exports,
    "fr-FR": $e831bc0a21a6f5ef$exports,
    "he-IL": $f7b31ab2aea5599c$exports,
    "hr-HR": $b9f8417294bd9217$exports,
    "hu-HU": $ae389e8071587137$exports,
    "it-IT": $b3c437be83c5b43c$exports,
    "ja-JP": $03db1b783f145fc7$exports,
    "ko-KR": $f11d8c43ea9f7ba1$exports,
    "lt-LT": $ef719e9091be7abe$exports,
    "lv-LV": $be45afc245c4ac9f$exports,
    "nb-NO": $aecddea55c29f6ce$exports,
    "nl-NL": $0bbc5349abf42763$exports,
    "pl-PL": $274b8a0ede7a4124$exports,
    "pt-BR": $b6cde9639b90d25e$exports,
    "pt-PT": $4f87283b5db19abe$exports,
    "ro-RO": $0999abe7beac7999$exports,
    "ru-RU": $ba473569082afa8d$exports,
    "sk-SK": $baefb2493240a5c0$exports,
    "sl-SI": $fb52009015f93f39$exports,
    "sr-SP": $c9b959ceb4b3c3a4$exports,
    "sv-SE": $8598b018e9660aee$exports,
    "tr-TR": $7c48445f4eaf06f4$exports,
    "uk-UA": $4e27050ebf56cdd9$exports,
    "zh-CN": $06ea93cbb3142213$exports,
    "zh-TW": $9c6a5108952aa436$exports
};








function $9a1dbc987be70e85$var$ActionMenu(props, ref) {
    props = $lJf9X$reactspectrumutils.useSlotProps(props, 'actionmenu');
    let formatMessage = $lJf9X$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($48164cbfe2ce83bd$exports)));
    let buttonProps = $lJf9X$reactariautils.filterDOMProps(props, {
        labelable: true
    });
    if (buttonProps['aria-label'] === undefined) buttonProps['aria-label'] = formatMessage('moreActions');
    return(/*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($0a5a3ef3fb6afa6f$export$27d2ad3c5815583e, {
        isOpen: props.isOpen,
        defaultOpen: props.defaultOpen,
        onOpenChange: props.onOpenChange,
        align: props.align,
        direction: props.direction,
        shouldFlip: props.shouldFlip
    }, /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($lJf9X$reactspectrumbutton.ActionButton, {
        ref: ref,
        ...props,
        ...buttonProps
    }, /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement(($parcel$interopDefault($lJf9X$spectrumiconsworkflowMore)), null)), /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).createElement($e2bade5edf19d13a$export$d9b273488cd8ce6f, {
        children: props.children,
        items: props.items,
        disabledKeys: props.disabledKeys,
        onAction: props.onAction
    })));
}
/**
 * ActionMenu combines an ActionButton with a Menu for simple "more actions" use cases.
 */ let $9a1dbc987be70e85$export$ed57a210b9d13bb6 = /*#__PURE__*/ ($parcel$interopDefault($lJf9X$react)).forwardRef($9a1dbc987be70e85$var$ActionMenu);



$parcel$exportWildcard(module.exports, $0a5a3ef3fb6afa6f$exports);
$parcel$exportWildcard(module.exports, $e2bade5edf19d13a$exports);
$parcel$exportWildcard(module.exports, $9a1dbc987be70e85$exports);
