// @react-spectrum/menu/dist/module.js
import "./main.css";
import {Item as $790472812827a49e$re_export$Item, Section as $790472812827a49e$re_export$Section} from "@react-stately/collections";
import {useDOMRef as $hphZ6$useDOMRef, useIsMobileDevice as $hphZ6$useIsMobileDevice, unwrapDOMRef as $hphZ6$unwrapDOMRef, classNames as $hphZ6$classNames, SlotProvider as $hphZ6$SlotProvider, useStyleProps as $hphZ6$useStyleProps, ClearSlots as $hphZ6$ClearSlots, useSlotProps as $hphZ6$useSlotProps} from "@react-spectrum/utils";
import {useOverlayPosition as $hphZ6$useOverlayPosition, DismissButton as $hphZ6$DismissButton} from "@react-aria/overlays";
import {FocusScope as $hphZ6$FocusScope, FocusRing as $hphZ6$FocusRing} from "@react-aria/focus";
import {Tray as $hphZ6$Tray, Popover as $hphZ6$Popover} from "@react-spectrum/overlays";
import {PressResponder as $hphZ6$PressResponder, useHover as $hphZ6$useHover} from "@react-aria/interactions";
import $hphZ6$react, {useRef as $hphZ6$useRef, Fragment as $hphZ6$Fragment, forwardRef as $hphZ6$forwardRef, useContext as $hphZ6$useContext} from "react";
import {useMenuTrigger as $hphZ6$useMenuTrigger, useMenu as $hphZ6$useMenu, useMenuItem as $hphZ6$useMenuItem, useMenuSection as $hphZ6$useMenuSection} from "@react-aria/menu";
import {useMenuTriggerState as $hphZ6$useMenuTriggerState} from "@react-stately/menu";
import {mergeProps as $hphZ6$mergeProps, useSyncRef as $hphZ6$useSyncRef, filterDOMProps as $hphZ6$filterDOMProps} from "@react-aria/utils";
import {useTreeState as $hphZ6$useTreeState} from "@react-stately/tree";
import $hphZ6$spectrumiconsuiCheckmarkMedium from "@spectrum-icons/ui/CheckmarkMedium";
import {Grid as $hphZ6$Grid} from "@react-spectrum/layout";
import {Text as $hphZ6$Text} from "@react-spectrum/text";
import {useSeparator as $hphZ6$useSeparator} from "@react-aria/separator";
import {ActionButton as $hphZ6$ActionButton} from "@react-spectrum/button";
import $hphZ6$spectrumiconsworkflowMore from "@spectrum-icons/workflow/More";
import {useMessageFormatter as $hphZ6$useMessageFormatter} from "@react-aria/i18n";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $170f90dc767586c6$exports = {};

$parcel$export($170f90dc767586c6$exports, "MenuTrigger", () => $170f90dc767586c6$export$27d2ad3c5815583e);




const $03d317390d176f75$export$c7e742effb1c51e2 = $hphZ6$react.createContext({
});
function $03d317390d176f75$export$21c7ab35b39f78ec() {
    return $hphZ6$useContext($03d317390d176f75$export$c7e742effb1c51e2);
}





var $2a670e3d20f05637$exports = {};

$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-popover", () => $2a670e3d20f05637$export$d959f30786125b, (v) => $2a670e3d20f05637$export$d959f30786125b = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu", () => $2a670e3d20f05637$export$fae6af6decd50ac5, (v) => $2a670e3d20f05637$export$fae6af6decd50ac5 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-sectionHeading", () => $2a670e3d20f05637$export$c20ea4b35020fbdd, (v) => $2a670e3d20f05637$export$c20ea4b35020fbdd = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-checkmark", () => $2a670e3d20f05637$export$8e15a726e4f7e62d, (v) => $2a670e3d20f05637$export$8e15a726e4f7e62d = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-item", () => $2a670e3d20f05637$export$45faad328e6c532b, (v) => $2a670e3d20f05637$export$45faad328e6c532b = v);
$parcel$export($2a670e3d20f05637$exports, "is-selected", () => $2a670e3d20f05637$export$1e0fb04f31d3c22a, (v) => $2a670e3d20f05637$export$1e0fb04f31d3c22a = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Icon", () => $2a670e3d20f05637$export$d374b04f30360026, (v) => $2a670e3d20f05637$export$d374b04f30360026 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-itemLabel", () => $2a670e3d20f05637$export$1aa6419c5c650567, (v) => $2a670e3d20f05637$export$1aa6419c5c650567 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-itemIcon", () => $2a670e3d20f05637$export$f431001e1bb54633, (v) => $2a670e3d20f05637$export$f431001e1bb54633 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-itemLabel--wrapping", () => $2a670e3d20f05637$export$465092164a8e1d98, (v) => $2a670e3d20f05637$export$465092164a8e1d98 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-divider", () => $2a670e3d20f05637$export$e892026f534efed8, (v) => $2a670e3d20f05637$export$e892026f534efed8 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-itemGrid", () => $2a670e3d20f05637$export$ee5f52784f57f9c8, (v) => $2a670e3d20f05637$export$ee5f52784f57f9c8 = v);
$parcel$export($2a670e3d20f05637$exports, "is-selectable", () => $2a670e3d20f05637$export$fbd22ba224473193, (v) => $2a670e3d20f05637$export$fbd22ba224473193 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-end", () => $2a670e3d20f05637$export$926a1bb74ed65667, (v) => $2a670e3d20f05637$export$926a1bb74ed65667 = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-icon", () => $2a670e3d20f05637$export$cbbae63ef9e44c7a, (v) => $2a670e3d20f05637$export$cbbae63ef9e44c7a = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-description", () => $2a670e3d20f05637$export$36f1275051e39ead, (v) => $2a670e3d20f05637$export$36f1275051e39ead = v);
$parcel$export($2a670e3d20f05637$exports, "spectrum-Menu-keyboard", () => $2a670e3d20f05637$export$4f9817a7dff6ba0f, (v) => $2a670e3d20f05637$export$4f9817a7dff6ba0f = v);
$parcel$export($2a670e3d20f05637$exports, "focus-ring", () => $2a670e3d20f05637$export$f39a09f249340e2a, (v) => $2a670e3d20f05637$export$f39a09f249340e2a = v);
$parcel$export($2a670e3d20f05637$exports, "is-focused", () => $2a670e3d20f05637$export$e7dc768d35940237, (v) => $2a670e3d20f05637$export$e7dc768d35940237 = v);
$parcel$export($2a670e3d20f05637$exports, "is-hovered", () => $2a670e3d20f05637$export$b8813cd5d7824ce7, (v) => $2a670e3d20f05637$export$b8813cd5d7824ce7 = v);
$parcel$export($2a670e3d20f05637$exports, "is-highlighted", () => $2a670e3d20f05637$export$451c613d0772eb99, (v) => $2a670e3d20f05637$export$451c613d0772eb99 = v);
$parcel$export($2a670e3d20f05637$exports, "is-open", () => $2a670e3d20f05637$export$a9781837241c946d, (v) => $2a670e3d20f05637$export$a9781837241c946d = v);
$parcel$export($2a670e3d20f05637$exports, "is-active", () => $2a670e3d20f05637$export$20fd0f7cd4e6112f, (v) => $2a670e3d20f05637$export$20fd0f7cd4e6112f = v);
$parcel$export($2a670e3d20f05637$exports, "is-disabled", () => $2a670e3d20f05637$export$d35bc1e505d1ebbf, (v) => $2a670e3d20f05637$export$d35bc1e505d1ebbf = v);
var $2a670e3d20f05637$export$d959f30786125b;
var $2a670e3d20f05637$export$fae6af6decd50ac5;
var $2a670e3d20f05637$export$c20ea4b35020fbdd;
var $2a670e3d20f05637$export$8e15a726e4f7e62d;
var $2a670e3d20f05637$export$45faad328e6c532b;
var $2a670e3d20f05637$export$1e0fb04f31d3c22a;
var $2a670e3d20f05637$export$d374b04f30360026;
var $2a670e3d20f05637$export$1aa6419c5c650567;
var $2a670e3d20f05637$export$f431001e1bb54633;
var $2a670e3d20f05637$export$465092164a8e1d98;
var $2a670e3d20f05637$export$e892026f534efed8;
var $2a670e3d20f05637$export$ee5f52784f57f9c8;
var $2a670e3d20f05637$export$fbd22ba224473193;
var $2a670e3d20f05637$export$926a1bb74ed65667;
var $2a670e3d20f05637$export$cbbae63ef9e44c7a;
var $2a670e3d20f05637$export$36f1275051e39ead;
var $2a670e3d20f05637$export$4f9817a7dff6ba0f;
var $2a670e3d20f05637$export$f39a09f249340e2a;
var $2a670e3d20f05637$export$e7dc768d35940237;
var $2a670e3d20f05637$export$b8813cd5d7824ce7;
var $2a670e3d20f05637$export$451c613d0772eb99;
var $2a670e3d20f05637$export$a9781837241c946d;
var $2a670e3d20f05637$export$20fd0f7cd4e6112f;
var $2a670e3d20f05637$export$d35bc1e505d1ebbf;
$2a670e3d20f05637$export$d959f30786125b = "spectrum-Menu-popover_7ea47c";
$2a670e3d20f05637$export$fae6af6decd50ac5 = "spectrum-Menu_7ea47c";
$2a670e3d20f05637$export$c20ea4b35020fbdd = "spectrum-Menu-sectionHeading_7ea47c";
$2a670e3d20f05637$export$8e15a726e4f7e62d = "spectrum-Menu-checkmark_7ea47c";
$2a670e3d20f05637$export$45faad328e6c532b = "spectrum-Menu-item_7ea47c";
$2a670e3d20f05637$export$1e0fb04f31d3c22a = "is-selected_7ea47c";
$2a670e3d20f05637$export$d374b04f30360026 = "spectrum-Icon_7ea47c";
$2a670e3d20f05637$export$1aa6419c5c650567 = "spectrum-Menu-itemLabel_7ea47c";
$2a670e3d20f05637$export$f431001e1bb54633 = "spectrum-Menu-itemIcon_7ea47c";
$2a670e3d20f05637$export$465092164a8e1d98 = "spectrum-Menu-itemLabel--wrapping_7ea47c";
$2a670e3d20f05637$export$e892026f534efed8 = "spectrum-Menu-divider_7ea47c";
$2a670e3d20f05637$export$ee5f52784f57f9c8 = "spectrum-Menu-itemGrid_7ea47c";
$2a670e3d20f05637$export$fbd22ba224473193 = "is-selectable_7ea47c";
$2a670e3d20f05637$export$926a1bb74ed65667 = "spectrum-Menu-end_7ea47c";
$2a670e3d20f05637$export$cbbae63ef9e44c7a = "spectrum-Menu-icon_7ea47c";
$2a670e3d20f05637$export$36f1275051e39ead = "spectrum-Menu-description_7ea47c";
$2a670e3d20f05637$export$4f9817a7dff6ba0f = "spectrum-Menu-keyboard_7ea47c";
$2a670e3d20f05637$export$f39a09f249340e2a = "focus-ring_7ea47c";
$2a670e3d20f05637$export$e7dc768d35940237 = "is-focused_7ea47c";
$2a670e3d20f05637$export$b8813cd5d7824ce7 = "is-hovered_7ea47c";
$2a670e3d20f05637$export$451c613d0772eb99 = "is-highlighted_7ea47c";
$2a670e3d20f05637$export$a9781837241c946d = "is-open_7ea47c";
$2a670e3d20f05637$export$20fd0f7cd4e6112f = "is-active_7ea47c";
$2a670e3d20f05637$export$d35bc1e505d1ebbf = "is-disabled_7ea47c";




function $170f90dc767586c6$var$MenuTrigger(props, ref) {
    let menuPopoverRef = $hphZ6$useRef();
    let triggerRef = $hphZ6$useRef();
    let domRef = $hphZ6$useDOMRef(ref);
    let menuTriggerRef = domRef || triggerRef;
    let menuRef = $hphZ6$useRef();
    let { children: children , align: align = 'start' , shouldFlip: shouldFlip = true , direction: direction = 'bottom' , closeOnSelect: closeOnSelect , trigger: trigger = 'press'  } = props;
    let [menuTrigger, menu] = $hphZ6$react.Children.toArray(children);
    let state = $hphZ6$useMenuTriggerState(props);
    let { menuTriggerProps: menuTriggerProps , menuProps: menuProps  } = $hphZ6$useMenuTrigger({
        trigger: trigger
    }, state, menuTriggerRef);
    let initialPlacement;
    switch(direction){
        case 'left':
        case 'right':
        case 'start':
        case 'end':
            initialPlacement = `${direction} ${align === 'end' ? 'bottom' : 'top'}`;
            break;
        case 'bottom':
        case 'top':
        default:
            initialPlacement = `${direction} ${align}`;
    }
    let isMobile = $hphZ6$useIsMobileDevice();
    let { overlayProps: positionProps , placement: placement  } = $hphZ6$useOverlayPosition({
        targetRef: menuTriggerRef,
        overlayRef: $hphZ6$unwrapDOMRef(menuPopoverRef),
        scrollRef: menuRef,
        placement: initialPlacement,
        shouldFlip: shouldFlip,
        isOpen: state.isOpen && !isMobile,
        onClose: state.close
    });
    let menuContext = {
        ...menuProps,
        ref: menuRef,
        onClose: state.close,
        closeOnSelect: closeOnSelect,
        autoFocus: state.focusStrategy || true,
        UNSAFE_style: isMobile ? {
            width: '100%',
            maxHeight: 'inherit'
        } : undefined,
        UNSAFE_className: $hphZ6$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), {
            'spectrum-Menu-popover': !isMobile
        })
    };
    let contents = /*#__PURE__*/ $hphZ6$react.createElement($hphZ6$FocusScope, {
        restoreFocus: true,
        contain: isMobile
    }, /*#__PURE__*/ $hphZ6$react.createElement($hphZ6$DismissButton, {
        onDismiss: state.close
    }), menu, /*#__PURE__*/ $hphZ6$react.createElement($hphZ6$DismissButton, {
        onDismiss: state.close
    }));
    // On small screen devices, the menu is rendered in a tray, otherwise a popover.
    let overlay;
    if (isMobile) overlay = /*#__PURE__*/ $hphZ6$react.createElement($hphZ6$Tray, {
        isOpen: state.isOpen,
        onClose: state.close
    }, contents);
    else overlay = /*#__PURE__*/ $hphZ6$react.createElement($hphZ6$Popover, {
        isOpen: state.isOpen,
        UNSAFE_style: positionProps.style,
        ref: menuPopoverRef,
        placement: placement,
        hideArrow: true,
        onClose: state.close,
        shouldCloseOnBlur: true
    }, contents);
    return(/*#__PURE__*/ $hphZ6$react.createElement($hphZ6$Fragment, null, /*#__PURE__*/ $hphZ6$react.createElement($hphZ6$SlotProvider, {
        slots: {
            actionButton: {
                holdAffordance: trigger === 'longPress'
            }
        }
    }, /*#__PURE__*/ $hphZ6$react.createElement($hphZ6$PressResponder, {
        ...menuTriggerProps,
        ref: menuTriggerRef,
        isPressed: state.isOpen
    }, menuTrigger)), /*#__PURE__*/ $hphZ6$react.createElement($03d317390d176f75$export$c7e742effb1c51e2.Provider, {
        value: menuContext
    }, overlay)));
}
/**
 * The MenuTrigger serves as a wrapper around a Menu and its associated trigger,
 * linking the Menu's open state with the trigger's press state.
 */ let $170f90dc767586c6$export$27d2ad3c5815583e = /*#__PURE__*/ $hphZ6$forwardRef($170f90dc767586c6$var$MenuTrigger);


var $b45af854eb24cf27$exports = {};

$parcel$export($b45af854eb24cf27$exports, "Menu", () => $b45af854eb24cf27$export$d9b273488cd8ce6f);













function $d250bb6f5beebd38$export$2ce376c2cc3355c8(props) {
    let { item: item , state: state , isVirtualized: isVirtualized , onAction: onAction  } = props;
    let { onClose: onClose , closeOnSelect: closeOnSelect  } = $03d317390d176f75$export$21c7ab35b39f78ec();
    let { rendered: rendered , key: key  } = item;
    let isSelected = state.selectionManager.isSelected(key);
    let isDisabled = state.disabledKeys.has(key);
    let ref = $hphZ6$useRef();
    let { menuItemProps: menuItemProps , labelProps: labelProps , descriptionProps: descriptionProps , keyboardShortcutProps: keyboardShortcutProps  } = $hphZ6$useMenuItem({
        isSelected: isSelected,
        isDisabled: isDisabled,
        'aria-label': item['aria-label'],
        key: key,
        onClose: onClose,
        closeOnSelect: closeOnSelect,
        isVirtualized: isVirtualized,
        onAction: onAction
    }, state, ref);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $hphZ6$useHover({
        isDisabled: isDisabled
    });
    let contents = typeof rendered === 'string' ? /*#__PURE__*/ $hphZ6$react.createElement($hphZ6$Text, null, rendered) : rendered;
    return(/*#__PURE__*/ $hphZ6$react.createElement($hphZ6$FocusRing, {
        focusRingClass: $hphZ6$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'focus-ring')
    }, /*#__PURE__*/ $hphZ6$react.createElement("li", {
        ...$hphZ6$mergeProps(menuItemProps, hoverProps),
        ref: ref,
        className: $hphZ6$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'spectrum-Menu-item', {
            'is-disabled': isDisabled,
            'is-selected': isSelected,
            'is-selectable': state.selectionManager.selectionMode !== 'none',
            'is-hovered': isHovered
        })
    }, /*#__PURE__*/ $hphZ6$react.createElement($hphZ6$Grid, {
        UNSAFE_className: $hphZ6$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'spectrum-Menu-itemGrid')
    }, /*#__PURE__*/ $hphZ6$react.createElement($hphZ6$ClearSlots, null, /*#__PURE__*/ $hphZ6$react.createElement($hphZ6$SlotProvider, {
        slots: {
            text: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports))['spectrum-Menu-itemLabel'],
                ...labelProps
            },
            end: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports))['spectrum-Menu-end'],
                ...descriptionProps
            },
            icon: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports))['spectrum-Menu-icon'],
                size: 'S'
            },
            description: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports))['spectrum-Menu-description'],
                ...descriptionProps
            },
            keyboard: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports))['spectrum-Menu-keyboard'],
                ...keyboardShortcutProps
            }
        }
    }, contents, isSelected && /*#__PURE__*/ $hphZ6$react.createElement($hphZ6$spectrumiconsuiCheckmarkMedium, {
        slot: "checkmark",
        UNSAFE_className: $hphZ6$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'spectrum-Menu-checkmark')
    })))))));
}








function $12c3830829151e87$export$4b1545b4f2016d26(props) {
    let { item: item1 , state: state , onAction: onAction  } = props;
    let { itemProps: itemProps , headingProps: headingProps , groupProps: groupProps  } = $hphZ6$useMenuSection({
        heading: item1.rendered,
        'aria-label': item1['aria-label']
    });
    let { separatorProps: separatorProps  } = $hphZ6$useSeparator({
        elementType: 'li'
    });
    return(/*#__PURE__*/ $hphZ6$react.createElement($hphZ6$Fragment, null, item1.key !== state.collection.getFirstKey() && /*#__PURE__*/ $hphZ6$react.createElement("li", {
        ...separatorProps,
        className: $hphZ6$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'spectrum-Menu-divider')
    }), /*#__PURE__*/ $hphZ6$react.createElement("li", itemProps, item1.rendered && /*#__PURE__*/ $hphZ6$react.createElement("span", {
        ...headingProps,
        className: $hphZ6$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'spectrum-Menu-sectionHeading')
    }, item1.rendered), /*#__PURE__*/ $hphZ6$react.createElement("ul", {
        ...groupProps,
        className: $hphZ6$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'spectrum-Menu')
    }, [
        ...item1.childNodes
    ].map((node)=>{
        let item = /*#__PURE__*/ $hphZ6$react.createElement($d250bb6f5beebd38$export$2ce376c2cc3355c8, {
            key: node.key,
            item: node,
            state: state,
            onAction: onAction
        });
        if (node.wrapper) item = node.wrapper(item);
        return item;
    })))));
}







function $b45af854eb24cf27$var$Menu(props, ref) {
    let contextProps = $hphZ6$useContext($03d317390d176f75$export$c7e742effb1c51e2);
    let completeProps = {
        ...$hphZ6$mergeProps(contextProps, props)
    };
    let domRef = $hphZ6$useDOMRef(ref);
    let state = $hphZ6$useTreeState(completeProps);
    let { menuProps: menuProps  } = $hphZ6$useMenu(completeProps, state, domRef);
    let { styleProps: styleProps  } = $hphZ6$useStyleProps(completeProps);
    $hphZ6$useSyncRef(contextProps, domRef);
    return(/*#__PURE__*/ $hphZ6$react.createElement("ul", {
        ...menuProps,
        ...styleProps,
        ref: domRef,
        className: $hphZ6$classNames((/*@__PURE__*/$parcel$interopDefault($2a670e3d20f05637$exports)), 'spectrum-Menu', styleProps.className)
    }, [
        ...state.collection
    ].map((item)=>{
        if (item.type === 'section') return(/*#__PURE__*/ $hphZ6$react.createElement($12c3830829151e87$export$4b1545b4f2016d26, {
            key: item.key,
            item: item,
            state: state,
            onAction: completeProps.onAction
        }));
        let menuItem = /*#__PURE__*/ $hphZ6$react.createElement($d250bb6f5beebd38$export$2ce376c2cc3355c8, {
            key: item.key,
            item: item,
            state: state,
            onAction: completeProps.onAction
        });
        if (item.wrapper) menuItem = item.wrapper(menuItem);
        return menuItem;
    })));
}
/**
 * Menus display a list of actions or options that a user can choose.
 */ // forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
const $b45af854eb24cf27$export$d9b273488cd8ce6f = /*#__PURE__*/ $hphZ6$react.forwardRef($b45af854eb24cf27$var$Menu);


var $8f8cfc365e73cc6c$exports = {};

$parcel$export($8f8cfc365e73cc6c$exports, "ActionMenu", () => $8f8cfc365e73cc6c$export$ed57a210b9d13bb6);


var $26a25f4b97c49a14$exports = {};
var $426212842a402207$exports = {};
$426212842a402207$exports = JSON.parse("{\"moreActions\":\"المزيد من الإجراءات\"}");


var $d08989437b736f18$exports = {};
$d08989437b736f18$exports = JSON.parse("{\"moreActions\":\"Повече действия\"}");


var $356dcb98202ddf3f$exports = {};
$356dcb98202ddf3f$exports = JSON.parse("{\"moreActions\":\"Další akce\"}");


var $879efb803d21ea12$exports = {};
$879efb803d21ea12$exports = JSON.parse("{\"moreActions\":\"Flere handlinger\"}");


var $d3ec50f14284484c$exports = {};
$d3ec50f14284484c$exports = JSON.parse("{\"moreActions\":\"Mehr Aktionen\"}");


var $c484d6c0ebc7d6d1$exports = {};
$c484d6c0ebc7d6d1$exports = JSON.parse("{\"moreActions\":\"Περισσότερες ενέργειες\"}");


var $2b006d57e8bcff4b$exports = {};
$2b006d57e8bcff4b$exports = JSON.parse("{\"moreActions\":\"More actions\"}");


var $93d0bf9fbed0999d$exports = {};
$93d0bf9fbed0999d$exports = JSON.parse("{\"moreActions\":\"Más acciones\"}");


var $a47c032f335f8486$exports = {};
$a47c032f335f8486$exports = JSON.parse("{\"moreActions\":\"Veel toiminguid\"}");


var $4cd40551f07f5b04$exports = {};
$4cd40551f07f5b04$exports = JSON.parse("{\"moreActions\":\"Lisää toimintoja\"}");


var $d4c865e9bcb71631$exports = {};
$d4c865e9bcb71631$exports = JSON.parse("{\"moreActions\":\"Autres actions\"}");


var $399d3b360d1f94d2$exports = {};
$399d3b360d1f94d2$exports = JSON.parse("{\"moreActions\":\"פעולות נוספות\"}");


var $6ffb924d46ce95db$exports = {};
$6ffb924d46ce95db$exports = JSON.parse("{\"moreActions\":\"Dodatne radnje\"}");


var $d08fd3789e6cb4e6$exports = {};
$d08fd3789e6cb4e6$exports = JSON.parse("{\"moreActions\":\"További lehetőségek\"}");


var $59d2fd52da9c738d$exports = {};
$59d2fd52da9c738d$exports = JSON.parse("{\"moreActions\":\"Altre azioni\"}");


var $9065ecb12e2f4025$exports = {};
$9065ecb12e2f4025$exports = JSON.parse("{\"moreActions\":\"その他のアクション\"}");


var $b39084b4281540d0$exports = {};
$b39084b4281540d0$exports = JSON.parse("{\"moreActions\":\"기타 작업\"}");


var $adb050a34b096a7b$exports = {};
$adb050a34b096a7b$exports = JSON.parse("{\"moreActions\":\"Daugiau veiksmų\"}");


var $a85366a96ab08e05$exports = {};
$a85366a96ab08e05$exports = JSON.parse("{\"moreActions\":\"Citas darbības\"}");


var $d8b1ea4c1578794a$exports = {};
$d8b1ea4c1578794a$exports = JSON.parse("{\"moreActions\":\"Flere handlinger\"}");


var $94999cd835525635$exports = {};
$94999cd835525635$exports = JSON.parse("{\"moreActions\":\"Meer handelingen\"}");


var $6e45402a14d3c873$exports = {};
$6e45402a14d3c873$exports = JSON.parse("{\"moreActions\":\"Więcej akcji\"}");


var $12cdf50b2d4d9e5b$exports = {};
$12cdf50b2d4d9e5b$exports = JSON.parse("{\"moreActions\":\"Mais ações\"}");


var $ef45c1b3b4623292$exports = {};
$ef45c1b3b4623292$exports = JSON.parse("{\"moreActions\":\"Mais ações\"}");


var $253dbbe0e1c3e4ad$exports = {};
$253dbbe0e1c3e4ad$exports = JSON.parse("{\"moreActions\":\"Mai multe acțiuni\"}");


var $3c58f66e34a71293$exports = {};
$3c58f66e34a71293$exports = JSON.parse("{\"moreActions\":\"Дополнительные действия\"}");


var $25af22fc5e5b693a$exports = {};
$25af22fc5e5b693a$exports = JSON.parse("{\"moreActions\":\"Ďalšie akcie\"}");


var $8d66a9f12915c16a$exports = {};
$8d66a9f12915c16a$exports = JSON.parse("{\"moreActions\":\"Več možnosti\"}");


var $eac7b0f370856164$exports = {};
$eac7b0f370856164$exports = JSON.parse("{\"moreActions\":\"Dodatne radnje\"}");


var $90f69b7bd813093a$exports = {};
$90f69b7bd813093a$exports = JSON.parse("{\"moreActions\":\"Fler åtgärder\"}");


var $0247f4c544e16816$exports = {};
$0247f4c544e16816$exports = JSON.parse("{\"moreActions\":\"Daha fazla eylem\"}");


var $94bf39621317a23f$exports = {};
$94bf39621317a23f$exports = JSON.parse("{\"moreActions\":\"Більше дій\"}");


var $f7dd6cc238786e06$exports = {};
$f7dd6cc238786e06$exports = JSON.parse("{\"moreActions\":\"更多操作\"}");


var $4f686012d2b3ba51$exports = {};
$4f686012d2b3ba51$exports = JSON.parse("{\"moreActions\":\"更多動作\"}");


$26a25f4b97c49a14$exports = {
    "ar-AE": $426212842a402207$exports,
    "bg-BG": $d08989437b736f18$exports,
    "cs-CZ": $356dcb98202ddf3f$exports,
    "da-DK": $879efb803d21ea12$exports,
    "de-DE": $d3ec50f14284484c$exports,
    "el-GR": $c484d6c0ebc7d6d1$exports,
    "en-US": $2b006d57e8bcff4b$exports,
    "es-ES": $93d0bf9fbed0999d$exports,
    "et-EE": $a47c032f335f8486$exports,
    "fi-FI": $4cd40551f07f5b04$exports,
    "fr-FR": $d4c865e9bcb71631$exports,
    "he-IL": $399d3b360d1f94d2$exports,
    "hr-HR": $6ffb924d46ce95db$exports,
    "hu-HU": $d08fd3789e6cb4e6$exports,
    "it-IT": $59d2fd52da9c738d$exports,
    "ja-JP": $9065ecb12e2f4025$exports,
    "ko-KR": $b39084b4281540d0$exports,
    "lt-LT": $adb050a34b096a7b$exports,
    "lv-LV": $a85366a96ab08e05$exports,
    "nb-NO": $d8b1ea4c1578794a$exports,
    "nl-NL": $94999cd835525635$exports,
    "pl-PL": $6e45402a14d3c873$exports,
    "pt-BR": $12cdf50b2d4d9e5b$exports,
    "pt-PT": $ef45c1b3b4623292$exports,
    "ro-RO": $253dbbe0e1c3e4ad$exports,
    "ru-RU": $3c58f66e34a71293$exports,
    "sk-SK": $25af22fc5e5b693a$exports,
    "sl-SI": $8d66a9f12915c16a$exports,
    "sr-SP": $eac7b0f370856164$exports,
    "sv-SE": $90f69b7bd813093a$exports,
    "tr-TR": $0247f4c544e16816$exports,
    "uk-UA": $94bf39621317a23f$exports,
    "zh-CN": $f7dd6cc238786e06$exports,
    "zh-TW": $4f686012d2b3ba51$exports
};








function $8f8cfc365e73cc6c$var$ActionMenu(props, ref) {
    props = $hphZ6$useSlotProps(props, 'actionmenu');
    let formatMessage = $hphZ6$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($26a25f4b97c49a14$exports)));
    let buttonProps = $hphZ6$filterDOMProps(props, {
        labelable: true
    });
    if (buttonProps['aria-label'] === undefined) buttonProps['aria-label'] = formatMessage('moreActions');
    return(/*#__PURE__*/ $hphZ6$react.createElement($170f90dc767586c6$export$27d2ad3c5815583e, {
        isOpen: props.isOpen,
        defaultOpen: props.defaultOpen,
        onOpenChange: props.onOpenChange,
        align: props.align,
        direction: props.direction,
        shouldFlip: props.shouldFlip
    }, /*#__PURE__*/ $hphZ6$react.createElement($hphZ6$ActionButton, {
        ref: ref,
        ...props,
        ...buttonProps
    }, /*#__PURE__*/ $hphZ6$react.createElement($hphZ6$spectrumiconsworkflowMore, null)), /*#__PURE__*/ $hphZ6$react.createElement($b45af854eb24cf27$export$d9b273488cd8ce6f, {
        children: props.children,
        items: props.items,
        disabledKeys: props.disabledKeys,
        onAction: props.onAction
    })));
}
/**
 * ActionMenu combines an ActionButton with a Menu for simple "more actions" use cases.
 */ let $8f8cfc365e73cc6c$export$ed57a210b9d13bb6 = /*#__PURE__*/ $hphZ6$react.forwardRef($8f8cfc365e73cc6c$var$ActionMenu);





export {$790472812827a49e$re_export$Item as Item, $790472812827a49e$re_export$Section as Section, $170f90dc767586c6$export$27d2ad3c5815583e as MenuTrigger, $b45af854eb24cf27$export$d9b273488cd8ce6f as Menu, $8f8cfc365e73cc6c$export$ed57a210b9d13bb6 as ActionMenu};
