// @react-spectrum/provider/dist/main.js
require("./main.css");
var $16ubb$reactspectrumutils = require("@react-spectrum/utils");
var $16ubb$clsx = require("clsx");
var $16ubb$reactariautils = require("@react-aria/utils");
var $16ubb$reactariai18n = require("@react-aria/i18n");
var $16ubb$reactariaoverlays = require("@react-aria/overlays");
var $16ubb$react = require("react");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $462ab2d1d29a0dad$exports = {};

$parcel$export($462ab2d1d29a0dad$exports, "useProvider", () => $462ab2d1d29a0dad$export$693cdb10cec23617);
$parcel$export($462ab2d1d29a0dad$exports, "Provider", () => $462ab2d1d29a0dad$export$2881499e37b75b9a);
$parcel$export($462ab2d1d29a0dad$exports, "useProviderProps", () => $462ab2d1d29a0dad$export$521c373ccc32c300);






var $c22adb08bea77d7d$exports = {};

$parcel$export($c22adb08bea77d7d$exports, "spectrum", () => $c22adb08bea77d7d$export$3311ab3a441bc141, (v) => $c22adb08bea77d7d$export$3311ab3a441bc141 = v);
var $c22adb08bea77d7d$export$3311ab3a441bc141;
$c22adb08bea77d7d$export$3311ab3a441bc141 = "spectrum_b37d53";


var $22af6b4a30f89e48$exports = {};

$parcel$export($22af6b4a30f89e48$exports, "spectrum", () => $22af6b4a30f89e48$export$3311ab3a441bc141, (v) => $22af6b4a30f89e48$export$3311ab3a441bc141 = v);
$parcel$export($22af6b4a30f89e48$exports, "spectrum-Body", () => $22af6b4a30f89e48$export$a30bf7810c8453d1, (v) => $22af6b4a30f89e48$export$a30bf7810c8453d1 = v);
$parcel$export($22af6b4a30f89e48$exports, "spectrum-Body--italic", () => $22af6b4a30f89e48$export$7b2dac1166f6ec4c, (v) => $22af6b4a30f89e48$export$7b2dac1166f6ec4c = v);
var $22af6b4a30f89e48$export$3311ab3a441bc141;
var $22af6b4a30f89e48$export$a30bf7810c8453d1;
var $22af6b4a30f89e48$export$7b2dac1166f6ec4c;
$22af6b4a30f89e48$export$3311ab3a441bc141 = "spectrum_2a241c";
$22af6b4a30f89e48$export$a30bf7810c8453d1 = "spectrum-Body_2a241c";
$22af6b4a30f89e48$export$7b2dac1166f6ec4c = "spectrum-Body--italic_2a241c";



function $6aece5ef0da74ae8$export$6343629ee1b29116(theme, defaultColorScheme) {
    let matchesDark = $16ubb$reactspectrumutils.useMediaQuery('(prefers-color-scheme: dark)');
    let matchesLight = $16ubb$reactspectrumutils.useMediaQuery('(prefers-color-scheme: light)');
    // importance OS > default > omitted
    if (theme.dark && matchesDark) return 'dark';
    if (theme.light && matchesLight) return 'light';
    if (theme.dark && defaultColorScheme === 'dark') return 'dark';
    if (theme.light && defaultColorScheme === 'light') return 'light';
    if (!theme.dark) return 'light';
    if (!theme.light) return 'dark';
    return 'light';
}
function $6aece5ef0da74ae8$export$a8d2043b2d807f4d(theme) {
    let matchesFine = $16ubb$reactspectrumutils.useMediaQuery('(any-pointer: fine)');
    if (matchesFine && theme.medium) return 'medium';
    if (theme.large) return 'large';
    return 'medium';
}


var $085c38a918b6013b$exports = {};
$085c38a918b6013b$exports = JSON.parse("{\"name\":\"@react-spectrum/provider\",\"version\":\"3.3.0\",\"description\":\"Spectrum UI components in React\",\"license\":\"Apache-2.0\",\"main\":\"dist/main.js\",\"module\":\"dist/module.js\",\"types\":\"dist/types.d.ts\",\"source\":\"src/index.ts\",\"files\":[\"dist\",\"src\"],\"sideEffects\":[\"*.css\"],\"targets\":{\"main\":{\"includeNodeModules\":[\"@adobe/spectrum-css-temp\"]},\"module\":{\"includeNodeModules\":[\"@adobe/spectrum-css-temp\"]}},\"repository\":{\"type\":\"git\",\"url\":\"https://github.com/adobe/react-spectrum\"},\"dependencies\":{\"@babel/runtime\":\"^7.6.2\",\"@react-aria/i18n\":\"^3.3.5\",\"@react-aria/overlays\":\"^3.7.4\",\"@react-aria/utils\":\"^3.11.1\",\"@react-spectrum/utils\":\"^3.6.4\",\"@react-types/provider\":\"^3.4.0\",\"@react-types/shared\":\"^3.11.0\",\"clsx\":\"^1.1.1\"},\"devDependencies\":{\"@adobe/spectrum-css-temp\":\"3.0.0-alpha.1\"},\"peerDependencies\":{\"react\":\"^16.8.0 || ^17.0.0-rc.1\",\"react-dom\":\"^16.8.0 || ^17.0.0-rc.1\"},\"publishConfig\":{\"access\":\"public\"}}");


const $462ab2d1d29a0dad$var$Context = /*#__PURE__*/ ($parcel$interopDefault($16ubb$react)).createContext(null);
$462ab2d1d29a0dad$var$Context.displayName = 'ProviderContext';
const $462ab2d1d29a0dad$var$DEFAULT_BREAKPOINTS = {
    S: 640,
    M: 768,
    L: 1024,
    XL: 1280,
    XXL: 1536
};
function $462ab2d1d29a0dad$var$Provider(props, ref) {
    let prevContext = $462ab2d1d29a0dad$export$693cdb10cec23617();
    let prevColorScheme = prevContext && prevContext.colorScheme;
    let prevBreakpoints = prevContext && prevContext.breakpoints;
    let { theme: theme = prevContext && prevContext.theme , defaultColorScheme: defaultColorScheme  } = props;
    // Hooks must always be called.
    let autoColorScheme = $6aece5ef0da74ae8$export$6343629ee1b29116(theme, defaultColorScheme);
    let autoScale = $6aece5ef0da74ae8$export$a8d2043b2d807f4d(theme);
    let { locale: prevLocale  } = $16ubb$reactariai18n.useLocale();
    // if the new theme doesn't support the prevColorScheme, we must resort to the auto
    let usePrevColorScheme = !!theme[prevColorScheme];
    // importance of color scheme props > parent > auto:(OS > default > omitted)
    let { colorScheme: colorScheme = usePrevColorScheme ? prevColorScheme : autoColorScheme , scale: scale = prevContext ? prevContext.scale : autoScale , locale: locale = prevContext ? prevLocale : null , breakpoints: breakpoints = prevContext ? prevBreakpoints : $462ab2d1d29a0dad$var$DEFAULT_BREAKPOINTS , children: children , isQuiet: isQuiet , isEmphasized: isEmphasized , isDisabled: isDisabled , isRequired: isRequired , isReadOnly: isReadOnly , validationState: validationState , ...otherProps } = props;
    // select only the props with values so undefined props don't overwrite prevContext values
    let currentProps = {
        version: $085c38a918b6013b$exports.version,
        theme: theme,
        breakpoints: breakpoints,
        colorScheme: colorScheme,
        scale: scale,
        isQuiet: isQuiet,
        isEmphasized: isEmphasized,
        isDisabled: isDisabled,
        isRequired: isRequired,
        isReadOnly: isReadOnly,
        validationState: validationState
    };
    let matchedBreakpoints = $16ubb$reactspectrumutils.useMatchedBreakpoints(breakpoints);
    let filteredProps = {
    };
    Object.entries(currentProps).forEach(([key, value])=>value !== undefined && (filteredProps[key] = value)
    );
    // Merge options with parent provider
    let context = Object.assign({
    }, prevContext, filteredProps);
    // Only wrap in a DOM node if the theme, colorScheme, or scale changed
    let contents = children;
    let domProps = $16ubb$reactariautils.filterDOMProps(otherProps);
    let { styleProps: styleProps  } = $16ubb$reactspectrumutils.useStyleProps(otherProps, undefined, {
        matchedBreakpoints: matchedBreakpoints
    });
    if (!prevContext || props.locale || theme !== prevContext.theme || colorScheme !== prevContext.colorScheme || scale !== prevContext.scale || Object.keys(domProps).length > 0 || otherProps.UNSAFE_className || Object.keys(styleProps.style).length > 0) contents = /*#__PURE__*/ ($parcel$interopDefault($16ubb$react)).createElement($462ab2d1d29a0dad$var$ProviderWrapper, {
        ...props,
        UNSAFE_style: {
            isolation: !prevContext ? 'isolate' : undefined,
            ...styleProps.style
        },
        ref: ref
    }, contents);
    return(/*#__PURE__*/ ($parcel$interopDefault($16ubb$react)).createElement($462ab2d1d29a0dad$var$Context.Provider, {
        value: context
    }, /*#__PURE__*/ ($parcel$interopDefault($16ubb$react)).createElement($16ubb$reactariai18n.I18nProvider, {
        locale: locale
    }, /*#__PURE__*/ ($parcel$interopDefault($16ubb$react)).createElement($16ubb$reactspectrumutils.BreakpointProvider, {
        matchedBreakpoints: matchedBreakpoints
    }, /*#__PURE__*/ ($parcel$interopDefault($16ubb$react)).createElement($16ubb$reactariaoverlays.ModalProvider, null, contents)))));
}
/**
 * Provider is the container for all React Spectrum applications.
 * It defines the theme, locale, and other application level settings,
 * and can also be used to provide common properties to a group of components.
 */ let $462ab2d1d29a0dad$export$2881499e37b75b9a = /*#__PURE__*/ ($parcel$interopDefault($16ubb$react)).forwardRef($462ab2d1d29a0dad$var$Provider);
const $462ab2d1d29a0dad$var$ProviderWrapper = /*#__PURE__*/ ($parcel$interopDefault($16ubb$react)).forwardRef(function ProviderWrapper(props, ref) {
    let { children: children , ...otherProps } = props;
    let { locale: locale , direction: direction  } = $16ubb$reactariai18n.useLocale();
    let { theme: theme , colorScheme: colorScheme , scale: scale  } = $462ab2d1d29a0dad$export$693cdb10cec23617();
    let { modalProviderProps: modalProviderProps  } = $16ubb$reactariaoverlays.useModalProvider();
    let { styleProps: styleProps  } = $16ubb$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $16ubb$reactspectrumutils.useDOMRef(ref);
    let themeKey = Object.keys(theme[colorScheme])[0];
    let scaleKey = Object.keys(theme[scale])[0];
    let className = ($parcel$interopDefault($16ubb$clsx))(styleProps.className, (/*@__PURE__*/$parcel$interopDefault($c22adb08bea77d7d$exports))['spectrum'], (/*@__PURE__*/$parcel$interopDefault($22af6b4a30f89e48$exports))['spectrum'], theme[colorScheme][themeKey], theme[scale][scaleKey], theme.global ? Object.values(theme.global) : null, {
        'react-spectrum-provider': $16ubb$reactspectrumutils.shouldKeepSpectrumClassNames,
        spectrum: $16ubb$reactspectrumutils.shouldKeepSpectrumClassNames,
        [themeKey]: $16ubb$reactspectrumutils.shouldKeepSpectrumClassNames,
        [scaleKey]: $16ubb$reactspectrumutils.shouldKeepSpectrumClassNames
    });
    var _colorScheme, ref1;
    let style = {
        ...styleProps.style,
        // This ensures that browser native UI like scrollbars are rendered in the right color scheme.
        // See https://web.dev/color-scheme/.
        colorScheme: (ref1 = (_colorScheme = props.colorScheme) !== null && _colorScheme !== void 0 ? _colorScheme : colorScheme) !== null && ref1 !== void 0 ? ref1 : Object.keys(theme).filter((k)=>k === 'light' || k === 'dark'
        ).join(' ')
    };
    let hasWarned = $16ubb$react.useRef(false);
    $16ubb$react.useEffect(()=>{
        if (direction && domRef.current) {
            let closestDir = domRef.current.parentElement.closest('[dir]');
            let dir = closestDir && closestDir.getAttribute('dir');
            if (dir && dir !== direction && !hasWarned.current) {
                console.warn(`Language directions cannot be nested. ${direction} inside ${dir}.`);
                hasWarned.current = true;
            }
        }
    }, [
        direction,
        domRef,
        hasWarned
    ]);
    return(/*#__PURE__*/ ($parcel$interopDefault($16ubb$react)).createElement("div", {
        ...$16ubb$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ...modalProviderProps,
        className: className,
        style: style,
        lang: locale,
        dir: direction,
        ref: domRef
    }, children));
});
function $462ab2d1d29a0dad$export$693cdb10cec23617() {
    return $16ubb$react.useContext($462ab2d1d29a0dad$var$Context);
}
function $462ab2d1d29a0dad$export$521c373ccc32c300(props) {
    let context = $462ab2d1d29a0dad$export$693cdb10cec23617();
    if (!context) return props;
    return Object.assign({
    }, {
        isQuiet: context.isQuiet,
        isEmphasized: context.isEmphasized,
        isDisabled: context.isDisabled,
        isRequired: context.isRequired,
        isReadOnly: context.isReadOnly,
        validationState: context.validationState
    }, props);
}


$parcel$exportWildcard(module.exports, $462ab2d1d29a0dad$exports);
