// @react-spectrum/provider/dist/module.js
import "./main.css";
import {useMatchedBreakpoints as $g69uj$useMatchedBreakpoints, useStyleProps as $g69uj$useStyleProps, BreakpointProvider as $g69uj$BreakpointProvider, useDOMRef as $g69uj$useDOMRef, shouldKeepSpectrumClassNames as $g69uj$shouldKeepSpectrumClassNames, useMediaQuery as $g69uj$useMediaQuery} from "@react-spectrum/utils";
import $g69uj$clsx from "clsx";
import {filterDOMProps as $g69uj$filterDOMProps} from "@react-aria/utils";
import {useLocale as $g69uj$useLocale, I18nProvider as $g69uj$I18nProvider} from "@react-aria/i18n";
import {ModalProvider as $g69uj$ModalProvider, useModalProvider as $g69uj$useModalProvider} from "@react-aria/overlays";
import $g69uj$react, {useRef as $g69uj$useRef, useEffect as $g69uj$useEffect, useContext as $g69uj$useContext} from "react";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $58cbe3eb1d813e08$exports = {};

$parcel$export($58cbe3eb1d813e08$exports, "useProvider", () => $58cbe3eb1d813e08$export$693cdb10cec23617);
$parcel$export($58cbe3eb1d813e08$exports, "Provider", () => $58cbe3eb1d813e08$export$2881499e37b75b9a);
$parcel$export($58cbe3eb1d813e08$exports, "useProviderProps", () => $58cbe3eb1d813e08$export$521c373ccc32c300);






var $eae14c1337c762cd$exports = {};

$parcel$export($eae14c1337c762cd$exports, "spectrum", () => $eae14c1337c762cd$export$3311ab3a441bc141, (v) => $eae14c1337c762cd$export$3311ab3a441bc141 = v);
var $eae14c1337c762cd$export$3311ab3a441bc141;
$eae14c1337c762cd$export$3311ab3a441bc141 = "spectrum_b37d53";


var $d17261c02cf551d7$exports = {};

$parcel$export($d17261c02cf551d7$exports, "spectrum", () => $d17261c02cf551d7$export$3311ab3a441bc141, (v) => $d17261c02cf551d7$export$3311ab3a441bc141 = v);
$parcel$export($d17261c02cf551d7$exports, "spectrum-Body", () => $d17261c02cf551d7$export$a30bf7810c8453d1, (v) => $d17261c02cf551d7$export$a30bf7810c8453d1 = v);
$parcel$export($d17261c02cf551d7$exports, "spectrum-Body--italic", () => $d17261c02cf551d7$export$7b2dac1166f6ec4c, (v) => $d17261c02cf551d7$export$7b2dac1166f6ec4c = v);
var $d17261c02cf551d7$export$3311ab3a441bc141;
var $d17261c02cf551d7$export$a30bf7810c8453d1;
var $d17261c02cf551d7$export$7b2dac1166f6ec4c;
$d17261c02cf551d7$export$3311ab3a441bc141 = "spectrum_2a241c";
$d17261c02cf551d7$export$a30bf7810c8453d1 = "spectrum-Body_2a241c";
$d17261c02cf551d7$export$7b2dac1166f6ec4c = "spectrum-Body--italic_2a241c";



function $c52cb170dc97a87d$export$6343629ee1b29116(theme, defaultColorScheme) {
    let matchesDark = $g69uj$useMediaQuery('(prefers-color-scheme: dark)');
    let matchesLight = $g69uj$useMediaQuery('(prefers-color-scheme: light)');
    // importance OS > default > omitted
    if (theme.dark && matchesDark) return 'dark';
    if (theme.light && matchesLight) return 'light';
    if (theme.dark && defaultColorScheme === 'dark') return 'dark';
    if (theme.light && defaultColorScheme === 'light') return 'light';
    if (!theme.dark) return 'light';
    if (!theme.light) return 'dark';
    return 'light';
}
function $c52cb170dc97a87d$export$a8d2043b2d807f4d(theme) {
    let matchesFine = $g69uj$useMediaQuery('(any-pointer: fine)');
    if (matchesFine && theme.medium) return 'medium';
    if (theme.large) return 'large';
    return 'medium';
}


var $170f096dc832ad09$exports = {};
$170f096dc832ad09$exports = JSON.parse("{\"name\":\"@react-spectrum/provider\",\"version\":\"3.3.0\",\"description\":\"Spectrum UI components in React\",\"license\":\"Apache-2.0\",\"main\":\"dist/main.js\",\"module\":\"dist/module.js\",\"types\":\"dist/types.d.ts\",\"source\":\"src/index.ts\",\"files\":[\"dist\",\"src\"],\"sideEffects\":[\"*.css\"],\"targets\":{\"main\":{\"includeNodeModules\":[\"@adobe/spectrum-css-temp\"]},\"module\":{\"includeNodeModules\":[\"@adobe/spectrum-css-temp\"]}},\"repository\":{\"type\":\"git\",\"url\":\"https://github.com/adobe/react-spectrum\"},\"dependencies\":{\"@babel/runtime\":\"^7.6.2\",\"@react-aria/i18n\":\"^3.3.5\",\"@react-aria/overlays\":\"^3.7.4\",\"@react-aria/utils\":\"^3.11.1\",\"@react-spectrum/utils\":\"^3.6.4\",\"@react-types/provider\":\"^3.4.0\",\"@react-types/shared\":\"^3.11.0\",\"clsx\":\"^1.1.1\"},\"devDependencies\":{\"@adobe/spectrum-css-temp\":\"3.0.0-alpha.1\"},\"peerDependencies\":{\"react\":\"^16.8.0 || ^17.0.0-rc.1\",\"react-dom\":\"^16.8.0 || ^17.0.0-rc.1\"},\"publishConfig\":{\"access\":\"public\"}}");


const $58cbe3eb1d813e08$var$Context = /*#__PURE__*/ $g69uj$react.createContext(null);
$58cbe3eb1d813e08$var$Context.displayName = 'ProviderContext';
const $58cbe3eb1d813e08$var$DEFAULT_BREAKPOINTS = {
    S: 640,
    M: 768,
    L: 1024,
    XL: 1280,
    XXL: 1536
};
function $58cbe3eb1d813e08$var$Provider(props, ref) {
    let prevContext = $58cbe3eb1d813e08$export$693cdb10cec23617();
    let prevColorScheme = prevContext && prevContext.colorScheme;
    let prevBreakpoints = prevContext && prevContext.breakpoints;
    let { theme: theme = prevContext && prevContext.theme , defaultColorScheme: defaultColorScheme  } = props;
    // Hooks must always be called.
    let autoColorScheme = $c52cb170dc97a87d$export$6343629ee1b29116(theme, defaultColorScheme);
    let autoScale = $c52cb170dc97a87d$export$a8d2043b2d807f4d(theme);
    let { locale: prevLocale  } = $g69uj$useLocale();
    // if the new theme doesn't support the prevColorScheme, we must resort to the auto
    let usePrevColorScheme = !!theme[prevColorScheme];
    // importance of color scheme props > parent > auto:(OS > default > omitted)
    let { colorScheme: colorScheme = usePrevColorScheme ? prevColorScheme : autoColorScheme , scale: scale = prevContext ? prevContext.scale : autoScale , locale: locale = prevContext ? prevLocale : null , breakpoints: breakpoints = prevContext ? prevBreakpoints : $58cbe3eb1d813e08$var$DEFAULT_BREAKPOINTS , children: children , isQuiet: isQuiet , isEmphasized: isEmphasized , isDisabled: isDisabled , isRequired: isRequired , isReadOnly: isReadOnly , validationState: validationState , ...otherProps } = props;
    // select only the props with values so undefined props don't overwrite prevContext values
    let currentProps = {
        version: $170f096dc832ad09$exports.version,
        theme: theme,
        breakpoints: breakpoints,
        colorScheme: colorScheme,
        scale: scale,
        isQuiet: isQuiet,
        isEmphasized: isEmphasized,
        isDisabled: isDisabled,
        isRequired: isRequired,
        isReadOnly: isReadOnly,
        validationState: validationState
    };
    let matchedBreakpoints = $g69uj$useMatchedBreakpoints(breakpoints);
    let filteredProps = {
    };
    Object.entries(currentProps).forEach(([key, value])=>value !== undefined && (filteredProps[key] = value)
    );
    // Merge options with parent provider
    let context = Object.assign({
    }, prevContext, filteredProps);
    // Only wrap in a DOM node if the theme, colorScheme, or scale changed
    let contents = children;
    let domProps = $g69uj$filterDOMProps(otherProps);
    let { styleProps: styleProps  } = $g69uj$useStyleProps(otherProps, undefined, {
        matchedBreakpoints: matchedBreakpoints
    });
    if (!prevContext || props.locale || theme !== prevContext.theme || colorScheme !== prevContext.colorScheme || scale !== prevContext.scale || Object.keys(domProps).length > 0 || otherProps.UNSAFE_className || Object.keys(styleProps.style).length > 0) contents = /*#__PURE__*/ $g69uj$react.createElement($58cbe3eb1d813e08$var$ProviderWrapper, {
        ...props,
        UNSAFE_style: {
            isolation: !prevContext ? 'isolate' : undefined,
            ...styleProps.style
        },
        ref: ref
    }, contents);
    return(/*#__PURE__*/ $g69uj$react.createElement($58cbe3eb1d813e08$var$Context.Provider, {
        value: context
    }, /*#__PURE__*/ $g69uj$react.createElement($g69uj$I18nProvider, {
        locale: locale
    }, /*#__PURE__*/ $g69uj$react.createElement($g69uj$BreakpointProvider, {
        matchedBreakpoints: matchedBreakpoints
    }, /*#__PURE__*/ $g69uj$react.createElement($g69uj$ModalProvider, null, contents)))));
}
/**
 * Provider is the container for all React Spectrum applications.
 * It defines the theme, locale, and other application level settings,
 * and can also be used to provide common properties to a group of components.
 */ let $58cbe3eb1d813e08$export$2881499e37b75b9a = /*#__PURE__*/ $g69uj$react.forwardRef($58cbe3eb1d813e08$var$Provider);
const $58cbe3eb1d813e08$var$ProviderWrapper = /*#__PURE__*/ $g69uj$react.forwardRef(function ProviderWrapper(props, ref) {
    let { children: children , ...otherProps } = props;
    let { locale: locale , direction: direction  } = $g69uj$useLocale();
    let { theme: theme , colorScheme: colorScheme , scale: scale  } = $58cbe3eb1d813e08$export$693cdb10cec23617();
    let { modalProviderProps: modalProviderProps  } = $g69uj$useModalProvider();
    let { styleProps: styleProps  } = $g69uj$useStyleProps(otherProps);
    let domRef = $g69uj$useDOMRef(ref);
    let themeKey = Object.keys(theme[colorScheme])[0];
    let scaleKey = Object.keys(theme[scale])[0];
    let className = $g69uj$clsx(styleProps.className, (/*@__PURE__*/$parcel$interopDefault($eae14c1337c762cd$exports))['spectrum'], (/*@__PURE__*/$parcel$interopDefault($d17261c02cf551d7$exports))['spectrum'], theme[colorScheme][themeKey], theme[scale][scaleKey], theme.global ? Object.values(theme.global) : null, {
        'react-spectrum-provider': $g69uj$shouldKeepSpectrumClassNames,
        spectrum: $g69uj$shouldKeepSpectrumClassNames,
        [themeKey]: $g69uj$shouldKeepSpectrumClassNames,
        [scaleKey]: $g69uj$shouldKeepSpectrumClassNames
    });
    var _colorScheme, ref1;
    let style = {
        ...styleProps.style,
        // This ensures that browser native UI like scrollbars are rendered in the right color scheme.
        // See https://web.dev/color-scheme/.
        colorScheme: (ref1 = (_colorScheme = props.colorScheme) !== null && _colorScheme !== void 0 ? _colorScheme : colorScheme) !== null && ref1 !== void 0 ? ref1 : Object.keys(theme).filter((k)=>k === 'light' || k === 'dark'
        ).join(' ')
    };
    let hasWarned = $g69uj$useRef(false);
    $g69uj$useEffect(()=>{
        if (direction && domRef.current) {
            let closestDir = domRef.current.parentElement.closest('[dir]');
            let dir = closestDir && closestDir.getAttribute('dir');
            if (dir && dir !== direction && !hasWarned.current) {
                console.warn(`Language directions cannot be nested. ${direction} inside ${dir}.`);
                hasWarned.current = true;
            }
        }
    }, [
        direction,
        domRef,
        hasWarned
    ]);
    return(/*#__PURE__*/ $g69uj$react.createElement("div", {
        ...$g69uj$filterDOMProps(otherProps),
        ...styleProps,
        ...modalProviderProps,
        className: className,
        style: style,
        lang: locale,
        dir: direction,
        ref: domRef
    }, children));
});
function $58cbe3eb1d813e08$export$693cdb10cec23617() {
    return $g69uj$useContext($58cbe3eb1d813e08$var$Context);
}
function $58cbe3eb1d813e08$export$521c373ccc32c300(props) {
    let context = $58cbe3eb1d813e08$export$693cdb10cec23617();
    if (!context) return props;
    return Object.assign({
    }, {
        isQuiet: context.isQuiet,
        isEmphasized: context.isEmphasized,
        isDisabled: context.isDisabled,
        isRequired: context.isRequired,
        isReadOnly: context.isReadOnly,
        validationState: context.validationState
    }, props);
}




export {$58cbe3eb1d813e08$export$693cdb10cec23617 as useProvider, $58cbe3eb1d813e08$export$2881499e37b75b9a as Provider, $58cbe3eb1d813e08$export$521c373ccc32c300 as useProviderProps};
