// @react-spectrum/view/dist/main.js
var $fPAGM$reactspectrumutils = require("@react-spectrum/utils");
var $fPAGM$reactariautils = require("@react-aria/utils");
var $fPAGM$react = require("react");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $2c69fdec3406267c$exports = {};

$parcel$export($2c69fdec3406267c$exports, "View", () => $2c69fdec3406267c$export$27a5bd065ad55220);



function $2c69fdec3406267c$var$View(props, ref) {
    props = $fPAGM$reactspectrumutils.useSlotProps(props);
    let { elementType: ElementType = 'div' , children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $fPAGM$reactspectrumutils.useStyleProps(props, $fPAGM$reactspectrumutils.viewStyleProps);
    let domRef = $fPAGM$reactspectrumutils.useDOMRef(ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($fPAGM$react)).createElement(ElementType, {
        ...$fPAGM$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ ($parcel$interopDefault($fPAGM$react)).createElement($fPAGM$reactspectrumutils.ClearSlots, null, children)));
}
/**
 * View is a general purpose container with no specific semantics that can be used for custom styling purposes.
 * It supports Spectrum style props to ensure consistency with other Spectrum components.
 */ const $2c69fdec3406267c$export$27a5bd065ad55220 = /*#__PURE__*/ $fPAGM$react.forwardRef($2c69fdec3406267c$var$View);


var $733f1a63a8c61e18$exports = {};

$parcel$export($733f1a63a8c61e18$exports, "Content", () => $733f1a63a8c61e18$export$7c6e2c02157bb7d2);




function $733f1a63a8c61e18$var$Content(props, ref) {
    props = $fPAGM$reactspectrumutils.useSlotProps(props, 'content');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $fPAGM$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $fPAGM$reactspectrumutils.useDOMRef(ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($fPAGM$react)).createElement("section", {
        ...$fPAGM$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ ($parcel$interopDefault($fPAGM$react)).createElement($fPAGM$reactspectrumutils.ClearSlots, null, children)));
}
/**
 * Content represents the primary content within a Spectrum container.
 */ const $733f1a63a8c61e18$export$7c6e2c02157bb7d2 = /*#__PURE__*/ $fPAGM$react.forwardRef($733f1a63a8c61e18$var$Content);


var $c97cfd39d4a20a59$exports = {};

$parcel$export($c97cfd39d4a20a59$exports, "Footer", () => $c97cfd39d4a20a59$export$a06f1c675e846f6f);



function $c97cfd39d4a20a59$var$Footer(props, ref) {
    props = $fPAGM$reactspectrumutils.useSlotProps(props, 'footer');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $fPAGM$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $fPAGM$reactspectrumutils.useDOMRef(ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($fPAGM$react)).createElement("footer", {
        ...$fPAGM$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ ($parcel$interopDefault($fPAGM$react)).createElement($fPAGM$reactspectrumutils.ClearSlots, null, children)));
}
/**
 * Footer represents a footer within a Spectrum container.
 */ const $c97cfd39d4a20a59$export$a06f1c675e846f6f = /*#__PURE__*/ $fPAGM$react.forwardRef($c97cfd39d4a20a59$var$Footer);


var $ab06cf38b8fa4da7$exports = {};

$parcel$export($ab06cf38b8fa4da7$exports, "Header", () => $ab06cf38b8fa4da7$export$8b251419efc915eb);



function $ab06cf38b8fa4da7$var$Header(props, ref) {
    props = $fPAGM$reactspectrumutils.useSlotProps(props, 'header');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $fPAGM$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $fPAGM$reactspectrumutils.useDOMRef(ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($fPAGM$react)).createElement("header", {
        ...$fPAGM$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ ($parcel$interopDefault($fPAGM$react)).createElement($fPAGM$reactspectrumutils.ClearSlots, null, children)));
}
/**
 * Header represents a header within a Spectrum container.
 */ const $ab06cf38b8fa4da7$export$8b251419efc915eb = /*#__PURE__*/ $fPAGM$react.forwardRef($ab06cf38b8fa4da7$var$Header);


$parcel$exportWildcard(module.exports, $2c69fdec3406267c$exports);
$parcel$exportWildcard(module.exports, $733f1a63a8c61e18$exports);
$parcel$exportWildcard(module.exports, $c97cfd39d4a20a59$exports);
$parcel$exportWildcard(module.exports, $ab06cf38b8fa4da7$exports);
