// @react-spectrum/view/dist/module.js
import {useSlotProps as $5c7up$useSlotProps, useStyleProps as $5c7up$useStyleProps, viewStyleProps as $5c7up$viewStyleProps, useDOMRef as $5c7up$useDOMRef, ClearSlots as $5c7up$ClearSlots} from "@react-spectrum/utils";
import {filterDOMProps as $5c7up$filterDOMProps} from "@react-aria/utils";
import $5c7up$react, {forwardRef as $5c7up$forwardRef} from "react";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $a18b6277175c703a$exports = {};

$parcel$export($a18b6277175c703a$exports, "View", () => $a18b6277175c703a$export$27a5bd065ad55220);



function $a18b6277175c703a$var$View(props, ref) {
    props = $5c7up$useSlotProps(props);
    let { elementType: ElementType = 'div' , children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $5c7up$useStyleProps(props, $5c7up$viewStyleProps);
    let domRef = $5c7up$useDOMRef(ref);
    return(/*#__PURE__*/ $5c7up$react.createElement(ElementType, {
        ...$5c7up$filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ $5c7up$react.createElement($5c7up$ClearSlots, null, children)));
}
/**
 * View is a general purpose container with no specific semantics that can be used for custom styling purposes.
 * It supports Spectrum style props to ensure consistency with other Spectrum components.
 */ const $a18b6277175c703a$export$27a5bd065ad55220 = /*#__PURE__*/ $5c7up$forwardRef($a18b6277175c703a$var$View);


var $0f56b7cc4c3950b8$exports = {};

$parcel$export($0f56b7cc4c3950b8$exports, "Content", () => $0f56b7cc4c3950b8$export$7c6e2c02157bb7d2);




function $0f56b7cc4c3950b8$var$Content(props, ref) {
    props = $5c7up$useSlotProps(props, 'content');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $5c7up$useStyleProps(otherProps);
    let domRef = $5c7up$useDOMRef(ref);
    return(/*#__PURE__*/ $5c7up$react.createElement("section", {
        ...$5c7up$filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ $5c7up$react.createElement($5c7up$ClearSlots, null, children)));
}
/**
 * Content represents the primary content within a Spectrum container.
 */ const $0f56b7cc4c3950b8$export$7c6e2c02157bb7d2 = /*#__PURE__*/ $5c7up$forwardRef($0f56b7cc4c3950b8$var$Content);


var $82b2dd3d9c6b1686$exports = {};

$parcel$export($82b2dd3d9c6b1686$exports, "Footer", () => $82b2dd3d9c6b1686$export$a06f1c675e846f6f);



function $82b2dd3d9c6b1686$var$Footer(props, ref) {
    props = $5c7up$useSlotProps(props, 'footer');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $5c7up$useStyleProps(otherProps);
    let domRef = $5c7up$useDOMRef(ref);
    return(/*#__PURE__*/ $5c7up$react.createElement("footer", {
        ...$5c7up$filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ $5c7up$react.createElement($5c7up$ClearSlots, null, children)));
}
/**
 * Footer represents a footer within a Spectrum container.
 */ const $82b2dd3d9c6b1686$export$a06f1c675e846f6f = /*#__PURE__*/ $5c7up$forwardRef($82b2dd3d9c6b1686$var$Footer);


var $f0ba792aac11eeb7$exports = {};

$parcel$export($f0ba792aac11eeb7$exports, "Header", () => $f0ba792aac11eeb7$export$8b251419efc915eb);



function $f0ba792aac11eeb7$var$Header(props, ref) {
    props = $5c7up$useSlotProps(props, 'header');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $5c7up$useStyleProps(otherProps);
    let domRef = $5c7up$useDOMRef(ref);
    return(/*#__PURE__*/ $5c7up$react.createElement("header", {
        ...$5c7up$filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ $5c7up$react.createElement($5c7up$ClearSlots, null, children)));
}
/**
 * Header represents a header within a Spectrum container.
 */ const $f0ba792aac11eeb7$export$8b251419efc915eb = /*#__PURE__*/ $5c7up$forwardRef($f0ba792aac11eeb7$var$Header);




export {$a18b6277175c703a$export$27a5bd065ad55220 as View, $0f56b7cc4c3950b8$export$7c6e2c02157bb7d2 as Content, $82b2dd3d9c6b1686$export$a06f1c675e846f6f as Footer, $f0ba792aac11eeb7$export$8b251419efc915eb as Header};
