// @react-spectrum/divider/dist/main.js
require("./main.css");
var $3FgJE$reactspectrumutils = require("@react-spectrum/utils");
var $3FgJE$react = require("react");
var $3FgJE$reactariaseparator = require("@react-aria/separator");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $30abbf2bb1fbe6c4$exports = {};

$parcel$export($30abbf2bb1fbe6c4$exports, "Divider", () => $30abbf2bb1fbe6c4$export$2e0a83ec2e27ecbb);


var $286c1bcc35fc04bc$exports = {};

$parcel$export($286c1bcc35fc04bc$exports, "spectrum-Rule", () => $286c1bcc35fc04bc$export$9ebf9cdaed6878bc, (v) => $286c1bcc35fc04bc$export$9ebf9cdaed6878bc = v);
$parcel$export($286c1bcc35fc04bc$exports, "spectrum-Rule--large", () => $286c1bcc35fc04bc$export$243baa0ec4ab58cc, (v) => $286c1bcc35fc04bc$export$243baa0ec4ab58cc = v);
$parcel$export($286c1bcc35fc04bc$exports, "spectrum-Rule--medium", () => $286c1bcc35fc04bc$export$a8d73b58ca350eb9, (v) => $286c1bcc35fc04bc$export$a8d73b58ca350eb9 = v);
$parcel$export($286c1bcc35fc04bc$exports, "spectrum-Rule--small", () => $286c1bcc35fc04bc$export$b4b727880cfe37d, (v) => $286c1bcc35fc04bc$export$b4b727880cfe37d = v);
$parcel$export($286c1bcc35fc04bc$exports, "spectrum-Rule--horizontal", () => $286c1bcc35fc04bc$export$563c1b24466c05b8, (v) => $286c1bcc35fc04bc$export$563c1b24466c05b8 = v);
$parcel$export($286c1bcc35fc04bc$exports, "spectrum-Rule--vertical", () => $286c1bcc35fc04bc$export$76f93463a8fc15a8, (v) => $286c1bcc35fc04bc$export$76f93463a8fc15a8 = v);
var $286c1bcc35fc04bc$export$9ebf9cdaed6878bc;
var $286c1bcc35fc04bc$export$243baa0ec4ab58cc;
var $286c1bcc35fc04bc$export$a8d73b58ca350eb9;
var $286c1bcc35fc04bc$export$b4b727880cfe37d;
var $286c1bcc35fc04bc$export$563c1b24466c05b8;
var $286c1bcc35fc04bc$export$76f93463a8fc15a8;
$286c1bcc35fc04bc$export$9ebf9cdaed6878bc = "spectrum-Rule_612bb8";
$286c1bcc35fc04bc$export$243baa0ec4ab58cc = "spectrum-Rule--large_612bb8";
$286c1bcc35fc04bc$export$a8d73b58ca350eb9 = "spectrum-Rule--medium_612bb8";
$286c1bcc35fc04bc$export$b4b727880cfe37d = "spectrum-Rule--small_612bb8";
$286c1bcc35fc04bc$export$563c1b24466c05b8 = "spectrum-Rule--horizontal_612bb8";
$286c1bcc35fc04bc$export$76f93463a8fc15a8 = "spectrum-Rule--vertical_612bb8";



let $30abbf2bb1fbe6c4$var$sizeMap = {
    S: 'small',
    M: 'medium',
    L: 'large'
};
function $30abbf2bb1fbe6c4$var$Divider(props, ref) {
    props = $3FgJE$reactspectrumutils.useSlotProps(props, 'divider');
    let { size: size = 'L' , orientation: orientation = 'horizontal' , ...otherProps } = props;
    let domRef = $3FgJE$reactspectrumutils.useDOMRef(ref);
    let { styleProps: styleProps  } = $3FgJE$reactspectrumutils.useStyleProps(otherProps);
    let weight = $30abbf2bb1fbe6c4$var$sizeMap[size];
    let Element = 'hr';
    if (orientation === 'vertical') Element = 'div';
    let { separatorProps: separatorProps  } = $3FgJE$reactariaseparator.useSeparator({
        ...props,
        elementType: Element
    });
    return(/*#__PURE__*/ ($parcel$interopDefault($3FgJE$react)).createElement(Element, {
        ...styleProps,
        className: $3FgJE$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($286c1bcc35fc04bc$exports)), 'spectrum-Rule', `spectrum-Rule--${weight}`, {
            'spectrum-Rule--vertical': orientation === 'vertical',
            'spectrum-Rule--horizontal': orientation === 'horizontal'
        }, styleProps.className),
        // @ts-ignore https://github.com/Microsoft/TypeScript/issues/28892
        ref: domRef,
        ...separatorProps
    }));
}
/**
 * Dividers bring clarity to a layout by grouping and dividing content in close proximity.
 * They can also be used to establish rhythm and hierarchy.
 */ let $30abbf2bb1fbe6c4$export$2e0a83ec2e27ecbb = /*#__PURE__*/ ($parcel$interopDefault($3FgJE$react)).forwardRef($30abbf2bb1fbe6c4$var$Divider);


$parcel$exportWildcard(module.exports, $30abbf2bb1fbe6c4$exports);
