// @react-spectrum/divider/dist/module.js
import "./main.css";
import {useSlotProps as $iMhIx$useSlotProps, useDOMRef as $iMhIx$useDOMRef, useStyleProps as $iMhIx$useStyleProps, classNames as $iMhIx$classNames} from "@react-spectrum/utils";
import $iMhIx$react from "react";
import {useSeparator as $iMhIx$useSeparator} from "@react-aria/separator";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $26836dbb9a290939$exports = {};

$parcel$export($26836dbb9a290939$exports, "Divider", () => $26836dbb9a290939$export$2e0a83ec2e27ecbb);


var $cee1b1036c239ff2$exports = {};

$parcel$export($cee1b1036c239ff2$exports, "spectrum-Rule", () => $cee1b1036c239ff2$export$9ebf9cdaed6878bc, (v) => $cee1b1036c239ff2$export$9ebf9cdaed6878bc = v);
$parcel$export($cee1b1036c239ff2$exports, "spectrum-Rule--large", () => $cee1b1036c239ff2$export$243baa0ec4ab58cc, (v) => $cee1b1036c239ff2$export$243baa0ec4ab58cc = v);
$parcel$export($cee1b1036c239ff2$exports, "spectrum-Rule--medium", () => $cee1b1036c239ff2$export$a8d73b58ca350eb9, (v) => $cee1b1036c239ff2$export$a8d73b58ca350eb9 = v);
$parcel$export($cee1b1036c239ff2$exports, "spectrum-Rule--small", () => $cee1b1036c239ff2$export$b4b727880cfe37d, (v) => $cee1b1036c239ff2$export$b4b727880cfe37d = v);
$parcel$export($cee1b1036c239ff2$exports, "spectrum-Rule--horizontal", () => $cee1b1036c239ff2$export$563c1b24466c05b8, (v) => $cee1b1036c239ff2$export$563c1b24466c05b8 = v);
$parcel$export($cee1b1036c239ff2$exports, "spectrum-Rule--vertical", () => $cee1b1036c239ff2$export$76f93463a8fc15a8, (v) => $cee1b1036c239ff2$export$76f93463a8fc15a8 = v);
var $cee1b1036c239ff2$export$9ebf9cdaed6878bc;
var $cee1b1036c239ff2$export$243baa0ec4ab58cc;
var $cee1b1036c239ff2$export$a8d73b58ca350eb9;
var $cee1b1036c239ff2$export$b4b727880cfe37d;
var $cee1b1036c239ff2$export$563c1b24466c05b8;
var $cee1b1036c239ff2$export$76f93463a8fc15a8;
$cee1b1036c239ff2$export$9ebf9cdaed6878bc = "spectrum-Rule_612bb8";
$cee1b1036c239ff2$export$243baa0ec4ab58cc = "spectrum-Rule--large_612bb8";
$cee1b1036c239ff2$export$a8d73b58ca350eb9 = "spectrum-Rule--medium_612bb8";
$cee1b1036c239ff2$export$b4b727880cfe37d = "spectrum-Rule--small_612bb8";
$cee1b1036c239ff2$export$563c1b24466c05b8 = "spectrum-Rule--horizontal_612bb8";
$cee1b1036c239ff2$export$76f93463a8fc15a8 = "spectrum-Rule--vertical_612bb8";



let $26836dbb9a290939$var$sizeMap = {
    S: 'small',
    M: 'medium',
    L: 'large'
};
function $26836dbb9a290939$var$Divider(props, ref) {
    props = $iMhIx$useSlotProps(props, 'divider');
    let { size: size = 'L' , orientation: orientation = 'horizontal' , ...otherProps } = props;
    let domRef = $iMhIx$useDOMRef(ref);
    let { styleProps: styleProps  } = $iMhIx$useStyleProps(otherProps);
    let weight = $26836dbb9a290939$var$sizeMap[size];
    let Element = 'hr';
    if (orientation === 'vertical') Element = 'div';
    let { separatorProps: separatorProps  } = $iMhIx$useSeparator({
        ...props,
        elementType: Element
    });
    return(/*#__PURE__*/ $iMhIx$react.createElement(Element, {
        ...styleProps,
        className: $iMhIx$classNames((/*@__PURE__*/$parcel$interopDefault($cee1b1036c239ff2$exports)), 'spectrum-Rule', `spectrum-Rule--${weight}`, {
            'spectrum-Rule--vertical': orientation === 'vertical',
            'spectrum-Rule--horizontal': orientation === 'horizontal'
        }, styleProps.className),
        // @ts-ignore https://github.com/Microsoft/TypeScript/issues/28892
        ref: domRef,
        ...separatorProps
    }));
}
/**
 * Dividers bring clarity to a layout by grouping and dividing content in close proximity.
 * They can also be used to establish rhythm and hierarchy.
 */ let $26836dbb9a290939$export$2e0a83ec2e27ecbb = /*#__PURE__*/ $iMhIx$react.forwardRef($26836dbb9a290939$var$Divider);




export {$26836dbb9a290939$export$2e0a83ec2e27ecbb as Divider};
