// @react-spectrum/illustratedmessage/dist/main.js
require("./main.css");
var $7Lfy3$reactspectrumutils = require("@react-spectrum/utils");
var $7Lfy3$reactariautils = require("@react-aria/utils");
var $7Lfy3$reactspectrumlayout = require("@react-spectrum/layout");
var $7Lfy3$react = require("react");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $05168ed4c51edc9a$exports = {};

$parcel$export($05168ed4c51edc9a$exports, "IllustratedMessage", () => $05168ed4c51edc9a$export$406dbc84c317ece0);




var $cf90b57028be8e07$exports = {};

$parcel$export($cf90b57028be8e07$exports, "spectrum-IllustratedMessage", () => $cf90b57028be8e07$export$e22c6753c239ec31, (v) => $cf90b57028be8e07$export$e22c6753c239ec31 = v);
$parcel$export($cf90b57028be8e07$exports, "spectrum-IllustratedMessage-heading", () => $cf90b57028be8e07$export$5df23faa5ed344c4, (v) => $cf90b57028be8e07$export$5df23faa5ed344c4 = v);
$parcel$export($cf90b57028be8e07$exports, "spectrum-IllustratedMessage-description", () => $cf90b57028be8e07$export$cf9d959111789fe7, (v) => $cf90b57028be8e07$export$cf9d959111789fe7 = v);
$parcel$export($cf90b57028be8e07$exports, "spectrum-IllustratedMessage--cta", () => $cf90b57028be8e07$export$29744d89c9a76d50, (v) => $cf90b57028be8e07$export$29744d89c9a76d50 = v);
var $cf90b57028be8e07$export$e22c6753c239ec31;
var $cf90b57028be8e07$export$5df23faa5ed344c4;
var $cf90b57028be8e07$export$cf9d959111789fe7;
var $cf90b57028be8e07$export$29744d89c9a76d50;
$cf90b57028be8e07$export$e22c6753c239ec31 = "spectrum-IllustratedMessage_02bd98";
$cf90b57028be8e07$export$5df23faa5ed344c4 = "spectrum-IllustratedMessage-heading_02bd98";
$cf90b57028be8e07$export$cf9d959111789fe7 = "spectrum-IllustratedMessage-description_02bd98";
$cf90b57028be8e07$export$29744d89c9a76d50 = "spectrum-IllustratedMessage--cta_02bd98";


function $05168ed4c51edc9a$var$IllustratedMessage(props, ref) {
    props = $7Lfy3$reactspectrumutils.useSlotProps(props, 'illustration');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $7Lfy3$reactspectrumutils.useStyleProps(otherProps);
    let headingClassName = $7Lfy3$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($cf90b57028be8e07$exports)), 'spectrum-IllustratedMessage-heading');
    let contentClassName = $7Lfy3$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($cf90b57028be8e07$exports)), 'spectrum-IllustratedMessage-description');
    let slots = {
        heading: {
            UNSAFE_className: headingClassName
        },
        content: {
            UNSAFE_className: contentClassName
        }
    };
    return(/*#__PURE__*/ ($parcel$interopDefault($7Lfy3$react)).createElement($7Lfy3$reactspectrumlayout.Flex, {
        ...$7Lfy3$reactariautils.filterDOMProps(otherProps),
        UNSAFE_style: styleProps.style,
        isHidden: styleProps.hidden,
        UNSAFE_className: $7Lfy3$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($cf90b57028be8e07$exports)), 'spectrum-IllustratedMessage', styleProps.className),
        ref: ref
    }, /*#__PURE__*/ ($parcel$interopDefault($7Lfy3$react)).createElement($7Lfy3$reactspectrumutils.SlotProvider, {
        slots: slots
    }, children)));
}
/**
 * An IllustratedMessage displays an illustration and a message, usually
 * for an empty state or an error page.
 */ let $05168ed4c51edc9a$export$406dbc84c317ece0 = /*#__PURE__*/ $7Lfy3$react.forwardRef($05168ed4c51edc9a$var$IllustratedMessage);


$parcel$exportWildcard(module.exports, $05168ed4c51edc9a$exports);
