// @react-spectrum/illustratedmessage/dist/module.js
import "./main.css";
import {useSlotProps as $jxkDN$useSlotProps, useStyleProps as $jxkDN$useStyleProps, classNames as $jxkDN$classNames, SlotProvider as $jxkDN$SlotProvider} from "@react-spectrum/utils";
import {filterDOMProps as $jxkDN$filterDOMProps} from "@react-aria/utils";
import {Flex as $jxkDN$Flex} from "@react-spectrum/layout";
import $jxkDN$react, {forwardRef as $jxkDN$forwardRef} from "react";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $cb79048b4081bd76$exports = {};

$parcel$export($cb79048b4081bd76$exports, "IllustratedMessage", () => $cb79048b4081bd76$export$406dbc84c317ece0);




var $8469598f5dd71cc1$exports = {};

$parcel$export($8469598f5dd71cc1$exports, "spectrum-IllustratedMessage", () => $8469598f5dd71cc1$export$e22c6753c239ec31, (v) => $8469598f5dd71cc1$export$e22c6753c239ec31 = v);
$parcel$export($8469598f5dd71cc1$exports, "spectrum-IllustratedMessage-heading", () => $8469598f5dd71cc1$export$5df23faa5ed344c4, (v) => $8469598f5dd71cc1$export$5df23faa5ed344c4 = v);
$parcel$export($8469598f5dd71cc1$exports, "spectrum-IllustratedMessage-description", () => $8469598f5dd71cc1$export$cf9d959111789fe7, (v) => $8469598f5dd71cc1$export$cf9d959111789fe7 = v);
$parcel$export($8469598f5dd71cc1$exports, "spectrum-IllustratedMessage--cta", () => $8469598f5dd71cc1$export$29744d89c9a76d50, (v) => $8469598f5dd71cc1$export$29744d89c9a76d50 = v);
var $8469598f5dd71cc1$export$e22c6753c239ec31;
var $8469598f5dd71cc1$export$5df23faa5ed344c4;
var $8469598f5dd71cc1$export$cf9d959111789fe7;
var $8469598f5dd71cc1$export$29744d89c9a76d50;
$8469598f5dd71cc1$export$e22c6753c239ec31 = "spectrum-IllustratedMessage_02bd98";
$8469598f5dd71cc1$export$5df23faa5ed344c4 = "spectrum-IllustratedMessage-heading_02bd98";
$8469598f5dd71cc1$export$cf9d959111789fe7 = "spectrum-IllustratedMessage-description_02bd98";
$8469598f5dd71cc1$export$29744d89c9a76d50 = "spectrum-IllustratedMessage--cta_02bd98";


function $cb79048b4081bd76$var$IllustratedMessage(props, ref) {
    props = $jxkDN$useSlotProps(props, 'illustration');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $jxkDN$useStyleProps(otherProps);
    let headingClassName = $jxkDN$classNames((/*@__PURE__*/$parcel$interopDefault($8469598f5dd71cc1$exports)), 'spectrum-IllustratedMessage-heading');
    let contentClassName = $jxkDN$classNames((/*@__PURE__*/$parcel$interopDefault($8469598f5dd71cc1$exports)), 'spectrum-IllustratedMessage-description');
    let slots = {
        heading: {
            UNSAFE_className: headingClassName
        },
        content: {
            UNSAFE_className: contentClassName
        }
    };
    return(/*#__PURE__*/ $jxkDN$react.createElement($jxkDN$Flex, {
        ...$jxkDN$filterDOMProps(otherProps),
        UNSAFE_style: styleProps.style,
        isHidden: styleProps.hidden,
        UNSAFE_className: $jxkDN$classNames((/*@__PURE__*/$parcel$interopDefault($8469598f5dd71cc1$exports)), 'spectrum-IllustratedMessage', styleProps.className),
        ref: ref
    }, /*#__PURE__*/ $jxkDN$react.createElement($jxkDN$SlotProvider, {
        slots: slots
    }, children)));
}
/**
 * An IllustratedMessage displays an illustration and a message, usually
 * for an empty state or an error page.
 */ let $cb79048b4081bd76$export$406dbc84c317ece0 = /*#__PURE__*/ $jxkDN$forwardRef($cb79048b4081bd76$var$IllustratedMessage);




export {$cb79048b4081bd76$export$406dbc84c317ece0 as IllustratedMessage};
