// @react-spectrum/buttongroup/dist/main.js
require("./main.css");
var $asYiO$reactspectrumutils = require("@react-spectrum/utils");
var $asYiO$reactariautils = require("@react-aria/utils");
var $asYiO$reactspectrumprovider = require("@react-spectrum/provider");
var $asYiO$react = require("react");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $5eef4b35ad265768$exports = {};

$parcel$export($5eef4b35ad265768$exports, "ButtonGroup", () => $5eef4b35ad265768$export$69b1032f2ecdf404);




var $9600c79dc6dc775a$exports = {};

$parcel$export($9600c79dc6dc775a$exports, "spectrum-ButtonGroup", () => $9600c79dc6dc775a$export$a92faf51352ac608, (v) => $9600c79dc6dc775a$export$a92faf51352ac608 = v);
$parcel$export($9600c79dc6dc775a$exports, "spectrum-ButtonGroup-Button", () => $9600c79dc6dc775a$export$7184de2337c1927f, (v) => $9600c79dc6dc775a$export$7184de2337c1927f = v);
$parcel$export($9600c79dc6dc775a$exports, "spectrum-ButtonGroup--alignEnd", () => $9600c79dc6dc775a$export$30e96c53d7beca11, (v) => $9600c79dc6dc775a$export$30e96c53d7beca11 = v);
$parcel$export($9600c79dc6dc775a$exports, "spectrum-ButtonGroup--alignCenter", () => $9600c79dc6dc775a$export$252bc7c4ae05757, (v) => $9600c79dc6dc775a$export$252bc7c4ae05757 = v);
$parcel$export($9600c79dc6dc775a$exports, "spectrum-ButtonGroup--vertical", () => $9600c79dc6dc775a$export$84d9c0e1917a4eaf, (v) => $9600c79dc6dc775a$export$84d9c0e1917a4eaf = v);
var $9600c79dc6dc775a$export$a92faf51352ac608;
var $9600c79dc6dc775a$export$7184de2337c1927f;
var $9600c79dc6dc775a$export$30e96c53d7beca11;
var $9600c79dc6dc775a$export$252bc7c4ae05757;
var $9600c79dc6dc775a$export$84d9c0e1917a4eaf;
$9600c79dc6dc775a$export$a92faf51352ac608 = "spectrum-ButtonGroup_b10c2f";
$9600c79dc6dc775a$export$7184de2337c1927f = "spectrum-ButtonGroup-Button_b10c2f";
$9600c79dc6dc775a$export$30e96c53d7beca11 = "spectrum-ButtonGroup--alignEnd_b10c2f";
$9600c79dc6dc775a$export$252bc7c4ae05757 = "spectrum-ButtonGroup--alignCenter_b10c2f";
$9600c79dc6dc775a$export$84d9c0e1917a4eaf = "spectrum-ButtonGroup--vertical_b10c2f";


function $5eef4b35ad265768$var$ButtonGroup(props, ref) {
    let { scale: scale  } = $asYiO$reactspectrumprovider.useProvider();
    props = $asYiO$reactspectrumprovider.useProviderProps(props);
    props = $asYiO$reactspectrumutils.useSlotProps(props, 'buttonGroup');
    let { children: children , orientation: orientation = 'horizontal' , isDisabled: isDisabled , align: align = 'start' , ...otherProps } = props;
    let { styleProps: styleProps  } = $asYiO$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $asYiO$reactspectrumutils.useDOMRef(ref);
    let [hasOverflow, setHasOverflow] = $asYiO$reactariautils.useValueEffect(false);
    let checkForOverflow = $asYiO$react.useCallback(()=>{
        let computeHasOverflow = ()=>{
            if (domRef.current && orientation === 'horizontal') {
                let buttonGroupChildren = Array.from(domRef.current.children);
                let maxX = domRef.current.offsetWidth + 1; // + 1 to account for rounding errors
                // If any buttons have negative X positions (align="end") or extend beyond
                // the width of the button group (align="start"), then switch to vertical.
                if (buttonGroupChildren.some((child)=>child.offsetLeft < 0 || child.offsetLeft + child.offsetWidth > maxX
                )) return true;
                return false;
            }
        };
        if (orientation === 'horizontal') setHasOverflow(function*() {
            // Force to horizontal for measurement.
            yield false;
            // Measure, and update if there is overflow.
            yield computeHasOverflow();
        });
    }, [
        domRef,
        orientation,
        scale,
        setHasOverflow,
        children
    ]);
    // There are two main reasons we need to remeasure:
    // 1. Internal changes: Check for initial overflow or when orientation/scale/children change (from checkForOverflow dep array)
    $asYiO$react.useEffect(()=>{
        checkForOverflow();
    }, [
        checkForOverflow
    ]);
    // 2. External changes: buttongroup won't change size due to any parents changing size, so listen to its container for size changes to figure out if we should remeasure
    let parent = $asYiO$react.useRef();
    $asYiO$react.useEffect(()=>{
        parent.current = domRef.current.parentElement;
    }, [
        domRef.current
    ]);
    $asYiO$reactspectrumutils.useResizeObserver({
        ref: parent,
        onResize: checkForOverflow
    });
    return(/*#__PURE__*/ ($parcel$interopDefault($asYiO$react)).createElement("div", {
        ...$asYiO$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef,
        className: $asYiO$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($9600c79dc6dc775a$exports)), 'spectrum-ButtonGroup', {
            'spectrum-ButtonGroup--vertical': orientation === 'vertical' || hasOverflow,
            'spectrum-ButtonGroup--alignEnd': align === 'end',
            'spectrum-ButtonGroup--alignCenter': align === 'center'
        }, styleProps.className)
    }, /*#__PURE__*/ ($parcel$interopDefault($asYiO$react)).createElement($asYiO$reactspectrumutils.SlotProvider, {
        slots: {
            button: {
                UNSAFE_className: $asYiO$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($9600c79dc6dc775a$exports)), 'spectrum-ButtonGroup-Button')
            }
        }
    }, /*#__PURE__*/ ($parcel$interopDefault($asYiO$react)).createElement($asYiO$reactspectrumprovider.Provider, {
        isDisabled: isDisabled
    }, children))));
}
/**
 * ButtonGroup handles overflow for a grouping of buttons whose actions are related to each other.
 */ let $5eef4b35ad265768$export$69b1032f2ecdf404 = /*#__PURE__*/ ($parcel$interopDefault($asYiO$react)).forwardRef($5eef4b35ad265768$var$ButtonGroup);


$parcel$exportWildcard(module.exports, $5eef4b35ad265768$exports);
