// @react-spectrum/buttongroup/dist/module.js
import "./main.css";
import {useSlotProps as $dqQfL$useSlotProps, useStyleProps as $dqQfL$useStyleProps, useDOMRef as $dqQfL$useDOMRef, useResizeObserver as $dqQfL$useResizeObserver, classNames as $dqQfL$classNames, SlotProvider as $dqQfL$SlotProvider} from "@react-spectrum/utils";
import {useValueEffect as $dqQfL$useValueEffect, filterDOMProps as $dqQfL$filterDOMProps} from "@react-aria/utils";
import {useProvider as $dqQfL$useProvider, useProviderProps as $dqQfL$useProviderProps, Provider as $dqQfL$Provider} from "@react-spectrum/provider";
import $dqQfL$react, {useCallback as $dqQfL$useCallback, useEffect as $dqQfL$useEffect, useRef as $dqQfL$useRef} from "react";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $5e39a78e785c6a30$exports = {};

$parcel$export($5e39a78e785c6a30$exports, "ButtonGroup", () => $5e39a78e785c6a30$export$69b1032f2ecdf404);




var $78cc0793b2d6df26$exports = {};

$parcel$export($78cc0793b2d6df26$exports, "spectrum-ButtonGroup", () => $78cc0793b2d6df26$export$a92faf51352ac608, (v) => $78cc0793b2d6df26$export$a92faf51352ac608 = v);
$parcel$export($78cc0793b2d6df26$exports, "spectrum-ButtonGroup-Button", () => $78cc0793b2d6df26$export$7184de2337c1927f, (v) => $78cc0793b2d6df26$export$7184de2337c1927f = v);
$parcel$export($78cc0793b2d6df26$exports, "spectrum-ButtonGroup--alignEnd", () => $78cc0793b2d6df26$export$30e96c53d7beca11, (v) => $78cc0793b2d6df26$export$30e96c53d7beca11 = v);
$parcel$export($78cc0793b2d6df26$exports, "spectrum-ButtonGroup--alignCenter", () => $78cc0793b2d6df26$export$252bc7c4ae05757, (v) => $78cc0793b2d6df26$export$252bc7c4ae05757 = v);
$parcel$export($78cc0793b2d6df26$exports, "spectrum-ButtonGroup--vertical", () => $78cc0793b2d6df26$export$84d9c0e1917a4eaf, (v) => $78cc0793b2d6df26$export$84d9c0e1917a4eaf = v);
var $78cc0793b2d6df26$export$a92faf51352ac608;
var $78cc0793b2d6df26$export$7184de2337c1927f;
var $78cc0793b2d6df26$export$30e96c53d7beca11;
var $78cc0793b2d6df26$export$252bc7c4ae05757;
var $78cc0793b2d6df26$export$84d9c0e1917a4eaf;
$78cc0793b2d6df26$export$a92faf51352ac608 = "spectrum-ButtonGroup_b10c2f";
$78cc0793b2d6df26$export$7184de2337c1927f = "spectrum-ButtonGroup-Button_b10c2f";
$78cc0793b2d6df26$export$30e96c53d7beca11 = "spectrum-ButtonGroup--alignEnd_b10c2f";
$78cc0793b2d6df26$export$252bc7c4ae05757 = "spectrum-ButtonGroup--alignCenter_b10c2f";
$78cc0793b2d6df26$export$84d9c0e1917a4eaf = "spectrum-ButtonGroup--vertical_b10c2f";


function $5e39a78e785c6a30$var$ButtonGroup(props, ref) {
    let { scale: scale  } = $dqQfL$useProvider();
    props = $dqQfL$useProviderProps(props);
    props = $dqQfL$useSlotProps(props, 'buttonGroup');
    let { children: children , orientation: orientation = 'horizontal' , isDisabled: isDisabled , align: align = 'start' , ...otherProps } = props;
    let { styleProps: styleProps  } = $dqQfL$useStyleProps(otherProps);
    let domRef = $dqQfL$useDOMRef(ref);
    let [hasOverflow, setHasOverflow] = $dqQfL$useValueEffect(false);
    let checkForOverflow = $dqQfL$useCallback(()=>{
        let computeHasOverflow = ()=>{
            if (domRef.current && orientation === 'horizontal') {
                let buttonGroupChildren = Array.from(domRef.current.children);
                let maxX = domRef.current.offsetWidth + 1; // + 1 to account for rounding errors
                // If any buttons have negative X positions (align="end") or extend beyond
                // the width of the button group (align="start"), then switch to vertical.
                if (buttonGroupChildren.some((child)=>child.offsetLeft < 0 || child.offsetLeft + child.offsetWidth > maxX
                )) return true;
                return false;
            }
        };
        if (orientation === 'horizontal') setHasOverflow(function*() {
            // Force to horizontal for measurement.
            yield false;
            // Measure, and update if there is overflow.
            yield computeHasOverflow();
        });
    }, [
        domRef,
        orientation,
        scale,
        setHasOverflow,
        children
    ]);
    // There are two main reasons we need to remeasure:
    // 1. Internal changes: Check for initial overflow or when orientation/scale/children change (from checkForOverflow dep array)
    $dqQfL$useEffect(()=>{
        checkForOverflow();
    }, [
        checkForOverflow
    ]);
    // 2. External changes: buttongroup won't change size due to any parents changing size, so listen to its container for size changes to figure out if we should remeasure
    let parent = $dqQfL$useRef();
    $dqQfL$useEffect(()=>{
        parent.current = domRef.current.parentElement;
    }, [
        domRef.current
    ]);
    $dqQfL$useResizeObserver({
        ref: parent,
        onResize: checkForOverflow
    });
    return(/*#__PURE__*/ $dqQfL$react.createElement("div", {
        ...$dqQfL$filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef,
        className: $dqQfL$classNames((/*@__PURE__*/$parcel$interopDefault($78cc0793b2d6df26$exports)), 'spectrum-ButtonGroup', {
            'spectrum-ButtonGroup--vertical': orientation === 'vertical' || hasOverflow,
            'spectrum-ButtonGroup--alignEnd': align === 'end',
            'spectrum-ButtonGroup--alignCenter': align === 'center'
        }, styleProps.className)
    }, /*#__PURE__*/ $dqQfL$react.createElement($dqQfL$SlotProvider, {
        slots: {
            button: {
                UNSAFE_className: $dqQfL$classNames((/*@__PURE__*/$parcel$interopDefault($78cc0793b2d6df26$exports)), 'spectrum-ButtonGroup-Button')
            }
        }
    }, /*#__PURE__*/ $dqQfL$react.createElement($dqQfL$Provider, {
        isDisabled: isDisabled
    }, children))));
}
/**
 * ButtonGroup handles overflow for a grouping of buttons whose actions are related to each other.
 */ let $5e39a78e785c6a30$export$69b1032f2ecdf404 = /*#__PURE__*/ $dqQfL$react.forwardRef($5e39a78e785c6a30$var$ButtonGroup);




export {$5e39a78e785c6a30$export$69b1032f2ecdf404 as ButtonGroup};
