// @react-spectrum/picker/dist/main.js
require("./main.css");
var $278M0$reactstatelycollections = require("@react-stately/collections");
var $278M0$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $278M0$spectrumiconsuiChevronDownMedium = require("@spectrum-icons/ui/ChevronDownMedium");
var $278M0$reactspectrumutils = require("@react-spectrum/utils");
var $278M0$reactariaoverlays = require("@react-aria/overlays");
var $278M0$reactspectrumlabel = require("@react-spectrum/label");
var $278M0$reactspectrumbutton = require("@react-spectrum/button");
var $278M0$reactariafocus = require("@react-aria/focus");
var $278M0$reactariaselect = require("@react-aria/select");
var $278M0$reactspectrumlistbox = require("@react-spectrum/listbox");
var $278M0$reactariautils = require("@react-aria/utils");
var $278M0$reactspectrumoverlays = require("@react-spectrum/overlays");
var $278M0$reactariainteractions = require("@react-aria/interactions");
var $278M0$reactspectrumprogress = require("@react-spectrum/progress");
var $278M0$react = require("react");
var $278M0$reactspectrumtext = require("@react-spectrum/text");
var $278M0$reactariai18n = require("@react-aria/i18n");
var $278M0$reactspectrumprovider = require("@react-spectrum/provider");
var $278M0$reactstatelyselect = require("@react-stately/select");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Item", () => $278M0$reactstatelycollections.Item);
$parcel$export(module.exports, "Section", () => $278M0$reactstatelycollections.Section);
var $78a207659b4003ba$exports = {};

$parcel$export($78a207659b4003ba$exports, "Picker", () => $78a207659b4003ba$export$ba25329847403e11);








var $cefc25c5a02ce315$exports = {};
var $fababa2c623756fb$exports = {};
$fababa2c623756fb$exports = JSON.parse("{\"loading\":\"جارٍ التحميل...\",\"placeholder\":\"حدد خيارًا...\"}");


var $d9deb081ffe0684a$exports = {};
$d9deb081ffe0684a$exports = JSON.parse("{\"loading\":\"Зареждане...\",\"placeholder\":\"Изберете опция\"}");


var $40e1ad1f2bd3fea8$exports = {};
$40e1ad1f2bd3fea8$exports = JSON.parse("{\"loading\":\"Načítání...\",\"placeholder\":\"Vyberte vhodnou možnost...\"}");


var $c5994d2a4bb5a813$exports = {};
$c5994d2a4bb5a813$exports = JSON.parse("{\"loading\":\"Indlæser ...\",\"placeholder\":\"Vælg en mulighed ...\"}");


var $77cd5d6187b728c2$exports = {};
$77cd5d6187b728c2$exports = JSON.parse("{\"loading\":\"Laden...\",\"placeholder\":\"Option auswählen...\"}");


var $0df1a4f5a7b161a0$exports = {};
$0df1a4f5a7b161a0$exports = JSON.parse("{\"loading\":\"Φόρτωση...\",\"placeholder\":\"Επιλέξτε…\"}");


var $3d73a3243a752fc9$exports = {};
$3d73a3243a752fc9$exports = JSON.parse("{\"placeholder\":\"Select an option…\",\"loading\":\"Loading…\"}");


var $1eb7b7d4408bbd6f$exports = {};
$1eb7b7d4408bbd6f$exports = JSON.parse("{\"loading\":\"Cargando…\",\"placeholder\":\"Seleccione una opción…\"}");


var $cb1b783598252e70$exports = {};
$cb1b783598252e70$exports = JSON.parse("{\"loading\":\"Laadimine...\",\"placeholder\":\"Valige valikuline...\"}");


var $65916dddfe655d3a$exports = {};
$65916dddfe655d3a$exports = JSON.parse("{\"loading\":\"Ladataan…\",\"placeholder\":\"Valitse vaihtoehto...\"}");


var $2b2519a7c2b509d5$exports = {};
$2b2519a7c2b509d5$exports = JSON.parse("{\"loading\":\"Chargement...\",\"placeholder\":\"Sélectionnez une option...\"}");


var $1bf296ea20afa4ce$exports = {};
$1bf296ea20afa4ce$exports = JSON.parse("{\"loading\":\"טוען...\",\"placeholder\":\"בחר אפשרות...\"}");


var $4f821cf433b06678$exports = {};
$4f821cf433b06678$exports = JSON.parse("{\"loading\":\"Učitavam...\",\"placeholder\":\"Odaberite opciju\"}");


var $65bd8ed249f8604c$exports = {};
$65bd8ed249f8604c$exports = JSON.parse("{\"loading\":\"Betöltés folyamatban…\",\"placeholder\":\"Válasszon egy opciót…\"}");


var $3dc1df76a923cf10$exports = {};
$3dc1df76a923cf10$exports = JSON.parse("{\"loading\":\"Caricamento...\",\"placeholder\":\"Seleziona un’opzione...\"}");


var $2f6d8cc43b353102$exports = {};
$2f6d8cc43b353102$exports = JSON.parse("{\"loading\":\"読み込み中...\",\"placeholder\":\"オプションを選択...\"}");


var $70cb5e6373ddc1dd$exports = {};
$70cb5e6373ddc1dd$exports = JSON.parse("{\"loading\":\"로드 중\",\"placeholder\":\"선택 사항 선택\"}");


var $534ea7f69936727f$exports = {};
$534ea7f69936727f$exports = JSON.parse("{\"loading\":\"Įkeliama...\",\"placeholder\":\"Pasirinkite parinktį...\"}");


var $deb2e1f904acaa05$exports = {};
$deb2e1f904acaa05$exports = JSON.parse("{\"loading\":\"Notiek ielāde...\",\"placeholder\":\"Atlasiet opciju...\"}");


var $4734955e71b2f055$exports = {};
$4734955e71b2f055$exports = JSON.parse("{\"loading\":\"Laster inn ...\",\"placeholder\":\"Velg et alternativ...\"}");


var $07df67d2484000f1$exports = {};
$07df67d2484000f1$exports = JSON.parse("{\"loading\":\"Laden...\",\"placeholder\":\"Optie selecteren...\"}");


var $0ba4d6d2ce678874$exports = {};
$0ba4d6d2ce678874$exports = JSON.parse("{\"loading\":\"Ładowanie...\",\"placeholder\":\"Wybierz opcję...\"}");


var $86d9568764ba2284$exports = {};
$86d9568764ba2284$exports = JSON.parse("{\"loading\":\"Carregando...\",\"placeholder\":\"Selecione uma opção...\"}");


var $87aa5698b818216a$exports = {};
$87aa5698b818216a$exports = JSON.parse("{\"loading\":\"A carregar...\",\"placeholder\":\"Selecionar uma opção...\"}");


var $4646dda640cb8454$exports = {};
$4646dda640cb8454$exports = JSON.parse("{\"loading\":\"Se încarcă...\",\"placeholder\":\"Selectați o opțiune\"}");


var $ef856b11c71c7dc2$exports = {};
$ef856b11c71c7dc2$exports = JSON.parse("{\"loading\":\"Загрузка...\",\"placeholder\":\"Выбрать параметр...\"}");


var $2688b290198cf4f4$exports = {};
$2688b290198cf4f4$exports = JSON.parse("{\"loading\":\"Načítava sa...\",\"placeholder\":\"Vyberte možnosť...\"}");


var $a9190ba2e8b3e3af$exports = {};
$a9190ba2e8b3e3af$exports = JSON.parse("{\"loading\":\"Nalaganje...\",\"placeholder\":\"Izberite možnost\"}");


var $767bd631d55ff027$exports = {};
$767bd631d55ff027$exports = JSON.parse("{\"loading\":\"Učitavam...\",\"placeholder\":\"Izaberite opciju\"}");


var $b88a4a1c1d60b63b$exports = {};
$b88a4a1c1d60b63b$exports = JSON.parse("{\"loading\":\"Läser in...\",\"placeholder\":\"Välj ett alternativ...\"}");


var $17877d43c6813450$exports = {};
$17877d43c6813450$exports = JSON.parse("{\"loading\":\"Yükleniyor...\",\"placeholder\":\"Bir seçim yapın…\"}");


var $3474d91d4ea302e4$exports = {};
$3474d91d4ea302e4$exports = JSON.parse("{\"loading\":\"Завантаження…\",\"placeholder\":\"Виберіть опцію...\"}");


var $eb05080be2b99b0e$exports = {};
$eb05080be2b99b0e$exports = JSON.parse("{\"loading\":\"正在加载...\",\"placeholder\":\"选择一个选项...\"}");


var $55f7df6adea6e9a9$exports = {};
$55f7df6adea6e9a9$exports = JSON.parse("{\"loading\":\"正在載入\",\"placeholder\":\"選取一個選項\"}");


$cefc25c5a02ce315$exports = {
    "ar-AE": $fababa2c623756fb$exports,
    "bg-BG": $d9deb081ffe0684a$exports,
    "cs-CZ": $40e1ad1f2bd3fea8$exports,
    "da-DK": $c5994d2a4bb5a813$exports,
    "de-DE": $77cd5d6187b728c2$exports,
    "el-GR": $0df1a4f5a7b161a0$exports,
    "en-US": $3d73a3243a752fc9$exports,
    "es-ES": $1eb7b7d4408bbd6f$exports,
    "et-EE": $cb1b783598252e70$exports,
    "fi-FI": $65916dddfe655d3a$exports,
    "fr-FR": $2b2519a7c2b509d5$exports,
    "he-IL": $1bf296ea20afa4ce$exports,
    "hr-HR": $4f821cf433b06678$exports,
    "hu-HU": $65bd8ed249f8604c$exports,
    "it-IT": $3dc1df76a923cf10$exports,
    "ja-JP": $2f6d8cc43b353102$exports,
    "ko-KR": $70cb5e6373ddc1dd$exports,
    "lt-LT": $534ea7f69936727f$exports,
    "lv-LV": $deb2e1f904acaa05$exports,
    "nb-NO": $4734955e71b2f055$exports,
    "nl-NL": $07df67d2484000f1$exports,
    "pl-PL": $0ba4d6d2ce678874$exports,
    "pt-BR": $86d9568764ba2284$exports,
    "pt-PT": $87aa5698b818216a$exports,
    "ro-RO": $4646dda640cb8454$exports,
    "ru-RU": $ef856b11c71c7dc2$exports,
    "sk-SK": $2688b290198cf4f4$exports,
    "sl-SI": $a9190ba2e8b3e3af$exports,
    "sr-SP": $767bd631d55ff027$exports,
    "sv-SE": $b88a4a1c1d60b63b$exports,
    "tr-TR": $17877d43c6813450$exports,
    "uk-UA": $3474d91d4ea302e4$exports,
    "zh-CN": $eb05080be2b99b0e$exports,
    "zh-TW": $55f7df6adea6e9a9$exports
};








var $71f3bb48ea2f223e$exports = {};

$parcel$export($71f3bb48ea2f223e$exports, "spectrum-Dropdown", () => $71f3bb48ea2f223e$export$f33c6833ddc773bd, (v) => $71f3bb48ea2f223e$export$f33c6833ddc773bd = v);
$parcel$export($71f3bb48ea2f223e$exports, "spectrum-Dropdown-chevron", () => $71f3bb48ea2f223e$export$abbb0cf96dbbb8a0, (v) => $71f3bb48ea2f223e$export$abbb0cf96dbbb8a0 = v);
$parcel$export($71f3bb48ea2f223e$exports, "spectrum-Dropdown-trigger", () => $71f3bb48ea2f223e$export$37a6d27a7cb47ce5, (v) => $71f3bb48ea2f223e$export$37a6d27a7cb47ce5 = v);
$parcel$export($71f3bb48ea2f223e$exports, "spectrum-Dropdown-label", () => $71f3bb48ea2f223e$export$1235a00795b885e7, (v) => $71f3bb48ea2f223e$export$1235a00795b885e7 = v);
$parcel$export($71f3bb48ea2f223e$exports, "is-placeholder", () => $71f3bb48ea2f223e$export$e5b2f5233e4e5194, (v) => $71f3bb48ea2f223e$export$e5b2f5233e4e5194 = v);
$parcel$export($71f3bb48ea2f223e$exports, "spectrum-Icon", () => $71f3bb48ea2f223e$export$d374b04f30360026, (v) => $71f3bb48ea2f223e$export$d374b04f30360026 = v);
$parcel$export($71f3bb48ea2f223e$exports, "spectrum-Dropdown-invalidIcon", () => $71f3bb48ea2f223e$export$a32fdb5a063ea755, (v) => $71f3bb48ea2f223e$export$a32fdb5a063ea755 = v);
$parcel$export($71f3bb48ea2f223e$exports, "spectrum-Dropdown-progressCircle", () => $71f3bb48ea2f223e$export$18281c34cabbd13f, (v) => $71f3bb48ea2f223e$export$18281c34cabbd13f = v);
$parcel$export($71f3bb48ea2f223e$exports, "spectrum-Dropdown--quiet", () => $71f3bb48ea2f223e$export$b2306417a70c7a62, (v) => $71f3bb48ea2f223e$export$b2306417a70c7a62 = v);
$parcel$export($71f3bb48ea2f223e$exports, "spectrum-Dropdown-popover--quiet", () => $71f3bb48ea2f223e$export$cbeaf986973c0633, (v) => $71f3bb48ea2f223e$export$cbeaf986973c0633 = v);
$parcel$export($71f3bb48ea2f223e$exports, "spectrum-Field", () => $71f3bb48ea2f223e$export$f6d480ae1e56eba0, (v) => $71f3bb48ea2f223e$export$f6d480ae1e56eba0 = v);
$parcel$export($71f3bb48ea2f223e$exports, "spectrum-Dropdown-fieldWrapper--quiet", () => $71f3bb48ea2f223e$export$66ddc2410b0e9eba, (v) => $71f3bb48ea2f223e$export$66ddc2410b0e9eba = v);
$parcel$export($71f3bb48ea2f223e$exports, "spectrum-Dropdown-fieldWrapper--positionSide", () => $71f3bb48ea2f223e$export$c0b4e9596538c7b1, (v) => $71f3bb48ea2f223e$export$c0b4e9596538c7b1 = v);
$parcel$export($71f3bb48ea2f223e$exports, "is-hovered", () => $71f3bb48ea2f223e$export$b8813cd5d7824ce7, (v) => $71f3bb48ea2f223e$export$b8813cd5d7824ce7 = v);
$parcel$export($71f3bb48ea2f223e$exports, "is-selected", () => $71f3bb48ea2f223e$export$1e0fb04f31d3c22a, (v) => $71f3bb48ea2f223e$export$1e0fb04f31d3c22a = v);
$parcel$export($71f3bb48ea2f223e$exports, "is-invalid", () => $71f3bb48ea2f223e$export$14a30de2866685fa, (v) => $71f3bb48ea2f223e$export$14a30de2866685fa = v);
$parcel$export($71f3bb48ea2f223e$exports, "is-disabled", () => $71f3bb48ea2f223e$export$d35bc1e505d1ebbf, (v) => $71f3bb48ea2f223e$export$d35bc1e505d1ebbf = v);
$parcel$export($71f3bb48ea2f223e$exports, "focus-ring", () => $71f3bb48ea2f223e$export$f39a09f249340e2a, (v) => $71f3bb48ea2f223e$export$f39a09f249340e2a = v);
var $71f3bb48ea2f223e$export$f33c6833ddc773bd;
var $71f3bb48ea2f223e$export$abbb0cf96dbbb8a0;
var $71f3bb48ea2f223e$export$37a6d27a7cb47ce5;
var $71f3bb48ea2f223e$export$1235a00795b885e7;
var $71f3bb48ea2f223e$export$e5b2f5233e4e5194;
var $71f3bb48ea2f223e$export$d374b04f30360026;
var $71f3bb48ea2f223e$export$a32fdb5a063ea755;
var $71f3bb48ea2f223e$export$18281c34cabbd13f;
var $71f3bb48ea2f223e$export$b2306417a70c7a62;
var $71f3bb48ea2f223e$export$cbeaf986973c0633;
var $71f3bb48ea2f223e$export$f6d480ae1e56eba0;
var $71f3bb48ea2f223e$export$66ddc2410b0e9eba;
var $71f3bb48ea2f223e$export$c0b4e9596538c7b1;
var $71f3bb48ea2f223e$export$b8813cd5d7824ce7;
var $71f3bb48ea2f223e$export$1e0fb04f31d3c22a;
var $71f3bb48ea2f223e$export$14a30de2866685fa;
var $71f3bb48ea2f223e$export$d35bc1e505d1ebbf;
var $71f3bb48ea2f223e$export$f39a09f249340e2a;
$71f3bb48ea2f223e$export$f33c6833ddc773bd = "spectrum-Dropdown_6d8ec0";
$71f3bb48ea2f223e$export$abbb0cf96dbbb8a0 = "spectrum-Dropdown-chevron_6d8ec0";
$71f3bb48ea2f223e$export$37a6d27a7cb47ce5 = "spectrum-Dropdown-trigger_6d8ec0";
$71f3bb48ea2f223e$export$1235a00795b885e7 = "spectrum-Dropdown-label_6d8ec0";
$71f3bb48ea2f223e$export$e5b2f5233e4e5194 = "is-placeholder_6d8ec0";
$71f3bb48ea2f223e$export$d374b04f30360026 = "spectrum-Icon_6d8ec0";
$71f3bb48ea2f223e$export$a32fdb5a063ea755 = "spectrum-Dropdown-invalidIcon_6d8ec0";
$71f3bb48ea2f223e$export$18281c34cabbd13f = "spectrum-Dropdown-progressCircle_6d8ec0";
$71f3bb48ea2f223e$export$b2306417a70c7a62 = "spectrum-Dropdown--quiet_6d8ec0";
$71f3bb48ea2f223e$export$cbeaf986973c0633 = "spectrum-Dropdown-popover--quiet_6d8ec0";
$71f3bb48ea2f223e$export$f6d480ae1e56eba0 = "spectrum-Field_6d8ec0";
$71f3bb48ea2f223e$export$66ddc2410b0e9eba = "spectrum-Dropdown-fieldWrapper--quiet_6d8ec0";
$71f3bb48ea2f223e$export$c0b4e9596538c7b1 = "spectrum-Dropdown-fieldWrapper--positionSide_6d8ec0";
$71f3bb48ea2f223e$export$b8813cd5d7824ce7 = "is-hovered_6d8ec0";
$71f3bb48ea2f223e$export$1e0fb04f31d3c22a = "is-selected_6d8ec0";
$71f3bb48ea2f223e$export$14a30de2866685fa = "is-invalid_6d8ec0";
$71f3bb48ea2f223e$export$d35bc1e505d1ebbf = "is-disabled_6d8ec0";
$71f3bb48ea2f223e$export$f39a09f249340e2a = "focus-ring_6d8ec0";






function $78a207659b4003ba$var$Picker(props, ref) {
    props = $278M0$reactspectrumutils.useSlotProps(props, 'picker');
    props = $278M0$reactspectrumprovider.useProviderProps(props);
    let formatMessage = $278M0$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($cefc25c5a02ce315$exports)));
    let { autoComplete: autoComplete , isDisabled: isDisabled , direction: direction = 'bottom' , align: align = 'start' , shouldFlip: shouldFlip = true , placeholder: placeholder = formatMessage('placeholder') , validationState: validationState , isQuiet: isQuiet , label: label , labelPosition: labelPosition = 'top' , menuWidth: menuWidth , name: name , autoFocus: autoFocus  } = props;
    let state = $278M0$reactstatelyselect.useSelectState(props);
    let domRef = $278M0$reactspectrumutils.useDOMRef(ref);
    let popoverRef = $278M0$react.useRef();
    let unwrappedPopoverRef = $278M0$reactspectrumutils.useUnwrapDOMRef(popoverRef);
    let triggerRef = $278M0$react.useRef();
    let unwrappedTriggerRef = $278M0$reactspectrumutils.useUnwrapDOMRef(triggerRef);
    let listboxRef = $278M0$react.useRef();
    // We create the listbox layout in Picker and pass it to ListBoxBase below
    // so that the layout information can be cached even while the listbox is not mounted.
    // We also use the layout as the keyboard delegate for type to select.
    let layout = $278M0$reactspectrumlistbox.useListBoxLayout(state);
    let { labelProps: labelProps , triggerProps: triggerProps , valueProps: valueProps , menuProps: menuProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $278M0$reactariaselect.useSelect({
        ...props,
        keyboardDelegate: layout
    }, state, unwrappedTriggerRef);
    let isMobile = $278M0$reactspectrumutils.useIsMobileDevice();
    let { overlayProps: overlayProps , placement: placement , updatePosition: updatePosition  } = $278M0$reactariaoverlays.useOverlayPosition({
        targetRef: unwrappedTriggerRef,
        overlayRef: unwrappedPopoverRef,
        scrollRef: listboxRef,
        placement: `${direction} ${align}`,
        shouldFlip: shouldFlip,
        isOpen: state.isOpen && !isMobile,
        onClose: state.close
    });
    let { hoverProps: hoverProps , isHovered: isHovered  } = $278M0$reactariainteractions.useHover({
        isDisabled: isDisabled
    });
    // Update position once the ListBox has rendered. This ensures that
    // it flips properly when it doesn't fit in the available space.
    // TODO: add ResizeObserver to useOverlayPosition so we don't need this.
    $278M0$reactariautils.useLayoutEffect(()=>{
        if (state.isOpen) requestAnimationFrame(()=>{
            updatePosition();
        });
    }, [
        state.isOpen,
        updatePosition
    ]);
    let isLoadingInitial = props.isLoading && state.collection.size === 0;
    let isLoadingMore = props.isLoading && state.collection.size > 0;
    // On small screen devices, the listbox is rendered in a tray, otherwise a popover.
    let listbox = /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement($278M0$reactariafocus.FocusScope, {
        restoreFocus: true,
        contain: isMobile
    }, /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement($278M0$reactariaoverlays.DismissButton, {
        onDismiss: ()=>state.close()
    }), /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement($278M0$reactspectrumlistbox.ListBoxBase, {
        ...menuProps,
        ref: listboxRef,
        disallowEmptySelection: true,
        autoFocus: state.focusStrategy || true,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        width: isMobile ? '100%' : undefined,
        // Set max height: inherit so Tray scrolling works
        UNSAFE_style: {
            maxHeight: 'inherit'
        },
        isLoading: isLoadingMore,
        onLoadMore: props.onLoadMore
    }), /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement($278M0$reactariaoverlays.DismissButton, {
        onDismiss: ()=>state.close()
    }));
    // Measure the width of the button to inform the width of the menu (below).
    let [buttonWidth, setButtonWidth] = $278M0$react.useState(null);
    let { scale: scale  } = $278M0$reactspectrumprovider.useProvider();
    let onResize = $278M0$react.useCallback(()=>{
        if (!isMobile && unwrappedTriggerRef.current) {
            let width = unwrappedTriggerRef.current.offsetWidth;
            setButtonWidth(width);
        }
    }, [
        unwrappedTriggerRef,
        setButtonWidth,
        isMobile
    ]);
    $278M0$reactariautils.useResizeObserver({
        ref: unwrappedTriggerRef,
        onResize: onResize
    });
    $278M0$reactariautils.useLayoutEffect(onResize, [
        scale,
        state.selectedKey,
        onResize
    ]);
    let overlay;
    if (isMobile) overlay = /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement($278M0$reactspectrumoverlays.Tray, {
        isOpen: state.isOpen,
        onClose: state.close
    }, listbox);
    else {
        // If quiet, use the default width, otherwise match the width of the button. This can be overridden by the menuWidth prop.
        // Always have a minimum width of the button width. When quiet, there is an extra offset to add.
        // Not using style props for this because they don't support `calc`.
        let width = isQuiet ? null : buttonWidth;
        let style = {
            ...overlayProps.style,
            width: menuWidth ? $278M0$reactspectrumutils.dimensionValue(menuWidth) : width,
            minWidth: isQuiet ? `calc(${buttonWidth}px + calc(2 * var(--spectrum-dropdown-quiet-offset)))` : buttonWidth
        };
        overlay = /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement($278M0$reactspectrumoverlays.Popover, {
            isOpen: state.isOpen,
            UNSAFE_style: style,
            UNSAFE_className: $278M0$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($71f3bb48ea2f223e$exports)), 'spectrum-Dropdown-popover', {
                'spectrum-Dropdown-popover--quiet': isQuiet
            }),
            ref: popoverRef,
            placement: placement,
            hideArrow: true,
            shouldCloseOnBlur: true,
            onClose: state.close
        }, listbox);
    }
    let contents = state.selectedItem ? state.selectedItem.rendered : placeholder;
    if (typeof contents === 'string') contents = /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement($278M0$reactspectrumtext.Text, null, contents);
    let picker = /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement("div", {
        className: $278M0$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($71f3bb48ea2f223e$exports)), 'spectrum-Dropdown', {
            'is-invalid': validationState === 'invalid',
            'is-disabled': isDisabled,
            'spectrum-Dropdown--quiet': isQuiet
        })
    }, /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement($278M0$reactariaselect.HiddenSelect, {
        autoComplete: autoComplete,
        isDisabled: isDisabled,
        state: state,
        triggerRef: unwrappedTriggerRef,
        label: label,
        name: name
    }), /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement($278M0$reactariainteractions.PressResponder, $278M0$reactariautils.mergeProps(hoverProps, triggerProps), /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement($278M0$reactspectrumbutton.FieldButton, {
        ref: triggerRef,
        isActive: state.isOpen,
        isQuiet: isQuiet,
        isDisabled: isDisabled,
        validationState: validationState,
        autoFocus: autoFocus,
        UNSAFE_className: $278M0$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($71f3bb48ea2f223e$exports)), 'spectrum-Dropdown-trigger', {
            'is-hovered': isHovered
        })
    }, /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement($278M0$reactspectrumutils.SlotProvider, {
        slots: {
            icon: {
                UNSAFE_className: $278M0$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($71f3bb48ea2f223e$exports)), 'spectrum-Icon'),
                size: 'S'
            },
            text: {
                ...valueProps,
                UNSAFE_className: $278M0$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($71f3bb48ea2f223e$exports)), 'spectrum-Dropdown-label', {
                    'is-placeholder': !state.selectedItem
                })
            },
            description: {
                isHidden: true
            }
        }
    }, contents), isLoadingInitial && /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement($278M0$reactspectrumprogress.ProgressCircle, {
        isIndeterminate: true,
        size: "S",
        "aria-label": formatMessage('loading'),
        UNSAFE_className: $278M0$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($71f3bb48ea2f223e$exports)), 'spectrum-Dropdown-progressCircle')
    }), validationState === 'invalid' && !isLoadingInitial && /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement(($parcel$interopDefault($278M0$spectrumiconsuiAlertMedium)), {
        UNSAFE_className: $278M0$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($71f3bb48ea2f223e$exports)), 'spectrum-Dropdown-invalidIcon')
    }), /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement(($parcel$interopDefault($278M0$spectrumiconsuiChevronDownMedium)), {
        UNSAFE_className: $278M0$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($71f3bb48ea2f223e$exports)), 'spectrum-Dropdown-chevron')
    }))), state.collection.size === 0 ? null : overlay);
    let wrapperClassName = label ? $278M0$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($71f3bb48ea2f223e$exports)), 'spectrum-Field', {
        'spectrum-Dropdown-fieldWrapper--quiet': isQuiet,
        'spectrum-Dropdown-fieldWrapper--positionSide': labelPosition === 'side'
    }) : '';
    return(/*#__PURE__*/ ($parcel$interopDefault($278M0$react)).createElement($278M0$reactspectrumlabel.Field, {
        ...props,
        ref: domRef,
        wrapperClassName: wrapperClassName,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        showErrorIcon: false,
        includeNecessityIndicatorInAccessibilityName: true,
        elementType: "span"
    }, picker));
}
/**
 * Pickers allow users to choose a single option from a collapsible list of options when space is limited.
 */ // forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
const $78a207659b4003ba$export$ba25329847403e11 = /*#__PURE__*/ ($parcel$interopDefault($278M0$react)).forwardRef($78a207659b4003ba$var$Picker);



$parcel$exportWildcard(module.exports, $78a207659b4003ba$exports);
