// @react-spectrum/picker/dist/module.js
import "./main.css";
import {Item as $3a6c13040c4e3df3$re_export$Item, Section as $3a6c13040c4e3df3$re_export$Section} from "@react-stately/collections";
import $6udtC$spectrumiconsuiAlertMedium from "@spectrum-icons/ui/AlertMedium";
import $6udtC$spectrumiconsuiChevronDownMedium from "@spectrum-icons/ui/ChevronDownMedium";
import {useSlotProps as $6udtC$useSlotProps, useDOMRef as $6udtC$useDOMRef, useUnwrapDOMRef as $6udtC$useUnwrapDOMRef, useIsMobileDevice as $6udtC$useIsMobileDevice, dimensionValue as $6udtC$dimensionValue, classNames as $6udtC$classNames, SlotProvider as $6udtC$SlotProvider} from "@react-spectrum/utils";
import {useOverlayPosition as $6udtC$useOverlayPosition, DismissButton as $6udtC$DismissButton} from "@react-aria/overlays";
import {Field as $6udtC$Field} from "@react-spectrum/label";
import {FieldButton as $6udtC$FieldButton} from "@react-spectrum/button";
import {FocusScope as $6udtC$FocusScope} from "@react-aria/focus";
import {useSelect as $6udtC$useSelect, HiddenSelect as $6udtC$HiddenSelect} from "@react-aria/select";
import {useListBoxLayout as $6udtC$useListBoxLayout, ListBoxBase as $6udtC$ListBoxBase} from "@react-spectrum/listbox";
import {useLayoutEffect as $6udtC$useLayoutEffect, useResizeObserver as $6udtC$useResizeObserver, mergeProps as $6udtC$mergeProps} from "@react-aria/utils";
import {Tray as $6udtC$Tray, Popover as $6udtC$Popover} from "@react-spectrum/overlays";
import {useHover as $6udtC$useHover, PressResponder as $6udtC$PressResponder} from "@react-aria/interactions";
import {ProgressCircle as $6udtC$ProgressCircle} from "@react-spectrum/progress";
import $6udtC$react, {useRef as $6udtC$useRef, useState as $6udtC$useState, useCallback as $6udtC$useCallback} from "react";
import {Text as $6udtC$Text} from "@react-spectrum/text";
import {useMessageFormatter as $6udtC$useMessageFormatter} from "@react-aria/i18n";
import {useProviderProps as $6udtC$useProviderProps, useProvider as $6udtC$useProvider} from "@react-spectrum/provider";
import {useSelectState as $6udtC$useSelectState} from "@react-stately/select";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $545e62b74e11b79a$exports = {};

$parcel$export($545e62b74e11b79a$exports, "Picker", () => $545e62b74e11b79a$export$ba25329847403e11);








var $fd58ba5494fe5263$exports = {};
var $cb9d87210831034e$exports = {};
$cb9d87210831034e$exports = JSON.parse("{\"loading\":\"جارٍ التحميل...\",\"placeholder\":\"حدد خيارًا...\"}");


var $e4f4717ed3dd7747$exports = {};
$e4f4717ed3dd7747$exports = JSON.parse("{\"loading\":\"Зареждане...\",\"placeholder\":\"Изберете опция\"}");


var $f74e7c6ce45e8dc0$exports = {};
$f74e7c6ce45e8dc0$exports = JSON.parse("{\"loading\":\"Načítání...\",\"placeholder\":\"Vyberte vhodnou možnost...\"}");


var $347370f2135ce1c8$exports = {};
$347370f2135ce1c8$exports = JSON.parse("{\"loading\":\"Indlæser ...\",\"placeholder\":\"Vælg en mulighed ...\"}");


var $aece270c2afb98ce$exports = {};
$aece270c2afb98ce$exports = JSON.parse("{\"loading\":\"Laden...\",\"placeholder\":\"Option auswählen...\"}");


var $1f9049e0fae6723d$exports = {};
$1f9049e0fae6723d$exports = JSON.parse("{\"loading\":\"Φόρτωση...\",\"placeholder\":\"Επιλέξτε…\"}");


var $87c5e0de88507b46$exports = {};
$87c5e0de88507b46$exports = JSON.parse("{\"placeholder\":\"Select an option…\",\"loading\":\"Loading…\"}");


var $60cce2863a5a54c0$exports = {};
$60cce2863a5a54c0$exports = JSON.parse("{\"loading\":\"Cargando…\",\"placeholder\":\"Seleccione una opción…\"}");


var $2633ab1ddc1f598a$exports = {};
$2633ab1ddc1f598a$exports = JSON.parse("{\"loading\":\"Laadimine...\",\"placeholder\":\"Valige valikuline...\"}");


var $0776be2f13f45c41$exports = {};
$0776be2f13f45c41$exports = JSON.parse("{\"loading\":\"Ladataan…\",\"placeholder\":\"Valitse vaihtoehto...\"}");


var $3bbcecb76f55555a$exports = {};
$3bbcecb76f55555a$exports = JSON.parse("{\"loading\":\"Chargement...\",\"placeholder\":\"Sélectionnez une option...\"}");


var $d909be4594bbb99a$exports = {};
$d909be4594bbb99a$exports = JSON.parse("{\"loading\":\"טוען...\",\"placeholder\":\"בחר אפשרות...\"}");


var $6ab62d2615ce3b56$exports = {};
$6ab62d2615ce3b56$exports = JSON.parse("{\"loading\":\"Učitavam...\",\"placeholder\":\"Odaberite opciju\"}");


var $5ca7e3d39c58e83b$exports = {};
$5ca7e3d39c58e83b$exports = JSON.parse("{\"loading\":\"Betöltés folyamatban…\",\"placeholder\":\"Válasszon egy opciót…\"}");


var $2e8eda03b246a6db$exports = {};
$2e8eda03b246a6db$exports = JSON.parse("{\"loading\":\"Caricamento...\",\"placeholder\":\"Seleziona un’opzione...\"}");


var $5c1e587020cc4365$exports = {};
$5c1e587020cc4365$exports = JSON.parse("{\"loading\":\"読み込み中...\",\"placeholder\":\"オプションを選択...\"}");


var $8f46d222f34acaea$exports = {};
$8f46d222f34acaea$exports = JSON.parse("{\"loading\":\"로드 중\",\"placeholder\":\"선택 사항 선택\"}");


var $40f8abca73f744ee$exports = {};
$40f8abca73f744ee$exports = JSON.parse("{\"loading\":\"Įkeliama...\",\"placeholder\":\"Pasirinkite parinktį...\"}");


var $f0aad4b264883246$exports = {};
$f0aad4b264883246$exports = JSON.parse("{\"loading\":\"Notiek ielāde...\",\"placeholder\":\"Atlasiet opciju...\"}");


var $d5d06454ec250c46$exports = {};
$d5d06454ec250c46$exports = JSON.parse("{\"loading\":\"Laster inn ...\",\"placeholder\":\"Velg et alternativ...\"}");


var $aa78de6e5ff3fe59$exports = {};
$aa78de6e5ff3fe59$exports = JSON.parse("{\"loading\":\"Laden...\",\"placeholder\":\"Optie selecteren...\"}");


var $e590d75095b88b15$exports = {};
$e590d75095b88b15$exports = JSON.parse("{\"loading\":\"Ładowanie...\",\"placeholder\":\"Wybierz opcję...\"}");


var $e7f89f2290f36b52$exports = {};
$e7f89f2290f36b52$exports = JSON.parse("{\"loading\":\"Carregando...\",\"placeholder\":\"Selecione uma opção...\"}");


var $8ae36db39b001657$exports = {};
$8ae36db39b001657$exports = JSON.parse("{\"loading\":\"A carregar...\",\"placeholder\":\"Selecionar uma opção...\"}");


var $6a651e7cdab5ad77$exports = {};
$6a651e7cdab5ad77$exports = JSON.parse("{\"loading\":\"Se încarcă...\",\"placeholder\":\"Selectați o opțiune\"}");


var $1f53d7d3f9f4aeaa$exports = {};
$1f53d7d3f9f4aeaa$exports = JSON.parse("{\"loading\":\"Загрузка...\",\"placeholder\":\"Выбрать параметр...\"}");


var $1276f024e5d1575d$exports = {};
$1276f024e5d1575d$exports = JSON.parse("{\"loading\":\"Načítava sa...\",\"placeholder\":\"Vyberte možnosť...\"}");


var $40daeedafd7e3c47$exports = {};
$40daeedafd7e3c47$exports = JSON.parse("{\"loading\":\"Nalaganje...\",\"placeholder\":\"Izberite možnost\"}");


var $3c76c3468818c13c$exports = {};
$3c76c3468818c13c$exports = JSON.parse("{\"loading\":\"Učitavam...\",\"placeholder\":\"Izaberite opciju\"}");


var $d1d1eb385b3f9a02$exports = {};
$d1d1eb385b3f9a02$exports = JSON.parse("{\"loading\":\"Läser in...\",\"placeholder\":\"Välj ett alternativ...\"}");


var $1c7fffb7d8e1c236$exports = {};
$1c7fffb7d8e1c236$exports = JSON.parse("{\"loading\":\"Yükleniyor...\",\"placeholder\":\"Bir seçim yapın…\"}");


var $780ab87056452357$exports = {};
$780ab87056452357$exports = JSON.parse("{\"loading\":\"Завантаження…\",\"placeholder\":\"Виберіть опцію...\"}");


var $87d45b3f98d0db52$exports = {};
$87d45b3f98d0db52$exports = JSON.parse("{\"loading\":\"正在加载...\",\"placeholder\":\"选择一个选项...\"}");


var $6b357d958f930a11$exports = {};
$6b357d958f930a11$exports = JSON.parse("{\"loading\":\"正在載入\",\"placeholder\":\"選取一個選項\"}");


$fd58ba5494fe5263$exports = {
    "ar-AE": $cb9d87210831034e$exports,
    "bg-BG": $e4f4717ed3dd7747$exports,
    "cs-CZ": $f74e7c6ce45e8dc0$exports,
    "da-DK": $347370f2135ce1c8$exports,
    "de-DE": $aece270c2afb98ce$exports,
    "el-GR": $1f9049e0fae6723d$exports,
    "en-US": $87c5e0de88507b46$exports,
    "es-ES": $60cce2863a5a54c0$exports,
    "et-EE": $2633ab1ddc1f598a$exports,
    "fi-FI": $0776be2f13f45c41$exports,
    "fr-FR": $3bbcecb76f55555a$exports,
    "he-IL": $d909be4594bbb99a$exports,
    "hr-HR": $6ab62d2615ce3b56$exports,
    "hu-HU": $5ca7e3d39c58e83b$exports,
    "it-IT": $2e8eda03b246a6db$exports,
    "ja-JP": $5c1e587020cc4365$exports,
    "ko-KR": $8f46d222f34acaea$exports,
    "lt-LT": $40f8abca73f744ee$exports,
    "lv-LV": $f0aad4b264883246$exports,
    "nb-NO": $d5d06454ec250c46$exports,
    "nl-NL": $aa78de6e5ff3fe59$exports,
    "pl-PL": $e590d75095b88b15$exports,
    "pt-BR": $e7f89f2290f36b52$exports,
    "pt-PT": $8ae36db39b001657$exports,
    "ro-RO": $6a651e7cdab5ad77$exports,
    "ru-RU": $1f53d7d3f9f4aeaa$exports,
    "sk-SK": $1276f024e5d1575d$exports,
    "sl-SI": $40daeedafd7e3c47$exports,
    "sr-SP": $3c76c3468818c13c$exports,
    "sv-SE": $d1d1eb385b3f9a02$exports,
    "tr-TR": $1c7fffb7d8e1c236$exports,
    "uk-UA": $780ab87056452357$exports,
    "zh-CN": $87d45b3f98d0db52$exports,
    "zh-TW": $6b357d958f930a11$exports
};








var $fd4e983208183241$exports = {};

$parcel$export($fd4e983208183241$exports, "spectrum-Dropdown", () => $fd4e983208183241$export$f33c6833ddc773bd, (v) => $fd4e983208183241$export$f33c6833ddc773bd = v);
$parcel$export($fd4e983208183241$exports, "spectrum-Dropdown-chevron", () => $fd4e983208183241$export$abbb0cf96dbbb8a0, (v) => $fd4e983208183241$export$abbb0cf96dbbb8a0 = v);
$parcel$export($fd4e983208183241$exports, "spectrum-Dropdown-trigger", () => $fd4e983208183241$export$37a6d27a7cb47ce5, (v) => $fd4e983208183241$export$37a6d27a7cb47ce5 = v);
$parcel$export($fd4e983208183241$exports, "spectrum-Dropdown-label", () => $fd4e983208183241$export$1235a00795b885e7, (v) => $fd4e983208183241$export$1235a00795b885e7 = v);
$parcel$export($fd4e983208183241$exports, "is-placeholder", () => $fd4e983208183241$export$e5b2f5233e4e5194, (v) => $fd4e983208183241$export$e5b2f5233e4e5194 = v);
$parcel$export($fd4e983208183241$exports, "spectrum-Icon", () => $fd4e983208183241$export$d374b04f30360026, (v) => $fd4e983208183241$export$d374b04f30360026 = v);
$parcel$export($fd4e983208183241$exports, "spectrum-Dropdown-invalidIcon", () => $fd4e983208183241$export$a32fdb5a063ea755, (v) => $fd4e983208183241$export$a32fdb5a063ea755 = v);
$parcel$export($fd4e983208183241$exports, "spectrum-Dropdown-progressCircle", () => $fd4e983208183241$export$18281c34cabbd13f, (v) => $fd4e983208183241$export$18281c34cabbd13f = v);
$parcel$export($fd4e983208183241$exports, "spectrum-Dropdown--quiet", () => $fd4e983208183241$export$b2306417a70c7a62, (v) => $fd4e983208183241$export$b2306417a70c7a62 = v);
$parcel$export($fd4e983208183241$exports, "spectrum-Dropdown-popover--quiet", () => $fd4e983208183241$export$cbeaf986973c0633, (v) => $fd4e983208183241$export$cbeaf986973c0633 = v);
$parcel$export($fd4e983208183241$exports, "spectrum-Field", () => $fd4e983208183241$export$f6d480ae1e56eba0, (v) => $fd4e983208183241$export$f6d480ae1e56eba0 = v);
$parcel$export($fd4e983208183241$exports, "spectrum-Dropdown-fieldWrapper--quiet", () => $fd4e983208183241$export$66ddc2410b0e9eba, (v) => $fd4e983208183241$export$66ddc2410b0e9eba = v);
$parcel$export($fd4e983208183241$exports, "spectrum-Dropdown-fieldWrapper--positionSide", () => $fd4e983208183241$export$c0b4e9596538c7b1, (v) => $fd4e983208183241$export$c0b4e9596538c7b1 = v);
$parcel$export($fd4e983208183241$exports, "is-hovered", () => $fd4e983208183241$export$b8813cd5d7824ce7, (v) => $fd4e983208183241$export$b8813cd5d7824ce7 = v);
$parcel$export($fd4e983208183241$exports, "is-selected", () => $fd4e983208183241$export$1e0fb04f31d3c22a, (v) => $fd4e983208183241$export$1e0fb04f31d3c22a = v);
$parcel$export($fd4e983208183241$exports, "is-invalid", () => $fd4e983208183241$export$14a30de2866685fa, (v) => $fd4e983208183241$export$14a30de2866685fa = v);
$parcel$export($fd4e983208183241$exports, "is-disabled", () => $fd4e983208183241$export$d35bc1e505d1ebbf, (v) => $fd4e983208183241$export$d35bc1e505d1ebbf = v);
$parcel$export($fd4e983208183241$exports, "focus-ring", () => $fd4e983208183241$export$f39a09f249340e2a, (v) => $fd4e983208183241$export$f39a09f249340e2a = v);
var $fd4e983208183241$export$f33c6833ddc773bd;
var $fd4e983208183241$export$abbb0cf96dbbb8a0;
var $fd4e983208183241$export$37a6d27a7cb47ce5;
var $fd4e983208183241$export$1235a00795b885e7;
var $fd4e983208183241$export$e5b2f5233e4e5194;
var $fd4e983208183241$export$d374b04f30360026;
var $fd4e983208183241$export$a32fdb5a063ea755;
var $fd4e983208183241$export$18281c34cabbd13f;
var $fd4e983208183241$export$b2306417a70c7a62;
var $fd4e983208183241$export$cbeaf986973c0633;
var $fd4e983208183241$export$f6d480ae1e56eba0;
var $fd4e983208183241$export$66ddc2410b0e9eba;
var $fd4e983208183241$export$c0b4e9596538c7b1;
var $fd4e983208183241$export$b8813cd5d7824ce7;
var $fd4e983208183241$export$1e0fb04f31d3c22a;
var $fd4e983208183241$export$14a30de2866685fa;
var $fd4e983208183241$export$d35bc1e505d1ebbf;
var $fd4e983208183241$export$f39a09f249340e2a;
$fd4e983208183241$export$f33c6833ddc773bd = "spectrum-Dropdown_6d8ec0";
$fd4e983208183241$export$abbb0cf96dbbb8a0 = "spectrum-Dropdown-chevron_6d8ec0";
$fd4e983208183241$export$37a6d27a7cb47ce5 = "spectrum-Dropdown-trigger_6d8ec0";
$fd4e983208183241$export$1235a00795b885e7 = "spectrum-Dropdown-label_6d8ec0";
$fd4e983208183241$export$e5b2f5233e4e5194 = "is-placeholder_6d8ec0";
$fd4e983208183241$export$d374b04f30360026 = "spectrum-Icon_6d8ec0";
$fd4e983208183241$export$a32fdb5a063ea755 = "spectrum-Dropdown-invalidIcon_6d8ec0";
$fd4e983208183241$export$18281c34cabbd13f = "spectrum-Dropdown-progressCircle_6d8ec0";
$fd4e983208183241$export$b2306417a70c7a62 = "spectrum-Dropdown--quiet_6d8ec0";
$fd4e983208183241$export$cbeaf986973c0633 = "spectrum-Dropdown-popover--quiet_6d8ec0";
$fd4e983208183241$export$f6d480ae1e56eba0 = "spectrum-Field_6d8ec0";
$fd4e983208183241$export$66ddc2410b0e9eba = "spectrum-Dropdown-fieldWrapper--quiet_6d8ec0";
$fd4e983208183241$export$c0b4e9596538c7b1 = "spectrum-Dropdown-fieldWrapper--positionSide_6d8ec0";
$fd4e983208183241$export$b8813cd5d7824ce7 = "is-hovered_6d8ec0";
$fd4e983208183241$export$1e0fb04f31d3c22a = "is-selected_6d8ec0";
$fd4e983208183241$export$14a30de2866685fa = "is-invalid_6d8ec0";
$fd4e983208183241$export$d35bc1e505d1ebbf = "is-disabled_6d8ec0";
$fd4e983208183241$export$f39a09f249340e2a = "focus-ring_6d8ec0";






function $545e62b74e11b79a$var$Picker(props, ref) {
    props = $6udtC$useSlotProps(props, 'picker');
    props = $6udtC$useProviderProps(props);
    let formatMessage = $6udtC$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($fd58ba5494fe5263$exports)));
    let { autoComplete: autoComplete , isDisabled: isDisabled , direction: direction = 'bottom' , align: align = 'start' , shouldFlip: shouldFlip = true , placeholder: placeholder = formatMessage('placeholder') , validationState: validationState , isQuiet: isQuiet , label: label , labelPosition: labelPosition = 'top' , menuWidth: menuWidth , name: name , autoFocus: autoFocus  } = props;
    let state = $6udtC$useSelectState(props);
    let domRef = $6udtC$useDOMRef(ref);
    let popoverRef = $6udtC$useRef();
    let unwrappedPopoverRef = $6udtC$useUnwrapDOMRef(popoverRef);
    let triggerRef = $6udtC$useRef();
    let unwrappedTriggerRef = $6udtC$useUnwrapDOMRef(triggerRef);
    let listboxRef = $6udtC$useRef();
    // We create the listbox layout in Picker and pass it to ListBoxBase below
    // so that the layout information can be cached even while the listbox is not mounted.
    // We also use the layout as the keyboard delegate for type to select.
    let layout = $6udtC$useListBoxLayout(state);
    let { labelProps: labelProps , triggerProps: triggerProps , valueProps: valueProps , menuProps: menuProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $6udtC$useSelect({
        ...props,
        keyboardDelegate: layout
    }, state, unwrappedTriggerRef);
    let isMobile = $6udtC$useIsMobileDevice();
    let { overlayProps: overlayProps , placement: placement , updatePosition: updatePosition  } = $6udtC$useOverlayPosition({
        targetRef: unwrappedTriggerRef,
        overlayRef: unwrappedPopoverRef,
        scrollRef: listboxRef,
        placement: `${direction} ${align}`,
        shouldFlip: shouldFlip,
        isOpen: state.isOpen && !isMobile,
        onClose: state.close
    });
    let { hoverProps: hoverProps , isHovered: isHovered  } = $6udtC$useHover({
        isDisabled: isDisabled
    });
    // Update position once the ListBox has rendered. This ensures that
    // it flips properly when it doesn't fit in the available space.
    // TODO: add ResizeObserver to useOverlayPosition so we don't need this.
    $6udtC$useLayoutEffect(()=>{
        if (state.isOpen) requestAnimationFrame(()=>{
            updatePosition();
        });
    }, [
        state.isOpen,
        updatePosition
    ]);
    let isLoadingInitial = props.isLoading && state.collection.size === 0;
    let isLoadingMore = props.isLoading && state.collection.size > 0;
    // On small screen devices, the listbox is rendered in a tray, otherwise a popover.
    let listbox = /*#__PURE__*/ $6udtC$react.createElement($6udtC$FocusScope, {
        restoreFocus: true,
        contain: isMobile
    }, /*#__PURE__*/ $6udtC$react.createElement($6udtC$DismissButton, {
        onDismiss: ()=>state.close()
    }), /*#__PURE__*/ $6udtC$react.createElement($6udtC$ListBoxBase, {
        ...menuProps,
        ref: listboxRef,
        disallowEmptySelection: true,
        autoFocus: state.focusStrategy || true,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        width: isMobile ? '100%' : undefined,
        // Set max height: inherit so Tray scrolling works
        UNSAFE_style: {
            maxHeight: 'inherit'
        },
        isLoading: isLoadingMore,
        onLoadMore: props.onLoadMore
    }), /*#__PURE__*/ $6udtC$react.createElement($6udtC$DismissButton, {
        onDismiss: ()=>state.close()
    }));
    // Measure the width of the button to inform the width of the menu (below).
    let [buttonWidth, setButtonWidth] = $6udtC$useState(null);
    let { scale: scale  } = $6udtC$useProvider();
    let onResize = $6udtC$useCallback(()=>{
        if (!isMobile && unwrappedTriggerRef.current) {
            let width = unwrappedTriggerRef.current.offsetWidth;
            setButtonWidth(width);
        }
    }, [
        unwrappedTriggerRef,
        setButtonWidth,
        isMobile
    ]);
    $6udtC$useResizeObserver({
        ref: unwrappedTriggerRef,
        onResize: onResize
    });
    $6udtC$useLayoutEffect(onResize, [
        scale,
        state.selectedKey,
        onResize
    ]);
    let overlay;
    if (isMobile) overlay = /*#__PURE__*/ $6udtC$react.createElement($6udtC$Tray, {
        isOpen: state.isOpen,
        onClose: state.close
    }, listbox);
    else {
        // If quiet, use the default width, otherwise match the width of the button. This can be overridden by the menuWidth prop.
        // Always have a minimum width of the button width. When quiet, there is an extra offset to add.
        // Not using style props for this because they don't support `calc`.
        let width = isQuiet ? null : buttonWidth;
        let style = {
            ...overlayProps.style,
            width: menuWidth ? $6udtC$dimensionValue(menuWidth) : width,
            minWidth: isQuiet ? `calc(${buttonWidth}px + calc(2 * var(--spectrum-dropdown-quiet-offset)))` : buttonWidth
        };
        overlay = /*#__PURE__*/ $6udtC$react.createElement($6udtC$Popover, {
            isOpen: state.isOpen,
            UNSAFE_style: style,
            UNSAFE_className: $6udtC$classNames((/*@__PURE__*/$parcel$interopDefault($fd4e983208183241$exports)), 'spectrum-Dropdown-popover', {
                'spectrum-Dropdown-popover--quiet': isQuiet
            }),
            ref: popoverRef,
            placement: placement,
            hideArrow: true,
            shouldCloseOnBlur: true,
            onClose: state.close
        }, listbox);
    }
    let contents = state.selectedItem ? state.selectedItem.rendered : placeholder;
    if (typeof contents === 'string') contents = /*#__PURE__*/ $6udtC$react.createElement($6udtC$Text, null, contents);
    let picker = /*#__PURE__*/ $6udtC$react.createElement("div", {
        className: $6udtC$classNames((/*@__PURE__*/$parcel$interopDefault($fd4e983208183241$exports)), 'spectrum-Dropdown', {
            'is-invalid': validationState === 'invalid',
            'is-disabled': isDisabled,
            'spectrum-Dropdown--quiet': isQuiet
        })
    }, /*#__PURE__*/ $6udtC$react.createElement($6udtC$HiddenSelect, {
        autoComplete: autoComplete,
        isDisabled: isDisabled,
        state: state,
        triggerRef: unwrappedTriggerRef,
        label: label,
        name: name
    }), /*#__PURE__*/ $6udtC$react.createElement($6udtC$PressResponder, $6udtC$mergeProps(hoverProps, triggerProps), /*#__PURE__*/ $6udtC$react.createElement($6udtC$FieldButton, {
        ref: triggerRef,
        isActive: state.isOpen,
        isQuiet: isQuiet,
        isDisabled: isDisabled,
        validationState: validationState,
        autoFocus: autoFocus,
        UNSAFE_className: $6udtC$classNames((/*@__PURE__*/$parcel$interopDefault($fd4e983208183241$exports)), 'spectrum-Dropdown-trigger', {
            'is-hovered': isHovered
        })
    }, /*#__PURE__*/ $6udtC$react.createElement($6udtC$SlotProvider, {
        slots: {
            icon: {
                UNSAFE_className: $6udtC$classNames((/*@__PURE__*/$parcel$interopDefault($fd4e983208183241$exports)), 'spectrum-Icon'),
                size: 'S'
            },
            text: {
                ...valueProps,
                UNSAFE_className: $6udtC$classNames((/*@__PURE__*/$parcel$interopDefault($fd4e983208183241$exports)), 'spectrum-Dropdown-label', {
                    'is-placeholder': !state.selectedItem
                })
            },
            description: {
                isHidden: true
            }
        }
    }, contents), isLoadingInitial && /*#__PURE__*/ $6udtC$react.createElement($6udtC$ProgressCircle, {
        isIndeterminate: true,
        size: "S",
        "aria-label": formatMessage('loading'),
        UNSAFE_className: $6udtC$classNames((/*@__PURE__*/$parcel$interopDefault($fd4e983208183241$exports)), 'spectrum-Dropdown-progressCircle')
    }), validationState === 'invalid' && !isLoadingInitial && /*#__PURE__*/ $6udtC$react.createElement($6udtC$spectrumiconsuiAlertMedium, {
        UNSAFE_className: $6udtC$classNames((/*@__PURE__*/$parcel$interopDefault($fd4e983208183241$exports)), 'spectrum-Dropdown-invalidIcon')
    }), /*#__PURE__*/ $6udtC$react.createElement($6udtC$spectrumiconsuiChevronDownMedium, {
        UNSAFE_className: $6udtC$classNames((/*@__PURE__*/$parcel$interopDefault($fd4e983208183241$exports)), 'spectrum-Dropdown-chevron')
    }))), state.collection.size === 0 ? null : overlay);
    let wrapperClassName = label ? $6udtC$classNames((/*@__PURE__*/$parcel$interopDefault($fd4e983208183241$exports)), 'spectrum-Field', {
        'spectrum-Dropdown-fieldWrapper--quiet': isQuiet,
        'spectrum-Dropdown-fieldWrapper--positionSide': labelPosition === 'side'
    }) : '';
    return(/*#__PURE__*/ $6udtC$react.createElement($6udtC$Field, {
        ...props,
        ref: domRef,
        wrapperClassName: wrapperClassName,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        showErrorIcon: false,
        includeNecessityIndicatorInAccessibilityName: true,
        elementType: "span"
    }, picker));
}
/**
 * Pickers allow users to choose a single option from a collapsible list of options when space is limited.
 */ // forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
const $545e62b74e11b79a$export$ba25329847403e11 = /*#__PURE__*/ $6udtC$react.forwardRef($545e62b74e11b79a$var$Picker);





export {$3a6c13040c4e3df3$re_export$Item as Item, $3a6c13040c4e3df3$re_export$Section as Section, $545e62b74e11b79a$export$ba25329847403e11 as Picker};
