// @react-spectrum/statuslight/dist/main.js
require("./main.css");
var $lPlPc$reactspectrumutils = require("@react-spectrum/utils");
var $lPlPc$reactariautils = require("@react-aria/utils");
var $lPlPc$react = require("react");
var $lPlPc$reactspectrumprovider = require("@react-spectrum/provider");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $5f006adc0ac94144$exports = {};

$parcel$export($5f006adc0ac94144$exports, "StatusLight", () => $5f006adc0ac94144$export$5f84c37a31c6e41c);



var $0a01bece09d42ba7$exports = {};

$parcel$export($0a01bece09d42ba7$exports, "spectrum-StatusLight", () => $0a01bece09d42ba7$export$11d3697e7e5c1bef, (v) => $0a01bece09d42ba7$export$11d3697e7e5c1bef = v);
$parcel$export($0a01bece09d42ba7$exports, "spectrum-StatusLight--neutral", () => $0a01bece09d42ba7$export$bbf75dd99797a93e, (v) => $0a01bece09d42ba7$export$bbf75dd99797a93e = v);
$parcel$export($0a01bece09d42ba7$exports, "is-disabled", () => $0a01bece09d42ba7$export$d35bc1e505d1ebbf, (v) => $0a01bece09d42ba7$export$d35bc1e505d1ebbf = v);
$parcel$export($0a01bece09d42ba7$exports, "spectrum-StatusLight--negative", () => $0a01bece09d42ba7$export$1974c2646528dc4f, (v) => $0a01bece09d42ba7$export$1974c2646528dc4f = v);
$parcel$export($0a01bece09d42ba7$exports, "spectrum-StatusLight--notice", () => $0a01bece09d42ba7$export$30a5e512ed4618b, (v) => $0a01bece09d42ba7$export$30a5e512ed4618b = v);
$parcel$export($0a01bece09d42ba7$exports, "spectrum-StatusLight--positive", () => $0a01bece09d42ba7$export$141dae5af9534f54, (v) => $0a01bece09d42ba7$export$141dae5af9534f54 = v);
$parcel$export($0a01bece09d42ba7$exports, "spectrum-StatusLight--info", () => $0a01bece09d42ba7$export$945a5d9a4e8494e6, (v) => $0a01bece09d42ba7$export$945a5d9a4e8494e6 = v);
$parcel$export($0a01bece09d42ba7$exports, "spectrum-StatusLight--active", () => $0a01bece09d42ba7$export$b40c603f4b579265, (v) => $0a01bece09d42ba7$export$b40c603f4b579265 = v);
$parcel$export($0a01bece09d42ba7$exports, "spectrum-StatusLight--celery", () => $0a01bece09d42ba7$export$e9a22f483095fe3e, (v) => $0a01bece09d42ba7$export$e9a22f483095fe3e = v);
$parcel$export($0a01bece09d42ba7$exports, "spectrum-StatusLight--yellow", () => $0a01bece09d42ba7$export$b22c4af54e28c6c, (v) => $0a01bece09d42ba7$export$b22c4af54e28c6c = v);
$parcel$export($0a01bece09d42ba7$exports, "spectrum-StatusLight--fuchsia", () => $0a01bece09d42ba7$export$cbaba642d0afd312, (v) => $0a01bece09d42ba7$export$cbaba642d0afd312 = v);
$parcel$export($0a01bece09d42ba7$exports, "spectrum-StatusLight--indigo", () => $0a01bece09d42ba7$export$8b77e0c07c11635a, (v) => $0a01bece09d42ba7$export$8b77e0c07c11635a = v);
$parcel$export($0a01bece09d42ba7$exports, "spectrum-StatusLight--seafoam", () => $0a01bece09d42ba7$export$d14bcd2d3ae8238c, (v) => $0a01bece09d42ba7$export$d14bcd2d3ae8238c = v);
$parcel$export($0a01bece09d42ba7$exports, "spectrum-StatusLight--chartreuse", () => $0a01bece09d42ba7$export$f05bd1f10c0cc15f, (v) => $0a01bece09d42ba7$export$f05bd1f10c0cc15f = v);
$parcel$export($0a01bece09d42ba7$exports, "spectrum-StatusLight--magenta", () => $0a01bece09d42ba7$export$43990cc5ba9a0059, (v) => $0a01bece09d42ba7$export$43990cc5ba9a0059 = v);
$parcel$export($0a01bece09d42ba7$exports, "spectrum-StatusLight--purple", () => $0a01bece09d42ba7$export$975baced4f8182b4, (v) => $0a01bece09d42ba7$export$975baced4f8182b4 = v);
var $0a01bece09d42ba7$export$11d3697e7e5c1bef;
var $0a01bece09d42ba7$export$bbf75dd99797a93e;
var $0a01bece09d42ba7$export$d35bc1e505d1ebbf;
var $0a01bece09d42ba7$export$1974c2646528dc4f;
var $0a01bece09d42ba7$export$30a5e512ed4618b;
var $0a01bece09d42ba7$export$141dae5af9534f54;
var $0a01bece09d42ba7$export$945a5d9a4e8494e6;
var $0a01bece09d42ba7$export$b40c603f4b579265;
var $0a01bece09d42ba7$export$e9a22f483095fe3e;
var $0a01bece09d42ba7$export$b22c4af54e28c6c;
var $0a01bece09d42ba7$export$cbaba642d0afd312;
var $0a01bece09d42ba7$export$8b77e0c07c11635a;
var $0a01bece09d42ba7$export$d14bcd2d3ae8238c;
var $0a01bece09d42ba7$export$f05bd1f10c0cc15f;
var $0a01bece09d42ba7$export$43990cc5ba9a0059;
var $0a01bece09d42ba7$export$975baced4f8182b4;
$0a01bece09d42ba7$export$11d3697e7e5c1bef = "spectrum-StatusLight_c020b6";
$0a01bece09d42ba7$export$bbf75dd99797a93e = "spectrum-StatusLight--neutral_c020b6";
$0a01bece09d42ba7$export$d35bc1e505d1ebbf = "is-disabled_c020b6";
$0a01bece09d42ba7$export$1974c2646528dc4f = "spectrum-StatusLight--negative_c020b6";
$0a01bece09d42ba7$export$30a5e512ed4618b = "spectrum-StatusLight--notice_c020b6";
$0a01bece09d42ba7$export$141dae5af9534f54 = "spectrum-StatusLight--positive_c020b6";
$0a01bece09d42ba7$export$945a5d9a4e8494e6 = "spectrum-StatusLight--info_c020b6";
$0a01bece09d42ba7$export$b40c603f4b579265 = "spectrum-StatusLight--active_c020b6";
$0a01bece09d42ba7$export$e9a22f483095fe3e = "spectrum-StatusLight--celery_c020b6";
$0a01bece09d42ba7$export$b22c4af54e28c6c = "spectrum-StatusLight--yellow_c020b6";
$0a01bece09d42ba7$export$cbaba642d0afd312 = "spectrum-StatusLight--fuchsia_c020b6";
$0a01bece09d42ba7$export$8b77e0c07c11635a = "spectrum-StatusLight--indigo_c020b6";
$0a01bece09d42ba7$export$d14bcd2d3ae8238c = "spectrum-StatusLight--seafoam_c020b6";
$0a01bece09d42ba7$export$f05bd1f10c0cc15f = "spectrum-StatusLight--chartreuse_c020b6";
$0a01bece09d42ba7$export$43990cc5ba9a0059 = "spectrum-StatusLight--magenta_c020b6";
$0a01bece09d42ba7$export$975baced4f8182b4 = "spectrum-StatusLight--purple_c020b6";



function $5f006adc0ac94144$var$StatusLight(props, ref) {
    let { variant: variant , children: children , isDisabled: isDisabled , role: role , ...otherProps } = $lPlPc$reactspectrumprovider.useProviderProps(props);
    let domRef = $lPlPc$reactspectrumutils.useDOMRef(ref);
    let { styleProps: styleProps  } = $lPlPc$reactspectrumutils.useStyleProps(otherProps);
    if (!children && !props['aria-label']) console.warn('If no children are provided, an aria-label must be specified');
    if (!role && (props['aria-label'] || props['aria-labelledby'])) console.warn('A labelled StatusLight must have a role.');
    return(/*#__PURE__*/ ($parcel$interopDefault($lPlPc$react)).createElement("div", {
        ...$lPlPc$reactariautils.filterDOMProps(otherProps, {
            labelable: !!role
        }),
        ...styleProps,
        role: role,
        className: $lPlPc$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($0a01bece09d42ba7$exports)), 'spectrum-StatusLight', `spectrum-StatusLight--${variant}`, {
            'is-disabled': isDisabled
        }, styleProps.className),
        ref: domRef
    }, children));
}
/**
 * Status lights are used to color code categories and labels commonly found in data visualization.
 * When status lights have a semantic meaning, they should use semantic variant colors.
 */ let $5f006adc0ac94144$export$5f84c37a31c6e41c = /*#__PURE__*/ $lPlPc$react.forwardRef($5f006adc0ac94144$var$StatusLight);


$parcel$exportWildcard(module.exports, $5f006adc0ac94144$exports);
