// @react-spectrum/statuslight/dist/module.js
import "./main.css";
import {useDOMRef as $2knHf$useDOMRef, useStyleProps as $2knHf$useStyleProps, classNames as $2knHf$classNames} from "@react-spectrum/utils";
import {filterDOMProps as $2knHf$filterDOMProps} from "@react-aria/utils";
import $2knHf$react, {forwardRef as $2knHf$forwardRef} from "react";
import {useProviderProps as $2knHf$useProviderProps} from "@react-spectrum/provider";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $9d1f104f9a5d54c4$exports = {};

$parcel$export($9d1f104f9a5d54c4$exports, "StatusLight", () => $9d1f104f9a5d54c4$export$5f84c37a31c6e41c);



var $bf7f989bef646f91$exports = {};

$parcel$export($bf7f989bef646f91$exports, "spectrum-StatusLight", () => $bf7f989bef646f91$export$11d3697e7e5c1bef, (v) => $bf7f989bef646f91$export$11d3697e7e5c1bef = v);
$parcel$export($bf7f989bef646f91$exports, "spectrum-StatusLight--neutral", () => $bf7f989bef646f91$export$bbf75dd99797a93e, (v) => $bf7f989bef646f91$export$bbf75dd99797a93e = v);
$parcel$export($bf7f989bef646f91$exports, "is-disabled", () => $bf7f989bef646f91$export$d35bc1e505d1ebbf, (v) => $bf7f989bef646f91$export$d35bc1e505d1ebbf = v);
$parcel$export($bf7f989bef646f91$exports, "spectrum-StatusLight--negative", () => $bf7f989bef646f91$export$1974c2646528dc4f, (v) => $bf7f989bef646f91$export$1974c2646528dc4f = v);
$parcel$export($bf7f989bef646f91$exports, "spectrum-StatusLight--notice", () => $bf7f989bef646f91$export$30a5e512ed4618b, (v) => $bf7f989bef646f91$export$30a5e512ed4618b = v);
$parcel$export($bf7f989bef646f91$exports, "spectrum-StatusLight--positive", () => $bf7f989bef646f91$export$141dae5af9534f54, (v) => $bf7f989bef646f91$export$141dae5af9534f54 = v);
$parcel$export($bf7f989bef646f91$exports, "spectrum-StatusLight--info", () => $bf7f989bef646f91$export$945a5d9a4e8494e6, (v) => $bf7f989bef646f91$export$945a5d9a4e8494e6 = v);
$parcel$export($bf7f989bef646f91$exports, "spectrum-StatusLight--active", () => $bf7f989bef646f91$export$b40c603f4b579265, (v) => $bf7f989bef646f91$export$b40c603f4b579265 = v);
$parcel$export($bf7f989bef646f91$exports, "spectrum-StatusLight--celery", () => $bf7f989bef646f91$export$e9a22f483095fe3e, (v) => $bf7f989bef646f91$export$e9a22f483095fe3e = v);
$parcel$export($bf7f989bef646f91$exports, "spectrum-StatusLight--yellow", () => $bf7f989bef646f91$export$b22c4af54e28c6c, (v) => $bf7f989bef646f91$export$b22c4af54e28c6c = v);
$parcel$export($bf7f989bef646f91$exports, "spectrum-StatusLight--fuchsia", () => $bf7f989bef646f91$export$cbaba642d0afd312, (v) => $bf7f989bef646f91$export$cbaba642d0afd312 = v);
$parcel$export($bf7f989bef646f91$exports, "spectrum-StatusLight--indigo", () => $bf7f989bef646f91$export$8b77e0c07c11635a, (v) => $bf7f989bef646f91$export$8b77e0c07c11635a = v);
$parcel$export($bf7f989bef646f91$exports, "spectrum-StatusLight--seafoam", () => $bf7f989bef646f91$export$d14bcd2d3ae8238c, (v) => $bf7f989bef646f91$export$d14bcd2d3ae8238c = v);
$parcel$export($bf7f989bef646f91$exports, "spectrum-StatusLight--chartreuse", () => $bf7f989bef646f91$export$f05bd1f10c0cc15f, (v) => $bf7f989bef646f91$export$f05bd1f10c0cc15f = v);
$parcel$export($bf7f989bef646f91$exports, "spectrum-StatusLight--magenta", () => $bf7f989bef646f91$export$43990cc5ba9a0059, (v) => $bf7f989bef646f91$export$43990cc5ba9a0059 = v);
$parcel$export($bf7f989bef646f91$exports, "spectrum-StatusLight--purple", () => $bf7f989bef646f91$export$975baced4f8182b4, (v) => $bf7f989bef646f91$export$975baced4f8182b4 = v);
var $bf7f989bef646f91$export$11d3697e7e5c1bef;
var $bf7f989bef646f91$export$bbf75dd99797a93e;
var $bf7f989bef646f91$export$d35bc1e505d1ebbf;
var $bf7f989bef646f91$export$1974c2646528dc4f;
var $bf7f989bef646f91$export$30a5e512ed4618b;
var $bf7f989bef646f91$export$141dae5af9534f54;
var $bf7f989bef646f91$export$945a5d9a4e8494e6;
var $bf7f989bef646f91$export$b40c603f4b579265;
var $bf7f989bef646f91$export$e9a22f483095fe3e;
var $bf7f989bef646f91$export$b22c4af54e28c6c;
var $bf7f989bef646f91$export$cbaba642d0afd312;
var $bf7f989bef646f91$export$8b77e0c07c11635a;
var $bf7f989bef646f91$export$d14bcd2d3ae8238c;
var $bf7f989bef646f91$export$f05bd1f10c0cc15f;
var $bf7f989bef646f91$export$43990cc5ba9a0059;
var $bf7f989bef646f91$export$975baced4f8182b4;
$bf7f989bef646f91$export$11d3697e7e5c1bef = "spectrum-StatusLight_c020b6";
$bf7f989bef646f91$export$bbf75dd99797a93e = "spectrum-StatusLight--neutral_c020b6";
$bf7f989bef646f91$export$d35bc1e505d1ebbf = "is-disabled_c020b6";
$bf7f989bef646f91$export$1974c2646528dc4f = "spectrum-StatusLight--negative_c020b6";
$bf7f989bef646f91$export$30a5e512ed4618b = "spectrum-StatusLight--notice_c020b6";
$bf7f989bef646f91$export$141dae5af9534f54 = "spectrum-StatusLight--positive_c020b6";
$bf7f989bef646f91$export$945a5d9a4e8494e6 = "spectrum-StatusLight--info_c020b6";
$bf7f989bef646f91$export$b40c603f4b579265 = "spectrum-StatusLight--active_c020b6";
$bf7f989bef646f91$export$e9a22f483095fe3e = "spectrum-StatusLight--celery_c020b6";
$bf7f989bef646f91$export$b22c4af54e28c6c = "spectrum-StatusLight--yellow_c020b6";
$bf7f989bef646f91$export$cbaba642d0afd312 = "spectrum-StatusLight--fuchsia_c020b6";
$bf7f989bef646f91$export$8b77e0c07c11635a = "spectrum-StatusLight--indigo_c020b6";
$bf7f989bef646f91$export$d14bcd2d3ae8238c = "spectrum-StatusLight--seafoam_c020b6";
$bf7f989bef646f91$export$f05bd1f10c0cc15f = "spectrum-StatusLight--chartreuse_c020b6";
$bf7f989bef646f91$export$43990cc5ba9a0059 = "spectrum-StatusLight--magenta_c020b6";
$bf7f989bef646f91$export$975baced4f8182b4 = "spectrum-StatusLight--purple_c020b6";



function $9d1f104f9a5d54c4$var$StatusLight(props, ref) {
    let { variant: variant , children: children , isDisabled: isDisabled , role: role , ...otherProps } = $2knHf$useProviderProps(props);
    let domRef = $2knHf$useDOMRef(ref);
    let { styleProps: styleProps  } = $2knHf$useStyleProps(otherProps);
    if (!children && !props['aria-label']) console.warn('If no children are provided, an aria-label must be specified');
    if (!role && (props['aria-label'] || props['aria-labelledby'])) console.warn('A labelled StatusLight must have a role.');
    return(/*#__PURE__*/ $2knHf$react.createElement("div", {
        ...$2knHf$filterDOMProps(otherProps, {
            labelable: !!role
        }),
        ...styleProps,
        role: role,
        className: $2knHf$classNames((/*@__PURE__*/$parcel$interopDefault($bf7f989bef646f91$exports)), 'spectrum-StatusLight', `spectrum-StatusLight--${variant}`, {
            'is-disabled': isDisabled
        }, styleProps.className),
        ref: domRef
    }, children));
}
/**
 * Status lights are used to color code categories and labels commonly found in data visualization.
 * When status lights have a semantic meaning, they should use semantic variant colors.
 */ let $9d1f104f9a5d54c4$export$5f84c37a31c6e41c = /*#__PURE__*/ $2knHf$forwardRef($9d1f104f9a5d54c4$var$StatusLight);




export {$9d1f104f9a5d54c4$export$5f84c37a31c6e41c as StatusLight};
