// @react-spectrum/actiongroup/dist/main.js
require("./main.css");
var $4VTOr$reactstatelycollections = require("@react-stately/collections");
var $4VTOr$reactspectrumbutton = require("@react-spectrum/button");
var $4VTOr$spectrumiconsuiChevronDownMedium = require("@spectrum-icons/ui/ChevronDownMedium");
var $4VTOr$reactspectrumutils = require("@react-spectrum/utils");
var $4VTOr$reactariautils = require("@react-aria/utils");
var $4VTOr$reactspectrummenu = require("@react-spectrum/menu");
var $4VTOr$reactstatelylist = require("@react-stately/list");
var $4VTOr$spectrumiconsworkflowMore = require("@spectrum-icons/workflow/More");
var $4VTOr$reactariainteractions = require("@react-aria/interactions");
var $4VTOr$reactspectrumprovider = require("@react-spectrum/provider");
var $4VTOr$react = require("react");
var $4VTOr$reactspectrumtext = require("@react-spectrum/text");
var $4VTOr$reactspectrumtooltip = require("@react-spectrum/tooltip");
var $4VTOr$reactariaactiongroup = require("@react-aria/actiongroup");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Item", () => $4VTOr$reactstatelycollections.Item);
var $202ea682547fd820$exports = {};

$parcel$export($202ea682547fd820$exports, "ActionGroup", () => $202ea682547fd820$export$c21a5597f732a168);

var $fdb409b291c9958f$exports = {};

$parcel$export($fdb409b291c9958f$exports, "spectrum-Button", () => $fdb409b291c9958f$export$1db4cca5b4ade39a, (v) => $fdb409b291c9958f$export$1db4cca5b4ade39a = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton", () => $fdb409b291c9958f$export$53da69f51b770d3, (v) => $fdb409b291c9958f$export$53da69f51b770d3 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-LogicButton", () => $fdb409b291c9958f$export$36ae67a131730482, (v) => $fdb409b291c9958f$export$36ae67a131730482 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-FieldButton", () => $fdb409b291c9958f$export$a024c99ecf6b5741, (v) => $fdb409b291c9958f$export$a024c99ecf6b5741 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ClearButton", () => $fdb409b291c9958f$export$b752ce409e5660c, (v) => $fdb409b291c9958f$export$b752ce409e5660c = v);
$parcel$export($fdb409b291c9958f$exports, "focus-ring", () => $fdb409b291c9958f$export$f39a09f249340e2a, (v) => $fdb409b291c9958f$export$f39a09f249340e2a = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Icon", () => $fdb409b291c9958f$export$d374b04f30360026, (v) => $fdb409b291c9958f$export$d374b04f30360026 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton--emphasized", () => $fdb409b291c9958f$export$5aabb61a966e266f, (v) => $fdb409b291c9958f$export$5aabb61a966e266f = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton--staticColor", () => $fdb409b291c9958f$export$29b7b5d6a5a497c5, (v) => $fdb409b291c9958f$export$29b7b5d6a5a497c5 = v);
$parcel$export($fdb409b291c9958f$exports, "is-hovered", () => $fdb409b291c9958f$export$b8813cd5d7824ce7, (v) => $fdb409b291c9958f$export$b8813cd5d7824ce7 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Button-label", () => $fdb409b291c9958f$export$c081ba562e204942, (v) => $fdb409b291c9958f$export$c081ba562e204942 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton-label", () => $fdb409b291c9958f$export$ebb4635c66bacdb9, (v) => $fdb409b291c9958f$export$ebb4635c66bacdb9 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton-hold", () => $fdb409b291c9958f$export$4a4f01c443eff992, (v) => $fdb409b291c9958f$export$4a4f01c443eff992 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton--quiet", () => $fdb409b291c9958f$export$40e1f009544bacf3, (v) => $fdb409b291c9958f$export$40e1f009544bacf3 = v);
$parcel$export($fdb409b291c9958f$exports, "is-disabled", () => $fdb409b291c9958f$export$d35bc1e505d1ebbf, (v) => $fdb409b291c9958f$export$d35bc1e505d1ebbf = v);
$parcel$export($fdb409b291c9958f$exports, "is-open", () => $fdb409b291c9958f$export$a9781837241c946d, (v) => $fdb409b291c9958f$export$a9781837241c946d = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-FieldButton--quiet", () => $fdb409b291c9958f$export$5ca7b274e4fff9fc, (v) => $fdb409b291c9958f$export$5ca7b274e4fff9fc = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ClearButton--overBackground", () => $fdb409b291c9958f$export$116f2fccf941d34d, (v) => $fdb409b291c9958f$export$116f2fccf941d34d = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ClearButton--small", () => $fdb409b291c9958f$export$49280ffa5236ba8b, (v) => $fdb409b291c9958f$export$49280ffa5236ba8b = v);
$parcel$export($fdb409b291c9958f$exports, "is-focused", () => $fdb409b291c9958f$export$e7dc768d35940237, (v) => $fdb409b291c9958f$export$e7dc768d35940237 = v);
$parcel$export($fdb409b291c9958f$exports, "is-active", () => $fdb409b291c9958f$export$20fd0f7cd4e6112f, (v) => $fdb409b291c9958f$export$20fd0f7cd4e6112f = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Button--cta", () => $fdb409b291c9958f$export$b818fdf0052a2cab, (v) => $fdb409b291c9958f$export$b818fdf0052a2cab = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Button--primary", () => $fdb409b291c9958f$export$c1c3511f5e96e926, (v) => $fdb409b291c9958f$export$c1c3511f5e96e926 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Button--secondary", () => $fdb409b291c9958f$export$4f1b75526afd8ff3, (v) => $fdb409b291c9958f$export$4f1b75526afd8ff3 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Button--warning", () => $fdb409b291c9958f$export$6f97cef817ff1cac, (v) => $fdb409b291c9958f$export$6f97cef817ff1cac = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Button--overBackground", () => $fdb409b291c9958f$export$5cda1128f9112d2e, (v) => $fdb409b291c9958f$export$5cda1128f9112d2e = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-Button--quiet", () => $fdb409b291c9958f$export$25f049c049371040, (v) => $fdb409b291c9958f$export$25f049c049371040 = v);
$parcel$export($fdb409b291c9958f$exports, "is-selected", () => $fdb409b291c9958f$export$1e0fb04f31d3c22a, (v) => $fdb409b291c9958f$export$1e0fb04f31d3c22a = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton--staticWhite", () => $fdb409b291c9958f$export$ccc536e1adbdc059, (v) => $fdb409b291c9958f$export$ccc536e1adbdc059 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-ActionButton--staticBlack", () => $fdb409b291c9958f$export$8f7c9db66978d20f, (v) => $fdb409b291c9958f$export$8f7c9db66978d20f = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-LogicButton--and", () => $fdb409b291c9958f$export$e52daab43a62c528, (v) => $fdb409b291c9958f$export$e52daab43a62c528 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-LogicButton--or", () => $fdb409b291c9958f$export$da76be6126aaa633, (v) => $fdb409b291c9958f$export$da76be6126aaa633 = v);
$parcel$export($fdb409b291c9958f$exports, "is-placeholder", () => $fdb409b291c9958f$export$e5b2f5233e4e5194, (v) => $fdb409b291c9958f$export$e5b2f5233e4e5194 = v);
$parcel$export($fdb409b291c9958f$exports, "spectrum-FieldButton--invalid", () => $fdb409b291c9958f$export$c9f503f672e8a3c1, (v) => $fdb409b291c9958f$export$c9f503f672e8a3c1 = v);
var $fdb409b291c9958f$export$1db4cca5b4ade39a;
var $fdb409b291c9958f$export$53da69f51b770d3;
var $fdb409b291c9958f$export$36ae67a131730482;
var $fdb409b291c9958f$export$a024c99ecf6b5741;
var $fdb409b291c9958f$export$b752ce409e5660c;
var $fdb409b291c9958f$export$f39a09f249340e2a;
var $fdb409b291c9958f$export$d374b04f30360026;
var $fdb409b291c9958f$export$5aabb61a966e266f;
var $fdb409b291c9958f$export$29b7b5d6a5a497c5;
var $fdb409b291c9958f$export$b8813cd5d7824ce7;
var $fdb409b291c9958f$export$c081ba562e204942;
var $fdb409b291c9958f$export$ebb4635c66bacdb9;
var $fdb409b291c9958f$export$4a4f01c443eff992;
var $fdb409b291c9958f$export$40e1f009544bacf3;
var $fdb409b291c9958f$export$d35bc1e505d1ebbf;
var $fdb409b291c9958f$export$a9781837241c946d;
var $fdb409b291c9958f$export$5ca7b274e4fff9fc;
var $fdb409b291c9958f$export$116f2fccf941d34d;
var $fdb409b291c9958f$export$49280ffa5236ba8b;
var $fdb409b291c9958f$export$e7dc768d35940237;
var $fdb409b291c9958f$export$20fd0f7cd4e6112f;
var $fdb409b291c9958f$export$b818fdf0052a2cab;
var $fdb409b291c9958f$export$c1c3511f5e96e926;
var $fdb409b291c9958f$export$4f1b75526afd8ff3;
var $fdb409b291c9958f$export$6f97cef817ff1cac;
var $fdb409b291c9958f$export$5cda1128f9112d2e;
var $fdb409b291c9958f$export$25f049c049371040;
var $fdb409b291c9958f$export$1e0fb04f31d3c22a;
var $fdb409b291c9958f$export$ccc536e1adbdc059;
var $fdb409b291c9958f$export$8f7c9db66978d20f;
var $fdb409b291c9958f$export$e52daab43a62c528;
var $fdb409b291c9958f$export$da76be6126aaa633;
var $fdb409b291c9958f$export$e5b2f5233e4e5194;
var $fdb409b291c9958f$export$c9f503f672e8a3c1;
$fdb409b291c9958f$export$1db4cca5b4ade39a = "spectrum-Button_e2d99e";
$fdb409b291c9958f$export$53da69f51b770d3 = "spectrum-ActionButton_e2d99e";
$fdb409b291c9958f$export$36ae67a131730482 = "spectrum-LogicButton_e2d99e";
$fdb409b291c9958f$export$a024c99ecf6b5741 = "spectrum-FieldButton_e2d99e";
$fdb409b291c9958f$export$b752ce409e5660c = "spectrum-ClearButton_e2d99e";
$fdb409b291c9958f$export$f39a09f249340e2a = "focus-ring_e2d99e";
$fdb409b291c9958f$export$d374b04f30360026 = "spectrum-Icon_e2d99e";
$fdb409b291c9958f$export$5aabb61a966e266f = "spectrum-ActionButton--emphasized_e2d99e";
$fdb409b291c9958f$export$29b7b5d6a5a497c5 = "spectrum-ActionButton--staticColor_e2d99e";
$fdb409b291c9958f$export$b8813cd5d7824ce7 = "is-hovered_e2d99e";
$fdb409b291c9958f$export$c081ba562e204942 = "spectrum-Button-label_e2d99e";
$fdb409b291c9958f$export$ebb4635c66bacdb9 = "spectrum-ActionButton-label_e2d99e";
$fdb409b291c9958f$export$4a4f01c443eff992 = "spectrum-ActionButton-hold_e2d99e";
$fdb409b291c9958f$export$40e1f009544bacf3 = "spectrum-ActionButton--quiet_e2d99e";
$fdb409b291c9958f$export$d35bc1e505d1ebbf = "is-disabled_e2d99e";
$fdb409b291c9958f$export$a9781837241c946d = "is-open_e2d99e";
$fdb409b291c9958f$export$5ca7b274e4fff9fc = "spectrum-FieldButton--quiet_e2d99e";
$fdb409b291c9958f$export$116f2fccf941d34d = "spectrum-ClearButton--overBackground_e2d99e";
$fdb409b291c9958f$export$49280ffa5236ba8b = "spectrum-ClearButton--small_e2d99e";
$fdb409b291c9958f$export$e7dc768d35940237 = "is-focused_e2d99e";
$fdb409b291c9958f$export$20fd0f7cd4e6112f = "is-active_e2d99e";
$fdb409b291c9958f$export$b818fdf0052a2cab = "spectrum-Button--cta_e2d99e";
$fdb409b291c9958f$export$c1c3511f5e96e926 = "spectrum-Button--primary_e2d99e";
$fdb409b291c9958f$export$4f1b75526afd8ff3 = "spectrum-Button--secondary_e2d99e";
$fdb409b291c9958f$export$6f97cef817ff1cac = "spectrum-Button--warning_e2d99e";
$fdb409b291c9958f$export$5cda1128f9112d2e = "spectrum-Button--overBackground_e2d99e";
$fdb409b291c9958f$export$25f049c049371040 = "spectrum-Button--quiet_e2d99e";
$fdb409b291c9958f$export$1e0fb04f31d3c22a = "is-selected_e2d99e";
$fdb409b291c9958f$export$ccc536e1adbdc059 = "spectrum-ActionButton--staticWhite_e2d99e";
$fdb409b291c9958f$export$8f7c9db66978d20f = "spectrum-ActionButton--staticBlack_e2d99e";
$fdb409b291c9958f$export$e52daab43a62c528 = "spectrum-LogicButton--and_e2d99e";
$fdb409b291c9958f$export$da76be6126aaa633 = "spectrum-LogicButton--or_e2d99e";
$fdb409b291c9958f$export$e5b2f5233e4e5194 = "is-placeholder_e2d99e";
$fdb409b291c9958f$export$c9f503f672e8a3c1 = "spectrum-FieldButton--invalid_e2d99e";











var $361fb80069a51025$exports = {};

$parcel$export($361fb80069a51025$exports, "flex-container", () => $361fb80069a51025$export$69d7a39fa31a000b, (v) => $361fb80069a51025$export$69d7a39fa31a000b = v);
$parcel$export($361fb80069a51025$exports, "flex-gap", () => $361fb80069a51025$export$31a9da8b58047a44, (v) => $361fb80069a51025$export$31a9da8b58047a44 = v);
$parcel$export($361fb80069a51025$exports, "spectrum-ActionGroup", () => $361fb80069a51025$export$cd2a778fcee9ca5c, (v) => $361fb80069a51025$export$cd2a778fcee9ca5c = v);
$parcel$export($361fb80069a51025$exports, "spectrum-ActionGroup-item", () => $361fb80069a51025$export$94ffe66966d2f53b, (v) => $361fb80069a51025$export$94ffe66966d2f53b = v);
$parcel$export($361fb80069a51025$exports, "spectrum-ActionGroup--vertical", () => $361fb80069a51025$export$b76f0727b879d41c, (v) => $361fb80069a51025$export$b76f0727b879d41c = v);
$parcel$export($361fb80069a51025$exports, "spectrum-ActionGroup--quiet", () => $361fb80069a51025$export$72623314301ff89b, (v) => $361fb80069a51025$export$72623314301ff89b = v);
$parcel$export($361fb80069a51025$exports, "spectrum-ActionGroup--justified", () => $361fb80069a51025$export$565332043c1dc688, (v) => $361fb80069a51025$export$565332043c1dc688 = v);
$parcel$export($361fb80069a51025$exports, "spectrum-ActionGroup--compact", () => $361fb80069a51025$export$a8ccf3eb925252bd, (v) => $361fb80069a51025$export$a8ccf3eb925252bd = v);
$parcel$export($361fb80069a51025$exports, "spectrum-ActionGroup-item--isDisabled", () => $361fb80069a51025$export$e5274b4bc2184bc9, (v) => $361fb80069a51025$export$e5274b4bc2184bc9 = v);
$parcel$export($361fb80069a51025$exports, "is-selected", () => $361fb80069a51025$export$1e0fb04f31d3c22a, (v) => $361fb80069a51025$export$1e0fb04f31d3c22a = v);
$parcel$export($361fb80069a51025$exports, "is-hovered", () => $361fb80069a51025$export$b8813cd5d7824ce7, (v) => $361fb80069a51025$export$b8813cd5d7824ce7 = v);
$parcel$export($361fb80069a51025$exports, "spectrum-ActionGroup--overflowCollapse", () => $361fb80069a51025$export$8e862a39302377c1, (v) => $361fb80069a51025$export$8e862a39302377c1 = v);
$parcel$export($361fb80069a51025$exports, "spectrum-ActionGroup-menu", () => $361fb80069a51025$export$a56eb57dbecb1bdd, (v) => $361fb80069a51025$export$a56eb57dbecb1bdd = v);
$parcel$export($361fb80069a51025$exports, "spectrum-ActionGroup-menu-chevron", () => $361fb80069a51025$export$96da682f48d2c62a, (v) => $361fb80069a51025$export$96da682f48d2c62a = v);
$parcel$export($361fb80069a51025$exports, "spectrum-ActionGroup-menu-contents", () => $361fb80069a51025$export$1733ee5cd2fc5d66, (v) => $361fb80069a51025$export$1733ee5cd2fc5d66 = v);
$parcel$export($361fb80069a51025$exports, "spectrum-ActionGroup-item--iconOnly", () => $361fb80069a51025$export$db27a979adeba484, (v) => $361fb80069a51025$export$db27a979adeba484 = v);
var $361fb80069a51025$export$69d7a39fa31a000b;
var $361fb80069a51025$export$31a9da8b58047a44;
var $361fb80069a51025$export$cd2a778fcee9ca5c;
var $361fb80069a51025$export$94ffe66966d2f53b;
var $361fb80069a51025$export$b76f0727b879d41c;
var $361fb80069a51025$export$72623314301ff89b;
var $361fb80069a51025$export$565332043c1dc688;
var $361fb80069a51025$export$a8ccf3eb925252bd;
var $361fb80069a51025$export$e5274b4bc2184bc9;
var $361fb80069a51025$export$1e0fb04f31d3c22a;
var $361fb80069a51025$export$b8813cd5d7824ce7;
var $361fb80069a51025$export$8e862a39302377c1;
var $361fb80069a51025$export$a56eb57dbecb1bdd;
var $361fb80069a51025$export$96da682f48d2c62a;
var $361fb80069a51025$export$1733ee5cd2fc5d66;
var $361fb80069a51025$export$db27a979adeba484;
$361fb80069a51025$export$69d7a39fa31a000b = "flex-container_7684cb";
$361fb80069a51025$export$31a9da8b58047a44 = "flex-gap_7684cb";
$361fb80069a51025$export$cd2a778fcee9ca5c = "spectrum-ActionGroup_7684cb";
$361fb80069a51025$export$94ffe66966d2f53b = "spectrum-ActionGroup-item_7684cb";
$361fb80069a51025$export$b76f0727b879d41c = "spectrum-ActionGroup--vertical_7684cb";
$361fb80069a51025$export$72623314301ff89b = "spectrum-ActionGroup--quiet_7684cb";
$361fb80069a51025$export$565332043c1dc688 = "spectrum-ActionGroup--justified_7684cb";
$361fb80069a51025$export$a8ccf3eb925252bd = "spectrum-ActionGroup--compact_7684cb";
$361fb80069a51025$export$e5274b4bc2184bc9 = "spectrum-ActionGroup-item--isDisabled_7684cb";
$361fb80069a51025$export$1e0fb04f31d3c22a = "is-selected_7684cb";
$361fb80069a51025$export$b8813cd5d7824ce7 = "is-hovered_7684cb";
$361fb80069a51025$export$8e862a39302377c1 = "spectrum-ActionGroup--overflowCollapse_7684cb";
$361fb80069a51025$export$a56eb57dbecb1bdd = "spectrum-ActionGroup-menu_7684cb";
$361fb80069a51025$export$96da682f48d2c62a = "spectrum-ActionGroup-menu-chevron_7684cb";
$361fb80069a51025$export$1733ee5cd2fc5d66 = "spectrum-ActionGroup-menu-contents_7684cb";
$361fb80069a51025$export$db27a979adeba484 = "spectrum-ActionGroup-item--iconOnly_7684cb";







function $202ea682547fd820$var$ActionGroup(props, ref) {
    props = $4VTOr$reactspectrumprovider.useProviderProps(props);
    props = $4VTOr$reactspectrumutils.useSlotProps(props, 'actionGroup');
    let { isEmphasized: isEmphasized , density: density , isJustified: isJustified , isDisabled: isDisabled , orientation: orientation = 'horizontal' , isQuiet: isQuiet , staticColor: staticColor , overflowMode: overflowMode = 'wrap' , onAction: onAction , buttonLabelBehavior: buttonLabelBehavior , summaryIcon: summaryIcon , ...otherProps } = props;
    let domRef = $4VTOr$reactspectrumutils.useDOMRef(ref);
    let wrapperRef = $4VTOr$react.useRef(null);
    let state = $4VTOr$reactstatelylist.useListState({
        ...props,
        suppressTextValueWarning: true
    });
    let { actionGroupProps: actionGroupProps  } = $4VTOr$reactariaactiongroup.useActionGroup(props, state, domRef);
    let isVertical = orientation === 'vertical';
    let providerProps = {
        isEmphasized: isEmphasized,
        isDisabled: isDisabled,
        isQuiet: isQuiet
    };
    let { styleProps: styleProps  } = $4VTOr$reactspectrumutils.useStyleProps(props);
    // Only hide button text if every item contains more than just plain text (we assume an icon).
    let isIconCollapsible = $4VTOr$react.useMemo(()=>[
            ...state.collection
        ].every((item)=>typeof item.rendered !== 'string'
        )
    , [
        state.collection
    ]);
    let [{ visibleItems: visibleItems1 , hideButtonText: hideButtonText1 , isMeasuring: isMeasuring1  }, setVisibleItems] = $4VTOr$reactariautils.useValueEffect({
        visibleItems: state.collection.size,
        hideButtonText: buttonLabelBehavior === 'hide' && isIconCollapsible,
        isMeasuring: false
    });
    let selectionMode = state.selectionManager.selectionMode;
    let updateOverflow = $4VTOr$react.useCallback(()=>{
        if (overflowMode === 'wrap') return;
        if (orientation === 'vertical' && selectionMode !== 'none') // Collapsing vertical action groups with selection is currently unsupported by Spectrum.
        return;
        let computeVisibleItems = (visibleItems)=>{
            let listItems = Array.from(domRef.current.children);
            let containerSize = orientation === 'horizontal' ? wrapperRef.current.offsetWidth : wrapperRef.current.offsetHeight;
            let isShowingMenu = visibleItems < state.collection.size;
            let calculatedSize = 0;
            let newVisibleItems = 0;
            if (isShowingMenu) calculatedSize += orientation === 'horizontal' ? $202ea682547fd820$var$outerWidth(listItems.pop(), false, true) : $202ea682547fd820$var$outerHeight(listItems.pop(), false, true);
            for (let [i, item] of listItems.entries()){
                calculatedSize += orientation === 'horizontal' ? $202ea682547fd820$var$outerWidth(item, i === 0, i === listItems.length - 1) : $202ea682547fd820$var$outerHeight(item, i === 0, i === listItems.length - 1);
                if (calculatedSize <= containerSize) newVisibleItems++;
                else break;
            }
            // If selection is enabled, and not all of the items fit, collapse all of them into a dropdown
            // immediately rather than having some visible and some not.
            if (selectionMode !== 'none' && newVisibleItems < state.collection.size) return 0;
            return newVisibleItems;
        };
        setVisibleItems(function*() {
            let hideButtonText = buttonLabelBehavior === 'hide' && isIconCollapsible;
            // Update to show all items.
            yield {
                visibleItems: state.collection.size,
                hideButtonText: hideButtonText,
                isMeasuring: true
            };
            // Measure, and update to show the items that fit.
            let newVisibleItems = computeVisibleItems(state.collection.size);
            let isMeasuring = newVisibleItems < state.collection.size && newVisibleItems > 0;
            // If not all of the buttons fit, and buttonLabelBehavior is 'collapse', then first try hiding
            // the button text and only showing icons. Only if that still doesn't fit collapse into a menu.
            if (newVisibleItems < state.collection.size && buttonLabelBehavior === 'collapse' && isIconCollapsible) {
                yield {
                    visibleItems: state.collection.size,
                    hideButtonText: true,
                    isMeasuring: true
                };
                newVisibleItems = computeVisibleItems(state.collection.size);
                isMeasuring = newVisibleItems < state.collection.size && newVisibleItems > 0;
                hideButtonText = true;
            }
            yield {
                visibleItems: newVisibleItems,
                hideButtonText: hideButtonText,
                isMeasuring: isMeasuring
            };
            // If the number of items is less than the number of children,
            // then update again to ensure that the menu fits.
            if (isMeasuring) yield {
                visibleItems: computeVisibleItems(newVisibleItems),
                hideButtonText: hideButtonText,
                isMeasuring: false
            };
        });
    }, [
        domRef,
        state.collection,
        setVisibleItems,
        overflowMode,
        selectionMode,
        buttonLabelBehavior,
        isIconCollapsible,
        orientation
    ]);
    // Watch the parent element for size changes. Watching only the action group itself may not work
    // in all scenarios because it may not shrink when available space is reduced.
    let parentRef = $4VTOr$react.useMemo(()=>({
            get current () {
                return wrapperRef.current.parentElement;
            }
        })
    , [
        wrapperRef
    ]);
    $4VTOr$reactariautils.useResizeObserver({
        ref: overflowMode !== 'wrap' ? parentRef : null,
        onResize: updateOverflow
    });
    $4VTOr$reactariautils.useLayoutEffect(updateOverflow, [
        updateOverflow,
        state.collection
    ]);
    let children = [
        ...state.collection
    ];
    let menuItem = null;
    let menuProps = {
    };
    // If there are no visible items, don't apply any props to the action group container
    // and pass all aria labeling props through to the menu button.
    if (overflowMode === 'collapse' && visibleItems1 === 0) {
        menuProps = $4VTOr$reactariautils.filterDOMProps(props, {
            labelable: true
        });
        actionGroupProps = {
        };
    }
    if (overflowMode === 'collapse' && visibleItems1 < state.collection.size) {
        let menuChildren = children.slice(visibleItems1);
        children = children.slice(0, visibleItems1);
        menuItem = /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($202ea682547fd820$var$ActionGroupMenu, {
            ...menuProps,
            items: menuChildren,
            onAction: onAction,
            isDisabled: isDisabled,
            isEmphasized: isEmphasized,
            staticColor: staticColor,
            state: state,
            summaryIcon: summaryIcon,
            hideButtonText: hideButtonText1,
            isOnlyItem: visibleItems1 === 0,
            orientation: orientation
        });
    }
    let style = {
        ...styleProps.style,
        // While measuring, take up as much space as possible.
        flexBasis: isMeasuring1 ? '100%' : undefined
    };
    return(/*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement("div", {
        ...styleProps,
        style: style,
        className: $4VTOr$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($361fb80069a51025$exports)), 'flex-container', styleProps.className),
        ref: wrapperRef
    }, /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement("div", {
        ...actionGroupProps,
        ref: domRef,
        className: $4VTOr$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($361fb80069a51025$exports)), 'flex-gap', 'spectrum-ActionGroup', {
            'spectrum-ActionGroup--quiet': isQuiet,
            'spectrum-ActionGroup--vertical': isVertical,
            'spectrum-ActionGroup--compact': density === 'compact',
            'spectrum-ActionGroup--justified': isJustified && !isMeasuring1,
            'spectrum-ActionGroup--overflowCollapse': overflowMode === 'collapse'
        }, otherProps.UNSAFE_className)
    }, /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($4VTOr$reactspectrumprovider.Provider, providerProps, children.map((item)=>/*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($202ea682547fd820$var$ActionGroupItem, {
            key: item.key,
            onAction: onAction,
            isDisabled: isDisabled,
            isEmphasized: isEmphasized,
            staticColor: staticColor,
            item: item,
            state: state,
            hideButtonText: hideButtonText1,
            orientation: orientation
        })
    ), menuItem))));
}
/**
 * An ActionGroup is a grouping of ActionButtons that are related to one another.
 */ const $202ea682547fd820$export$c21a5597f732a168 = /*#__PURE__*/ $4VTOr$react.forwardRef($202ea682547fd820$var$ActionGroup);
function $202ea682547fd820$var$ActionGroupItem({ item: item , state: state , isDisabled: isDisabled , isEmphasized: isEmphasized , staticColor: staticColor , onAction: onAction , hideButtonText: hideButtonText , orientation: orientation  }) {
    let ref1 = $4VTOr$react.useRef();
    let { buttonProps: buttonProps  } = $4VTOr$reactariaactiongroup.useActionGroupItem({
        key: item.key
    }, state);
    isDisabled = isDisabled || state.disabledKeys.has(item.key);
    let isSelected = state.selectionManager.isSelected(item.key);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $4VTOr$reactariainteractions.useHover({
        isDisabled: isDisabled
    });
    if (onAction && !isDisabled) buttonProps = $4VTOr$reactariautils.mergeProps(buttonProps, {
        onPress: ()=>onAction(item.key)
    });
    // If button text is hidden, we need to show it as a tooltip instead, so
    // go find the text element in the DOM after rendering.
    let textId = $4VTOr$reactariautils.useId();
    let [textContent, setTextContent] = $4VTOr$react.useState('');
    $4VTOr$reactariautils.useLayoutEffect(()=>{
        var ref;
        if (hideButtonText) setTextContent((ref = document.getElementById(textId)) === null || ref === void 0 ? void 0 : ref.textContent);
    }, [
        hideButtonText,
        item.rendered,
        textId
    ]);
    let button = // Use a PressResponder to send DOM props through.
    // ActionButton doesn't allow overriding the role by default.
    /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($4VTOr$reactariainteractions.PressResponder, $4VTOr$reactariautils.mergeProps(buttonProps, hoverProps), /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($4VTOr$reactspectrumutils.ClearSlots, null, /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($4VTOr$reactspectrumutils.SlotProvider, {
        slots: {
            text: {
                id: hideButtonText ? textId : null,
                isHidden: hideButtonText
            }
        }
    }, /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($4VTOr$reactspectrumbutton.ActionButton, {
        ref: ref1,
        UNSAFE_className: $4VTOr$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($361fb80069a51025$exports)), 'spectrum-ActionGroup-item', {
            'is-selected': isSelected,
            'is-hovered': isHovered,
            'spectrum-ActionGroup-item--iconOnly': hideButtonText,
            'spectrum-ActionGroup-item--isDisabled': isDisabled
        }, $4VTOr$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($fdb409b291c9958f$exports)), {
            'spectrum-ActionButton--emphasized': isEmphasized,
            'is-selected': isSelected
        })),
        isDisabled: isDisabled,
        staticColor: staticColor,
        "aria-label": item['aria-label'],
        "aria-labelledby": item['aria-label'] == null && hideButtonText ? textId : undefined
    }, item.rendered))));
    if (hideButtonText && textContent) button = /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($4VTOr$reactspectrumtooltip.TooltipTrigger, {
        placement: orientation === 'vertical' ? 'end' : 'top'
    }, button, /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($4VTOr$reactspectrumtooltip.Tooltip, null, textContent));
    if (item.wrapper) button = item.wrapper(button);
    return button;
}
function $202ea682547fd820$var$ActionGroupMenu({ state: state , isDisabled: isDisabled , isEmphasized: isEmphasized , staticColor: staticColor , items: items , onAction: onAction , summaryIcon: summaryIcon , hideButtonText: hideButtonText , isOnlyItem: isOnlyItem , orientation: orientation , ...otherProps }) {
    // Use the key of the first item within the menu as the key of the button.
    // The key must actually exist in the collection for focus to work correctly.
    let key = items[0].key;
    let { buttonProps: buttonProps  } = $4VTOr$reactariaactiongroup.useActionGroupItem({
        key: key
    }, state);
    // The menu button shouldn't act like an actual action group item.
    delete buttonProps.onPress;
    delete buttonProps.role;
    delete buttonProps['aria-checked'];
    let { hoverProps: hoverProps , isHovered: isHovered  } = $4VTOr$reactariainteractions.useHover({
        isDisabled: isDisabled
    });
    // If no aria-label or aria-labelledby is given, provide a default one.
    let ariaLabel = otherProps['aria-label'] || (otherProps['aria-labelledby'] ? null : '…');
    let ariaLabelledby = otherProps['aria-labelledby'];
    let textId = $4VTOr$reactariautils.useId();
    let id = $4VTOr$reactariautils.useId();
    // Summary icon only applies when selection is enabled.
    if (state.selectionManager.selectionMode === 'none') summaryIcon = null;
    let iconOnly = false;
    // If there is a selection, show the selected state on the menu button.
    let isSelected = state.selectionManager.selectionMode !== 'none' && !state.selectionManager.isEmpty;
    // If single selection and empty selection is not allowed, swap the contents of the button to the selected item (like a Picker).
    if (!summaryIcon && state.selectionManager.selectionMode === 'single' && state.selectionManager.disallowEmptySelection) {
        let selectedItem = state.collection.getItem(state.selectionManager.firstSelectedKey);
        if (selectedItem) {
            summaryIcon = selectedItem.rendered;
            if (typeof summaryIcon === 'string') summaryIcon = /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($4VTOr$reactspectrumtext.Text, null, summaryIcon);
            iconOnly = hideButtonText;
            ariaLabelledby = `${ariaLabelledby !== null && ariaLabelledby !== void 0 ? ariaLabelledby : id} ${textId}`;
        }
    }
    if (summaryIcon) // If there's a custom summary icon, also add a chevron.
    summaryIcon = /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement(($parcel$interopDefault($4VTOr$react)).Fragment, null, /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement(($parcel$interopDefault($4VTOr$spectrumiconsuiChevronDownMedium)), {
        UNSAFE_className: $4VTOr$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($361fb80069a51025$exports)), 'spectrum-ActionGroup-menu-chevron')
    }), /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement("span", {
        className: $4VTOr$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($361fb80069a51025$exports)), 'spectrum-ActionGroup-menu-contents', {
            'spectrum-ActionGroup-item--iconOnly': iconOnly
        })
    }, summaryIcon));
    return(// Use a PressResponder to send DOM props through.
    /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($4VTOr$reactspectrummenu.MenuTrigger, {
        align: isOnlyItem ? 'start' : 'end',
        direction: orientation === 'vertical' ? 'end' : 'bottom'
    }, /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($4VTOr$reactspectrumutils.SlotProvider, {
        slots: {
            text: {
                id: hideButtonText ? textId : null,
                isHidden: hideButtonText,
                UNSAFE_className: $4VTOr$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($361fb80069a51025$exports)), 'spectrum-ActionGroup-menu-text')
            }
        }
    }, /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($4VTOr$reactariainteractions.PressResponder, $4VTOr$reactariautils.mergeProps(buttonProps, hoverProps), /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($4VTOr$reactspectrumbutton.ActionButton, {
        ...otherProps,
        id: id,
        "aria-label": ariaLabel,
        "aria-labelledby": ariaLabelledby,
        UNSAFE_className: $4VTOr$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($361fb80069a51025$exports)), 'spectrum-ActionGroup-item', 'spectrum-ActionGroup-menu', {
            'is-hovered': isHovered,
            'is-selected': isSelected
        }, $4VTOr$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($fdb409b291c9958f$exports)), {
            'is-selected': isSelected,
            'spectrum-ActionButton--emphasized': isEmphasized
        })),
        isDisabled: isDisabled,
        staticColor: staticColor
    }, summaryIcon || /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement(($parcel$interopDefault($4VTOr$spectrumiconsworkflowMore)), null)))), /*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($4VTOr$reactspectrummenu.Menu, {
        items: items,
        disabledKeys: state.disabledKeys,
        selectionMode: state.selectionManager.selectionMode,
        selectedKeys: state.selectionManager.selectedKeys,
        disallowEmptySelection: state.selectionManager.disallowEmptySelection,
        onSelectionChange: (keys)=>state.selectionManager.setSelectedKeys(keys)
        ,
        onAction: onAction
    }, (node)=>/*#__PURE__*/ ($parcel$interopDefault($4VTOr$react)).createElement($4VTOr$reactspectrummenu.Item, {
            textValue: node.textValue
        }, node.rendered)
    )));
}
function $202ea682547fd820$var$outerWidth(element, ignoreLeftMargin, ignoreRightMargin) {
    let style = window.getComputedStyle(element);
    return element.offsetWidth + (ignoreLeftMargin ? 0 : $202ea682547fd820$var$toNumber(style.marginLeft)) + (ignoreRightMargin ? 0 : $202ea682547fd820$var$toNumber(style.marginRight));
}
function $202ea682547fd820$var$outerHeight(element, ignoreTopMargin, ignoreBottomMargin) {
    let style = window.getComputedStyle(element);
    return element.offsetHeight + (ignoreTopMargin ? 0 : $202ea682547fd820$var$toNumber(style.marginTop)) + (ignoreBottomMargin ? 0 : $202ea682547fd820$var$toNumber(style.marginBottom));
}
function $202ea682547fd820$var$toNumber(value) {
    let parsed = parseInt(value, 10);
    return isNaN(parsed) ? 0 : parsed;
}



$parcel$exportWildcard(module.exports, $202ea682547fd820$exports);
