// @react-spectrum/actiongroup/dist/module.js
import "./main.css";
import {Item as $867cb5dee5ee8fb8$re_export$Item} from "@react-stately/collections";
import {ActionButton as $3z8pF$ActionButton} from "@react-spectrum/button";
import $3z8pF$spectrumiconsuiChevronDownMedium from "@spectrum-icons/ui/ChevronDownMedium";
import {useSlotProps as $3z8pF$useSlotProps, useDOMRef as $3z8pF$useDOMRef, useStyleProps as $3z8pF$useStyleProps, classNames as $3z8pF$classNames, ClearSlots as $3z8pF$ClearSlots, SlotProvider as $3z8pF$SlotProvider} from "@react-spectrum/utils";
import {useValueEffect as $3z8pF$useValueEffect, useResizeObserver as $3z8pF$useResizeObserver, useLayoutEffect as $3z8pF$useLayoutEffect, filterDOMProps as $3z8pF$filterDOMProps, mergeProps as $3z8pF$mergeProps, useId as $3z8pF$useId} from "@react-aria/utils";
import {MenuTrigger as $3z8pF$MenuTrigger, Menu as $3z8pF$Menu, Item as $3z8pF$Item} from "@react-spectrum/menu";
import {useListState as $3z8pF$useListState} from "@react-stately/list";
import $3z8pF$spectrumiconsworkflowMore from "@spectrum-icons/workflow/More";
import {useHover as $3z8pF$useHover, PressResponder as $3z8pF$PressResponder} from "@react-aria/interactions";
import {useProviderProps as $3z8pF$useProviderProps, Provider as $3z8pF$Provider} from "@react-spectrum/provider";
import $3z8pF$react, {useRef as $3z8pF$useRef, useMemo as $3z8pF$useMemo, useCallback as $3z8pF$useCallback, forwardRef as $3z8pF$forwardRef, useState as $3z8pF$useState} from "react";
import {Text as $3z8pF$Text} from "@react-spectrum/text";
import {TooltipTrigger as $3z8pF$TooltipTrigger, Tooltip as $3z8pF$Tooltip} from "@react-spectrum/tooltip";
import {useActionGroup as $3z8pF$useActionGroup, useActionGroupItem as $3z8pF$useActionGroupItem} from "@react-aria/actiongroup";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $092fae2a3c3dd9ff$exports = {};

$parcel$export($092fae2a3c3dd9ff$exports, "ActionGroup", () => $092fae2a3c3dd9ff$export$c21a5597f732a168);

var $8682ec7d9494680f$exports = {};

$parcel$export($8682ec7d9494680f$exports, "spectrum-Button", () => $8682ec7d9494680f$export$1db4cca5b4ade39a, (v) => $8682ec7d9494680f$export$1db4cca5b4ade39a = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton", () => $8682ec7d9494680f$export$53da69f51b770d3, (v) => $8682ec7d9494680f$export$53da69f51b770d3 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-LogicButton", () => $8682ec7d9494680f$export$36ae67a131730482, (v) => $8682ec7d9494680f$export$36ae67a131730482 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-FieldButton", () => $8682ec7d9494680f$export$a024c99ecf6b5741, (v) => $8682ec7d9494680f$export$a024c99ecf6b5741 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ClearButton", () => $8682ec7d9494680f$export$b752ce409e5660c, (v) => $8682ec7d9494680f$export$b752ce409e5660c = v);
$parcel$export($8682ec7d9494680f$exports, "focus-ring", () => $8682ec7d9494680f$export$f39a09f249340e2a, (v) => $8682ec7d9494680f$export$f39a09f249340e2a = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Icon", () => $8682ec7d9494680f$export$d374b04f30360026, (v) => $8682ec7d9494680f$export$d374b04f30360026 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton--emphasized", () => $8682ec7d9494680f$export$5aabb61a966e266f, (v) => $8682ec7d9494680f$export$5aabb61a966e266f = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton--staticColor", () => $8682ec7d9494680f$export$29b7b5d6a5a497c5, (v) => $8682ec7d9494680f$export$29b7b5d6a5a497c5 = v);
$parcel$export($8682ec7d9494680f$exports, "is-hovered", () => $8682ec7d9494680f$export$b8813cd5d7824ce7, (v) => $8682ec7d9494680f$export$b8813cd5d7824ce7 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button-label", () => $8682ec7d9494680f$export$c081ba562e204942, (v) => $8682ec7d9494680f$export$c081ba562e204942 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton-label", () => $8682ec7d9494680f$export$ebb4635c66bacdb9, (v) => $8682ec7d9494680f$export$ebb4635c66bacdb9 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton-hold", () => $8682ec7d9494680f$export$4a4f01c443eff992, (v) => $8682ec7d9494680f$export$4a4f01c443eff992 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton--quiet", () => $8682ec7d9494680f$export$40e1f009544bacf3, (v) => $8682ec7d9494680f$export$40e1f009544bacf3 = v);
$parcel$export($8682ec7d9494680f$exports, "is-disabled", () => $8682ec7d9494680f$export$d35bc1e505d1ebbf, (v) => $8682ec7d9494680f$export$d35bc1e505d1ebbf = v);
$parcel$export($8682ec7d9494680f$exports, "is-open", () => $8682ec7d9494680f$export$a9781837241c946d, (v) => $8682ec7d9494680f$export$a9781837241c946d = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-FieldButton--quiet", () => $8682ec7d9494680f$export$5ca7b274e4fff9fc, (v) => $8682ec7d9494680f$export$5ca7b274e4fff9fc = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ClearButton--overBackground", () => $8682ec7d9494680f$export$116f2fccf941d34d, (v) => $8682ec7d9494680f$export$116f2fccf941d34d = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ClearButton--small", () => $8682ec7d9494680f$export$49280ffa5236ba8b, (v) => $8682ec7d9494680f$export$49280ffa5236ba8b = v);
$parcel$export($8682ec7d9494680f$exports, "is-focused", () => $8682ec7d9494680f$export$e7dc768d35940237, (v) => $8682ec7d9494680f$export$e7dc768d35940237 = v);
$parcel$export($8682ec7d9494680f$exports, "is-active", () => $8682ec7d9494680f$export$20fd0f7cd4e6112f, (v) => $8682ec7d9494680f$export$20fd0f7cd4e6112f = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--cta", () => $8682ec7d9494680f$export$b818fdf0052a2cab, (v) => $8682ec7d9494680f$export$b818fdf0052a2cab = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--primary", () => $8682ec7d9494680f$export$c1c3511f5e96e926, (v) => $8682ec7d9494680f$export$c1c3511f5e96e926 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--secondary", () => $8682ec7d9494680f$export$4f1b75526afd8ff3, (v) => $8682ec7d9494680f$export$4f1b75526afd8ff3 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--warning", () => $8682ec7d9494680f$export$6f97cef817ff1cac, (v) => $8682ec7d9494680f$export$6f97cef817ff1cac = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--overBackground", () => $8682ec7d9494680f$export$5cda1128f9112d2e, (v) => $8682ec7d9494680f$export$5cda1128f9112d2e = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-Button--quiet", () => $8682ec7d9494680f$export$25f049c049371040, (v) => $8682ec7d9494680f$export$25f049c049371040 = v);
$parcel$export($8682ec7d9494680f$exports, "is-selected", () => $8682ec7d9494680f$export$1e0fb04f31d3c22a, (v) => $8682ec7d9494680f$export$1e0fb04f31d3c22a = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton--staticWhite", () => $8682ec7d9494680f$export$ccc536e1adbdc059, (v) => $8682ec7d9494680f$export$ccc536e1adbdc059 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-ActionButton--staticBlack", () => $8682ec7d9494680f$export$8f7c9db66978d20f, (v) => $8682ec7d9494680f$export$8f7c9db66978d20f = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-LogicButton--and", () => $8682ec7d9494680f$export$e52daab43a62c528, (v) => $8682ec7d9494680f$export$e52daab43a62c528 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-LogicButton--or", () => $8682ec7d9494680f$export$da76be6126aaa633, (v) => $8682ec7d9494680f$export$da76be6126aaa633 = v);
$parcel$export($8682ec7d9494680f$exports, "is-placeholder", () => $8682ec7d9494680f$export$e5b2f5233e4e5194, (v) => $8682ec7d9494680f$export$e5b2f5233e4e5194 = v);
$parcel$export($8682ec7d9494680f$exports, "spectrum-FieldButton--invalid", () => $8682ec7d9494680f$export$c9f503f672e8a3c1, (v) => $8682ec7d9494680f$export$c9f503f672e8a3c1 = v);
var $8682ec7d9494680f$export$1db4cca5b4ade39a;
var $8682ec7d9494680f$export$53da69f51b770d3;
var $8682ec7d9494680f$export$36ae67a131730482;
var $8682ec7d9494680f$export$a024c99ecf6b5741;
var $8682ec7d9494680f$export$b752ce409e5660c;
var $8682ec7d9494680f$export$f39a09f249340e2a;
var $8682ec7d9494680f$export$d374b04f30360026;
var $8682ec7d9494680f$export$5aabb61a966e266f;
var $8682ec7d9494680f$export$29b7b5d6a5a497c5;
var $8682ec7d9494680f$export$b8813cd5d7824ce7;
var $8682ec7d9494680f$export$c081ba562e204942;
var $8682ec7d9494680f$export$ebb4635c66bacdb9;
var $8682ec7d9494680f$export$4a4f01c443eff992;
var $8682ec7d9494680f$export$40e1f009544bacf3;
var $8682ec7d9494680f$export$d35bc1e505d1ebbf;
var $8682ec7d9494680f$export$a9781837241c946d;
var $8682ec7d9494680f$export$5ca7b274e4fff9fc;
var $8682ec7d9494680f$export$116f2fccf941d34d;
var $8682ec7d9494680f$export$49280ffa5236ba8b;
var $8682ec7d9494680f$export$e7dc768d35940237;
var $8682ec7d9494680f$export$20fd0f7cd4e6112f;
var $8682ec7d9494680f$export$b818fdf0052a2cab;
var $8682ec7d9494680f$export$c1c3511f5e96e926;
var $8682ec7d9494680f$export$4f1b75526afd8ff3;
var $8682ec7d9494680f$export$6f97cef817ff1cac;
var $8682ec7d9494680f$export$5cda1128f9112d2e;
var $8682ec7d9494680f$export$25f049c049371040;
var $8682ec7d9494680f$export$1e0fb04f31d3c22a;
var $8682ec7d9494680f$export$ccc536e1adbdc059;
var $8682ec7d9494680f$export$8f7c9db66978d20f;
var $8682ec7d9494680f$export$e52daab43a62c528;
var $8682ec7d9494680f$export$da76be6126aaa633;
var $8682ec7d9494680f$export$e5b2f5233e4e5194;
var $8682ec7d9494680f$export$c9f503f672e8a3c1;
$8682ec7d9494680f$export$1db4cca5b4ade39a = "spectrum-Button_e2d99e";
$8682ec7d9494680f$export$53da69f51b770d3 = "spectrum-ActionButton_e2d99e";
$8682ec7d9494680f$export$36ae67a131730482 = "spectrum-LogicButton_e2d99e";
$8682ec7d9494680f$export$a024c99ecf6b5741 = "spectrum-FieldButton_e2d99e";
$8682ec7d9494680f$export$b752ce409e5660c = "spectrum-ClearButton_e2d99e";
$8682ec7d9494680f$export$f39a09f249340e2a = "focus-ring_e2d99e";
$8682ec7d9494680f$export$d374b04f30360026 = "spectrum-Icon_e2d99e";
$8682ec7d9494680f$export$5aabb61a966e266f = "spectrum-ActionButton--emphasized_e2d99e";
$8682ec7d9494680f$export$29b7b5d6a5a497c5 = "spectrum-ActionButton--staticColor_e2d99e";
$8682ec7d9494680f$export$b8813cd5d7824ce7 = "is-hovered_e2d99e";
$8682ec7d9494680f$export$c081ba562e204942 = "spectrum-Button-label_e2d99e";
$8682ec7d9494680f$export$ebb4635c66bacdb9 = "spectrum-ActionButton-label_e2d99e";
$8682ec7d9494680f$export$4a4f01c443eff992 = "spectrum-ActionButton-hold_e2d99e";
$8682ec7d9494680f$export$40e1f009544bacf3 = "spectrum-ActionButton--quiet_e2d99e";
$8682ec7d9494680f$export$d35bc1e505d1ebbf = "is-disabled_e2d99e";
$8682ec7d9494680f$export$a9781837241c946d = "is-open_e2d99e";
$8682ec7d9494680f$export$5ca7b274e4fff9fc = "spectrum-FieldButton--quiet_e2d99e";
$8682ec7d9494680f$export$116f2fccf941d34d = "spectrum-ClearButton--overBackground_e2d99e";
$8682ec7d9494680f$export$49280ffa5236ba8b = "spectrum-ClearButton--small_e2d99e";
$8682ec7d9494680f$export$e7dc768d35940237 = "is-focused_e2d99e";
$8682ec7d9494680f$export$20fd0f7cd4e6112f = "is-active_e2d99e";
$8682ec7d9494680f$export$b818fdf0052a2cab = "spectrum-Button--cta_e2d99e";
$8682ec7d9494680f$export$c1c3511f5e96e926 = "spectrum-Button--primary_e2d99e";
$8682ec7d9494680f$export$4f1b75526afd8ff3 = "spectrum-Button--secondary_e2d99e";
$8682ec7d9494680f$export$6f97cef817ff1cac = "spectrum-Button--warning_e2d99e";
$8682ec7d9494680f$export$5cda1128f9112d2e = "spectrum-Button--overBackground_e2d99e";
$8682ec7d9494680f$export$25f049c049371040 = "spectrum-Button--quiet_e2d99e";
$8682ec7d9494680f$export$1e0fb04f31d3c22a = "is-selected_e2d99e";
$8682ec7d9494680f$export$ccc536e1adbdc059 = "spectrum-ActionButton--staticWhite_e2d99e";
$8682ec7d9494680f$export$8f7c9db66978d20f = "spectrum-ActionButton--staticBlack_e2d99e";
$8682ec7d9494680f$export$e52daab43a62c528 = "spectrum-LogicButton--and_e2d99e";
$8682ec7d9494680f$export$da76be6126aaa633 = "spectrum-LogicButton--or_e2d99e";
$8682ec7d9494680f$export$e5b2f5233e4e5194 = "is-placeholder_e2d99e";
$8682ec7d9494680f$export$c9f503f672e8a3c1 = "spectrum-FieldButton--invalid_e2d99e";











var $d6b91cc4198181d4$exports = {};

$parcel$export($d6b91cc4198181d4$exports, "flex-container", () => $d6b91cc4198181d4$export$69d7a39fa31a000b, (v) => $d6b91cc4198181d4$export$69d7a39fa31a000b = v);
$parcel$export($d6b91cc4198181d4$exports, "flex-gap", () => $d6b91cc4198181d4$export$31a9da8b58047a44, (v) => $d6b91cc4198181d4$export$31a9da8b58047a44 = v);
$parcel$export($d6b91cc4198181d4$exports, "spectrum-ActionGroup", () => $d6b91cc4198181d4$export$cd2a778fcee9ca5c, (v) => $d6b91cc4198181d4$export$cd2a778fcee9ca5c = v);
$parcel$export($d6b91cc4198181d4$exports, "spectrum-ActionGroup-item", () => $d6b91cc4198181d4$export$94ffe66966d2f53b, (v) => $d6b91cc4198181d4$export$94ffe66966d2f53b = v);
$parcel$export($d6b91cc4198181d4$exports, "spectrum-ActionGroup--vertical", () => $d6b91cc4198181d4$export$b76f0727b879d41c, (v) => $d6b91cc4198181d4$export$b76f0727b879d41c = v);
$parcel$export($d6b91cc4198181d4$exports, "spectrum-ActionGroup--quiet", () => $d6b91cc4198181d4$export$72623314301ff89b, (v) => $d6b91cc4198181d4$export$72623314301ff89b = v);
$parcel$export($d6b91cc4198181d4$exports, "spectrum-ActionGroup--justified", () => $d6b91cc4198181d4$export$565332043c1dc688, (v) => $d6b91cc4198181d4$export$565332043c1dc688 = v);
$parcel$export($d6b91cc4198181d4$exports, "spectrum-ActionGroup--compact", () => $d6b91cc4198181d4$export$a8ccf3eb925252bd, (v) => $d6b91cc4198181d4$export$a8ccf3eb925252bd = v);
$parcel$export($d6b91cc4198181d4$exports, "spectrum-ActionGroup-item--isDisabled", () => $d6b91cc4198181d4$export$e5274b4bc2184bc9, (v) => $d6b91cc4198181d4$export$e5274b4bc2184bc9 = v);
$parcel$export($d6b91cc4198181d4$exports, "is-selected", () => $d6b91cc4198181d4$export$1e0fb04f31d3c22a, (v) => $d6b91cc4198181d4$export$1e0fb04f31d3c22a = v);
$parcel$export($d6b91cc4198181d4$exports, "is-hovered", () => $d6b91cc4198181d4$export$b8813cd5d7824ce7, (v) => $d6b91cc4198181d4$export$b8813cd5d7824ce7 = v);
$parcel$export($d6b91cc4198181d4$exports, "spectrum-ActionGroup--overflowCollapse", () => $d6b91cc4198181d4$export$8e862a39302377c1, (v) => $d6b91cc4198181d4$export$8e862a39302377c1 = v);
$parcel$export($d6b91cc4198181d4$exports, "spectrum-ActionGroup-menu", () => $d6b91cc4198181d4$export$a56eb57dbecb1bdd, (v) => $d6b91cc4198181d4$export$a56eb57dbecb1bdd = v);
$parcel$export($d6b91cc4198181d4$exports, "spectrum-ActionGroup-menu-chevron", () => $d6b91cc4198181d4$export$96da682f48d2c62a, (v) => $d6b91cc4198181d4$export$96da682f48d2c62a = v);
$parcel$export($d6b91cc4198181d4$exports, "spectrum-ActionGroup-menu-contents", () => $d6b91cc4198181d4$export$1733ee5cd2fc5d66, (v) => $d6b91cc4198181d4$export$1733ee5cd2fc5d66 = v);
$parcel$export($d6b91cc4198181d4$exports, "spectrum-ActionGroup-item--iconOnly", () => $d6b91cc4198181d4$export$db27a979adeba484, (v) => $d6b91cc4198181d4$export$db27a979adeba484 = v);
var $d6b91cc4198181d4$export$69d7a39fa31a000b;
var $d6b91cc4198181d4$export$31a9da8b58047a44;
var $d6b91cc4198181d4$export$cd2a778fcee9ca5c;
var $d6b91cc4198181d4$export$94ffe66966d2f53b;
var $d6b91cc4198181d4$export$b76f0727b879d41c;
var $d6b91cc4198181d4$export$72623314301ff89b;
var $d6b91cc4198181d4$export$565332043c1dc688;
var $d6b91cc4198181d4$export$a8ccf3eb925252bd;
var $d6b91cc4198181d4$export$e5274b4bc2184bc9;
var $d6b91cc4198181d4$export$1e0fb04f31d3c22a;
var $d6b91cc4198181d4$export$b8813cd5d7824ce7;
var $d6b91cc4198181d4$export$8e862a39302377c1;
var $d6b91cc4198181d4$export$a56eb57dbecb1bdd;
var $d6b91cc4198181d4$export$96da682f48d2c62a;
var $d6b91cc4198181d4$export$1733ee5cd2fc5d66;
var $d6b91cc4198181d4$export$db27a979adeba484;
$d6b91cc4198181d4$export$69d7a39fa31a000b = "flex-container_7684cb";
$d6b91cc4198181d4$export$31a9da8b58047a44 = "flex-gap_7684cb";
$d6b91cc4198181d4$export$cd2a778fcee9ca5c = "spectrum-ActionGroup_7684cb";
$d6b91cc4198181d4$export$94ffe66966d2f53b = "spectrum-ActionGroup-item_7684cb";
$d6b91cc4198181d4$export$b76f0727b879d41c = "spectrum-ActionGroup--vertical_7684cb";
$d6b91cc4198181d4$export$72623314301ff89b = "spectrum-ActionGroup--quiet_7684cb";
$d6b91cc4198181d4$export$565332043c1dc688 = "spectrum-ActionGroup--justified_7684cb";
$d6b91cc4198181d4$export$a8ccf3eb925252bd = "spectrum-ActionGroup--compact_7684cb";
$d6b91cc4198181d4$export$e5274b4bc2184bc9 = "spectrum-ActionGroup-item--isDisabled_7684cb";
$d6b91cc4198181d4$export$1e0fb04f31d3c22a = "is-selected_7684cb";
$d6b91cc4198181d4$export$b8813cd5d7824ce7 = "is-hovered_7684cb";
$d6b91cc4198181d4$export$8e862a39302377c1 = "spectrum-ActionGroup--overflowCollapse_7684cb";
$d6b91cc4198181d4$export$a56eb57dbecb1bdd = "spectrum-ActionGroup-menu_7684cb";
$d6b91cc4198181d4$export$96da682f48d2c62a = "spectrum-ActionGroup-menu-chevron_7684cb";
$d6b91cc4198181d4$export$1733ee5cd2fc5d66 = "spectrum-ActionGroup-menu-contents_7684cb";
$d6b91cc4198181d4$export$db27a979adeba484 = "spectrum-ActionGroup-item--iconOnly_7684cb";







function $092fae2a3c3dd9ff$var$ActionGroup(props, ref) {
    props = $3z8pF$useProviderProps(props);
    props = $3z8pF$useSlotProps(props, 'actionGroup');
    let { isEmphasized: isEmphasized , density: density , isJustified: isJustified , isDisabled: isDisabled , orientation: orientation = 'horizontal' , isQuiet: isQuiet , staticColor: staticColor , overflowMode: overflowMode = 'wrap' , onAction: onAction , buttonLabelBehavior: buttonLabelBehavior , summaryIcon: summaryIcon , ...otherProps } = props;
    let domRef = $3z8pF$useDOMRef(ref);
    let wrapperRef = $3z8pF$useRef(null);
    let state = $3z8pF$useListState({
        ...props,
        suppressTextValueWarning: true
    });
    let { actionGroupProps: actionGroupProps  } = $3z8pF$useActionGroup(props, state, domRef);
    let isVertical = orientation === 'vertical';
    let providerProps = {
        isEmphasized: isEmphasized,
        isDisabled: isDisabled,
        isQuiet: isQuiet
    };
    let { styleProps: styleProps  } = $3z8pF$useStyleProps(props);
    // Only hide button text if every item contains more than just plain text (we assume an icon).
    let isIconCollapsible = $3z8pF$useMemo(()=>[
            ...state.collection
        ].every((item)=>typeof item.rendered !== 'string'
        )
    , [
        state.collection
    ]);
    let [{ visibleItems: visibleItems1 , hideButtonText: hideButtonText1 , isMeasuring: isMeasuring1  }, setVisibleItems] = $3z8pF$useValueEffect({
        visibleItems: state.collection.size,
        hideButtonText: buttonLabelBehavior === 'hide' && isIconCollapsible,
        isMeasuring: false
    });
    let selectionMode = state.selectionManager.selectionMode;
    let updateOverflow = $3z8pF$useCallback(()=>{
        if (overflowMode === 'wrap') return;
        if (orientation === 'vertical' && selectionMode !== 'none') // Collapsing vertical action groups with selection is currently unsupported by Spectrum.
        return;
        let computeVisibleItems = (visibleItems)=>{
            let listItems = Array.from(domRef.current.children);
            let containerSize = orientation === 'horizontal' ? wrapperRef.current.offsetWidth : wrapperRef.current.offsetHeight;
            let isShowingMenu = visibleItems < state.collection.size;
            let calculatedSize = 0;
            let newVisibleItems = 0;
            if (isShowingMenu) calculatedSize += orientation === 'horizontal' ? $092fae2a3c3dd9ff$var$outerWidth(listItems.pop(), false, true) : $092fae2a3c3dd9ff$var$outerHeight(listItems.pop(), false, true);
            for (let [i, item] of listItems.entries()){
                calculatedSize += orientation === 'horizontal' ? $092fae2a3c3dd9ff$var$outerWidth(item, i === 0, i === listItems.length - 1) : $092fae2a3c3dd9ff$var$outerHeight(item, i === 0, i === listItems.length - 1);
                if (calculatedSize <= containerSize) newVisibleItems++;
                else break;
            }
            // If selection is enabled, and not all of the items fit, collapse all of them into a dropdown
            // immediately rather than having some visible and some not.
            if (selectionMode !== 'none' && newVisibleItems < state.collection.size) return 0;
            return newVisibleItems;
        };
        setVisibleItems(function*() {
            let hideButtonText = buttonLabelBehavior === 'hide' && isIconCollapsible;
            // Update to show all items.
            yield {
                visibleItems: state.collection.size,
                hideButtonText: hideButtonText,
                isMeasuring: true
            };
            // Measure, and update to show the items that fit.
            let newVisibleItems = computeVisibleItems(state.collection.size);
            let isMeasuring = newVisibleItems < state.collection.size && newVisibleItems > 0;
            // If not all of the buttons fit, and buttonLabelBehavior is 'collapse', then first try hiding
            // the button text and only showing icons. Only if that still doesn't fit collapse into a menu.
            if (newVisibleItems < state.collection.size && buttonLabelBehavior === 'collapse' && isIconCollapsible) {
                yield {
                    visibleItems: state.collection.size,
                    hideButtonText: true,
                    isMeasuring: true
                };
                newVisibleItems = computeVisibleItems(state.collection.size);
                isMeasuring = newVisibleItems < state.collection.size && newVisibleItems > 0;
                hideButtonText = true;
            }
            yield {
                visibleItems: newVisibleItems,
                hideButtonText: hideButtonText,
                isMeasuring: isMeasuring
            };
            // If the number of items is less than the number of children,
            // then update again to ensure that the menu fits.
            if (isMeasuring) yield {
                visibleItems: computeVisibleItems(newVisibleItems),
                hideButtonText: hideButtonText,
                isMeasuring: false
            };
        });
    }, [
        domRef,
        state.collection,
        setVisibleItems,
        overflowMode,
        selectionMode,
        buttonLabelBehavior,
        isIconCollapsible,
        orientation
    ]);
    // Watch the parent element for size changes. Watching only the action group itself may not work
    // in all scenarios because it may not shrink when available space is reduced.
    let parentRef = $3z8pF$useMemo(()=>({
            get current () {
                return wrapperRef.current.parentElement;
            }
        })
    , [
        wrapperRef
    ]);
    $3z8pF$useResizeObserver({
        ref: overflowMode !== 'wrap' ? parentRef : null,
        onResize: updateOverflow
    });
    $3z8pF$useLayoutEffect(updateOverflow, [
        updateOverflow,
        state.collection
    ]);
    let children = [
        ...state.collection
    ];
    let menuItem = null;
    let menuProps = {
    };
    // If there are no visible items, don't apply any props to the action group container
    // and pass all aria labeling props through to the menu button.
    if (overflowMode === 'collapse' && visibleItems1 === 0) {
        menuProps = $3z8pF$filterDOMProps(props, {
            labelable: true
        });
        actionGroupProps = {
        };
    }
    if (overflowMode === 'collapse' && visibleItems1 < state.collection.size) {
        let menuChildren = children.slice(visibleItems1);
        children = children.slice(0, visibleItems1);
        menuItem = /*#__PURE__*/ $3z8pF$react.createElement($092fae2a3c3dd9ff$var$ActionGroupMenu, {
            ...menuProps,
            items: menuChildren,
            onAction: onAction,
            isDisabled: isDisabled,
            isEmphasized: isEmphasized,
            staticColor: staticColor,
            state: state,
            summaryIcon: summaryIcon,
            hideButtonText: hideButtonText1,
            isOnlyItem: visibleItems1 === 0,
            orientation: orientation
        });
    }
    let style = {
        ...styleProps.style,
        // While measuring, take up as much space as possible.
        flexBasis: isMeasuring1 ? '100%' : undefined
    };
    return(/*#__PURE__*/ $3z8pF$react.createElement("div", {
        ...styleProps,
        style: style,
        className: $3z8pF$classNames((/*@__PURE__*/$parcel$interopDefault($d6b91cc4198181d4$exports)), 'flex-container', styleProps.className),
        ref: wrapperRef
    }, /*#__PURE__*/ $3z8pF$react.createElement("div", {
        ...actionGroupProps,
        ref: domRef,
        className: $3z8pF$classNames((/*@__PURE__*/$parcel$interopDefault($d6b91cc4198181d4$exports)), 'flex-gap', 'spectrum-ActionGroup', {
            'spectrum-ActionGroup--quiet': isQuiet,
            'spectrum-ActionGroup--vertical': isVertical,
            'spectrum-ActionGroup--compact': density === 'compact',
            'spectrum-ActionGroup--justified': isJustified && !isMeasuring1,
            'spectrum-ActionGroup--overflowCollapse': overflowMode === 'collapse'
        }, otherProps.UNSAFE_className)
    }, /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$Provider, providerProps, children.map((item)=>/*#__PURE__*/ $3z8pF$react.createElement($092fae2a3c3dd9ff$var$ActionGroupItem, {
            key: item.key,
            onAction: onAction,
            isDisabled: isDisabled,
            isEmphasized: isEmphasized,
            staticColor: staticColor,
            item: item,
            state: state,
            hideButtonText: hideButtonText1,
            orientation: orientation
        })
    ), menuItem))));
}
/**
 * An ActionGroup is a grouping of ActionButtons that are related to one another.
 */ const $092fae2a3c3dd9ff$export$c21a5597f732a168 = /*#__PURE__*/ $3z8pF$forwardRef($092fae2a3c3dd9ff$var$ActionGroup);
function $092fae2a3c3dd9ff$var$ActionGroupItem({ item: item , state: state , isDisabled: isDisabled , isEmphasized: isEmphasized , staticColor: staticColor , onAction: onAction , hideButtonText: hideButtonText , orientation: orientation  }) {
    let ref1 = $3z8pF$useRef();
    let { buttonProps: buttonProps  } = $3z8pF$useActionGroupItem({
        key: item.key
    }, state);
    isDisabled = isDisabled || state.disabledKeys.has(item.key);
    let isSelected = state.selectionManager.isSelected(item.key);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $3z8pF$useHover({
        isDisabled: isDisabled
    });
    if (onAction && !isDisabled) buttonProps = $3z8pF$mergeProps(buttonProps, {
        onPress: ()=>onAction(item.key)
    });
    // If button text is hidden, we need to show it as a tooltip instead, so
    // go find the text element in the DOM after rendering.
    let textId = $3z8pF$useId();
    let [textContent, setTextContent] = $3z8pF$useState('');
    $3z8pF$useLayoutEffect(()=>{
        var ref;
        if (hideButtonText) setTextContent((ref = document.getElementById(textId)) === null || ref === void 0 ? void 0 : ref.textContent);
    }, [
        hideButtonText,
        item.rendered,
        textId
    ]);
    let button = // Use a PressResponder to send DOM props through.
    // ActionButton doesn't allow overriding the role by default.
    /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$PressResponder, $3z8pF$mergeProps(buttonProps, hoverProps), /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$ClearSlots, null, /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$SlotProvider, {
        slots: {
            text: {
                id: hideButtonText ? textId : null,
                isHidden: hideButtonText
            }
        }
    }, /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$ActionButton, {
        ref: ref1,
        UNSAFE_className: $3z8pF$classNames((/*@__PURE__*/$parcel$interopDefault($d6b91cc4198181d4$exports)), 'spectrum-ActionGroup-item', {
            'is-selected': isSelected,
            'is-hovered': isHovered,
            'spectrum-ActionGroup-item--iconOnly': hideButtonText,
            'spectrum-ActionGroup-item--isDisabled': isDisabled
        }, $3z8pF$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), {
            'spectrum-ActionButton--emphasized': isEmphasized,
            'is-selected': isSelected
        })),
        isDisabled: isDisabled,
        staticColor: staticColor,
        "aria-label": item['aria-label'],
        "aria-labelledby": item['aria-label'] == null && hideButtonText ? textId : undefined
    }, item.rendered))));
    if (hideButtonText && textContent) button = /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$TooltipTrigger, {
        placement: orientation === 'vertical' ? 'end' : 'top'
    }, button, /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$Tooltip, null, textContent));
    if (item.wrapper) button = item.wrapper(button);
    return button;
}
function $092fae2a3c3dd9ff$var$ActionGroupMenu({ state: state , isDisabled: isDisabled , isEmphasized: isEmphasized , staticColor: staticColor , items: items , onAction: onAction , summaryIcon: summaryIcon , hideButtonText: hideButtonText , isOnlyItem: isOnlyItem , orientation: orientation , ...otherProps }) {
    // Use the key of the first item within the menu as the key of the button.
    // The key must actually exist in the collection for focus to work correctly.
    let key = items[0].key;
    let { buttonProps: buttonProps  } = $3z8pF$useActionGroupItem({
        key: key
    }, state);
    // The menu button shouldn't act like an actual action group item.
    delete buttonProps.onPress;
    delete buttonProps.role;
    delete buttonProps['aria-checked'];
    let { hoverProps: hoverProps , isHovered: isHovered  } = $3z8pF$useHover({
        isDisabled: isDisabled
    });
    // If no aria-label or aria-labelledby is given, provide a default one.
    let ariaLabel = otherProps['aria-label'] || (otherProps['aria-labelledby'] ? null : '…');
    let ariaLabelledby = otherProps['aria-labelledby'];
    let textId = $3z8pF$useId();
    let id = $3z8pF$useId();
    // Summary icon only applies when selection is enabled.
    if (state.selectionManager.selectionMode === 'none') summaryIcon = null;
    let iconOnly = false;
    // If there is a selection, show the selected state on the menu button.
    let isSelected = state.selectionManager.selectionMode !== 'none' && !state.selectionManager.isEmpty;
    // If single selection and empty selection is not allowed, swap the contents of the button to the selected item (like a Picker).
    if (!summaryIcon && state.selectionManager.selectionMode === 'single' && state.selectionManager.disallowEmptySelection) {
        let selectedItem = state.collection.getItem(state.selectionManager.firstSelectedKey);
        if (selectedItem) {
            summaryIcon = selectedItem.rendered;
            if (typeof summaryIcon === 'string') summaryIcon = /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$Text, null, summaryIcon);
            iconOnly = hideButtonText;
            ariaLabelledby = `${ariaLabelledby !== null && ariaLabelledby !== void 0 ? ariaLabelledby : id} ${textId}`;
        }
    }
    if (summaryIcon) // If there's a custom summary icon, also add a chevron.
    summaryIcon = /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$react.Fragment, null, /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$spectrumiconsuiChevronDownMedium, {
        UNSAFE_className: $3z8pF$classNames((/*@__PURE__*/$parcel$interopDefault($d6b91cc4198181d4$exports)), 'spectrum-ActionGroup-menu-chevron')
    }), /*#__PURE__*/ $3z8pF$react.createElement("span", {
        className: $3z8pF$classNames((/*@__PURE__*/$parcel$interopDefault($d6b91cc4198181d4$exports)), 'spectrum-ActionGroup-menu-contents', {
            'spectrum-ActionGroup-item--iconOnly': iconOnly
        })
    }, summaryIcon));
    return(// Use a PressResponder to send DOM props through.
    /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$MenuTrigger, {
        align: isOnlyItem ? 'start' : 'end',
        direction: orientation === 'vertical' ? 'end' : 'bottom'
    }, /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$SlotProvider, {
        slots: {
            text: {
                id: hideButtonText ? textId : null,
                isHidden: hideButtonText,
                UNSAFE_className: $3z8pF$classNames((/*@__PURE__*/$parcel$interopDefault($d6b91cc4198181d4$exports)), 'spectrum-ActionGroup-menu-text')
            }
        }
    }, /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$PressResponder, $3z8pF$mergeProps(buttonProps, hoverProps), /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$ActionButton, {
        ...otherProps,
        id: id,
        "aria-label": ariaLabel,
        "aria-labelledby": ariaLabelledby,
        UNSAFE_className: $3z8pF$classNames((/*@__PURE__*/$parcel$interopDefault($d6b91cc4198181d4$exports)), 'spectrum-ActionGroup-item', 'spectrum-ActionGroup-menu', {
            'is-hovered': isHovered,
            'is-selected': isSelected
        }, $3z8pF$classNames((/*@__PURE__*/$parcel$interopDefault($8682ec7d9494680f$exports)), {
            'is-selected': isSelected,
            'spectrum-ActionButton--emphasized': isEmphasized
        })),
        isDisabled: isDisabled,
        staticColor: staticColor
    }, summaryIcon || /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$spectrumiconsworkflowMore, null)))), /*#__PURE__*/ $3z8pF$react.createElement($3z8pF$Menu, {
        items: items,
        disabledKeys: state.disabledKeys,
        selectionMode: state.selectionManager.selectionMode,
        selectedKeys: state.selectionManager.selectedKeys,
        disallowEmptySelection: state.selectionManager.disallowEmptySelection,
        onSelectionChange: (keys)=>state.selectionManager.setSelectedKeys(keys)
        ,
        onAction: onAction
    }, (node)=>/*#__PURE__*/ $3z8pF$react.createElement($3z8pF$Item, {
            textValue: node.textValue
        }, node.rendered)
    )));
}
function $092fae2a3c3dd9ff$var$outerWidth(element, ignoreLeftMargin, ignoreRightMargin) {
    let style = window.getComputedStyle(element);
    return element.offsetWidth + (ignoreLeftMargin ? 0 : $092fae2a3c3dd9ff$var$toNumber(style.marginLeft)) + (ignoreRightMargin ? 0 : $092fae2a3c3dd9ff$var$toNumber(style.marginRight));
}
function $092fae2a3c3dd9ff$var$outerHeight(element, ignoreTopMargin, ignoreBottomMargin) {
    let style = window.getComputedStyle(element);
    return element.offsetHeight + (ignoreTopMargin ? 0 : $092fae2a3c3dd9ff$var$toNumber(style.marginTop)) + (ignoreBottomMargin ? 0 : $092fae2a3c3dd9ff$var$toNumber(style.marginBottom));
}
function $092fae2a3c3dd9ff$var$toNumber(value) {
    let parsed = parseInt(value, 10);
    return isNaN(parsed) ? 0 : parsed;
}





export {$867cb5dee5ee8fb8$re_export$Item as Item, $092fae2a3c3dd9ff$export$c21a5597f732a168 as ActionGroup};
