// @react-stately/tooltip/dist/main.js
var $9WA31$react = require("react");
var $9WA31$reactstatelyoverlays = require("@react-stately/overlays");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $a631e6fca33e6b30$exports = {};

$parcel$export($a631e6fca33e6b30$exports, "useTooltipTriggerState", () => $a631e6fca33e6b30$export$4d40659c25ecb50b);


const $a631e6fca33e6b30$var$TOOLTIP_DELAY = 1500; // this seems to be a 1.5 second delay, check with design
const $a631e6fca33e6b30$var$TOOLTIP_COOLDOWN = 500;
let $a631e6fca33e6b30$var$tooltips = {
};
let $a631e6fca33e6b30$var$tooltipId = 0;
let $a631e6fca33e6b30$var$globalWarmedUp = false;
let $a631e6fca33e6b30$var$globalWarmUpTimeout = null;
let $a631e6fca33e6b30$var$globalCooldownTimeout = null;
function $a631e6fca33e6b30$export$4d40659c25ecb50b(props = {
}) {
    let { delay: delay = $a631e6fca33e6b30$var$TOOLTIP_DELAY  } = props;
    let { isOpen: isOpen , open: open , close: close  } = $9WA31$reactstatelyoverlays.useOverlayTriggerState(props);
    let id = $9WA31$react.useMemo(()=>`${++$a631e6fca33e6b30$var$tooltipId}`
    , []);
    let closeTimeout = $9WA31$react.useRef();
    let ensureTooltipEntry = ()=>{
        $a631e6fca33e6b30$var$tooltips[id] = hideTooltip;
    };
    let closeOpenTooltips = ()=>{
        for(let hideTooltipId in $a631e6fca33e6b30$var$tooltips)if (hideTooltipId !== id) {
            $a631e6fca33e6b30$var$tooltips[hideTooltipId](true);
            delete $a631e6fca33e6b30$var$tooltips[hideTooltipId];
        }
    };
    let showTooltip = ()=>{
        clearTimeout(closeTimeout.current);
        closeTimeout.current = null;
        closeOpenTooltips();
        ensureTooltipEntry();
        $a631e6fca33e6b30$var$globalWarmedUp = true;
        open();
        if ($a631e6fca33e6b30$var$globalWarmUpTimeout) {
            clearTimeout($a631e6fca33e6b30$var$globalWarmUpTimeout);
            $a631e6fca33e6b30$var$globalWarmUpTimeout = null;
        }
        if ($a631e6fca33e6b30$var$globalCooldownTimeout) {
            clearTimeout($a631e6fca33e6b30$var$globalCooldownTimeout);
            $a631e6fca33e6b30$var$globalCooldownTimeout = null;
        }
    };
    let hideTooltip = (immediate)=>{
        if (immediate) {
            clearTimeout(closeTimeout.current);
            closeTimeout.current = null;
            close();
        } else if (!closeTimeout.current) closeTimeout.current = setTimeout(()=>{
            closeTimeout.current = null;
            close();
        }, $a631e6fca33e6b30$var$TOOLTIP_COOLDOWN);
        if ($a631e6fca33e6b30$var$globalWarmUpTimeout) {
            clearTimeout($a631e6fca33e6b30$var$globalWarmUpTimeout);
            $a631e6fca33e6b30$var$globalWarmUpTimeout = null;
        }
        if ($a631e6fca33e6b30$var$globalWarmedUp) {
            if ($a631e6fca33e6b30$var$globalCooldownTimeout) clearTimeout($a631e6fca33e6b30$var$globalCooldownTimeout);
            $a631e6fca33e6b30$var$globalCooldownTimeout = setTimeout(()=>{
                delete $a631e6fca33e6b30$var$tooltips[id];
                $a631e6fca33e6b30$var$globalCooldownTimeout = null;
                $a631e6fca33e6b30$var$globalWarmedUp = false;
            }, $a631e6fca33e6b30$var$TOOLTIP_COOLDOWN);
        }
    };
    let warmupTooltip = ()=>{
        closeOpenTooltips();
        ensureTooltipEntry();
        if (!isOpen && !$a631e6fca33e6b30$var$globalWarmUpTimeout && !$a631e6fca33e6b30$var$globalWarmedUp) $a631e6fca33e6b30$var$globalWarmUpTimeout = setTimeout(()=>{
            $a631e6fca33e6b30$var$globalWarmUpTimeout = null;
            $a631e6fca33e6b30$var$globalWarmedUp = true;
            showTooltip();
        }, delay);
        else if (!isOpen) showTooltip();
    };
    // eslint-disable-next-line arrow-body-style
    $9WA31$react.useEffect(()=>{
        return ()=>{
            clearTimeout(closeTimeout.current);
            let tooltip = $a631e6fca33e6b30$var$tooltips[id];
            if (tooltip) delete $a631e6fca33e6b30$var$tooltips[id];
        };
    }, [
        id
    ]);
    return {
        isOpen: isOpen,
        open: (immediate)=>{
            if (!immediate && delay > 0 && !closeTimeout.current) warmupTooltip();
            else showTooltip();
        },
        close: hideTooltip
    };
}


$parcel$exportWildcard(module.exports, $a631e6fca33e6b30$exports);
