// @react-stately/tooltip/dist/module.js
import {useMemo as $gaC1N$useMemo, useRef as $gaC1N$useRef, useEffect as $gaC1N$useEffect} from "react";
import {useOverlayTriggerState as $gaC1N$useOverlayTriggerState} from "@react-stately/overlays";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $80c0593dd8f9401b$exports = {};

$parcel$export($80c0593dd8f9401b$exports, "useTooltipTriggerState", () => $80c0593dd8f9401b$export$4d40659c25ecb50b);


const $80c0593dd8f9401b$var$TOOLTIP_DELAY = 1500; // this seems to be a 1.5 second delay, check with design
const $80c0593dd8f9401b$var$TOOLTIP_COOLDOWN = 500;
let $80c0593dd8f9401b$var$tooltips = {
};
let $80c0593dd8f9401b$var$tooltipId = 0;
let $80c0593dd8f9401b$var$globalWarmedUp = false;
let $80c0593dd8f9401b$var$globalWarmUpTimeout = null;
let $80c0593dd8f9401b$var$globalCooldownTimeout = null;
function $80c0593dd8f9401b$export$4d40659c25ecb50b(props = {
}) {
    let { delay: delay = $80c0593dd8f9401b$var$TOOLTIP_DELAY  } = props;
    let { isOpen: isOpen , open: open , close: close  } = $gaC1N$useOverlayTriggerState(props);
    let id = $gaC1N$useMemo(()=>`${++$80c0593dd8f9401b$var$tooltipId}`
    , []);
    let closeTimeout = $gaC1N$useRef();
    let ensureTooltipEntry = ()=>{
        $80c0593dd8f9401b$var$tooltips[id] = hideTooltip;
    };
    let closeOpenTooltips = ()=>{
        for(let hideTooltipId in $80c0593dd8f9401b$var$tooltips)if (hideTooltipId !== id) {
            $80c0593dd8f9401b$var$tooltips[hideTooltipId](true);
            delete $80c0593dd8f9401b$var$tooltips[hideTooltipId];
        }
    };
    let showTooltip = ()=>{
        clearTimeout(closeTimeout.current);
        closeTimeout.current = null;
        closeOpenTooltips();
        ensureTooltipEntry();
        $80c0593dd8f9401b$var$globalWarmedUp = true;
        open();
        if ($80c0593dd8f9401b$var$globalWarmUpTimeout) {
            clearTimeout($80c0593dd8f9401b$var$globalWarmUpTimeout);
            $80c0593dd8f9401b$var$globalWarmUpTimeout = null;
        }
        if ($80c0593dd8f9401b$var$globalCooldownTimeout) {
            clearTimeout($80c0593dd8f9401b$var$globalCooldownTimeout);
            $80c0593dd8f9401b$var$globalCooldownTimeout = null;
        }
    };
    let hideTooltip = (immediate)=>{
        if (immediate) {
            clearTimeout(closeTimeout.current);
            closeTimeout.current = null;
            close();
        } else if (!closeTimeout.current) closeTimeout.current = setTimeout(()=>{
            closeTimeout.current = null;
            close();
        }, $80c0593dd8f9401b$var$TOOLTIP_COOLDOWN);
        if ($80c0593dd8f9401b$var$globalWarmUpTimeout) {
            clearTimeout($80c0593dd8f9401b$var$globalWarmUpTimeout);
            $80c0593dd8f9401b$var$globalWarmUpTimeout = null;
        }
        if ($80c0593dd8f9401b$var$globalWarmedUp) {
            if ($80c0593dd8f9401b$var$globalCooldownTimeout) clearTimeout($80c0593dd8f9401b$var$globalCooldownTimeout);
            $80c0593dd8f9401b$var$globalCooldownTimeout = setTimeout(()=>{
                delete $80c0593dd8f9401b$var$tooltips[id];
                $80c0593dd8f9401b$var$globalCooldownTimeout = null;
                $80c0593dd8f9401b$var$globalWarmedUp = false;
            }, $80c0593dd8f9401b$var$TOOLTIP_COOLDOWN);
        }
    };
    let warmupTooltip = ()=>{
        closeOpenTooltips();
        ensureTooltipEntry();
        if (!isOpen && !$80c0593dd8f9401b$var$globalWarmUpTimeout && !$80c0593dd8f9401b$var$globalWarmedUp) $80c0593dd8f9401b$var$globalWarmUpTimeout = setTimeout(()=>{
            $80c0593dd8f9401b$var$globalWarmUpTimeout = null;
            $80c0593dd8f9401b$var$globalWarmedUp = true;
            showTooltip();
        }, delay);
        else if (!isOpen) showTooltip();
    };
    // eslint-disable-next-line arrow-body-style
    $gaC1N$useEffect(()=>{
        return ()=>{
            clearTimeout(closeTimeout.current);
            let tooltip = $80c0593dd8f9401b$var$tooltips[id];
            if (tooltip) delete $80c0593dd8f9401b$var$tooltips[id];
        };
    }, [
        id
    ]);
    return {
        isOpen: isOpen,
        open: (immediate)=>{
            if (!immediate && delay > 0 && !closeTimeout.current) warmupTooltip();
            else showTooltip();
        },
        close: hideTooltip
    };
}




export {$80c0593dd8f9401b$export$4d40659c25ecb50b as useTooltipTriggerState};
