// @react-spectrum/link/dist/main.js
require("./main.css");
var $1vAVm$reactspectrumutils = require("@react-spectrum/utils");
var $1vAVm$reactariafocus = require("@react-aria/focus");
var $1vAVm$reactariautils = require("@react-aria/utils");
var $1vAVm$react = require("react");
var $1vAVm$reactariainteractions = require("@react-aria/interactions");
var $1vAVm$reactarialink = require("@react-aria/link");
var $1vAVm$reactspectrumprovider = require("@react-spectrum/provider");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $c813c9dd9e1d28bf$exports = {};

$parcel$export($c813c9dd9e1d28bf$exports, "Link", () => $c813c9dd9e1d28bf$export$a6c7ac8248d6e38a);




var $451aaf8155691d97$exports = {};

$parcel$export($451aaf8155691d97$exports, "spectrum-Link", () => $451aaf8155691d97$export$d85dd8b94975f388, (v) => $451aaf8155691d97$export$d85dd8b94975f388 = v);
$parcel$export($451aaf8155691d97$exports, "is-hovered", () => $451aaf8155691d97$export$b8813cd5d7824ce7, (v) => $451aaf8155691d97$export$b8813cd5d7824ce7 = v);
$parcel$export($451aaf8155691d97$exports, "focus-ring", () => $451aaf8155691d97$export$f39a09f249340e2a, (v) => $451aaf8155691d97$export$f39a09f249340e2a = v);
$parcel$export($451aaf8155691d97$exports, "is-disabled", () => $451aaf8155691d97$export$d35bc1e505d1ebbf, (v) => $451aaf8155691d97$export$d35bc1e505d1ebbf = v);
$parcel$export($451aaf8155691d97$exports, "spectrum-Link--secondary", () => $451aaf8155691d97$export$ab09ed292dcb9a79, (v) => $451aaf8155691d97$export$ab09ed292dcb9a79 = v);
$parcel$export($451aaf8155691d97$exports, "spectrum-Link--overBackground", () => $451aaf8155691d97$export$679b6267148797da, (v) => $451aaf8155691d97$export$679b6267148797da = v);
$parcel$export($451aaf8155691d97$exports, "spectrum-Link--quiet", () => $451aaf8155691d97$export$81921ea93435e997, (v) => $451aaf8155691d97$export$81921ea93435e997 = v);
var $451aaf8155691d97$export$d85dd8b94975f388;
var $451aaf8155691d97$export$b8813cd5d7824ce7;
var $451aaf8155691d97$export$f39a09f249340e2a;
var $451aaf8155691d97$export$d35bc1e505d1ebbf;
var $451aaf8155691d97$export$ab09ed292dcb9a79;
var $451aaf8155691d97$export$679b6267148797da;
var $451aaf8155691d97$export$81921ea93435e997;
$451aaf8155691d97$export$d85dd8b94975f388 = "spectrum-Link_5a03c7";
$451aaf8155691d97$export$b8813cd5d7824ce7 = "is-hovered_5a03c7";
$451aaf8155691d97$export$f39a09f249340e2a = "focus-ring_5a03c7";
$451aaf8155691d97$export$d35bc1e505d1ebbf = "is-disabled_5a03c7";
$451aaf8155691d97$export$ab09ed292dcb9a79 = "spectrum-Link--secondary_5a03c7";
$451aaf8155691d97$export$679b6267148797da = "spectrum-Link--overBackground_5a03c7";
$451aaf8155691d97$export$81921ea93435e997 = "spectrum-Link--quiet_5a03c7";





function $c813c9dd9e1d28bf$export$a6c7ac8248d6e38a(props) {
    props = $1vAVm$reactspectrumprovider.useProviderProps(props);
    props = $1vAVm$reactspectrumutils.useSlotProps(props, 'link');
    let { variant: variant = 'primary' , isQuiet: isQuiet , children: children , href: // @ts-ignore
    href  } = props;
    let { styleProps: styleProps  } = $1vAVm$reactspectrumutils.useStyleProps(props);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $1vAVm$reactariainteractions.useHover({
    });
    if (href) console.warn('href is deprecated, please use an anchor element as children');
    let ref = $1vAVm$react.useRef();
    let { linkProps: linkProps  } = $1vAVm$reactarialink.useLink({
        ...props,
        elementType: typeof children === 'string' ? 'span' : 'a'
    }, ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($1vAVm$react)).createElement($1vAVm$reactariafocus.FocusRing, {
        focusRingClass: $1vAVm$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($451aaf8155691d97$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($1vAVm$react)).cloneElement($1vAVm$reactspectrumutils.getWrappedElement(children), {
        ...styleProps,
        ...$1vAVm$reactariautils.mergeProps(linkProps, hoverProps),
        ref: ref,
        className: $1vAVm$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($451aaf8155691d97$exports)), 'spectrum-Link', {
            'spectrum-Link--quiet': isQuiet,
            [`spectrum-Link--${variant}`]: variant,
            'is-hovered': isHovered
        }, styleProps.className)
    })));
}


$parcel$exportWildcard(module.exports, $c813c9dd9e1d28bf$exports);
