// @react-spectrum/link/dist/module.js
import "./main.css";
import {useSlotProps as $iIJLa$useSlotProps, useStyleProps as $iIJLa$useStyleProps, classNames as $iIJLa$classNames, getWrappedElement as $iIJLa$getWrappedElement} from "@react-spectrum/utils";
import {FocusRing as $iIJLa$FocusRing} from "@react-aria/focus";
import {mergeProps as $iIJLa$mergeProps} from "@react-aria/utils";
import $iIJLa$react, {useRef as $iIJLa$useRef} from "react";
import {useHover as $iIJLa$useHover} from "@react-aria/interactions";
import {useLink as $iIJLa$useLink} from "@react-aria/link";
import {useProviderProps as $iIJLa$useProviderProps} from "@react-spectrum/provider";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $679ffbd705ba6cf7$exports = {};

$parcel$export($679ffbd705ba6cf7$exports, "Link", () => $679ffbd705ba6cf7$export$a6c7ac8248d6e38a);




var $c9f2f68a832cd862$exports = {};

$parcel$export($c9f2f68a832cd862$exports, "spectrum-Link", () => $c9f2f68a832cd862$export$d85dd8b94975f388, (v) => $c9f2f68a832cd862$export$d85dd8b94975f388 = v);
$parcel$export($c9f2f68a832cd862$exports, "is-hovered", () => $c9f2f68a832cd862$export$b8813cd5d7824ce7, (v) => $c9f2f68a832cd862$export$b8813cd5d7824ce7 = v);
$parcel$export($c9f2f68a832cd862$exports, "focus-ring", () => $c9f2f68a832cd862$export$f39a09f249340e2a, (v) => $c9f2f68a832cd862$export$f39a09f249340e2a = v);
$parcel$export($c9f2f68a832cd862$exports, "is-disabled", () => $c9f2f68a832cd862$export$d35bc1e505d1ebbf, (v) => $c9f2f68a832cd862$export$d35bc1e505d1ebbf = v);
$parcel$export($c9f2f68a832cd862$exports, "spectrum-Link--secondary", () => $c9f2f68a832cd862$export$ab09ed292dcb9a79, (v) => $c9f2f68a832cd862$export$ab09ed292dcb9a79 = v);
$parcel$export($c9f2f68a832cd862$exports, "spectrum-Link--overBackground", () => $c9f2f68a832cd862$export$679b6267148797da, (v) => $c9f2f68a832cd862$export$679b6267148797da = v);
$parcel$export($c9f2f68a832cd862$exports, "spectrum-Link--quiet", () => $c9f2f68a832cd862$export$81921ea93435e997, (v) => $c9f2f68a832cd862$export$81921ea93435e997 = v);
var $c9f2f68a832cd862$export$d85dd8b94975f388;
var $c9f2f68a832cd862$export$b8813cd5d7824ce7;
var $c9f2f68a832cd862$export$f39a09f249340e2a;
var $c9f2f68a832cd862$export$d35bc1e505d1ebbf;
var $c9f2f68a832cd862$export$ab09ed292dcb9a79;
var $c9f2f68a832cd862$export$679b6267148797da;
var $c9f2f68a832cd862$export$81921ea93435e997;
$c9f2f68a832cd862$export$d85dd8b94975f388 = "spectrum-Link_5a03c7";
$c9f2f68a832cd862$export$b8813cd5d7824ce7 = "is-hovered_5a03c7";
$c9f2f68a832cd862$export$f39a09f249340e2a = "focus-ring_5a03c7";
$c9f2f68a832cd862$export$d35bc1e505d1ebbf = "is-disabled_5a03c7";
$c9f2f68a832cd862$export$ab09ed292dcb9a79 = "spectrum-Link--secondary_5a03c7";
$c9f2f68a832cd862$export$679b6267148797da = "spectrum-Link--overBackground_5a03c7";
$c9f2f68a832cd862$export$81921ea93435e997 = "spectrum-Link--quiet_5a03c7";





function $679ffbd705ba6cf7$export$a6c7ac8248d6e38a(props) {
    props = $iIJLa$useProviderProps(props);
    props = $iIJLa$useSlotProps(props, 'link');
    let { variant: variant = 'primary' , isQuiet: isQuiet , children: children , href: // @ts-ignore
    href  } = props;
    let { styleProps: styleProps  } = $iIJLa$useStyleProps(props);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $iIJLa$useHover({
    });
    if (href) console.warn('href is deprecated, please use an anchor element as children');
    let ref = $iIJLa$useRef();
    let { linkProps: linkProps  } = $iIJLa$useLink({
        ...props,
        elementType: typeof children === 'string' ? 'span' : 'a'
    }, ref);
    return(/*#__PURE__*/ $iIJLa$react.createElement($iIJLa$FocusRing, {
        focusRingClass: $iIJLa$classNames((/*@__PURE__*/$parcel$interopDefault($c9f2f68a832cd862$exports)), 'focus-ring')
    }, /*#__PURE__*/ $iIJLa$react.cloneElement($iIJLa$getWrappedElement(children), {
        ...styleProps,
        ...$iIJLa$mergeProps(linkProps, hoverProps),
        ref: ref,
        className: $iIJLa$classNames((/*@__PURE__*/$parcel$interopDefault($c9f2f68a832cd862$exports)), 'spectrum-Link', {
            'spectrum-Link--quiet': isQuiet,
            [`spectrum-Link--${variant}`]: variant,
            'is-hovered': isHovered
        }, styleProps.className)
    })));
}




export {$679ffbd705ba6cf7$export$a6c7ac8248d6e38a as Link};
